<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="XSun" />

<meta name="date" content="2022-07-26" />

<title>ACAT analysis for random SNPs</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-sm-12 col-md-4 col-lg-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-sm-12 col-md-8 col-lg-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">factor_analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/xsun1229/factor_analysis">
    <span class="fas fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">ACAT analysis for random SNPs</h1>
<h4 class="author">XSun</h4>
<h4 class="date">2022-07-26</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-07-29
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>factor_analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you'll want to first commit it to the Git repo. If you're still working on the analysis, you can ignore this warning. When you're finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it's best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200623code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200623)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200623code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200623)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomxsun1229factoranalysistree625b2a6cf2f0f53c2fef76183a3e287ca8a7a041targetblank625b2a6a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/xsun1229/factor_analysis/tree/625b2a6cf2f0f53c2fef76183a3e287ca8a7a041" target="_blank">625b2a6</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomxsun1229factoranalysistree625b2a6cf2f0f53c2fef76183a3e287ca8a7a041targetblank625b2a6a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/xsun1229/factor_analysis/tree/625b2a6cf2f0f53c2fef76183a3e287ca8a7a041" target="_blank">625b2a6</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    analysis/.Rhistory
    Ignored:    output/.Rhistory

Unstaged changes:
    Modified:   analysis/ACAT_random.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/ACAT_random.Rmd</code>) and HTML (<code>docs/ACAT_random.html</code>) files. If you've configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/625b2a6cf2f0f53c2fef76183a3e287ca8a7a041/analysis/ACAT_random.Rmd" target="_blank">625b2a6</a>
</td>
<td>
XSun
</td>
<td>
2022-07-29
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/625b2a6cf2f0f53c2fef76183a3e287ca8a7a041/docs/ACAT_random.html" target="_blank">625b2a6</a>
</td>
<td>
XSun
</td>
<td>
2022-07-29
</td>
<td>
update
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>It was shown that ACAT p-values deviate significantly from null. One concern is that the GWAS SNPs may be somewhat different from other SNPs, so the results may be due to chance. It would be useful to repeat the analysis with random SNPs that match GWAS SNPs for allele frequency and LD.</p>
</div>
<div id="original-acat-results" class="section level1">
<h1>Original ACAT results</h1>
<p><img src="figure/ACAT_random.Rmd/unnamed-chunk-2-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-1">
Past versions of unnamed-chunk-2-1.png
</button>
</p>
<div id="fig-unnamed-chunk-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/625b2a6cf2f0f53c2fef76183a3e287ca8a7a041/docs/figure/ACAT_random.Rmd/unnamed-chunk-2-1.png" target="_blank">625b2a6</a>
</td>
<td>
XSun
</td>
<td>
2022-07-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="random-acat-results" class="section level1">
<h1>Random ACAT results</h1>
<p>We did the analysis follows:</p>
<ol style="list-style-type: decimal">
<li><p>Find random SNPs that match GWAS SNPs for allele frequency and LD using <a href="http://www.mulinlab.org/vsampler/index.html">vSampler</a></p></li>
<li><p>For each SNP, do association test PEER ~ SNP, select the PEER factors do not associate with SNP as covariates. For each expression factor, fo association test factor ~ SNP + SNP-specific covariates.</p></li>
<li><p>Aggregate p-values using ACAT.</p></li>
<li><p>Make QQ plot for each cell type.</p></li>
</ol>
<p><img src="figure/ACAT_random.Rmd/unnamed-chunk-3-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/625b2a6cf2f0f53c2fef76183a3e287ca8a7a041/docs/figure/ACAT_random.Rmd/unnamed-chunk-3-1.png" target="_blank">625b2a6</a>
</td>
<td>
XSun
</td>
<td>
2022-07-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>The QQ plots shift to y-axis and show strong signal, which is not expected for random SNPs. This could be due to: (1) Even random SNPs tend to be associated with the expression factors. (2) There is really no such trend, but ACAT gave significant p-values, in other words, the ACAT p-values are not calibrated.</p>
</div>
<div id="results-checking" class="section level1">
<h1>Results checking</h1>
<p>Take random SNPs matching a GWAS trait, say asthma, or IBD, or some blood cell traits. We have p-values of these SNPs with all factors. We show the QQ plots of these p-values.</p>
<p>Then we chose a pair of trait-factors that have very strong ACAT signal and made QQ plot the p-values from factor ~ SNP association tests.</p>
<div id="for-all-traits" class="section level2">
<h2>For all traits</h2>
<p>Several immune diseases and cell type trait can have strong signals (IBD,rbc,asthma, see below) while others not.</p>
<p><a href="https://photos.app.goo.gl/ouXA6kiRXWu2NCio8">All plots are here</a></p>
</div>
<div id="eur.ibd" class="section level2">
<h2>EUR.IBD</h2>
<p><img src="figure/ACAT_random.Rmd/unnamed-chunk-5-1.png" width="864" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/625b2a6cf2f0f53c2fef76183a3e287ca8a7a041/docs/figure/ACAT_random.Rmd/unnamed-chunk-5-1.png" target="_blank">625b2a6</a>
</td>
<td>
XSun
</td>
<td>
2022-07-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="pair-1-from-ibd-trait" class="section level3">
<h3>pair 1 from IBD trait</h3>
<p>From B cell dataset, pair: pwy203_pc4 - EUR.IBD, ACAT p-value:2.57244e-57</p>
<p><img src="figure/ACAT_random.Rmd/unnamed-chunk-6-1.png" width="384" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/625b2a6cf2f0f53c2fef76183a3e287ca8a7a041/docs/figure/ACAT_random.Rmd/unnamed-chunk-6-1.png" target="_blank">625b2a6</a>
</td>
<td>
XSun
</td>
<td>
2022-07-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>top snps</p>
<pre class="r"><code>index &lt;- which.min(as.numeric(as.character(results_linear$P)))
results_linear[index,]</code></pre>
<pre><code>   CHR       SNP       BP A1 TEST NMISS   BETA   STAT         P
43   6 rs9270832 32570228  T  ADD   254 -1.874 -22.17 5.044e-59</code></pre>
<p>The origin snp:</p>
<pre class="r"><code>final_map[index_origin,]</code></pre>
<pre><code>        rs_or LABEL CHR      POS REF ALT    MAF inLDvariants_LD1     chrpos
430 rs9271176 query   6 32578127   A   G 0.3231             6543 6_32578127
        rs_vs LABEL_vs CHR_vs   POS_vs REF_vs ALT_vs MAF_vs inLDvariants_LD1_vs
430 rs9270832 control1      6 32570228      T      C 0.2744                6544
     chrpos_vs
430 6_32570228</code></pre>
<pre class="r"><code>index_origin &lt;- which(final_map$rs_vs == snp)</code></pre>
<p>QQ plot for the real data.</p>
<p><img src="figure/ACAT_random.Rmd/unnamed-chunk-10-1.png" width="384" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-1">
Past versions of unnamed-chunk-10-1.png
</button>
</p>
<div id="fig-unnamed-chunk-10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/625b2a6cf2f0f53c2fef76183a3e287ca8a7a041/docs/figure/ACAT_random.Rmd/unnamed-chunk-10-1.png" target="_blank">625b2a6</a>
</td>
<td>
XSun
</td>
<td>
2022-07-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>p_acat &lt;- ACAT(as.numeric(as.character(results_linear$P)))
p_acat</code></pre>
<pre><code>[1] 0.9894268</code></pre>
</div>
<div id="pair-2-from-ibd-trait" class="section level3">
<h3>Pair 2 from IBD trait</h3>
<p>From CD14_positive_monocyte, pair: pwy312_pc3 - EUR.IBD, ACAT p-value:1.604200e-59 <img src="figure/ACAT_random.Rmd/unnamed-chunk-12-1.png" width="384" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-1">
Past versions of unnamed-chunk-12-1.png
</button>
</p>
<div id="fig-unnamed-chunk-12-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/625b2a6cf2f0f53c2fef76183a3e287ca8a7a041/docs/figure/ACAT_random.Rmd/unnamed-chunk-12-1.png" target="_blank">625b2a6</a>
</td>
<td>
XSun
</td>
<td>
2022-07-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>top snps</p>
<pre class="r"><code>index &lt;- which.min(as.numeric(as.character(results_linear$P)))
results_linear[index,]</code></pre>
<pre><code>   CHR       SNP       BP A1 TEST NMISS   BETA   STAT         P
43   6 rs9270832 32570228  T  ADD   261 -1.823 -22.69 3.085e-61</code></pre>
<p>The origin snp:</p>
<pre class="r"><code>final_map[index_origin,]</code></pre>
<pre><code>        rs_or LABEL CHR      POS REF ALT    MAF inLDvariants_LD1     chrpos
430 rs9271176 query   6 32578127   A   G 0.3231             6543 6_32578127
        rs_vs LABEL_vs CHR_vs   POS_vs REF_vs ALT_vs MAF_vs inLDvariants_LD1_vs
430 rs9270832 control1      6 32570228      T      C 0.2744                6544
     chrpos_vs
430 6_32570228</code></pre>
<p>QQ plot for the real data.</p>
<p><img src="figure/ACAT_random.Rmd/unnamed-chunk-16-1.png" width="384" style="display: block; margin: auto;" /></p>
<pre class="r"><code>p_acat &lt;- ACAT(as.numeric(as.character(results_linear$P)))
p_acat</code></pre>
<pre><code>[1] 0.3482019</code></pre>
</div>
</div>
<div id="rbc" class="section level2">
<h2>rbc</h2>
<p><img src="figure/ACAT_random.Rmd/unnamed-chunk-18-1.png" width="864" style="display: block; margin: auto;" /></p>
<div id="pair-1-from-rbc" class="section level3">
<h3>Pair 1 from rbc</h3>
<p>From T cell (CD8+) dataset, pair: pwy311_pc5 - rbc, ACAT p-value:2.637203e-11 <img src="figure/ACAT_random.Rmd/unnamed-chunk-19-1.png" width="384" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-19-1">
Past versions of unnamed-chunk-19-1.png
</button>
</p>
<div id="fig-unnamed-chunk-19-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/625b2a6cf2f0f53c2fef76183a3e287ca8a7a041/docs/figure/ACAT_random.Rmd/unnamed-chunk-19-1.png" target="_blank">625b2a6</a>
</td>
<td>
XSun
</td>
<td>
2022-07-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>top snps</p>
<pre class="r"><code>index1 &lt;- which.min(as.numeric(as.character(results_linear$P)))
results_linear[index1,]</code></pre>
<pre><code>    CHR       SNP       BP A1 TEST NMISS   BETA  STAT         P
138   6 rs2647083 32679722  G  ADD   286 0.7466 7.788 1.579e-13</code></pre>
<pre class="r"><code>index2 &lt;- which(as.numeric(as.character(results_linear$P)) == sort(as.numeric(as.character(results_linear$P)))[2])
results_linear[index2,]</code></pre>
<pre><code>    CHR        SNP       BP A1 TEST NMISS    BETA   STAT        P
137   6 rs17423698 32358368  C  ADD   286 -0.7387 -4.954 1.31e-06</code></pre>
<p>The origin snp:</p>
<pre class="r"><code>final_map[index_origin1,]</code></pre>
<pre><code>          rs_or LABEL CHR      POS REF ALT    MAF inLDvariants_LD1     chrpos
1475 rs17884428 query   6 31238759   C   T 0.3807             3078 6_31238759
         rs_vs LABEL_vs CHR_vs   POS_vs REF_vs ALT_vs MAF_vs
1475 rs2647083 control2      6 32679722      C      G 0.3459
     inLDvariants_LD1_vs  chrpos_vs
1475                3083 6_32679722</code></pre>
<pre class="r"><code>final_map[index_origin2,]</code></pre>
<pre><code>         rs_or LABEL CHR      POS REF ALT    MAF inLDvariants_LD1     chrpos
1497 rs1136917 query   6 29912281   G   A 0.1133              866 6_29912281
          rs_vs LABEL_vs CHR_vs   POS_vs REF_vs ALT_vs MAF_vs
1497 rs17423698 control1      6 32358368      T      C  0.159
     inLDvariants_LD1_vs  chrpos_vs
1497                 888 6_32358368</code></pre>
<p>These two origin variants have LD R2 &lt; 0.1 and physical distance &gt; 1Mb</p>
<p>QQ plot for the real data. <img src="figure/ACAT_random.Rmd/unnamed-chunk-23-1.png" width="384" style="display: block; margin: auto;" /></p>
<pre class="r"><code>p_acat &lt;- ACAT(as.numeric(as.character(results_linear$P)))
p_acat</code></pre>
<pre><code>[1] 0.05769225</code></pre>
</div>
</div>
<div id="allergy" class="section level2">
<h2>Allergy</h2>
<p><img src="figure/ACAT_random.Rmd/unnamed-chunk-25-1.png" width="864" style="display: block; margin: auto;" /></p>
<div id="pair-1-from-allergy" class="section level3">
<h3>Pair 1 from allergy</h3>
<p>From B cell dataset, pair: pwy236_pc4 - rbc, ACAT p-value:2.1553e-17 <img src="figure/ACAT_random.Rmd/unnamed-chunk-26-1.png" width="288" style="display: block; margin: auto;" /></p>
<p>top snps</p>
<pre class="r"><code>index1 &lt;- which.min(as.numeric(as.character(results_linear$P)))
results_linear[index1,]</code></pre>
<pre><code>   CHR       SNP       BP A1 TEST NMISS   BETA   STAT         P
60   6 rs9271667 32592710  C  ADD   254 -1.217 -9.826 3.079e-19</code></pre>
<pre class="r"><code>index2 &lt;- which(as.numeric(as.character(results_linear$P)) == sort(as.numeric(as.character(results_linear$P)))[2])
results_linear[index2,]</code></pre>
<pre><code>   CHR        SNP       BP A1 TEST NMISS   BETA  STAT         P
59   6 rs28895129 32420952  G  ADD   254 0.8147 4.007 8.307e-05</code></pre>
<p>The origin snp:</p>
<pre class="r"><code>final_map[index_origin1,]</code></pre>
<pre><code>         rs_or LABEL CHR      POS REF ALT    MAF inLDvariants_LD1     chrpos
705 rs17211510 query   6 32602430   C   A 0.2644             4540 6_32602430
        rs_vs LABEL_vs CHR_vs   POS_vs REF_vs ALT_vs MAF_vs inLDvariants_LD1_vs
705 rs9271667 control1      6 32592710      T      C 0.3121                4534
     chrpos_vs
705 6_32592710</code></pre>
<pre class="r"><code>final_map[index_origin2,]</code></pre>
<pre><code>        rs_or LABEL CHR      POS REF ALT    MAF inLDvariants_LD1     chrpos
694 rs2854001 query   6 31323012   G   A 0.2117             1303 6_31323012
         rs_vs LABEL_vs CHR_vs   POS_vs REF_vs ALT_vs MAF_vs
694 rs28895129 control1      6 32420952      A      G  0.166
    inLDvariants_LD1_vs  chrpos_vs
694                1288 6_32420952</code></pre>
<p>These two origin variants have LD R2 &lt; 0.1 and physical distance &gt; 1Mb</p>
<p>QQ plot for the real data. <img src="figure/ACAT_random.Rmd/unnamed-chunk-30-1.png" width="384" style="display: block; margin: auto;" /></p>
<pre class="r"><code>p_acat &lt;- ACAT(as.numeric(as.character(results_linear$P)))
p_acat</code></pre>
<pre><code>[1] 0.8346754</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.4 (2021-02-15)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
 [1] LC_CTYPE=C                 LC_NUMERIC=C              
 [3] LC_TIME=C                  LC_COLLATE=C              
 [5] LC_MONETARY=C              LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] ACAT_0.91     gridExtra_2.3 ggplot2_3.3.5

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.8        highr_0.8         pillar_1.6.5      compiler_4.0.4   
 [5] bslib_0.2.4       later_1.1.0.1     jquerylib_0.1.3   git2r_0.28.0     
 [9] workflowr_1.6.2   tools_4.0.4       digest_0.6.27     jsonlite_1.7.2   
[13] evaluate_0.14     lifecycle_1.0.0   tibble_3.1.6      gtable_0.3.0     
[17] pkgconfig_2.0.3   rlang_0.4.12      DBI_1.1.1         yaml_2.2.1       
[21] xfun_0.29         withr_2.4.3       dplyr_1.0.7       stringr_1.4.0    
[25] knitr_1.31        generics_0.1.0    fs_1.5.0          vctrs_0.3.8      
[29] sass_0.3.1        tidyselect_1.1.0  rprojroot_2.0.2   grid_4.0.4       
[33] glue_1.6.2        R6_2.5.0          fansi_0.4.2       rmarkdown_2.7    
[37] farver_2.1.0      purrr_0.3.4       magrittr_2.0.1    whisker_0.4      
[41] scales_1.1.1      promises_1.2.0.1  ellipsis_0.3.2    htmltools_0.5.1.1
[45] assertthat_0.2.1  colorspace_2.0-0  httpuv_1.5.5      labeling_0.4.2   
[49] utf8_1.2.2        stringi_1.5.3     munsell_0.5.0     crayon_1.4.1     </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
