<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="XSun" />

<meta name="date" content="2022-07-26" />

<title>ACAT analysis for random SNPs</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.22/datatables.js"></script>
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">factor_analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/xsun1229/factor_analysis">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">ACAT analysis for random SNPs</h1>
<h4 class="author">XSun</h4>
<h4 class="date">2022-07-26</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-09-20
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>factor_analysis/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has staged changes. To know which version of the
R Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200623code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20200623)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200623code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200623)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomxsun1229factoranalysistreed52e8ea6c9d3a64403639a73c70dda17f9d7adf2targetblankd52e8eaa">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/xsun1229/factor_analysis/tree/d52e8ea6c9d3a64403639a73c70dda17f9d7adf2" target="_blank">d52e8ea</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomxsun1229factoranalysistreed52e8ea6c9d3a64403639a73c70dda17f9d7adf2targetblankd52e8eaa"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/xsun1229/factor_analysis/tree/d52e8ea6c9d3a64403639a73c70dda17f9d7adf2" target="_blank">d52e8ea</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    analysis/.Rhistory
    Ignored:    output/.Rhistory

Staged changes:
    Modified:   analysis/ACAT_random.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/ACAT_random.Rmd</code>) and HTML
(<code>docs/ACAT_random.html</code>) files. If you’ve configured a
remote Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/d52e8ea6c9d3a64403639a73c70dda17f9d7adf2/analysis/ACAT_random.Rmd" target="_blank">d52e8ea</a>
</td>
<td>
XSun
</td>
<td>
2022-09-20
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/d52e8ea6c9d3a64403639a73c70dda17f9d7adf2/docs/ACAT_random.html" target="_blank">d52e8ea</a>
</td>
<td>
XSun
</td>
<td>
2022-09-20
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/d8d055ea04a5c554584aeb5724065b394b60e3ee/analysis/ACAT_random.Rmd" target="_blank">d8d055e</a>
</td>
<td>
XSun
</td>
<td>
2022-09-20
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/d8d055ea04a5c554584aeb5724065b394b60e3ee/docs/ACAT_random.html" target="_blank">d8d055e</a>
</td>
<td>
XSun
</td>
<td>
2022-09-20
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/627a691fa68b3c98b5c8b99d9cafa6fc33d0b5a1/analysis/ACAT_random.Rmd" target="_blank">627a691</a>
</td>
<td>
XSun
</td>
<td>
2022-09-02
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/627a691fa68b3c98b5c8b99d9cafa6fc33d0b5a1/docs/ACAT_random.html" target="_blank">627a691</a>
</td>
<td>
XSun
</td>
<td>
2022-09-02
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/f80ea6929a1afd813a43aea04e8529aa29d9f3e7/analysis/ACAT_random.Rmd" target="_blank">f80ea69</a>
</td>
<td>
XSun
</td>
<td>
2022-08-29
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/b1b827a3c736cb2f5d5f1287cf2eef72e169a69e/analysis/ACAT_random.Rmd" target="_blank">b1b827a</a>
</td>
<td>
XSun
</td>
<td>
2022-08-17
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/b1b827a3c736cb2f5d5f1287cf2eef72e169a69e/docs/ACAT_random.html" target="_blank">b1b827a</a>
</td>
<td>
XSun
</td>
<td>
2022-08-17
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/e4382dd341f0e01d3d231418c2463ec88777fc6f/analysis/ACAT_random.Rmd" target="_blank">e4382dd</a>
</td>
<td>
XSun
</td>
<td>
2022-08-08
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/e4382dd341f0e01d3d231418c2463ec88777fc6f/docs/ACAT_random.html" target="_blank">e4382dd</a>
</td>
<td>
XSun
</td>
<td>
2022-08-08
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/cebf58a84c177756602d4318f9d272817c5c979e/analysis/ACAT_random.Rmd" target="_blank">cebf58a</a>
</td>
<td>
XSun
</td>
<td>
2022-07-29
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/cebf58a84c177756602d4318f9d272817c5c979e/docs/ACAT_random.html" target="_blank">cebf58a</a>
</td>
<td>
XSun
</td>
<td>
2022-07-29
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/5a499c2a0e94afce0fc5cadbd8962f165301bc1c/analysis/ACAT_random.Rmd" target="_blank">5a499c2</a>
</td>
<td>
XSun
</td>
<td>
2022-07-29
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/5a499c2a0e94afce0fc5cadbd8962f165301bc1c/docs/ACAT_random.html" target="_blank">5a499c2</a>
</td>
<td>
XSun
</td>
<td>
2022-07-29
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/625b2a6cf2f0f53c2fef76183a3e287ca8a7a041/analysis/ACAT_random.Rmd" target="_blank">625b2a6</a>
</td>
<td>
XSun
</td>
<td>
2022-07-29
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/625b2a6cf2f0f53c2fef76183a3e287ca8a7a041/docs/ACAT_random.html" target="_blank">625b2a6</a>
</td>
<td>
XSun
</td>
<td>
2022-07-29
</td>
<td>
update
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>It was shown that ACAT p-values deviate significantly from null. One
concern is that the GWAS SNPs may be somewhat different from other SNPs,
so the results may be due to chance. It would be useful to repeat the
analysis with random SNPs that match GWAS SNPs for allele frequency and
LD.</p>
</div>
<div id="original-acat-results" class="section level1">
<h1>Original ACAT results</h1>
<p><img src="figure/ACAT_random.Rmd/unnamed-chunk-2-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-1">
Past versions of unnamed-chunk-2-1.png
</button>
</p>
<div id="fig-unnamed-chunk-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/d8d055ea04a5c554584aeb5724065b394b60e3ee/docs/figure/ACAT_random.Rmd/unnamed-chunk-2-1.png" target="_blank">d8d055e</a>
</td>
<td>
XSun
</td>
<td>
2022-09-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/625b2a6cf2f0f53c2fef76183a3e287ca8a7a041/docs/figure/ACAT_random.Rmd/unnamed-chunk-2-1.png" target="_blank">625b2a6</a>
</td>
<td>
XSun
</td>
<td>
2022-07-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="random-acat-results" class="section level1">
<h1>Random ACAT results</h1>
<p>We did the analysis follows:</p>
<ol style="list-style-type: decimal">
<li><p>Find random SNPs that match GWAS SNPs for allele frequency and LD
using <a
href="http://www.mulinlab.org/vsampler/index.html">vSampler</a></p></li>
<li><p>For each SNP, do association test PEER ~ SNP, select the PEER
factors do not associate with SNP as covariates. For each expression
factor, fo association test factor ~ SNP + SNP-specific
covariates.</p></li>
<li><p>Aggregate p-values using ACAT.</p></li>
<li><p>Make QQ plot for each cell type.</p></li>
</ol>
<p><img src="figure/ACAT_random.Rmd/unnamed-chunk-3-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/d8d055ea04a5c554584aeb5724065b394b60e3ee/docs/figure/ACAT_random.Rmd/unnamed-chunk-3-1.png" target="_blank">d8d055e</a>
</td>
<td>
XSun
</td>
<td>
2022-09-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/625b2a6cf2f0f53c2fef76183a3e287ca8a7a041/docs/figure/ACAT_random.Rmd/unnamed-chunk-3-1.png" target="_blank">625b2a6</a>
</td>
<td>
XSun
</td>
<td>
2022-07-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>The QQ plots shift to y-axis and show strong signal, which is not
expected for random SNPs. This could be due to: (1) Even random SNPs
tend to be associated with the expression factors. (2) There is really
no such trend, but ACAT gave significant p-values, in other words, the
ACAT p-values are not calibrated.</p>
</div>
<div id="results-checking" class="section level1">
<h1>Results checking</h1>
<p>Take random SNPs matching a GWAS trait, say asthma, or IBD, or some
blood cell traits. We have p-values of these SNPs with all factors. We
show the QQ plots of these p-values.</p>
<p>Then we chose a pair of trait-factors that have very strong ACAT
signal and made QQ plot the p-values from factor ~ SNP association
tests.</p>
<div id="for-all-traits" class="section level2">
<h2>For all traits</h2>
<p>Several immune diseases and cell type trait can have strong signals
(IBD,rbc,asthma, see below) while others not.</p>
<p><a href="https://photos.app.goo.gl/ouXA6kiRXWu2NCio8">All plots are
here</a></p>
</div>
<div id="eur.ibd" class="section level2">
<h2>EUR.IBD</h2>
<p><img src="figure/ACAT_random.Rmd/unnamed-chunk-5-1.png" width="864" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/d8d055ea04a5c554584aeb5724065b394b60e3ee/docs/figure/ACAT_random.Rmd/unnamed-chunk-5-1.png" target="_blank">d8d055e</a>
</td>
<td>
XSun
</td>
<td>
2022-09-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/625b2a6cf2f0f53c2fef76183a3e287ca8a7a041/docs/figure/ACAT_random.Rmd/unnamed-chunk-5-1.png" target="_blank">625b2a6</a>
</td>
<td>
XSun
</td>
<td>
2022-07-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="pair-1-from-ibd-trait" class="section level3">
<h3>pair 1 from IBD trait</h3>
<p>From B cell dataset, pair: pwy203_pc4 - EUR.IBD, ACAT
p-value:2.57244e-57</p>
<p><img src="figure/ACAT_random.Rmd/unnamed-chunk-6-1.png" width="384" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/d8d055ea04a5c554584aeb5724065b394b60e3ee/docs/figure/ACAT_random.Rmd/unnamed-chunk-6-1.png" target="_blank">d8d055e</a>
</td>
<td>
XSun
</td>
<td>
2022-09-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/625b2a6cf2f0f53c2fef76183a3e287ca8a7a041/docs/figure/ACAT_random.Rmd/unnamed-chunk-6-1.png" target="_blank">625b2a6</a>
</td>
<td>
XSun
</td>
<td>
2022-07-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>top snps</p>
<pre class="r"><code>index &lt;- which.min(as.numeric(as.character(results_linear$P)))
results_linear[index,]</code></pre>
<pre><code>   CHR       SNP       BP A1 TEST NMISS   BETA   STAT         P
43   6 rs9270832 32570228  T  ADD   254 -1.874 -22.17 5.044e-59</code></pre>
<p>The origin snp:</p>
<pre class="r"><code>final_map[index_origin,]</code></pre>
<pre><code>        rs_or LABEL CHR      POS REF ALT    MAF inLDvariants_LD1     chrpos
430 rs9271176 query   6 32578127   A   G 0.3231             6543 6_32578127
        rs_vs LABEL_vs CHR_vs   POS_vs REF_vs ALT_vs MAF_vs inLDvariants_LD1_vs
430 rs9270832 control1      6 32570228      T      C 0.2744                6544
     chrpos_vs
430 6_32570228</code></pre>
<pre class="r"><code>index_origin &lt;- which(final_map$rs_vs == snp)</code></pre>
<p>QQ plot for the real data.</p>
<p><img src="figure/ACAT_random.Rmd/unnamed-chunk-10-1.png" width="384" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-1">
Past versions of unnamed-chunk-10-1.png
</button>
</p>
<div id="fig-unnamed-chunk-10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/d8d055ea04a5c554584aeb5724065b394b60e3ee/docs/figure/ACAT_random.Rmd/unnamed-chunk-10-1.png" target="_blank">d8d055e</a>
</td>
<td>
XSun
</td>
<td>
2022-09-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/5a499c2a0e94afce0fc5cadbd8962f165301bc1c/docs/figure/ACAT_random.Rmd/unnamed-chunk-10-1.png" target="_blank">5a499c2</a>
</td>
<td>
XSun
</td>
<td>
2022-07-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/625b2a6cf2f0f53c2fef76183a3e287ca8a7a041/docs/figure/ACAT_random.Rmd/unnamed-chunk-10-1.png" target="_blank">625b2a6</a>
</td>
<td>
XSun
</td>
<td>
2022-07-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>p_acat &lt;- ACAT(as.numeric(as.character(results_linear$P)))
p_acat</code></pre>
<pre><code>[1] 0.9894268</code></pre>
</div>
<div id="pair-2-from-ibd-trait" class="section level3">
<h3>Pair 2 from IBD trait</h3>
<p>From CD14_positive_monocyte, pair: pwy312_pc3 - EUR.IBD, ACAT
p-value:1.604200e-59
<img src="figure/ACAT_random.Rmd/unnamed-chunk-12-1.png" width="384" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-1">
Past versions of unnamed-chunk-12-1.png
</button>
</p>
<div id="fig-unnamed-chunk-12-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/d8d055ea04a5c554584aeb5724065b394b60e3ee/docs/figure/ACAT_random.Rmd/unnamed-chunk-12-1.png" target="_blank">d8d055e</a>
</td>
<td>
XSun
</td>
<td>
2022-09-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/5a499c2a0e94afce0fc5cadbd8962f165301bc1c/docs/figure/ACAT_random.Rmd/unnamed-chunk-12-1.png" target="_blank">5a499c2</a>
</td>
<td>
XSun
</td>
<td>
2022-07-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/625b2a6cf2f0f53c2fef76183a3e287ca8a7a041/docs/figure/ACAT_random.Rmd/unnamed-chunk-12-1.png" target="_blank">625b2a6</a>
</td>
<td>
XSun
</td>
<td>
2022-07-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>top snps</p>
<pre class="r"><code>index &lt;- which.min(as.numeric(as.character(results_linear$P)))
results_linear[index,]</code></pre>
<pre><code>   CHR       SNP       BP A1 TEST NMISS   BETA   STAT         P
43   6 rs9270832 32570228  T  ADD   261 -1.823 -22.69 3.085e-61</code></pre>
<p>The origin snp:</p>
<pre class="r"><code>final_map[index_origin,]</code></pre>
<pre><code>        rs_or LABEL CHR      POS REF ALT    MAF inLDvariants_LD1     chrpos
430 rs9271176 query   6 32578127   A   G 0.3231             6543 6_32578127
        rs_vs LABEL_vs CHR_vs   POS_vs REF_vs ALT_vs MAF_vs inLDvariants_LD1_vs
430 rs9270832 control1      6 32570228      T      C 0.2744                6544
     chrpos_vs
430 6_32570228</code></pre>
<p>QQ plot for the real data.</p>
<p><img src="figure/ACAT_random.Rmd/unnamed-chunk-16-1.png" width="384" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-16-1">
Past versions of unnamed-chunk-16-1.png
</button>
</p>
<div id="fig-unnamed-chunk-16-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/d8d055ea04a5c554584aeb5724065b394b60e3ee/docs/figure/ACAT_random.Rmd/unnamed-chunk-16-1.png" target="_blank">d8d055e</a>
</td>
<td>
XSun
</td>
<td>
2022-09-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/5a499c2a0e94afce0fc5cadbd8962f165301bc1c/docs/figure/ACAT_random.Rmd/unnamed-chunk-16-1.png" target="_blank">5a499c2</a>
</td>
<td>
XSun
</td>
<td>
2022-07-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>p_acat &lt;- ACAT(as.numeric(as.character(results_linear$P)))
p_acat</code></pre>
<pre><code>[1] 0.3482019</code></pre>
</div>
</div>
<div id="rbc" class="section level2">
<h2>rbc</h2>
<p><img src="figure/ACAT_random.Rmd/unnamed-chunk-18-1.png" width="864" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-1">
Past versions of unnamed-chunk-18-1.png
</button>
</p>
<div id="fig-unnamed-chunk-18-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/d8d055ea04a5c554584aeb5724065b394b60e3ee/docs/figure/ACAT_random.Rmd/unnamed-chunk-18-1.png" target="_blank">d8d055e</a>
</td>
<td>
XSun
</td>
<td>
2022-09-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/5a499c2a0e94afce0fc5cadbd8962f165301bc1c/docs/figure/ACAT_random.Rmd/unnamed-chunk-18-1.png" target="_blank">5a499c2</a>
</td>
<td>
XSun
</td>
<td>
2022-07-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="pair-1-from-rbc" class="section level3">
<h3>Pair 1 from rbc</h3>
<p>From T cell (CD8+) dataset, pair: pwy311_pc5 - rbc, ACAT
p-value:2.637203e-11
<img src="figure/ACAT_random.Rmd/unnamed-chunk-19-1.png" width="384" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-19-1">
Past versions of unnamed-chunk-19-1.png
</button>
</p>
<div id="fig-unnamed-chunk-19-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/d8d055ea04a5c554584aeb5724065b394b60e3ee/docs/figure/ACAT_random.Rmd/unnamed-chunk-19-1.png" target="_blank">d8d055e</a>
</td>
<td>
XSun
</td>
<td>
2022-09-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/5a499c2a0e94afce0fc5cadbd8962f165301bc1c/docs/figure/ACAT_random.Rmd/unnamed-chunk-19-1.png" target="_blank">5a499c2</a>
</td>
<td>
XSun
</td>
<td>
2022-07-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/625b2a6cf2f0f53c2fef76183a3e287ca8a7a041/docs/figure/ACAT_random.Rmd/unnamed-chunk-19-1.png" target="_blank">625b2a6</a>
</td>
<td>
XSun
</td>
<td>
2022-07-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>top snps</p>
<pre class="r"><code>index1 &lt;- which.min(as.numeric(as.character(results_linear$P)))
results_linear[index1,]</code></pre>
<pre><code>    CHR       SNP       BP A1 TEST NMISS   BETA  STAT         P
138   6 rs2647083 32679722  G  ADD   286 0.7466 7.788 1.579e-13</code></pre>
<pre class="r"><code>index2 &lt;- which(as.numeric(as.character(results_linear$P)) == sort(as.numeric(as.character(results_linear$P)))[2])
results_linear[index2,]</code></pre>
<pre><code>    CHR        SNP       BP A1 TEST NMISS    BETA   STAT        P
137   6 rs17423698 32358368  C  ADD   286 -0.7387 -4.954 1.31e-06</code></pre>
<p>The origin snp:</p>
<pre class="r"><code>final_map[index_origin1,]</code></pre>
<pre><code>          rs_or LABEL CHR      POS REF ALT    MAF inLDvariants_LD1     chrpos
1475 rs17884428 query   6 31238759   C   T 0.3807             3078 6_31238759
         rs_vs LABEL_vs CHR_vs   POS_vs REF_vs ALT_vs MAF_vs
1475 rs2647083 control2      6 32679722      C      G 0.3459
     inLDvariants_LD1_vs  chrpos_vs
1475                3083 6_32679722</code></pre>
<pre class="r"><code>final_map[index_origin2,]</code></pre>
<pre><code>         rs_or LABEL CHR      POS REF ALT    MAF inLDvariants_LD1     chrpos
1497 rs1136917 query   6 29912281   G   A 0.1133              866 6_29912281
          rs_vs LABEL_vs CHR_vs   POS_vs REF_vs ALT_vs MAF_vs
1497 rs17423698 control1      6 32358368      T      C  0.159
     inLDvariants_LD1_vs  chrpos_vs
1497                 888 6_32358368</code></pre>
<p>These two origin variants have LD R2 &lt; 0.1 and physical distance
&gt; 1Mb</p>
<p>QQ plot for the real data.
<img src="figure/ACAT_random.Rmd/unnamed-chunk-23-1.png" width="384" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-1">
Past versions of unnamed-chunk-23-1.png
</button>
</p>
<div id="fig-unnamed-chunk-23-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/d8d055ea04a5c554584aeb5724065b394b60e3ee/docs/figure/ACAT_random.Rmd/unnamed-chunk-23-1.png" target="_blank">d8d055e</a>
</td>
<td>
XSun
</td>
<td>
2022-09-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/5a499c2a0e94afce0fc5cadbd8962f165301bc1c/docs/figure/ACAT_random.Rmd/unnamed-chunk-23-1.png" target="_blank">5a499c2</a>
</td>
<td>
XSun
</td>
<td>
2022-07-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>p_acat &lt;- ACAT(as.numeric(as.character(results_linear$P)))
p_acat</code></pre>
<pre><code>[1] 0.05769225</code></pre>
</div>
</div>
<div id="allergy" class="section level2">
<h2>Allergy</h2>
<p><img src="figure/ACAT_random.Rmd/unnamed-chunk-25-1.png" width="864" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-25-1">
Past versions of unnamed-chunk-25-1.png
</button>
</p>
<div id="fig-unnamed-chunk-25-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/d8d055ea04a5c554584aeb5724065b394b60e3ee/docs/figure/ACAT_random.Rmd/unnamed-chunk-25-1.png" target="_blank">d8d055e</a>
</td>
<td>
XSun
</td>
<td>
2022-09-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/5a499c2a0e94afce0fc5cadbd8962f165301bc1c/docs/figure/ACAT_random.Rmd/unnamed-chunk-25-1.png" target="_blank">5a499c2</a>
</td>
<td>
XSun
</td>
<td>
2022-07-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="pair-1-from-allergy" class="section level3">
<h3>Pair 1 from allergy</h3>
<p>From B cell dataset, pair: pwy236_pc4 - rbc, ACAT p-value:2.1553e-17
<img src="figure/ACAT_random.Rmd/unnamed-chunk-26-1.png" width="288" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-26-1">
Past versions of unnamed-chunk-26-1.png
</button>
</p>
<div id="fig-unnamed-chunk-26-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/d8d055ea04a5c554584aeb5724065b394b60e3ee/docs/figure/ACAT_random.Rmd/unnamed-chunk-26-1.png" target="_blank">d8d055e</a>
</td>
<td>
XSun
</td>
<td>
2022-09-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/5a499c2a0e94afce0fc5cadbd8962f165301bc1c/docs/figure/ACAT_random.Rmd/unnamed-chunk-26-1.png" target="_blank">5a499c2</a>
</td>
<td>
XSun
</td>
<td>
2022-07-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>top snps</p>
<pre class="r"><code>index1 &lt;- which.min(as.numeric(as.character(results_linear$P)))
results_linear[index1,]</code></pre>
<pre><code>   CHR       SNP       BP A1 TEST NMISS   BETA   STAT         P
60   6 rs9271667 32592710  C  ADD   254 -1.217 -9.826 3.079e-19</code></pre>
<pre class="r"><code>index2 &lt;- which(as.numeric(as.character(results_linear$P)) == sort(as.numeric(as.character(results_linear$P)))[2])
results_linear[index2,]</code></pre>
<pre><code>   CHR        SNP       BP A1 TEST NMISS   BETA  STAT         P
59   6 rs28895129 32420952  G  ADD   254 0.8147 4.007 8.307e-05</code></pre>
<p>The origin snp:</p>
<pre class="r"><code>final_map[index_origin1,]</code></pre>
<pre><code>         rs_or LABEL CHR      POS REF ALT    MAF inLDvariants_LD1     chrpos
705 rs17211510 query   6 32602430   C   A 0.2644             4540 6_32602430
        rs_vs LABEL_vs CHR_vs   POS_vs REF_vs ALT_vs MAF_vs inLDvariants_LD1_vs
705 rs9271667 control1      6 32592710      T      C 0.3121                4534
     chrpos_vs
705 6_32592710</code></pre>
<pre class="r"><code>final_map[index_origin2,]</code></pre>
<pre><code>        rs_or LABEL CHR      POS REF ALT    MAF inLDvariants_LD1     chrpos
694 rs2854001 query   6 31323012   G   A 0.2117             1303 6_31323012
         rs_vs LABEL_vs CHR_vs   POS_vs REF_vs ALT_vs MAF_vs
694 rs28895129 control1      6 32420952      A      G  0.166
    inLDvariants_LD1_vs  chrpos_vs
694                1288 6_32420952</code></pre>
<p>These two origin variants have LD R2 &lt; 0.1 and physical distance
&gt; 1Mb</p>
<p>QQ plot for the real data.
<img src="figure/ACAT_random.Rmd/unnamed-chunk-30-1.png" width="384" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-30-1">
Past versions of unnamed-chunk-30-1.png
</button>
</p>
<div id="fig-unnamed-chunk-30-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/d8d055ea04a5c554584aeb5724065b394b60e3ee/docs/figure/ACAT_random.Rmd/unnamed-chunk-30-1.png" target="_blank">d8d055e</a>
</td>
<td>
XSun
</td>
<td>
2022-09-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/5a499c2a0e94afce0fc5cadbd8962f165301bc1c/docs/figure/ACAT_random.Rmd/unnamed-chunk-30-1.png" target="_blank">5a499c2</a>
</td>
<td>
XSun
</td>
<td>
2022-07-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>p_acat &lt;- ACAT(as.numeric(as.character(results_linear$P)))
p_acat</code></pre>
<pre><code>[1] 0.8346754</code></pre>
</div>
</div>
</div>
<div id="comparing-acat-for-real-data-and-random-data-mhc-removed"
class="section level1">
<h1>Comparing ACAT for real data and random data – MHC removed</h1>
<p>We found that the variants in MHC area may lead to the deviation in
random ACAT. So we removed the SNPs in MHC area and re-plot the qq
plots.</p>
<p><img src="figure/ACAT_random.Rmd/unnamed-chunk-32-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-32-1">
Past versions of unnamed-chunk-32-1.png
</button>
</p>
<div id="fig-unnamed-chunk-32-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/d8d055ea04a5c554584aeb5724065b394b60e3ee/docs/figure/ACAT_random.Rmd/unnamed-chunk-32-1.png" target="_blank">d8d055e</a>
</td>
<td>
XSun
</td>
<td>
2022-09-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/627a691fa68b3c98b5c8b99d9cafa6fc33d0b5a1/docs/figure/ACAT_random.Rmd/unnamed-chunk-32-1.png" target="_blank">627a691</a>
</td>
<td>
XSun
</td>
<td>
2022-09-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/cebf58a84c177756602d4318f9d272817c5c979e/docs/figure/ACAT_random.Rmd/unnamed-chunk-32-1.png" target="_blank">cebf58a</a>
</td>
<td>
XSun
</td>
<td>
2022-07-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>We pooled all cell types to check if the inflations from random data
(Leukocyte (CD15+), Platelet, T cell (CD8+)) dominant the whole data
set.</p>
<p><img src="figure/ACAT_random.Rmd/unnamed-chunk-33-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-33-1">
Past versions of unnamed-chunk-33-1.png
</button>
</p>
<div id="fig-unnamed-chunk-33-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/d8d055ea04a5c554584aeb5724065b394b60e3ee/docs/figure/ACAT_random.Rmd/unnamed-chunk-33-1.png" target="_blank">d8d055e</a>
</td>
<td>
XSun
</td>
<td>
2022-09-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/627a691fa68b3c98b5c8b99d9cafa6fc33d0b5a1/docs/figure/ACAT_random.Rmd/unnamed-chunk-33-1.png" target="_blank">627a691</a>
</td>
<td>
XSun
</td>
<td>
2022-09-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/cebf58a84c177756602d4318f9d272817c5c979e/docs/figure/ACAT_random.Rmd/unnamed-chunk-33-1.png" target="_blank">cebf58a</a>
</td>
<td>
XSun
</td>
<td>
2022-07-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>We noticed that there is some inflations for random data (Leukocyte
(CD15+), Platelet, T cell (CD8+)). We compute genomic inflation factors
for all cell types.</p>
<p><span class="math display">\[ chisq = qchisq(1-pvalue,1)\]</span>
<span class="math display">\[ lambda =
\frac{median(chisq)}{qchisq(0.5,1)} \]</span> The lambda are 1.103569
1.110509 1.133734 1.179766 1.045502 4.582142 accordingly.</p>
<p>Then we corrected the p-values with the inflation factor – lambda
(devide the chi-squares by inflation factors accoring to <a
href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3133943/"
class="uri">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3133943/</a>
and compute the p-values):</p>
<p><span class="math display">\[ chisq_{adjusted} =
\frac{chisq}{lambda}\]</span></p>
<p><span class="math display">\[ pvalue_{adjusted} = 1 -
pchisq(chisq_{adjusted}) \]</span></p>
<p>The adjusted p-values are shown in the qqplots below (comparing
unadjusted real p-values).</p>
<p><img src="figure/ACAT_random.Rmd/unnamed-chunk-34-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-34-1">
Past versions of unnamed-chunk-34-1.png
</button>
</p>
<div id="fig-unnamed-chunk-34-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/d8d055ea04a5c554584aeb5724065b394b60e3ee/docs/figure/ACAT_random.Rmd/unnamed-chunk-34-1.png" target="_blank">d8d055e</a>
</td>
<td>
XSun
</td>
<td>
2022-09-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/627a691fa68b3c98b5c8b99d9cafa6fc33d0b5a1/docs/figure/ACAT_random.Rmd/unnamed-chunk-34-1.png" target="_blank">627a691</a>
</td>
<td>
XSun
</td>
<td>
2022-09-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/e4382dd341f0e01d3d231418c2463ec88777fc6f/docs/figure/ACAT_random.Rmd/unnamed-chunk-34-1.png" target="_blank">e4382dd</a>
</td>
<td>
XSun
</td>
<td>
2022-08-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="top-pairs-from-t-cell-cd8" class="section level1">
<h1>Top pairs from T cell CD8+</h1>
<div id="random-acat" class="section level2">
<h2>Random ACAT</h2>
<p>We check several top pairs (top20) that have lowest p-values.</p>
<pre><code>   pdw_pwy288_pc4    ret_pwy158_pc1    pdw_pwy291_pc1    pdw_pwy102_pc3 
     1.271423e-06      3.710175e-06      6.523194e-06      7.455455e-06 
   pdw_pwy289_pc5    pdw_pwy228_pc5    pdw_pwy129_pc4    pdw_pwy165_pc3 
     8.618332e-06      8.774918e-06      1.122459e-05      1.191599e-05 
   pdw_pwy202_pc3    pdw_pwy303_pc1     pdw_pwy86_pc4 EUR.CD_pwy259_pc4 
     1.341401e-05      1.463348e-05      1.549829e-05      1.652491e-05 
 lymph_pwy284_pc3  lymph_pwy300_pc1    pdw_pwy292_pc1 EUR.CD_pwy131_pc1 
     1.868668e-05      2.309532e-05      2.353322e-05      2.389009e-05 
   pdw_pwy137_pc5 EUR.CD_pwy299_pc1    plt_pwy155_pc3    hgb_pwy216_pc2 
     2.541057e-05      3.095121e-05      3.980136e-05      4.210378e-05 </code></pre>
<pre><code>traits
EUR.CD    hgb  lymph    pdw    plt    ret 
     3      1      2     12      1      1 </code></pre>
<p>12 of 20 are related to trait pdw (platelet distribution width).</p>
</div>
<div id="real-acat" class="section level2">
<h2>Real ACAT</h2>
<p>For comparison, we also checked the top 20 pairs from real data</p>
<pre><code> lymph_pwy212_pc3   mono_pwy284_pc2    wbc_pwy284_pc2    hct_pwy141_pc3 
     3.407572e-06      5.849411e-06      5.991581e-06      1.092830e-05 
   hgb_pwy168_pc4   mono_pwy293_pc3    wbc_pwy293_pc3    rbc_pwy141_pc3 
     1.369486e-05      1.475678e-05      1.512792e-05      1.699808e-05 
   hct_pwy256_pc2    rbc_pwy256_pc2    hct_pwy154_pc1 EUR.CD_pwy217_pc5 
     1.727633e-05      2.688218e-05      3.243143e-05      4.113453e-05 
    hct_pwy16_pc2    pct_pwy322_pc2    hct_pwy167_pc1    hct_pwy139_pc1 
     4.189620e-05      4.462366e-05      4.656165e-05      4.818949e-05 
   rbc_pwy154_pc1    hct_pwy210_pc1   mono_pwy125_pc4    hct_pwy273_pc1 
     5.043930e-05      5.106646e-05      5.317304e-05      5.520883e-05 </code></pre>
<pre><code>traits_real
EUR.CD    hct    hgb  lymph   mono    pct    rbc    wbc 
     1      8      1      1      3      1      3      2 </code></pre>
<p>The top pairs are not related to pdw.</p>
<p>We removed the pairs related to pdw for both real data and random
data, then made the qq-plot:</p>
<p><img src="figure/ACAT_random.Rmd/unnamed-chunk-37-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-37-1">
Past versions of unnamed-chunk-37-1.png
</button>
</p>
<div id="fig-unnamed-chunk-37-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/d8d055ea04a5c554584aeb5724065b394b60e3ee/docs/figure/ACAT_random.Rmd/unnamed-chunk-37-1.png" target="_blank">d8d055e</a>
</td>
<td>
XSun
</td>
<td>
2022-09-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;The top1 pair of random data  (top 1 dot in red) is ret_pwy158_pc1 &quot;</code></pre>
<pre><code>[1] &quot;p-value = 3.71017525187992e-06 &quot;</code></pre>
<pre><code>[1] &quot;The top1 pair of real data (top 1 dot in blue) is lymph_pwy212_pc3 &quot;</code></pre>
<pre><code>[1] &quot;p-value = 3.40757177707754e-06 &quot;</code></pre>
</div>
</div>
<div id="top-pairs-from-leukocyte-cd15" class="section level1">
<h1>Top pairs from Leukocyte (CD15+)</h1>
<div id="random-acat-1" class="section level2">
<h2>Random ACAT</h2>
<p>We also check the top 20 pairs from Leukocyte (CD15+).</p>
<pre><code> wbc_pwy151_pc1 gran_pwy271_pc1 neut_pwy271_pc1  hct_pwy163_pc3 gran_pwy185_pc5 
   5.269953e-06    8.021516e-06    8.191818e-06    1.999401e-05    2.186851e-05 
neut_pwy185_pc5  hct_pwy290_pc4  gran_pwy93_pc3  neut_pwy93_pc3  hct_pwy269_pc1 
   2.233374e-05    2.380974e-05    2.406965e-05    2.457972e-05    3.407876e-05 
 gran_pwy56_pc3  neut_pwy56_pc3  hct_pwy116_pc4  hct_pwy314_pc4  hct_pwy151_pc5 
   4.757533e-05    4.859534e-05    5.407810e-05    7.126153e-05    7.663900e-05 
gran_pwy256_pc4 neut_pwy256_pc4 gran_pwy230_pc5 neut_pwy230_pc5  hct_pwy224_pc1 
   8.034297e-05    8.206683e-05    9.064552e-05    9.266286e-05    1.046540e-04 </code></pre>
<pre><code>traits
gran  hct neut  wbc 
   6    7    6    1 </code></pre>
<pre><code>pwy
pwy116 pwy151 pwy163 pwy185 pwy224 pwy230 pwy256 pwy269 pwy271 pwy290 pwy314 
     1      2      1      2      1      2      2      1      2      1      1 
 pwy56  pwy93 
     2      2 </code></pre>
<p>There is not a dominant pathway or trait.</p>
</div>
<div id="real-acat-1" class="section level2">
<h2>Real ACAT</h2>
<p>For comparison, we also checked the top 20 pairs from real data</p>
<pre><code> lymph_pwy212_pc3   mono_pwy284_pc2    wbc_pwy284_pc2    hct_pwy141_pc3 
     3.407572e-06      5.849411e-06      5.991581e-06      1.092830e-05 
   hgb_pwy168_pc4   mono_pwy293_pc3    wbc_pwy293_pc3    rbc_pwy141_pc3 
     1.369486e-05      1.475678e-05      1.512792e-05      1.699808e-05 
   hct_pwy256_pc2    rbc_pwy256_pc2    hct_pwy154_pc1 EUR.CD_pwy217_pc5 
     1.727633e-05      2.688218e-05      3.243143e-05      4.113453e-05 
    hct_pwy16_pc2    pct_pwy322_pc2    hct_pwy167_pc1    hct_pwy139_pc1 
     4.189620e-05      4.462366e-05      4.656165e-05      4.818949e-05 
   rbc_pwy154_pc1    hct_pwy210_pc1   mono_pwy125_pc4    hct_pwy273_pc1 
     5.043930e-05      5.106646e-05      5.317304e-05      5.520883e-05 </code></pre>
<pre><code>traits_real
EUR.CD    hct    hgb  lymph   mono    pct    rbc    wbc 
     1      8      1      1      3      1      3      2 </code></pre>
<p>There are three dominant traits in random data (gran, hct and neut),
hct is also the dominant trait in real data. So we removed gran and neut
in both random data and real data.</p>
<p><img src="figure/ACAT_random.Rmd/unnamed-chunk-40-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-40-1">
Past versions of unnamed-chunk-40-1.png
</button>
</p>
<div id="fig-unnamed-chunk-40-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/d8d055ea04a5c554584aeb5724065b394b60e3ee/docs/figure/ACAT_random.Rmd/unnamed-chunk-40-1.png" target="_blank">d8d055e</a>
</td>
<td>
XSun
</td>
<td>
2022-09-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;The top3 pairs of random data  (top 3 dot in red) is wbc_pwy151_pc1,hct_pwy163_pc3,hct_pwy290_pc4 &quot;</code></pre>
<pre><code>[1] &quot;p-value = 5.26995343553681e-06,1.99940063134465e-05,2.38097448835839e-05 &quot;</code></pre>
<pre><code>[1] &quot;The top3 pairs of real data (top 1 dot in blue) is hgb_pwy280_pc3,lymph_pwy18_pc4,rdw_pwy210_pc3 &quot;</code></pre>
<pre><code>[1] &quot;p-value = 1.924151714091e-05,2.1727275875024e-05,2.56802806524958e-05 &quot;</code></pre>
<p>If we remove the hct as well, we have:</p>
<p><img src="figure/ACAT_random.Rmd/unnamed-chunk-41-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-41-1">
Past versions of unnamed-chunk-41-1.png
</button>
</p>
<div id="fig-unnamed-chunk-41-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/d8d055ea04a5c554584aeb5724065b394b60e3ee/docs/figure/ACAT_random.Rmd/unnamed-chunk-41-1.png" target="_blank">d8d055e</a>
</td>
<td>
XSun
</td>
<td>
2022-09-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;The top1 pair of random data  (top 1 dot in red) is wbc_pwy151_pc1 &quot;</code></pre>
<pre><code>[1] &quot;p-value = 5.26995343553681e-06 &quot;</code></pre>
<pre><code>[1] &quot;The top1 pair of real data (top 1 dot in blue) is hgb_pwy280_pc3 &quot;</code></pre>
<pre><code>[1] &quot;p-value = 1.924151714091e-05 &quot;</code></pre>
</div>
</div>
<div id="checking-the-top-20-pairs-for-all-cell-types"
class="section level1">
<h1>Checking the top 20 pairs for all cell types</h1>
<div id="random-acat-2" class="section level2">
<h2>Random ACAT:</h2>
<pre><code>[1] &quot;The traits included in top pairs, B cell (CD19+) &quot;
traits
       EUR.UC           hct           hgb           mcv           mpv 
            1             2             2             2             1 
  myeloid_wbc          neut           pct           pdw           rdw 
            1             1             3             2             3 
          ret ukb.allasthma 
            1             1 
[1] &quot;The pathways included in top pairs, B cell (CD19+) &quot;
pwy
pwy114 pwy122 pwy123 pwy124 pwy125 pwy137 pwy162 pwy163 pwy177 pwy186 pwy202 
     2      1      1      1      1      1      1      1      1      1      1 
pwy259 pwy268 pwy271 pwy276 pwy284 pwy298 pwy318  pwy89 
     1      1      1      1      1      1      1      1 
[1] &quot;The traits included in top pairs, Monocyte (CD14+) &quot;
traits
      EUR.IBD          baso           hct          mchc          mono 
            1             1             2             1             1 
         neut           plt           rbc           rdw           ret 
            1             9             1             1             1 
ukb.allasthma 
            1 
[1] &quot;The pathways included in top pairs, Monocyte (CD14+) &quot;
pwy
pwy116 pwy118 pwy119 pwy121 pwy122 pwy150 pwy161 pwy173 pwy174 pwy187 pwy236 
     2      1      1      2      1      1      1      2      1      1      1 
pwy286 pwy287 pwy290 pwy304  pwy54 
     1      1      1      2      1 
[1] &quot;The traits included in top pairs, Leukocyte (CD15+) &quot;
traits
gran  hct neut  wbc 
   6    7    6    1 
[1] &quot;The pathways included in top pairs, Leukocyte (CD15+) &quot;
pwy
pwy116 pwy151 pwy163 pwy185 pwy224 pwy230 pwy256 pwy269 pwy271 pwy290 pwy314 
     1      2      1      2      1      2      2      1      2      1      1 
 pwy56  pwy93 
     2      2 
[1] &quot;The traits included in top pairs, Platelet &quot;
traits
         baso            eo           hct         lymph           mcv 
            8             1             2             1             1 
  myeloid_wbc           pdw           plt ukb.allasthma 
            1             1             4             1 
[1] &quot;The pathways included in top pairs, Platelet &quot;
pwy
pwy116 pwy127 pwy136 pwy150 pwy158 pwy159 pwy177 pwy181 pwy192 pwy200 pwy235 
     1      1      1      1      1      1      3      1      2      1      1 
pwy275 pwy294 pwy301 pwy315 pwy316  pwy91 
     1      1      1      1      1      1 
[1] &quot;The traits included in top pairs, T cell (CD4+) &quot;
traits
EUR.UC    T1D   gran    hgb   mchc   mono    mpv   neut    pct    ret 
     2      1      1      1      1      1      1      1      1     10 
[1] &quot;The pathways included in top pairs, T cell (CD4+) &quot;
pwy
  pwy1 pwy117 pwy124 pwy131 pwy135 pwy145 pwy151 pwy195 pwy217 pwy222 pwy283 
     1      1      1      1      1      1      1      2      1      1      2 
pwy289 pwy292 pwy305  pwy86  pwy94  pwy96 
     1      1      2      1      1      1 
[1] &quot;The traits included in top pairs, T cell (CD8+) &quot;
traits
EUR.CD    hgb  lymph    pdw    plt    ret 
     3      1      2     12      1      1 
[1] &quot;The pathways included in top pairs, T cell (CD8+) &quot;
pwy
pwy102 pwy129 pwy131 pwy137 pwy155 pwy158 pwy165 pwy202 pwy216 pwy228 pwy259 
     1      1      1      1      1      1      1      1      1      1      1 
pwy284 pwy288 pwy289 pwy291 pwy292 pwy299 pwy300 pwy303  pwy86 
     1      1      1      1      1      1      1      1      1 </code></pre>
<div id="htmlwidget-c3f4891eb8e9a8c25198" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-c3f4891eb8e9a8c25198">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20"],["ukb.allasthma_pwy268_pc5","neut_pwy177_pc3","hgb_pwy137_pc2","myeloid_wbc_pwy276_pc4","hct_pwy202_pc3","rdw_pwy114_pc4","pdw_pwy122_pc2","pdw_pwy114_pc5","mpv_pwy186_pc3","pct_pwy298_pc1","hct_pwy124_pc3","mcv_pwy284_pc2","rdw_pwy162_pc3","rdw_pwy318_pc4","EUR.UC_pwy123_pc4","pct_pwy89_pc1","mcv_pwy271_pc2","pct_pwy259_pc3","hgb_pwy125_pc3","ret_pwy163_pc1"],["plt_pwy173_pc2","plt_pwy286_pc2","plt_pwy116_pc3","ukb.allasthma_pwy116_pc5","plt_pwy54_pc3","plt_pwy122_pc4","EUR.IBD_pwy121_pc3","hct_pwy173_pc4","neut_pwy121_pc3","rdw_pwy287_pc3","mono_pwy304_pc5","mchc_pwy118_pc3","hct_pwy119_pc5","rbc_pwy174_pc4","plt_pwy290_pc3","ret_pwy161_pc4","plt_pwy236_pc2","plt_pwy150_pc2","plt_pwy187_pc3","baso_pwy304_pc3"],["wbc_pwy151_pc1","gran_pwy271_pc1","neut_pwy271_pc1","hct_pwy163_pc3","gran_pwy185_pc5","neut_pwy185_pc5","hct_pwy290_pc4","gran_pwy93_pc3","neut_pwy93_pc3","hct_pwy269_pc1","gran_pwy56_pc3","neut_pwy56_pc3","hct_pwy116_pc4","hct_pwy314_pc4","hct_pwy151_pc5","gran_pwy256_pc4","neut_pwy256_pc4","gran_pwy230_pc5","neut_pwy230_pc5","hct_pwy224_pc1"],["baso_pwy275_pc5","ukb.allasthma_pwy91_pc3","baso_pwy159_pc4","plt_pwy192_pc2","plt_pwy316_pc4","baso_pwy116_pc2","eo_pwy192_pc2","myeloid_wbc_pwy301_pc4","hct_pwy177_pc3","baso_pwy136_pc4","plt_pwy177_pc3","pdw_pwy177_pc3","baso_pwy235_pc1","plt_pwy294_pc3","baso_pwy315_pc5","baso_pwy181_pc4","baso_pwy150_pc4","mcv_pwy200_pc5","hct_pwy158_pc4","lymph_pwy127_pc5"],["ret_pwy86_pc2","ret_pwy151_pc2","ret_pwy217_pc2","gran_pwy283_pc2","ret_pwy305_pc3","ret_pwy289_pc3","EUR.UC_pwy222_pc1","ret_pwy283_pc3","EUR.UC_pwy145_pc5","neut_pwy195_pc2","mono_pwy117_pc5","pct_pwy96_pc5","ret_pwy94_pc2","mpv_pwy135_pc5","ret_pwy131_pc2","hgb_pwy124_pc5","ret_pwy195_pc4","T1D_pwy305_pc4","ret_pwy292_pc3","mchc_pwy1_pc4"],["pdw_pwy288_pc4","ret_pwy158_pc1","pdw_pwy291_pc1","pdw_pwy102_pc3","pdw_pwy289_pc5","pdw_pwy228_pc5","pdw_pwy129_pc4","pdw_pwy165_pc3","pdw_pwy202_pc3","pdw_pwy303_pc1","pdw_pwy86_pc4","EUR.CD_pwy259_pc4","lymph_pwy284_pc3","lymph_pwy300_pc1","pdw_pwy292_pc1","EUR.CD_pwy131_pc1","pdw_pwy137_pc5","EUR.CD_pwy299_pc1","plt_pwy155_pc3","hgb_pwy216_pc2"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>B cell (CD19+)<\/th>\n      <th>Monocyte (CD14+)<\/th>\n      <th>Leukocyte (CD15+)<\/th>\n      <th>Platelet<\/th>\n      <th>T cell (CD4+)<\/th>\n      <th>T cell (CD8+)<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":20,"columnDefs":[{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[10,20,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="real-acat-2" class="section level2">
<h2>Real ACAT</h2>
<pre><code>[1] &quot;The traits included in top pairs, B cell (CD19+) &quot;
traits
baso   eo  hct  hgb mchc  pct  plt  rbc  rdw  ret 
   1    2    3    2    3    1    1    2    3    2 
[1] &quot;The pathways included in top pairs, B cell (CD19+) &quot;
pwy
pwy120 pwy150 pwy155 pwy191 pwy266 pwy298 pwy299 pwy300 pwy315  pwy89 
     3      1      2      1      3      3      1      1      4      1 
[1] &quot;The traits included in top pairs, Monocyte (CD14+) &quot;
traits
     EUR.CD        gran         mch myeloid_wbc        neut         wbc 
          3           5           1           5           5           1 
[1] &quot;The pathways included in top pairs, Monocyte (CD14+) &quot;
pwy
pwy126 pwy142 pwy143 pwy145 pwy171 pwy230 pwy236 pwy273 pwy299 
     1      1      3      4      1      3      1      3      3 
[1] &quot;The traits included in top pairs, Leukocyte (CD15+) &quot;
traits
EUR.IBD  EUR.UC allergy    baso      eo     hct     hgb   lymph     mch    mchc 
      1       2       1       1       3       1       1       1       1       2 
    mcv    mono     pdw     rdw     ret 
      1       1       1       2       1 
[1] &quot;The pathways included in top pairs, Leukocyte (CD15+) &quot;
pwy
pwy129 pwy145  pwy16  pwy18 pwy192 pwy205 pwy210 pwy234 pwy255 pwy269 pwy276 
     1      1      1      1      1      1      3      1      1      1      2 
pwy280 pwy282 pwy291 pwy296 pwy314 
     1      1      1      1      2 
[1] &quot;The traits included in top pairs, Platelet &quot;
traits
   eo lymph   mch   mcv   mpv   pct   pdw   plt   rbc   ret 
    1     1     3     2     1     1     1     1     1     8 
[1] &quot;The pathways included in top pairs, Platelet &quot;
pwy
  pwy1 pwy106 pwy181 pwy202 pwy211 pwy234 pwy271 pwy279 pwy295 pwy301 pwy302 
     6      1      1      1      1      1      1      1      1      3      2 
 pwy96 
     1 
[1] &quot;The traits included in top pairs, T cell (CD4+) &quot;
traits
       EUR.CD       EUR.IBD        EUR.UC          baso          gran 
            2             1             1             1             2 
          hgb          mchc           mcv   myeloid_wbc          neut 
            1             1             1             1             3 
          pdw           rdw ukb.allasthma           wbc 
            2             1             2             1 
[1] &quot;The pathways included in top pairs, T cell (CD4+) &quot;
pwy
pwy139 pwy155 pwy163 pwy210 pwy216 pwy224 pwy276 pwy283 pwy302 pwy305 pwy310 
     1      1      1      4      1      1      1      1      3      1      1 
pwy313  pwy86  pwy94 
     1      1      2 
[1] &quot;The traits included in top pairs, T cell (CD8+) &quot;
traits
EUR.CD    hct    hgb  lymph   mono    pct    rbc    wbc 
     1      8      1      1      3      1      3      2 
[1] &quot;The pathways included in top pairs, T cell (CD8+) &quot;
pwy
pwy125 pwy139 pwy141 pwy154  pwy16 pwy167 pwy168 pwy210 pwy212 pwy217 pwy256 
     1      1      2      2      1      1      1      1      1      1      2 
pwy273 pwy284 pwy293 pwy322 
     1      2      2      1 </code></pre>
<div id="htmlwidget-cfefb73e51b4928e765c" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-cfefb73e51b4928e765c">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20"],["rdw_pwy315_pc1","mchc_pwy315_pc1","ret_pwy315_pc1","eo_pwy191_pc3","pct_pwy315_pc1","rdw_pwy298_pc1","mchc_pwy298_pc1","hct_pwy120_pc2","baso_pwy89_pc2","hgb_pwy120_pc2","hct_pwy266_pc2","hgb_pwy266_pc2","rbc_pwy120_pc2","ret_pwy298_pc1","plt_pwy299_pc3","mchc_pwy155_pc1","rbc_pwy266_pc2","eo_pwy150_pc5","rdw_pwy155_pc1","hct_pwy300_pc2"],["gran_pwy230_pc5","myeloid_wbc_pwy230_pc5","neut_pwy230_pc5","myeloid_wbc_pwy273_pc5","gran_pwy273_pc5","neut_pwy273_pc5","myeloid_wbc_pwy143_pc5","gran_pwy143_pc5","neut_pwy143_pc5","gran_pwy299_pc5","myeloid_wbc_pwy299_pc5","neut_pwy299_pc5","gran_pwy145_pc4","myeloid_wbc_pwy145_pc4","neut_pwy145_pc4","wbc_pwy145_pc4","EUR.CD_pwy236_pc5","EUR.CD_pwy171_pc5","EUR.CD_pwy142_pc5","mch_pwy126_pc2"],["hgb_pwy280_pc3","lymph_pwy18_pc4","rdw_pwy210_pc3","EUR.UC_pwy314_pc3","mch_pwy210_pc3","mono_pwy296_pc3","ret_pwy269_pc4","EUR.IBD_pwy314_pc3","mchc_pwy291_pc4","mcv_pwy210_pc3","rdw_pwy129_pc3","eo_pwy234_pc2","hct_pwy192_pc2","pdw_pwy205_pc1","baso_pwy276_pc1","mchc_pwy276_pc1","eo_pwy255_pc2","EUR.UC_pwy145_pc3","eo_pwy282_pc5","allergy_pwy16_pc2"],["ret_pwy271_pc5","ret_pwy1_pc1","pdw_pwy1_pc1","lymph_pwy1_pc1","mpv_pwy1_pc1","plt_pwy1_pc1","pct_pwy1_pc1","ret_pwy181_pc3","rbc_pwy301_pc2","mcv_pwy301_pc2","ret_pwy202_pc4","mch_pwy301_pc2","eo_pwy295_pc3","mch_pwy211_pc5","ret_pwy96_pc5","mch_pwy302_pc1","ret_pwy302_pc1","mcv_pwy106_pc5","ret_pwy234_pc4","ret_pwy279_pc5"],["EUR.CD_pwy210_pc5","gran_pwy302_pc4","neut_pwy302_pc4","pdw_pwy283_pc3","wbc_pwy302_pc4","EUR.CD_pwy163_pc4","neut_pwy224_pc4","EUR.UC_pwy155_pc3","ukb.allasthma_pwy139_pc3","rdw_pwy94_pc3","mchc_pwy94_pc2","pdw_pwy305_pc3","EUR.IBD_pwy216_pc4","baso_pwy313_pc4","neut_pwy210_pc3","gran_pwy210_pc3","myeloid_wbc_pwy210_pc3","ukb.allasthma_pwy276_pc2","mcv_pwy86_pc3","hgb_pwy310_pc2"],["lymph_pwy212_pc3","mono_pwy284_pc2","wbc_pwy284_pc2","hct_pwy141_pc3","hgb_pwy168_pc4","mono_pwy293_pc3","wbc_pwy293_pc3","rbc_pwy141_pc3","hct_pwy256_pc2","rbc_pwy256_pc2","hct_pwy154_pc1","EUR.CD_pwy217_pc5","hct_pwy16_pc2","pct_pwy322_pc2","hct_pwy167_pc1","hct_pwy139_pc1","rbc_pwy154_pc1","hct_pwy210_pc1","mono_pwy125_pc4","hct_pwy273_pc1"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>B cell (CD19+)<\/th>\n      <th>Monocyte (CD14+)<\/th>\n      <th>Leukocyte (CD15+)<\/th>\n      <th>Platelet<\/th>\n      <th>T cell (CD4+)<\/th>\n      <th>T cell (CD8+)<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":20,"columnDefs":[{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[10,20,25,50,100]}},"evals":[],"jsHooks":[]}</script>
<!-- We also compared the adjusted random p-values with adjusted real p-values. The inflation factors for real p-values are 5.775381 6.671074 4.687092 4.605726 5.453085 4.84691.  -->
<!-- ```{r echo=FALSE, message=FALSE, warning=FALSE, paged.print=FALSE,fig.height=10, fig.width=15} -->
<!-- load("/project2/xinhe/xsun/CEDAR/ACAT_randomsnp/pval_gc_corrected_random_MHCremoved.rdata") -->
<!-- acat_random <- pval_corrected -->
<!-- load("/project2/xinhe/xsun/CEDAR/ACAT_randomsnp/pval_gc_corrected_real_MHCremoved.rdata") -->
<!-- acat_real <- pval_corrected -->
<!-- type <- c("B cell (CD19+)","Monocyte (CD14+)","Leukocyte (CD15+)","Platelet","T cell (CD4+)","T cell (CD8+)") -->
<!-- p <- list() -->
<!-- for (i in 1:length(acat_random)){ -->
<!--   p_random <- acat_random[[i]] -->
<!--   p_real <- acat_real[[i]] -->
<!--   if (length(p_random) > length(p_real)) { -->
<!--     p_real[length(p_real)+1] <- 1 -->
<!--   } -->
<!--   p_random[is.na(p_random)] <- 1 -->
<!--   plotdata <- data.frame(observed_random = -log10(sort(as.numeric(as.character(p_random)))), -->
<!--                          expected = -log10(ppoints(length(p_random))), -->
<!--                          observed_real = -log10(sort(as.numeric(as.character(p_real)))) ) -->
<!--   dat.m <- melt(plotdata, id.vars = "expected") -->
<!--   p[[i]]<- ggplot(dat.m, aes(expected, value, colour = variable)) + theme_bw(base_line_size =0.3) + -->
<!--     geom_point(shape = 1, size = 1.5) + -->
<!--     geom_abline(intercept = 0, slope = 1, alpha = 0.6, color = "red") + -->
<!--     labs(x = expression(paste("Expected -log"[10],"(p-value)")), -->
<!--          y = expression(paste("Observed -log"[10],"(p-value)"))) + -->
<!--     theme(axis.title.x = element_text(size = 10), -->
<!--           axis.text.x = element_text(size = 8, color = "black"), -->
<!--           axis.title.y = element_text(size = 10), -->
<!--           axis.text.y = element_text(size = 8, color = "black"), -->
<!--           text= element_text(family="Times New Roman")) + -->
<!--     scale_colour_manual(values = c("red", "steelblue")) + -->
<!--     ggtitle(type[i]) + theme(plot.title = element_text(hjust = 0.5)) -->
<!-- } -->
<!-- all <- grid.arrange(p[[1]],p[[2]],p[[3]],p[[4]],p[[5]],p[[6]], nrow = 2) -->
<!-- ``` -->
<!-- ```{r echo=FALSE, message=FALSE, warning=FALSE, paged.print=FALSE,fig.height=10, fig.width=15} -->
<!-- load("/project2/xinhe/xsun/CEDAR/ACAT_randomsnp/ACAT_results_remove_MHC_realdata.rdata") -->
<!-- type <- c("B cell (CD19+)","Monocyte (CD14+)","Leukocyte (CD15+)","Platelet","T cell (CD4+)","T cell (CD8+)") -->
<!-- plot <- list() -->
<!-- for (i in 1:length(type)){ -->
<!--   p <- acat_allcelltypes[[i]] -->
<!--   p[is.na(p)] <- 1 -->
<!--   plotdata <- data.frame(observed = -log10(sort(p)), -->
<!--                          expected = -log10(ppoints(length(p)))) -->
<!--   #   -->
<!--   plot[[i]] <- ggplot(plotdata) + theme_bw(base_line_size =0.3) + -->
<!--     geom_point(aes(expected, observed), shape = 1, size = 1.5,color = "steelblue") + -->
<!--     geom_abline(intercept = 0, slope = 1, alpha = 0.6, color = "red") + -->
<!--     ggtitle(type[i]) + theme(plot.title = element_text(hjust = 0.5)) + -->
<!--     labs(x = expression(paste("Expected -log"[10],"(p-value)")), -->
<!--          y = expression(paste("Observed -log"[10],"(p-value)"))) + -->
<!--     theme(axis.title.x = element_text(size = 10), -->
<!--           axis.text.x = element_text(size = 8, color = "black"), -->
<!--           axis.title.y = element_text(size = 10), -->
<!--           axis.text.y = element_text(size = 8, color = "black"), -->
<!--           text= element_text(family="Times New Roman"))  -->
<!-- } -->
<!-- all <- grid.arrange(plot[[1]],plot[[2]],plot[[3]],plot[[4]],plot[[5]],plot[[6]], nrow = 2) -->
<!-- ``` -->
<!-- ## ACAT for random data -- MHC removed -->
<!-- ```{r echo=FALSE, message=FALSE, warning=FALSE, paged.print=FALSE,fig.height=10, fig.width=15} -->
<!-- load("/project2/xinhe/xsun/CEDAR/ACAT_randomsnp/ACAT_results_remove_MHC_random.rdata") -->
<!-- type <- c("B cell (CD19+)","Monocyte (CD14+)","Leukocyte (CD15+)","Platelet","T cell (CD4+)","T cell (CD8+)") -->
<!-- plot <- list() -->
<!-- for (i in 1:length(type)){ -->
<!--   p <- acat_allcelltypes[[i]] -->
<!--   p[is.na(p)] <- 1 -->
<!--   plotdata <- data.frame(observed = -log10(sort(p)), -->
<!--                          expected = -log10(ppoints(length(p)))) -->
<!--   #   -->
<!--   plot[[i]] <- ggplot(plotdata) + theme_bw(base_line_size =0.3) + -->
<!--     geom_point(aes(expected, observed), shape = 1, size = 1.5,color = "steelblue") + -->
<!--     geom_abline(intercept = 0, slope = 1, alpha = 0.6, color = "red") + -->
<!--     ggtitle(type[i]) + theme(plot.title = element_text(hjust = 0.5)) + -->
<!--     labs(x = expression(paste("Expected -log"[10],"(p-value)")), -->
<!--          y = expression(paste("Observed -log"[10],"(p-value)"))) + -->
<!--     theme(axis.title.x = element_text(size = 10), -->
<!--           axis.text.x = element_text(size = 8, color = "black"), -->
<!--           axis.title.y = element_text(size = 10), -->
<!--           axis.text.y = element_text(size = 8, color = "black"), -->
<!--           text= element_text(family="Times New Roman"))  -->
<!-- } -->
<!-- all <- grid.arrange(plot[[1]],plot[[2]],plot[[3]],plot[[4]],plot[[5]],plot[[6]], nrow = 2) -->
<!-- ``` -->
<!-- # Correct original ACAT results with random ones -- All SNPs included -->
<!-- There is still slight deviation from null in the random SNPs after removing SNPs in MHC area. So we correct for real ACAT p-values using the distribution of ACAT p-values from random SNPs as null. In this way, we got empirical ACAT p-values computed from   sum(p_origin[j] < acat_p_random)/length(p_origin). -->
<!-- ```{r echo=FALSE, message=FALSE, warning=FALSE, paged.print=FALSE,fig.height=10, fig.width=15} -->
<!-- setwd("/project2/xinhe/xsun/CEDAR/") -->
<!-- type <- c("B cell (CD19+)","Monocyte (CD14+)","Leukocyte (CD15+)","Platelet","T cell (CD4+)","T cell (CD8+)") -->
<!-- path <- c("4.B_cell_kegg","6.CD14_positive_monocyte_kegg","8.CD15_positive_leukocyte_kegg","16.platelet_kegg","10.T_cell_kegg","14.thymocyte_kegg") -->
<!-- load("/project2/xinhe/xsun/CEDAR/ACAT_randomsnp/ACAT_results.rdata") -->
<!-- p_correct_allcelltypes <- list() -->
<!-- for (i in 1:length(type)){ -->
<!--   file <- paste0("/project2/xinhe/xsun/CEDAR/",path[i],"/7.geneset40_peer/acat.rdata") -->
<!--   load(file) -->
<!--   p_origin <- as.numeric(list$pval_acat) -->
<!--   acat_cell_random <- acat_allcelltypes[[i]] -->
<!--   acat_cell_random[acat_cell_random == "NaN"] <- 0.00001 -->
<!--   p_correct <- c() -->
<!--   for (j in 1:length(p_origin)) { -->
<!--     p_correct[j] <- sum(p_origin[j] < acat_cell_random)/length(p_origin) -->
<!--   } -->
<!--   p_correct_allcelltypes[[i]] <- p_correct -->
<!-- } -->
<!-- p <- list() -->
<!-- for (i in 1:length(p_correct_allcelltypes)){ -->
<!--   plotdata <- data.frame(observed = -log10(sort(as.numeric(as.character(p_correct_allcelltypes[[i]])))), -->
<!--                          expected = -log10(ppoints(length(p_correct_allcelltypes[[i]])))) -->
<!--   #   -->
<!--   p[[i]] <- ggplot(plotdata) + theme_bw(base_line_size =0.3) + -->
<!--     geom_point(aes(expected, observed), shape = 1, size = 1.5, color= "steelblue") + -->
<!--     geom_abline(intercept = 0, slope = 1, alpha = 0.6, color = "red") + -->
<!--     ggtitle(type[i]) + theme(plot.title = element_text(hjust = 0.5)) + -->
<!--     labs(x = expression(paste("Expected -log"[10],"(p-value)")), -->
<!--          y = expression(paste("Observed -log"[10],"(p-value)"))) + -->
<!--     theme(axis.title.x = element_text(size = 10), -->
<!--           axis.text.x = element_text(size = 8, color = "black"), -->
<!--           axis.title.y = element_text(size = 10), -->
<!--           axis.text.y = element_text(size = 8, color = "black"), -->
<!--           text= element_text(family="Times New Roman"))  -->
<!-- } -->
<!-- all <- grid.arrange(p[[1]],p[[2]],p[[3]],p[[4]],p[[5]],p[[6]], nrow = 2) -->
<!-- ``` -->
<!-- # Correct original ACAT results with random ones -- MHC area removed -->
<!-- ```{r echo=FALSE, message=FALSE, warning=FALSE, paged.print=FALSE,fig.height=10, fig.width=15} -->
<!-- type <- c("B cell (CD19+)","Monocyte (CD14+)","Leukocyte (CD15+)","Platelet","T cell (CD4+)","T cell (CD8+)") -->
<!-- load("/project2/xinhe/xsun/CEDAR/ACAT_randomsnp/ACAT_results_remove_MHC_random.rdata") -->
<!-- acat_random <- acat_allcelltypes -->
<!-- load("/project2/xinhe/xsun/CEDAR/ACAT_randomsnp/ACAT_results_remove_MHC_realdata.rdata") -->
<!-- acat_real <- acat_allcelltypes -->
<!-- p_correct_allcelltypes <- list() -->
<!-- for (i in 1:length(acat_real)){ -->
<!--   p_origin <- as.numeric(as.character(acat_real[[i]])) -->
<!--   acat_cell_random <- as.numeric(as.character(acat_random[[i]])) -->
<!--   acat_cell_random[acat_cell_random == "NaN"] <- 0.00001 -->
<!--   p_correct <- c() -->
<!--   for (j in 1:length(p_origin)) { -->
<!--     p_correct[j] <- sum(p_origin[j] < acat_cell_random)/length(p_origin) -->
<!--   } -->
<!--   p_correct_allcelltypes[[i]] <- p_correct -->
<!-- } -->
<!-- p <- list() -->
<!-- for (i in 1:length(p_correct_allcelltypes)){ -->
<!--   plotdata <- data.frame(observed = -log10(sort(as.numeric(as.character(p_correct_allcelltypes[[i]])))), -->
<!--                          expected = -log10(ppoints(length(p_correct_allcelltypes[[i]])))) -->
<!--   #   -->
<!--   p[[i]] <- ggplot(plotdata) + theme_bw(base_line_size =0.3) + -->
<!--     geom_point(aes(expected, observed), shape = 1, size = 1.5, color= "steelblue") + -->
<!--     geom_abline(intercept = 0, slope = 1, alpha = 0.6, color = "red") + -->
<!--     ggtitle(type[i]) + theme(plot.title = element_text(hjust = 0.5)) + -->
<!--     labs(x = expression(paste("Expected -log"[10],"(p-value)")), -->
<!--          y = expression(paste("Observed -log"[10],"(p-value)"))) + -->
<!--     theme(axis.title.x = element_text(size = 10), -->
<!--           axis.text.x = element_text(size = 8, color = "black"), -->
<!--           axis.title.y = element_text(size = 10), -->
<!--           axis.text.y = element_text(size = 8, color = "black"), -->
<!--           text= element_text(family="Times New Roman"))  -->
<!-- } -->
<!-- all <- grid.arrange(p[[1]],p[[2]],p[[3]],p[[4]],p[[5]],p[[6]], nrow = 2) -->
<!-- ``` -->
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
[1] C

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] data.table_1.14.2 ACAT_0.91         gridExtra_2.3     ggplot2_3.3.5    

loaded via a namespace (and not attached):
 [1] tidyselect_1.1.2  xfun_0.30         bslib_0.3.1       reshape2_1.4.4   
 [5] purrr_0.3.4       colorspace_2.0-3  vctrs_0.4.1       generics_0.1.2   
 [9] htmltools_0.5.2   yaml_2.3.5        utf8_1.2.2        rlang_1.0.5      
[13] jquerylib_0.1.4   later_1.3.0       pillar_1.7.0      glue_1.6.2       
[17] withr_2.5.0       DBI_1.1.2         plyr_1.8.7        lifecycle_1.0.1  
[21] stringr_1.4.0     munsell_0.5.0     gtable_0.3.0      workflowr_1.7.0  
[25] htmlwidgets_1.5.4 evaluate_0.15     labeling_0.4.2    knitr_1.39       
[29] fastmap_1.1.0     crosstalk_1.2.0   httpuv_1.6.5      fansi_1.0.3      
[33] highr_0.9         Rcpp_1.0.8.3      DT_0.22           promises_1.2.0.1 
[37] scales_1.2.0      jsonlite_1.8.0    farver_2.1.0      fs_1.5.2         
[41] digest_0.6.29     stringi_1.7.6     dplyr_1.0.9       grid_4.2.0       
[45] rprojroot_2.0.3   cli_3.3.0         tools_4.2.0       magrittr_2.0.3   
[49] sass_0.4.1        tibble_3.1.7      crayon_1.5.1      whisker_0.4      
[53] pkgconfig_2.0.3   ellipsis_0.3.2    assertthat_0.2.1  rmarkdown_2.14   
[57] rstudioapi_0.13   R6_2.5.1          git2r_0.30.1      compiler_4.2.0   </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
