<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="XSun" />

<meta name="date" content="2020-10-19" />

<title>catalog_gwas_peer_sep_ldclumping_r01d1000</title>

<script src="site_libs/jquery-1.12.4/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.2/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.15/datatables.js"></script>
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.1.0.1/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">factor_analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/xsun1229/factor_analysis">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">catalog_gwas_peer_sep_ldclumping_r01d1000</h1>
<h4 class="author">XSun</h4>
<h4 class="date">2020-10-19</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-10-19
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>factor_analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div class="panel-group" id="workflowr-checks">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200623code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200623)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200623code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200623)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it difficult for you and others to run your code on a different machine. Change the absolute path(s) below to the suggested relative path(s) to make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/factor_analysis/output/sum_sep_tog_pval_peer_wgpc_ldcl_d1k.rdata
</td>
<td style="text-align:left;">
output/sum_sep_tog_pval_peer_wgpc_ldcl_d1k.rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/factor_analysis/output/info_pval5e8_peer_ld_d1k.rdata
</td>
<td style="text-align:left;">
output/info_pval5e8_peer_ld_d1k.rdata
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomxsun1229factoranalysistree5374205dd44ccc3c3a110d38b4219f30a8da6863targetblank5374205a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/xsun1229/factor_analysis/tree/5374205dd44ccc3c3a110d38b4219f30a8da6863" target="_blank">5374205</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomxsun1229factoranalysistree5374205dd44ccc3c3a110d38b4219f30a8da6863targetblank5374205a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/xsun1229/factor_analysis/tree/5374205dd44ccc3c3a110d38b4219f30a8da6863" target="_blank">5374205</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    analysis/.Rhistory

Unstaged changes:
    Modified:   delete.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/catalog_gwas_peer_sep_ldclumping_r01d1000.Rmd</code>) and HTML (<code>docs/catalog_gwas_peer_sep_ldclumping_r01d1000.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/5374205dd44ccc3c3a110d38b4219f30a8da6863/analysis/catalog_gwas_peer_sep_ldclumping_r01d1000.Rmd" target="_blank">5374205</a>
</td>
<td>
XSun
</td>
<td>
2020-10-19
</td>
<td>
wflow_publish(c(“analysis/index.Rmd”, “analysis/catalog_gwas_peer_sep_ldclumping_r01d1000.Rmd”))
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/95057728e6167eca0f09f3481cd209cb10c070b2/analysis/catalog_gwas_peer_sep_ldclumping_r01d1000.Rmd" target="_blank">9505772</a>
</td>
<td>
XSun
</td>
<td>
2020-10-19
</td>
<td>
update
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>In this part, I considered the traits separately. I selected the SNPs with pval &lt; 5e-8 for each traits. Then, I did LD Clumping for these SNPs to eliminate the LD and select a smaller subset of SNPs. Finally, I did GWAS analysis for the plier_canonical factors with the SNPs.</p>
</div>
<div id="material-and-methods" class="section level2">
<h2>Material and Methods</h2>
<ol style="list-style-type: decimal">
<li><p>I used the GWAS results of 12 immune traits: platelet count, white blood cell count, myeloid white cell count, lymphocyte counts, red blood cell count, granulocyte count, eosinophil count, neutrophil count. These data are from <em>Astle WJ, Elding H, Jiang T, et al. The Allelic Landscape of Human Blood Cell Trait Variation and Links to Common Complex Disease. Cell. 2016;167(5):1415-1429.e19. <a href="doi:10.1016/j.cell.2016.10.042" class="uri">doi:10.1016/j.cell.2016.10.042</a></em> . I also used IBD,Ulcerative colitist,Crohn’s disease, asthma, BMI,and T2D collected by our lab.</p></li>
<li><p>Computed the fdr of each SNP according to their p-values from GWAS data. Chose those SNPs with pval &lt; 5e-8.</p></li>
<li><p>Did LD Clumping for the SNPs in obtained from step 2. I used default settings of PLINK LD clumping:</p>
<p>–clump-p1 0.0001 Significance threshold for index SNPs</p>
<p>–clump-p2 0.01 Secondary significance threshold for clumped SNPs</p>
<p>–clump-r2 0.1 LD threshold for clumping</p>
<p>–clump-kb 1000 Physical distance threshold for clumping</p></li>
</ol>
<p>For each trait, I got a subset of SNPs that are not in LD with each other.</p>
<ol start="4" style="list-style-type: decimal">
<li><p>Did GWAS analysis for plier_canonical factors and SNPs from 3. The association tests were corrected by 10 genotype PCs of whole genome.</p></li>
<li><p>For each trait, I made a plot of association with LV(indicating by beta in GWAS) vs association with trait(indicating by ln(odds ratio) or beta in GWAS) to show if the variants have the correlated effect direction. The effect sizes of Catalog GWAS and factor association tests are harmonized by <a href="https://mrcieu.github.io/TwoSampleMR/articles/introduction.html">TwoSampleMR</a> R package to make the effect alleles in these two analysis identical. The LVs have more than one significant SNPs with FDR&lt;0.2 are included in the plotting.Besides, for each plots, I fitted the points with intercept = 0. The pvalues and r-squared are shown on the plots.</p></li>
<li><p>For the traits and LVs in 5, I made an info table to show more details of the SNPs.</p></li>
<li><p>For several LVs we are interested in, I did gene set enrichment analysis to test if the LVs are correlated with some KEGG/REACTOME pathways. I used two kind gene sets to do GSEA: 1. genes that used to compute LVs; 2. Sorting the genes in 1 by their loadings, take the top 25% as the gene set. For both gene sets, the gene scores used as input of GSEA are the gene loadings.</p></li>
</ol>
</div>
<div id="results---pval-5e-8" class="section level2">
<h2>Results - pval &lt; 5e-8</h2>
<div id="overview" class="section level3">
<h3>Overview</h3>
<p>After filtering by ‘pval &lt; 5e-8’ and LD Clumping, for each trait, I got :</p>
<p>platelet count trait contains 688 SNPs with pval&lt;5e-8.</p>
<p>white blood cell count trait contains 368 SNPs with pval&lt;5e-8.</p>
<p>myeloid white cell count trait contains 319 SNPs with pval&lt;5e-8.</p>
<p>lymphocyte count trait contains 436 SNPs with pval&lt;5e-8.</p>
<p>red blood cell count trait contains 466 SNPs with pval&lt;5e-8.</p>
<p>granulocyte count trait contains 316 SNPs with pval&lt;5e-8.</p>
<p>eosinophil count trait contains 491 SNPs with pval&lt;5e-8.</p>
<p>neutrophil count trait contains 317 SNPs with pval&lt;5e-8.</p>
<p>IBD trait contains 116 SNPs with pval&lt;5e-8.</p>
<p>Ulcerative colitist trait contains 73 SNPs with pval&lt;5e-8.</p>
<p>Crohn’s disease trait contains 96 SNPs with pval&lt;5e-8.</p>
<p>asthma trait contains 186 SNPs with pval&lt;5e-8.</p>
<p>BMI trait contains 104 SNPs with pval&lt;5e-8.</p>
<p>T2D contains 14 SNPs with pval&lt;5e-8.</p>
</div>
<div id="summary-table" class="section level3">
<h3>Summary table</h3>
<p>I used ‘qvalue’ R package to compute the fdr from p-values for each SNP and made a table to show the number of SNPs that pass the threshold. The thresholds are ‘fdr &lt; 0.1’,‘fdr &lt; 0.2’,‘pval &lt; 5e-8’.</p>
<div id="htmlwidget-c3f4891eb8e9a8c25198" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-c3f4891eb8e9a8c25198">{"x":{"filter":"none","data":[["factor1","factor2","factor3","factor4","factor5","factor6","factor7","factor8","factor9","factor10","factor11","factor12","factor13","factor14","factor15","factor16","factor17","factor18","factor19","factor20","factor21","factor22","factor23","factor24","factor25","factor26","factor27","factor28","factor29","factor30","factor31","factor32","factor33","factor34","factor35","factor36","factor37","factor38","factor39","factor40","factor41","factor42","factor43","factor44","factor45","factor46","factor47","factor48","factor49","factor50","factor51","factor52","factor53","factor54","factor55","factor56","factor57","factor58","factor59","factor60"],[0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,1,1,0,0,0,0,2,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,2,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0],[0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,2,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0],[0,0,0,0,0,2,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,1,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,3,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,1,1,0,0,1,0,1,0,0,0,0,1,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,3,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,1,0,1,0,0,0,0,0,0,0,0,1,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,2,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,3,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,1,0,0,1,0,1,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,3,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,2,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,1,1,0,0,1,0,1,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,3,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,1],[0,0,0,0,0,0,2,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,1,1,0,1,0,0,0,0,0,0,1,0,0,1],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,1,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,1,1,1,0,0,0,0,1,0,2,0,1,0,0,0,0,0,1,0,0,0,0,0,0,0,0,2,0,0,0,0,0,0,0,0,0,0,0,0,2,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,1,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,2,0,0,0,0,0,0,1,0,0,0,0,0,1,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,1,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,3,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,1,1,0,0,1,0,1,1,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,4,0,0,0,0,0,0,0,0,1,0,0,0,0,1,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>bmi_fdr010_num<\/th>\n      <th>bmi_fdr020_num<\/th>\n      <th>bmi_pval5e8_num<\/th>\n      <th>eo_fdr010_num<\/th>\n      <th>eo_fdr020_num<\/th>\n      <th>eo_pval5e8_num<\/th>\n      <th>EUR.CD_fdr010_num<\/th>\n      <th>EUR.CD_fdr020_num<\/th>\n      <th>EUR.CD_pval5e8_num<\/th>\n      <th>EUR.IBD_fdr010_num<\/th>\n      <th>EUR.IBD_fdr020_num<\/th>\n      <th>EUR.IBD_pval5e8_num<\/th>\n      <th>EUR.UC_fdr010_num<\/th>\n      <th>EUR.UC_fdr020_num<\/th>\n      <th>EUR.UC_pval5e8_num<\/th>\n      <th>gran_fdr010_num<\/th>\n      <th>gran_fdr020_num<\/th>\n      <th>gran_pval5e8_num<\/th>\n      <th>lymph_fdr010_num<\/th>\n      <th>lymph_fdr020_num<\/th>\n      <th>lymph_pval5e8_num<\/th>\n      <th>myeloid_wbc_fdr010_num<\/th>\n      <th>myeloid_wbc_fdr020_num<\/th>\n      <th>myeloid_wbc_pval5e8_num<\/th>\n      <th>neut_fdr010_num<\/th>\n      <th>neut_fdr020_num<\/th>\n      <th>neut_pval5e8_num<\/th>\n      <th>plt_fdr010_num<\/th>\n      <th>plt_fdr020_num<\/th>\n      <th>plt_pval5e8_num<\/th>\n      <th>rbc_fdr010_num<\/th>\n      <th>rbc_fdr020_num<\/th>\n      <th>rbc_pval5e8_num<\/th>\n      <th>T2D_fdr010_num<\/th>\n      <th>T2D_fdr020_num<\/th>\n      <th>T2D_pval5e8_num<\/th>\n      <th>ukb.allasthma_fdr010_num<\/th>\n      <th>ukb.allasthma_fdr020_num<\/th>\n      <th>ukb.allasthma_pval5e8_num<\/th>\n      <th>wbc_fdr010_num<\/th>\n      <th>wbc_fdr020_num<\/th>\n      <th>wbc_pval5e8_num<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":20,"columnDefs":[{"className":"dt-right","targets":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[10,20,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="info-table" class="section level3">
<h3>Info table</h3>
<p>For each trait, I made a table to show the info of snps with fdr&gt;0.2 in the factor ~ SNP + genotype pcs association test. For each trait,The LVs have more than one significant SNPs with FDR&lt;0.2 are included.</p>
<p>The suffix ’_assoc’ here means that results are from factor ~ SNP + genotype pcs association test. The suffix ’_gwas’ here means results are from original GWAS results files. For EUR.CD, EUR.IBD, EUR.UC, the effectsize_gwas here means ‘ln(OR)’, for others, it means ‘beta’.</p>
<p>‘snp_ld’ here means the snps that in LD with the snp in each line.’ld_r2’ means the LD r-squared which is corresponding to the ‘snp_ld’ column. ‘cis-eqtl’ column indicates whether the snp is a cis-eqtl according to GTEx data. ‘cis_gene_hgnc’ and ‘cis_gene_hgnc’ is the genes that the snp influence when it act as cis-eqtl. ‘func’ and ‘func_gene’ are obtained from ANNOVAR, which indicating the snp function within the genes.</p>
<div id="htmlwidget-cfefb73e51b4928e765c" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-cfefb73e51b4928e765c">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33"],["bmi","bmi","bmi","bmi","eo","eo","EUR.IBD","EUR.IBD","gran","gran","gran","lymph","lymph","myeloid_wbc","myeloid_wbc","myeloid_wbc","neut","neut","neut","plt","plt","rbc","rbc","rbc","rbc","rbc","rbc","ukb.allasthma","ukb.allasthma","wbc","wbc","wbc","wbc"],["LV14","LV14","LV40","LV40","LV57","LV57","LV6","LV6","LV43","LV43","LV43","LV44","LV44","LV43","LV43","LV43","LV43","LV43","LV43","LV7","LV7","LV11","LV11","LV28","LV28","LV41","LV41","LV24","LV24","LV43","LV43","LV43","LV43"],["chr19_18344015_A_G_b38","chr3_85758440_T_G_b38","chr10_85651147_A_G_b38","chr9_126698635_A_G_b38","chr6_30338176_A_G_b38","chr8_129662115_CTACAT_C_b38","chr16_50349166_C_T_b38","chr6_14721666_T_C_b38","chr10_44386212_C_T_b38","chr17_39987826_G_T_b38","chr6_109286353_G_A_b38","chr12_112565131_G_T_b38","chr20_1986403_C_T_b38","chr10_44384812_T_C_b38","chr17_39987826_G_T_b38","chr6_109286353_G_A_b38","chr10_44384812_T_C_b38","chr17_39987826_G_T_b38","chr6_109286353_G_A_b38","chr12_111429489_A_C_b38","chr9_5067832_T_C_b38","chr2_126662850_G_A_b38","chr6_43842065_T_C_b38","chr6_26092913_G_A_b38","chr8_129591389_C_T_b38","chr7_100144470_G_A_b38","chr7_100370748_C_T_b38","chr17_40021896_A_G_b38","chr3_33026456_C_T_b38","chr10_44386212_C_T_b38","chr17_2101961_CAGA_C_b38","chr17_39987826_G_T_b38","chr6_109300291_T_C_b38"],["rs17724992","rs13078960","rs7899106","rs10733682","rs9261659","rs146895157","rs9934775","rs9370774","rs17885289","rs118009374","rs9400271","rs73209657","rs13037430","rs72790862","rs118009374","rs9400271","rs72790862","rs118009374","rs9400271","rs61944375","rs73393498","rs28387101","rs9472135","rs1800562","rs10107630","rs4729578","rs62482199","rs2302776","rs72854772","rs17885289","rs71902577","rs118009374","rs6927569"],["-0.0194","0.0297","0.0395","0.0174","0.03385428","-0.02825609","-0.139595456234745","-0.130703648321444","-0.02326956","-0.08451142","0.03462562","0.08388152","-0.04181903","-0.02433963","-0.08176227","0.03495081","-0.0220783","-0.08172991","0.03322504","0.0887707","-0.09418124","-0.04877128","-0.03799032","-0.04991289","-0.02196291","-0.0328844","0.05599315","-0.0576873","-0.0699484","-0.02783845","-0.03205232","-0.07104802","-0.029455"],["3.415e-08","1.737e-14","2.96e-08","1.83e-08","2.049e-09","3.914e-15","1.715e-09","2.536e-09","2.677e-09","2.712e-18","5.496e-22","5.043e-12","9.109e-09","5.443e-10","3.886e-17","2.861e-22","1.541e-08","2.799e-17","1.848e-20","1.193e-09","5.997e-13","4.07e-08","4.987e-23","5.157e-15","1.209e-09","2.172e-14","6.19e-23","3.4e-20","3.6e-09","1.005e-12","7.173e-13","2.102e-13","2.321e-16"],["-0.009436","0.01367","0.03254","0.01524","-0.04142","0.02331","-0.01138","0.009851","0.01968","-0.05133","0.01899","-0.07439","-0.04047","0.02003","-0.05133","0.01899","0.02003","-0.05133","0.01899","-0.03167","-0.02717","0.01812","-0.008757","-0.02747","-0.01289","-0.01905","0.0267","0.01112","-0.02087","0.01968","-0.02172","-0.05133","-0.0184"],["0.001317","2.573e-05","0.0006072","0.001123","0.0001102","0.0002048","0.0008663","0.001007","0.0002933","0.0003855","0.0001321","0.0004051","0.0001279","0.0002282","0.0003855","0.0001321","0.0002282","0.0003855","0.0001321","0.0001809","0.0001332","0.0001659","0.0006756","0.0001122","0.0004628","0.0004502","0.0003737","0.001032","0.001088","0.0002933","0.0001748","0.0003855","0.0003175"],["0.14785221594302","0.0133554759649393","0.117303745331803","0.185580467973068","0.0977186816723455","0.156674994348622","0.119206855990096","0.128746066202813","0.076857055804324","0.0908671041066077","0.0491766760881988","0.179796082155159","0.0876680086744259","0.0712800215249686","0.0994105273859499","0.051473457228084","0.0751069288468551","0.108579210800774","0.0529680984937062","0.164111009838468","0.133687413881551","0.114959143286857","0.195941440531276","0.0968334658489444","0.190636700891226","0.171619111106082","0.156034049648692","0.157697606655427","0.161952724840863","0.0898553787609563","0.0627910617871024","0.110497419198664","0.0953097752206111"],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,"T",null,"T",null,null,null,"T",null,null,null,null,"T",null,null,"T",null,null,null,null,null,null,"T","T",null,null,null,null,null,"T"],[null,null,null,null,"VARS2,ZFP57,HCG4P5,HLA-T,TRIM26,IFITM4P,RPL23AP1",null,"ADCY7,BRD7",null,null,null,"CD164,SMPD2,CCDC162P",null,null,null,null,"CD164,SMPD2,CCDC162P",null,null,"CD164,SMPD2,CCDC162P",null,null,null,null,null,null,"LAMTOR4,AP4M1","MOSPD3,C7orf43,GIGYF1,EPHB4",null,null,null,null,null,"CD164,SMPD2,CCDC162P"],[null,null,null,null,"ENSG00000137411,ENSG00000204644,ENSG00000227766,ENSG00000231130,ENSG00000234127,ENSG00000235821,ENSG00000239257",null,"ENSG00000121281,ENSG00000166164",null,null,null,"ENSG00000135535,ENSG00000135587,ENSG00000203799",null,null,null,null,"ENSG00000135535,ENSG00000135587,ENSG00000203799",null,null,"ENSG00000135535,ENSG00000135587,ENSG00000203799",null,null,null,null,null,null,"ENSG00000188186,ENSG00000221838","ENSG00000106330,ENSG00000146826,ENSG00000146830,ENSG00000196411",null,null,null,null,null,"ENSG00000135535,ENSG00000135587,ENSG00000203799"],["intronic","intronic","intronic","UTR3","intronic","ncRNA_intronic","intronic","intergenic","intergenic","intronic","ncRNA_exonic","intergenic","ncRNA_intronic","intronic","intronic","ncRNA_exonic","intronic","intronic","ncRNA_exonic","intronic","intronic","intronic","intergenic","exonic","ncRNA_intronic","intergenic","intergenic","intronic","intronic","intergenic","intronic","intronic","ncRNA_exonic"],["PGPEP1","CADM2","GRID1","LMX1B(NM_002316:c.*2184A&gt;G,NM_001174147:c.*2184A&gt;G,NM_001174146:c.*2184A&gt;G)","TRIM39,TRIM39-RPP21","CCDC26","BRD7","LINC01108(dist=436212),JARID2(dist=524403)","CXCL12(dist=1115),TMEM72-AS1(dist=424812)","PSMD3","CCDC162P","PTPN11(dist=55213),RPH3A(dist=10105)","PDYN-AS1","CXCL12","PSMD3","CCDC162P","CXCL12","PSMD3","CCDC162P","SH2B3","JAK2","GYPC","VEGFA(dist=55579),LINC02537(dist=2811)","HFE","CCDC26","MBLAC1(dist=15977),LAMTOR4(dist=4448)","STAG3L5P-PVRIG2P-PILRB(dist=2917),PILRA(dist=2739)","MED24","GLB1","CXCL12(dist=1115),TMEM72-AS1(dist=424812)","SMG6","PSMD3","CCDC162P"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Traits<\/th>\n      <th>factors<\/th>\n      <th>SNP_gtexid<\/th>\n      <th>SNP_rsid<\/th>\n      <th>BETA_gwas<\/th>\n      <th>P_gwas<\/th>\n      <th>BETA_assoc<\/th>\n      <th>P_assoc<\/th>\n      <th>fdr_assoc<\/th>\n      <th>snp_ld<\/th>\n      <th>ld_r2<\/th>\n      <th>cis_eqtl<\/th>\n      <th>cis_genes_hgnc<\/th>\n      <th>cis_genes_ensembl<\/th>\n      <th>func<\/th>\n      <th>func_gene<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":20,"order":[],"autoWidth":false,"orderClasses":false,"columnDefs":[{"orderable":false,"targets":0}],"lengthMenu":[10,20,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="effect-size-plots" class="section level3">
<h3>Effect size plots</h3>
<p>For each trait, I made a plot of association with LV(indicating by beta in GWAS) vs association with trait(indicating by ln(odds ratio) or beta in GWAS) to show if the variants have the correlated effect direction. The effect sizes of Catalog GWAS and factor association tests are harmonized by <a href="https://mrcieu.github.io/TwoSampleMR/articles/introduction.html">TwoSampleMR</a> R package to make the effect alleles in these two analysis identical. The LVs have more than one significant SNPs with FDR&lt;0.2 are included in the plotting.Besides, for each plots, I fitted the points with intercept = 0. The pvalues and r-squared are shown on the plots.</p>
<div id="bmi" class="section level4">
<h4>BMI</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv14_bmi_pval5e8_d1k_peer.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv40_bmi_pval5e8_d1k_peer.jpeg" width="30%">
</figure>
</div>
<div id="eosinophil-count" class="section level4">
<h4>Eosinophil count</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv157_eo_pval5e8_d1k_peer.jpeg" width="30%">
</figure>
</div>
<div id="crohns-disease" class="section level4">
<h4>Crohn’s disease</h4>
<p>None of the LVs have &gt;1 SNPs at FDR&lt;0.2.</p>
</div>
<div id="ibd" class="section level4">
<h4>IBD</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv6_EUR.IBD_pval5e8_d1k_peer.jpeg" width="30%">
</figure>
</div>
<div id="ulcerative-colitist" class="section level4">
<h4>Ulcerative colitist</h4>
<p>None of the LVs have &gt;1 SNPs at FDR&lt;0.2.</p>
</div>
<div id="granulocyte-count" class="section level4">
<h4>granulocyte count</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_gran_pval5e8_d1k_peer.jpeg" width="30%">
</figure>
</div>
<div id="lymphocyte-count" class="section level4">
<h4>lymphocyte count</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv44_lymph_pval5e8_d1k_peer.jpeg" width="30%">
</figure>
</div>
<div id="myeloid-white-cell-count" class="section level4">
<h4>myeloid white cell count</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_myeloid_wbc_pval5e8_d1k_peer.jpeg" width="30%">
</figure>
</div>
<div id="neutrophil-count" class="section level4">
<h4>neutrophil count</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_neut_pval5e8_d1k_peer.jpeg" width="30%">
</figure>
</div>
<div id="plt" class="section level4">
<h4>plt</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_plt_pval5e8_d1k_peer.jpeg" width="30%">
</figure>
</div>
<div id="rbc" class="section level4">
<h4>rbc</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv11_rbc_pval5e8_d1k_peer.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv28_rbc_pval5e8_d1k_peer.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv41_rbc_pval5e8_d1k_peer.jpeg" width="30%">
</figure>
<p>###T2D</p>
<p>None of the LVs have &gt;1 SNPs at FDR&lt;0.2.</p>
</div>
<div id="asthma" class="section level4">
<h4>asthma</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv24_ukb.allasthma_pval5e8_d1k_peer.jpeg" width="30%">
</figure>
</div>
<div id="wbc" class="section level4">
<h4>wbc</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_wbc_pval5e8_d1k_peer.jpeg" width="30%">
</figure>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] workflowr_1.6.2

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.5        rstudioapi_0.11   whisker_0.3-2     knitr_1.30       
 [5] magrittr_1.5      R6_2.4.1          rlang_0.4.8       highr_0.8        
 [9] stringr_1.4.0     tools_3.6.1       DT_0.15           xfun_0.18        
[13] git2r_0.26.1      crosstalk_1.1.0.1 htmltools_0.5.0   ellipsis_0.3.1   
[17] rprojroot_1.3-2   yaml_2.2.1        digest_0.6.25     tibble_3.0.3     
[21] lifecycle_0.2.0   crayon_1.3.4      later_1.1.0.1     htmlwidgets_1.5.2
[25] vctrs_0.3.4       promises_1.1.1    fs_1.5.0          glue_1.4.2       
[29] evaluate_0.14     rmarkdown_1.13    stringi_1.5.3     compiler_3.6.1   
[33] pillar_1.4.6      backports_1.1.10  jsonlite_1.7.1    httpuv_1.5.1     
[37] pkgconfig_2.0.3  </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
