<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="XSun" />

<meta name="date" content="2020-11-19" />

<title>catalog_gwas_pliercanon_sep_ldclumping_r01d1000_adipose_sub</title>

<script src="site_libs/jquery-1.12.4/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.2/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.15/datatables.js"></script>
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.1.0.1/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">factor_analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/xsun1229/factor_analysis">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">catalog_gwas_pliercanon_sep_ldclumping_r01d1000_adipose_sub</h1>
<h4 class="author">XSun</h4>
<h4 class="date">2020-11-19</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-02-02
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>factor_analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div class="panel-group" id="workflowr-checks">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200623code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200623)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200623code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200623)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it difficult for you and others to run your code on a different machine. Change the absolute path(s) below to the suggested relative path(s) to make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/factor_analysis/output/sum_pliercanon_wgpc_ldcl_d1k_ageco_adp.rdata
</td>
<td style="text-align:left;">
output/sum_pliercanon_wgpc_ldcl_d1k_ageco_adp.rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/factor_analysis/output/sum_sep_tog_pval_pliercanon_wgpc_ldcl_d1k_ageco_adp.rdata
</td>
<td style="text-align:left;">
output/sum_sep_tog_pval_pliercanon_wgpc_ldcl_d1k_ageco_adp.rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/factor_analysis/output/info_pval5e8_pliercanon_ld_d1k_ageco_adp.rdata
</td>
<td style="text-align:left;">
output/info_pval5e8_pliercanon_ld_d1k_ageco_adp.rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/factor_analysis/output/EM_SUM_new_turn_func_g_adp.rdata
</td>
<td style="text-align:left;">
output/EM_SUM_new_turn_func_g_adp.rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/factor_analysis/output/resampling_sum_g_adp.rdata
</td>
<td style="text-align:left;">
output/resampling_sum_g_adp.rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/factor_analysis/output/lv35top025_pliercanon_adp.rdata
</td>
<td style="text-align:left;">
output/lv35top025_pliercanon_adp.rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/factor_analysis/output/LDL_35_20816003.coloc.results_adp_500.rdata
</td>
<td style="text-align:left;">
output/LDL_35_20816003.coloc.results_adp_500.rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/factor_analysis/output/LDL_35_45149969.coloc.results_adp_500.rdata
</td>
<td style="text-align:left;">
output/LDL_35_45149969.coloc.results_adp_500.rdata
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomxsun1229factoranalysistree92a0304f48f962e8a97abf36f9273567ab0b1a8ctargetblank92a0304a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/xsun1229/factor_analysis/tree/92a0304f48f962e8a97abf36f9273567ab0b1a8c" target="_blank">92a0304</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomxsun1229factoranalysistree92a0304f48f962e8a97abf36f9273567ab0b1a8ctargetblank92a0304a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/xsun1229/factor_analysis/tree/92a0304f48f962e8a97abf36f9273567ab0b1a8c" target="_blank">92a0304</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    analysis/.Rhistory

Unstaged changes:
    Deleted:    resampling_sum_g_adp.rdata
    Deleted:    sum_sep_tog_pval_pliercanon_wgpc_ldcl_d1k_ageco_adp.rdata

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/catalog_gwas_pliercanon_sep_ldclumping_r01d1000_adipose_sub.Rmd</code>) and HTML (<code>docs/catalog_gwas_pliercanon_sep_ldclumping_r01d1000_adipose_sub.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/a9f3fb97cfa9e3cb5158aa103a8a1b7231bb9346/docs/catalog_gwas_pliercanon_sep_ldclumping_r01d1000_adipose_sub.html" target="_blank">a9f3fb9</a>
</td>
<td>
XSun
</td>
<td>
2021-02-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/e4bf324e1c0756e1152921d7fd8d6a6ae7fa78f9/analysis/catalog_gwas_pliercanon_sep_ldclumping_r01d1000_adipose_sub.Rmd" target="_blank">e4bf324</a>
</td>
<td>
XSun
</td>
<td>
2021-02-01
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/d33bc448bffe130c5e3bbbed58a611ee175096b9/docs/catalog_gwas_pliercanon_sep_ldclumping_r01d1000_adipose_sub.html" target="_blank">d33bc44</a>
</td>
<td>
XSun
</td>
<td>
2020-11-30
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/c7567a36e47a4ab26fd8503e037c67b67f397e88/analysis/catalog_gwas_pliercanon_sep_ldclumping_r01d1000_adipose_sub.Rmd" target="_blank">c7567a3</a>
</td>
<td>
XSun
</td>
<td>
2020-11-30
</td>
<td>
wflow_publish(c(“analysis/catalog_gwas_pliercanon_sep_ldclumping_r01d1000_adipose_sub.Rmd”,
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/c02b975d21c7343d1409b276d8f0c359ab3af9b4/docs/catalog_gwas_pliercanon_sep_ldclumping_r01d1000_adipose_sub.html" target="_blank">c02b975</a>
</td>
<td>
XSun
</td>
<td>
2020-11-30
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/dd8cb608277db75581b7e3b262b0cde43b916f9a/docs/catalog_gwas_pliercanon_sep_ldclumping_r01d1000_adipose_sub.html" target="_blank">dd8cb60</a>
</td>
<td>
XSun
</td>
<td>
2020-11-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/da041554a6012b5c861a77282b51b53bb7178474/analysis/catalog_gwas_pliercanon_sep_ldclumping_r01d1000_adipose_sub.Rmd" target="_blank">da04155</a>
</td>
<td>
XSun
</td>
<td>
2020-11-26
</td>
<td>
wflow_publish(“analysis/catalog_gwas_pliercanon_sep_ldclumping_r01d1000_adipose_sub.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/db5a9a37145997b1280feeca79f7e33d464d952d/docs/catalog_gwas_pliercanon_sep_ldclumping_r01d1000_adipose_sub.html" target="_blank">db5a9a3</a>
</td>
<td>
XSun
</td>
<td>
2020-11-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/884f405edea63d324984297da9620801c4b90a15/analysis/catalog_gwas_pliercanon_sep_ldclumping_r01d1000_adipose_sub.Rmd" target="_blank">884f405</a>
</td>
<td>
XSun
</td>
<td>
2020-11-26
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/931a5146627601ee97988273be0c15b1d1d36e54/docs/catalog_gwas_pliercanon_sep_ldclumping_r01d1000_adipose_sub.html" target="_blank">931a514</a>
</td>
<td>
XSun
</td>
<td>
2020-11-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/e2a4aac638bd0e67d3426450ddebbd2e1f4eb3cd/analysis/catalog_gwas_pliercanon_sep_ldclumping_r01d1000_adipose_sub.Rmd" target="_blank">e2a4aac</a>
</td>
<td>
XSun
</td>
<td>
2020-11-24
</td>
<td>
wflow_publish(c(“analysis/catalog_gwas_flash_sep_ldclumping_r01d1000.Rmd”,
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/77d4653edf35b39c04d2b7293c15335fbf38f261/docs/catalog_gwas_pliercanon_sep_ldclumping_r01d1000_adipose_sub.html" target="_blank">77d4653</a>
</td>
<td>
XSun
</td>
<td>
2020-11-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/675bc6be224322408210c7db114a893fb26fb4f6/analysis/catalog_gwas_pliercanon_sep_ldclumping_r01d1000_adipose_sub.Rmd" target="_blank">675bc6b</a>
</td>
<td>
XSun
</td>
<td>
2020-11-24
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/691daa9aa0fd4e3ed192b6cd4ddabb5e65bbc3c3/analysis/catalog_gwas_pliercanon_sep_ldclumping_r01d1000_adipose_sub.Rmd" target="_blank">691daa9</a>
</td>
<td>
XSun
</td>
<td>
2020-11-23
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/1e9dc8a1391b07d42b092404ac4c8fd47f37ca02/docs/catalog_gwas_pliercanon_sep_ldclumping_r01d1000_adipose_sub.html" target="_blank">1e9dc8a</a>
</td>
<td>
XSun
</td>
<td>
2020-11-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/2fb7f998e6338956e0788c908b5d633db6d33b01/analysis/catalog_gwas_pliercanon_sep_ldclumping_r01d1000_adipose_sub.Rmd" target="_blank">2fb7f99</a>
</td>
<td>
XSun
</td>
<td>
2020-11-21
</td>
<td>
wflow_publish(“analysis/catalog_gwas_pliercanon_sep_ldclumping_r01d1000_adipose_sub.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/f17379876f34c4424b8ecb534e8d3e784f32412e/docs/catalog_gwas_pliercanon_sep_ldclumping_r01d1000_adipose_sub.html" target="_blank">f173798</a>
</td>
<td>
XSun
</td>
<td>
2020-11-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/d3741d1fede5b5527f0a3badbaf3459831b8f487/analysis/catalog_gwas_pliercanon_sep_ldclumping_r01d1000_adipose_sub.Rmd" target="_blank">d3741d1</a>
</td>
<td>
XSun
</td>
<td>
2020-11-20
</td>
<td>
wflow_publish(c(“analysis/catalog_gwas_flash_sep_ldclumping_r01d1000.Rmd”,
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/4f8ac40ea29af865346c022a54c1bc7dba66147a/docs/catalog_gwas_pliercanon_sep_ldclumping_r01d1000_adipose_sub.html" target="_blank">4f8ac40</a>
</td>
<td>
XSun
</td>
<td>
2020-11-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/5afff748bd3d9dc7637531d171e8022249bd1651/analysis/catalog_gwas_pliercanon_sep_ldclumping_r01d1000_adipose_sub.Rmd" target="_blank">5afff74</a>
</td>
<td>
XSun
</td>
<td>
2020-11-20
</td>
<td>
wflow_publish(“analysis/catalog_gwas_pliercanon_sep_ldclumping_r01d1000_adipose_sub.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/dd6f247fbff84a909338cda8257242911dbc910d/docs/catalog_gwas_pliercanon_sep_ldclumping_r01d1000_adipose_sub.html" target="_blank">dd6f247</a>
</td>
<td>
XSun
</td>
<td>
2020-11-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/12d9814bcf0026b0ca1ca76842abc0cd50e2f9c9/analysis/catalog_gwas_pliercanon_sep_ldclumping_r01d1000_adipose_sub.Rmd" target="_blank">12d9814</a>
</td>
<td>
XSun
</td>
<td>
2020-11-19
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/811c970c75e3f805268950ac36d7d776657ae17f/docs/catalog_gwas_pliercanon_sep_ldclumping_r01d1000_adipose_sub.html" target="_blank">811c970</a>
</td>
<td>
XSun
</td>
<td>
2020-11-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/e947318db6d17651b1bc7b4cc8a8d48400a111d7/analysis/catalog_gwas_pliercanon_sep_ldclumping_r01d1000_adipose_sub.Rmd" target="_blank">e947318</a>
</td>
<td>
XSun
</td>
<td>
2020-11-19
</td>
<td>
update
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>The analysis in this part is the same with <a href="https://xsun1229.github.io/factor_analysis/catalog_gwas_pliercanon_sep_ldclumping_r01d1000.html">PLIER-Canonical</a>. But in this part, I used Adipose Subcutaneous expression data. It contains 581 samples. I used PLIER with Canonical Pathway and inferred 142 LVs first. And used these 142 LVs to do the following analysis.</p>
</div>
<div id="material-and-methods" class="section level2">
<h2>Material and Methods</h2>
<p>This part is the same with <a href="https://xsun1229.github.io/factor_analysis/catalog_gwas_pliercanon_sep_ldclumping_r01d1000.html">PLIER-Canonical</a>.</p>
</div>
<div id="snps-after-filter" class="section level2">
<h2>Snps after filter</h2>
<p>After filtering by ‘pval &lt; 5e-8’ and LD Clumping, for each trait, I got :</p>
<p>BMI trait contains 102 SNPs with pval&lt;5e-8.</p>
<p>T1D trait contains 65 SNPs with pval&lt;5e-8</p>
<p>T2D trait contains 14 SNPs with pval&lt;5e-8. T2D_2 contains 4 SNPs with pval&lt;5e-8.</p>
<p>TG trait contains 152 SNPs with pval&lt;5e-8</p>
<p>TC trait contains 254 SNPs with pval&lt;5e-8</p>
<p>HDL trait contains 225 SNPs with pval&lt;5e-8.</p>
<p>LDL trait contains 203 SNPs with pval&lt;5e-8.</p>
<p>WHR trait contains 37 SNPs with pval&lt;5e-8.</p>
</div>
<div id="results---pval-5e-8-120-peer-factors-covariants-10-pcs-gtexsequencing-platformsequencing-protocolsex-age" class="section level2">
<h2>Results - pval &lt; 5e-8 &amp; 120 PEER factors &amp; covariants: 10 PCs + GTEx:Sequencing platform,Sequencing protocol,Sex + AGE</h2>
<div id="summary-table" class="section level3">
<h3>Summary table</h3>
<p>I used ‘qvalue’ R package to compute the fdr from p-values for each SNP and made a table to show the number of SNPs that pass the threshold. The thresholds are ‘fdr &lt; 0.1’,‘fdr &lt; 0.2’,‘pval &lt; 5e-8’. The ‘num_significant_pairs’ indicates the number of significant pairs under each threshold. If a trait~factor pair has as least 1 significant SNP, we named it as ‘significant pair’.</p>
<table>
<caption>number of significant trait-LV associations across all traits</caption>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">x</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">num_sigpairs_fdr010</td>
<td align="right">37</td>
</tr>
<tr class="even">
<td align="left">num_sigpairs_fdr020</td>
<td align="right">113</td>
</tr>
</tbody>
</table>
<div id="htmlwidget-c3f4891eb8e9a8c25198" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-c3f4891eb8e9a8c25198">{"x":{"filter":"none","data":[["num_significant_pairs","LV1","LV2","LV3","LV4","LV5","LV6","LV7","LV8","LV9","LV10","LV11","LV12","LV13","LV14","LV15","LV16","LV17","LV18","LV19","LV20","LV21","LV22","LV23","LV24","LV25","LV26","LV27","LV28","LV29","LV30","LV31","LV32","LV33","LV34","LV35","LV36","LV37","LV38","LV39","LV40","LV41","LV42","LV43","LV44","LV45","LV46","LV47","LV48","LV49","LV50","LV51","LV52","LV53","LV54","LV55","LV56","LV57","LV58","LV59","LV60","LV61","LV62","LV63","LV64","LV65","LV66","LV67","LV68","LV69","LV70","LV71","LV72","LV73","LV74","LV75","LV76","LV77","LV78","LV79","LV80","LV81","LV82","LV83","LV84","LV85","LV86","LV87","LV88","LV89","LV90","LV91","LV92","LV93","LV94","LV95","LV96","LV97","LV98","LV99","LV100","LV101","LV102","LV103","LV104","LV105","LV106","LV107","LV108","LV109","LV110","LV111","LV112","LV113","LV114","LV115","LV116","LV117","LV118","LV119","LV120","LV121","LV122","LV123","LV124","LV125","LV126","LV127","LV128","LV129","LV130","LV131","LV132","LV133","LV134","LV135","LV136","LV137","LV138","LV139","LV140","LV141","LV142"],[3,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,2,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[11,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,1,0,1,1,0,1,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,2,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,2,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[4,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1],[11,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,2,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,2,0,0,0,0,0,0,0,0,0,0,1,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,1,0,1,0,0,0,0,1],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[2,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[5,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[15,1,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,1,0,0,1,0,0,0,1,0,0,0,0,0,0,0,1,0,1,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[4,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,1,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[12,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,1,0,0,1,0,0,0,1,0,0,0,0,0,1,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>bmi_fdr010_num<\/th>\n      <th>bmi_fdr020_num<\/th>\n      <th>bmi_pval5e8_num<\/th>\n      <th>HDL_fdr010_num<\/th>\n      <th>HDL_fdr020_num<\/th>\n      <th>HDL_pval5e8_num<\/th>\n      <th>LDL_fdr010_num<\/th>\n      <th>LDL_fdr020_num<\/th>\n      <th>LDL_pval5e8_num<\/th>\n      <th>T2D_2_fdr010_num<\/th>\n      <th>T2D_2_fdr020_num<\/th>\n      <th>T2D_2_pval5e8_num<\/th>\n      <th>T2D_fdr010_num<\/th>\n      <th>T2D_fdr020_num<\/th>\n      <th>T2D_pval5e8_num<\/th>\n      <th>WHR_fdr010_num<\/th>\n      <th>WHR_fdr020_num<\/th>\n      <th>WHR_pval5e8_num<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":20,"columnDefs":[{"className":"dt-right","targets":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[10,20,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="info-table" class="section level3">
<h3>Info table</h3>
<p>For each trait, I made a table to show the info of snps with fdr&gt;0.2 in the factor ~ SNP + genotype pcs association test. For each trait,The LVs have more than one significant SNPs with FDR&lt;0.2 are included.</p>
<p>The suffix ’_assoc’ here means that results are from factor ~ SNP + genotype pcs association test. The suffix ’_gwas’ here means results are from original GWAS results files.</p>
<p>‘snp_ld’ here means the snps that in LD with the snp in each line.’ld_r2’ means the LD r-squared which is corresponding to the ‘snp_ld’ column. ‘cis-eqtl’ column indicates whether the snp is a cis-eqtl according to GTEx data. ‘cis_gene_hgnc’ and ‘cis_gene_hgnc’ is the genes that the snp influence when it act as cis-eqtl. ‘func’ and ‘func_gene’ are obtained from ANNOVAR, which indicating the snp function within the genes.</p>
<div id="htmlwidget-cfefb73e51b4928e765c" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-cfefb73e51b4928e765c">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23"],["bmi","bmi","bmi","bmi","HDL","HDL","HDL","HDL","HDL","HDL","LDL","LDL","LDL","LDL","T1D","T1D","T1D","T1D","T1D","TC","TC","TG","TG"],["LV78","LV78","LV100","LV100","LV55","LV55","LV105","LV105","LV126","LV126","LV35","LV35","LV94","LV94","LV17","LV17","LV17","LV86","LV86","LV133","LV133","LV6","LV6"],["chr15_67784830_T_C_b38","chr16_53769662_T_A_b38","chr15_67784830_T_C_b38","chr16_53769662_T_A_b38","chr11_116634723_G_A_b38","chr20_44413724_C_T_b38","chr11_48324650_G_T_b38","chr11_49593964_A_C_b38","chr11_48324650_G_T_b38","chr11_49593964_A_C_b38","chr19_45149969_G_A_b38","chr2_20816003_G_A_b38","chr5_75360714_T_C_b38","chr5_75676165_C_G_b38","chr1_113468684_G_A_b38","chr16_11257354_A_G_b38","chr6_90267049_G_A_b38","chr1_113302202_A_C_b38","chr11_2150692_T_C_b38","chr1_55030366_T_C_b38","chr15_58399009_C_T_b38","chr19_7224420_G_A_b38","chr8_20084670_G_T_b38"],["rs16951275","rs1558902","rs16951275","rs1558902","rs10892001","rs1800961","rs7949865","rs7925397","rs7949865","rs7925397","rs8108762","rs492399","rs12916","rs6892146","rs72687939","rs193778","rs72928038","rs773588","rs10840442","rs11206510","rs74019428","rs7248104","rs13263508"],["-0.0311","0.0818","-0.0311","0.0818","-0.0211","-0.127","0.045","0.0373","0.045","0.0373","-0.0394","0.0629","0.0733","0.0398","0.179985496996407","0.131028262406404","0.179985496996407","0.193014186332698","0.152978564597081","-0.069","-0.0805","-0.0222","0.0517"],["1.911e-17","7.51e-153","1.911e-17","7.51e-153","1.37e-08","1.639e-34","1.37e-18","1.393e-10","1.37e-18","1.393e-10","1.45e-20","1.225e-09","7.792e-78","2.548e-15","3.01062798111745e-13","2.21465488778786e-10","3.1908916729109e-14","2.45778648347229e-22","1.44397578043858e-11","1.128e-41","2.007e-08","5.045e-10","3.383e-48"],["-0.06648","0.06081","0.06207","-0.06394","-0.03541","0.1032","-0.08149","-0.08282","0.1349","0.1334","-0.1641","0.2967","-0.06983","-0.08561","0.1051","-0.08563","0.1005","-0.03247","0.04394","0.07139","0.157","0.04232","-0.04289"],["0.0002976","0.0004996","0.00148","0.0005639","0.0007835","0.0007917","0.0003666","0.0003233","0.0001998","0.0002473","3.035e-05","0.0003612","0.0003309","0.0004296","0.002311","0.002496","0.002729","0.002749","0.0009077","0.0002525","4.542e-05","0.000859","0.00122"],["0.0545969610896357","0.0813064186983676","0.178252723608334","0.101526585486637","0.180028721841158","0.181028614482801","0.147350060945609","0.133509047454052","0.0714465938205948","0.0814629433854559","0.0222875511491437","0.178482119070835","0.140783528832845","0.174500282001957","0.169807186796223","0.17943378330878","0.191532474259917","0.19716804113426","0.11054962012019","0.155326456158451","0.033562839409675","0.111499529165682","0.133014165206592"],[null,null,null,null,null,null,"chr11_49593964_A_C_b38","chr11_48324650_G_T_b38","chr11_49593964_A_C_b38","chr11_48324650_G_T_b38",null,null,"chr5_75676165_C_G_b38","chr5_75360714_T_C_b38",null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,"0.812851","0.812851","0.812851","0.812851",null,null,"0.0895167","0.0895167",null,null,null,null,null,null,null,null,null],["T",null,"T",null,null,null,"T",null,"T",null,null,null,null,"T",null,null,null,null,null,null,null,null,null],["IQCH,MAP2K5,SKOR1",null,"IQCH,MAP2K5,SKOR1",null,null,null,"PTPRJ,C1QTNF4",null,"PTPRJ,C1QTNF4",null,null,null,null,"POC5,ANKDD1B",null,null,null,null,null,null,null,null,null],["ENSG00000103599,ENSG00000137764,ENSG00000188779",null,"ENSG00000103599,ENSG00000137764,ENSG00000188779",null,null,null,"ENSG00000149177,ENSG00000172247",null,"ENSG00000149177,ENSG00000172247",null,null,null,null,"ENSG00000152359,ENSG00000189045",null,null,null,null,null,null,null,null,null],["intronic","intronic","intronic","intronic","intergenic","exonic","upstream","ncRNA_intronic","upstream","ncRNA_intronic","UTR3","intronic","UTR3","intronic","intronic","intergenic","intronic","intergenic","intronic","intergenic","intergenic","intronic","intergenic"],["MAP2K5","FTO","MAP2K5","FTO","LINC00900(dist=874523),LINC02702(dist=4699)","HNF4A","OR4C3(dist=270)","LOC440040","OR4C3(dist=270)","LOC440040","NKPD1(NM_198478:c.*1969C&gt;T)","LDAH","HMGCR(NM_001130996:c.*372T&gt;C,NM_001364187:c.*372T&gt;C,NM_000859:c.*372T&gt;C)","POC5","MAGI3","SOCS1(dist=1172),TNP2(dist=10503)","BACH2","LOC643441(dist=95949),MAGI3(dist=88313)","INS-IGF2","BSND(dist=13194),PCSK9(dist=9110)","AQP9(dist=213098),LIPC(dist=32982)","INSR","LPL(dist=117411),SLC18A1(dist=60185)"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Traits<\/th>\n      <th>factors<\/th>\n      <th>SNP_gtexid<\/th>\n      <th>SNP_rsid<\/th>\n      <th>BETA_gwas<\/th>\n      <th>P_gwas<\/th>\n      <th>BETA_assoc<\/th>\n      <th>P_assoc<\/th>\n      <th>fdr_assoc<\/th>\n      <th>snp_ld<\/th>\n      <th>ld_r2<\/th>\n      <th>cis_eqtl<\/th>\n      <th>cis_genes_hgnc<\/th>\n      <th>cis_genes_ensembl<\/th>\n      <th>func<\/th>\n      <th>func_gene<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":20,"order":[],"autoWidth":false,"orderClasses":false,"columnDefs":[{"orderable":false,"targets":0}],"lengthMenu":[10,20,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="effectsize-summary-26-traits" class="section level3">
<h3>Effectsize summary (26 traits)</h3>
<p>To summarize the results above, I made a scatter plot of all trait-LV pairs. The x-axis shows the R2 in fitting, and the y-axis show the -log10-p values from resampling. Below are plots for three FDR cutoffs, 0.2, 0.3 and 0.5. To make the plot more informative, Each point represent a trait-LV pair. Different colors show the number of SNPs passing the FDR threshold for each pair.</p>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsizesum_0.2_adp.jpeg" width="60%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsizesum_0.2_adp.jpeg" width="60%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsizesum_0.2_adp.jpeg" width="60%">
</figure>
<p><a href="https://drive.google.com/file/d/1TdP2HHARmdjqlFy1nix2M079UDbcJNvm/view?usp=sharing">Summary Table</a></p>
</div>
<div id="colocalization-1000kb" class="section level3">
<h3>Colocalization – 1000kb</h3>
<p>The colocalization analysis was performed using the approximate Bayes factor test implemented in the Coloc package. Coloc computes five posterior probabilities (PP0, PP1, PP2, PP3 and PP4), each corresponding to a hypothesis: H0, no association with either trait; H1, association with trait 1 but not with trait 2; H2, association with trait 2 but not with trait 1; H3, association with trait 1 and trait 2, two independent SNPs; H4, association with trait 1 and trait 2, one shared SNP. We ran Coloc with the default parameters and used PP4 to assess evidence of colocalization. We visualized the colocalization of factor - QTLs and GWAS associations using the <a href="https://github.com/boxiangliu/locuscomparer">LocusCompareR package</a>.</p>
<p>SNP selection: 1. Chose the SNPs in the info table. 2. For each SNP, the region used in colocalization analysis is between [pos-1000kb, pos+1000kb]. 3. All SNPs in this region are included in alalysis.</p>
<p><a href="https://drive.google.com/file/d/1-v6P5PDzzo-qnUln8Oy0CATDWz3-xGED/view?usp=sharing">Summary Table</a></p>
</div>
</div>
<div id="analysis-based-on-em-algorithm" class="section level2">
<h2>Analysis based on EM algorithm</h2>
<div id="em-algorithm-summary-table" class="section level3">
<h3>EM algorithm summary table</h3>
<div id="htmlwidget-5ba5287ce9a3b08bdb5d" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-5ba5287ce9a3b08bdb5d">{"x":{"filter":"none","data":[["bmi_lv100","bmi_lv78","HDL_lv105","HDL_lv126","HDL_lv55","LDL_lv35","LDL_lv94","T1D_lv17","T1D_lv86","TC_lv133","TG_lv6"],[-0.505215625241871,0.829804623164148,-0.260293143034486,0.141507652940162,-0.521804090495213,0.0352358559031186,-0.750516851559926,0.809157688680764,-1.50330924324739,-0.649217307685811,-0.563394234763715],[-0.547889306534834,0.963167582313075,-0.253548303071626,0.132068117402819,-0.564079420047567,0.0385215080459838,-1.05623678041828,0.765017584049116,-1.38732995741396,-0.659556042679222,-0.579175174643761],[0.0330073324517239,0.0330073324517239,0.0591240322485987,0.0591240322485987,0.0591240322485987,0.0891284228414716,0.0891284228414716,0.251551837529763,0.251551837529763,0.0643598359172237,0.0581475643212667],[-0.0260525323029785,-0.0260525323029785,0.0495761071104162,0.0495761071104162,0.0495761071104162,0.0777012676893654,0.0777012676893654,0.207730352825048,0.207730352825048,0.0541293534398118,0.0500215911897451],[5,9,5,5,5,5,8,6,6,4,4],[0.199802936763772,0.226236609690972,1,1,0.183225815551747,1,0.00902212905980663,1,0.732126140364768,1,1],[1.64381427482067,1.46436669638433,-0.855628122093322,-0.16902217452315,1.77126327908695,-0.367296775583782,6.81844105520031,-0.293129885481875,0.117168375855272,-1.69822040477447,-1.6989080516455],[0.068,0.065,0.528,0.598,0.082,0.527,0.073,0.431,0.366,0.46046046046046,0.603]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>gamma_initial<\/th>\n      <th>gamma_final<\/th>\n      <th>sigma_init<\/th>\n      <th>sigma_final<\/th>\n      <th>iteration_times<\/th>\n      <th>LRT_p<\/th>\n      <th>LRT_stat<\/th>\n      <th>pval_resampling<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":20,"columnDefs":[{"className":"dt-right","targets":[1,2,3,4,5,6,7,8]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[10,20,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div id="resampling-summary-table-summary-of-the-above-analyses" class="section level2">
<h2>Resampling summary table (summary of the above analyses)</h2>
<div id="htmlwidget-0b516daec537d8efb4b6" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-0b516daec537d8efb4b6">{"x":{"filter":"none","data":[["bmi_lv100","bmi_lv78","HDL_lv105","HDL_lv126","HDL_lv55","LDL_lv35","LDL_lv94","T1D_lv17","T1D_lv86","TC_lv133","TG_lv6"],[2,2,2,2,2,2,2,3,2,2,2],[0.442,0.462,0.109,0.1,0.297,0.053,0.331,0.36,0.955,0.186,0.324],[0.199802936763772,0.226236609690972,1,1,0.183225815551747,1,0.00902212905980663,1,0.732126140364768,1,1],[0.068,0.065,0.528,0.598,0.082,0.527,0.073,0.431,0.366,0.46046046046046,0.603]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>num_snp_fdr02<\/th>\n      <th>pval_resampling<\/th>\n      <th>pval_EM<\/th>\n      <th>pval_EM_resampling<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":20,"columnDefs":[{"className":"dt-right","targets":[1,2,3,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[10,20,25,50,100]}},"evals":[],"jsHooks":[]}</script>
<!-- ### Enrichment analysis -->
<!-- There is only one promising pair: LDL-LV35 -->
<!-- **LDL - LV35** -->
<!-- ```{r echo=FALSE} -->
<!-- load("/project2/xinhe/xsun/website/factor_analysis/output/lv35top025_pliercanon_adp.rdata") -->
<!-- DT::datatable(ORA_results, options = list(pageLength =5)) -->
<!-- ``` -->
<!-- ### Effect size plots & Resampling -->
<!-- For each trait, I made a plot of association with LV(indicating by beta in GWAS) vs association with trait(indicating by ln(odds ratio) or beta in GWAS) to show if the variants have the correlated effect direction. The effect sizes of Catalog GWAS and factor association tests are harmonized by [TwoSampleMR](https://mrcieu.github.io/TwoSampleMR/articles/introduction.html) R package to make the effect alleles in these two analysis identical. The LVs have more than one significant SNPs with FDR<0.2 are included in the plotting.Besides, for each plots, I fitted the points with intercept = 0. The pvalues and r-squared are shown on the plots. -->
<!-- I also relaxed the fdr threshold of the SNPs that used to make effect size plots(from 0.2 to 0.3/0.5) -->
<!-- For all pairs , I did resampling. I resampled the SNPs without replacement and fitted the points with intercept = 0 again and recorded the pvalues and r-squared. The resampling was repeated 1000 times. -->
<!-- I made a histogram to show the rsquared distribution from resampling. The red line in the plots are the rsquared in the origin analysis. The r_mean values in the plots are the mean values of rsquared in point fitting. The 'p-value from resampling' is computed by: (number of more extreme values)/(times of resampling). -->
<!-- #### BMI  -->
<!-- LV78 -->
<!-- fdr0.2 -->
<!-- <figure class="half"> -->
<!--     <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv78_bmi_pval5e8_pliercanon_d1k_ageco_adp.jpeg" width="30%"> -->
<!--     <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_78_bmi_adp.jpeg" width="30%"> -->
<!-- </figure> -->
<!-- p-value from resampling: 0.336 -->
<!-- fdr0.3  -->
<!-- <figure class="half"> -->
<!--     <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv78_bmi_pval5e8_pliercanon_d1k_ageco_fdr03_adp.jpeg" width="30%"> -->
<!--     <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_78_bmi_fdr03_adp.jpeg" width="30%"> -->
<!-- </figure> -->
<!-- p-value from resampling: 0.336 -->
<!-- fdr0.5 -->
<!-- <figure class="half"> -->
<!--     <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv78_bmi_pval5e8_pliercanon_d1k_ageco_fdr05_adp.jpeg" width="30%"> -->
<!--     <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_78_bmi_fdr05_adp.jpeg" width="30%"> -->
<!-- </figure> -->
<!-- p-value from resampling: 0.432 -->
<!-- LV100 -->
<!-- fdr0.2 -->
<!-- <figure class="half"> -->
<!--     <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv100_bmi_pval5e8_pliercanon_d1k_ageco_adp.jpeg" width="30%"> -->
<!--     <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_100_bmi_adp.jpeg" width="30%"> -->
<!-- </figure> -->
<!-- p-value from resampling: 0.305 -->
<!-- fdr0.3  -->
<!-- <figure class="half"> -->
<!--     <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv100_bmi_pval5e8_pliercanon_d1k_ageco_fdr03_adp.jpeg" width="30%"> -->
<!--     <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_100_bmi_fdr03_adp.jpeg" width="30%"> -->
<!-- </figure> -->
<!-- p-value from resampling: 0.305 -->
<!-- fdr0.5 -->
<!-- <figure class="half"> -->
<!--     <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv100_bmi_pval5e8_pliercanon_d1k_ageco_fdr05_adp.jpeg" width="30%"> -->
<!--     <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_100_bmi_fdr05_adp.jpeg" width="30%"> -->
<!-- </figure> -->
<!-- p-value from resampling: 0.607 -->
<!-- #### HDL -->
<!-- None of the LVs have >1 SNPs at FDR<0.2. -->
<!-- #### LDL -->
<!-- LV35 -->
<!-- fdr0.2 -->
<!-- <figure class="half"> -->
<!--     <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv35_LDL_pval5e8_pliercanon_d1k_ageco_adp.jpeg" width="30%"> -->
<!--     <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_35_LDL_adp.jpeg" width="30%"> -->
<!-- </figure> -->
<!-- p-value from resampling: 0.037 -->
<!-- fdr0.3  -->
<!-- <figure class="half"> -->
<!--     <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv35_LDL_pval5e8_pliercanon_d1k_ageco_fdr03_adp.jpeg" width="30%"> -->
<!--     <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_35_LDL_fdr03_adp.jpeg" width="30%"> -->
<!-- </figure> -->
<!-- p-value from resampling: 0.037 -->
<!-- fdr0.5 -->
<!-- <figure class="half"> -->
<!--     <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv35_LDL_pval5e8_pliercanon_d1k_ageco_fdr05_adp.jpeg" width="30%"> -->
<!--     <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_35_LDL_fdr05_adp.jpeg" width="30%"> -->
<!-- </figure> -->
<!-- p-value from resampling: 0.037 -->
<!-- LV94 -->
<!-- fdr0.2 -->
<!-- <figure class="half"> -->
<!--     <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv94_LDL_pval5e8_pliercanon_d1k_ageco_adp.jpeg" width="30%"> -->
<!--     <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_94_LDL_adp.jpeg" width="30%"> -->
<!-- </figure> -->
<!-- p-value from resampling: 0.295 -->
<!-- fdr0.3  -->
<!-- <figure class="half"> -->
<!--     <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv94_LDL_pval5e8_pliercanon_d1k_ageco_fdr03_adp.jpeg" width="30%"> -->
<!--     <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_94_LDL_fdr03_adp.jpeg" width="30%"> -->
<!-- </figure> -->
<!-- p-value from resampling: 0.295 -->
<!-- fdr0.5 -->
<!-- <figure class="half"> -->
<!--     <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv94_LDL_pval5e8_pliercanon_d1k_ageco_fdr05_adp.jpeg" width="30%"> -->
<!--     <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_94_LDL_fdr05_adp.jpeg" width="30%"> -->
<!-- </figure> -->
<!-- p-value from resampling: 0.295 -->
<!-- #### T2D_2 -->
<!-- None of the LVs have >1 SNPs at FDR<0.2. -->
<!-- #### T2D -->
<!-- None of the LVs have >1 SNPs at FDR<0.2. -->
<!-- #### WHR -->
<!-- None of the LVs have >1 SNPs at FDR<0.2. -->
<!-- #### LDL -->
<!-- <figure class="half"> -->
<!--     <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv35_LDL_pval5e8_pliercanon_d1k_ageco_reverse_adp.jpeg" width="30%"> -->
<!--     <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv35_LDL_pval5e8_pliercanon_d1k_ageco_reverse_remove_adp.jpeg" width="30%"> -->
<!-- </figure> -->
<!-- ### QQplots - promising pairs -->
<!-- #### LDL -->
<!-- LV35 -->
<!-- <figure class="half"> -->
<!--     <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qq_unif_lv35_LDL_pliercanon_adp.jpeg" width="30%"> -->
<!-- </figure> -->
<!-- ### Colocalization -- 500kb -->
<!-- The colocalization analysis was performed using the approximate Bayes factor test implemented in the Coloc package. Coloc computes five posterior probabilities (PP0, PP1, PP2, PP3 and PP4), each corresponding to a hypothesis: H0, no association with either trait; H1, association with trait 1 but not with trait 2; H2, association with trait 2 but not with trait 1; H3, association with trait 1 and trait 2, two independent SNPs; H4, association with trait 1 and trait 2, one shared SNP. We ran Coloc with the default parameters and used PP4 to assess evidence of colocalization. We visualized the colocalization of factor - QTLs and GWAS associations using the [LocusCompareR package](https://github.com/boxiangliu/locuscomparer). -->
<!-- Chromosome selection: I first sorted the SNPs by their pvalues from factor association tests, then I selected the first two SNPs and chose their chromosome to do the colocalization test adn visualization.  -->
<!-- #### LDL -->
<!-- lv35 -->
<!-- ```{r echo=FALSE} -->
<!-- load("/project2/xinhe/xsun/website/factor_analysis/output/LDL_35_20816003.coloc.results_adp_500.rdata") -->
<!-- note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP") -->
<!-- pval <- cbind(results$summary,note) -->
<!-- knitr::kable(pval, caption = "pvalues in coloclization") -->
<!-- ``` -->
<!-- <figure class="half"> -->
<!--     <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/LDL_35_20816003.coloc_adp_500.jpeg" width="80%"> -->
<!-- </figure> -->
<!-- ```{r echo=FALSE} -->
<!-- load("/project2/xinhe/xsun/website/factor_analysis/output/LDL_35_45149969.coloc.results_adp_500.rdata") -->
<!-- note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP") -->
<!-- pval <- cbind(results$summary,note) -->
<!-- knitr::kable(pval, caption = "pvalues in coloclization") -->
<!-- ``` -->
<!-- <figure class="half"> -->
<!--     <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/LDL_35_45149969.coloc_adp_500.jpeg" width="80%"> -->
<!-- </figure> -->
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] workflowr_1.6.2

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.5        rstudioapi_0.11   whisker_0.3-2     knitr_1.30       
 [5] magrittr_1.5      R6_2.4.1          rlang_0.4.8       highr_0.8        
 [9] stringr_1.4.0     tools_3.6.1       DT_0.15           xfun_0.18        
[13] git2r_0.26.1      crosstalk_1.1.0.1 htmltools_0.5.0   ellipsis_0.3.1   
[17] rprojroot_1.3-2   yaml_2.2.1        digest_0.6.25     tibble_3.0.3     
[21] lifecycle_0.2.0   crayon_1.3.4      later_1.1.0.1     htmlwidgets_1.5.2
[25] vctrs_0.3.4       promises_1.1.1    fs_1.5.0          glue_1.4.2       
[29] evaluate_0.14     rmarkdown_1.13    stringi_1.5.3     compiler_3.6.1   
[33] pillar_1.4.6      backports_1.1.10  jsonlite_1.7.1    httpuv_1.5.1     
[37] pkgconfig_2.0.3  </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
