<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="XSun" />

<meta name="date" content="2020-10-19" />

<title>catalog_gwas_peer_sep_ldclumping_r01d1000</title>

<script src="site_libs/jquery-1.12.4/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.2/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.15/datatables.js"></script>
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.1.0.1/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">factor_analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/xsun1229/factor_analysis">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">catalog_gwas_peer_sep_ldclumping_r01d1000</h1>
<h4 class="author">XSun</h4>
<h4 class="date">2020-10-19</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-11-20
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>factor_analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div class="panel-group" id="workflowr-checks">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200623code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200623)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200623code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200623)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it difficult for you and others to run your code on a different machine. Change the absolute path(s) below to the suggested relative path(s) to make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/factor_analysis/output/sum_sep_tog_pval_peer_wgpc_ldcl_d1k_ageco.rdata
</td>
<td style="text-align:left;">
output/sum_sep_tog_pval_peer_wgpc_ldcl_d1k_ageco.rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/factor_analysis/output/info_pval5e8_peer_ld_d1k_ageco.rdata
</td>
<td style="text-align:left;">
output/info_pval5e8_peer_ld_d1k_ageco.rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/factor_analysis/output/asthma_2_50_5866925.coloc.results_peer.rdata
</td>
<td style="text-align:left;">
output/asthma_2_50_5866925.coloc.results_peer.rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/factor_analysis/output/asthma_2_50_80386214.coloc.results_peer.rdata
</td>
<td style="text-align:left;">
output/asthma_2_50_80386214.coloc.results_peer.rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/factor_analysis/output/bmi_14_18344015.coloc.results_peer.rdata
</td>
<td style="text-align:left;">
output/bmi_14_18344015.coloc.results_peer.rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/factor_analysis/output/bmi_14_85758440.coloc.results_peer.rdata
</td>
<td style="text-align:left;">
output/bmi_14_85758440.coloc.results_peer.rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/factor_analysis/output/EUR.UC_77_32483933.coloc.results_peer.rdata
</td>
<td style="text-align:left;">
output/EUR.UC_77_32483933.coloc.results_peer.rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/factor_analysis/output/EUR.UC_77_32636617.coloc.results_peer.rdata
</td>
<td style="text-align:left;">
output/EUR.UC_77_32636617.coloc.results_peer.rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/factor_analysis/output/lymph_66_137031531.coloc.results_peer.rdata
</td>
<td style="text-align:left;">
output/lymph_66_137031531.coloc.results_peer.rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/factor_analysis/output/lymph_66_29892532.coloc.results_peer.rdata
</td>
<td style="text-align:left;">
output/lymph_66_29892532.coloc.results_peer.rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/factor_analysis/output/lymph_76_112677823.coloc.results_peer.rdata
</td>
<td style="text-align:left;">
output/lymph_76_112677823.coloc.results_peer.rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/factor_analysis/output/lymph_76_62808178.coloc.results_peer.rdata
</td>
<td style="text-align:left;">
output/lymph_76_62808178.coloc.results_peer.rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/factor_analysis/output/rbc_73_100749368.coloc.results_peer.rdata
</td>
<td style="text-align:left;">
output/rbc_73_100749368.coloc.results_peer.rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/factor_analysis/output/rbc_73_100933277.coloc.results_peer.rdata
</td>
<td style="text-align:left;">
output/rbc_73_100933277.coloc.results_peer.rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/factor_analysis/output/wbc_11_102727686.coloc.results_peer.rdata
</td>
<td style="text-align:left;">
output/wbc_11_102727686.coloc.results_peer.rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/factor_analysis/output/wbc_11_141296923.coloc.results_peer.rdata
</td>
<td style="text-align:left;">
output/wbc_11_141296923.coloc.results_peer.rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/factor_analysis/output/wbc_68_142168399.coloc.results_peer.rdata
</td>
<td style="text-align:left;">
output/wbc_68_142168399.coloc.results_peer.rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/factor_analysis/output/wbc_68_31018955.coloc.results_peer.rdata
</td>
<td style="text-align:left;">
output/wbc_68_31018955.coloc.results_peer.rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/factor_analysis/output/sum_sep_tog_pval_peer_wgpc_ldcl_d1k.rdata
</td>
<td style="text-align:left;">
output/sum_sep_tog_pval_peer_wgpc_ldcl_d1k.rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/factor_analysis/output/info_pval5e8_peer_ld_d1k.rdata
</td>
<td style="text-align:left;">
output/info_pval5e8_peer_ld_d1k.rdata
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomxsun1229factoranalysistree7c67709ea5a8ccc9523581104062c8d1ed10301atargetblank7c67709a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/xsun1229/factor_analysis/tree/7c67709ea5a8ccc9523581104062c8d1ed10301a" target="_blank">7c67709</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomxsun1229factoranalysistree7c67709ea5a8ccc9523581104062c8d1ed10301atargetblank7c67709a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/xsun1229/factor_analysis/tree/7c67709ea5a8ccc9523581104062c8d1ed10301a" target="_blank">7c67709</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    analysis/.Rhistory

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/catalog_gwas_peer_sep_ldclumping_r01d1000.Rmd</code>) and HTML (<code>docs/catalog_gwas_peer_sep_ldclumping_r01d1000.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/7c67709ea5a8ccc9523581104062c8d1ed10301a/analysis/catalog_gwas_peer_sep_ldclumping_r01d1000.Rmd" target="_blank">7c67709</a>
</td>
<td>
XSun
</td>
<td>
2020-11-20
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/e947318db6d17651b1bc7b4cc8a8d48400a111d7/analysis/catalog_gwas_peer_sep_ldclumping_r01d1000.Rmd" target="_blank">e947318</a>
</td>
<td>
XSun
</td>
<td>
2020-11-19
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/8e48c613d9bfc99b899edccbcb8905e05b497a9e/docs/catalog_gwas_peer_sep_ldclumping_r01d1000.html" target="_blank">8e48c61</a>
</td>
<td>
XSun
</td>
<td>
2020-11-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/e7099022b829d88b85541439cb818fcf45c85fab/analysis/catalog_gwas_peer_sep_ldclumping_r01d1000.Rmd" target="_blank">e709902</a>
</td>
<td>
XSun
</td>
<td>
2020-11-19
</td>
<td>
wflow_publish(“analysis/catalog_gwas_peer_sep_ldclumping_r01d1000.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/6bf3af3c05a8fe4b3290ad9f497c720501fd741e/docs/catalog_gwas_peer_sep_ldclumping_r01d1000.html" target="_blank">6bf3af3</a>
</td>
<td>
XSun
</td>
<td>
2020-11-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/87f58b8de24214e4c1410d5f2c964cc9e3e65a3b/analysis/catalog_gwas_peer_sep_ldclumping_r01d1000.Rmd" target="_blank">87f58b8</a>
</td>
<td>
XSun
</td>
<td>
2020-11-19
</td>
<td>
updata
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/76ff242d9c62ab7deabad512fda318dcf0b5cd9b/docs/catalog_gwas_peer_sep_ldclumping_r01d1000.html" target="_blank">76ff242</a>
</td>
<td>
XSun
</td>
<td>
2020-10-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/3bcebcf8225ba63901278c6a86e7743638c8d0a0/analysis/catalog_gwas_peer_sep_ldclumping_r01d1000.Rmd" target="_blank">3bcebcf</a>
</td>
<td>
XSun
</td>
<td>
2020-10-29
</td>
<td>
wflow_publish(“analysis/catalog_gwas_peer_sep_ldclumping_r01d1000.Rmd”)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/178f2784d2d6c33c1ac1ee14522b43de778d2719/analysis/catalog_gwas_peer_sep_ldclumping_r01d1000.Rmd" target="_blank">178f278</a>
</td>
<td>
XSun
</td>
<td>
2020-10-29
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/deadf213066f385f5de350d87280f4bbedc94864/docs/catalog_gwas_peer_sep_ldclumping_r01d1000.html" target="_blank">deadf21</a>
</td>
<td>
XSun
</td>
<td>
2020-10-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/07b2bfdf2104d0b29f531534428fc2aaf3a5f597/analysis/catalog_gwas_peer_sep_ldclumping_r01d1000.Rmd" target="_blank">07b2bfd</a>
</td>
<td>
XSun
</td>
<td>
2020-10-24
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/c099975b144883df9f9f397ba4c53a4c19face1f/docs/catalog_gwas_peer_sep_ldclumping_r01d1000.html" target="_blank">c099975</a>
</td>
<td>
XSun
</td>
<td>
2020-10-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/fe7bf17c83adbec493934196a830a7dc3d03ce43/analysis/catalog_gwas_peer_sep_ldclumping_r01d1000.Rmd" target="_blank">fe7bf17</a>
</td>
<td>
XSun
</td>
<td>
2020-10-24
</td>
<td>
wflow_publish(“analysis/catalog_gwas_peer_sep_ldclumping_r01d1000.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/73f7bffe8c29fed53046c3d13210b17ec23f3db4/docs/catalog_gwas_peer_sep_ldclumping_r01d1000.html" target="_blank">73f7bff</a>
</td>
<td>
XSun
</td>
<td>
2020-10-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/c8c56223aa2598c97a0ad61c58eaa8270d00f47e/analysis/catalog_gwas_peer_sep_ldclumping_r01d1000.Rmd" target="_blank">c8c5622</a>
</td>
<td>
XSun
</td>
<td>
2020-10-24
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/3a794762fc1f9c2e9818758209fc75180de3c891/docs/catalog_gwas_peer_sep_ldclumping_r01d1000.html" target="_blank">3a79476</a>
</td>
<td>
XSun
</td>
<td>
2020-10-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/ea8765cefcb94fe73ace4b9f320c43f1aad1b876/analysis/catalog_gwas_peer_sep_ldclumping_r01d1000.Rmd" target="_blank">ea8765c</a>
</td>
<td>
XSun
</td>
<td>
2020-10-20
</td>
<td>
wflow_publish(“analysis/catalog_gwas_peer_sep_ldclumping_r01d1000.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/b5dcb2c0b1937b659a4760568d9ddff54c8ae24c/docs/catalog_gwas_peer_sep_ldclumping_r01d1000.html" target="_blank">b5dcb2c</a>
</td>
<td>
XSun
</td>
<td>
2020-10-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/cad96c37738b66dc27d9d70d6877ad18e90d24ae/analysis/catalog_gwas_peer_sep_ldclumping_r01d1000.Rmd" target="_blank">cad96c3</a>
</td>
<td>
XSun
</td>
<td>
2020-10-20
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/9f98c45c96ef4a4c7b46c34f0ed13359aa4d464f/docs/catalog_gwas_peer_sep_ldclumping_r01d1000.html" target="_blank">9f98c45</a>
</td>
<td>
XSun
</td>
<td>
2020-10-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/5374205dd44ccc3c3a110d38b4219f30a8da6863/analysis/catalog_gwas_peer_sep_ldclumping_r01d1000.Rmd" target="_blank">5374205</a>
</td>
<td>
XSun
</td>
<td>
2020-10-19
</td>
<td>
wflow_publish(c(“analysis/index.Rmd”, “analysis/catalog_gwas_peer_sep_ldclumping_r01d1000.Rmd”))
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/95057728e6167eca0f09f3481cd209cb10c070b2/analysis/catalog_gwas_peer_sep_ldclumping_r01d1000.Rmd" target="_blank">9505772</a>
</td>
<td>
XSun
</td>
<td>
2020-10-19
</td>
<td>
update
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>In this part, I considered the traits separately. I selected the SNPs with pval &lt; 5e-8 for each traits. Then, I did LD Clumping for these SNPs to eliminate the LD and select a smaller subset of SNPs. After that, I did association tests for the PEER factors with the SNPs. I also did colozalization analysis for some significant factor~snp pairs.</p>
</div>
<div id="material-and-methods" class="section level2">
<h2>Material and Methods</h2>
<ol style="list-style-type: decimal">
<li>Catalog GWAS data:</li>
</ol>
<ul>
<li><p>Platelet count, white blood cell count, myeloid white cell count, lymphocyte counts, red blood cell count, granulocyte count, eosinophil count, neutrophil count from <em>Astle WJ, Elding H, Jiang T, et al. The Allelic Landscape of Human Blood Cell Trait Variation and Links to Common Complex Disease. Cell. 2016;167(5):1415-1429.e19. <a href="doi:10.1016/j.cell.2016.10.042" class="uri">doi:10.1016/j.cell.2016.10.042</a></em>.</p></li>
<li><p>T2D. I first used data from our lab collaction <em>Morris et al. Large-scale association analysis provides insights into the genetic architecture and pathophysiology of type 2 diabetes. Nat Genet. 2012 Sep;44(9):981-90. doi: 10.1038/ng.2383. Epub 2012 Aug 12. PMID: 22885922; PMCID: PMC3442244.</em> but it doesn’t contain MAF info of variants.So I added this from <a href="https://www.ebi.ac.uk/gwas/home">GWAS Catalog</a>: <em>Wood AR et al. Variants in the FTO and CDKAL1 loci have recessive effects on risk of obesity and type 2 diabetes, respectively. Diabetologia. 2016 Jun;59(6):1214-21. doi: 10.1007/s00125-016-3908-5. Epub 2016 Mar 10. PMID: 26961502; PMCID: PMC4869698.</em>.</p></li>
<li><p>Asthma. I first used data from our lab collaction <em>Zhu et al, Shared Genetics of Asthma and Mental Health Disorders: A Large-Scale Genome-Wide Cross-Trait Analysis. European Respiratory Journal, 2019 (PMID: 31619474)</em> but it doesn’t contain MAF info of variants.So I added this from <a href="https://www.ebi.ac.uk/gwas/home">GWAS Catalog</a>: <em>Manuel A.R. et al. Genetic Architectures of Childhood- and Adult-Onset Asthma Are Partly Distinct,The American Journal of Human Genetics,Volume 104, Issue 4,2019,Pages 665-684,ISSN 0002-9297,<a href="https://doi.org/10.1016/j.ajhg.2019.02.022" class="uri">https://doi.org/10.1016/j.ajhg.2019.02.022</a>.</em>.</p></li>
<li><p>IBD,Ulcerative colitist,Crohn’s disease data are from lab collection <em>Liu, van Sommeren et al, Nature Genetics, 2015</em></p></li>
<li><p>Waist-hip ratio data are from <em>Shungin et al. New genetic loci link adipose and insulin biology to body fat distribution. Nature. 2015 Feb 12;518(7538):187-196. doi: 10.1038/nature14132. PMID: 25673412; PMCID: PMC4338562</em>.</p></li>
<li><p>BMI data are also from lab collection: <em>Locke AE, Kahali B, Berndt SI, Justice AE, Pers TH, Day FR, Powell C, Vedantam S, Buchkovich ML, Yang J, Croteau-Chonka DC, Esko T et al. (2015). Genetic studies of body mass index yield new insights for obesity biology. Nature 518, 197-206</em></p></li>
<li><p>HDL &amp; LDL data are downloaded from <a href="https://www.ebi.ac.uk/gwas/home">GWAS Catalog</a>. <em>Global Lipids Genetics Consortium., Willer, C., Schmidt, E. et al. Discovery and refinement of loci associated with lipid levels. Nat Genet 45, 12741283 (2013). <a href="https://doi.org/10.1038/ng.2797" class="uri">https://doi.org/10.1038/ng.2797</a></em></p></li>
</ul>
<ol start="2" style="list-style-type: decimal">
<li><p>Filtered the SNPs using pval &lt; 5e-8 as cut off from the GWAS Catalog data.</p></li>
<li><p>Did LD Clumping for the SNPs in step 2. The PLINK LD Clumping patameters are:</p>
<p>–clump-p1 0.0001 Significance threshold for index SNPs</p>
<p>–clump-p2 0.01 Secondary significance threshold for clumped SNPs</p>
<p>–clump-r2 0.1 LD threshold for clumping</p>
<p>–clump-kb 1000 Physical distance threshold for clumping</p></li>
</ol>
<p>For each trait, I got a subset of SNPs that are not in LD with each other.</p>
<ol start="4" style="list-style-type: decimal">
<li><p>Did association tests for plier_canonical factors and SNPs in 3. The association tests were corrected by 1)10 genotype PCs of whole genome; 2)10 PCs + GTEx:Sequencing platform,Sequencing protocol,Sex; 3)10 PCs + GTEx:Sequencing platform,Sequencing protocol,Sex + AGE</p></li>
<li><p>For each trait, I made a plot of association with LV(indicating by beta in GWAS) vs association with trait(indicating by ln(odds ratio) or beta in GWAS) to show if the variants have the correlated effect direction. The effect sizes of Catalog GWAS and factor association tests are harmonized by <a href="https://mrcieu.github.io/TwoSampleMR/articles/introduction.html">TwoSampleMR</a> R package to make the effect alleles in these two analysis identical. The LVs have more than one significant SNPs with FDR&lt;0.2 are included in the plotting.Besides, for each plots, I fitted the points with intercept = 0. The pvalues and r-squared are shown on the plots.</p></li>
<li><p>For the traits and LVs in 5, I made an info table to show more details of the SNPs.</p></li>
<li><p>For several LVs we are interested in, I did gene set enrichment analysis to test if the LVs are correlated with some KEGG/REACTOME pathways. I used two kind gene sets to do GSEA: 1. genes that used to compute LVs; 2. Sorting the genes in 1 by their loadings, take the top 25% as the gene set. For both gene sets, the gene scores used as input of GSEA are the gene loadings.</p></li>
</ol>
</div>
<div id="snps-after-filter" class="section level2">
<h2>Snps after filter</h2>
<p>After filtering by ‘pval &lt; 5e-8’ and LD Clumping, for each trait, I got :</p>
<p>platelet count trait contains 688 SNPs with pval&lt;5e-8.</p>
<p>white blood cell count trait contains 368 SNPs with pval&lt;5e-8.</p>
<p>myeloid white cell count trait contains 319 SNPs with pval&lt;5e-8.</p>
<p>lymphocyte count trait contains 436 SNPs with pval&lt;5e-8.</p>
<p>red blood cell count trait contains 466 SNPs with pval&lt;5e-8.</p>
<p>granulocyte count trait contains 316 SNPs with pval&lt;5e-8.</p>
<p>eosinophil count trait contains 491 SNPs with pval&lt;5e-8.</p>
<p>neutrophil count trait contains 317 SNPs with pval&lt;5e-8.</p>
<p>IBD trait contains 116 SNPs with pval&lt;5e-8.</p>
<p>Ulcerative colitist trait contains 73 SNPs with pval&lt;5e-8.</p>
<p>Crohn’s disease trait contains 96 SNPs with pval&lt;5e-8.</p>
<p>BMI trait contains 104 SNPs with pval&lt;5e-8.</p>
<p>T2D contains 14 SNPs with pval&lt;5e-8. T2D_2 contains 4 SNPs with pval&lt;5e-8.</p>
<p>Asthma trait contains 186 SNPs with pval&lt;5e-8. Asthma_2 trait contains 112 SNPs with pval&lt;5e-8.</p>
<p>HDL trait contains 227 SNPs with pval&lt;5e-8.</p>
<p>LDL trait contains 204 SNPs with pval&lt;5e-8.</p>
<p>WHR trait contains 36 SNPs with pval&lt;5e-8.</p>
</div>
<div id="results---pval-5e-8-120-peer-factors-covariants-10-pcs-gtexsequencing-platformsequencing-protocolsex-age" class="section level2">
<h2>Results - pval &lt; 5e-8 &amp; 120 PEER factors &amp; covariants: 10 PCs + GTEx:Sequencing platform,Sequencing protocol,Sex + AGE</h2>
<div id="summary-table" class="section level3">
<h3>Summary table</h3>
<p>I used ‘qvalue’ R package to compute the fdr from p-values for each SNP and made a table to show the number of SNPs that pass the threshold. The thresholds are ‘fdr &lt; 0.1’,‘fdr &lt; 0.2’,‘pval &lt; 5e-8’. The ‘num_significant_pairs’ indicates the number of significant pairs under each threshold. If a trait~factor pair has as least 1 significant SNP, we named it as ‘significant pair’.</p>
<div id="htmlwidget-c3f4891eb8e9a8c25198" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-c3f4891eb8e9a8c25198">{"x":{"filter":"none","data":[["num_significant_pairs","factor1","factor2","factor3","factor4","factor5","factor6","factor7","factor8","factor9","factor10","factor11","factor12","factor13","factor14","factor15","factor16","factor17","factor18","factor19","factor20","factor21","factor22","factor23","factor24","factor25","factor26","factor27","factor28","factor29","factor30","factor31","factor32","factor33","factor34","factor35","factor36","factor37","factor38","factor39","factor40","factor41","factor42","factor43","factor44","factor45","factor46","factor47","factor48","factor49","factor50","factor51","factor52","factor53","factor54","factor55","factor56","factor57","factor58","factor59","factor60","factor61","factor62","factor63","factor64","factor65","factor66","factor67","factor68","factor69","factor70","factor71","factor72","factor73","factor74","factor75","factor76","factor77","factor78","factor79","factor80","factor81","factor82","factor83","factor84","factor85","factor86","factor87","factor88","factor89","factor90","factor91","factor92","factor93","factor94","factor95","factor96","factor97","factor98","factor99","factor100","factor101","factor102","factor103","factor104","factor105","factor106","factor107","factor108","factor109","factor110","factor111","factor112","factor113","factor114","factor115","factor116","factor117","factor118","factor119","factor120"],[5,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0],[18,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,1,0,0,0,0,0,0,0,0,0,0,2,0,1,0,0,0,0,3,0,1,0,0,0,0,0,0,0,0,1,0,0,0,1,0,0,0,0,0,0,0,0,1,0,0,1,0,0,0,1,0,0,0,0,2,1,0,1,0,0,0,0,0,0,0,0,0,1,0,0,0,4,0,0,0,0,1,0,1,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[7,0,0,0,0,0,0,0,1,1,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0],[14,0,0,0,0,0,0,0,1,1,0,0,0,0,2,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,1,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,2,0,0,0,0,1,0,0,0,0,0,0,0,0,1,0,0,1,0,0,0,1,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,1,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[2,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[11,0,0,0,0,0,0,0,0,3,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,1,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,2,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,1,0,0,1,1,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[4,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0],[8,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,2,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,2,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,1,0,0,0,1,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[5,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0],[12,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,1,0,0,1,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,2,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,1,0,0,0,0,0,1,0,0,2],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[3,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,2,0,0,0,0,0,0,0,0,0,0],[10,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,1,0,0,0,2,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,2,0,0,0,0,0,0,0,0,0,1],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[4,0,0,0,0,0,0,0,0,0,0,0,1,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,2,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,2,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[13,0,0,0,0,0,0,0,0,0,0,0,1,0,0,1,0,1,0,0,0,0,1,1,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,3,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,3,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,2,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,1,0,0,0,0,0,1,1,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[6,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,1,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[16,2,1,0,0,0,0,0,0,2,0,0,0,0,0,1,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,2,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,1,0,0,0,0,0,1,0,0,1,0,0,0,0,0,1,0,0,0,1,0,0,0,1,0,0,0,2,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[5,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,2,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[12,0,0,0,0,1,0,0,2,0,0,1,0,0,0,0,0,0,0,2,0,0,0,0,0,0,0,0,1,0,1,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,1,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[6,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,1,3,0,0,0,2,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[14,0,0,0,0,1,0,1,0,0,0,0,0,0,0,0,0,0,0,0,2,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,2,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,2,0,0,0,0,1,3,0,0,0,2,0,0,0,0,0,0,0,0,0,0,1,1,0,0,0,0,1,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[5,0,0,0,0,0,0,0,0,0,0,0,1,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,2,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,2,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[12,0,0,0,0,0,0,0,0,0,0,0,1,0,0,1,0,1,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,3,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,2,0,0,0,1,0,0,3,0,0,0,0,0,0,0,0,2,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,1,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[6,0,0,0,0,0,0,0,0,0,0,0,1,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,2,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,2,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0],[16,0,0,0,0,0,0,0,0,1,0,0,1,0,0,1,0,1,0,0,0,0,0,0,0,0,0,1,0,0,0,0,1,1,0,0,0,0,0,0,0,0,0,3,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,3,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,1,0,0,0,0,0,1,1,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[4,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,2,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[17,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,1,0,0,1,0,0,0,0,0,0,0,1,0,0,1,0,0,0,0,0,1,1,0,2,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,1,0,2,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,1,0,1,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[7,0,1,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[16,0,1,1,1,0,1,0,0,1,0,1,0,1,0,0,0,0,0,2,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,2,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,2,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,1,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[2,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0],[4,0,0,0,0,0,0,0,0,3,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[7,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,2,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[15,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,2,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,1,0,0,1,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,1,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,1,0,0,0,1,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,1,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[5,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,2,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[19,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,1,0,0,0,0,3,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,2,0,0,1,0,0,0,0,0,0,1,2,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,1,3,0,0,0,0,0,1,0,3,0,1,0,1,0,1,0,1,0,0,0,0,0,0,0,0,1,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[3,0,0,0,0,0,0,0,0,0,0,0,1,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,3,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[16,0,0,0,0,0,0,0,0,1,0,2,1,0,0,1,0,1,1,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,4,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,2,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,1,0,0,0,0,0,0,0,0,0,0,0,0,1,0,1,0,0,0,0,0,0,1,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[2,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[13,0,0,0,0,0,0,0,0,0,2,0,0,0,1,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,2,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,1,1,0,0,0,0,1,0,0,0,0,1,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,1,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>asthma_2_fdr010_num<\/th>\n      <th>asthma_2_fdr020_num<\/th>\n      <th>asthma_2_pval5e8_num<\/th>\n      <th>bmi_fdr010_num<\/th>\n      <th>bmi_fdr020_num<\/th>\n      <th>bmi_pval5e8_num<\/th>\n      <th>eo_fdr010_num<\/th>\n      <th>eo_fdr020_num<\/th>\n      <th>eo_pval5e8_num<\/th>\n      <th>EUR.CD_fdr010_num<\/th>\n      <th>EUR.CD_fdr020_num<\/th>\n      <th>EUR.CD_pval5e8_num<\/th>\n      <th>EUR.IBD_fdr010_num<\/th>\n      <th>EUR.IBD_fdr020_num<\/th>\n      <th>EUR.IBD_pval5e8_num<\/th>\n      <th>EUR.UC_fdr010_num<\/th>\n      <th>EUR.UC_fdr020_num<\/th>\n      <th>EUR.UC_pval5e8_num<\/th>\n      <th>gran_fdr010_num<\/th>\n      <th>gran_fdr020_num<\/th>\n      <th>gran_pval5e8_num<\/th>\n      <th>HDL_fdr010_num<\/th>\n      <th>HDL_fdr020_num<\/th>\n      <th>HDL_pval5e8_num<\/th>\n      <th>LDL_fdr010_num<\/th>\n      <th>LDL_fdr020_num<\/th>\n      <th>LDL_pval5e8_num<\/th>\n      <th>lymph_fdr010_num<\/th>\n      <th>lymph_fdr020_num<\/th>\n      <th>lymph_pval5e8_num<\/th>\n      <th>myeloid_wbc_fdr010_num<\/th>\n      <th>myeloid_wbc_fdr020_num<\/th>\n      <th>myeloid_wbc_pval5e8_num<\/th>\n      <th>neut_fdr010_num<\/th>\n      <th>neut_fdr020_num<\/th>\n      <th>neut_pval5e8_num<\/th>\n      <th>plt_fdr010_num<\/th>\n      <th>plt_fdr020_num<\/th>\n      <th>plt_pval5e8_num<\/th>\n      <th>rbc_fdr010_num<\/th>\n      <th>rbc_fdr020_num<\/th>\n      <th>rbc_pval5e8_num<\/th>\n      <th>T2D_2_fdr010_num<\/th>\n      <th>T2D_2_fdr020_num<\/th>\n      <th>T2D_2_pval5e8_num<\/th>\n      <th>T2D_fdr010_num<\/th>\n      <th>T2D_fdr020_num<\/th>\n      <th>T2D_pval5e8_num<\/th>\n      <th>ukb.allasthma_fdr010_num<\/th>\n      <th>ukb.allasthma_fdr020_num<\/th>\n      <th>ukb.allasthma_pval5e8_num<\/th>\n      <th>wbc_fdr010_num<\/th>\n      <th>wbc_fdr020_num<\/th>\n      <th>wbc_pval5e8_num<\/th>\n      <th>WHR_fdr010_num<\/th>\n      <th>WHR_fdr020_num<\/th>\n      <th>WHR_pval5e8_num<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":20,"columnDefs":[{"className":"dt-right","targets":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[10,20,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="info-table" class="section level3">
<h3>Info table</h3>
<p>For each trait, I made a table to show the info of snps with fdr&gt;0.2 in the factor ~ SNP + genotype pcs association test. For each trait,The LVs have more than one significant SNPs with FDR&lt;0.2 are included.</p>
<p>The suffix ’_assoc’ here means that results are from factor ~ SNP + genotype pcs association test. The suffix ’_gwas’ here means results are from original GWAS results files. For EUR.CD, EUR.IBD, EUR.UC, the effectsize_gwas here means ‘ln(OR)’, for others, it means ‘beta’.</p>
<p>‘snp_ld’ here means the snps that in LD with the snp in each line.’ld_r2’ means the LD r-squared which is corresponding to the ‘snp_ld’ column. ‘cis-eqtl’ column indicates whether the snp is a cis-eqtl according to GTEx data. ‘cis_gene_hgnc’ and ‘cis_gene_hgnc’ is the genes that the snp influence when it act as cis-eqtl. ‘func’ and ‘func_gene’ are obtained from ANNOVAR, which indicating the snp function within the genes.</p>
<div id="htmlwidget-cfefb73e51b4928e765c" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-cfefb73e51b4928e765c">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97","98","99","100","101","102","103","104","105","106","107","108","109","110","111","112","113","114"],["asthma_2","asthma_2","asthma_2","asthma_2","asthma_2","asthma_2","asthma_2","asthma_2","asthma_2","asthma_2","asthma_2","bmi","bmi","bmi","bmi","eo","eo","eo","eo","eo","EUR.CD","EUR.CD","EUR.CD","EUR.CD","EUR.IBD","EUR.IBD","EUR.IBD","EUR.IBD","EUR.UC","EUR.UC","EUR.UC","EUR.UC","gran","gran","gran","gran","gran","gran","gran","gran","HDL","HDL","HDL","HDL","HDL","HDL","HDL","HDL","LDL","LDL","LDL","LDL","lymph","lymph","lymph","lymph","lymph","lymph","lymph","lymph","lymph","lymph","lymph","myeloid_wbc","myeloid_wbc","myeloid_wbc","myeloid_wbc","myeloid_wbc","myeloid_wbc","myeloid_wbc","myeloid_wbc","myeloid_wbc","myeloid_wbc","neut","neut","neut","neut","neut","neut","plt","plt","plt","plt","rbc","rbc","rbc","rbc","rbc","rbc","ukb.allasthma","ukb.allasthma","ukb.allasthma","ukb.allasthma","ukb.allasthma","ukb.allasthma","ukb.allasthma","ukb.allasthma","ukb.allasthma","ukb.allasthma","ukb.allasthma","ukb.allasthma","ukb.allasthma","wbc","wbc","wbc","wbc","wbc","wbc","wbc","wbc","WHR","WHR","WHR","WHR"],["LV50","LV50","LV57","LV57","LV57","LV93","LV93","LV110","LV110","LV110","LV110","LV14","LV14","LV79","LV79","LV9","LV9","LV9","LV57","LV57","LV75","LV75","LV91","LV91","LV91","LV91","LV120","LV120","LV77","LV77","LV110","LV110","LV43","LV43","LV43","LV79","LV79","LV79","LV96","LV96","LV1","LV1","LV9","LV9","LV38","LV38","LV108","LV108","LV8","LV8","LV19","LV19","LV20","LV20","LV44","LV44","LV66","LV66","LV72","LV72","LV72","LV76","LV76","LV43","LV43","LV43","LV72","LV72","LV79","LV79","LV79","LV88","LV88","LV43","LV43","LV43","LV79","LV79","LV79","LV50","LV50","LV81","LV81","LV19","LV19","LV41","LV41","LV73","LV73","LV24","LV24","LV24","LV47","LV47","LV58","LV58","LV94","LV94","LV94","LV102","LV102","LV102","LV11","LV11","LV43","LV43","LV43","LV43","LV68","LV68","LV10","LV10","LV39","LV39"],["chr8_80386214_C_A_b38","chr9_5866925_G_A_b38","chr16_11119164_T_A_b38","chr3_196035621_G_A_b38","chr5_132447521_C_T_b38","chr6_32423918_T_C_b38","chr6_32714617_T_G_b38","chr1_198671359_C_T_b38","chr10_9012377_A_C_b38","chr10_9069863_C_T_b38","chr12_47792780_T_A_b38","chr19_18344015_A_G_b38","chr3_85758440_T_G_b38","chr2_62825913_G_A_b38","chr9_117616205_T_C_b38","chr6_25480100_T_G_b38","chr6_27970706_A_G_b38","chr9_6534466_G_A_b38","chr6_30338176_A_G_b38","chr8_129662115_CTACAT_C_b38","chr2_233244476_G_A_b38","chr5_40438414_C_G_b38","chr1_67240275_G_A_b38","chr16_50865029_T_G_b38","chr1_67043149_G_A_b38","chr6_31557671_C_T_b38","chr10_99530544_T_C_b38","chr6_31748659_A_G_b38","chr6_32483933_A_T_b38","chr6_32636617_T_C_b38","chr6_32577904_T_A_b38","chr6_32578764_T_C_b38","chr10_44386212_C_T_b38","chr17_39987826_G_T_b38","chr6_109286353_G_A_b38","chr17_39157532_T_G_b38","chr3_128604581_G_C_b38","chr8_60738272_A_T_b38","chr11_118251337_C_G_b38","chr6_30644330_A_C_b38","chr16_56950678_T_C_b38","chr5_56564954_A_G_b38","chr15_43919075_T_C_b38","chr3_136213517_C_T_b38","chr11_65623846_G_A_b38","chr16_56973280_T_C_b38","chr4_88820118_G_A_b38","chr7_150832361_T_C_b38","chr1_150986360_A_G_b38","chr2_21125702_A_G_b38","chr1_109421983_T_C_b38","chr1_109497688_A_G_b38","chr1_101238976_G_T_b38","chr7_22724719_G_A_b38","chr20_1986403_C_T_b38","chr8_128000328_C_T_b38","chr6_29892532_A_G_b38","chr9_137031531_G_A_b38","chr1_56442048_G_A_b38","chr22_41364168_T_G_b38","chr6_31148894_G_A_b38","chr10_62808178_C_T_b38","chr12_112677823_T_C_b38","chr10_44384812_T_C_b38","chr17_39987826_G_T_b38","chr6_109286353_G_A_b38","chr6_31148894_G_A_b38","chr6_32660643_C_T_b38","chr17_39157532_T_G_b38","chr3_128604581_G_C_b38","chr8_60738272_A_T_b38","chr18_23507567_C_T_b38","chr6_31148894_G_A_b38","chr10_44384812_T_C_b38","chr17_39987826_G_T_b38","chr6_109286353_G_A_b38","chr17_39157532_T_G_b38","chr3_128604581_G_C_b38","chr8_60738272_A_T_b38","chr14_69023286_C_T_b38","chr9_4780052_C_T_b38","chr6_33692411_T_C_b38","chr9_70455368_C_T_b38","chr14_68044056_C_T_b38","chr2_60399111_C_T_b38","chr7_100144470_G_A_b38","chr7_100370748_C_T_b38","chr7_100749368_C_T_b38","chr7_100933277_C_T_b38","chr17_40021896_A_G_b38","chr19_6580881_T_C_b38","chr3_33026456_C_T_b38","chr1_152502757_A_G_b38","chr1_153302967_G_A_b38","chr2_227840005_T_C_b38","chr3_196641320_G_C_b38","chr16_27370327_C_A_b38","chr5_132698896_C_A_b38","chr6_32606581_G_A_b38","chr16_27354132_C_G_b38","chr17_48486345_G_T_b38","chr5_111245164_C_A_b38","chr10_102727686_G_A_b38","chr8_141296923_C_T_b38","chr10_44386212_C_T_b38","chr17_2101961_CAGA_C_b38","chr17_39987826_G_T_b38","chr6_109300291_T_C_b38","chr5_142168399_T_A_b38","chr6_31018955_C_CT_b38","chr6_126960402_C_T_b38","chr8_71580003_T_G_b38","chr1_119006795_C_T_b38","chr7_25818994_C_A_b38"],["rs7824278","rs66632892","rs35441874","rs11715524","rs7703152","rs28895026","rs3997848","rs17668708","rs1775554","rs72784452","rs2544026","rs17724992","rs13078960","rs11688816","rs1928295","rs34991172","rs13216117","rs2282162","rs9261659","rs146895157","rs78583556","rs150566648","rs11209026","rs141976787","rs113843670","rs2230365","rs6584283","rs116043458","rs114765062","rs9272347","rs34902830","rs35528115","rs17885289","rs118009374","rs9400271","rs801241","rs6782228","rs7846314","rs1715460","rs45503895","rs9938160","rs3936511","rs492571","rs687339","rs12801636","rs9930761","rs3822072","rs17173637","rs267733","rs531593","rs17647543","rs6666119","rs3737577","rs62449495","rs13037430","rs10087240","rs9259312","rs117125309","rs4319365","rs9611562","rs9263760","rs9990","rs232935","rs72790862","rs118009374","rs9400271","rs9263760","rs9273505","rs801241","rs6782228","rs7846314","rs891389","rs9263760","rs72790862","rs118009374","rs9400271","rs801241","rs6782228","rs7846314","rs28535566","rs144745077","rs4711338","rs11142447","rs72725174","rs3948623","rs4729578","rs62482199","rs80134657","rs7797690","rs2302776","rs2029745","rs72854772","rs6669608","rs821429","rs7423358","rs11185551","rs77315098","rs10455052","rs2760995","rs3024622","rs1553754","rs75503445","rs7917772","rs76942920","rs17885289","rs71902577","rs118009374","rs6927569","rs2288822","rs9281102","rs987763","rs12549058","rs2765539","rs10245353"],["1.06233","-0.946701","-1.08093","0.949974","1.06801","-1.10054","-0.95045","-1.09563","-1.10914","-1.11632","-1.06001","-0.0194","0.0297","-0.0172","-0.0188","-0.03559993","-0.05857272","-0.02840131","0.03385428","-0.02825609","0.235301344917802","0.36279362494691","-0.995198666979028","0.684009060648921","-0.318705012687104","0.153802047633545","0.180302853917034","-0.378299440035395","-0.817894410017962","-0.379100598676836","-0.395697962349193","-0.284599827210502","-0.02326956","-0.08451142","0.03462562","0.03047516","-0.03102483","0.05996328","-0.02369268","0.03587079","0.0599","-0.0308","-0.0663","0.0316","0.0235","0.0619","-0.0251","-0.0363","-0.0331","-0.0694","-0.0801","-0.0299","0.02222877","-0.02920904","-0.04181903","0.02405023","0.03894995","0.1170279","0.02171254","0.02443549","0.048283","0.03014244","-0.02010865","-0.02433963","-0.08176227","0.03495081","0.03776033","0.03366127","0.02944234","-0.03977305","0.05674002","-0.02485763","0.03776033","-0.0220783","-0.08172991","0.03322504","0.03062596","-0.03407596","0.06456402","0.04747107","-0.1133533","0.04786607","-0.0239581","0.03792288","0.04403375","-0.0328844","0.05599315","0.06801273","0.0311448","-0.0576873","0.0363865","-0.0699484","0.0446196","0.0903323","-0.0411032","-0.0460908","-0.0877008","0.0646093","-0.0897591","0.0545226","-0.0347811","-0.0671227","-0.02219608","-0.06154283","-0.02783845","-0.03205232","-0.07104802","-0.029455","-0.02110647","0.02333396","-0.02","0.04","-0.027","0.027"],["1.4e-12","9.9e-09","6e-16","8.5e-10","3.7e-11","1.3e-19","1e-08","1.6e-11","7.8e-35","1.7e-12","1.2e-09","3.415e-08","1.737e-14","1.893e-08","7.91e-10","3.079e-08","1.078e-25","3.112e-15","2.049e-09","3.914e-15","4.098e-08","2.656e-29","1.052e-54","1.142e-11","4.121e-16","1.968e-11","1.701e-26","2.818e-08","1.037e-08","2.111e-49","9.787e-11","6.304e-09","2.677e-09","2.712e-18","5.496e-22","1.391e-08","2.859e-14","1.592e-38","1.978e-08","1.284e-09","3.17e-25","2.959e-09","1.265e-12","7.108e-13","3.147e-08","9.551e-10","4.058e-12","1.899e-08","5.285e-09","7.034e-18","3.347e-24","1.323e-12","1.311e-08","3.442e-11","9.109e-09","4.612e-11","2.033e-18","1.07e-13","2.228e-09","1.578e-08","3.134e-27","3.531e-08","4.032e-08","5.443e-10","3.886e-17","2.861e-22","2.845e-17","3.186e-19","4.532e-08","2.353e-22","1.683e-34","6.446e-11","2.845e-17","1.541e-08","2.799e-17","1.848e-20","1.091e-08","5.603e-17","1.365e-44","1.862e-15","5.394e-14","3.072e-39","6.13e-11","3.186e-15","1.42e-10","2.172e-14","6.19e-23","7.121e-13","6.865e-17","3.4e-20","2.1e-08","3.6e-09","6.8e-11","2.4e-08","9.6e-09","1.4e-08","1.1e-13","3.5e-12","4.7e-17","1.6e-17","2e-08","7.2e-09","2.001e-09","7.443e-12","1.005e-12","7.173e-13","2.102e-13","2.321e-16","6.686e-09","3.827e-10","1.1e-08","3.2e-10","1.1e-12","1.6e-10"],["-0.1696","0.1715","0.1676","0.131","-0.1649","-0.195","0.1289","-0.1762","-0.1008","-0.1926","0.1207","-0.01719","0.02526","0.1357","-0.1557","-0.02393","-0.01884","-0.00966","-0.297","0.1729","-0.2777","0.2389","0.2655","0.5701","0.2818","-0.1863","0.1052","-0.323","0.4147","0.1561","-0.4983","0.4874","0.1658","-0.4921","0.1739","0.2455","0.1732","0.1953","0.1529","0.2393","-0.006025","-0.007617","-0.02213","-0.01214","0.1643","-0.3354","0.114","0.2057","0.01357","-0.02303","-0.07306","-0.03633","0.03406","0.03977","-0.3735","0.1621","-0.2017","-0.7782","-0.1615","0.2233","0.2052","-0.2292","0.1692","0.1687","-0.4921","0.1739","0.2052","0.1506","0.2455","0.1732","0.1953","0.1498","0.1788","0.1687","-0.4921","0.1739","0.2455","0.1732","0.1953","0.2984","0.9392","-0.1425","-0.1433","-0.04606","-0.06712","-0.1703","0.2382","-0.4738","-0.1831","0.04849","-0.04921","-0.09185","-0.1848","-0.4147","0.1831","-0.2123","0.2488","-0.1633","-0.1631","-0.1239","-0.1289","-0.2355","0.01386","0.03067","0.1658","-0.19","-0.4921","-0.1709","-0.162","0.1535","-0.009066","-0.01669","-0.1295","0.1638"],["0.000398","0.001847","0.001824","0.004892","0.005141","8.491e-05","0.001696","0.001019","0.002682","0.002905","0.001657","0.00139","2.239e-05","0.001208","0.000378","0.000226","0.0007151","0.001046","0.0002283","0.0002675","0.0008753","0.0004773","0.001809","0.0003603","0.0005609","0.001207","0.001134","0.0005181","0.0003001","0.000825","0.0005025","0.0001741","0.000691","0.0001529","9.673e-05","1.299e-05","0.0001359","0.0003466","0.0005614","0.000621","0.0007506","0.0004256","0.0002059","0.0005184","0.0003099","5.112e-05","0.0007974","0.0004057","0.0006101","0.0003981","0.0002891","0.0003296","0.0004769","0.0004667","9.938e-05","0.0003487","3.552e-05","0.0001729","0.0001727","3.817e-05","0.0001618","0.0001057","7.091e-05","0.0005571","0.0001529","9.673e-05","0.0001618","0.0003143","1.299e-05","0.0001359","0.0003466","0.0007916","0.0005172","0.0005571","0.0001529","9.673e-05","1.299e-05","0.0001359","0.0003466","1.697e-05","6.516e-06","0.0004437","0.0004979","0.0002405","0.000209","0.0004756","0.0004099","0.0002486","9.697e-05","0.0008347","0.001308","0.0007998","0.0007248","0.0004788","0.0007779","0.0006055","0.001434","0.002413","0.0007818","0.0009066","0.0003714","0.00017","0.0003087","0.0005094","0.000691","0.0002579","0.0001529","0.0001947","0.0003264","0.0005767","0.002114","0.003936","0.003362","0.001722"],["0.0733926835219887","0.184086795229037","0.136933139592798","0.194004329349068","0.197854980806859","0.0363137367803715","0.195493127935023","0.104381884115727","0.190089177435928","0.19991833426485","0.140260659148748","0.164808627311671","0.0119085677219918","0.192926356963283","0.083512095237533","0.103800452684709","0.159369128943212","0.193299423588222","0.167933898160831","0.186507126222744","0.164085549419436","0.0945374172656292","0.157758466997134","0.06756502769822","0.0926486939261689","0.155306604458421","0.171722319267541","0.102708403491488","0.0486018547517842","0.100529458425843","0.0873744411208906","0.0337743811483556","0.147259600885105","0.0588914730074046","0.0449540931252058","0.0124155460390785","0.0758298426096531","0.170817821344396","0.129824203830449","0.136541625726156","0.169263415634953","0.127342980596232","0.0806688632462804","0.13546740576974","0.143333646888693","0.0318773522459543","0.189953949145631","0.133008663579189","0.115789340493538","0.0967338387809577","0.0742385110890042","0.0810482962415972","0.199843374280362","0.197130870449549","0.0807600099133779","0.175998726346032","0.0373867265835895","0.124668639386821","0.0957064614044164","0.0310134765335704","0.0908292616875012","0.0923705217169303","0.0640293584956439","0.143964258817474","0.0619905679193172","0.0464538989308661","0.102555767185979","0.169847713291869","0.012476060815226","0.0759603390471379","0.175572782056592","0.1782623515732","0.140312676857815","0.161655316958838","0.0623988499831518","0.0460434808274669","0.0125138293841159","0.0808196383810042","0.197147139236809","0.0250639062640565","0.0103315425031108","0.179911723244853","0.190953957052654","0.129337636428544","0.119383659794914","0.184448342328996","0.173024085272095","0.168199822828891","0.0870709916651762","0.131652542579363","0.17028424270278","0.128552298005863","0.19021261714401","0.143525620818315","0.199387621040798","0.172126917120881","0.132046332078164","0.175850117491565","0.100949817455736","0.14757273657048","0.0784871913703515","0.047770155268503","0.147027824906705","0.19935514616024","0.193032437447101","0.0878810057133952","0.0603213021964558","0.0713448351658603","0.14354705515596","0.183450046049058","0.118270455602161","0.163474257600371","0.120006679287783","0.0823405678382328"],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"chr6_27970706_A_G_b38","chr6_25480100_T_G_b38",null,null,null,null,null,null,null,null,null,null,null,"chr6_32636617_T_C_b38","chr6_32483933_A_T_b38",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"chr1_109497688_A_G_b38","chr1_109421983_T_C_b38",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"0.326403","0.326403",null,null,null,null,null,null,null,null,null,null,null,"0.0746541","0.0746541",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"0.091373","0.091373",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,"T","T",null,null,null,"T",null,null,null,null,"T","T",null,"T",null,null,null,null,null,null,"T",null,"T","T","T",null,null,null,null,"T",null,null,null,"T","T",null,null,null,null,"T",null,null,null,null,null,"T","T",null,null,null,null,"T",null,null,"T","T",null,null,null,null,"T","T","T",null,null,null,null,"T",null,null,"T",null,null,null,null,null,"T",null,null,null,"T","T",null,null,null,null,null,null,null,null,null,null,null,"T","T","T",null,"T","T",null,null,null,"T",null,"T",null,null,null,null],[null,null,null,null,null,"HLA-DRB9,HLA-DRB5,ZBTB12,HLA-DQB2","HLA-DRB9,CYP21A1P,HLA-DRB6,HLA-DQB2,HLA-DQA2,HLA-DOB",null,null,null,"RP1-197B17.3,RP5-1057I20.5",null,null,null,null,"BTN3A2,U91328.19,U91328.22","ZKSCAN4,RP1-265C24.5",null,"VARS2,ZFP57,HCG4P5,HLA-T,TRIM26,IFITM4P,RPL23AP1",null,null,null,null,null,null,"LY6G5C,XXbac-BPG248L24.12,MIR6891",null,"LY6G5C,HLA-C,HLA-DQA2","HLA-DRB1,FKBPL,ZBTB12,MICB,HLA-DQA2","HLA-DQB1,HLA-DRB1,HLA-DRB9,HLA-DQA1,CYP21A1P,HLA-DQB1-AS1,HLA-DRB6,HLA-DQB2,HLA-DQA2,STK19B",null,null,null,null,"CD164,SMPD2,CCDC162P",null,null,null,"MPZL2,MPZL3,AMICA1,CD3E,HSPE1P18","VARS2,DDX39B,HLA-A,HCG4P5,HLA-T,IFITM4P,RPL23AP1",null,null,null,null,"RNASEH2C,MAP3K11,KCNK7,SIPA1",null,null,null,null,null,"GSTM3,GSTM4","GSTM4,AMIGO1",null,null,null,null,"HLA-J,HCG9,HLA-F,HLA-H,HLA-F-AS1,MICD,HCG4P7,HLA-K,IFITM4P,DDX39BP2,RPL23AP1,HCP5B",null,null,"PMM1,DESI1,TEF,MEI1","VARS2,MICB,HLA-S,XXbac-BPG248L24.12",null,null,null,null,"CD164,SMPD2,CCDC162P","VARS2,MICB,HLA-S,XXbac-BPG248L24.12","HLA-DQB1,HLA-DRB1,HLA-DQA1,CYP21A1P,HLA-DQB1-AS1,HLA-DRB6,HLA-DQB2,HLA-DQA2",null,null,null,null,"VARS2,MICB,HLA-S,XXbac-BPG248L24.12",null,null,"CD164,SMPD2,CCDC162P",null,null,null,null,null,"BAK1,ITPR3",null,null,null,"LAMTOR4,AP4M1","MOSPD3,C7orf43,GIGYF1,EPHB4",null,null,null,null,null,null,null,null,null,null,null,"HLA-DRB1,HLA-DRB9,FKBPL,ZBTB12,HLA-DRB6,HLA-DQA2,STK19B","IL4R,CTD-3203P2.1","HOXB3,HOXB2,HOXB4,HOXB-AS1",null,"TMEM180,SFXN2,C10orf32","SLC45A4,PTP4A3,LINC01300",null,null,null,"CD164,SMPD2,CCDC162P",null,"CYP21A1P,ZBTB12,LINC00243,HLA-S,HCG18,HLA-L,XXbac-BPG248L24.12,XXbac-BPG283O16.9",null,null,null,null],[null,null,null,null,null,"ENSG00000196301,ENSG00000198502,ENSG00000204366,ENSG00000232629","ENSG00000196301,ENSG00000204338,ENSG00000229391,ENSG00000232629,ENSG00000237541,ENSG00000241106",null,null,null,"ENSG00000257433,ENSG00000276691",null,null,null,null,"ENSG00000186470,ENSG00000272462,ENSG00000272810","ENSG00000187626,ENSG00000219392",null,"ENSG00000137411,ENSG00000204644,ENSG00000227766,ENSG00000231130,ENSG00000234127,ENSG00000235821,ENSG00000239257",null,null,null,null,null,null,"ENSG00000204428,ENSG00000271581,ENSG00000277402",null,"ENSG00000204428,ENSG00000204525,ENSG00000237541","ENSG00000196126,ENSG00000204315,ENSG00000204366,ENSG00000204516,ENSG00000237541","ENSG00000179344,ENSG00000196126,ENSG00000196301,ENSG00000196735,ENSG00000204338,ENSG00000223534,ENSG00000229391,ENSG00000232629,ENSG00000237541,ENSG00000250535",null,null,null,null,"ENSG00000135535,ENSG00000135587,ENSG00000203799",null,null,null,"ENSG00000149573,ENSG00000160588,ENSG00000160593,ENSG00000198851,ENSG00000255163","ENSG00000137411,ENSG00000198563,ENSG00000206503,ENSG00000227766,ENSG00000231130,ENSG00000235821,ENSG00000239257",null,null,null,null,"ENSG00000172922,ENSG00000173327,ENSG00000173338,ENSG00000213445",null,null,null,null,null,"ENSG00000134202,ENSG00000168765","ENSG00000168765,ENSG00000181754",null,null,null,null,"ENSG00000204622,ENSG00000204625,ENSG00000204642,ENSG00000206341,ENSG00000214922,ENSG00000229390,ENSG00000230521,ENSG00000230795,ENSG00000235821,ENSG00000238024,ENSG00000239257,ENSG00000281831",null,null,"ENSG00000100417,ENSG00000100418,ENSG00000167074,ENSG00000167077","ENSG00000137411,ENSG00000204516,ENSG00000225851,ENSG00000271581",null,null,null,null,"ENSG00000135535,ENSG00000135587,ENSG00000203799","ENSG00000137411,ENSG00000204516,ENSG00000225851,ENSG00000271581","ENSG00000179344,ENSG00000196126,ENSG00000196735,ENSG00000204338,ENSG00000223534,ENSG00000229391,ENSG00000232629,ENSG00000237541",null,null,null,null,"ENSG00000137411,ENSG00000204516,ENSG00000225851,ENSG00000271581",null,null,"ENSG00000135535,ENSG00000135587,ENSG00000203799",null,null,null,null,null,"ENSG00000030110,ENSG00000096433",null,null,null,"ENSG00000188186,ENSG00000221838","ENSG00000106330,ENSG00000146826,ENSG00000146830,ENSG00000196411",null,null,null,null,null,null,null,null,null,null,null,"ENSG00000196126,ENSG00000196301,ENSG00000204315,ENSG00000204366,ENSG00000229391,ENSG00000237541,ENSG00000250535","ENSG00000077238,ENSG00000259940","ENSG00000120093,ENSG00000173917,ENSG00000182742,ENSG00000230148",null,"ENSG00000138111,ENSG00000156398,ENSG00000166275","ENSG00000022567,ENSG00000184489,ENSG00000253595",null,null,null,"ENSG00000135535,ENSG00000135587,ENSG00000203799",null,"ENSG00000204338,ENSG00000204366,ENSG00000214894,ENSG00000225851,ENSG00000231074,ENSG00000243753,ENSG00000271581,ENSG00000280128",null,null,null,null],["intergenic","intergenic","intronic","intergenic","ncRNA_intronic","intergenic","intergenic","intronic","intergenic","intergenic","intronic","intronic","intronic","intronic","intergenic","intronic","intergenic","intronic","intronic","ncRNA_intronic","intergenic","intergenic","exonic","intergenic","intronic","exonic","ncRNA_intronic","ncRNA_intronic","intergenic","upstream","downstream","downstream","intergenic","intronic","ncRNA_exonic","intronic","intergenic","intronic","intronic","UTR3","intergenic","ncRNA_intronic","UTR5","intergenic","intronic","intronic","intronic","intergenic","exonic","intronic","intronic","UTR3","UTR5","downstream","ncRNA_intronic","ncRNA_intronic","intergenic","exonic","intergenic","intergenic","intronic","UTR3","intronic","intronic","intronic","ncRNA_exonic","intronic","ncRNA_exonic","intronic","intergenic","intronic","intronic","intronic","intronic","intronic","ncRNA_exonic","intronic","intergenic","intronic","intergenic","intergenic","intronic","intergenic","intronic","intergenic","intergenic","intergenic","intronic","intergenic","intronic","downstream","intronic","intergenic","intronic","intergenic","intronic","intergenic","intronic","intergenic","intronic","intergenic","intronic","intronic","intergenic","intergenic","intronic","intronic","ncRNA_exonic","intergenic","intronic","intergenic","intergenic","intergenic","intergenic"],["MIR5708(dist=144741),ZBTB10(dist=99379)","ERMP1(dist=33808),MLANA(dist=23964)","CLEC16A","SDHAP1(dist=45342),TFRC(dist=13663)","C5orf56","TSBP1-AS1(dist=16189),HLA-DRA(dist=15969)","HLA-DQB1(dist=47960),HLA-DQA2(dist=26774)","PTPRC","LINC02676(dist=97781),LOC101928272(dist=184989)","LINC02676(dist=155267),LOC101928272(dist=127503)","HDAC7","PGPEP1","CADM2","EHBP1","ASTN2(dist=201148),LOC101928797(dist=32401)","CARMIL1","OR2B6(dist=12524),OR1F12(dist=102610)","GLDC","TRIM39,TRIM39-RPP21","CCDC26","INPP5D(dist=36573),ATG16L1(dist=7095)","LINC00603(dist=385090),PTGER4(dist=241501)","IL23R","LINC02168(dist=57400),MIR548AI(dist=13744)","SLC35D1","NFKBIL1","LINC01475","MSH5-SAPCD1","HLA-DRA(dist=38887),HLA-DRB5(dist=33420)","HLA-DQA1(dist=789)","HLA-DRB1(dist=866)","HLA-DRB1(dist=6)","CXCL12(dist=1115),TMEM72-AS1(dist=424812)","PSMD3","CCDC162P","ARL5C","LINC01565(dist=28495),RPN1(dist=15388)","CHD7","MPZL3","ATAT1(NM_001318763:c.*1315A&gt;C,NM_001190724:c.*1315A&gt;C,NM_001254952:c.*768A&gt;C,NM_001318762:c.*1315A&gt;C,NM_024909:c.*768A&gt;C)","HERPUD1(dist=5814),CETP(dist=11272)","C5orf67","FRMD5(NM_001286491:c.-16875A&gt;G)","MSL2(dist=17182),PCCB(dist=36808)","PCNX3","CETP","FAM13A","TMEM176A(dist=27243),AOC1(dist=20156)","ANXA9","TDRD15","PSMA5","CYB561D1(NM_001134402:c.*1429A&gt;G,NM_001134400:c.*1429A&gt;G,NM_001134403:c.*1821A&gt;G,NM_182580:c.*1429A&gt;G,NM_001134404:c.*1913A&gt;G)","S1PR1(NM_001400:c.-9G&gt;T,NM_001320730:c.-9G&gt;T)","LOC541472(dist=676)","PDYN-AS1","PVT1","HLA-H(dist=1453),HCG4B(dist=32060)","FUT7","LINC01767(dist=26082),PLPP3(dist=52713)","ZC3H7B(dist=4021),TEF(dist=3165)","CCHCR1","ADO(NM_032804:c.*2306C&gt;T)","RPH3A","CXCL12","PSMD3","CCDC162P","CCHCR1","HLA-DQB1-AS1","ARL5C","LINC01565(dist=28495),RPN1(dist=15388)","CHD7","RMC1","CCHCR1","CXCL12","PSMD3","CCDC162P","ARL5C","LINC01565(dist=28495),RPN1(dist=15388)","CHD7","ACTN1-AS1(dist=35823),DCAF5(dist=27595)","AK3(dist=38009),RCL1(dist=12782)","ITPR3","KLF9(dist=40711),TRPM3(dist=73692)","RAD51B","MIR4432HG(dist=7736),BCL11A(dist=51409)","MBLAC1(dist=15977),LAMTOR4(dist=4448)","STAG3L5P-PVRIG2P-PILRB(dist=2917),PILRA(dist=2739)","ZAN","ACHE(dist=36283),MUC3A(dist=16257)","MED24","CD70(dist=767)","GLB1","CRNN(dist=88494),LCE5A(dist=8046)","PGLYRP3","CCL20(dist=22449),DAW1(dist=31371)","NRROS","IL4R(dist=5549),IL21R(dist=31847)","KIF3A","HLA-DRB1(dist=16745),HLA-DQA1(dist=30825)","IL4R","LOC101927166(dist=20305),HOXB1(dist=42181)","CAMK4","SFXN2","LOC105375787(dist=40106),LINC01300(dist=43626)","CXCL12(dist=1115),TMEM72-AS1(dist=424812)","SMG6","PSMD3","CCDC162P","NDFIP1(dist=13959),SPRY4(dist=142036)","MUC22","MIR588(dist=475689),RSPO3(dist=158269)","EYA1(dist=31909),MSC(dist=261539)","LOC105378933(dist=5390),WARS2(dist=24421)","LOC646588(dist=28875),MIR148A(dist=130925)"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Traits<\/th>\n      <th>factors<\/th>\n      <th>SNP_gtexid<\/th>\n      <th>SNP_rsid<\/th>\n      <th>BETA_gwas<\/th>\n      <th>P_gwas<\/th>\n      <th>BETA_assoc<\/th>\n      <th>P_assoc<\/th>\n      <th>fdr_assoc<\/th>\n      <th>snp_ld<\/th>\n      <th>ld_r2<\/th>\n      <th>cis_eqtl<\/th>\n      <th>cis_genes_hgnc<\/th>\n      <th>cis_genes_ensembl<\/th>\n      <th>func<\/th>\n      <th>func_gene<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":20,"order":[],"autoWidth":false,"orderClasses":false,"columnDefs":[{"orderable":false,"targets":0}],"lengthMenu":[10,20,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="enrichment-analysis" class="section level3">
<h3>Enrichment analysis</h3>
<p>There are some promising trait-factor pairs when considering SNPs at fdr&lt;0.2 (i.e. BMI-LV14, Ulcerative colitist - LV77, Lymphocyte-LV66,76, Neutrophil - LV43, RBC-LV73, Asthma - LV47, WBC-LV11,68). When relaxing the cutoff to fdr&lt;0.5, there appeared some other promising pairs(i.e. Eocinophil count - LV9, Granulocyte count - LV43,96, Myeloid white cell count -LV43, WBC-LV43).</p>
<p>I did ORA for all the factors mentioned above with WebGestalt. The analysis settings: for each factor, I sorted the genes their absolute loading values, taking the top 5% as the gene set. The function database here are Reactome pathway; Disgenet + GLAD4U + OMIM disease dataset; geneontology biological process. The reference set affy hugene 2 0 st v1. Minimum number of genes for a category is 5, maximum number of genes for a category is 2000(default settings). All categories that have fdr&lt;0.01 are listed. If all the categories don’t have fdr&lt;0.01, I just listed the top 20 categories that related to each factors.</p>
<p><a href="http://www.webgestalt.org/results/1603472381/#">BMI - LV14</a></p>
<p><a href="http://www.webgestalt.org/results/1603472369/#">Eocinophil - LV9</a></p>
<p><a href="http://www.webgestalt.org/results/1603502010/#">Ulcerative colitist - LV77</a></p>
<p><a href="http://www.webgestalt.org/results/1603472386/#">Granulocyte/Neutrophil/WBC/Myeloid white cell - LV43</a></p>
<p><a href="http://www.webgestalt.org/results/1603472435/#">Granulocyte - LV96</a></p>
<p><a href="http://www.webgestalt.org/results/1603502255/#">Lymphocyte-LV66</a></p>
<p><a href="http://www.webgestalt.org/results/1603472431/#">Lymphocyte-LV76</a></p>
<p><a href="http://www.webgestalt.org/results/1603502267/#">RBC - LV73</a></p>
<p><a href="http://www.webgestalt.org/results/1603472390/#">Asthma - LV47</a></p>
<p><a href="http://www.webgestalt.org/results/1603472375/#">WBC - LV11</a></p>
<p><a href="http://www.webgestalt.org/results/1603472406/#">WBC - LV68</a></p>
</div>
<div id="effect-size-plots" class="section level3">
<h3>Effect size plots</h3>
<p>For each trait, I made a plot of association with LV(indicating by beta in GWAS) vs association with trait(indicating by ln(odds ratio) or beta in GWAS) to show if the variants have the correlated effect direction. The effect sizes of Catalog GWAS and factor association tests are harmonized by <a href="https://mrcieu.github.io/TwoSampleMR/articles/introduction.html">TwoSampleMR</a> R package to make the effect alleles in these two analysis identical. The LVs have more than one significant SNPs with FDR&lt;0.2 are included in the plotting.Besides, for each plots, I fitted the points with intercept = 0. The pvalues and r-squared are shown on the plots.</p>
<div id="asthma_2" class="section level4">
<h4>Asthma_2</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv50_asthma_2_pval5e8_peer_d1k_ageco.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv57_asthma_2_pval5e8_peer_d1k_ageco.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv93_asthma_2_pval5e8_peer_d1k_ageco.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv110_asthma_2_pval5e8_peer_d1k_ageco.jpeg" width="30%">
</figure>
</div>
<div id="bmi" class="section level4">
<h4>BMI</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv14_bmi_pval5e8_peer_d1k_ageco.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv79_bmi_pval5e8_peer_d1k_ageco.jpeg" width="30%">
</figure>
</div>
<div id="eosinophil-count" class="section level4">
<h4>Eosinophil count</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv9_eo_pval5e8_peer_d1k_ageco.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv57_eo_pval5e8_peer_d1k_ageco.jpeg" width="30%">
</figure>
</div>
<div id="crohns-disease" class="section level4">
<h4>Crohn’s disease</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv75_EUR.CD_pval5e8_peer_d1k_ageco.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv91_EUR.CD_pval5e8_peer_d1k_ageco.jpeg" width="30%">
</figure>
</div>
<div id="ibd" class="section level4">
<h4>IBD</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv91_EUR.IBD_pval5e8_peer_d1k_ageco.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv120_EUR.IBD_pval5e8_peer_d1k_ageco.jpeg" width="30%">
</figure>
</div>
<div id="ulcerative-colitist" class="section level4">
<h4>Ulcerative colitist</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv77_EUR.UC_pval5e8_peer_d1k_ageco.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv110_EUR.UC_pval5e8_peer_d1k_ageco.jpeg" width="30%">
</figure>
</div>
<div id="granulocyte-count" class="section level4">
<h4>granulocyte count</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_gran_pval5e8_peer_d1k_ageco.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv79_gran_pval5e8_peer_d1k_ageco.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv96_gran_pval5e8_peer_d1k_ageco.jpeg" width="30%">
</figure>
</div>
<div id="hdl" class="section level4">
<h4>HDL</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv1_HDL_pval5e8_peer_d1k_ageco.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv9_HDL_pval5e8_peer_d1k_ageco.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv38_HDL_pval5e8_peer_d1k_ageco.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv108_HDL_pval5e8_peer_d1k_ageco.jpeg" width="30%">
</figure>
</div>
<div id="ldl" class="section level4">
<h4>LDL</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv8_LDL_pval5e8_peer_d1k_ageco.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv19_LDL_pval5e8_peer_d1k_ageco.jpeg" width="30%">
</figure>
</div>
<div id="lymphocyte-count" class="section level4">
<h4>lymphocyte count</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv72_lymph_pval5e8_peer_d1k_ageco.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv20_lymph_pval5e8_peer_d1k_ageco.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv44_lymph_pval5e8_peer_d1k_ageco.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv66_lymph_pval5e8_peer_d1k_ageco.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv76_lymph_pval5e8_peer_d1k_ageco.jpeg" width="30%">
</figure>
</div>
<div id="myeloid-white-cell-count" class="section level4">
<h4>myeloid white cell count</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_myeloid_wbc_pval5e8_peer_d1k_ageco.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv79_myeloid_wbc_pval5e8_peer_d1k_ageco.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv72_myeloid_wbc_pval5e8_peer_d1k_ageco.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv88_myeloid_wbc_pval5e8_peer_d1k_ageco.jpeg" width="30%">
</figure>
</div>
<div id="neutrophil-count" class="section level4">
<h4>neutrophil count</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv79_neut_pval5e8_peer_d1k_ageco.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_neut_pval5e8_peer_d1k_ageco.jpeg" width="30%">
</figure>
</div>
<div id="plt" class="section level4">
<h4>plt</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv50_plt_pval5e8_peer_d1k_ageco.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv81_plt_pval5e8_peer_d1k_ageco.jpeg" width="30%">
</figure>
</div>
<div id="rbc" class="section level4">
<h4>rbc</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv19_rbc_pval5e8_peer_d1k_ageco.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv41_rbc_pval5e8_peer_d1k_ageco.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv73_rbc_pval5e8_peer_d1k_ageco.jpeg" width="30%">
</figure>
</div>
<div id="t2d_2" class="section level4">
<h4>T2D_2</h4>
<p><img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv9_T2D_2_pval5e8_peer_d1k_ageco.jpeg" width="30%"></p>
</div>
<div id="t2d" class="section level4">
<h4>T2D</h4>
<p><img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv19_T2D_pval5e8_peer_d1k_ageco.jpeg" width="30%"></p>
</div>
<div id="asthma" class="section level4">
<h4>asthma</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv24_ukb.allasthma_pval5e8_peer_d1k_ageco.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv94_ukb.allasthma_pval5e8_peer_d1k_ageco.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv102_ukb.allasthma_pval5e8_peer_d1k_ageco.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv47_ukb.allasthma_pval5e8_peer_d1k_ageco.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv58_ukb.allasthma_pval5e8_peer_d1k_ageco.jpeg" width="30%">
</figure>
</div>
<div id="wbc" class="section level4">
<h4>wbc</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_wbc_pval5e8_peer_d1k_ageco.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv11_wbc_pval5e8_peer_d1k_ageco.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv68_wbc_pval5e8_peer_d1k_ageco.jpeg" width="30%">
</figure>
</div>
<div id="whr" class="section level4">
<h4>WHR</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv10_WHR_pval5e8_peer_d1k_ageco.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv39_WHR_pval5e8_peer_d1k_ageco.jpeg" width="30%">
</figure>
</div>
</div>
<div id="resampling" class="section level3">
<h3>Resampling</h3>
<p>For some promising trait-factor pairs, I did resampling. I resampled the SNPs without replacement, I fitted the points with intercept = 0 again and recorded the pvalues and r-squared. The resampling was repeated 1000 times. The following plots are the resampling results.</p>
<p>I made a histogram to show the pvalues/rsquared distribution from resampling. The red line in the plots are the pvalue/rsquared in the origin analysis. The p_mean/r_mean values in the plots are the mean values of the resampling. The ‘prob of getting more extreme values’ is computed by: (number of more extreme values)/(times of resampling)</p>
<div id="asthma_2-1" class="section level4">
<h4>Asthma_2</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_peer_50_asthma_2.jpeg" width="40%">
</figure>
</div>
<div id="bmi-1" class="section level4">
<h4>BMI</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_peer_14_bmi.jpeg" width="40%">
</figure>
</div>
<div id="ulcerative-colitist-1" class="section level4">
<h4>Ulcerative colitist</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_peer_77_EUR.UC.jpeg" width="40%">
</figure>
</div>
<div id="lymphocyte-count-1" class="section level4">
<h4>lymphocyte count</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_peer_66_lymph.jpeg" width="40%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_peer_76_lymph.jpeg" width="40%">
</figure>
</div>
<div id="red-blood-cell-count" class="section level4">
<h4>red blood cell count</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_peer_73_rbc.jpeg" width="40%">
</figure>
</div>
<div id="asthma-1" class="section level4">
<h4>asthma</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_peer_47_asthma.jpeg" width="40%">
</figure>
</div>
<div id="white-blood-cell-count" class="section level4">
<h4>white blood cell count</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_peer_11_wbc.jpeg" width="40%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_peer_68_wbc.jpeg" width="40%">
</figure>
</div>
</div>
<div id="effect-size-plots--more-snps" class="section level3">
<h3>Effect size plots- more SNPs</h3>
<p>For some promising trait-factor pairs , I relaxed the fdr threshold of the SNPs that used to make effect size plots(from 0.2 to 0.3/0.5)</p>
<div id="asthma_2-2" class="section level4">
<h4>Asthma_2</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv50_asthma_2_pval5e8_peer_d1k_ageco.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv50_asthma_2_pval5e8_peer_d1k_ageco_fdr03.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv50_asthma_2_pval5e8_peer_d1k_ageco_fdr05.jpeg" width="30%">
</figure>
</div>
<div id="bmi-2" class="section level4">
<h4>BMI</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv14_bmi_pval5e8_peer_d1k_ageco.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv14_bmi_pval5e8_peer_d1k_ageco_fdr03.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv14_bmi_pval5e8_peer_d1k_ageco_fdr05.jpeg" width="30%">
</figure>
</div>
<div id="eosinophil-count-1" class="section level4">
<h4>Eosinophil count</h4>
<p>The Eo ~lv9 is not very promising pairs when considering SNPs at fdr&lt;0.2, but the fitting result at fdr &lt;0.5 is better than the former result . So I post the plots here too.</p>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv9_eo_pval5e8_peer_d1k_ageco.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv9_eo_pval5e8_peer_d1k_ageco_fdr03.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv9_eo_pval5e8_peer_d1k_ageco_fdr05.jpeg" width="30%">
</figure>
</div>
<div id="ulcerative-colitist-2" class="section level4">
<h4>Ulcerative colitist</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv77_EUR.UC_pval5e8_peer_d1k_ageco.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv77_EUR.UC_pval5e8_peer_d1k_ageco_fdr03.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv77_EUR.UC_pval5e8_peer_d1k_ageco_fdr05.jpeg" width="30%">
</figure>
</div>
<div id="granulocyte-count-1" class="section level4">
<h4>Granulocyte count</h4>
<p>The gran ~lv43,96 is not very promising pairs when considering SNPs at fdr&lt;0.2, but the fitting result at fdr &lt;0.5 is better than the former result . So I post the plots here too.</p>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_gran_pval5e8_peer_d1k_ageco.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_gran_pval5e8_peer_d1k_ageco_fdr03.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_gran_pval5e8_peer_d1k_ageco_fdr05.jpeg" width="30%">
</figure>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv96_gran_pval5e8_peer_d1k_ageco.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv96_gran_pval5e8_peer_d1k_ageco_fdr03.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv96_gran_pval5e8_peer_d1k_ageco_fdr05.jpeg" width="30%">
</figure>
</div>
<div id="lymphocyte-count-2" class="section level4">
<h4>lymphocyte count</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv66_lymph_pval5e8_peer_d1k_ageco.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv66_lymph_pval5e8_peer_d1k_ageco_fdr03.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv66_lymph_pval5e8_peer_d1k_ageco_fdr05.jpeg" width="30%">
</figure>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv76_lymph_pval5e8_peer_d1k_ageco.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv76_lymph_pval5e8_peer_d1k_ageco_fdr03.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv76_lymph_pval5e8_peer_d1k_ageco_fdr05.jpeg" width="30%">
</figure>
</div>
<div id="myeloid-white-cell-count-1" class="section level4">
<h4>Myeloid white cell count</h4>
<p>The myeloid_wbc ~lv43 is not very promising pairs when considering SNPs at fdr&lt;0.2, but the fitting result at fdr &lt;0.5 is better than the former result . So I post the plots here too.</p>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_myeloid_wbc_pval5e8_peer_d1k_ageco.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_myeloid_wbc_pval5e8_peer_d1k_ageco_fdr03.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_myeloid_wbc_pval5e8_peer_d1k_ageco_fdr05.jpeg" width="30%">
</figure>
</div>
<div id="neutrophil-count-1" class="section level4">
<h4>Neutrophil count</h4>
<p>The neut ~lv43 is not very promising pairs when considering SNPs at fdr&lt;0.2, but the fitting result at fdr &lt;0.5 is better than the former result . So I post the plots here too.</p>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_neut_pval5e8_peer_d1k_ageco.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_neut_pval5e8_peer_d1k_ageco_fdr03.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_neut_pval5e8_peer_d1k_ageco_fdr05.jpeg" width="30%">
</figure>
</div>
<div id="red-blood-cell-count-1" class="section level4">
<h4>red blood cell count</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv73_rbc_pval5e8_peer_d1k_ageco.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv73_rbc_pval5e8_peer_d1k_ageco_fdr03.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv73_rbc_pval5e8_peer_d1k_ageco_fdr05.jpeg" width="30%">
</figure>
</div>
<div id="asthma-2" class="section level4">
<h4>asthma</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv47_ukb.allasthma_pval5e8_peer_d1k_ageco.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv47_ukb.allasthma_pval5e8_peer_d1k_ageco_fdr03.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv47_ukb.allasthma_pval5e8_peer_d1k_ageco_fdr05.jpeg" width="30%">
</figure>
</div>
<div id="white-blood-cell-count-1" class="section level4">
<h4>white blood cell count</h4>
<p>The wbc ~lv43 is not very promising pairs when considering SNPs at fdr&lt;0.2, but the fitting result at fdr &lt;0.5 is better than the former result . So I post the plots here too.</p>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv11_wbc_pval5e8_peer_d1k_ageco.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv11_wbc_pval5e8_peer_d1k_ageco_fdr03.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv11_wbc_pval5e8_peer_d1k_ageco_fdr05.jpeg" width="30%">
</figure>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv68_wbc_pval5e8_peer_d1k_ageco.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv68_wbc_pval5e8_peer_d1k_ageco_fdr03.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv68_wbc_pval5e8_peer_d1k_ageco_fdr05.jpeg" width="30%">
</figure>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_wbc_pval5e8_peer_d1k_ageco.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_wbc_pval5e8_peer_d1k_ageco_fdr03.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_wbc_pval5e8_peer_d1k_ageco_fdr05.jpeg" width="30%">
</figure>
</div>
</div>
<div id="effect-size-plots---checking-the-reverse-causality" class="section level3">
<h3>Effect size plots - checking the reverse causality</h3>
<p>To check if the effect size correlation is due to reverse causality: i.e. trait -&gt; LV (trait causally affect LV), instead of LV -&gt; trait (which is what we like to see). I used all SNPs associated with traits(pval&lt;5E-8). The x-axis is the effects of these SNPs on trait, and y-axis is the effects on LV.</p>
<p>Some pair show p &lt; 0.05, the result may be driven by the possible causal effect of LV -&gt; trait. To test this, I removed the SNPs that are associated with LVs at FDR &lt; 0.2 and made the plots again.</p>
<div id="asthma_2-3" class="section level4">
<h4>Asthma_2</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv50_asthma_2_pval5e8_peer_d1k_ageco_reverse.jpeg" width="30%">
</figure>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv50_asthma_2_pval5e8_peer_d1k_ageco_reverse_remove.jpeg" width="30%">
</figure>
</div>
<div id="bmi-3" class="section level4">
<h4>BMI</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv14_bmi_pval5e8_peer_d1k_ageco_reverse.jpeg" width="30%">
</figure>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv14_bmi_pval5e8_peer_d1k_ageco_reverse_remove.jpeg" width="30%">
</figure>
</div>
<div id="eosinophil-count-2" class="section level4">
<h4>Eosinophil count</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv9_eo_pval5e8_peer_d1k_ageco_reverse.jpeg" width="30%">
</figure>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv9_eo_pval5e8_peer_d1k_ageco_reverse_remove.jpeg" width="30%">
</figure>
</div>
<div id="ulcerative-colistist" class="section level4">
<h4>Ulcerative colistist</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv77_EUR.UC_pval5e8_peer_d1k_ageco_reverse.jpeg" width="30%">
</figure>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv77_EUR.UC_pval5e8_peer_d1k_ageco_reverse_remove.jpeg" width="30%">
</figure>
</div>
<div id="granulocyte-count-2" class="section level4">
<h4>Granulocyte count</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_gran_pval5e8_peer_d1k_ageco_reverse.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv96_gran_pval5e8_peer_d1k_ageco_reverse.jpeg" width="30%">
</figure>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_gran_pval5e8_peer_d1k_ageco_reverse_remove.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv96_gran_pval5e8_peer_d1k_ageco_reverse_remove.jpeg" width="30%">
</figure>
</div>
<div id="lymphocyte-count-3" class="section level4">
<h4>lymphocyte count</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv66_lymph_pval5e8_peer_d1k_ageco_reverse.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv76_lymph_pval5e8_peer_d1k_ageco_reverse.jpeg" width="30%">
</figure>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv66_lymph_pval5e8_peer_d1k_ageco_reverse_remove.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv76_lymph_pval5e8_peer_d1k_ageco_reverse_remove.jpeg" width="30%">
</figure>
</div>
<div id="myeloid-white-cell-count-2" class="section level4">
<h4>Myeloid white cell count</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_myeloid_wbc_pval5e8_peer_d1k_ageco_reverse.jpeg" width="30%">
</figure>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_myeloid_wbc_pval5e8_peer_d1k_ageco_reverse_remove.jpeg" width="30%">
</figure>
</div>
<div id="neutrophil-count-2" class="section level4">
<h4>Neutrophil count</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_neut_pval5e8_peer_d1k_ageco_reverse.jpeg" width="30%">
</figure>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_neut_pval5e8_peer_d1k_ageco_reverse_remove.jpeg" width="30%">
</figure>
</div>
<div id="red-blood-cell-count-2" class="section level4">
<h4>red blood cell count</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv73_rbc_pval5e8_peer_d1k_ageco_reverse.jpeg" width="30%">
</figure>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv73_rbc_pval5e8_peer_d1k_ageco_reverse_remove.jpeg" width="30%">
</figure>
</div>
<div id="asthma-3" class="section level4">
<h4>asthma</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv47_ukb.allasthma_pval5e8_peer_d1k_ageco_reverse.jpeg" width="30%">
</figure>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv47_ukb.allasthma_pval5e8_peer_d1k_ageco_reverse_remove.jpeg" width="30%">
</figure>
</div>
<div id="white-blood-cell-count-2" class="section level4">
<h4>white blood cell count</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv11_wbc_pval5e8_peer_d1k_ageco_reverse.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv68_wbc_pval5e8_peer_d1k_ageco_reverse.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_wbc_pval5e8_peer_d1k_ageco_reverse.jpeg" width="30%">
</figure>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv11_wbc_pval5e8_peer_d1k_ageco_reverse_remove.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv68_wbc_pval5e8_peer_d1k_ageco_reverse_remove.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_wbc_pval5e8_peer_d1k_ageco_reverse_remove.jpeg" width="30%">
</figure>
</div>
</div>
<div id="qqplots---promising-pairs" class="section level3">
<h3>QQplots - promising pairs</h3>
<div id="asthma_2-4" class="section level4">
<h4>Asthma_2</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qq_unif_lv50_Asthma_2_peer_d1k_ageco.jpeg" width="30%">
</figure>
</div>
<div id="bmi-4" class="section level4">
<h4>BMI</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qq_unif_lv14_bmi_peer_d1k_ageco.jpeg" width="30%">
</figure>
</div>
<div id="eosinophil-count-3" class="section level4">
<h4>Eosinophil count</h4>
<p>The Eo ~lv9 is not very promising pairs when considering SNPs at fdr&lt;0.2, but the fitting result at fdr &lt;0.5 is better than the former result . So I post the plots here too.</p>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qq_unif_lv9_eo_peer_d1k_ageco.jpeg" width="30%">
</figure>
</div>
<div id="ulcerative-colitist-3" class="section level4">
<h4>Ulcerative colitist</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qq_unif_lv77_EUR.UC_peer_d1k_ageco.jpeg" width="30%">
</figure>
</div>
<div id="granulocyte-count-3" class="section level4">
<h4>Granulocyte count</h4>
<p>The gran ~lv43,96 is not very promising pairs when considering SNPs at fdr&lt;0.2, but the fitting result at fdr &lt;0.5 is better than the former result . So I post the plots here too.</p>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qq_unif_lv43_gran_peer_d1k_ageco.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qq_unif_lv96_gran_peer_d1k_ageco.jpeg" width="30%">
</figure>
</div>
<div id="lymphocyte-count-4" class="section level4">
<h4>lymphocyte count</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qq_unif_lv66_lymph_peer_d1k_ageco.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qq_unif_lv76_lymph_peer_d1k_ageco.jpeg" width="30%">
</figure>
</div>
<div id="myeloid-white-cell-count-3" class="section level4">
<h4>Myeloid white cell count</h4>
<p>The myeloid_wbc ~lv43 is not very promising pairs when considering SNPs at fdr&lt;0.2, but the fitting result at fdr &lt;0.5 is better than the former result . So I post the plots here too.</p>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qq_unif_lv43_myeloid_wbc_peer_d1k_ageco.jpeg" width="30%">
</figure>
</div>
<div id="neutrophil-count-3" class="section level4">
<h4>Neutrophil count</h4>
<p>The neut ~lv43 is not very promising pairs when considering SNPs at fdr&lt;0.2, but the fitting result at fdr &lt;0.5 is better than the former result . So I post the plots here too.</p>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qq_unif_lv43_neut_peer_d1k_ageco.jpeg" width="30%">
</figure>
</div>
<div id="red-blood-cell-count-3" class="section level4">
<h4>red blood cell count</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qq_unif_lv73_rbc_peer_d1k_ageco.jpeg" width="30%">
</figure>
</div>
<div id="asthma-4" class="section level4">
<h4>asthma</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qq_unif_lv47_ukb.allasthma_peer_d1k_ageco.jpeg" width="30%">
</figure>
</div>
<div id="white-blood-cell-count-3" class="section level4">
<h4>white blood cell count</h4>
<p>The wbc ~lv43 is not very promising pairs when considering SNPs at fdr&lt;0.2, but the fitting result at fdr &lt;0.5 is better than the former result . So I post the plots here too.</p>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qq_unif_lv11_wbc_peer_d1k_ageco.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qq_unif_lv68_wbc_peer_d1k_ageco.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qq_unif_lv43_wbc_peer_d1k_ageco.jpeg" width="30%">
</figure>
</div>
</div>
<div id="colocalization" class="section level3">
<h3>Colocalization</h3>
<p>The colocalization analysis was performed using the approximate Bayes factor test implemented in the Coloc package. Coloc computes five posterior probabilities (PP0, PP1, PP2, PP3 and PP4), each corresponding to a hypothesis: H0, no association with either trait; H1, association with trait 1 but not with trait 2; H2, association with trait 2 but not with trait 1; H3, association with trait 1 and trait 2, two independent SNPs; H4, association with trait 1 and trait 2, one shared SNP. We ran Coloc with the default parameters and used PP4 to assess evidence of colocalization. We visualized the colocalization of factor - QTLs and GWAS associations using the <a href="https://github.com/boxiangliu/locuscomparer">LocusCompareR package</a>.</p>
<p>Chromosome selection: I first sorted the SNPs by their pvalues from factor association tests, then I selected the first two SNPs and chose their chromosome to do the colocalization test adn visualization.</p>
<div id="asthma_2-5" class="section level4">
<h4>Asthma_2</h4>
<p>LV50</p>
<table>
<caption>pvalues in coloclization</caption>
<colgroup>
<col width="11%" />
<col width="23%" />
<col width="65%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="left"></th>
<th align="left">note</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">nsnps</td>
<td align="left">486</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">PP.H0.abf</td>
<td align="left">4.83941533135763e-06</td>
<td align="left">no association with either trait</td>
</tr>
<tr class="odd">
<td align="left">PP.H1.abf</td>
<td align="left">1.89300837494408e-07</td>
<td align="left">association with trait 1 but not with trait 2</td>
</tr>
<tr class="even">
<td align="left">PP.H2.abf</td>
<td align="left">0.925530825013238</td>
<td align="left">association with trait 2 but not with trait 1</td>
</tr>
<tr class="odd">
<td align="left">PP.H3.abf</td>
<td align="left">0.0361651984388548</td>
<td align="left">association with trait 1 and trait 2,two independent SNPs</td>
</tr>
<tr class="even">
<td align="left">PP.H4.abf</td>
<td align="left">0.0382989478317374</td>
<td align="left">association with trait 1 and trait 2, one shared SNP</td>
</tr>
</tbody>
</table>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/asthma_2_50_5866925.coloc_peer.jpeg" width="80%">
</figure>
<table>
<caption>pvalues in coloclization</caption>
<colgroup>
<col width="11%" />
<col width="23%" />
<col width="65%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="left"></th>
<th align="left">note</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">nsnps</td>
<td align="left">673</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">PP.H0.abf</td>
<td align="left">4.20278123503938e-07</td>
<td align="left">no association with either trait</td>
</tr>
<tr class="odd">
<td align="left">PP.H1.abf</td>
<td align="left">2.02466203032563e-08</td>
<td align="left">association with trait 1 but not with trait 2</td>
</tr>
<tr class="even">
<td align="left">PP.H2.abf</td>
<td align="left">0.896384639487441</td>
<td align="left">association with trait 2 but not with trait 1</td>
</tr>
<tr class="odd">
<td align="left">PP.H3.abf</td>
<td align="left">0.0431222437699809</td>
<td align="left">association with trait 1 and trait 2,two independent SNPs</td>
</tr>
<tr class="even">
<td align="left">PP.H4.abf</td>
<td align="left">0.0604926762178344</td>
<td align="left">association with trait 1 and trait 2, one shared SNP</td>
</tr>
</tbody>
</table>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/asthma_2_50_80386214.coloc_peer.jpeg" width="80%">
</figure>
</div>
<div id="bmi-5" class="section level4">
<h4>BMI</h4>
<p>LV14</p>
<table>
<caption>pvalues in coloclization</caption>
<colgroup>
<col width="11%" />
<col width="23%" />
<col width="65%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="left"></th>
<th align="left">note</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">nsnps</td>
<td align="left">127</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">PP.H0.abf</td>
<td align="left">0.0876139284655906</td>
<td align="left">no association with either trait</td>
</tr>
<tr class="odd">
<td align="left">PP.H1.abf</td>
<td align="left">0.000651648193592992</td>
<td align="left">association with trait 1 but not with trait 2</td>
</tr>
<tr class="even">
<td align="left">PP.H2.abf</td>
<td align="left">0.857580708973003</td>
<td align="left">association with trait 2 but not with trait 1</td>
</tr>
<tr class="odd">
<td align="left">PP.H3.abf</td>
<td align="left">0.00633062528836015</td>
<td align="left">association with trait 1 and trait 2,two independent SNPs</td>
</tr>
<tr class="even">
<td align="left">PP.H4.abf</td>
<td align="left">0.047823089079453</td>
<td align="left">association with trait 1 and trait 2, one shared SNP</td>
</tr>
</tbody>
</table>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/bmi_14_18344015.coloc_peer.jpeg" width="80%">
</figure>
<table>
<caption>pvalues in coloclization</caption>
<colgroup>
<col width="11%" />
<col width="23%" />
<col width="65%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="left"></th>
<th align="left">note</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">nsnps</td>
<td align="left">177</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">PP.H0.abf</td>
<td align="left">1.1343708338057e-08</td>
<td align="left">no association with either trait</td>
</tr>
<tr class="odd">
<td align="left">PP.H1.abf</td>
<td align="left">9.80140597466771e-11</td>
<td align="left">association with trait 1 but not with trait 2</td>
</tr>
<tr class="even">
<td align="left">PP.H2.abf</td>
<td align="left">0.948145462445653</td>
<td align="left">association with trait 2 but not with trait 1</td>
</tr>
<tr class="odd">
<td align="left">PP.H3.abf</td>
<td align="left">0.00814864032368767</td>
<td align="left">association with trait 1 and trait 2,two independent SNPs</td>
</tr>
<tr class="even">
<td align="left">PP.H4.abf</td>
<td align="left">0.0437058857889365</td>
<td align="left">association with trait 1 and trait 2, one shared SNP</td>
</tr>
</tbody>
</table>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/bmi_14_85758440.coloc_peer.jpeg" width="80%">
</figure>
</div>
<div id="ulcerative-colitist-4" class="section level4">
<h4>Ulcerative colitist</h4>
<p>lv77</p>
<table>
<caption>pvalues in coloclization</caption>
<colgroup>
<col width="11%" />
<col width="23%" />
<col width="65%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="left"></th>
<th align="left">note</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">nsnps</td>
<td align="left">527</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">PP.H0.abf</td>
<td align="left">2.69025336435801e-34</td>
<td align="left">no association with either trait</td>
</tr>
<tr class="odd">
<td align="left">PP.H1.abf</td>
<td align="left">1.11678684479166e-35</td>
<td align="left">association with trait 1 but not with trait 2</td>
</tr>
<tr class="even">
<td align="left">PP.H2.abf</td>
<td align="left">0.923318774730398</td>
<td align="left">association with trait 2 but not with trait 1</td>
</tr>
<tr class="odd">
<td align="left">PP.H3.abf</td>
<td align="left">0.0382907228379451</td>
<td align="left">association with trait 1 and trait 2,two independent SNPs</td>
</tr>
<tr class="even">
<td align="left">PP.H4.abf</td>
<td align="left">0.0383905024316569</td>
<td align="left">association with trait 1 and trait 2, one shared SNP</td>
</tr>
</tbody>
</table>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/EUR.UC_77_32483933.coloc_peer.jpeg" width="80%">
</figure>
<table>
<caption>pvalues in coloclization</caption>
<colgroup>
<col width="11%" />
<col width="23%" />
<col width="65%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="left"></th>
<th align="left">note</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">nsnps</td>
<td align="left">1959</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">PP.H0.abf</td>
<td align="left">1.04159326845804e-42</td>
<td align="left">no association with either trait</td>
</tr>
<tr class="odd">
<td align="left">PP.H1.abf</td>
<td align="left">1.39587558379482e-43</td>
<td align="left">association with trait 1 but not with trait 2</td>
</tr>
<tr class="even">
<td align="left">PP.H2.abf</td>
<td align="left">0.846217398506928</td>
<td align="left">association with trait 2 but not with trait 1</td>
</tr>
<tr class="odd">
<td align="left">PP.H3.abf</td>
<td align="left">0.113364135973167</td>
<td align="left">association with trait 1 and trait 2,two independent SNPs</td>
</tr>
<tr class="even">
<td align="left">PP.H4.abf</td>
<td align="left">0.0404184655199037</td>
<td align="left">association with trait 1 and trait 2, one shared SNP</td>
</tr>
</tbody>
</table>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/outputEUR.UC_77_32636617.coloc_peer.jpeg" width="80%">
</figure>
</div>
<div id="lymphocyte-count-5" class="section level4">
<h4>lymphocyte count</h4>
<p>lv66</p>
<table>
<caption>pvalues in coloclization</caption>
<colgroup>
<col width="11%" />
<col width="22%" />
<col width="65%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="left"></th>
<th align="left">note</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">nsnps</td>
<td align="left">656</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">PP.H0.abf</td>
<td align="left">0.914227993994197</td>
<td align="left">no association with either trait</td>
</tr>
<tr class="odd">
<td align="left">PP.H1.abf</td>
<td align="left">0.0511759375107596</td>
<td align="left">association with trait 1 but not with trait 2</td>
</tr>
<tr class="even">
<td align="left">PP.H2.abf</td>
<td align="left">0.0302135843760485</td>
<td align="left">association with trait 2 but not with trait 1</td>
</tr>
<tr class="odd">
<td align="left">PP.H3.abf</td>
<td align="left">0.00168857842041938</td>
<td align="left">association with trait 1 and trait 2,two independent SNPs</td>
</tr>
<tr class="even">
<td align="left">PP.H4.abf</td>
<td align="left">0.00269390569857591</td>
<td align="left">association with trait 1 and trait 2, one shared SNP</td>
</tr>
</tbody>
</table>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/lymph_66_137031531.coloc_peer.jpeg" width="80%">
</figure>
<table>
<caption>pvalues in coloclization</caption>
<colgroup>
<col width="11%" />
<col width="23%" />
<col width="65%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="left"></th>
<th align="left">note</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">nsnps</td>
<td align="left">30</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">PP.H0.abf</td>
<td align="left">0.995412325574295</td>
<td align="left">no association with either trait</td>
</tr>
<tr class="odd">
<td align="left">PP.H1.abf</td>
<td align="left">0.00293182681284496</td>
<td align="left">association with trait 1 but not with trait 2</td>
</tr>
<tr class="even">
<td align="left">PP.H2.abf</td>
<td align="left">0.00149470204967773</td>
<td align="left">association with trait 2 but not with trait 1</td>
</tr>
<tr class="odd">
<td align="left">PP.H3.abf</td>
<td align="left">4.24550428538446e-06</td>
<td align="left">association with trait 1 and trait 2,two independent SNPs</td>
</tr>
<tr class="even">
<td align="left">PP.H4.abf</td>
<td align="left">0.000156900058896911</td>
<td align="left">association with trait 1 and trait 2, one shared SNP</td>
</tr>
</tbody>
</table>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/lymph_66_29892532.coloc_peer.jpeg" width="80%">
</figure>
<p>lv76</p>
<table>
<caption>pvalues in coloclization</caption>
<colgroup>
<col width="11%" />
<col width="23%" />
<col width="65%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="left"></th>
<th align="left">note</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">nsnps</td>
<td align="left">422</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">PP.H0.abf</td>
<td align="left">8.03177768881441e-24</td>
<td align="left">no association with either trait</td>
</tr>
<tr class="odd">
<td align="left">PP.H1.abf</td>
<td align="left">4.65471991572009e-25</td>
<td align="left">association with trait 1 but not with trait 2</td>
</tr>
<tr class="even">
<td align="left">PP.H2.abf</td>
<td align="left">0.901065420225599</td>
<td align="left">association with trait 2 but not with trait 1</td>
</tr>
<tr class="odd">
<td align="left">PP.H3.abf</td>
<td align="left">0.0521733987810033</td>
<td align="left">association with trait 1 and trait 2,two independent SNPs</td>
</tr>
<tr class="even">
<td align="left">PP.H4.abf</td>
<td align="left">0.046761180993398</td>
<td align="left">association with trait 1 and trait 2, one shared SNP</td>
</tr>
</tbody>
</table>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/lymph_76_112677823.coloc_peer.jpeg" width="80%">
</figure>
<table>
<caption>pvalues in coloclization</caption>
<colgroup>
<col width="11%" />
<col width="23%" />
<col width="65%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="left"></th>
<th align="left">note</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">nsnps</td>
<td align="left">699</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">PP.H0.abf</td>
<td align="left">0.00604489554081038</td>
<td align="left">no association with either trait</td>
</tr>
<tr class="odd">
<td align="left">PP.H1.abf</td>
<td align="left">0.000375604585947078</td>
<td align="left">association with trait 1 but not with trait 2</td>
</tr>
<tr class="even">
<td align="left">PP.H2.abf</td>
<td align="left">0.748326460541052</td>
<td align="left">association with trait 2 but not with trait 1</td>
</tr>
<tr class="odd">
<td align="left">PP.H3.abf</td>
<td align="left">0.0462989296779427</td>
<td align="left">association with trait 1 and trait 2,two independent SNPs</td>
</tr>
<tr class="even">
<td align="left">PP.H4.abf</td>
<td align="left">0.198954109654248</td>
<td align="left">association with trait 1 and trait 2, one shared SNP</td>
</tr>
</tbody>
</table>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/lymph_76_62808178.coloc_peer.jpeg" width="80%">
</figure>
</div>
<div id="red-blood-cell-count-4" class="section level4">
<h4>red blood cell count</h4>
<p>lv73</p>
<table>
<caption>pvalues in coloclization</caption>
<colgroup>
<col width="11%" />
<col width="23%" />
<col width="65%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="left"></th>
<th align="left">note</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">nsnps</td>
<td align="left">525</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">PP.H0.abf</td>
<td align="left">4.35025968726479e-86</td>
<td align="left">no association with either trait</td>
</tr>
<tr class="odd">
<td align="left">PP.H1.abf</td>
<td align="left">2.13517879197461e-87</td>
<td align="left">association with trait 1 but not with trait 2</td>
</tr>
<tr class="even">
<td align="left">PP.H2.abf</td>
<td align="left">0.919638037276592</td>
<td align="left">association with trait 2 but not with trait 1</td>
</tr>
<tr class="odd">
<td align="left">PP.H3.abf</td>
<td align="left">0.045102083052625</td>
<td align="left">association with trait 1 and trait 2,two independent SNPs</td>
</tr>
<tr class="even">
<td align="left">PP.H4.abf</td>
<td align="left">0.0352598796707963</td>
<td align="left">association with trait 1 and trait 2, one shared SNP</td>
</tr>
</tbody>
</table>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/rbc_73_100749368.coloc_peer.jpeg" width="80%">
</figure>
<table>
<caption>pvalues in coloclization</caption>
<colgroup>
<col width="11%" />
<col width="23%" />
<col width="65%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="left"></th>
<th align="left">note</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">nsnps</td>
<td align="left">548</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">PP.H0.abf</td>
<td align="left">0.939203328600614</td>
<td align="left">no association with either trait</td>
</tr>
<tr class="odd">
<td align="left">PP.H1.abf</td>
<td align="left">0.0374365853167522</td>
<td align="left">association with trait 1 but not with trait 2</td>
</tr>
<tr class="even">
<td align="left">PP.H2.abf</td>
<td align="left">0.0209483211437356</td>
<td align="left">association with trait 2 but not with trait 1</td>
</tr>
<tr class="odd">
<td align="left">PP.H3.abf</td>
<td align="left">0.000833420412248058</td>
<td align="left">association with trait 1 and trait 2,two independent SNPs</td>
</tr>
<tr class="even">
<td align="left">PP.H4.abf</td>
<td align="left">0.00157834452665028</td>
<td align="left">association with trait 1 and trait 2, one shared SNP</td>
</tr>
</tbody>
</table>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/rbc_73_100933277.coloc_peer.jpeg" width="80%">
</figure>
</div>
<div id="white-blood-cell-count-4" class="section level4">
<h4>white blood cell count</h4>
<p>lv11</p>
<table>
<caption>pvalues in coloclization</caption>
<colgroup>
<col width="11%" />
<col width="23%" />
<col width="65%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="left"></th>
<th align="left">note</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">nsnps</td>
<td align="left">489</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">PP.H0.abf</td>
<td align="left">0.00506051897138509</td>
<td align="left">no association with either trait</td>
</tr>
<tr class="odd">
<td align="left">PP.H1.abf</td>
<td align="left">0.000178830028493028</td>
<td align="left">association with trait 1 but not with trait 2</td>
</tr>
<tr class="even">
<td align="left">PP.H2.abf</td>
<td align="left">0.925003567522857</td>
<td align="left">association with trait 2 but not with trait 1</td>
</tr>
<tr class="odd">
<td align="left">PP.H3.abf</td>
<td align="left">0.0326509274768405</td>
<td align="left">association with trait 1 and trait 2,two independent SNPs</td>
</tr>
<tr class="even">
<td align="left">PP.H4.abf</td>
<td align="left">0.0371061560004238</td>
<td align="left">association with trait 1 and trait 2, one shared SNP</td>
</tr>
</tbody>
</table>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/wbc_11_102727686.coloc_peer.jpeg" width="80%">
</figure>
<table>
<caption>pvalues in coloclization</caption>
<colgroup>
<col width="11%" />
<col width="22%" />
<col width="65%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="left"></th>
<th align="left">note</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">nsnps</td>
<td align="left">844</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">PP.H0.abf</td>
<td align="left">0.904342003175634</td>
<td align="left">no association with either trait</td>
</tr>
<tr class="odd">
<td align="left">PP.H1.abf</td>
<td align="left">0.0524456584918885</td>
<td align="left">association with trait 1 but not with trait 2</td>
</tr>
<tr class="even">
<td align="left">PP.H2.abf</td>
<td align="left">0.0383806925868013</td>
<td align="left">association with trait 2 but not with trait 1</td>
</tr>
<tr class="odd">
<td align="left">PP.H3.abf</td>
<td align="left">0.00222320954972216</td>
<td align="left">association with trait 1 and trait 2,two independent SNPs</td>
</tr>
<tr class="even">
<td align="left">PP.H4.abf</td>
<td align="left">0.00260843619595447</td>
<td align="left">association with trait 1 and trait 2, one shared SNP</td>
</tr>
</tbody>
</table>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/wbc_11_141296923.coloc_peer.jpeg" width="80%">
</figure>
<p>lv68</p>
<table>
<caption>pvalues in coloclization</caption>
<colgroup>
<col width="11%" />
<col width="23%" />
<col width="65%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="left"></th>
<th align="left">note</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">nsnps</td>
<td align="left">800</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">PP.H0.abf</td>
<td align="left">3.61093816625653e-08</td>
<td align="left">no association with either trait</td>
</tr>
<tr class="odd">
<td align="left">PP.H1.abf</td>
<td align="left">2.85433608632546e-09</td>
<td align="left">association with trait 1 but not with trait 2</td>
</tr>
<tr class="even">
<td align="left">PP.H2.abf</td>
<td align="left">0.799758261820341</td>
<td align="left">association with trait 2 but not with trait 1</td>
</tr>
<tr class="odd">
<td align="left">PP.H3.abf</td>
<td align="left">0.0630812820469758</td>
<td align="left">association with trait 1 and trait 2,two independent SNPs</td>
</tr>
<tr class="even">
<td align="left">PP.H4.abf</td>
<td align="left">0.137160417168966</td>
<td align="left">association with trait 1 and trait 2, one shared SNP</td>
</tr>
</tbody>
</table>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/wbc_68_142168399.coloc_peer.jpeg" width="80%">
</figure>
<table>
<caption>pvalues in coloclization</caption>
<colgroup>
<col width="11%" />
<col width="23%" />
<col width="65%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="left"></th>
<th align="left">note</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">nsnps</td>
<td align="left">61</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">PP.H0.abf</td>
<td align="left">0.992285776285355</td>
<td align="left">no association with either trait</td>
</tr>
<tr class="odd">
<td align="left">PP.H1.abf</td>
<td align="left">0.00422989487427057</td>
<td align="left">association with trait 1 but not with trait 2</td>
</tr>
<tr class="even">
<td align="left">PP.H2.abf</td>
<td align="left">0.00321630024115643</td>
<td align="left">association with trait 2 but not with trait 1</td>
</tr>
<tr class="odd">
<td align="left">PP.H3.abf</td>
<td align="left">1.34558040229751e-05</td>
<td align="left">association with trait 1 and trait 2,two independent SNPs</td>
</tr>
<tr class="even">
<td align="left">PP.H4.abf</td>
<td align="left">0.000254572795195109</td>
<td align="left">association with trait 1 and trait 2, one shared SNP</td>
</tr>
</tbody>
</table>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/wbc_68_31018955.coloc_peer.jpeg" width="80%">
</figure>
</div>
</div>
</div>
<div id="resultsformer-results---pval-5e-8-60-peer-factors-covariants-10-whole-genome-pcs" class="section level2">
<h2>Results(former results) - pval &lt; 5e-8 &amp; 60 PEER factors &amp; covariants: 10 whole genome PCs</h2>
<p>In this part, the PEER factors are consistent with the the <a href="https://xsun1229.github.io/factor_analysis/eval_peer.html">former PEER factors</a></p>
<div id="summary-table-1" class="section level3">
<h3>Summary table</h3>
<p>I used ‘qvalue’ R package to compute the fdr from p-values for each SNP and made a table to show the number of SNPs that pass the threshold. The thresholds are ‘fdr &lt; 0.1’,‘fdr &lt; 0.2’,‘pval &lt; 5e-8’.</p>
<div id="htmlwidget-5ba5287ce9a3b08bdb5d" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-5ba5287ce9a3b08bdb5d">{"x":{"filter":"none","data":[["factor1","factor2","factor3","factor4","factor5","factor6","factor7","factor8","factor9","factor10","factor11","factor12","factor13","factor14","factor15","factor16","factor17","factor18","factor19","factor20","factor21","factor22","factor23","factor24","factor25","factor26","factor27","factor28","factor29","factor30","factor31","factor32","factor33","factor34","factor35","factor36","factor37","factor38","factor39","factor40","factor41","factor42","factor43","factor44","factor45","factor46","factor47","factor48","factor49","factor50","factor51","factor52","factor53","factor54","factor55","factor56","factor57","factor58","factor59","factor60"],[0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,1,1,0,0,0,0,2,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,2,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0],[0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,2,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0],[0,0,0,0,0,2,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,1,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,3,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,1,1,0,0,1,0,1,0,0,0,0,1,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,3,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,1,0,1,0,0,0,0,0,0,0,0,1,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,2,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,3,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,1,0,0,1,0,1,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,3,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,2,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,1,1,0,0,1,0,1,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,3,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,1],[0,0,0,0,0,0,2,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,1,1,0,1,0,0,0,0,0,0,1,0,0,1],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,1,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,1,1,1,0,0,0,0,1,0,2,0,1,0,0,0,0,0,1,0,0,0,0,0,0,0,0,2,0,0,0,0,0,0,0,0,0,0,0,0,2,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,1,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,2,0,0,0,0,0,0,1,0,0,0,0,0,1,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,1,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,3,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,1,1,0,0,1,0,1,1,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,4,0,0,0,0,0,0,0,0,1,0,0,0,0,1,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>bmi_fdr010_num<\/th>\n      <th>bmi_fdr020_num<\/th>\n      <th>bmi_pval5e8_num<\/th>\n      <th>eo_fdr010_num<\/th>\n      <th>eo_fdr020_num<\/th>\n      <th>eo_pval5e8_num<\/th>\n      <th>EUR.CD_fdr010_num<\/th>\n      <th>EUR.CD_fdr020_num<\/th>\n      <th>EUR.CD_pval5e8_num<\/th>\n      <th>EUR.IBD_fdr010_num<\/th>\n      <th>EUR.IBD_fdr020_num<\/th>\n      <th>EUR.IBD_pval5e8_num<\/th>\n      <th>EUR.UC_fdr010_num<\/th>\n      <th>EUR.UC_fdr020_num<\/th>\n      <th>EUR.UC_pval5e8_num<\/th>\n      <th>gran_fdr010_num<\/th>\n      <th>gran_fdr020_num<\/th>\n      <th>gran_pval5e8_num<\/th>\n      <th>lymph_fdr010_num<\/th>\n      <th>lymph_fdr020_num<\/th>\n      <th>lymph_pval5e8_num<\/th>\n      <th>myeloid_wbc_fdr010_num<\/th>\n      <th>myeloid_wbc_fdr020_num<\/th>\n      <th>myeloid_wbc_pval5e8_num<\/th>\n      <th>neut_fdr010_num<\/th>\n      <th>neut_fdr020_num<\/th>\n      <th>neut_pval5e8_num<\/th>\n      <th>plt_fdr010_num<\/th>\n      <th>plt_fdr020_num<\/th>\n      <th>plt_pval5e8_num<\/th>\n      <th>rbc_fdr010_num<\/th>\n      <th>rbc_fdr020_num<\/th>\n      <th>rbc_pval5e8_num<\/th>\n      <th>T2D_fdr010_num<\/th>\n      <th>T2D_fdr020_num<\/th>\n      <th>T2D_pval5e8_num<\/th>\n      <th>ukb.allasthma_fdr010_num<\/th>\n      <th>ukb.allasthma_fdr020_num<\/th>\n      <th>ukb.allasthma_pval5e8_num<\/th>\n      <th>wbc_fdr010_num<\/th>\n      <th>wbc_fdr020_num<\/th>\n      <th>wbc_pval5e8_num<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":20,"columnDefs":[{"className":"dt-right","targets":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[10,20,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="info-table-1" class="section level3">
<h3>Info table</h3>
<p>For each trait, I made a table to show the info of snps with fdr&gt;0.2 in the factor ~ SNP + genotype pcs association test. For each trait,The LVs have more than one significant SNPs with FDR&lt;0.2 are included.</p>
<p>The suffix ’_assoc’ here means that results are from factor ~ SNP + genotype pcs association test. The suffix ’_gwas’ here means results are from original GWAS results files. For EUR.CD, EUR.IBD, EUR.UC, the effectsize_gwas here means ‘ln(OR)’, for others, it means ‘beta’.</p>
<p>‘snp_ld’ here means the snps that in LD with the snp in each line.’ld_r2’ means the LD r-squared which is corresponding to the ‘snp_ld’ column. ‘cis-eqtl’ column indicates whether the snp is a cis-eqtl according to GTEx data. ‘cis_gene_hgnc’ and ‘cis_gene_hgnc’ is the genes that the snp influence when it act as cis-eqtl. ‘func’ and ‘func_gene’ are obtained from ANNOVAR, which indicating the snp function within the genes.</p>
<div id="htmlwidget-0b516daec537d8efb4b6" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-0b516daec537d8efb4b6">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33"],["bmi","bmi","bmi","bmi","eo","eo","EUR.IBD","EUR.IBD","gran","gran","gran","lymph","lymph","myeloid_wbc","myeloid_wbc","myeloid_wbc","neut","neut","neut","plt","plt","rbc","rbc","rbc","rbc","rbc","rbc","ukb.allasthma","ukb.allasthma","wbc","wbc","wbc","wbc"],["LV14","LV14","LV40","LV40","LV57","LV57","LV6","LV6","LV43","LV43","LV43","LV44","LV44","LV43","LV43","LV43","LV43","LV43","LV43","LV7","LV7","LV11","LV11","LV28","LV28","LV41","LV41","LV24","LV24","LV43","LV43","LV43","LV43"],["chr19_18344015_A_G_b38","chr3_85758440_T_G_b38","chr10_85651147_A_G_b38","chr9_126698635_A_G_b38","chr6_30338176_A_G_b38","chr8_129662115_CTACAT_C_b38","chr16_50349166_C_T_b38","chr6_14721666_T_C_b38","chr10_44386212_C_T_b38","chr17_39987826_G_T_b38","chr6_109286353_G_A_b38","chr12_112565131_G_T_b38","chr20_1986403_C_T_b38","chr10_44384812_T_C_b38","chr17_39987826_G_T_b38","chr6_109286353_G_A_b38","chr10_44384812_T_C_b38","chr17_39987826_G_T_b38","chr6_109286353_G_A_b38","chr12_111429489_A_C_b38","chr9_5067832_T_C_b38","chr2_126662850_G_A_b38","chr6_43842065_T_C_b38","chr6_26092913_G_A_b38","chr8_129591389_C_T_b38","chr7_100144470_G_A_b38","chr7_100370748_C_T_b38","chr17_40021896_A_G_b38","chr3_33026456_C_T_b38","chr10_44386212_C_T_b38","chr17_2101961_CAGA_C_b38","chr17_39987826_G_T_b38","chr6_109300291_T_C_b38"],["rs17724992","rs13078960","rs7899106","rs10733682","rs9261659","rs146895157","rs9934775","rs9370774","rs17885289","rs118009374","rs9400271","rs73209657","rs13037430","rs72790862","rs118009374","rs9400271","rs72790862","rs118009374","rs9400271","rs61944375","rs73393498","rs28387101","rs9472135","rs1800562","rs10107630","rs4729578","rs62482199","rs2302776","rs72854772","rs17885289","rs71902577","rs118009374","rs6927569"],["-0.0194","0.0297","0.0395","0.0174","0.03385428","-0.02825609","-0.139595456234745","-0.130703648321444","-0.02326956","-0.08451142","0.03462562","0.08388152","-0.04181903","-0.02433963","-0.08176227","0.03495081","-0.0220783","-0.08172991","0.03322504","0.0887707","-0.09418124","-0.04877128","-0.03799032","-0.04991289","-0.02196291","-0.0328844","0.05599315","-0.0576873","-0.0699484","-0.02783845","-0.03205232","-0.07104802","-0.029455"],["3.415e-08","1.737e-14","2.96e-08","1.83e-08","2.049e-09","3.914e-15","1.715e-09","2.536e-09","2.677e-09","2.712e-18","5.496e-22","5.043e-12","9.109e-09","5.443e-10","3.886e-17","2.861e-22","1.541e-08","2.799e-17","1.848e-20","1.193e-09","5.997e-13","4.07e-08","4.987e-23","5.157e-15","1.209e-09","2.172e-14","6.19e-23","3.4e-20","3.6e-09","1.005e-12","7.173e-13","2.102e-13","2.321e-16"],["-0.009436","0.01367","0.03254","0.01524","-0.04142","0.02331","-0.01138","0.009851","0.01968","-0.05133","0.01899","-0.07439","-0.04047","0.02003","-0.05133","0.01899","0.02003","-0.05133","0.01899","-0.03167","-0.02717","0.01812","-0.008757","-0.02747","-0.01289","-0.01905","0.0267","0.01112","-0.02087","0.01968","-0.02172","-0.05133","-0.0184"],["0.001317","2.573e-05","0.0006072","0.001123","0.0001102","0.0002048","0.0008663","0.001007","0.0002933","0.0003855","0.0001321","0.0004051","0.0001279","0.0002282","0.0003855","0.0001321","0.0002282","0.0003855","0.0001321","0.0001809","0.0001332","0.0001659","0.0006756","0.0001122","0.0004628","0.0004502","0.0003737","0.001032","0.001088","0.0002933","0.0001748","0.0003855","0.0003175"],["0.14785221594302","0.0133554759649393","0.117303745331803","0.185580467973068","0.0977186816723455","0.156674994348622","0.119206855990096","0.128746066202813","0.076857055804324","0.0908671041066077","0.0491766760881988","0.179796082155159","0.0876680086744259","0.0712800215249686","0.0994105273859499","0.051473457228084","0.0751069288468551","0.108579210800774","0.0529680984937062","0.164111009838468","0.133687413881551","0.114959143286857","0.195941440531276","0.0968334658489444","0.190636700891226","0.171619111106082","0.156034049648692","0.157697606655427","0.161952724840863","0.0898553787609563","0.0627910617871024","0.110497419198664","0.0953097752206111"],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,"T",null,"T",null,null,null,"T",null,null,null,null,"T",null,null,"T",null,null,null,null,null,null,"T","T",null,null,null,null,null,"T"],[null,null,null,null,"VARS2,ZFP57,HCG4P5,HLA-T,TRIM26,IFITM4P,RPL23AP1",null,"ADCY7,BRD7",null,null,null,"CD164,SMPD2,CCDC162P",null,null,null,null,"CD164,SMPD2,CCDC162P",null,null,"CD164,SMPD2,CCDC162P",null,null,null,null,null,null,"LAMTOR4,AP4M1","MOSPD3,C7orf43,GIGYF1,EPHB4",null,null,null,null,null,"CD164,SMPD2,CCDC162P"],[null,null,null,null,"ENSG00000137411,ENSG00000204644,ENSG00000227766,ENSG00000231130,ENSG00000234127,ENSG00000235821,ENSG00000239257",null,"ENSG00000121281,ENSG00000166164",null,null,null,"ENSG00000135535,ENSG00000135587,ENSG00000203799",null,null,null,null,"ENSG00000135535,ENSG00000135587,ENSG00000203799",null,null,"ENSG00000135535,ENSG00000135587,ENSG00000203799",null,null,null,null,null,null,"ENSG00000188186,ENSG00000221838","ENSG00000106330,ENSG00000146826,ENSG00000146830,ENSG00000196411",null,null,null,null,null,"ENSG00000135535,ENSG00000135587,ENSG00000203799"],["intronic","intronic","intronic","UTR3","intronic","ncRNA_intronic","intronic","intergenic","intergenic","intronic","ncRNA_exonic","intergenic","ncRNA_intronic","intronic","intronic","ncRNA_exonic","intronic","intronic","ncRNA_exonic","intronic","intronic","intronic","intergenic","exonic","ncRNA_intronic","intergenic","intergenic","intronic","intronic","intergenic","intronic","intronic","ncRNA_exonic"],["PGPEP1","CADM2","GRID1","LMX1B(NM_002316:c.*2184A&gt;G,NM_001174147:c.*2184A&gt;G,NM_001174146:c.*2184A&gt;G)","TRIM39,TRIM39-RPP21","CCDC26","BRD7","LINC01108(dist=436212),JARID2(dist=524403)","CXCL12(dist=1115),TMEM72-AS1(dist=424812)","PSMD3","CCDC162P","PTPN11(dist=55213),RPH3A(dist=10105)","PDYN-AS1","CXCL12","PSMD3","CCDC162P","CXCL12","PSMD3","CCDC162P","SH2B3","JAK2","GYPC","VEGFA(dist=55579),LINC02537(dist=2811)","HFE","CCDC26","MBLAC1(dist=15977),LAMTOR4(dist=4448)","STAG3L5P-PVRIG2P-PILRB(dist=2917),PILRA(dist=2739)","MED24","GLB1","CXCL12(dist=1115),TMEM72-AS1(dist=424812)","SMG6","PSMD3","CCDC162P"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Traits<\/th>\n      <th>factors<\/th>\n      <th>SNP_gtexid<\/th>\n      <th>SNP_rsid<\/th>\n      <th>BETA_gwas<\/th>\n      <th>P_gwas<\/th>\n      <th>BETA_assoc<\/th>\n      <th>P_assoc<\/th>\n      <th>fdr_assoc<\/th>\n      <th>snp_ld<\/th>\n      <th>ld_r2<\/th>\n      <th>cis_eqtl<\/th>\n      <th>cis_genes_hgnc<\/th>\n      <th>cis_genes_ensembl<\/th>\n      <th>func<\/th>\n      <th>func_gene<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":20,"order":[],"autoWidth":false,"orderClasses":false,"columnDefs":[{"orderable":false,"targets":0}],"lengthMenu":[10,20,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="effect-size-plots-1" class="section level3">
<h3>Effect size plots</h3>
<p>For each trait, I made a plot of association with LV(indicating by beta in GWAS) vs association with trait(indicating by ln(odds ratio) or beta in GWAS) to show if the variants have the correlated effect direction. The effect sizes of Catalog GWAS and factor association tests are harmonized by <a href="https://mrcieu.github.io/TwoSampleMR/articles/introduction.html">TwoSampleMR</a> R package to make the effect alleles in these two analysis identical. The LVs have more than one significant SNPs with FDR&lt;0.2 are included in the plotting.Besides, for each plots, I fitted the points with intercept = 0. The pvalues and r-squared are shown on the plots.</p>
<div id="bmi-6" class="section level4">
<h4>BMI</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv14_bmi_pval5e8_d1k_peer.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv40_bmi_pval5e8_d1k_peer.jpeg" width="30%">
</figure>
</div>
<div id="eosinophil-count-4" class="section level4">
<h4>Eosinophil count</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv57_eo_pval5e8_d1k_peer.jpeg" width="30%">
</figure>
</div>
<div id="crohns-disease-1" class="section level4">
<h4>Crohn’s disease</h4>
<p>None of the LVs have &gt;1 SNPs at FDR&lt;0.2.</p>
</div>
<div id="ibd-1" class="section level4">
<h4>IBD</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv6_EUR.IBD_pval5e8_d1k_peer.jpeg" width="30%">
</figure>
</div>
<div id="ulcerative-colitist-5" class="section level4">
<h4>Ulcerative colitist</h4>
<p>None of the LVs have &gt;1 SNPs at FDR&lt;0.2.</p>
</div>
<div id="granulocyte-count-4" class="section level4">
<h4>granulocyte count</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_gran_pval5e8_d1k_peer.jpeg" width="30%">
</figure>
</div>
<div id="lymphocyte-count-6" class="section level4">
<h4>lymphocyte count</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv44_lymph_pval5e8_d1k_peer.jpeg" width="30%">
</figure>
</div>
<div id="myeloid-white-cell-count-4" class="section level4">
<h4>myeloid white cell count</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_myeloid_wbc_pval5e8_d1k_peer.jpeg" width="30%">
</figure>
</div>
<div id="neutrophil-count-4" class="section level4">
<h4>neutrophil count</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_neut_pval5e8_d1k_peer.jpeg" width="30%">
</figure>
</div>
<div id="plt-1" class="section level4">
<h4>plt</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv7_plt_pval5e8_d1k_peer.jpeg" width="30%">
</figure>
</div>
<div id="rbc-1" class="section level4">
<h4>rbc</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv11_rbc_pval5e8_d1k_peer.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv28_rbc_pval5e8_d1k_peer.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv41_rbc_pval5e8_d1k_peer.jpeg" width="30%">
</figure>
</div>
<div id="t2d-1" class="section level4">
<h4>T2D</h4>
<p>None of the LVs have &gt;1 SNPs at FDR&lt;0.2.</p>
</div>
<div id="asthma-5" class="section level4">
<h4>asthma</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv24_ukb.allasthma_pval5e8_d1k_peer.jpeg" width="30%">
</figure>
</div>
<div id="wbc-1" class="section level4">
<h4>wbc</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_wbc_pval5e8_d1k_peer.jpeg" width="30%">
</figure>
</div>
</div>
<div id="effect-size-plots-more-snps" class="section level3">
<h3>effect size plots – more SNPs</h3>
<p>For some promising trait-factor pairs (i.e. BMI-LV14, LV40,LV76, RBC-LV28), I relaxed the fdr threshold of the SNPs that used to make effect size plots(from 0.2 to 0.3/0.5)</p>
<p>I also made a plot to show the the distribution of the SNPs’ fdr.</p>
<div id="bmi-7" class="section level4">
<h4>BMI</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv14_bmi_pval5e8_d1k_peer.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv14_bmi_pval5e8_d1k_peer_fdr03.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv14_bmi_pval5e8_d1k_peer_fdr05.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/fdrhist_lv14_bmi_pval5e8_d1k_peer.jpeg" width="30%">
</figure>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv40_bmi_pval5e8_d1k_peer.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv40_bmi_pval5e8_d1k_peer_fdr03.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv40_bmi_pval5e8_d1k_peer_fdr05.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/fdrhist_lv40_bmi_pval5e8_d1k_peer.jpeg" width="30%">
</figure>
</div>
<div id="ibd-2" class="section level4">
<h4>IBD</h4>
<p>The IBD~lv6 is not very promising pairs when considering SNPs at fdr&lt;0.2, but the fitting result at fdr &lt;0.5 is better than the former result. So I post the plots here too.</p>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv6_EUR.IBD_pval5e8_d1k_peer.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv6_EUR.IBD_pval5e8_d1k_peer_fdr03.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv6_EUR.IBD_pval5e8_d1k_peer_fdr05.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/fdrhist_lv6_EUR.IBD_pval5e8_d1k_peer.jpeg" width="30%">
</figure>
</div>
<div id="myeloid-white-cell-count-5" class="section level4">
<h4>myeloid white cell count</h4>
<p>The myeloid white cell count~lv6 is not very promising pairs when considering SNPs at fdr&lt;0.2, but the fitting result at fdr &lt;0.5 is better than the former result. So I post the plots here too.</p>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_myeloid_wbc_pval5e8_d1k_peer.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_myeloid_wbc_pval5e8_d1k_peer_fdr03.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_myeloid_wbc_pval5e8_d1k_peer_fdr05.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/fdrhist_lv43_myeloid_wbc_pval5e8_d1k_peer.jpeg" width="30%">
</figure>
</div>
<div id="neutrophil-count-5" class="section level4">
<h4>neutrophil count</h4>
<p>The neutrophil count~lv6 is not very promising pairs when considering SNPs at fdr&lt;0.2, but the fitting result at fdr &lt;0.5 is better than the former result. So I post the plots here too.</p>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_neut_pval5e8_d1k_peer.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_neut_pval5e8_d1k_peer_fdr03.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_neut_pval5e8_d1k_peer_fdr05.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/fdrhist_lv43_neut_pval5e8_d1k_peer.jpeg" width="30%">
</figure>
</div>
<div id="red-blood-cell-count-5" class="section level4">
<h4>red blood cell count</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv28_rbc_pval5e8_d1k_peer.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv28_rbc_pval5e8_d1k_peer_fdr03.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv28_rbc_pval5e8_d1k_peer_fdr05.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/fdrhist_lv28_rbc_pval5e8_d1k_peer.jpeg" width="30%">
</figure>
</div>
<div id="wbc-2" class="section level4">
<h4>wbc</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_wbc_pval5e8_d1k_peer.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_wbc_pval5e8_d1k_peer_fdr03.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_wbc_pval5e8_d1k_peer_fdr05.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/fdrhist_lv43_wbc_pval5e8_d1k_peer.jpeg" width="30%">
</figure>
</div>
</div>
<div id="effect-size-plots---checking-the-reverse-causality-1" class="section level3">
<h3>Effect size plots - checking the reverse causality</h3>
<p>To check if the effect size correlation is due to reverse causality: i.e. trait -&gt; LV (trait causally affect LV), instead of LV -&gt; trait (which is what we like to see). I used all SNPs associated with traits(pval&lt;5E-8). The x-axis is the effects of these SNPs on trait, and y-axis is the effects on LV.</p>
<p>Some pair show p &lt; 0.05, the result may be driven by the possible causal effect of LV -&gt; trait. To test this, I removed the SNPs that are associated with LVs at FDR &lt; 0.2 and made the plots again.</p>
<div id="bmi-8" class="section level4">
<h4>BMI</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv14_bmi_pval5e8_d1k_reverse_peer.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv40_bmi_pval5e8_d1k_reverse_peer.jpeg" width="30%">
</figure>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv14_bmi_pval5e8_d1k_reverse_remove_peer.jpeg" width="30%"> <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv40_bmi_pval5e8_d1k_reverse_remove_peer.jpeg" width="30%">
</figure>
</div>
<div id="ibd-3" class="section level4">
<h4>IBD</h4>
<p>The IBD~lv6 is not very promising pairs when considering SNPs at fdr&lt;0.2, but the fitting result at fdr &lt;0.5 is better than the former result. So I post the plots here too.</p>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv6_EUR.IBD_pval5e8_d1k_reverse_peer.jpeg" width="30%">
</figure>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv6_EUR.IBD_pval5e8_d1k_reverse_remove_peer.jpeg" width="30%">
</figure>
</div>
<div id="myeloid-white-cell-count-6" class="section level4">
<h4>myeloid white cell count</h4>
<p>The myeloid white cell count~lv6 is not very promising pairs when considering SNPs at fdr&lt;0.2, but the fitting result at fdr &lt;0.5 is better than the former result. So I post the plots here too.</p>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_myeloid_wbc_pval5e8_d1k_reverse_peer.jpeg" width="30%">
</figure>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_myeloid_wbc_pval5e8_d1k_reverse_remove_peer.jpeg" width="30%">
</figure>
</div>
<div id="neutrophil-count-6" class="section level4">
<h4>neutrophil count</h4>
<p>The neutrophil count~lv6 is not very promising pairs when considering SNPs at fdr&lt;0.2, but the fitting result at fdr &lt;0.5 is better than the former result. So I post the plots here too.</p>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_neut_pval5e8_d1k_reverse_peer.jpeg" width="30%">
</figure>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_neut_pval5e8_d1k_reverse_remove_peer.jpeg" width="30%">
</figure>
</div>
<div id="red-blood-cell-count-6" class="section level4">
<h4>red blood cell count</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv28_rbc_pval5e8_d1k_reverse_peer.jpeg" width="30%">
</figure>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv28_rbc_pval5e8_d1k_reverse_remove_peer.jpeg" width="30%">
</figure>
</div>
<div id="wbc-3" class="section level4">
<h4>wbc</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_wbc_pval5e8_d1k_reverse_peer.jpeg" width="30%">
</figure>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_wbc_pval5e8_d1k_reverse_remove_peer.jpeg" width="30%">
</figure>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] workflowr_1.6.2

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.5        rstudioapi_0.11   whisker_0.3-2     knitr_1.30       
 [5] magrittr_1.5      R6_2.4.1          rlang_0.4.8       highr_0.8        
 [9] stringr_1.4.0     tools_3.6.1       DT_0.15           xfun_0.18        
[13] git2r_0.26.1      crosstalk_1.1.0.1 htmltools_0.5.0   ellipsis_0.3.1   
[17] rprojroot_1.3-2   yaml_2.2.1        digest_0.6.25     tibble_3.0.3     
[21] lifecycle_0.2.0   crayon_1.3.4      later_1.1.0.1     htmlwidgets_1.5.2
[25] vctrs_0.3.4       promises_1.1.1    fs_1.5.0          glue_1.4.2       
[29] evaluate_0.14     rmarkdown_1.13    stringi_1.5.3     compiler_3.6.1   
[33] pillar_1.4.6      backports_1.1.10  jsonlite_1.7.1    httpuv_1.5.1     
[37] pkgconfig_2.0.3  </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
