<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="XSun" />

<meta name="date" content="2021-08-05" />

<title>Results for KEGG PCs from CEDAR data set (all KEGG pathways included)</title>

<script src="site_libs/jquery-1.12.4/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.2/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.15/datatables.js"></script>
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.1.0.1/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">factor_analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/xsun1229/factor_analysis">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Results for KEGG PCs from CEDAR data set (all KEGG pathways included)</h1>
<h4 class="author">XSun</h4>
<h4 class="date">2021-08-05</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-08-14
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>factor_analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div class="panel-group" id="workflowr-checks">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200623code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200623)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200623code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200623)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it difficult for you and others to run your code on a different machine. Change the absolute path(s) below to the suggested relative path(s) to make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/factor_analysis/output/pwy_name_bcell_cedar_qc_kegg_pruned_10peer.rdata
</td>
<td style="text-align:left;">
output/pwy_name_bcell_cedar_qc_kegg_pruned_10peer.rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/factor_analysis/output/mbe_bcell_cedar_qc_kegg_pruned_10peer.rdata
</td>
<td style="text-align:left;">
output/mbe_bcell_cedar_qc_kegg_pruned_10peer.rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/factor_analysis/output/pwy_name_cd14_cedar_qc_kegg_pruned_10peer.rdata
</td>
<td style="text-align:left;">
output/pwy_name_cd14_cedar_qc_kegg_pruned_10peer.rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/factor_analysis/output/mbe_cd14_cedar_qc_kegg_pruned_10peer.rdata
</td>
<td style="text-align:left;">
output/mbe_cd14_cedar_qc_kegg_pruned_10peer.rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/factor_analysis/output/pwy_name_cd15_cedar_qc_kegg_pruned_10peer.rdata
</td>
<td style="text-align:left;">
output/pwy_name_cd15_cedar_qc_kegg_pruned_10peer.rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/factor_analysis/output/mbe_cd15_cedar_qc_kegg_pruned_10peer.rdata
</td>
<td style="text-align:left;">
output/mbe_cd15_cedar_qc_kegg_pruned_10peer.rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/factor_analysis/output/pwy_name_tcell_cedar_qc_kegg_pruned_10peer.rdata
</td>
<td style="text-align:left;">
output/pwy_name_tcell_cedar_qc_kegg_pruned_10peer.rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/factor_analysis/output/mbe_tcell_cedar_qc_kegg_pruned_10peer.rdata
</td>
<td style="text-align:left;">
output/mbe_tcell_cedar_qc_kegg_pruned_10peer.rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/factor_analysis/output/pwy_name_thy_cedar_qc_kegg_pruned_10peer.rdata
</td>
<td style="text-align:left;">
output/pwy_name_thy_cedar_qc_kegg_pruned_10peer.rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/factor_analysis/output/mbe_thy_cedar_qc_kegg_pruned_10peer.rdata
</td>
<td style="text-align:left;">
output/mbe_thy_cedar_qc_kegg_pruned_10peer.rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/factor_analysis/output/pwy_name_plt_cedar_qc_kegg_pruned_10peer.rdata
</td>
<td style="text-align:left;">
output/pwy_name_plt_cedar_qc_kegg_pruned_10peer.rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/factor_analysis/output/mbe_plt_cedar_qc_kegg_pruned_10peer.rdata
</td>
<td style="text-align:left;">
output/mbe_plt_cedar_qc_kegg_pruned_10peer.rdata
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomxsun1229factoranalysistreee3611441275544b1d7bc60f3fe874a210d64d823targetblanke361144a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/xsun1229/factor_analysis/tree/e3611441275544b1d7bc60f3fe874a210d64d823" target="_blank">e361144</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomxsun1229factoranalysistreee3611441275544b1d7bc60f3fe874a210d64d823targetblanke361144a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/xsun1229/factor_analysis/tree/e3611441275544b1d7bc60f3fe874a210d64d823" target="_blank">e361144</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    analysis/.Rhistory
    Ignored:    output/.Rhistory

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/cedar_kegg_26_qc_d1k_pruned_10peer.Rmd</code>) and HTML (<code>docs/cedar_kegg_26_qc_d1k_pruned_10peer.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/fea1a60015f6b16ff6bca9cf1b15c26d121e099a/docs/cedar_kegg_26_qc_d1k_pruned_10peer.html" target="_blank">fea1a60</a>
</td>
<td>
XSun
</td>
<td>
2021-08-12
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/4fcd34b491cb2374612f81712363eb613173694e/analysis/cedar_kegg_26_qc_d1k_pruned_10peer.Rmd" target="_blank">4fcd34b</a>
</td>
<td>
XSun
</td>
<td>
2021-08-12
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/99de2c5ae8e523b1c6943d386b528cb7b18b8e3c/analysis/cedar_kegg_26_qc_d1k_pruned_10peer.Rmd" target="_blank">99de2c5</a>
</td>
<td>
XSun
</td>
<td>
2021-08-08
</td>
<td>
update
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="overview" class="section level1">
<h1><strong>Overview</strong></h1>
<p>In this part, when doing association test for PCs ~ variants, we added N PEER factors as covariates(N &lt;= <strong>10</strong>). The details are described in <a href="https://xsun1229.github.io/factor_analysis/factors_traits.html#513_association_test_covariates">5.1.3 N PEER factors</a></p>
<p>The other parts of the workflow are the same as described in <a href="https://xsun1229.github.io/factor_analysis/factors_traits.html#3_pipeline">main page</a>.</p>
</div>
<div id="enrichment-summary" class="section level1">
<h1><strong>Enrichment Summary</strong></h1>
<p><strong>The enrichment results are computed from adaptive resampling</strong></p>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/CEDAR_cell_sum_kegg_qc_pruned_10peer.jpg" width="150%">
</figure>
</div>
<div id="barplot-to-show-the-number-of-snps-have-fdr-0.2-that-each-pair-contains" class="section level1">
<h1><strong>Barplot to show the number of SNPs have FDR &lt;0.2 that each pair contains</strong></h1>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/all_kegg_hist_snp_all_CEDAR_10peer.jpg" width="150%">
</figure>
</div>
<div id="detailed-results-for-6-cell-types" class="section level1">
<h1><strong>Detailed results for 6 cell types</strong></h1>
<div id="b-cell" class="section level2">
<h2><strong>B cell</strong></h2>
<div id="effectsize-fitting-summary" class="section level3">
<h3>effectsize fitting summary</h3>
<div id="htmlwidget-c3f4891eb8e9a8c25198" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-c3f4891eb8e9a8c25198">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64"],["EUR.CD_pwy276_pc4","EUR.IBD_pwy44_pc4","T1D_pwy151_pc4","baso_pwy223_pc3","baso_pwy282_pc5","eo_pwy203_pc3","eo_pwy222_pc4","eo_pwy285_pc4","hct_pwy273_pc3","hct_pwy56_pc2","hgb_pwy262_pc2","hgb_pwy273_pc3","lymph_pwy106_pc3","lymph_pwy140_pc1","lymph_pwy163_pc5","lymph_pwy187_pc1","lymph_pwy18_pc3","lymph_pwy193_pc1","lymph_pwy196_pc1","lymph_pwy267_pc1","lymph_pwy270_pc1","lymph_pwy280_pc4","lymph_pwy281_pc2","lymph_pwy283_pc1","lymph_pwy291_pc1","lymph_pwy312_pc1","lymph_pwy319_pc1","lymph_pwy322_pc1","lymph_pwy331_pc1","mchc_pwy169_pc4","mcv_pwy118_pc1","mcv_pwy145_pc2","mcv_pwy186_pc5","mcv_pwy274_pc4","mcv_pwy289_pc4","mono_pwy129_pc4","mpv_pwy168_pc5","mpv_pwy176_pc4","mpv_pwy224_pc2","mpv_pwy332_pc4","mpv_pwy82_pc4","myeloid_wbc_pwy200_pc4","myeloid_wbc_pwy234_pc3","pct_pwy130_pc3","pct_pwy286_pc5","pct_pwy292_pc5","pct_pwy317_pc5","pct_pwy321_pc2","plt_pwy100_pc2","plt_pwy119_pc5","plt_pwy147_pc2","plt_pwy162_pc4","rbc_pwy314_pc3","rbc_pwy316_pc4","rdw_pwy112_pc2","ret_pwy101_pc2","ret_pwy101_pc3","ret_pwy168_pc2","ret_pwy186_pc2","ret_pwy222_pc2","ret_pwy310_pc3","ret_pwy332_pc3","wbc_pwy152_pc3","wbc_pwy331_pc4"],["05142","00513","04218","04921","05152","04726","04920","05162","05134","00564","05031","05134","03420","04145","04390","04660","00230","04672","04714","05110","05131","05146","05150","05160","05168","05222","05235","05321","05416","04530","04060","04210","04659","05135","05166","04114","04520","04620","04922","05418","00920","04723","04932","04115","05163","05169","05230","05320","03030","04061","04213","04380","05224","05226","04012","03040","03040","04520","04659","04920","05220","05418","04260","05416"],["Chagas disease","Various types of N-glycan biosynthesis","Cellular senescence","Oxytocin signaling pathway","Tuberculosis","Serotonergic synapse","Adipocytokine signaling pathway","Measles","Legionellosis","Glycerophospholipid metabolism","Amphetamine addiction","Legionellosis","Nucleotide excision repair","Phagosome","Hippo signaling pathway","T cell receptor signaling pathway","Purine metabolism","Intestinal immune network for IgA production","Thermogenesis","Vibrio cholerae infection","Shigellosis","Amoebiasis","Staphylococcus aureus infection","Hepatitis C","Herpes simplex virus 1 infection","Small cell lung cancer","PD-L1 expression and PD-1 checkpoint pathway in cancer","Inflammatory bowel disease","Viral myocarditis","Tight junction","Cytokine-cytokine receptor interaction","Apoptosis","Th17 cell differentiation","Yersinia infection","Human T-cell leukemia virus 1 infection","Oocyte meiosis","Adherens junction","Toll-like receptor signaling pathway","Glucagon signaling pathway","Fluid shear stress and atherosclerosis","Sulfur metabolism","Retrograde endocannabinoid signaling","Non-alcoholic fatty liver disease","p53 signaling pathway","Human cytomegalovirus infection","Epstein-Barr virus infection","Central carbon metabolism in cancer","Autoimmune thyroid disease","DNA replication","Viral protein interaction with cytokine and cytokine receptor","Longevity regulating pathway - multiple species","Osteoclast differentiation","Breast cancer","Gastric cancer","ErbB signaling pathway","Spliceosome","Spliceosome","Adherens junction","Th17 cell differentiation","Adipocytokine signaling pathway","Chronic myeloid leukemia","Fluid shear stress and atherosclerosis","Cardiac muscle contraction","Viral myocarditis"],["58","28","112","79","122","46","48","96","40","45","36","40","41","99","71","76","65","33","126","35","165","41","39","104","294","60","67","38","46","93","101","104","79","89","164","76","40","63","62","83","6","67","108","46","145","158","44","27","35","42","36","86","67","71","55","111","111","40","79","48","62","83","33","46"],[2,2,2,2,2,4,2,6,2,2,3,3,2,6,2,2,7,4,4,6,5,2,6,6,6,3,4,5,3,2,5,6,5,4,7,6,5,4,2,4,2,3,8,3,2,3,4,3,3,2,3,2,7,4,2,2,2,2,2,2,3,7,2,2],[2,2,2,2,2,4,2,6,2,2,3,3,2,6,2,2,7,4,4,6,5,2,6,6,6,3,4,5,3,2,5,6,5,4,7,6,5,4,2,4,2,3,8,3,2,3,4,3,3,2,3,2,7,4,2,2,2,2,2,2,3,7,2,2],[0,0.008,0.001,0.003,0.002,0,0.006,0,0.001,0.001,0,0.001,0.003,0,0.001,0,0.001,0.002,0.004,0,0,0,0,0,0.002,0.001,0.001,0.001,0.001,0.003,0,0,0,0.001,0,0,0.001,0,0.001,0,0.001,0.001,0,0,0.001,0.001,0.001,0.001,0,0.001,0,0.001,0.001,0,0,0.003,0,0,0.001,0.001,0.001,0,0.002,0.002],[0.476,-0.429,-0.457,-0.189,-0.164,-0.118,0.07,-0.105,0.088,0.136,0.067,0.08,0.134,0.083,-0.108,0.1,0.08,0.098,-0.067,0.114,-0.064,0.147,0.098,0.088,0.049,-0.113,0.108,-0.109,-0.085,0.075,-0.079,-0.064,0.127,-0.106,-0.082,-0.107,-0.079,0.094,0.131,-0.067,-0.126,0.103,0.084,0.117,-0.106,0.065,-0.132,0.082,0.075,0.106,-0.115,0.064,-0.092,-0.099,-0.035,-0.034,0.098,0.064,0.06,0.066,-0.063,-0.076,-0.146,-0.067],[1,1,1,1,1,0.994,1,0.936,1,1,1,0.998,1,0.955,1,1,0.884,0.974,0.954,0.943,0.969,1,0.941,0.982,0.865,0.998,0.983,0.956,0.998,1,0.968,0.966,0.985,0.986,0.97,0.95,0.949,0.99,1,0.997,1,0.998,0.873,0.999,1,0.998,0.981,0.999,0.999,1,1,1,0.85,0.993,1,1,1,1,1,1,0.999,0.911,1,1],[0,0,0,0,0.001,0,0,0,0,0,0.001,0.001,0.003,0.001,0.001,0.001,0.001,0.001,0,0,0,0,0.001,0,0,0,0.002,0.002,0.001,0,0.001,0,0,0,0,0.002,0,0,0.001,0,0,0.001,0,0,0,0.001,0.002,0.001,0,0.001,0,0,0,0,0,0.004,0.001,0.001,0.001,0.003,0,0.002,0.001,0.002],[5e-07,0.00082,5e-07,0.00037,0.00056,0.00047,0.0016,0.00077,0.00067,0.00101,0.0014,0.0012,0.0019,0.00032,0.0006,0.0013,0.00057,0.00063,0.0011,0.00031,0.00036,5e-07,0.0011,0.0001,0.00061,0.001,0.0013,0.0011,0.00096,0.0016,0.00056,0.00066,0.00029,0.00049,0.00014,0.0013,0.00066,0.00119,0.00064,0.0001,0.00079,0.0009,0.0015,0.00016,0.00087,0.0011,0.0011,0.0012,0.00032,0.00061,1e-05,0.0011,0.0019,0.00017,0.00015,0.0016,0.00074,0.0003,0.00068,0.00078,0.0017,0.00071,0.0019,0.0014]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>factor<\/th>\n      <th>pwy<\/th>\n      <th>pwy_name<\/th>\n      <th>pwy_gene_num<\/th>\n      <th>num_snp_fdr.0.2<\/th>\n      <th>num_snp_noMHC<\/th>\n      <th>pval_in_fitting<\/th>\n      <th>slope_in_fitting<\/th>\n      <th>rsq_in_fitting<\/th>\n      <th>pval_in_resampling<\/th>\n      <th>pval_in_resampling_2<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[5,6,7,8,9,10,11]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="checking-the-reverse-causality" class="section level3">
<h3>Checking the reverse causality</h3>
<p>**The suffix &quot;_reverse&quot; are the results for traits -&gt; LV (using all candidate SNPs after LD Clumping), the suffix &quot;_rremove&quot; are the results for traits -&gt; LV (using all candidate SNPs after LD Clumping but remove the SNPs have fdr &lt; 0.2)**</p>
<div id="htmlwidget-cfefb73e51b4928e765c" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-cfefb73e51b4928e765c">{"x":{"filter":"none","data":[["EUR.CD_pwy276_pc4","EUR.IBD_pwy44_pc4","T1D_pwy151_pc4","baso_pwy223_pc3","baso_pwy282_pc5","eo_pwy203_pc3","eo_pwy222_pc4","eo_pwy285_pc4","hct_pwy273_pc3","hct_pwy56_pc2","hgb_pwy262_pc2","hgb_pwy273_pc3","lymph_pwy106_pc3","lymph_pwy140_pc1","lymph_pwy163_pc5","lymph_pwy187_pc1","lymph_pwy18_pc3","lymph_pwy193_pc1","lymph_pwy196_pc1","lymph_pwy267_pc1","lymph_pwy270_pc1","lymph_pwy280_pc4","lymph_pwy281_pc2","lymph_pwy283_pc1","lymph_pwy291_pc1","lymph_pwy312_pc1","lymph_pwy319_pc1","lymph_pwy322_pc1","lymph_pwy331_pc1","mchc_pwy169_pc4","mcv_pwy118_pc1","mcv_pwy145_pc2","mcv_pwy186_pc5","mcv_pwy274_pc4","mcv_pwy289_pc4","mono_pwy129_pc4","mpv_pwy168_pc5","mpv_pwy176_pc4","mpv_pwy224_pc2","mpv_pwy332_pc4","mpv_pwy82_pc4","myeloid_wbc_pwy200_pc4","myeloid_wbc_pwy234_pc3","pct_pwy130_pc3","pct_pwy286_pc5","pct_pwy292_pc5","pct_pwy317_pc5","pct_pwy321_pc2","plt_pwy100_pc2","plt_pwy119_pc5","plt_pwy147_pc2","plt_pwy162_pc4","rbc_pwy314_pc3","rbc_pwy316_pc4","rdw_pwy112_pc2","ret_pwy101_pc2","ret_pwy101_pc3","ret_pwy168_pc2","ret_pwy186_pc2","ret_pwy222_pc2","ret_pwy310_pc3","ret_pwy332_pc3","wbc_pwy152_pc3","wbc_pwy331_pc4"],[0.00580490063791972,0.00218519868839717,-0.0212976083177026,0.0609607866301118,0.0114594446381417,0.00322822776638709,0.0197467047518476,0.0315201137310588,0.0664547783629885,-0.0147257547047133,-0.0291544330038462,0.0242745483100553,0.00884542531402666,0.0130627170066511,-0.0373358562740069,-0.0159378309451199,0.00319694355932398,0.0115698219064075,0.0075255254352129,-0.0088420800431166,-0.00304514298834491,-0.02393323467372,-0.00106939117320159,0.0472279857271545,-0.0073288998909895,0.0240684061510059,0.00572723826947197,0.0262053188536306,-0.0288651754858904,0.0115873180062263,-0.0325222952625808,0.0183453742861026,-0.00138531107359263,0.0235075631725792,-0.0423979673982231,0.0242334127274262,-0.0121168370096481,0.0112407474318204,0.0246988847281973,-0.0125423124508153,0.0106697278658849,0.0178343148458247,-0.00772348496082387,0.0193052234137312,-0.0269825708576477,-0.00496387206112203,-0.0119158903646142,0.00758661772948344,0.039914645172443,-0.0161033746548878,0.0105435790123831,0.0380314719133086,-0.00475954713536453,-0.0179371887793831,-0.0330411000506103,-0.0364540959815471,-0.00429522684619599,0.00378768399554374,-0.0173042656186624,0.0276402373116902,0.0368745722289106,0.0839959475551701,0.0120220680368206,-0.0248011237796852],[0.527671889561258,0.958321485102878,0.16589465850094,0.43149156601424,0.934734296611411,0.989458451460856,0.902117141344889,0.921242061296525,0.820391007967357,0.98500316136881,0.857128617133295,0.96376622962463,0.991444996707899,0.991606490650714,0.80631692989543,0.980527532329902,0.981834796483865,0.881345496123739,0.972673944099556,0.822496483406089,0.983614297240451,0.980571125285689,0.997997174709575,0.924599973288764,0.950247562396193,0.866276311849923,0.968897493118365,0.925926413150775,0.973046162731747,0.93660838133488,0.958733444825795,0.903183564524338,0.996886742425513,0.927498133023012,0.815464179728354,0.905834203857853,0.982328208246916,0.944012024699563,0.788030799659424,0.911588403830334,0.899335149934962,0.960115083091628,0.991673576773237,0.928981645118088,0.912126999380814,0.980257821240394,0.989756601845815,0.991152179379923,0.911933448333441,0.984245876352223,0.960980860158785,0.75817193222115,0.978552166063049,0.973496732261221,0.869456509616715,0.90739575018074,0.983880642810185,0.99084175645188,0.851896193792101,0.852521769258632,0.827816613018093,0.429677090407497,0.920677838027924,0.970312648428159],[0.0112745581681364,0.00557650200023296,-0.0251273300261996,0.0578614794497412,0.0197547698390022,0.00441041382509155,-0.0203764365800883,-0.00997055903912525,0.0734375655876001,-0.0536319788921986,-0.0200408637603928,0.0226531109816346,0.0062631427834301,-0.0221096337992837,-0.0337899796221421,-0.0159428426434189,-0.0170892141236187,0.0257460410104926,-0.0104460859866897,-0.00473271366579198,0.0013260243242037,-0.0231078320086393,-0.015527793963658,-0.000202729076852037,0.00482889806464826,0.0395970352204742,0.00769740846952693,0.0187970595594962,-0.0251994040810324,0.00853815648519607,0.00665224310469981,0.00196730627685149,0.0145289103691066,0.0115743579933145,0.00515809809303036,0.0259828068883863,-0.0189170899624642,-0.00267635758613416,0.0250587871355492,0.00511597769475003,0.021671266233974,0.0211796778714692,0.0293018644743732,0.0295795263554801,-0.0285252463475025,0.0111760082494534,-0.0133199080738109,0.018195120300156,0.0425626996899133,-0.0300618627608458,0.0110489612120386,0.0303949710172944,-0.0131543134111425,-0.027557932334151,-0.00451329161308145,0.0183267128013469,-0.0045948903124966,-0.00851095166371074,-0.0165046802242976,0.0161050851962905,0.0157522650068209,0.0469027724560078,0.00696167200596776,0.00652305786153917],[0.206865248356238,0.895259084277298,0.0922873877008054,0.454466552536632,0.888820935164334,0.985456043408839,0.898235601868884,0.97519951379031,0.794973172092042,0.946641737061153,0.902579217674287,0.965411797877591,0.993911151356452,0.985861416503016,0.822497516651434,0.980607192830011,0.904447403661805,0.737161780886285,0.961681295808064,0.899943339871439,0.992896862687794,0.980925608486607,0.970840801957143,0.999687052985207,0.966347995285964,0.787280517915813,0.957303990069238,0.946491367234312,0.976365539599474,0.953766354895052,0.991710828407766,0.989713220235381,0.966793818495511,0.964422531721982,0.977855064332309,0.898653354567627,0.972758275418709,0.986607390674425,0.77998832551486,0.963461144578174,0.794458984164483,0.95322415020363,0.968601477071151,0.889686704976951,0.906519455975444,0.956639923925736,0.988627141755744,0.978158320042545,0.906023163937119,0.971231671034178,0.958822202664967,0.80345816888211,0.939400046614815,0.957332108491394,0.981677667827429,0.953826635292741,0.982859018641206,0.979514010246521,0.860743907714405,0.913525966799329,0.925448201285248,0.65891959721292,0.953470400576864,0.99220419373343]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>mbe_est_reverse<\/th>\n      <th>mbe_pval_reverse<\/th>\n      <th>mbe_est_rremove<\/th>\n      <th>mbe_pval_rremove<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[1,2,3,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div id="monocyte-cd14" class="section level2">
<h2><strong>monocyte CD14</strong></h2>
<div id="effectsize-fitting-summary-1" class="section level3">
<h3>effectsize fitting summary</h3>
<div id="htmlwidget-5ba5287ce9a3b08bdb5d" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-5ba5287ce9a3b08bdb5d">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57"],["EUR.IBD_pwy183_pc5","EUR.IBD_pwy290_pc5","allergy_pwy311_pc4","baso_pwy185_pc1","baso_pwy186_pc1","baso_pwy225_pc1","baso_pwy242_pc1","baso_pwy257_pc4","baso_pwy274_pc1","baso_pwy276_pc1","baso_pwy286_pc1","baso_pwy287_pc1","baso_pwy292_pc1","baso_pwy293_pc1","eo_pwy199_pc4","eo_pwy244_pc3","gran_pwy188_pc3","gran_pwy250_pc2","gran_pwy268_pc2","gran_pwy307_pc2","hct_pwy241_pc1","hct_pwy314_pc2","hgb_pwy135_pc2","hgb_pwy192_pc4","hgb_pwy40_pc3","lymph_pwy152_pc1","lymph_pwy202_pc1","lymph_pwy255_pc1","lymph_pwy66_pc1","mch_pwy112_pc5","mch_pwy183_pc1","mchc_pwy114_pc4","mchc_pwy199_pc2","mchc_pwy284_pc5","mchc_pwy289_pc3","mcv_pwy164_pc1","mcv_pwy189_pc2","mcv_pwy302_pc3","myeloid_wbc_pwy205_pc4","myeloid_wbc_pwy250_pc2","myeloid_wbc_pwy321_pc5","myeloid_wbc_pwy54_pc3","neut_pwy204_pc3","neut_pwy236_pc1","neut_pwy250_pc2","pct_pwy222_pc2","pdw_pwy231_pc4","pdw_pwy268_pc3","pdw_pwy292_pc3","plt_pwy297_pc1","rbc_pwy268_pc1","rdw_pwy88_pc1","ret_pwy291_pc3","ukb.allasthma_pwy302_pc4","wbc_pwy16_pc3","wbc_pwy177_pc3","wbc_pwy250_pc2"],["04650","05167","05221","04658","04659","04923","04962","05014","05135","05142","05163","05164","05169","05170","04722","04966","04662","04975","05120","05217","04961","05224","04137","04670","00500","04260","04725","05010","00630","04012","04650","04015","04722","05161","05166","04392","04664","05212","04728","04975","05320","00562","04727","04934","04975","04920","04929","05120","05169","05204","05120","01521","05168","05212","00190","04621","04975"],["Natural killer cell mediated cytotoxicity","Kaposi sarcoma-associated herpesvirus infection","Acute myeloid leukemia","Th1 and Th2 cell differentiation","Th17 cell differentiation","Regulation of lipolysis in adipocytes","Vasopressin-regulated water reabsorption","Amyotrophic lateral sclerosis","Yersinia infection","Chagas disease","Human cytomegalovirus infection","Influenza A","Epstein-Barr virus infection","Human immunodeficiency virus 1 infection","Neurotrophin signaling pathway","Collecting duct acid secretion","B cell receptor signaling pathway","Fat digestion and absorption","Epithelial cell signaling in Helicobacter pylori infection","Basal cell carcinoma","Endocrine and other factor-regulated calcium reabsorption","Breast cancer","Mitophagy - animal","Leukocyte transendothelial migration","Starch and sucrose metabolism","Cardiac muscle contraction","Cholinergic synapse","Alzheimer disease","Glyoxylate and dicarboxylate metabolism","ErbB signaling pathway","Natural killer cell mediated cytotoxicity","Rap1 signaling pathway","Neurotrophin signaling pathway","Hepatitis B","Human T-cell leukemia virus 1 infection","Hippo signaling pathway - multiple species","Fc epsilon RI signaling pathway","Pancreatic cancer","Dopaminergic synapse","Fat digestion and absorption","Autoimmune thyroid disease","Inositol phosphate metabolism","GABAergic synapse","Cushing syndrome","Fat digestion and absorption","Adipocytokine signaling pathway","GnRH secretion","Epithelial cell signaling in Helicobacter pylori infection","Epstein-Barr virus infection","Chemical carcinogenesis","Epithelial cell signaling in Helicobacter pylori infection","EGFR tyrosine kinase inhibitor resistance","Herpes simplex virus 1 infection","Pancreatic cancer","Oxidative phosphorylation","NOD-like receptor signaling pathway","Fat digestion and absorption"],["77","132","54","65","77","24","33","39","92","68","152","121","155","140","88","17","68","10","53","24","30","69","56","63","23","36","58","119","23","59","77","109","88","113","157","11","50","57","74","10","23","52","37","76","10","47","28","53","155","25","53","53","273","57","84","127","10"],[4,3,2,4,4,4,2,3,2,2,5,6,6,5,4,4,6,5,7,4,2,2,2,2,2,5,7,5,2,3,2,4,2,3,6,2,5,3,4,4,4,4,2,8,5,4,4,7,8,2,3,2,6,5,8,5,6],[4,3,2,4,4,4,2,3,2,2,5,6,6,5,4,4,6,5,7,4,2,2,2,2,2,5,7,5,2,3,2,4,2,3,6,2,5,3,4,4,2,4,2,8,5,4,4,7,8,2,3,2,6,4,8,5,6],[0,0,0.001,0.001,0,0,0.008,0,0.011,0.002,0.001,0,0,0.003,0.001,0,0,0,0,0,0.001,0,0.019,0.009,0.016,0,0.001,0,0.001,0,0.007,0.007,0.006,0,0,0.003,0,0.002,0,0,0,0,0.001,0.001,0,0,0,0,0,0,0.007,0.001,0,0,0,0,0],[-0.515,0.372,0.21,0.125,0.112,0.202,-0.139,0.107,-0.091,-0.121,-0.073,0.09,0.069,-0.068,-0.074,0.141,0.073,-0.125,0.072,-0.096,-0.231,0.058,0.076,-0.068,-0.075,-0.044,0.049,0.03,-0.132,0.127,0.056,0.054,0.035,-0.104,-0.093,-0.136,0.163,0.13,-0.079,-0.134,-0.141,-0.103,0.069,-0.035,-0.125,0.061,-0.138,0.116,-0.071,0.122,0.058,-0.088,0.088,-0.135,0.064,0.066,-0.138],[0.996,1,1,0.985,0.994,0.996,1,1,1,1,0.956,0.988,0.959,0.906,0.983,0.993,0.985,0.999,0.928,0.997,1,1,0.999,1,0.999,0.986,0.882,0.983,1,1,1,0.937,1,0.999,0.966,1,0.973,0.995,0.994,0.995,0.994,0.996,1,0.838,0.991,0.993,0.995,0.944,0.867,1,0.985,1,0.935,0.975,0.845,0.983,0.985],[0.197,0.977,0.929,0.492,0.686,0.327,0.047,0.015,0.614,0.26,0.009,0.686,0.689,0.442,0.271,0.644,0.032,0.081,0.423,0.336,0.076,0.156,0.959,0.732,0.926,0.252,0.255,0.004,0.586,0.217,0.075,0.002,0.098,0.839,0.602,0.827,0.381,0.304,0.86,0.817,0.786,0.396,0.053,0.415,0.674,0.53,0.596,0.402,0.407,0.991,0.899,0.078,0.728,0.209,0.998,0.759,0.133],[0.00017,0.00043,0.00043,0.00103,0.00056,0.00058,0.0018,0.00016,0.00109,0.00048,0.0018,0.00048,0.0012,0.0019,0.0016,0.0014,0.00081,1e-05,0.0018,0.0016,0.00092,0.00077,0.0016,0.0016,0.0011,0.0008,0.0017,0.00108,0.0012,0.00032,0.00076,0.00072,0.0011,0.0011,0.00061,0.0017,0.00088,0.0013,0.00012,0.00066,0.0017,0.00046,0.00097,0.0019,0.00034,0.0017,0.00054,0.00016,0.00028,6e-05,0.00049,0.0019,0.001,0.00057,0.0014,0.001,0.00028]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>factor<\/th>\n      <th>pwy<\/th>\n      <th>pwy_name<\/th>\n      <th>pwy_gene_num<\/th>\n      <th>num_snp_fdr.0.2<\/th>\n      <th>num_snp_noMHC<\/th>\n      <th>pval_in_fitting<\/th>\n      <th>slope_in_fitting<\/th>\n      <th>rsq_in_fitting<\/th>\n      <th>pval_in_resampling<\/th>\n      <th>pval_in_resampling_2<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[5,6,7,8,9,10,11]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="checking-the-reverse-causality-1" class="section level3">
<h3>Checking the reverse causality</h3>
<p>**The suffix &quot;_reverse&quot; are the results for traits -&gt; LV (using all candidate SNPs after LD Clumping), the suffix &quot;_rremove&quot; are the results for traits -&gt; LV (using all candidate SNPs after LD Clumping but remove the SNPs have fdr &lt; 0.2)**</p>
<div id="htmlwidget-0b516daec537d8efb4b6" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-0b516daec537d8efb4b6">{"x":{"filter":"none","data":[["EUR.IBD_pwy183_pc5","EUR.IBD_pwy290_pc5","allergy_pwy311_pc4","baso_pwy185_pc1","baso_pwy186_pc1","baso_pwy225_pc1","baso_pwy257_pc4","baso_pwy274_pc1","baso_pwy276_pc1","baso_pwy287_pc1","baso_pwy292_pc1","eo_pwy199_pc4","eo_pwy244_pc3","gran_pwy188_pc3","gran_pwy250_pc2","hct_pwy241_pc1","hct_pwy314_pc2","hgb_pwy40_pc3","lymph_pwy152_pc1","lymph_pwy202_pc1","lymph_pwy255_pc1","lymph_pwy66_pc1","mch_pwy112_pc5","mch_pwy183_pc1","mchc_pwy114_pc4","mchc_pwy199_pc2","mchc_pwy284_pc5","mchc_pwy289_pc3","mcv_pwy189_pc2","mcv_pwy228_pc2","mcv_pwy302_pc3","myeloid_wbc_pwy205_pc4","myeloid_wbc_pwy250_pc2","myeloid_wbc_pwy54_pc3","neut_pwy204_pc3","neut_pwy250_pc2","pdw_pwy231_pc4","pdw_pwy268_pc3","pdw_pwy292_pc3","plt_pwy297_pc1","rbc_pwy268_pc1","rbc_pwy304_pc1","ret_pwy291_pc3","ukb.allasthma_pwy302_pc4","wbc_pwy16_pc3","wbc_pwy177_pc3","wbc_pwy250_pc2"],[-0.00684364953574024,3.52548825870436e-05,0.00754403996423214,0.175971920504875,-0.0143711259552681,0.0321974360269728,0.0227244951543693,0.0724723772486087,-0.0561108752937969,-0.0412059371531583,-0.021238298127372,-0.0244865973189334,-0.00822447486317657,-0.00482843496851793,-0.000604255449534463,-0.000469096796464896,0.0258951908355165,0.00533473245131688,-0.0108953391801521,0.0471951520809704,0.0449202429799129,-0.061304306090106,-0.0161959490145449,0.0501833390885906,-0.0521764289816034,0.0384080700013421,0.00407393234976894,0.000731388107734432,-0.0238275601835092,-0.00770980088791795,0.0346724158058489,0.00148164671348994,-0.00701779055433249,-0.0235291441293839,-0.0158986529156682,-0.0222921156858842,-0.00999378640301885,-0.0117047092484803,0.0138962786257828,0.0330437867458997,0.0558133618552752,0.0348123448427913,-0.0259101759031743,-0.000698030418501361,0.00151335738256009,-0.0132298989704331,0.00652659652481802],[0.422965687598589,0.998812924099449,0.725489038909774,0.654280140068668,0.837543728829753,0.702505478073232,0.752858232045153,0.352536692166513,0.591925331159634,0.627306179248061,0.903704357705209,0.87530825487419,0.973196006500675,0.988686102307811,0.997085314469074,0.996702095221395,0.937453914323939,0.982356226706339,0.966822736928696,0.579601021757965,0.833373431202,0.767176062508369,0.952533140330691,0.932256651289211,0.749040806242014,0.810385154710243,0.995137048331167,0.994490817446852,0.914691331100313,0.935731935251497,0.92206009028759,0.991782780181497,0.964511441007378,0.992007166708464,0.98929655981146,0.890721707324119,0.932565145174112,0.983426393195529,0.944565708682533,0.982653238945104,0.66270485753269,0.853751321402138,0.96110833392033,0.994706485395309,0.99694602712344,0.991291589595702,0.977270625679226],[-0.00617569058162992,0.000814518999442182,0.00683742734810977,0.210802926554756,-0.0187486637416967,0.0368528460667719,-0.0306884038856943,0.0526240000105673,0.013576715088881,-0.076159628960518,-0.00657890676104955,-0.0205240014466419,-0.0111367584832713,-0.0192863454483856,0.00226721856786227,-0.0127746987787454,0.0260178332997896,-0.00358719674992258,-0.0468186387783831,0.0287977327367024,-0.000245674267361551,-0.0617456223610819,-0.0158292411842638,-0.0553843039030113,-0.043335408295043,-2.17648533862302e-06,-0.0278094969037213,-0.0191191067857375,-0.0175616958432734,-0.00855025673017984,0.0392769964684323,0.0510255558379615,0.006230256596087,-0.0289831042039701,-0.000119036023374264,-0.0198496943683963,0.010300089477834,0.0129467223119555,0.0150960028199449,0.0330009218333664,0.0129225767865719,0.0250541546933043,-0.0263176137754613,0.0016585248051566,-0.00786751298526056,-0.0200897306926713,-0.00694318905306091],[0.460282220515932,0.971581856509188,0.749282383506424,0.588225766157821,0.786145667872747,0.662291992102329,0.670022236759854,0.500714251524959,0.897014279050406,0.371990722863464,0.969800462709551,0.892982212412724,0.96322529574452,0.954041448010055,0.989002814527955,0.910915399628311,0.936921275102588,0.988121229494126,0.857260334782424,0.730054745009361,0.999067730266343,0.760071062835805,0.954171746722067,0.923483306812592,0.790528190775236,0.999988903667273,0.967167444287896,0.85347379280945,0.938694297307624,0.929278040110796,0.909067984383733,0.723410795504861,0.967914540571847,0.989651691951924,0.999921204987674,0.899495697494869,0.931517519897782,0.981424915483328,0.940668783952448,0.982721547296566,0.920560791642091,0.894265296172593,0.960250841514268,0.987537692189531,0.983850826461069,0.986682861099992,0.975414721809452]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>mbe_est_reverse<\/th>\n      <th>mbe_pval_reverse<\/th>\n      <th>mbe_est_rremove<\/th>\n      <th>mbe_pval_rremove<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[1,2,3,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div id="neutrophils-cd15" class="section level2">
<h2><strong>neutrophils CD15</strong></h2>
<div id="effectsize-fitting-summary-2" class="section level3">
<h3>effectsize fitting summary</h3>
<div id="htmlwidget-d0109ea5b7e7d667f148" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-d0109ea5b7e7d667f148">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29"],["EUR.CD_pwy31_pc3","EUR.IBD_pwy322_pc4","baso_pwy322_pc4","eo_pwy258_pc4","hgb_pwy122_pc3","hgb_pwy233_pc3","hgb_pwy248_pc3","hgb_pwy309_pc3","mch_pwy10_pc3","mch_pwy161_pc4","mch_pwy212_pc4","mch_pwy26_pc5","mch_pwy332_pc4","mchc_pwy284_pc4","mcv_pwy183_pc3","mcv_pwy202_pc2","mcv_pwy310_pc2","mcv_pwy311_pc3","pct_pwy142_pc2","pdw_pwy122_pc3","plt_pwy152_pc3","plt_pwy44_pc3","rdw_pwy299_pc5","rdw_pwy92_pc3","ret_pwy234_pc3","ret_pwy332_pc5","ret_pwy94_pc2","wbc_pwy160_pc5","wbc_pwy324_pc4"],["00380","05321","05321","05016","04066","04931","04973","05219","00071","04371","04910","00310","05418","05161","04650","04725","05220","05221","04150","04066","04260","00513","05206","02010","04932","05418","03010","04370","05323"],["Tryptophan metabolism","Inflammatory bowel disease","Inflammatory bowel disease","Huntington disease","HIF-1 signaling pathway","Insulin resistance","Carbohydrate digestion and absorption","Bladder cancer","Fatty acid degradation","Apelin signaling pathway","Insulin signaling pathway","Lysine degradation","Fluid shear stress and atherosclerosis","Hepatitis B","Natural killer cell mediated cytotoxicity","Cholinergic synapse","Chronic myeloid leukemia","Acute myeloid leukemia","mTOR signaling pathway","HIF-1 signaling pathway","Cardiac muscle contraction","Various types of N-glycan biosynthesis","MicroRNAs in cancer","ABC transporters","Non-alcoholic fatty liver disease","Fluid shear stress and atherosclerosis","Ribosome","VEGF signaling pathway","Rheumatoid arthritis"],["15","32","32","146","67","56","22","23","18","69","87","29","84","105","66","48","56","47","81","67","38","24","99","12","101","84","105","35","48"],[2,3,2,4,2,2,2,2,2,4,5,4,2,2,2,3,2,5,4,3,2,3,4,2,2,2,2,2,3],[2,2,1,4,2,2,2,2,2,4,5,4,2,2,2,3,2,5,4,3,2,3,4,2,2,2,2,2,3],[0,0.001,0.001,0,0.006,0.003,0.003,0.007,0.001,0.001,0,0,0.002,0.002,0,0.001,0,0,0,0.002,0.001,0.001,0.001,0.001,0.018,0,0.002,0.001,0],[0.759,-0.469,-0.072,-0.084,0.05,-0.056,-0.069,0.068,0.072,0.067,0.07,0.083,0.067,0.071,0.094,0.104,0.093,-0.07,-0.117,-0.1,-0.099,0.189,-0.066,0.079,-0.103,0.055,0.147,-0.082,-0.066],[1,0.999,1,0.99,1,1,1,1,1,0.985,0.984,0.997,1,1,1,0.999,1,0.97,0.997,0.997,1,0.999,0.985,1,0.999,1,1,1,1],[0,0.001,0.001,0,0.003,0,0,0.002,0.001,0.001,0,0,0,0.002,0,0.002,0.001,0,0,0.001,0.001,0.001,0.001,0,0.001,0,0.001,0.001,0.001],[5e-07,0.00069,0.0011,0.0012,0.0009,0.00022,0.00031,0.0017,0.0016,0.00041,8e-05,0.0011,0.0017,0.0012,0.00065,0.0012,0.00053,0.001,0.001,0.0015,0.0014,0.0016,0.0018,0.00081,0.0011,0.0011,0.0019,0.0017,0.00046]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>factor<\/th>\n      <th>pwy<\/th>\n      <th>pwy_name<\/th>\n      <th>pwy_gene_num<\/th>\n      <th>num_snp_fdr.0.2<\/th>\n      <th>num_snp_noMHC<\/th>\n      <th>pval_in_fitting<\/th>\n      <th>slope_in_fitting<\/th>\n      <th>rsq_in_fitting<\/th>\n      <th>pval_in_resampling<\/th>\n      <th>pval_in_resampling_2<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[5,6,7,8,9,10,11]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="checking-the-reverse-causality-2" class="section level3">
<h3>Checking the reverse causality</h3>
<p>**The suffix &quot;_reverse&quot; are the results for traits -&gt; LV (using all candidate SNPs after LD Clumping), the suffix &quot;_rremove&quot; are the results for traits -&gt; LV (using all candidate SNPs after LD Clumping but remove the SNPs have fdr &lt; 0.2)**</p>
<div id="htmlwidget-7d2b77686a7a668a5a8b" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-7d2b77686a7a668a5a8b">{"x":{"filter":"none","data":[["EUR.CD_pwy31_pc3","EUR.IBD_pwy322_pc4","baso_pwy322_pc4","eo_pwy258_pc4","hgb_pwy122_pc3","hgb_pwy233_pc3","hgb_pwy248_pc3","hgb_pwy309_pc3","mch_pwy10_pc3","mch_pwy161_pc4","mch_pwy212_pc4","mch_pwy26_pc5","mch_pwy332_pc4","mchc_pwy284_pc4","mcv_pwy183_pc3","mcv_pwy202_pc2","mcv_pwy310_pc2","mcv_pwy311_pc3","pct_pwy142_pc2","pdw_pwy122_pc3","plt_pwy152_pc3","plt_pwy44_pc3","rdw_pwy299_pc5","rdw_pwy92_pc3","ret_pwy234_pc3","ret_pwy332_pc5","ret_pwy94_pc2","wbc_pwy160_pc5","wbc_pwy324_pc4"],[0.00253894895772744,-0.00567298174302255,0.028656239265004,0.0246334333508198,-0.0242131142219186,-0.0487346701210321,-0.00752614731171164,0.139354932167617,0.0290858873634345,-0.020565557432823,0.032305201727425,0.0112360206302355,-0.00688001096522228,-0.0710262127408079,-0.0379963048104734,-0.0263428750641097,-0.00686210461552861,-0.01098299508228,-0.0306723926686203,-0.0381517467306747,0.0172795701436925,-0.0207012993247258,0.0314439147752665,0.00460993575241986,-0.0101458286226297,-0.0383837532052702,0.0243551352427644,0.0377690153716213,0.0236835285203014],[0.706085751070376,0.787220311210823,0.963084528025654,0.799273383685441,0.902289266709681,0.598558083082675,0.894335621472986,0.760191934208647,0.93614540931074,0.922903621990397,0.816582958851305,0.902860591143882,0.993788318561542,0.311491384386613,0.725966413779726,0.913627531777036,0.950163950382926,0.994907998988574,0.900021793706587,0.703259398802406,0.78585475939709,0.983468616924092,0.930875921825467,0.970994149012343,0.946165944477365,0.900617936962574,0.919558012364948,0.776647714644205,0.935408894574126],[0.00253124430884966,-0.0100883531182752,0.0288100300697245,0.00718440864140568,-0.020220699602886,-0.003044148060523,0.012878550327617,0.11636355615102,-0.00845467554882262,0.00856126337459173,0.00946456744160606,-0.000430986765637797,-0.0631893896407369,-0.081420222429081,-0.0294200536519504,-0.0258681968625147,-0.0138355369628513,0.0216526480041761,-0.0422078721345613,-0.0296593974134538,0.000911346215708875,-0.0311350377189363,0.0249528092829046,-0.00595117430226821,0.051665159550808,-0.0347998946924726,0.0135148214966883,0.00735089716699555,0.0248616450392127],[0.710329474931724,0.627307631771347,0.962920446960582,0.941089405159578,0.918778270100834,0.973100525912064,0.814891242097107,0.797926634043573,0.981480067890538,0.969118006439371,0.943052157626506,0.996286963377048,0.94250759435422,0.228352545855922,0.785347423928499,0.916787798433889,0.898746726100894,0.989929898973147,0.864081138831512,0.766037191510149,0.98855430530429,0.974742813041136,0.946839646878962,0.962181207297948,0.729242564491446,0.908236715264525,0.955516683205377,0.955742124145658,0.931376168185324]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>mbe_est_reverse<\/th>\n      <th>mbe_pval_reverse<\/th>\n      <th>mbe_est_rremove<\/th>\n      <th>mbe_pval_rremove<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[1,2,3,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div id="t-cellcd4" class="section level2">
<h2><strong>T cell(CD4)</strong></h2>
<div id="effectsize-fitting-summary-3" class="section level3">
<h3>effectsize fitting summary</h3>
<div id="htmlwidget-1e347d3f3a7c82c854df" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-1e347d3f3a7c82c854df">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35"],["EUR.CD_pwy124_pc1","EUR.CD_pwy300_pc1","EUR.UC_pwy188_pc5","T1D_pwy72_pc3","allergy_pwy248_pc5","baso_pwy153_pc1","baso_pwy23_pc2","baso_pwy246_pc2","baso_pwy56_pc4","gran_pwy100_pc3","gran_pwy116_pc5","gran_pwy132_pc1","lymph_pwy162_pc3","lymph_pwy290_pc5","lymph_pwy42_pc1","mcv_pwy139_pc4","mcv_pwy154_pc5","mcv_pwy176_pc5","mcv_pwy273_pc2","mcv_pwy284_pc5","mcv_pwy297_pc2","mcv_pwy327_pc1","neut_pwy304_pc5","pct_pwy279_pc4","pdw_pwy262_pc5","pdw_pwy302_pc3","plt_pwy101_pc5","plt_pwy115_pc3","plt_pwy279_pc4","rdw_pwy294_pc4","rdw_pwy295_pc2","ukb.allasthma_pwy172_pc2","ukb.allasthma_pwy330_pc1","ukb.allasthma_pwy73_pc3","wbc_pwy269_pc1"],["04070","05210","04662","00750","04973","04261","00270","04971","00564","03030","04022","04122","04380","05167","00511","04144","04270","04620","05134","05161","05204","05340","05214","05145","05031","05212","03040","04020","05145","05200","05202","04610","05414","00760","05130"],["Phosphatidylinositol signaling system","Colorectal cancer","B cell receptor signaling pathway","Vitamin B6 metabolism","Carbohydrate digestion and absorption","Adrenergic signaling in cardiomyocytes","Cysteine and methionine metabolism","Gastric acid secretion","Glycerophospholipid metabolism","DNA replication","cGMP-PKG signaling pathway","Sulfur relay system","Osteoclast differentiation","Kaposi sarcoma-associated herpesvirus infection","Other glycan degradation","Endocytosis","Vascular smooth muscle contraction","Toll-like receptor signaling pathway","Legionellosis","Hepatitis B","Chemical carcinogenesis","Primary immunodeficiency","Glioma","Toxoplasmosis","Amphetamine addiction","Pancreatic cancer","Spliceosome","Calcium signaling pathway","Toxoplasmosis","Pathways in cancer","Transcriptional misregulation in cancer","Complement and coagulation cascades","Dilated cardiomyopathy","Nicotinate and nicotinamide metabolism","Pathogenic Escherichia coli infection"],["65","64","68","5","26","71","35","34","47","34","78","5","90","135","12","176","55","61","42","112","24","25","53","74","36","56","112","70","74","299","94","27","35","14","123"],[3,4,2,3,3,2,2,5,2,2,2,2,2,3,3,3,3,4,3,4,2,7,2,4,2,2,3,3,2,6,2,2,2,3,6],[2,2,2,3,3,1,2,4,2,2,2,2,2,3,3,2,3,4,3,4,2,7,2,4,2,2,3,3,2,6,2,2,2,3,6],[0,0,0,0,0,0,0.002,0,0,0,0.001,0.002,0,0.003,0,0.003,0,0,0.001,0,0,0,0,0,0,0,0.002,0.003,0.001,0,0.001,0.002,0,0.001,0],[1.012,0.963,-0.402,0.709,-0.147,-0.135,-0.143,-0.154,0.119,-0.105,0.142,0.114,-0.078,0.064,-0.11,-0.083,-0.087,-0.074,-0.096,-0.066,-0.057,-0.095,0.061,0.084,-0.094,-0.067,0.192,-0.246,0.076,0.076,-0.058,-0.147,0.181,0.179,-0.135],[1,0.992,1,0.999,1,1,1,0.991,1,1,1,1,1,0.995,0.999,0.995,0.999,0.995,0.998,0.992,1,0.976,1,0.993,1,1,0.996,0.994,1,0.996,1,1,1,0.998,0.937],[0,0,0,0,0.002,0,0.001,0,0.001,0,0,0.003,0,0.002,0.001,0.001,0.002,0.001,0,0.001,0,0,0.001,0,0,0,0.001,0,0,0,0.002,0.001,0,0.001,0],[5e-07,5e-07,5e-07,5e-07,0.001,5e-07,0.001,5e-07,5e-07,5e-07,5e-07,0.001,5e-07,0.001,0.001,0.001,0.001,0.001,5e-07,0.001,5e-07,5e-07,5e-07,5e-07,5e-07,5e-07,0.001,5e-07,5e-07,5e-07,0.001,0.001,5e-07,0.001,5e-07]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>factor<\/th>\n      <th>pwy<\/th>\n      <th>pwy_name<\/th>\n      <th>pwy_gene_num<\/th>\n      <th>num_snp_fdr.0.2<\/th>\n      <th>num_snp_noMHC<\/th>\n      <th>pval_in_fitting<\/th>\n      <th>slope_in_fitting<\/th>\n      <th>rsq_in_fitting<\/th>\n      <th>pval_in_resampling<\/th>\n      <th>pval_in_resampling_2<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[5,6,7,8,9,10,11]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="checking-the-reverse-causality-3" class="section level3">
<h3>Checking the reverse causality</h3>
<p>**The suffix &quot;_reverse&quot; are the results for traits -&gt; LV (using all candidate SNPs after LD Clumping), the suffix &quot;_rremove&quot; are the results for traits -&gt; LV (using all candidate SNPs after LD Clumping but remove the SNPs have fdr &lt; 0.2)**</p>
<div id="htmlwidget-32695f625600a1e2a463" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-32695f625600a1e2a463">{"x":{"filter":"none","data":[["EUR.CD_pwy124_pc1","EUR.CD_pwy300_pc1","EUR.UC_pwy188_pc5","T1D_pwy72_pc3","allergy_pwy248_pc5","baso_pwy153_pc1","baso_pwy23_pc2","baso_pwy246_pc2","baso_pwy56_pc4","gran_pwy100_pc3","gran_pwy116_pc5","gran_pwy132_pc1","lymph_pwy162_pc3","lymph_pwy290_pc5","lymph_pwy42_pc1","mcv_pwy139_pc4","mcv_pwy154_pc5","mcv_pwy176_pc5","mcv_pwy273_pc2","mcv_pwy284_pc5","mcv_pwy297_pc2","mcv_pwy327_pc1","neut_pwy304_pc5","pct_pwy279_pc4","pdw_pwy262_pc5","pdw_pwy302_pc3","plt_pwy101_pc5","plt_pwy115_pc3","plt_pwy279_pc4","rdw_pwy294_pc4","rdw_pwy295_pc2","ukb.allasthma_pwy172_pc2","ukb.allasthma_pwy330_pc1","ukb.allasthma_pwy73_pc3","wbc_pwy269_pc1"],[7.99617709614786e-05,-0.000681944065187512,0.00116081275948421,-0.00162841563963467,-0.0206539253462683,-0.0214230658692429,0.00711724498695254,-0.0341039990844365,0.0200010180988706,-0.0178951132042542,0.022005452185752,0.00954007345240271,-0.019297043978991,-0.0241040525845762,-0.00713454606944097,0.0251474008338839,0.00729285045967032,0.0231655332439793,0.023877102326983,0.0247487013139036,0.0366959012106918,0.0126894738348504,0.0204571324100566,0.030592050163003,-0.00141402525737711,-0.00599190498850888,0.00523031050522604,-0.0144259833370128,0.0220517515772549,-0.0165366648307526,-0.0279017756072513,-0.012207028634224,0.000443633447339664,-0.0006857174920194,0.0017089255226459],[0.991791495424332,0.973262995952826,0.919435083512509,0.834073956947344,0.92488173332499,0.761722940790194,0.923027623611125,0.719882840816103,0.785751711376004,0.869692765428358,0.959295636280418,0.974193007860589,0.916946736533039,0.910855343968536,0.942495886658246,0.985500102707561,0.987153046190859,0.856289150682551,0.860992952197057,0.877875730138841,0.960409323425535,0.982009695707144,0.943560900167383,0.910981207730804,0.997279434709163,0.956377157915764,0.977118335168871,0.961421874456906,0.961283184374337,0.925706242136353,0.980877441804903,0.843081963675785,0.993835657623466,0.973419513793484,0.994139148988667],[-0.00291821973091611,-0.00110326247383452,0.00529202578646881,-0.00385642192089874,0.00983743042711005,-0.0128320496595489,-0.0214702112096674,-0.0449750448162352,0.0158335062633803,-0.0189329065645349,0.0127500924009709,-0.00341753207420403,-0.00640991067679408,-0.020977881916977,-0.0134043532932839,0.0226130205761166,0.00318740387704253,-0.00874572735393997,0.0030927897987052,0.0233010484040364,-0.0271790677938952,0.0269649315347689,0.00961479825397937,0.00594976368950917,-0.0202606500830811,0.00314372284254816,0.00272811907522197,-0.0150490622132704,-0.0125854292900094,-0.012787995555076,-0.0238074469229588,0.016595128212904,0.0112596234228235,-0.000573979271011105,0.0082548746613007],[0.694641070356504,0.956907545540037,0.63007719252249,0.609081201404675,0.964224933685824,0.85523049436197,0.771414164774692,0.635382304794619,0.826780207472275,0.863589378611495,0.976684328156237,0.990779125885474,0.973052270125885,0.92354052506655,0.891062244413982,0.987348115368086,0.994340265626716,0.944863627694531,0.982108011960808,0.88265849786785,0.970645216775481,0.962709814040248,0.973328132927002,0.982154483878539,0.960126320763702,0.977384001375696,0.987830962317984,0.960549246091106,0.978014786213181,0.9427548773305,0.984176085086836,0.790825505230946,0.846718030923172,0.977509973254164,0.970961231019728]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>mbe_est_reverse<\/th>\n      <th>mbe_pval_reverse<\/th>\n      <th>mbe_est_rremove<\/th>\n      <th>mbe_pval_rremove<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[1,2,3,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div id="t-cellcd8" class="section level2">
<h2><strong>T cell(CD8)</strong></h2>
<div id="effectsize-fitting-summary-4" class="section level3">
<h3>effectsize fitting summary</h3>
<div id="htmlwidget-312b9245b08df84edcee" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-312b9245b08df84edcee">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41"],["EUR.CD_pwy96_pc4","EUR.IBD_pwy141_pc3","EUR.UC_pwy18_pc1","hgb_pwy283_pc2","hgb_pwy288_pc2","hgb_pwy306_pc4","lymph_pwy113_pc3","lymph_pwy138_pc5","lymph_pwy139_pc2","lymph_pwy143_pc4","lymph_pwy150_pc5","lymph_pwy177_pc3","lymph_pwy227_pc5","lymph_pwy242_pc1","lymph_pwy298_pc2","lymph_pwy95_pc3","mch_pwy286_pc5","mchc_pwy177_pc3","mcv_pwy163_pc4","mcv_pwy206_pc2","mcv_pwy293_pc1","mcv_pwy293_pc3","mcv_pwy317_pc2","mono_pwy117_pc4","mono_pwy213_pc4","myeloid_wbc_pwy10_pc3","myeloid_wbc_pwy125_pc4","myeloid_wbc_pwy160_pc1","neut_pwy10_pc3","pct_pwy103_pc1","pct_pwy181_pc2","pct_pwy270_pc3","pdw_pwy169_pc1","plt_pwy116_pc5","plt_pwy44_pc4","rbc_pwy217_pc4","rbc_pwy28_pc5","rdw_pwy179_pc5","rdw_pwy329_pc1","ukb.allasthma_pwy221_pc5","ukb.allasthma_pwy64_pc2"],["03015","04146","00230","05160","05165","05216","04014","04142","04144","04151","04217","04621","04925","04962","05205","03013","05163","04621","04390","04730","05170","05170","05230","04024","04911","00071","04071","04370","00071","03060","04630","05131","04530","04022","00513","04915","00340","04623","05412","04919","00604"],["mRNA surveillance pathway","Peroxisome","Purine metabolism","Hepatitis C","Human papillomavirus infection","Thyroid cancer","Ras signaling pathway","Lysosome","Endocytosis","PI3K-Akt signaling pathway","Necroptosis","NOD-like receptor signaling pathway","Aldosterone synthesis and secretion","Vasopressin-regulated water reabsorption","Proteoglycans in cancer","RNA transport","Human cytomegalovirus infection","NOD-like receptor signaling pathway","Hippo signaling pathway","Long-term depression","Human immunodeficiency virus 1 infection","Human immunodeficiency virus 1 infection","Central carbon metabolism in cancer","cAMP signaling pathway","Insulin secretion","Fatty acid degradation","Sphingolipid signaling pathway","VEGF signaling pathway","Fatty acid degradation","Protein export","JAK-STAT signaling pathway","Shigellosis","Tight junction","cGMP-PKG signaling pathway","Various types of N-glycan biosynthesis","Estrogen signaling pathway","Histidine metabolism","Cytosolic DNA-sensing pathway","Arrhythmogenic right ventricular cardiomyopathy","Thyroid hormone signaling pathway","Glycosphingolipid biosynthesis - ganglio series"],["72","59","66","109","195","29","119","104","179","178","97","118","49","33","111","137","152","118","76","30","145","145","49","94","31","28","81","39","28","20","83","162","93","78","30","66","8","41","33","84","11"],[2,2,2,3,2,2,2,3,2,3,4,2,3,2,2,2,4,3,4,7,3,12,2,4,4,2,3,2,2,4,9,8,4,2,2,4,3,3,9,2,3],[1,2,1,3,2,2,2,3,2,3,4,2,3,2,2,2,4,3,4,7,3,12,2,4,4,2,3,2,2,4,9,8,4,2,2,4,3,3,9,1,3],[0,0.001,0,0.005,0.002,0.02,0.069,0.008,0.083,0.021,0.016,0.001,0.001,0.033,0.088,0.048,0,0,0,0,0.001,0,0,0.001,0,0.006,0,0.001,0.001,0.001,0.001,0,0,0.001,0.001,0,0.001,0,0,0.011,0.001],[0.641,-0.493,-0.504,0.082,0.083,-0.07,-0.048,-0.081,-0.025,0.041,-0.041,0.088,0.071,0.078,-0.032,0.046,0.091,-0.168,0.101,-0.088,0.096,0.08,0.127,0.08,-0.095,0.076,-0.092,-0.071,0.078,-0.123,-0.09,0.07,0.117,-0.112,-0.149,0.113,0.138,-0.105,0.176,-0.126,0.218],[1,1,1,0.989,1,0.999,0.988,0.983,0.983,0.958,0.892,1,0.997,0.997,0.981,0.994,0.99,1,0.997,0.921,0.998,0.688,1,0.985,0.992,1,1,1,1,0.979,0.77,0.894,0.997,1,1,0.992,0.998,1,0.828,1,0.999],[0.003,0,0,0,0.002,0.001,0.001,0,0.001,0.001,0.002,0,0,0.002,0.001,0.001,0.001,0.001,0.001,0.002,0,0.001,0.001,0.001,0.001,0.002,0.001,0,0.002,0.001,0,0.001,0,0,0.004,0,0.001,0.001,0.001,0.002,0.001],[0.0018,5e-07,5e-07,0.0018,0.0016,0.0012,0.00082,0.0016,0.00049,0.0018,0.001,0.00022,8e-05,0.0011,0.00082,0.00039,0.00091,0.0016,0.00021,0.0012,0.00099,0.00078,0.00043,0.00045,0.0017,0.001,0.001,0.00039,0.0014,0.0018,0.0018,0.0016,9e-05,0.0011,0.0019,0.00062,0.0012,0.00029,0.00062,0.0019,0.0014]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>factor<\/th>\n      <th>pwy<\/th>\n      <th>pwy_name<\/th>\n      <th>pwy_gene_num<\/th>\n      <th>num_snp_fdr.0.2<\/th>\n      <th>num_snp_noMHC<\/th>\n      <th>pval_in_fitting<\/th>\n      <th>slope_in_fitting<\/th>\n      <th>rsq_in_fitting<\/th>\n      <th>pval_in_resampling<\/th>\n      <th>pval_in_resampling_2<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[5,6,7,8,9,10,11]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="checking-the-reverse-causality-4" class="section level3">
<h3>Checking the reverse causality</h3>
<p>**The suffix &quot;_reverse&quot; are the results for traits -&gt; LV (using all candidate SNPs after LD Clumping), the suffix &quot;_rremove&quot; are the results for traits -&gt; LV (using all candidate SNPs after LD Clumping but remove the SNPs have fdr &lt; 0.2)**</p>
<div id="htmlwidget-3cd2a26b2bcc583003e2" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-3cd2a26b2bcc583003e2">{"x":{"filter":"none","data":[["EUR.CD_pwy96_pc4","EUR.IBD_pwy141_pc3","EUR.UC_pwy18_pc1","hgb_pwy283_pc2","hgb_pwy288_pc2","hgb_pwy306_pc4","lymph_pwy113_pc3","lymph_pwy138_pc5","lymph_pwy139_pc2","lymph_pwy143_pc4","lymph_pwy150_pc5","lymph_pwy177_pc3","lymph_pwy227_pc5","lymph_pwy242_pc1","lymph_pwy298_pc2","lymph_pwy95_pc3","mch_pwy286_pc5","mchc_pwy177_pc3","mcv_pwy163_pc4","mcv_pwy206_pc2","mcv_pwy293_pc1","mcv_pwy293_pc3","mcv_pwy317_pc2","mono_pwy117_pc4","mono_pwy213_pc4","myeloid_wbc_pwy10_pc3","myeloid_wbc_pwy125_pc4","myeloid_wbc_pwy160_pc1","neut_pwy10_pc3","pct_pwy103_pc1","pct_pwy181_pc2","pct_pwy270_pc3","pdw_pwy169_pc1","plt_pwy116_pc5","plt_pwy44_pc4","rbc_pwy217_pc4","rbc_pwy28_pc5","rdw_pwy179_pc5","rdw_pwy329_pc1","ukb.allasthma_pwy221_pc5","ukb.allasthma_pwy64_pc2"],[0.00163547382212137,-0.00342824758768812,-0.000756243238402732,0.00514373629245313,0.0298285849286906,0.0161454849789173,-0.0102503865540022,-0.035500970893537,0.0164202618044911,0.038765107360387,-0.0431139394315885,-0.0331221646460769,0.00443911700732791,-0.00854542980706796,-0.0427102454525183,0.0842668854102673,0.0182518721462799,0.208772913908032,0.0193841660766338,0.0199321613850643,0.0371555166746376,-0.0157120985533119,0.0173892155103932,0.0424942368859682,0.0439396758969526,-0.0296694595043334,0.00365091116135297,0.0144239545971665,-0.0048120109291041,-0.0228125735507767,0.0116134841174613,0.0381395572586136,-0.0107421549378977,-0.00194178519827659,-0.0276372968140226,-0.00831512568476711,-0.0231905893610378,0.00626838464721402,-0.0033945811967353,-0.00514830322595294,-0.0141056957732326],[0.94010841638263,0.910477589293754,0.950869145608541,0.971066008142181,0.904305311970379,0.923651480263731,0.996978226722869,0.91316142819505,0.941864360916818,0.893226337994891,0.929611012624142,0.80600549050855,0.993882872518246,0.956451623657949,0.908833538196034,0.832962551652039,0.948992979523408,0.0915420095782351,0.912230878120618,0.870726633258047,0.974992232874431,0.925771845713402,0.955305721023492,0.988866391986659,0.857470962091406,0.843681306169702,0.97656675741453,0.880212712920162,0.967667108411319,0.941423553810362,0.987357374815049,0.846427902932626,0.877690801154008,0.995096390537541,0.809939246898856,0.983888060937405,0.808251013174773,0.946070977270515,0.986912574701195,0.981592944608198,0.701205702055262],[0.00307218346145621,-0.00379805322685423,0.00338704201440621,0.00565764164069904,0.0084525531746209,0.0200577477127712,0.0126722801599826,0.0130152197744398,-0.0570989073589274,0.0046304417606855,-0.0159954668124023,-0.0338272751957405,-0.00740678104707548,-0.0195369142048332,0.0360091968532608,0.0287980902039973,0.0355703866801527,0.201707617075105,0.0139278412078809,0.00203496249346991,0.0354653361907999,0.00223740374674009,0.0229295418025526,0.0515405558414272,0.047637028552292,-0.0109005731304688,-0.00295604707817532,0.0142656781004824,-0.00544137838007153,-0.0233416095247136,-0.0105437223604703,-0.0148707845817775,-0.00531599131965521,-0.00236409656271519,-0.0188545832380331,-0.00241504398242753,-0.0279496032430036,0.00628761896753005,0.00161765462276797,0.0100248305034984,-0.00918984425548164],[0.891375840028149,0.897980079575619,0.781766613296012,0.968115321686858,0.972798881542192,0.907447260163573,0.996344757285258,0.967793698485524,0.798643786190503,0.987337459624995,0.973410772428712,0.801081685141795,0.990069364842256,0.900644200112322,0.923658492645328,0.942311575096512,0.901580036821799,0.111224103449274,0.937345806494928,0.987058630116703,0.975465236349446,0.989245322231928,0.938869758709212,0.986919193425072,0.846782737469186,0.942648302893687,0.981016085742462,0.880622552790474,0.963096649069094,0.940422838773362,0.988468644179681,0.938706344461145,0.939498946762927,0.993971892386565,0.869564261432231,0.995267262239937,0.770705231596189,0.946099777637192,0.993626676283042,0.963497507536512,0.803707399153532]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>mbe_est_reverse<\/th>\n      <th>mbe_pval_reverse<\/th>\n      <th>mbe_est_rremove<\/th>\n      <th>mbe_pval_rremove<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[1,2,3,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div id="platelet" class="section level2">
<h2>platelet</h2>
<div id="effectsize-fitting-summary-5" class="section level3">
<h3>effectsize fitting summary</h3>
<div id="htmlwidget-9ad70954d77b9fd07fab" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-9ad70954d77b9fd07fab">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29"],["baso_pwy100_pc1","eo_pwy262_pc2","gran_pwy77_pc2","hgb_pwy192_pc5","lymph_pwy23_pc4","mch_pwy15_pc2","mono_pwy241_pc1","neut_pwy152_pc5","pct_pwy294_pc5","plt_pwy221_pc2","plt_pwy89_pc2","rbc_pwy197_pc5","ret_pwy110_pc1","ret_pwy119_pc3","ret_pwy188_pc1","ret_pwy18_pc1","ret_pwy192_pc5","ret_pwy218_pc1","ret_pwy223_pc1","ret_pwy243_pc2","ret_pwy247_pc4","ret_pwy258_pc4","ret_pwy300_pc1","ret_pwy309_pc1","ret_pwy321_pc5","ret_pwy64_pc2","ret_pwy67_pc2","ukb.allasthma_pwy189_pc1","wbc_pwy10_pc2"],["03030","05031","00790","04670","00270","00140","04961","04260","05200","04919","01522","04720","03460","04061","04662","00230","04670","04916","04921","04964","04972","05016","05210","05219","05320","00604","00640","04664","00071"],["DNA replication","Amphetamine addiction","Folate biosynthesis","Leukocyte transendothelial migration","Cysteine and methionine metabolism","Steroid hormone biosynthesis","Endocrine and other factor-regulated calcium reabsorption","Cardiac muscle contraction","Pathways in cancer","Thyroid hormone signaling pathway","Endocrine resistance","Long-term potentiation","Fanconi anemia pathway","Viral protein interaction with cytokine and cytokine receptor","B cell receptor signaling pathway","Purine metabolism","Leukocyte transendothelial migration","Melanogenesis","Oxytocin signaling pathway","Proximal tubule bicarbonate reclamation","Pancreatic secretion","Huntington disease","Colorectal cancer","Bladder cancer","Autoimmune thyroid disease","Glycosphingolipid biosynthesis - ganglio series","Propanoate metabolism","Fc epsilon RI signaling pathway","Fatty acid degradation"],["35","36","11","67","32","12","23","33","284","79","56","40","28","42","70","65","67","44","79","10","35","157","60","27","27","10","29","47","26"],[2,2,2,5,3,2,3,2,3,2,2,2,2,4,4,6,3,4,8,3,3,3,7,4,2,5,2,2,2],[1,2,2,5,3,2,3,2,3,2,2,2,2,4,4,6,3,4,8,3,3,3,7,4,2,5,2,2,2],[null,0.001,0.001,0,0.007,0.002,0.001,0.001,0,0.003,0,0.001,0.002,0,0.001,0.001,0.006,0.004,0,0.001,0.001,0.001,0.002,0.001,0.002,0.001,0,0,0],[0.106,0.08,0.072,0.06,0.083,0.09,-0.16,-0.073,-0.044,0.186,-0.085,-0.104,-0.073,-0.094,-0.09,-0.044,0.074,0.054,0.044,0.069,-0.077,-0.09,-0.064,-0.104,0.111,0.087,-0.094,0.149,-0.108],[1,1,1,0.979,0.986,1,0.997,1,0.999,1,1,1,1,0.997,0.979,0.917,0.989,0.953,0.905,0.998,0.999,0.997,0.811,0.979,1,0.953,1,1,1],[0,0,0,0,0.001,0.003,0.001,0.002,0.003,0.003,0.001,0,0,0,0.002,0.001,0.001,0.004,0,0,0,0,0.001,0.002,0.004,0.002,0,0,0],[5e-07,0.00045,0.0011,0.00037,0.0016,0.0016,0.0017,0.0017,0.0013,0.0016,0.00077,0.0013,0.00067,0.00021,0.0012,0.00096,0.0015,0.0019,5e-07,0.0015,0.00073,0.00029,0.0011,0.0012,0.0017,0.0018,5e-07,0.001,5e-07]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>factor<\/th>\n      <th>pwy<\/th>\n      <th>pwy_name<\/th>\n      <th>pwy_gene_num<\/th>\n      <th>num_snp_fdr.0.2<\/th>\n      <th>num_snp_noMHC<\/th>\n      <th>pval_in_fitting<\/th>\n      <th>slope_in_fitting<\/th>\n      <th>rsq_in_fitting<\/th>\n      <th>pval_in_resampling<\/th>\n      <th>pval_in_resampling_2<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[5,6,7,8,9,10,11]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="checking-the-reverse-causality-5" class="section level3">
<h3>Checking the reverse causality</h3>
<p>**The suffix &quot;_reverse&quot; are the results for traits -&gt; LV (using all candidate SNPs after LD Clumping), the suffix &quot;_rremove&quot; are the results for traits -&gt; LV (using all candidate SNPs after LD Clumping but remove the SNPs have fdr &lt; 0.2)**</p>
<div id="htmlwidget-12eab46753bdd639aac3" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-12eab46753bdd639aac3">{"x":{"filter":"none","data":[["baso_pwy100_pc1","eo_pwy262_pc2","gran_pwy77_pc2","hgb_pwy192_pc5","lymph_pwy23_pc4","mch_pwy15_pc2","mono_pwy241_pc1","neut_pwy152_pc5","pct_pwy294_pc5","plt_pwy221_pc2","plt_pwy89_pc2","rbc_pwy197_pc5","ret_pwy110_pc1","ret_pwy119_pc3","ret_pwy188_pc1","ret_pwy18_pc1","ret_pwy192_pc5","ret_pwy218_pc1","ret_pwy223_pc1","ret_pwy243_pc2","ret_pwy247_pc4","ret_pwy258_pc4","ret_pwy300_pc1","ret_pwy309_pc1","ret_pwy321_pc5","ret_pwy64_pc2","ret_pwy67_pc2","ukb.allasthma_pwy189_pc1","wbc_pwy10_pc2"],[0.0791116393678255,0.00874283596240488,-0.00556017012543819,-0.0873349883109764,0.0384525517272536,-0.00673134090197713,0.0269240284279793,-0.0186220953890697,-0.0108731387213155,0.0226798037323732,-0.0343122457492697,-0.0133526060464533,0.0170379254013859,0.0220497888161262,0.0110059329859737,0.0456583981432672,-0.0305257996636072,-0.0225387127586743,0.0674141535366743,0.0178376982756188,0.000358983893978149,-0.0204097987433727,0.0126006404670846,-0.0330665525654013,0.0161923552342031,-0.0199552036008335,0.0232069695047663,-0.0415117269938605,-0.002070504232357],[0.175605145684195,0.972574566856287,0.990655917300649,0.748935243866598,0.894813886202986,0.991416544957731,0.911038195493305,0.93789335471951,0.994530607093454,0.940237336458681,0.918115569378742,0.979623669283339,0.862451014622026,0.92499963525788,0.918576703244474,0.809153003481732,0.932790910540622,0.887008057440056,0.729894042094818,0.896905902237342,0.99708359543852,0.986803273196107,0.972197867068854,0.82205737178154,0.95223857305717,0.952631869390806,0.677597208420439,0.352993153686744,0.993506292693039],[0.0964452448941966,0.0242689859538672,0.026365779538029,-0.0665585893127751,0.00473051611791497,-0.0271149437922542,0.035817021550713,-0.00339410231887172,0.00185592082284103,0.0173258786239341,0.0195769029681472,-0.0148572183614686,0.00643872354417674,0.0132443319089752,-0.00650651228882992,-0.0196504004949283,-0.0186436129877325,-0.0247436198144761,-0.0268468395090551,0.0317851928248665,-0.00448776522998529,-0.0119830225803845,-0.0360644583052547,-0.0307208469133187,0.00359288033879857,-0.00338003605401305,0.0252004165627326,-0.0324466586171592,-0.0057766975713136],[0.0969099731838686,0.923041621297557,0.954855194722126,0.810543999519329,0.986935410282832,0.965913801805452,0.881649050982598,0.98842023366605,0.999067295720839,0.95562125348927,0.953047317013895,0.977081724091356,0.948805897543042,0.955189387270102,0.951630877253257,0.915573256071831,0.959247464301665,0.876388130053893,0.89022653269665,0.82252265370959,0.964186228311081,0.992636767741155,0.919969744871859,0.835539684562019,0.989164818158526,0.992212205766409,0.647403021745736,0.449995419157126,0.981215008376181]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>mbe_est_reverse<\/th>\n      <th>mbe_pval_reverse<\/th>\n      <th>mbe_est_rremove<\/th>\n      <th>mbe_pval_rremove<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[1,2,3,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] workflowr_1.6.2

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.5        rstudioapi_0.11   whisker_0.3-2     knitr_1.30       
 [5] magrittr_1.5      R6_2.4.1          rlang_0.4.8       highr_0.8        
 [9] stringr_1.4.0     tools_3.6.1       DT_0.15           xfun_0.18        
[13] git2r_0.26.1      crosstalk_1.1.0.1 htmltools_0.5.0   ellipsis_0.3.1   
[17] rprojroot_1.3-2   yaml_2.2.1        digest_0.6.25     tibble_3.0.3     
[21] lifecycle_0.2.0   crayon_1.3.4      later_1.1.0.1     htmlwidgets_1.5.2
[25] vctrs_0.3.4       promises_1.1.1    fs_1.5.0          glue_1.4.2       
[29] evaluate_0.14     rmarkdown_1.13    stringi_1.5.3     compiler_3.6.1   
[33] pillar_1.4.6      backports_1.1.10  jsonlite_1.7.1    httpuv_1.5.1     
[37] pkgconfig_2.0.3  </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
