<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="XSun" />

<meta name="date" content="2021-07-09" />

<title>KEGG PCs summary for 6 cell types from CEDAR dataset (pathways have &gt; 40 genes only), Pruned PCs</title>

<script src="site_libs/jquery-1.12.4/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.2/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.15/datatables.js"></script>
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.1.0.1/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">factor_analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/xsun1229/factor_analysis">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">KEGG PCs summary for 6 cell types from CEDAR dataset (pathways have &gt; 40 genes only), Pruned PCs</h1>
<h4 class="author">XSun</h4>
<h4 class="date">2021-07-09</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-07-10
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>factor_analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div class="panel-group" id="workflowr-checks">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200623code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200623)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200623code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200623)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it difficult for you and others to run your code on a different machine. Change the absolute path(s) below to the suggested relative path(s) to make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/factor_analysis/output/pwy_name_cd14_cedar_qc_kegg_40genes_pruned.rdata
</td>
<td style="text-align:left;">
output/pwy_name_cd14_cedar_qc_kegg_40genes_pruned.rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/factor_analysis/output/mbe_cd14_cedar_qc_kegg_40genes_pruned.rdata
</td>
<td style="text-align:left;">
output/mbe_cd14_cedar_qc_kegg_40genes_pruned.rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/factor_analysis/output/pwy_name_bcell_cedar_qc_kegg_40genes_pruned.rdata
</td>
<td style="text-align:left;">
output/pwy_name_bcell_cedar_qc_kegg_40genes_pruned.rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/factor_analysis/output/mbe_bcell_cedar_qc_kegg_40genes_pruned.rdata
</td>
<td style="text-align:left;">
output/mbe_bcell_cedar_qc_kegg_40genes_pruned.rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/factor_analysis/output/pwy_name_tcell_cedar_qc_kegg_40genes_pruned.rdata
</td>
<td style="text-align:left;">
output/pwy_name_tcell_cedar_qc_kegg_40genes_pruned.rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/factor_analysis/output/mbe_tcell_cedar_qc_kegg_40genes_pruned.rdata
</td>
<td style="text-align:left;">
output/mbe_tcell_cedar_qc_kegg_40genes_pruned.rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/factor_analysis/output/pwy_name_cd15_cedar_qc_kegg_40genes_pruned.rdata
</td>
<td style="text-align:left;">
output/pwy_name_cd15_cedar_qc_kegg_40genes_pruned.rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/factor_analysis/output/mbe_cd15_cedar_qc_kegg_40genes_pruned.rdata
</td>
<td style="text-align:left;">
output/mbe_cd15_cedar_qc_kegg_40genes_pruned.rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/factor_analysis/output/pwy_name_thy_cedar_qc_kegg_40genes_pruned.rdata
</td>
<td style="text-align:left;">
output/pwy_name_thy_cedar_qc_kegg_40genes_pruned.rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/factor_analysis/output/mbe_thy_cedar_qc_kegg_40genes_pruned.rdata
</td>
<td style="text-align:left;">
output/mbe_thy_cedar_qc_kegg_40genes_pruned.rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/factor_analysis/output/pwy_name_plt_cedar_qc_kegg_40genes_pruned.rdata
</td>
<td style="text-align:left;">
output/pwy_name_plt_cedar_qc_kegg_40genes_pruned.rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/factor_analysis/output/mbe_plt_cedar_qc_kegg_40genes_pruned.rdata
</td>
<td style="text-align:left;">
output/mbe_plt_cedar_qc_kegg_40genes_pruned.rdata
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomxsun1229factoranalysistree24b53437e47ec2d607ed7347dece8bd5a1b609d2targetblank24b5343a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/xsun1229/factor_analysis/tree/24b53437e47ec2d607ed7347dece8bd5a1b609d2" target="_blank">24b5343</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomxsun1229factoranalysistree24b53437e47ec2d607ed7347dece8bd5a1b609d2targetblank24b5343a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/xsun1229/factor_analysis/tree/24b53437e47ec2d607ed7347dece8bd5a1b609d2" target="_blank">24b5343</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    analysis/.Rhistory
    Ignored:    output/.Rhistory

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/cedar_kegg_26_qc_d1k_40genes_pruned.Rmd</code>) and HTML (<code>docs/cedar_kegg_26_qc_d1k_40genes_pruned.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/24b53437e47ec2d607ed7347dece8bd5a1b609d2/analysis/cedar_kegg_26_qc_d1k_40genes_pruned.Rmd" target="_blank">24b5343</a>
</td>
<td>
XSun
</td>
<td>
2021-07-10
</td>
<td>
update
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2><strong>Introduction</strong></h2>
<p>In this part, we just used the pathways have &gt; 40 genes.</p>
</div>
<div id="summary" class="section level2">
<h2><strong>Summary</strong></h2>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/CEDAR_cell_sum_40genes_pruned.png" width="80%">
</figure>
</div>
<div id="detailed-results-for-6-cell-types" class="section level2">
<h2><strong>Detailed results for 6 cell types</strong></h2>
<p>We just listed the pairs show sign consistancy (at p-value &lt; 0.05 in resampling) here</p>
<p>We did some literature searching for some disease related pairs (p-value &lt; 0.01 in resampling) in monocyte and B cell.</p>
<div id="monocyte-cd14" class="section level3">
<h3>monocyte CD14</h3>
<div id="histogram-to-show-the-number-of-snps-that-each-pair-contains" class="section level4">
<h4>Histogram to show the number of SNPs that each pair contains</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/snp_count_cd14_cedar_qc_kegg_40genes_pruned.jpeg" width="80%">
</figure>
</div>
<div id="effectsize-fitting-summary" class="section level4">
<h4>effectsize fitting summary</h4>
<div id="htmlwidget-c3f4891eb8e9a8c25198" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-c3f4891eb8e9a8c25198">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54"],["EUR.CD_pwy112_pc2","EUR.CD_pwy281_pc5","EUR.IBD_pwy121_pc1","EUR.IBD_pwy89_pc2","EUR.UC_pwy115_pc5","EUR.UC_pwy116_pc5","EUR.UC_pwy138_pc5","EUR.UC_pwy153_pc3","EUR.UC_pwy154_pc3","EUR.UC_pwy176_pc4","EUR.UC_pwy200_pc4","EUR.UC_pwy201_pc3","EUR.UC_pwy201_pc4","EUR.UC_pwy203_pc4","EUR.UC_pwy205_pc4","EUR.UC_pwy294_pc1","EUR.UC_pwy91_pc4","EUR.UC_pwy91_pc5","allergy_pwy167_pc3","allergy_pwy183_pc3","baso_pwy18_pc2","baso_pwy18_pc3","baso_pwy56_pc2","eo_pwy214_pc3","hgb_pwy135_pc2","lymph_pwy111_pc4","lymph_pwy181_pc2","lymph_pwy237_pc2","lymph_pwy291_pc4","lymph_pwy294_pc1","lymph_pwy299_pc1","lymph_pwy332_pc1","mchc_pwy186_pc5","mchc_pwy236_pc2","mchc_pwy315_pc2","mcv_pwy184_pc3","mono_pwy295_pc1","neut_pwy128_pc5","neut_pwy216_pc4","pct_pwy101_pc2","pct_pwy56_pc3","pdw_pwy150_pc5","pdw_pwy151_pc4","pdw_pwy153_pc2","pdw_pwy290_pc5","plt_pwy258_pc4","rdw_pwy116_pc1","rdw_pwy118_pc5","rdw_pwy186_pc4","ukb.allasthma_pwy201_pc4","ukb.allasthma_pwy276_pc3","wbc_pwy112_pc2","wbc_pwy155_pc5","wbc_pwy273_pc5"],["04012","05150","04064","01522","04020","04022","04142","04261","04270","04620","04723","04724","04724","04726","04728","05200","01524","01524","04514","04650","00230","00230","00564","04912","04137","04010","04630","04935","05168","05200","05206","05418","04659","04934","05225","04657","05202","04110","04914","03040","00564","04217","04218","04261","05167","05016","04022","04060","04659","04724","05142","04012","04310","05134"],["ErbB signaling pathway","Staphylococcus aureus infection","NF-kappa B signaling pathway","Endocrine resistance","Calcium signaling pathway","cGMP-PKG signaling pathway","Lysosome","Adrenergic signaling in cardiomyocytes","Vascular smooth muscle contraction","Toll-like receptor signaling pathway","Retrograde endocannabinoid signaling","Glutamatergic synapse","Glutamatergic synapse","Serotonergic synapse","Dopaminergic synapse","Pathways in cancer","Platinum drug resistance","Platinum drug resistance","Cell adhesion molecules","Natural killer cell mediated cytotoxicity","Purine metabolism","Purine metabolism","Glycerophospholipid metabolism","GnRH signaling pathway","Mitophagy - animal","MAPK signaling pathway","JAK-STAT signaling pathway","Growth hormone synthesis, secretion and action","Herpes simplex virus 1 infection","Pathways in cancer","MicroRNAs in cancer","Fluid shear stress and atherosclerosis","Th17 cell differentiation","Cushing syndrome","Hepatocellular carcinoma","IL-17 signaling pathway","Transcriptional misregulation in cancer","Cell cycle","Progesterone-mediated oocyte maturation","Spliceosome","Glycerophospholipid metabolism","Necroptosis","Cellular senescence","Adrenergic signaling in cardiomyocytes","Kaposi sarcoma-associated herpesvirus infection","Huntington disease","cGMP-PKG signaling pathway","Cytokine-cytokine receptor interaction","Th17 cell differentiation","Glutamatergic synapse","Chagas disease","ErbB signaling pathway","Wnt signaling pathway","Legionellosis"],["59","43","71","58","81","88","106","71","63","72","75","47","47","45","74","290","50","50","61","77","65","65","52","60","56","165","79","71","273","290","112","94","77","76","102","53","98","77","51","110","52","98","108","71","132","175","88","106","77","47","68","59","83","46"],[2,2,2,2,2,2,3,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,5,2,2,2,3,3,2,3,2,2,2,2,3,2,2,2,3,2,4,3,2,2,3,3,2,2,2,2],[1,0,2,2,2,2,3,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,5,2,2,2,3,3,2,3,2,2,2,2,3,2,2,2,3,2,4,3,2,2,3,3,2,2,2,2],[0.01,0.034,0.047,0.008,0.019,0.003,0.008,0.002,0.004,0.019,0.027,0.024,0.021,0.009,0.01,0.009,0.002,0.028,0.023,0.017,0.017,0.001,0.004,0.023,0.007,0.017,0.001,0.016,0.001,0.005,0.008,0.003,0.02,0.009,0.009,0.016,0.003,0.01,0.006,0.009,0.013,0.042,0.007,0.024,0.02,0.003,0.005,0.014,0.003,0,0.021,0.019,0.016,0.008],[-0.519,0.473,-0.556,0.429,-0.301,-0.349,-0.263,-0.297,-0.276,0.26,0.331,-0.324,0.508,-0.365,0.346,0.504,0.704,-0.343,0.095,0.211,0.068,0.064,-0.104,0.064,0.066,0.032,0.104,0.168,-0.04,-0.06,0.128,0.15,0.054,-0.106,0.104,0.076,-0.176,-0.066,-0.054,0.065,0.059,-0.057,-0.045,0.205,0.065,-0.117,-0.097,-0.039,-0.048,-0.145,-0.085,-0.086,0.056,-0.068],[1,0.997,0.995,1,0.999,1,0.985,1,1,0.999,0.998,0.999,0.999,1,1,1,1,0.998,0.999,0.999,0.999,1,1,0.999,1,0.999,0.947,0.999,1,1,0.984,0.995,0.999,0.983,1,0.999,1,1,0.989,1,1,0.996,0.986,0.999,0.875,0.994,1,1,0.993,0.999,0.999,0.999,0.999,1],[0.016,0.04,0.042,0.017,0.01,0.013,0.006,0,0.007,0.047,0.027,0.022,0.022,0.015,0.04,0.034,0.002,0.018,0.041,0.038,0.042,0.002,0.006,0.037,0.001,0.038,0.017,0.038,0.001,0.015,0.044,0.017,0.03,0.035,0.032,0.016,0.01,0.024,0.034,0.023,0.043,0.014,0.025,0.042,0.016,0.009,0.003,0.034,0.015,0.001,0.031,0.049,0.041,0.031]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>factor<\/th>\n      <th>pwy<\/th>\n      <th>pwy_name<\/th>\n      <th>pwy_gene_num<\/th>\n      <th>num_snp_fdr.0.2<\/th>\n      <th>num_snp_noMHC<\/th>\n      <th>pval_in_fitting<\/th>\n      <th>slope_in_fitting<\/th>\n      <th>rsq_in_fitting<\/th>\n      <th>pval_in_resampling<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[5,6,7,8,9,10]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="checking-the-reverse-causality" class="section level4">
<h4>Checking the reverse causality</h4>
<p>**The suffix &quot;_reverse&quot; are the results for traits -&gt; LV (using all candidate SNPs after LD Clumping), the suffix &quot;_rremove&quot; are the results for traits -&gt; LV (using all candidate SNPs after LD Clumping but remove the SNPs have fdr &lt; 0.2)**</p>
<div id="htmlwidget-cfefb73e51b4928e765c" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-cfefb73e51b4928e765c">{"x":{"filter":"none","data":[["EUR.CD_pwy112_pc2","EUR.CD_pwy281_pc5","EUR.IBD_pwy121_pc1","EUR.IBD_pwy89_pc2","EUR.UC_pwy115_pc5","EUR.UC_pwy116_pc5","EUR.UC_pwy138_pc5","EUR.UC_pwy153_pc3","EUR.UC_pwy154_pc3","EUR.UC_pwy176_pc4","EUR.UC_pwy200_pc4","EUR.UC_pwy201_pc3","EUR.UC_pwy201_pc4","EUR.UC_pwy203_pc4","EUR.UC_pwy205_pc4","EUR.UC_pwy294_pc1","EUR.UC_pwy91_pc4","EUR.UC_pwy91_pc5","allergy_pwy167_pc3","allergy_pwy183_pc3","baso_pwy18_pc2","baso_pwy18_pc3","baso_pwy56_pc2","eo_pwy214_pc3","hgb_pwy135_pc2","lymph_pwy111_pc4","lymph_pwy181_pc2","lymph_pwy237_pc2","lymph_pwy291_pc4","lymph_pwy294_pc1","lymph_pwy299_pc1","lymph_pwy332_pc1","mchc_pwy186_pc5","mchc_pwy236_pc2","mchc_pwy315_pc2","mcv_pwy184_pc3","mono_pwy295_pc1","neut_pwy128_pc5","neut_pwy216_pc4","pct_pwy101_pc2","pct_pwy56_pc3","pdw_pwy150_pc5","pdw_pwy151_pc4","pdw_pwy153_pc2","pdw_pwy290_pc5","plt_pwy258_pc4","rdw_pwy116_pc1","rdw_pwy118_pc5","rdw_pwy186_pc4","ukb.allasthma_pwy201_pc4","ukb.allasthma_pwy276_pc3","wbc_pwy112_pc2","wbc_pwy155_pc5","wbc_pwy273_pc5"],[-0.0483392468057606,0.0199449701646914,-0.00938223361417778,-0.00161369031944991,0.00816152658406821,-0.00922388954320841,0.0471960783700087,0.0641686035658848,-0.0075072887859875,-0.0543965492981235,-0.00356429670946579,0.0751773626050181,0.00880952927534473,-0.000611961591126864,0.017879253415396,-0.0150418464456887,-0.00806373113478642,-0.0323927686001504,-0.00481960529237035,0.00137112577928633,-0.071555560098993,0.0820368146655834,0.0135503001906496,-0.0329924371750963,-0.0512601647361137,-0.0949539051949237,0.00359076925903246,0.0154231050032827,-0.0318666873837046,0.0173711392983762,-0.000860847918968233,-0.00268230719039053,0.00629619574770146,0.0025419940143987,0.0116879544481332,0.00627213738944121,-0.0110749931124179,0.0447619058580742,0.0599386594052476,0.00166606708514294,-0.00669879453925049,-0.0562711777731586,-0.011597043130763,-0.025017613053949,-0.0673386147232407,-0.00759340116423779,-0.0121767075508483,0.0411091914281911,0.0244682711226396,-0.00345215057673975,0.0886140066198777,0.0042615979389975,-0.0436765986239358,0.000806544774029305],[0.0513781223780757,0.634045539467314,0.819148731816926,0.942200154615566,0.901024990759328,0.888532152894879,0.867336651363699,0.0453957937986944,0.708751426810532,0.229480629215666,0.790851661045003,0.0397695699571223,0.906869646413759,0.996485554724338,0.690509510517983,0.563001508511235,0.447769723483816,0.0944659531358807,0.965565658998151,0.979078547147721,0.239955063304003,0.568604889102091,0.964795450011902,0.977564214243917,0.852962368015702,0.883978349628576,0.989244341107643,0.961584391099153,0.902084124298784,0.960604380735861,0.994426663700835,0.986900737913281,0.990788531632776,0.996113062004583,0.963169376893632,0.968516018293365,0.985744445549692,0.789412737204774,0.854904977520395,0.998061664600051,0.972854153506945,0.868004286610381,0.976891341492997,0.876068048001676,0.983923951274513,0.982247632795467,0.987151199644762,0.99825468740406,0.89495166515414,0.98555287669418,0.676427061130347,0.990375700881878,0.9843762512641,0.997236057082329],[-0.0487469561604321,0.0182657207733543,-0.00830731186438083,0.000830173640068654,0.0131177199867198,-0.00977326245150234,0.0660274129992704,0.0632585851407343,-0.00267461718262041,-0.0571222603339834,-0.0101251526182975,0.0672597899284861,0.0120981175314583,-0.00301930351358148,0.0247401726829839,-0.0126956956551747,-0.0109200999173091,-0.0314004771990914,-0.0292384986382377,0.00334649429861145,-0.0491332289911206,0.0947642045311561,0.0571935431410466,-0.0328306706449979,-0.0780613841769462,-0.00557983408187113,-0.0180772259535118,0.0153105661521113,-0.0669622020146861,-0.0148128346576399,-0.0206275583183064,-0.00345791013112873,-0.0330476553690033,0.012922503034293,-0.0259408442020028,-0.0334274655938103,-0.0110682091198377,-0.00686565525593075,0.016892073731313,-0.0190560353364617,0.0275747170522997,-0.0493173300380718,-0.0479599344614492,-0.0247511826586191,-0.0857309532633828,-0.00764135330427784,-0.0453404001222371,9.8740650951612e-05,0.0251600376765673,-0.0153526943747195,0.121496606681212,-0.013397991758973,-0.00356363374074675,-0.0251280228840223],[0.0505255518562754,0.663125762037874,0.834055018995034,0.969671424688153,0.839258402823942,0.881135623454532,0.811504694525146,0.0453785110463309,0.892057561144188,0.193841827769443,0.441688739288131,0.0607534281233578,0.873306275170462,0.981968380792736,0.582230803924788,0.627751220490996,0.29553874481008,0.101422179404489,0.795755386995978,0.949269064729238,0.413320058237946,0.49815448850004,0.852414676130236,0.977748542748452,0.777206617866202,0.993147876560448,0.94725260989429,0.962366348544735,0.796156377409909,0.966583697853136,0.869561457567207,0.983171953738039,0.949762271798099,0.98001259597387,0.919677561497662,0.830560638873013,0.985271752350262,0.966504439236785,0.959475343039989,0.978040692841086,0.886244291415255,0.885055521506556,0.90329272052829,0.877641289329672,0.978964404875121,0.982019725213141,0.953404915597215,0.999995772750011,0.892354778564564,0.934586202800066,0.575665021140016,0.969598083216937,0.998683392525215,0.914510848272715]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>mbe_est_reverse<\/th>\n      <th>mbe_pval_reverse<\/th>\n      <th>mbe_est_rremove<\/th>\n      <th>mbe_pval_rremove<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[1,2,3,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div id="b-cell" class="section level3">
<h3>B cell</h3>
<div id="histogram-to-show-the-number-of-snps-that-each-pair-contains-1" class="section level4">
<h4>Histogram to show the number of SNPs that each pair contains</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/snp_count_bcell_cedar_qc_kegg_40genes_pruned.jpeg" width="80%">
</figure>
</div>
<div id="effectsize-fitting-summary-1" class="section level4">
<h4>effectsize fitting summary</h4>
<div id="htmlwidget-5ba5287ce9a3b08bdb5d" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-5ba5287ce9a3b08bdb5d">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37"],["EUR.CD_pwy151_pc3","EUR.CD_pwy180_pc1","EUR.CD_pwy185_pc4","T1D_pwy121_pc4","T1D_pwy188_pc5","T1D_pwy295_pc3","T1D_pwy311_pc3","allergy_pwy223_pc5","allergy_pwy224_pc5","baso_pwy317_pc3","baso_pwy319_pc2","eo_pwy144_pc4","eo_pwy235_pc3","eo_pwy283_pc1","hct_pwy210_pc2","hgb_pwy219_pc5","hgb_pwy259_pc5","lymph_pwy101_pc2","lymph_pwy121_pc5","lymph_pwy219_pc1","lymph_pwy234_pc1","lymph_pwy269_pc1","lymph_pwy310_pc1","lymph_pwy311_pc1","mchc_pwy144_pc5","mcv_pwy237_pc1","mcv_pwy310_pc5","mono_pwy176_pc4","mono_pwy202_pc1","myeloid_wbc_pwy120_pc1","neut_pwy184_pc3","plt_pwy304_pc1","rbc_pwy106_pc3","rbc_pwy304_pc4","rbc_pwy305_pc2","ret_pwy127_pc4","wbc_pwy284_pc1"],["04218","04625","04658","04064","04662","05202","05221","04921","04922","05230","05235","04152","04933","05160","04750","04917","05017","03040","04064","04917","04932","05130","05220","05221","04152","04935","05220","04620","04725","04062","04657","05214","03420","05214","05215","04080","05161"],["Cellular senescence","C-type lectin receptor signaling pathway","Th1 and Th2 cell differentiation","NF-kappa B signaling pathway","B cell receptor signaling pathway","Transcriptional misregulation in cancer","Acute myeloid leukemia","Oxytocin signaling pathway","Glucagon signaling pathway","Central carbon metabolism in cancer","PD-L1 expression and PD-1 checkpoint pathway in cancer","AMPK signaling pathway","AGE-RAGE signaling pathway in diabetic complications","Hepatitis C","Inflammatory mediator regulation of TRP channels","Prolactin signaling pathway","Spinocerebellar ataxia","Spliceosome","NF-kappa B signaling pathway","Prolactin signaling pathway","Non-alcoholic fatty liver disease","Pathogenic Escherichia coli infection","Chronic myeloid leukemia","Acute myeloid leukemia","AMPK signaling pathway","Growth hormone synthesis, secretion and action","Chronic myeloid leukemia","Toll-like receptor signaling pathway","Cholinergic synapse","Chemokine signaling pathway","IL-17 signaling pathway","Glioma","Nucleotide excision repair","Glioma","Prostate cancer","Neuroactive ligand-receptor interaction","Hepatitis B"],["112","73","64","69","70","99","49","79","62","44","67","83","51","104","48","42","52","111","69","42","108","123","62","49","83","67","62","63","51","109","46","50","41","50","58","71","110"],[2,2,2,2,2,2,3,2,2,2,2,2,2,2,2,2,2,2,2,3,2,2,2,2,2,3,2,2,3,2,5,2,2,2,4,2,3],[2,2,1,2,2,2,3,2,2,2,2,2,2,2,2,2,2,2,2,3,2,2,2,2,2,3,2,2,3,2,5,2,2,2,4,2,3],[0.011,0.008,0.018,0.035,0.047,0.024,0.002,0.02,0.03,0.009,0.029,0.009,0.013,0.009,0.03,0.004,0.046,0.001,0.024,0.002,0.002,0.007,0.012,0.011,0.016,0.001,0.003,0.007,0.001,0.021,0.031,0.014,0.013,0.006,0,0.001,0.004],[0.575,-0.478,-0.339,-0.307,0.246,0.22,0.301,0.081,0.1,-0.15,0.118,0.068,-0.05,-0.134,-0.125,0.063,0.067,0.056,-0.061,0.111,-0.074,0.105,0.091,0.091,-0.048,-0.109,-0.066,-0.057,-0.119,0.102,-0.068,-0.163,0.1,-0.07,0.108,-0.101,0.109],[1,1,0.999,0.997,0.994,0.999,0.996,0.999,0.998,1,0.998,1,1,1,0.998,1,0.995,1,0.999,0.996,1,1,1,1,0.999,0.998,1,1,0.997,0.999,0.728,1,1,1,0.993,1,0.992],[0.03,0.011,0.041,0.022,0.048,0.012,0.007,0.04,0.046,0.013,0.024,0.011,0.013,0.015,0.032,0.001,0.038,0.001,0.048,0.012,0.002,0.016,0.026,0.017,0.036,0.002,0.007,0.009,0.009,0.033,0.047,0.025,0.031,0.017,0.005,0,0.03]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>factor<\/th>\n      <th>pwy<\/th>\n      <th>pwy_name<\/th>\n      <th>pwy_gene_num<\/th>\n      <th>num_snp_fdr.0.2<\/th>\n      <th>num_snp_noMHC<\/th>\n      <th>pval_in_fitting<\/th>\n      <th>slope_in_fitting<\/th>\n      <th>rsq_in_fitting<\/th>\n      <th>pval_in_resampling<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[5,6,7,8,9,10]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="checking-the-reverse-causality-1" class="section level4">
<h4>Checking the reverse causality</h4>
<p>**The suffix &quot;_reverse&quot; are the results for traits -&gt; LV (using all candidate SNPs after LD Clumping), the suffix &quot;_rremove&quot; are the results for traits -&gt; LV (using all candidate SNPs after LD Clumping but remove the SNPs have fdr &lt; 0.2)**</p>
<div id="htmlwidget-0b516daec537d8efb4b6" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-0b516daec537d8efb4b6">{"x":{"filter":"none","data":[["EUR.CD_pwy151_pc3","EUR.CD_pwy180_pc1","EUR.CD_pwy185_pc4","T1D_pwy121_pc4","T1D_pwy188_pc5","T1D_pwy295_pc3","T1D_pwy311_pc3","allergy_pwy223_pc5","allergy_pwy224_pc5","baso_pwy317_pc3","baso_pwy319_pc2","eo_pwy144_pc4","eo_pwy235_pc3","eo_pwy283_pc1","hct_pwy210_pc2","hgb_pwy219_pc5","hgb_pwy259_pc5","lymph_pwy101_pc2","lymph_pwy121_pc5","lymph_pwy219_pc1","lymph_pwy234_pc1","lymph_pwy269_pc1","lymph_pwy310_pc1","lymph_pwy311_pc1","mchc_pwy144_pc5","mcv_pwy237_pc1","mcv_pwy310_pc5","mono_pwy176_pc4","mono_pwy202_pc1","myeloid_wbc_pwy120_pc1","neut_pwy184_pc3","plt_pwy304_pc1","rbc_pwy106_pc3","rbc_pwy304_pc4","rbc_pwy305_pc2","ret_pwy127_pc4","wbc_pwy284_pc1"],[-0.00175063194954994,0.0107265729796282,-0.0062407820654844,-0.0634850898974681,-0.0143469626222945,-0.0936693836379483,-0.000571298463149184,-0.0078195702717867,0.0181597197722301,-0.0484674846930577,0.0370610723239206,0.0184699053703277,-0.00711607355413335,-0.0107062909699316,-0.00879150262400863,0.0314335667422796,-0.0229209985141559,-0.0261836764761192,-0.0244135208348091,-0.0254286821864449,0.0182952923504125,0.00865846226739997,-0.0476300941442496,-0.000836201736833075,0.0794032933369451,0.000802155890815115,-0.0209870275057966,0.00470100688211161,-0.0164493204862044,0.0309251838443156,-0.00839789294991533,-0.00834702993840963,0.00847303592559712,0.0403381399659857,-0.0344424870437265,0.00177017330076268,0.0186071281668099],[0.97655809151451,0.566562636963742,0.854015129563213,0.0925876525678437,0.693471096601848,0.0167835255813016,0.965182542702145,0.905754177497624,0.657701811185994,0.712415457224003,0.73173196154579,0.898965854315474,0.982540760542793,0.973286815436498,0.997805987720067,0.8439261642598,0.871809769482944,0.934793942953457,0.955554093379971,0.977346942188235,0.983675877972056,0.977718702704144,0.905647854354113,0.996315684613981,0.600299290629456,0.994989451933441,0.995874662180452,0.996676700952903,0.846809241133738,0.950665441133011,0.99515838296499,0.957701169575508,0.972841092413757,0.800223597406579,0.87938777191722,0.999630806812795,0.98769084656198],[-0.0012563719336125,0.0139789683613287,-0.00555333481804521,-0.0576373041530918,-0.00247029452445524,-0.091328007045413,-0.00154359237366464,-0.00378928210770013,0.0207510658108916,-0.0138359417770619,0.0575322130410179,0.0314625862816271,0.0231975940506075,-0.028744337077848,-0.0215756208383784,-0.00844814450147524,-0.0402588024841961,-0.0347412627420915,-0.0127583809189566,0.00326165876935036,0.0412290702188294,-0.0126832401174628,-0.0154112283493237,0.00459983346565807,0.0886657322717372,0.0151733692831488,-0.0181186056791525,-0.0248462990603624,-0.0160380708939645,0.0447100883649032,-0.00152505730402908,-0.00111855280840345,-0.031417537979106,0.0384796307167292,-0.0327666039476746,0.00442767421492896,0.0134544789205648],[0.983370568303204,0.454114477896783,0.868994583300795,0.127349484867141,0.944771683904052,0.0214695545980346,0.902583174041074,0.953590044351037,0.605536732748868,0.914314120454094,0.600285234003116,0.829383596750444,0.943095091536232,0.92762986062064,0.994458774727254,0.957569724441011,0.774681972100557,0.91209319377091,0.976755176989418,0.997003718320974,0.963858508273912,0.967089193339579,0.969324444375333,0.980000064054597,0.544228947401263,0.906503092211366,0.996541233479647,0.982980925584247,0.850650778037176,0.930388218917222,0.999112004835405,0.994224582472184,0.902890599668221,0.810924857683953,0.883077809330874,0.999073666588756,0.991152680028101]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>mbe_est_reverse<\/th>\n      <th>mbe_pval_reverse<\/th>\n      <th>mbe_est_rremove<\/th>\n      <th>mbe_pval_rremove<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[1,2,3,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div id="t-cellcd4" class="section level3">
<h3>T cell(CD4)</h3>
<div id="histogram-to-show-the-number-of-snps-that-each-pair-contains-2" class="section level4">
<h4>Histogram to show the number of SNPs that each pair contains</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/snp_count_tcell_cedar_qc_kegg_40genes_pruned.jpeg" width="80%">
</figure>
</div>
<div id="effectsize-fitting-summary-2" class="section level4">
<h4>effectsize fitting summary</h4>
<div id="htmlwidget-d0109ea5b7e7d667f148" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-d0109ea5b7e7d667f148">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43"],["allergy_pwy219_pc5","baso_pwy94_pc2","eo_pwy113_pc1","eo_pwy120_pc1","eo_pwy143_pc1","eo_pwy153_pc1","eo_pwy162_pc4","eo_pwy178_pc4","eo_pwy212_pc1","eo_pwy256_pc2","eo_pwy304_pc1","eo_pwy310_pc1","eo_pwy311_pc1","EUR.CD_pwy187_pc1","EUR.CD_pwy300_pc1","EUR.CD_pwy86_pc4","gran_pwy120_pc1","hgb_pwy217_pc3","lymph_pwy322_pc5","lymph_pwy56_pc4","mch_pwy137_pc3","mch_pwy174_pc5","mch_pwy191_pc2","mch_pwy271_pc3","mch_pwy285_pc3","mch_pwy294_pc3","mch_pwy88_pc4","mcv_pwy165_pc5","mcv_pwy293_pc5","mcv_pwy88_pc4","mono_pwy228_pc1","neut_pwy93_pc5","pct_pwy119_pc3","pct_pwy126_pc1","pct_pwy145_pc3","pct_pwy161_pc3","pct_pwy221_pc5","plt_pwy183_pc2","plt_pwy266_pc5","rbc_pwy168_pc1","rbc_pwy303_pc1","rdw_pwy181_pc2","T1D_pwy137_pc4"],["04917","03010","04014","04062","04151","04261","04380","04622","04910","05012","05214","05220","05221","04660","05210","00983","04062","04915","05321","00564","04141","04612","04668","05132","05162","05200","01521","04510","05170","01521","04926","03008","04061","04072","04210","04371","04919","04650","05100","04520","05213","04630","04141"],["Prolactin signaling pathway","Ribosome","Ras signaling pathway","Chemokine signaling pathway","PI3K-Akt signaling pathway","Adrenergic signaling in cardiomyocytes","Osteoclast differentiation","RIG-I-like receptor signaling pathway","Insulin signaling pathway","Parkinson disease","Glioma","Chronic myeloid leukemia","Acute myeloid leukemia","T cell receptor signaling pathway","Colorectal cancer","Drug metabolism - other enzymes","Chemokine signaling pathway","Estrogen signaling pathway","Inflammatory bowel disease","Glycerophospholipid metabolism","Protein processing in endoplasmic reticulum","Antigen processing and presentation","TNF signaling pathway","Salmonella infection","Measles","Pathways in cancer","EGFR tyrosine kinase inhibitor resistance","Focal adhesion","Human immunodeficiency virus 1 infection","EGFR tyrosine kinase inhibitor resistance","Relaxin signaling pathway","Ribosome biogenesis in eukaryotes","Viral protein interaction with cytokine and cytokine receptor","Phospholipase D signaling pathway","Apoptosis","Apelin signaling pathway","Thyroid hormone signaling pathway","Natural killer cell mediated cytotoxicity","Bacterial invasion of epithelial cells","Adherens junction","Endometrial cancer","JAK-STAT signaling pathway","Protein processing in endoplasmic reticulum"],["43","126","120","112","177","71","90","42","97","87","53","63","52","84","64","41","112","67","44","47","133","59","75","162","99","299","49","99","146","49","68","63","46","77","110","70","83","88","48","41","43","86","133"],[2,2,2,2,2,2,2,3,3,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,3,2,4,2,3,2,2,2,2,3,2,2,3,2,2,2,2,2,2],[2,2,2,2,2,2,2,3,3,2,2,2,2,1,0,2,2,2,2,2,2,2,2,2,3,2,4,2,3,2,2,2,2,3,2,2,3,2,2,1,2,2,2],[0.024,0.024,0.066,0.077,0.071,0.127,0.04,0.107,0.052,0.038,0.079,0.084,0.086,0.008,0.036,0,0.014,0,0.05,0.015,0.003,0.007,0.008,0,0.001,0.005,0.007,0.022,0.002,0.024,0.014,0.018,0.026,0.001,0.018,0.005,0.003,0.001,0.012,0.005,0.011,0.02,0.011],[0.098,-0.057,-0.054,0.057,0.047,-0.063,-0.069,0.056,-0.061,0.058,-0.075,-0.064,0.064,0.672,0.946,0.491,0.082,-0.082,0.047,-0.043,0.047,-0.048,0.051,0.06,0.047,0.041,-0.125,-0.043,0.035,-0.123,0.086,-0.067,0.057,0.108,-0.061,0.061,-0.056,0.061,0.071,0.082,0.138,0.083,0.23],[0.999,0.999,0.989,0.985,0.988,0.961,0.996,0.798,0.9,0.996,0.984,0.983,0.982,1,0.997,1,0.999,1,0.994,0.999,1,1,1,1,0.998,1,0.937,0.999,0.996,0.999,1,0.999,0.998,0.998,0.999,1,0.995,1,1,1,1,0.999,1],[0.023,0.049,0.018,0.018,0.018,0.032,0.043,0.041,0.018,0.004,0.018,0.018,0.018,0.008,0.034,0,0.041,0,0.018,0.027,0,0.007,0.02,0,0.001,0.006,0.02,0.045,0.002,0.033,0.029,0.038,0.023,0.003,0.025,0.011,0.004,0.001,0.033,0.01,0.029,0.022,0.025]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>factor<\/th>\n      <th>pwy<\/th>\n      <th>pwy_name<\/th>\n      <th>pwy_gene_num<\/th>\n      <th>num_snp_fdr.0.2<\/th>\n      <th>num_snp_noMHC<\/th>\n      <th>pval_in_fitting<\/th>\n      <th>slope_in_fitting<\/th>\n      <th>rsq_in_fitting<\/th>\n      <th>pval_in_resampling<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[5,6,7,8,9,10]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="checking-the-reverse-causality-2" class="section level4">
<h4>Checking the reverse causality</h4>
<p>**The suffix &quot;_reverse&quot; are the results for traits -&gt; LV (using all candidate SNPs after LD Clumping), the suffix &quot;_rremove&quot; are the results for traits -&gt; LV (using all candidate SNPs after LD Clumping but remove the SNPs have fdr &lt; 0.2)**</p>
<div id="htmlwidget-7d2b77686a7a668a5a8b" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-7d2b77686a7a668a5a8b">{"x":{"filter":"none","data":[["allergy_pwy219_pc5","baso_pwy94_pc2","eo_pwy113_pc1","eo_pwy120_pc1","eo_pwy143_pc1","eo_pwy153_pc1","eo_pwy162_pc4","eo_pwy178_pc4","eo_pwy212_pc1","eo_pwy256_pc2","eo_pwy304_pc1","eo_pwy310_pc1","eo_pwy311_pc1","EUR.CD_pwy187_pc1","EUR.CD_pwy300_pc1","EUR.CD_pwy86_pc4","gran_pwy120_pc1","hgb_pwy217_pc3","lymph_pwy322_pc5","lymph_pwy56_pc4","mch_pwy137_pc3","mch_pwy174_pc5","mch_pwy191_pc2","mch_pwy271_pc3","mch_pwy285_pc3","mch_pwy294_pc3","mch_pwy88_pc4","mcv_pwy165_pc5","mcv_pwy293_pc5","mcv_pwy88_pc4","mono_pwy228_pc1","neut_pwy93_pc5","pct_pwy119_pc3","pct_pwy126_pc1","pct_pwy145_pc3","pct_pwy161_pc3","pct_pwy221_pc5","plt_pwy183_pc2","plt_pwy266_pc5","rbc_pwy168_pc1","rbc_pwy303_pc1","rdw_pwy181_pc2","T1D_pwy137_pc4"],[-0.0336882827189147,-0.0107447105046248,0.0454703643480698,0.00780493418070094,0.0214635102379024,0.0462305292258094,0.0323335702712093,-0.0196710955665633,0.0277008847156281,-0.0153236089802977,0.019845651229808,-0.0221303805445814,0.0271813244940642,-0.00477904095419968,0.0149295383461461,-0.0819393637156569,-0.0193711205133766,-0.0326330194243862,0.0377670206940088,0.0156469775915724,-0.0313611998956631,-0.041164522400706,0.0199041367035804,0.00622688961749418,-0.0241457592549956,0.0746679041848921,0.0438190685433071,0.0328594775722664,-0.183569563154968,-0.0121685049938982,-0.0202009952506099,0.0112680704486792,-0.0851352291935896,-0.0220297677500465,-0.0201777221177188,0.0127387569285915,0.00651343077007738,0.00864340978283451,0.0462005333688431,0.00278915281945302,-0.0136648076185448,-0.0315803264574228,-0.0170493988328007],[0.774803414235098,0.947927488847394,0.898544580687157,0.950646462807201,0.815321569174677,0.788668921097388,0.909595636689227,0.969862520415994,0.863993986095842,0.953641933535531,0.974530546203188,0.884060335305255,0.871015991180633,0.930788659526204,0.519405599967053,0.199273637300375,0.980206915843383,0.892564506900867,0.854121034846203,0.974465640144536,0.949877264077443,0.889979239579158,0.997294092274422,0.980511657256711,0.961604614820142,0.944589021868583,0.929089900739647,0.945125416762831,0.738614608921067,0.972080289364675,0.876448591777076,0.975153916972418,0.856360886286571,0.914201143128854,0.849484047466083,0.953571877624846,0.99065826397347,0.995618456433205,0.948203494014844,0.998884753632921,0.990802104508315,0.825191372953913,0.308468697467671],[-0.0597481142835967,-0.00413305374559769,0.00511686603180017,0.0295093010910783,0.0171472706460225,0.0381681964749561,0.0187544664729842,-0.0176296227647903,-0.0185655515316565,0.0315971089259897,0.0185779340233196,-0.0347781997822274,-0.0130400504515453,-0.00999057477286702,0.0142090364954893,-0.0845963638009946,-0.0223591552611584,-0.0519841947759794,0.0684996410496872,-0.00989459061193187,-0.0111976087193426,-0.00553066540694047,0.0371324386542717,0.0206121066210123,0.00161210063049566,-0.0269851773868375,0.024576216946544,-0.0225520528657235,-0.177777923978947,-0.00748076503532502,-0.0160784518829633,0.0576277814554196,-0.0551414016226843,-0.015670064242908,0.0109848283397724,-0.0114455095188148,0.0169177894190113,0.0134418474314018,0.0489307523712874,-0.0087287417014501,-0.0219518428632295,-0.0177459732201459,-0.0299225977158253],[0.610336256275364,0.979933591811757,0.988767850664648,0.813192660373291,0.856100913010765,0.82478090419126,0.947659475146424,0.972354306999559,0.906927533283624,0.904417602106977,0.977395709602808,0.820598364821115,0.938682007252477,0.851759317914185,0.542524393454501,0.191467198762694,0.976753884099717,0.831610073794904,0.736213192329163,0.983529719705249,0.98164783153203,0.985048679181053,0.994988568487289,0.935172424990371,0.997381760447651,0.979733161040218,0.959845110580029,0.963528979174592,0.748588009444827,0.982703984399505,0.902384220159395,0.875263827427464,0.906125493742693,0.939578363820005,0.916856893795713,0.958568448131028,0.975774322645472,0.993468714345967,0.943230372546835,0.996635465755421,0.985204381066893,0.902400692033656,0.0639699808639428]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>mbe_est_reverse<\/th>\n      <th>mbe_pval_reverse<\/th>\n      <th>mbe_est_rremove<\/th>\n      <th>mbe_pval_rremove<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[1,2,3,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div id="neutrophils-cd15" class="section level3">
<h3>neutrophils CD15</h3>
<div id="histogram-to-show-the-number-of-snps-that-each-pair-contains-3" class="section level4">
<h4>Histogram to show the number of SNPs that each pair contains</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/snp_count_cd15_cedar_qc_kegg_40genes_pruned.jpeg" width="80%">
</figure>
</div>
<div id="effectsize-fitting-summary-3" class="section level4">
<h4>effectsize fitting summary</h4>
<div id="htmlwidget-1e347d3f3a7c82c854df" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-1e347d3f3a7c82c854df">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32"],["EUR.CD_pwy56_pc4","EUR.IBD_pwy120_pc3","EUR.IBD_pwy182_pc4","EUR.UC_pwy304_pc2","gran_pwy298_pc4","hgb_pwy93_pc5","mchc_pwy183_pc4","mchc_pwy285_pc1","mchc_pwy315_pc2","mpv_pwy217_pc4","mpv_pwy233_pc5","mpv_pwy268_pc5","mpv_pwy94_pc1","myeloid_wbc_pwy214_pc3","myeloid_wbc_pwy228_pc3","myeloid_wbc_pwy305_pc2","neut_pwy162_pc1","neut_pwy298_pc4","pct_pwy135_pc3","pct_pwy316_pc5","pdw_pwy181_pc4","pdw_pwy291_pc4","pdw_pwy305_pc5","rbc_pwy122_pc2","rdw_pwy233_pc4","rdw_pwy268_pc4","rdw_pwy286_pc3","rdw_pwy299_pc4","rdw_pwy319_pc5","rdw_pwy324_pc5","ret_pwy128_pc3","ret_pwy296_pc5"],["00564","04062","04640","05214","05205","03008","04650","05162","05225","04915","04931","05120","03010","04912","04926","05215","04380","05205","04137","05226","04630","05168","05215","04066","04931","05120","05163","05206","05235","05323","04110","05203"],["Glycerophospholipid metabolism","Chemokine signaling pathway","Hematopoietic cell lineage","Glioma","Proteoglycans in cancer","Ribosome biogenesis in eukaryotes","Natural killer cell mediated cytotoxicity","Measles","Hepatocellular carcinoma","Estrogen signaling pathway","Insulin resistance","Epithelial cell signaling in Helicobacter pylori infection","Ribosome","GnRH signaling pathway","Relaxin signaling pathway","Prostate cancer","Osteoclast differentiation","Proteoglycans in cancer","Mitophagy - animal","Gastric cancer","JAK-STAT signaling pathway","Herpes simplex virus 1 infection","Prostate cancer","HIF-1 signaling pathway","Insulin resistance","Epithelial cell signaling in Helicobacter pylori infection","Human cytomegalovirus infection","MicroRNAs in cancer","PD-L1 expression and PD-1 checkpoint pathway in cancer","Rheumatoid arthritis","Cell cycle","Viral carcinogenesis"],["46","100","45","45","102","42","66","87","92","67","56","47","105","51","62","59","89","102","43","60","73","237","59","67","56","47","136","99","62","48","69","121"],[2,2,2,3,2,2,2,5,2,3,2,2,2,2,2,2,2,2,2,2,2,2,3,2,2,2,3,2,2,2,2,2],[2,2,1,3,2,2,2,5,2,3,2,2,2,2,2,2,2,2,2,2,2,2,3,2,2,2,3,2,2,1,2,2],[0.014,0.144,0.015,0.006,0.005,0.012,0.032,0.012,0.016,0.007,0.004,0.004,0.055,0.003,0.014,0.01,0.031,0.008,0.001,0.019,0.003,0.013,0.002,0.01,0.015,0.003,0.001,0.007,0.016,0.021,0.016,0],[0.421,0.269,-0.362,0.627,0.082,0.048,0.1,0.187,0.081,0.078,0.079,-0.049,-0.048,0.074,-0.068,-0.11,0.085,0.08,0.081,-0.064,-0.051,0.031,0.069,-0.089,-0.063,-0.068,0.047,-0.055,-0.061,-0.054,0.072,0.074],[1,0.95,0.999,0.988,1,1,0.997,0.824,0.999,0.986,1,1,0.992,1,1,1,0.998,1,1,0.999,1,1,0.997,1,0.999,1,0.998,1,0.999,0.999,0.999,1],[0.034,0.041,0.025,0.037,0.007,0.009,0.042,0.044,0.015,0.027,0.006,0.01,0.035,0.012,0.041,0.035,0.026,0.012,0.003,0.042,0.004,0.029,0.008,0.022,0.03,0.004,0.004,0.022,0.031,0.037,0.037,0.001]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>factor<\/th>\n      <th>pwy<\/th>\n      <th>pwy_name<\/th>\n      <th>pwy_gene_num<\/th>\n      <th>num_snp_fdr.0.2<\/th>\n      <th>num_snp_noMHC<\/th>\n      <th>pval_in_fitting<\/th>\n      <th>slope_in_fitting<\/th>\n      <th>rsq_in_fitting<\/th>\n      <th>pval_in_resampling<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[5,6,7,8,9,10]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="checking-the-reverse-causality-3" class="section level4">
<h4>Checking the reverse causality</h4>
<p>**The suffix &quot;_reverse&quot; are the results for traits -&gt; LV (using all candidate SNPs after LD Clumping), the suffix &quot;_rremove&quot; are the results for traits -&gt; LV (using all candidate SNPs after LD Clumping but remove the SNPs have fdr &lt; 0.2)**</p>
<div id="htmlwidget-32695f625600a1e2a463" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-32695f625600a1e2a463">{"x":{"filter":"none","data":[["EUR.CD_pwy56_pc4","EUR.IBD_pwy120_pc3","EUR.IBD_pwy182_pc4","EUR.UC_pwy304_pc2","gran_pwy298_pc4","hgb_pwy93_pc5","mchc_pwy183_pc4","mchc_pwy285_pc1","mchc_pwy315_pc2","mpv_pwy217_pc4","mpv_pwy233_pc5","mpv_pwy268_pc5","mpv_pwy94_pc1","myeloid_wbc_pwy214_pc3","myeloid_wbc_pwy228_pc3","myeloid_wbc_pwy305_pc2","neut_pwy162_pc1","neut_pwy298_pc4","pct_pwy135_pc3","pct_pwy316_pc5","pdw_pwy181_pc4","pdw_pwy291_pc4","pdw_pwy305_pc5","rbc_pwy122_pc2","rdw_pwy233_pc4","rdw_pwy268_pc4","rdw_pwy286_pc3","rdw_pwy299_pc4","rdw_pwy319_pc5","rdw_pwy324_pc5","ret_pwy128_pc3","ret_pwy296_pc5"],[-0.00363188048164798,0.000570693457285998,-0.0295944245467221,0.0578442149478131,0.0154469070035361,0.00231066938881597,-0.0119357277267653,0.0236940987619558,0.00659016506619281,-0.0100721663661432,-0.024478484438891,0.00997347514540081,-0.027136466135552,0.0222767908947006,0.0277285624238743,-0.0425392959159705,-0.0261565216438449,0.0172619216877337,-0.0158967440061897,-0.0118463646458373,0.0023950426685424,0.080189261822202,0.0308643438752156,-0.00527274759384078,0.00755256639133961,0.0607673680313425,-0.0227505224469589,-0.0196715504562555,-0.00775913316459409,0.00432152559600496,0.0210252391260344,-0.0379837258300899],[0.992819847817152,0.981800294408656,0.392597354981878,0.0604558345050147,0.956240474566979,0.994047776917414,0.894350993973148,0.716564141557156,0.944133609786901,0.97797622984929,0.937150680598015,0.969218861505057,0.959640145796158,0.872834977123072,0.960322485021406,0.533029631004161,0.832715162818593,0.95203028069044,0.959775173331808,0.983619466741109,0.990665138949298,0.960836494320397,0.850838397055784,0.994322895787984,0.973385928774702,0.616195324723244,0.939325256521756,0.899983012680462,0.981883923587242,0.984385063668704,0.944000780686128,0.901741320150866],[-0.00407578456688462,0.00178493596251084,-0.0296684104018099,0.0640480492828779,-0.0151331596042574,0.01830876580701,-0.00550844708184961,0.0379790045554369,0.0107832254669411,0.000741689415932001,0.0194542218319924,0.0191192527689736,0.0329840323441672,0.0371849552269889,0.0237940157792735,-0.0401874274169725,-0.0214201195852777,0.0227352268367298,0.0107664051614673,-0.00627438635431687,0.00066099582717527,0.0814054090445921,0.00381256731669666,-0.0227105403535823,0.0214691107584031,0.0454906152897436,-0.00857644586605133,-0.0234222043437491,0.0335642274159476,-0.0363691807702722,-0.0040379040708558,0.0163359444892635],[0.992394442482598,0.942529600372357,0.390611469458075,0.0391546472810669,0.95600609302293,0.953028134233239,0.950870914588695,0.558506175025233,0.908199521127859,0.998393724768113,0.949913726926726,0.94056502193985,0.951127480438686,0.785209692020617,0.965790707174423,0.55057717026978,0.861553496452361,0.936094140648459,0.97346214093586,0.991471183722968,0.997418683878937,0.960483134996247,0.981344343567917,0.975069019417696,0.924383454261204,0.704082474535585,0.976851065773722,0.879309526813126,0.920145000289014,0.869501907837563,0.989105283180113,0.95485252384317]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>mbe_est_reverse<\/th>\n      <th>mbe_pval_reverse<\/th>\n      <th>mbe_est_rremove<\/th>\n      <th>mbe_pval_rremove<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[1,2,3,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div id="t-cellcd8" class="section level3">
<h3>T cell(CD8)</h3>
<div id="histogram-to-show-the-number-of-snps-that-each-pair-contains-4" class="section level4">
<h4>Histogram to show the number of SNPs that each pair contains</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/snp_count_thy_cedar_qc_kegg_40genes_pruned.jpeg" width="80%">
</figure>
</div>
<div id="effectsize-fitting-summary-4" class="section level4">
<h4>effectsize fitting summary</h4>
<div id="htmlwidget-312b9245b08df84edcee" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-312b9245b08df84edcee">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38"],["EUR.CD_pwy219_pc5","EUR.CD_pwy237_pc5","EUR.CD_pwy41_pc1","EUR.IBD_pwy324_pc2","EUR.UC_pwy146_pc3","EUR.UC_pwy182_pc3","EUR.UC_pwy202_pc4","EUR.UC_pwy221_pc3","EUR.UC_pwy303_pc2","EUR.UC_pwy314_pc2","EUR.UC_pwy324_pc2","EUR.UC_pwy86_pc1","T1D_pwy101_pc2","hct_pwy129_pc2","hct_pwy218_pc5","hct_pwy304_pc5","hgb_pwy131_pc1","hgb_pwy282_pc3","lymph_pwy265_pc5","lymph_pwy287_pc3","lymph_pwy302_pc4","mchc_pwy200_pc1","mchc_pwy300_pc4","mono_pwy140_pc4","mono_pwy154_pc5","mono_pwy171_pc3","mpv_pwy265_pc2","mpv_pwy56_pc5","pdw_pwy188_pc5","rbc_pwy276_pc5","ukb.allasthma_pwy301_pc3","ukb.allasthma_pwy332_pc5","wbc_pwy127_pc1","wbc_pwy191_pc4","wbc_pwy304_pc4","wbc_pwy315_pc3","wbc_pwy86_pc2","wbc_pwy96_pc5"],["04917","04935","00510","05323","04211","04640","04725","04919","05213","05224","05323","00983","03040","04114","04916","05214","04120","05152","05034","05164","05212","04723","05210","04145","04270","04550","05034","00564","04662","05142","05211","05418","04080","04668","05214","05225","00983","03015"],["Prolactin signaling pathway","Growth hormone synthesis, secretion and action","N-Glycan biosynthesis","Rheumatoid arthritis","Longevity regulating pathway","Hematopoietic cell lineage","Cholinergic synapse","Thyroid hormone signaling pathway","Endometrial cancer","Breast cancer","Rheumatoid arthritis","Drug metabolism - other enzymes","Spliceosome","Oocyte meiosis","Melanogenesis","Glioma","Ubiquitin mediated proteolysis","Tuberculosis","Alcoholism","Influenza A","Pancreatic cancer","Retrograde endocannabinoid signaling","Colorectal cancer","Phagosome","Vascular smooth muscle contraction","Signaling pathways regulating pluripotency of stem cells","Alcoholism","Glycerophospholipid metabolism","B cell receptor signaling pathway","Chagas disease","Renal cell carcinoma","Fluid shear stress and atherosclerosis","Neuroactive ligand-receptor interaction","TNF signaling pathway","Glioma","Hepatocellular carcinoma","Drug metabolism - other enzymes","mRNA surveillance pathway"],["43","71","41","54","57","64","56","84","44","73","54","41","112","76","48","53","116","120","62","122","58","71","65","97","54","63","62","46","70","64","51","90","71","75","53","106","41","72"],[2,2,2,2,2,2,3,3,2,2,2,2,2,2,2,3,2,2,2,2,3,2,2,2,2,2,2,2,2,2,2,2,3,2,2,3,2,3],[2,2,2,2,2,2,3,2,2,2,2,2,2,2,2,3,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,1,3,2,2,3,2,3],[0.015,0.011,0.015,0.006,0.011,0.001,0.002,0.005,0.009,0.01,0.013,0.008,0.004,0.017,0.034,0.003,0.012,0.007,0.126,0.001,0.016,0.013,0.015,0.008,0.046,0.02,0.045,0.02,0.031,0.058,0.008,0.017,0.008,0.006,0.014,0,0.014,0.002],[-0.32,0.244,0.582,0.347,-0.512,0.513,-0.42,-0.42,0.351,-0.51,0.393,0.631,0.392,0.069,0.066,-0.071,-0.07,0.051,0.051,0.045,-0.118,-0.072,0.044,-0.05,-0.069,-0.075,0.074,0.07,0.101,-0.055,-0.168,-0.152,0.052,0.037,-0.043,0.053,-0.069,0.049],[0.999,1,0.999,1,1,1,0.996,0.99,1,1,1,1,1,0.999,0.997,0.995,1,1,0.961,1,0.969,1,0.999,1,0.995,0.999,0.995,0.999,0.998,0.992,1,0.999,0.985,1,1,1,1,0.995],[0.027,0.026,0.042,0.01,0.029,0.004,0.016,0.037,0.036,0.045,0.043,0.023,0.019,0.014,0.022,0.036,0.022,0.026,0.033,0.003,0.023,0.044,0.04,0.015,0.043,0.037,0.044,0.043,0.03,0.047,0.02,0.049,0.033,0.016,0.028,0,0.028,0.013]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>factor<\/th>\n      <th>pwy<\/th>\n      <th>pwy_name<\/th>\n      <th>pwy_gene_num<\/th>\n      <th>num_snp_fdr.0.2<\/th>\n      <th>num_snp_noMHC<\/th>\n      <th>pval_in_fitting<\/th>\n      <th>slope_in_fitting<\/th>\n      <th>rsq_in_fitting<\/th>\n      <th>pval_in_resampling<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[5,6,7,8,9,10]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="checking-the-reverse-causality-4" class="section level4">
<h4>Checking the reverse causality</h4>
<p>**The suffix &quot;_reverse&quot; are the results for traits -&gt; LV (using all candidate SNPs after LD Clumping), the suffix &quot;_rremove&quot; are the results for traits -&gt; LV (using all candidate SNPs after LD Clumping but remove the SNPs have fdr &lt; 0.2)**</p>
<div id="htmlwidget-3cd2a26b2bcc583003e2" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-3cd2a26b2bcc583003e2">{"x":{"filter":"none","data":[["EUR.CD_pwy219_pc5","EUR.CD_pwy237_pc5","EUR.CD_pwy41_pc1","EUR.IBD_pwy324_pc2","EUR.UC_pwy146_pc3","EUR.UC_pwy182_pc3","EUR.UC_pwy202_pc4","EUR.UC_pwy221_pc3","EUR.UC_pwy303_pc2","EUR.UC_pwy314_pc2","EUR.UC_pwy324_pc2","EUR.UC_pwy86_pc1","T1D_pwy101_pc2","hct_pwy129_pc2","hct_pwy218_pc5","hct_pwy304_pc5","hgb_pwy131_pc1","hgb_pwy282_pc3","lymph_pwy265_pc5","lymph_pwy287_pc3","lymph_pwy302_pc4","mchc_pwy200_pc1","mchc_pwy300_pc4","mono_pwy140_pc4","mono_pwy154_pc5","mono_pwy171_pc3","mpv_pwy265_pc2","mpv_pwy56_pc5","pdw_pwy188_pc5","rbc_pwy276_pc5","ukb.allasthma_pwy301_pc3","ukb.allasthma_pwy332_pc5","wbc_pwy127_pc1","wbc_pwy191_pc4","wbc_pwy304_pc4","wbc_pwy315_pc3","wbc_pwy86_pc2","wbc_pwy96_pc5"],[7.58321087168667e-06,-0.00503209368451785,0.0031531620578289,0.0255204747277853,-0.00380683670461712,-0.0209809283491249,0.02274149452826,-0.00168611636848492,-0.00383651776971128,0.0158114882859457,0.0147067222552923,0.0266096953609165,-0.0208273434073321,-0.00784910940059902,0.0110854390677559,0.0597502463274679,-0.0329527029977612,0.00252194638082059,0.0328369579703249,-0.00185643734277363,-0.00932295405095473,-0.0403724267030157,0.027194680792693,-0.0334175309837406,0.0111898686205834,0.0378385068725784,0.0236175978375268,0.0195929664907233,-0.0290288036865967,0.0171267653126819,-0.0155519394069596,-0.0138975074910945,-0.00560981949567108,-0.0267744992809682,0.0455690299719684,0.034298403811917,-0.0418385360118378,-0.0426370715870839],[0.99981741796587,0.761710666562154,0.922645610663168,0.89610452272446,0.90955229444654,0.292218040317523,0.0802835003652967,0.980329524885211,0.997165122154058,0.982687162195326,0.391032158175869,0.250911190674328,0.277807781188467,0.993010322871742,0.960888442566487,0.87471192980652,0.933733566127193,0.996609474807427,0.807707733714672,0.994439687614204,0.986175687213454,0.717355298942053,0.977168696366181,0.965738184245559,0.986108754641743,0.695466014091125,0.948904122242717,0.985456712995306,0.910390843377951,0.952888445670673,0.9894986484594,0.987457326979231,0.986619011368173,0.945909016717182,0.963618691001784,0.938065789812996,0.959507266079635,0.9926448050166],[0.00480432354976434,-0.00515334123421241,0.00278385036775242,0.030883710034888,0.00279135331611258,-0.0193071043159474,0.0216711325460945,0.000392955279372664,0.00780116466324898,0.00898785682020309,0.0125881335889346,0.0211540029781627,-0.0213215084984304,-0.0290334784298771,0.0147453540414304,0.00538106844576447,-0.0102006504102228,0.0387368418753979,-0.014903385140915,0.00849739383316361,-0.0101504915509238,-0.0236620593206318,0.0095562928580506,0.0329472867369027,-0.0240581848066146,0.0273080902545644,-0.0033393732056286,0.0193626853018785,0.00773513163173689,0.00699481667906809,0.00171429652287092,-0.0238137193964345,-0.0109403997666,-0.0829319173928553,-0.00938388198091644,0.00585018162431794,-0.0311188792652146,-0.0268870564380279],[0.886338883638048,0.748504917669769,0.931309358038373,0.87351028844477,0.933343826885435,0.330322536590673,0.0915990619875868,0.995456758438379,0.994010172591913,0.990009224009443,0.453310862559279,0.372419484134229,0.255989806318476,0.974160362157365,0.947632322281271,0.988749075697888,0.97975993096112,0.947915903796343,0.915627246823996,0.974291765456631,0.984877488923215,0.833143978013442,0.992094451259631,0.965765736606749,0.969557166478919,0.780724326256091,0.993052341690068,0.984970436824592,0.976573649940452,0.98052523463909,0.998833590534144,0.97856838488391,0.97338796350731,0.834702064880492,0.992438651820544,0.989211721410775,0.970408066165529,0.99548011218202]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>mbe_est_reverse<\/th>\n      <th>mbe_pval_reverse<\/th>\n      <th>mbe_est_rremove<\/th>\n      <th>mbe_pval_rremove<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[1,2,3,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div id="platelet" class="section level3">
<h3>Platelet</h3>
<div id="histogram-to-show-the-number-of-snps-that-each-pair-contains-5" class="section level4">
<h4>Histogram to show the number of SNPs that each pair contains</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/snp_count_plt_cedar_qc_kegg_40genes_pruned.jpeg" width="80%">
</figure>
</div>
<div id="effectsize-fitting-summary-5" class="section level4">
<h4>effectsize fitting summary</h4>
<div id="htmlwidget-9ad70954d77b9fd07fab" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-9ad70954d77b9fd07fab">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27"],["EUR.IBD_pwy162_pc1","EUR.IBD_pwy176_pc1","EUR.IBD_pwy287_pc1","T1D_pwy131_pc5","allergy_pwy259_pc5","baso_pwy180_pc2","hct_pwy187_pc2","hgb_pwy159_pc3","hgb_pwy187_pc2","hgb_pwy255_pc5","hgb_pwy315_pc3","mch_pwy228_pc1","mch_pwy323_pc5","mchc_pwy272_pc2","mcv_pwy120_pc3","mono_pwy161_pc4","mpv_pwy285_pc1","myeloid_wbc_pwy222_pc1","pct_pwy56_pc4","plt_pwy138_pc3","rdw_pwy114_pc3","ret_pwy150_pc2","ret_pwy161_pc1","ret_pwy169_pc5","ukb.allasthma_pwy138_pc1","ukb.allasthma_pwy284_pc3","wbc_pwy228_pc2"],["04380","04620","05164","04120","05017","04625","04660","04360","04660","05010","05225","04926","05322","05133","04062","04371","05162","04920","00564","04142","04015","04217","04371","04530","04142","05161","04926"],["Osteoclast differentiation","Toll-like receptor signaling pathway","Influenza A","Ubiquitin mediated proteolysis","Spinocerebellar ataxia","C-type lectin receptor signaling pathway","T cell receptor signaling pathway","Axon guidance","T cell receptor signaling pathway","Alzheimer disease","Hepatocellular carcinoma","Relaxin signaling pathway","Systemic lupus erythematosus","Pertussis","Chemokine signaling pathway","Apelin signaling pathway","Measles","Adipocytokine signaling pathway","Glycerophospholipid metabolism","Lysosome","Rap1 signaling pathway","Necroptosis","Apelin signaling pathway","Tight junction","Lysosome","Hepatitis B","Relaxin signaling pathway"],["86","63","119","113","52","73","76","84","76","111","99","65","62","49","109","72","96","48","45","99","104","106","72","93","99","110","65"],[2,2,2,2,2,2,2,2,2,2,3,2,2,2,2,2,2,2,2,2,2,2,2,3,2,3,3],[2,2,2,2,2,2,2,2,2,2,3,2,2,2,2,2,2,2,2,2,2,2,2,3,2,3,3],[0.003,0.001,0.007,0.007,0.008,0.042,0.137,0.072,0.08,0.037,0.02,0.018,0.065,0.004,0,0.001,0.046,0.092,0.003,0.002,0.016,0.009,0.045,0.002,0.001,0.007,0.006],[-0.303,-0.438,0.341,0.242,0.108,0.18,0.054,0.057,0.056,-0.047,-0.062,-0.165,0.065,-0.086,-0.102,0.054,-0.156,0.058,0.092,0.082,0.089,0.083,0.072,0.068,-0.104,-0.146,-0.126],[1,1,1,1,1,0.996,0.954,0.987,0.984,0.997,0.96,0.999,0.989,1,1,1,0.995,0.979,1,1,0.999,1,0.995,0.996,1,0.986,0.988],[0.007,0.004,0.015,0.016,0.019,0.035,0.026,0.049,0.002,0.029,0.021,0.027,0.028,0.005,0,0.002,0.042,0.008,0.003,0.002,0.017,0.019,0.048,0,0.002,0.042,0.01]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>factor<\/th>\n      <th>pwy<\/th>\n      <th>pwy_name<\/th>\n      <th>pwy_gene_num<\/th>\n      <th>num_snp_fdr.0.2<\/th>\n      <th>num_snp_noMHC<\/th>\n      <th>pval_in_fitting<\/th>\n      <th>slope_in_fitting<\/th>\n      <th>rsq_in_fitting<\/th>\n      <th>pval_in_resampling<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[5,6,7,8,9,10]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="checking-the-reverse-causality-5" class="section level4">
<h4>Checking the reverse causality</h4>
<p>**The suffix &quot;_reverse&quot; are the results for traits -&gt; LV (using all candidate SNPs after LD Clumping), the suffix &quot;_rremove&quot; are the results for traits -&gt; LV (using all candidate SNPs after LD Clumping but remove the SNPs have fdr &lt; 0.2)**</p>
<div id="htmlwidget-12eab46753bdd639aac3" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-12eab46753bdd639aac3">{"x":{"filter":"none","data":[["EUR.IBD_pwy162_pc1","EUR.IBD_pwy176_pc1","EUR.IBD_pwy287_pc1","T1D_pwy131_pc5","allergy_pwy259_pc5","baso_pwy180_pc2","hct_pwy187_pc2","hgb_pwy159_pc3","hgb_pwy187_pc2","hgb_pwy255_pc5","hgb_pwy315_pc3","mch_pwy228_pc1","mch_pwy323_pc5","mchc_pwy272_pc2","mcv_pwy120_pc3","mono_pwy161_pc4","mpv_pwy285_pc1","myeloid_wbc_pwy222_pc1","pct_pwy56_pc4","plt_pwy138_pc3","rdw_pwy114_pc3","ret_pwy150_pc2","ret_pwy161_pc1","ret_pwy169_pc5","ukb.allasthma_pwy138_pc1","ukb.allasthma_pwy284_pc3","wbc_pwy228_pc2"],[-0.000263406323048443,0.0101728266043386,-0.00203846494105697,0.00201938137135027,0.0124931522622802,0.00747976339479628,-0.034056866838764,-0.0573572473040187,-0.0504946816747314,-0.0177191006135544,-0.0201506886251153,0.0236453252188724,-0.00249407115374112,0.0775799584776422,0.0209396474246955,-0.0292401628513055,-0.00400570921623356,0.0232223673242089,0.0285831517054511,-0.0198280075261046,0.0126784950780525,0.00940627952224027,0.00768833621099674,0.0142106508454596,0.0185385573184327,0.00876185495564386,0.0372344714063324],[0.997683029195917,0.743038533430541,0.911166343626597,0.906995601450318,0.949542702267793,0.956670404281717,0.942829909386931,0.925593271381622,0.706025506127894,0.906405019138758,0.908683237888582,0.949596000701587,0.988328596474182,0.501935357586207,0.961220793387409,0.955984192995201,0.986171432640857,0.882872161345888,0.932376025621067,0.953919310211842,0.993617116778792,0.975586766357468,0.996007231446476,0.963776833273476,0.880466182423096,0.967028131053708,0.991405052647595],[0.00939197383013157,0.00707435773628351,0.0102890361833099,-0.0147631238358521,0.0107018716370018,0.00936910495627075,-0.0237776863232977,-0.0209508511985028,-0.0324468647106873,-0.0416018081897285,0.0611219223117949,0.0113815202418088,-0.0375466744234743,0.0339729620610516,0.0229570917220112,-0.0156829059634056,-0.00362054434805081,-0.000775881770882947,0.0313608276887738,0.00968467313185961,0.0136035206443648,0.00801750331753759,-0.00920236322384937,0.0252100143359471,0.00574229495751055,0.0264649397551064,0.0304476189321381],[0.918704027017197,0.818977573197014,0.562156430586618,0.376093596779577,0.955557823914034,0.946905287940425,0.959983080759041,0.973684839370201,0.807981836295941,0.780909469921112,0.722092056500051,0.975587925102316,0.820380849253712,0.768670076316271,0.957049744868829,0.976392301947489,0.987457166610324,0.995939882379488,0.923315058733673,0.977421654844949,0.993007276059419,0.979905343270803,0.99536426118209,0.936209913703276,0.962802459024826,0.901920888242552,0.99312900254618]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>mbe_est_reverse<\/th>\n      <th>mbe_pval_reverse<\/th>\n      <th>mbe_est_rremove<\/th>\n      <th>mbe_pval_rremove<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[1,2,3,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] workflowr_1.6.2

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.5        rstudioapi_0.11   whisker_0.3-2     knitr_1.30       
 [5] magrittr_1.5      R6_2.4.1          rlang_0.4.8       highr_0.8        
 [9] stringr_1.4.0     tools_3.6.1       DT_0.15           xfun_0.18        
[13] git2r_0.26.1      crosstalk_1.1.0.1 htmltools_0.5.0   ellipsis_0.3.1   
[17] rprojroot_1.3-2   yaml_2.2.1        digest_0.6.25     tibble_3.0.3     
[21] lifecycle_0.2.0   crayon_1.3.4      later_1.1.0.1     htmlwidgets_1.5.2
[25] vctrs_0.3.4       promises_1.1.1    fs_1.5.0          glue_1.4.2       
[29] evaluate_0.14     rmarkdown_1.13    stringi_1.5.3     compiler_3.6.1   
[33] pillar_1.4.6      backports_1.1.10  jsonlite_1.7.1    httpuv_1.5.1     
[37] pkgconfig_2.0.3  </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
