<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="XSun" />

<meta name="date" content="2021-06-28" />

<title>KEGG PCs summary for 6 cell types from CEDAR dataset (pathways have &gt; 40 genes only)</title>

<script src="site_libs/jquery-1.12.4/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.2/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.15/datatables.js"></script>
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.1.0.1/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">factor_analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/xsun1229/factor_analysis">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">KEGG PCs summary for 6 cell types from CEDAR dataset (pathways have &gt; 40 genes only)</h1>
<h4 class="author">XSun</h4>
<h4 class="date">2021-06-28</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-06-29
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>factor_analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div class="panel-group" id="workflowr-checks">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200623code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200623)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200623code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200623)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it difficult for you and others to run your code on a different machine. Change the absolute path(s) below to the suggested relative path(s) to make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/factor_analysis/output/pwy_name_cd14_cedar_qc_kegg_40genes.rdata
</td>
<td style="text-align:left;">
output/pwy_name_cd14_cedar_qc_kegg_40genes.rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/factor_analysis/output/mbe_cd14_cedar_qc_kegg_40genes.rdata
</td>
<td style="text-align:left;">
output/mbe_cd14_cedar_qc_kegg_40genes.rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/factor_analysis/output/pwy_name_bcell_cedar_qc_kegg_40genes.rdata
</td>
<td style="text-align:left;">
output/pwy_name_bcell_cedar_qc_kegg_40genes.rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/factor_analysis/output/mbe_bcell_cedar_qc_kegg_40genes.rdata
</td>
<td style="text-align:left;">
output/mbe_bcell_cedar_qc_kegg_40genes.rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/factor_analysis/output/pwy_name_tcell_cedar_qc_kegg_40genes.rdata
</td>
<td style="text-align:left;">
output/pwy_name_tcell_cedar_qc_kegg_40genes.rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/factor_analysis/output/mbe_tcell_cedar_qc_kegg_40genes.rdata
</td>
<td style="text-align:left;">
output/mbe_tcell_cedar_qc_kegg_40genes.rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/factor_analysis/output/pwy_name_cd15_cedar_qc_kegg_40genes.rdata
</td>
<td style="text-align:left;">
output/pwy_name_cd15_cedar_qc_kegg_40genes.rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/factor_analysis/output/mbe_cd15_cedar_qc_kegg_40genes.rdata
</td>
<td style="text-align:left;">
output/mbe_cd15_cedar_qc_kegg_40genes.rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/factor_analysis/output/pwy_name_thy_cedar_qc_kegg_40genes.rdata
</td>
<td style="text-align:left;">
output/pwy_name_thy_cedar_qc_kegg_40genes.rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/factor_analysis/output/mbe_thy_cedar_qc_kegg_40genes.rdata
</td>
<td style="text-align:left;">
output/mbe_thy_cedar_qc_kegg_40genes.rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/factor_analysis/output/pwy_name_plt_cedar_qc_kegg_40genes.rdata
</td>
<td style="text-align:left;">
output/pwy_name_plt_cedar_qc_kegg_40genes.rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/xsun/website/factor_analysis/output/mbe_plt_cedar_qc_kegg_40genes.rdata
</td>
<td style="text-align:left;">
output/mbe_plt_cedar_qc_kegg_40genes.rdata
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomxsun1229factoranalysistree2f686091f178d102385b6ad949593d92efbbf0d9targetblank2f68609a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/xsun1229/factor_analysis/tree/2f686091f178d102385b6ad949593d92efbbf0d9" target="_blank">2f68609</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomxsun1229factoranalysistree2f686091f178d102385b6ad949593d92efbbf0d9targetblank2f68609a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/xsun1229/factor_analysis/tree/2f686091f178d102385b6ad949593d92efbbf0d9" target="_blank">2f68609</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    analysis/.Rhistory
    Ignored:    output/.Rhistory

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/cedar_kegg_26_qc_d1k_40genes.Rmd</code>) and HTML (<code>docs/cedar_kegg_26_qc_d1k_40genes.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/2f686091f178d102385b6ad949593d92efbbf0d9/analysis/cedar_kegg_26_qc_d1k_40genes.Rmd" target="_blank">2f68609</a>
</td>
<td>
XSun
</td>
<td>
2021-06-29
</td>
<td>
wflow_publish(“analysis/cedar_kegg_26_qc_d1k_40genes.Rmd”)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/7a68b92966c5866ed64b99ef6b0eb375d6f739d0/analysis/cedar_kegg_26_qc_d1k_40genes.Rmd" target="_blank">7a68b92</a>
</td>
<td>
XSun
</td>
<td>
2021-06-29
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/1d6a6cacc48b0de93269fb04e051164ce6071414/docs/cedar_kegg_26_qc_d1k_40genes.html" target="_blank">1d6a6ca</a>
</td>
<td>
XSun
</td>
<td>
2021-06-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/4128057451cebfc02f26703a9a54a6e53f3e2158/docs/cedar_kegg_26_qc_d1k_40genes.html" target="_blank">4128057</a>
</td>
<td>
XSun
</td>
<td>
2021-06-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/a14f095b55221ed3c96e4ac6f1ab376d529d979e/analysis/cedar_kegg_26_qc_d1k_40genes.Rmd" target="_blank">a14f095</a>
</td>
<td>
XSun
</td>
<td>
2021-06-28
</td>
<td>
wflow_publish(c(“analysis/thesis.Rmd”, “analysis/cedar_kegg_26_qc_d1k_40genes.Rmd”))
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/c87de2ac4abf35e2aa3936181b1547f1d113acfa/analysis/cedar_kegg_26_qc_d1k_40genes.Rmd" target="_blank">c87de2a</a>
</td>
<td>
XSun
</td>
<td>
2021-06-28
</td>
<td>
update
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2><strong>Introduction</strong></h2>
<p>In this part, we just used the pathways have &gt; 40 genes.</p>
</div>
<div id="summary" class="section level2">
<h2><strong>Summary</strong></h2>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/CEDAR_cell_sum_40genes.png" width="80%">
</figure>
</div>
<div id="detailed-results-for-6-cell-types" class="section level2">
<h2><strong>Detailed results for 6 cell types</strong></h2>
<p>We just listed the pairs show sign consistancy (at p-value &lt; 0.05 in resampling) here</p>
<p>We did some literature searching for some disease related pairs (p-value &lt; 0.01 in resampling) in monocyte and B cell.</p>
<div id="monocyte-cd14" class="section level3">
<h3>monocyte CD14</h3>
<div id="histogram-to-show-the-number-of-snps-that-each-pair-contains" class="section level4">
<h4>Histogram to show the number of SNPs that each pair contains</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/snp_count_cd14_cedar_qc_kegg_40genes.jpeg" width="80%">
</figure>
</div>
<div id="effectsize-fitting-summary" class="section level4">
<h4>effectsize fitting summary</h4>
<div id="htmlwidget-c3f4891eb8e9a8c25198" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-c3f4891eb8e9a8c25198">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40"],["allergy_pwy112_pc2","allergy_pwy177_pc3","allergy_pwy275_pc2","allergy_pwy279_pc4","allergy_pwy290_pc3","baso_pwy18_pc3","EUR.CD_pwy112_pc2","EUR.IBD_pwy121_pc1","EUR.IBD_pwy223_pc1","EUR.IBD_pwy275_pc1","EUR.UC_pwy125_pc1","EUR.UC_pwy169_pc5","EUR.UC_pwy186_pc4","EUR.UC_pwy202_pc1","EUR.UC_pwy275_pc2","EUR.UC_pwy282_pc1","EUR.UC_pwy294_pc1","EUR.UC_pwy56_pc2","EUR.UC_pwy91_pc5","EUR.UC_pwy97_pc1","EUR.UC_pwy97_pc2","hgb_pwy135_pc2","lymph_pwy140_pc1","lymph_pwy237_pc2","lymph_pwy289_pc4","lymph_pwy291_pc4","lymph_pwy294_pc1","lymph_pwy299_pc1","lymph_pwy86_pc1","mchc_pwy186_pc5","mchc_pwy236_pc2","mchc_pwy247_pc4","mchc_pwy331_pc4","mcv_pwy184_pc3","mono_pwy54_pc5","pct_pwy150_pc2","rbc_pwy140_pc1","rbc_pwy228_pc1","rdw_pwy118_pc5","ukb.allasthma_pwy155_pc1"],["04012","04621","05140","05145","05167","00230","04012","04064","04921","05140","04071","04530","04659","04725","05140","05152","05200","00564","01524","03018","03018","04137","04145","04935","05166","05168","05200","05206","00983","04659","04934","04972","05416","04657","00562","04217","04145","04926","04060","04310"],["ErbB signaling pathway","NOD-like receptor signaling pathway","Leishmaniasis","Toxoplasmosis","Kaposi sarcoma-associated herpesvirus infection","Purine metabolism","ErbB signaling pathway","NF-kappa B signaling pathway","Oxytocin signaling pathway","Leishmaniasis","Sphingolipid signaling pathway","Tight junction","Th17 cell differentiation","Cholinergic synapse","Leishmaniasis","Tuberculosis","Pathways in cancer","Glycerophospholipid metabolism","Platinum drug resistance","RNA degradation","RNA degradation","Mitophagy - animal","Phagosome","Growth hormone synthesis, secretion and action","Human T-cell leukemia virus 1 infection","Herpes simplex virus 1 infection","Pathways in cancer","MicroRNAs in cancer","Drug metabolism - other enzymes","Th17 cell differentiation","Cushing syndrome","Pancreatic secretion","Viral myocarditis","IL-17 signaling pathway","Inositol phosphate metabolism","Necroptosis","Phagosome","Relaxin signaling pathway","Cytokine-cytokine receptor interaction","Wnt signaling pathway"],["59","127","58","76","132","65","59","71","86","58","84","91","77","58","58","129","290","52","50","65","65","56","113","71","157","273","290","112","41","77","76","42","43","53","52","98","113","70","106","83"],[2,2,2,2,2,2,2,2,2,2,2,3,2,2,2,3,2,2,2,2,2,2,3,2,2,3,2,2,2,2,2,2,2,2,2,2,2,2,2,2],[2,2,2,2,2,2,1,2,2,1,2,3,2,2,2,3,2,2,2,2,2,2,3,2,2,3,2,2,2,2,2,2,2,2,2,2,2,2,2,2],[0.01,0.012,0.009,0.004,0.021,0.003,0.025,0.015,0.021,0.012,0.114,0.007,0.01,0.027,0.013,0.009,0,0.105,0.008,0.011,0.064,0.002,0.003,0.015,0.019,0.004,0.001,0.007,0.036,0.023,0,0.001,0.012,0.015,0.001,0.006,0.005,0.009,0.006,0.005],[-0.144,0.132,-0.219,-0.199,0.166,0.062,-0.522,-0.508,0.622,-0.584,0.556,-0.334,0.268,-0.499,0.568,-0.53,0.499,0.486,-0.352,0.704,-0.465,0.067,0.075,0.17,0.027,-0.039,-0.06,0.14,0.128,0.052,-0.113,-0.094,-0.075,0.078,-0.069,0.059,0.061,-0.081,-0.038,-0.209],[1,1,1,1,0.999,1,0.998,0.999,0.999,1,0.968,0.986,1,0.998,1,0.982,1,0.973,1,1,0.99,1,0.993,0.999,0.999,0.993,1,1,0.997,0.999,1,1,1,0.999,1,1,1,1,1,1],[0.015,0.028,0.02,0.013,0.032,0.009,0.049,0.002,0.049,0.015,0.037,0.017,0.015,0.017,0.047,0.043,0,0.037,0.002,0.037,0.01,0,0.014,0.049,0.034,0.014,0.004,0.017,0.024,0.026,0.001,0.003,0.02,0.017,0.004,0.018,0.012,0.021,0.009,0.008]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>factor<\/th>\n      <th>pwy<\/th>\n      <th>pwy_name<\/th>\n      <th>pwy_gene_num<\/th>\n      <th>num_snp_fdr.0.2<\/th>\n      <th>num_snp_noMHC<\/th>\n      <th>pval_in_fitting<\/th>\n      <th>slope_in_fitting<\/th>\n      <th>rsq_in_fitting<\/th>\n      <th>pval_in_resampling<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[5,6,7,8,9,10]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
<p>EUR.IBD_pwy121_pc1: <a href="https://www.frontiersin.org/articles/10.3389/fped.2018.00317/full">Dysregulation in the NF-κβ pathway and factors that regulate it lead to a state of uncontrolled inflammation and altered immunity, as typically observed in IBD</a></p>
<p>ukb.allasthma_pwy155_pc1 <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4686695/">Taken together, these findings demonstrate that the Wnt/β-catenin signaling pathway is highly expressed and regulates the development of airway remodeling in chronic asthma</a></p>
<p>EUR.UC_pwy97_pc2 <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6077315/">Numerous miRNAs were found to be differentially expressed in IBD and were hypothesized to act as critical mediators in the IBD pathogenesis by regulating proteins in vital regulatory pathways such as in the expression of inflammatory cytokines</a></p>
</div>
<div id="checking-the-reverse-causality" class="section level4">
<h4>Checking the reverse causality</h4>
<p>**The suffix &quot;_reverse&quot; are the results for traits -&gt; LV (using all candidate SNPs after LD Clumping), the suffix &quot;_rremove&quot; are the results for traits -&gt; LV (using all candidate SNPs after LD Clumping but remove the SNPs have fdr &lt; 0.2)**</p>
<div id="htmlwidget-cfefb73e51b4928e765c" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-cfefb73e51b4928e765c">{"x":{"filter":"none","data":[["allergy_pwy112_pc2","allergy_pwy177_pc3","allergy_pwy275_pc2","allergy_pwy279_pc4","allergy_pwy290_pc3","baso_pwy18_pc3","EUR.CD_pwy112_pc2","EUR.IBD_pwy121_pc1","EUR.IBD_pwy223_pc1","EUR.IBD_pwy275_pc1","EUR.UC_pwy125_pc1","EUR.UC_pwy169_pc5","EUR.UC_pwy186_pc4","EUR.UC_pwy202_pc1","EUR.UC_pwy275_pc2","EUR.UC_pwy282_pc1","EUR.UC_pwy294_pc1","EUR.UC_pwy56_pc2","EUR.UC_pwy91_pc5","EUR.UC_pwy97_pc1","EUR.UC_pwy97_pc2","hgb_pwy135_pc2","lymph_pwy140_pc1","lymph_pwy237_pc2","lymph_pwy289_pc4","lymph_pwy291_pc4","lymph_pwy294_pc1","lymph_pwy299_pc1","lymph_pwy86_pc1","mchc_pwy186_pc5","mchc_pwy236_pc2","mchc_pwy247_pc4","mchc_pwy331_pc4","mcv_pwy184_pc3","mono_pwy54_pc5","pct_pwy150_pc2","rbc_pwy140_pc1","rbc_pwy228_pc1","rdw_pwy118_pc5","ukb.allasthma_pwy155_pc1"],[0.0144323539524986,0.0277494413756845,-0.0616720920033842,0.00884398282377497,0.0322979677796127,-0.051320511777174,-0.00124403320550215,0.00533661110078976,0.00564985066336865,0.00263160123104234,-0.00156831612590347,0.0281500914873432,0.0948276686799518,-0.00197373686806301,-0.0351362136386202,0.0143873193913082,0.0103042682493717,-0.00972304372677435,0.0140294504813081,0.00598505009857475,0.00186815517588457,0.00188249558588005,-0.0282933778383487,0.0161042219909859,-0.0598306690854002,0.0656188698263023,0.0759880097757382,0.0179938161490085,0.00624599240076051,0.0542931916076306,-0.0242892148743685,-0.0148597662249017,0.191768519741384,-0.0105952647688916,0.0267694321530705,0.0192462214079523,-0.00975072561675816,0.0221104406998318,-0.0424357647811959,0.0486143720554653],[0.840017448208292,0.966693994384009,0.317403611687059,0.888138676049519,0.93931376128603,0.510033390933468,0.885176451312727,0.580657816248324,0.759105717644958,0.990231917951687,0.972850225759536,0.502871032096996,0.206810517911895,0.883207014820539,0.423105137528257,0.408053314605268,0.843093818998248,0.320160586353723,0.454304689074041,0.744011637571855,0.949597224304896,0.98840812920517,0.903093213469728,0.96652137642342,0.941276891666686,0.918475912409738,0.497355300060461,0.889694411949346,0.951236233158505,0.743320838596075,0.876196362079283,0.906272991251014,0.0975609577245855,0.955456884052417,0.901285207733198,0.99044701318731,0.998095004961327,0.934504039179755,0.917205777244404,0.868481190468412],[0.0312655164748143,0.0171526830779403,-0.0618033940347065,-0.00229154936106823,0.0325293010442085,-0.0110405765407968,0.0017479280966699,0.00569103616888689,0.00482677652977626,0.0129500605887491,-0.0022433824932786,0.0223260253152184,0.0999126569345816,-0.00656351033727964,-0.0352619778525467,0.0123674908854108,0.0187261111985824,-0.00934841000123643,0.0130178109695107,0.0107292184237207,0.00263279992153564,-0.0261082631813725,-0.0402932136364491,0.0165628629001802,-0.023012422211643,0.061251806974397,0.0584743074480301,0.0169276452759419,-0.0110416117923453,0.0548865201569662,-0.00857338922277862,-0.0165796106968656,0.242841246337147,0.0149196129763283,0.0284209169116352,0.0634894604640817,0.0221465792276554,-0.00495714811388659,-0.0161497782892575,0.0507971594949357],[0.654623171114779,0.97920168695866,0.317359198764965,0.97107915210356,0.939387623559015,0.885138942369117,0.834008247243666,0.55500837475597,0.795276043788639,0.951516193164646,0.960986629468807,0.586966333134765,0.18353452644246,0.627844399139714,0.403766031402772,0.469734926187988,0.723245922922797,0.320231886365278,0.467486855290819,0.558284889309477,0.926463765973124,0.837380655620755,0.862282379598144,0.965121578369874,0.977785505406045,0.924221794288713,0.596404267393679,0.895817596179634,0.912463776966659,0.73566481297653,0.956521792032192,0.892708042483305,0.034394437420184,0.937546835636733,0.893748267824327,0.968834603784498,0.995618897797309,0.985247407245257,0.968003045931293,0.859694497171213]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>mbe_est_reverse<\/th>\n      <th>mbe_pval_reverse<\/th>\n      <th>mbe_est_rremove<\/th>\n      <th>mbe_pval_rremove<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[1,2,3,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div id="b-cell" class="section level3">
<h3>B cell</h3>
<div id="histogram-to-show-the-number-of-snps-that-each-pair-contains-1" class="section level4">
<h4>Histogram to show the number of SNPs that each pair contains</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/snp_count_bcell_cedar_qc_kegg_40genes.jpeg" width="80%">
</figure>
</div>
<div id="effectsize-fitting-summary-1" class="section level4">
<h4>effectsize fitting summary</h4>
<div id="htmlwidget-5ba5287ce9a3b08bdb5d" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-5ba5287ce9a3b08bdb5d">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45"],["EUR.CD_pwy106_pc3","EUR.CD_pwy180_pc1","EUR.CD_pwy298_pc5","EUR.UC_pwy163_pc5","T1D_pwy142_pc1","T1D_pwy188_pc5","eo_pwy283_pc1","gran_pwy184_pc3","hct_pwy235_pc1","hgb_pwy219_pc5","hgb_pwy235_pc1","hgb_pwy259_pc5","lymph_pwy120_pc5","lymph_pwy269_pc1","mch_pwy234_pc5","mch_pwy237_pc1","mchc_pwy144_pc5","mchc_pwy259_pc1","mchc_pwy274_pc2","mchc_pwy310_pc4","mcv_pwy237_pc1","mono_pwy128_pc2","mpv_pwy118_pc1","mpv_pwy119_pc1","mpv_pwy177_pc4","mpv_pwy181_pc4","mpv_pwy187_pc3","mpv_pwy270_pc5","mpv_pwy282_pc4","mpv_pwy282_pc5","mpv_pwy283_pc4","mpv_pwy293_pc5","neut_pwy235_pc1","pct_pwy154_pc1","pct_pwy182_pc4","pct_pwy312_pc3","pdw_pwy221_pc2","plt_pwy304_pc1","rbc_pwy318_pc5","rdw_pwy285_pc2","ret_pwy112_pc4","ret_pwy219_pc3","ret_pwy270_pc4","ukb.allasthma_pwy116_pc5","ukb.allasthma_pwy269_pc1"],["03420","04625","05205","04390","04150","04662","05160","04657","04933","04917","04933","05017","04062","05130","04932","04935","04152","05017","05135","05220","04935","04110","04060","04061","04621","04630","04660","05131","05152","05152","05160","05170","04933","04270","04640","05222","04919","05214","05231","05162","04012","04917","05131","04022","05130"],["Nucleotide excision repair","C-type lectin receptor signaling pathway","Proteoglycans in cancer","Hippo signaling pathway","mTOR signaling pathway","B cell receptor signaling pathway","Hepatitis C","IL-17 signaling pathway","AGE-RAGE signaling pathway in diabetic complications","Prolactin signaling pathway","AGE-RAGE signaling pathway in diabetic complications","Spinocerebellar ataxia","Chemokine signaling pathway","Pathogenic Escherichia coli infection","Non-alcoholic fatty liver disease","Growth hormone synthesis, secretion and action","AMPK signaling pathway","Spinocerebellar ataxia","Yersinia infection","Chronic myeloid leukemia","Growth hormone synthesis, secretion and action","Cell cycle","Cytokine-cytokine receptor interaction","Viral protein interaction with cytokine and cytokine receptor","NOD-like receptor signaling pathway","JAK-STAT signaling pathway","T cell receptor signaling pathway","Shigellosis","Tuberculosis","Tuberculosis","Hepatitis C","Human immunodeficiency virus 1 infection","AGE-RAGE signaling pathway in diabetic complications","Vascular smooth muscle contraction","Hematopoietic cell lineage","Small cell lung cancer","Thyroid hormone signaling pathway","Glioma","Choline metabolism in cancer","Measles","ErbB signaling pathway","Prolactin signaling pathway","Shigellosis","cGMP-PKG signaling pathway","Pathogenic Escherichia coli infection"],["41","73","105","71","90","70","104","46","51","42","51","52","109","123","108","67","83","52","89","62","67","92","101","42","116","77","76","165","122","122","104","138","51","57","62","60","79","50","52","96","55","42","165","81","123"],[2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,3,2,2,2,2,3,3,2,2,2,2,2,2,2,2,2,3,2,2,3,2,2,2,2,2],[2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,3,2,2,2,2,3,3,2,2,2,2,2,2,2,2,2,3,2,2,3,2,2,2,2,2],[0.087,0.005,0.033,0.024,0.012,0.02,0.006,0.059,0.003,0.015,0.005,0.01,0.005,0.009,0.019,0.026,0.003,0.002,0.038,0.012,0.001,0.013,0.01,0.005,0.01,0.01,0,0.004,0.021,0.016,0.03,0.018,0.005,0.021,0.012,0.04,0.006,0.011,0.012,0.006,0.019,0.02,0.005,0.01,0.002],[-0.633,-0.493,0.328,-0.346,-0.737,0.251,-0.134,-0.08,0.155,0.066,0.141,0.076,0.033,0.108,-0.066,-0.133,-0.045,0.2,-0.094,-0.061,-0.113,0.084,-0.053,-0.052,-0.043,-0.069,0.044,-0.036,-0.045,-0.064,-0.093,-0.045,0.11,0.159,-0.101,0.089,-0.104,-0.152,-0.053,0.075,0.091,-0.072,-0.057,0.089,-0.191],[0.982,1,0.997,0.999,1,0.999,1,0.992,1,0.999,1,1,1,1,0.999,0.998,1,1,0.997,1,0.998,1,1,1,1,0.98,0.999,1,0.999,0.999,0.998,0.999,1,0.999,1,0.996,0.988,1,1,0.989,0.999,0.999,1,1,1],[0.015,0.002,0.032,0.01,0.004,0.011,0.014,0.044,0.008,0.008,0.012,0.007,0.018,0.025,0.033,0.045,0.005,0.002,0.037,0.021,0.008,0.025,0.011,0.007,0.014,0.039,0.002,0.006,0.036,0.015,0.042,0.032,0.01,0.021,0.025,0.048,0.02,0.016,0.032,0.026,0.021,0.036,0.017,0.026,0.012]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>factor<\/th>\n      <th>pwy<\/th>\n      <th>pwy_name<\/th>\n      <th>pwy_gene_num<\/th>\n      <th>num_snp_fdr.0.2<\/th>\n      <th>num_snp_noMHC<\/th>\n      <th>pval_in_fitting<\/th>\n      <th>slope_in_fitting<\/th>\n      <th>rsq_in_fitting<\/th>\n      <th>pval_in_resampling<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[5,6,7,8,9,10]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
<p>EUR.CD_pwy180_pc1: <a href="https://academic.oup.com/gastro/article/7/5/312/5530700">Therefore, CLRs, as innate immune receptors, may play vital roles in IBD progression.</a></p>
<p>EUR.UC_pwy163_pc5: <a href="https://www.nature.com/articles/s41419-021-03395-3">The role of the Hippo pathway in the pathogenesis of inflammatory bowel disease</a></p>
</div>
<div id="checking-the-reverse-causality-1" class="section level4">
<h4>Checking the reverse causality</h4>
<p>**The suffix &quot;_reverse&quot; are the results for traits -&gt; LV (using all candidate SNPs after LD Clumping), the suffix &quot;_rremove&quot; are the results for traits -&gt; LV (using all candidate SNPs after LD Clumping but remove the SNPs have fdr &lt; 0.2)**</p>
<div id="htmlwidget-0b516daec537d8efb4b6" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-0b516daec537d8efb4b6">{"x":{"filter":"none","data":[["EUR.CD_pwy106_pc3","EUR.CD_pwy180_pc1","EUR.CD_pwy298_pc5","EUR.UC_pwy163_pc5","T1D_pwy142_pc1","T1D_pwy188_pc5","eo_pwy283_pc1","gran_pwy184_pc3","hct_pwy235_pc1","hgb_pwy219_pc5","hgb_pwy235_pc1","hgb_pwy259_pc5","lymph_pwy120_pc5","lymph_pwy269_pc1","mch_pwy234_pc5","mch_pwy237_pc1","mchc_pwy144_pc5","mchc_pwy259_pc1","mchc_pwy274_pc2","mchc_pwy310_pc4","mcv_pwy237_pc1","mono_pwy128_pc2","mpv_pwy118_pc1","mpv_pwy119_pc1","mpv_pwy177_pc4","mpv_pwy181_pc4","mpv_pwy187_pc3","mpv_pwy270_pc5","mpv_pwy282_pc4","mpv_pwy282_pc5","mpv_pwy283_pc4","mpv_pwy293_pc5","neut_pwy235_pc1","pct_pwy154_pc1","pct_pwy182_pc4","pct_pwy312_pc3","pdw_pwy221_pc2","plt_pwy304_pc1","rbc_pwy318_pc5","rdw_pwy285_pc2","ret_pwy112_pc4","ret_pwy219_pc3","ret_pwy270_pc4","ukb.allasthma_pwy116_pc5","ukb.allasthma_pwy269_pc1"],[-0.00010578972024966,0.00154184241434363,0.00555968698461351,-0.0193737498985396,0.0495169708025855,-0.0190452280749485,0.00559891980833704,-0.0125936895301137,-0.0356236900950213,0.0367673901834991,-0.0414761246747908,-0.0734760270542498,-0.0344861223980253,0.0391486707188218,0.0252695197726638,0.020707954668886,-0.0498911348944198,-0.0160971422610015,0.0386754598446153,-0.0726914269849814,0.0201889983347971,0.00629594641236153,0.0443173540953303,-0.0455662669260697,0.0173085966545941,-0.0529358691252142,-0.00129070513861151,-0.02417042358298,0.0204566431216087,-0.00413337056534147,0.0224409725124115,-0.018809121102219,0.0140687124732803,0.00255652746306012,-0.00379108135201278,0.0223949727595283,-0.00182402093839151,-0.00130410317361473,-0.00517803727206356,-0.0198049465673105,-0.00746271120204511,0.0304103781424025,-0.00870216412305069,-0.124551696373002,0.016961587471485],[0.9980495985347,0.977941656661029,0.827010804185721,0.772942840574512,0.280084291326897,0.403667735203923,0.939301791458248,0.99432360045871,0.948154291920715,0.731222504198314,0.841081174074251,0.68996934199131,0.899143743455802,0.697391093913375,0.970904407106234,0.89392000219404,0.981895624826112,0.915022204059879,0.876099490169512,0.89451665517099,0.883858939403186,0.981470020180191,0.978521283775323,0.933003566060639,0.974648646424945,0.981753391017825,0.997192253387008,0.974494001935663,0.994426930830483,0.990502857491089,0.917140133044344,0.956527548837671,0.959422045695973,0.97173229128895,0.994338724336239,0.911219882953721,0.994280821059758,0.999396157977125,0.993027586913709,0.97205494771893,0.977251399025354,0.882782789048096,0.960528338973656,0.151759478963159,0.620764983969024],[0.00285092142579479,-0.00101169068819518,0.00721527648907294,-0.023388863585597,0.0512297891749387,-0.0256259806703332,-0.00767024305259056,-0.00564685556259903,-0.0214067813885759,-0.00392203529210633,-0.0542247802502001,-0.0270394653726331,0.0476438095795366,0.0310133885639399,0.021591613645267,0.0211492401594793,0.0134446160871882,-0.0279399465109584,0.0327979956447031,-0.0560832454275335,0.0141266186592688,0.0138399637781053,0.0392668742866178,-0.0497035934368775,0.015442594477264,0.054860946238108,-0.0107140690329892,-0.033431324703912,-0.0173534114577123,-0.00229770200027701,0.0216571567079207,-0.029344323231939,-0.011581797366766,0.00738243033113406,0.0194245679797138,0.00682721173375533,0.0108011804720203,-0.00450902061202108,0.0219147649486331,0.0321988259983925,0.0309969216750208,0.041606486953782,0.0420590931681453,-0.117901979399701,0.0130753561722958],[0.950201385113042,0.98536536935253,0.776233184036401,0.730921017728491,0.265127686108151,0.246882023849031,0.917950576231928,0.997501657729774,0.96860596876564,0.970081588265407,0.800455962672827,0.882511002744162,0.857531038734272,0.762288621275455,0.975290921711737,0.893807978046024,0.9950582277047,0.848573288043578,0.892503180291269,0.919533684620986,0.918466561294539,0.960010940292216,0.981005979168318,0.927771110865533,0.978393142415363,0.981514089820153,0.976845992289198,0.96443641961811,0.995258184978608,0.994677879038419,0.920181490140879,0.932739350903081,0.966547641025697,0.919823996232931,0.97098316401477,0.972928269686942,0.966414261202896,0.997847578562977,0.970494877391443,0.955242401049921,0.905336617768737,0.843865209745372,0.809578690561685,0.172343326824645,0.711653637649384]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>mbe_est_reverse<\/th>\n      <th>mbe_pval_reverse<\/th>\n      <th>mbe_est_rremove<\/th>\n      <th>mbe_pval_rremove<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[1,2,3,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div id="t-cellcd4" class="section level3">
<h3>T cell(CD4)</h3>
<div id="histogram-to-show-the-number-of-snps-that-each-pair-contains-2" class="section level4">
<h4>Histogram to show the number of SNPs that each pair contains</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/snp_count_tcell_cedar_qc_kegg_40genes.jpeg" width="80%">
</figure>
</div>
<div id="effectsize-fitting-summary-2" class="section level4">
<h4>effectsize fitting summary</h4>
<div id="htmlwidget-d0109ea5b7e7d667f148" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-d0109ea5b7e7d667f148">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42"],["EUR.IBD_pwy191_pc4","EUR.IBD_pwy266_pc2","T1D_pwy96_pc4","allergy_pwy300_pc5","allergy_pwy302_pc5","baso_pwy138_pc1","baso_pwy189_pc4","eo_pwy113_pc1","eo_pwy120_pc1","eo_pwy143_pc1","eo_pwy153_pc1","eo_pwy178_pc4","eo_pwy1_pc1","eo_pwy292_pc1","eo_pwy304_pc1","eo_pwy310_pc1","hgb_pwy201_pc2","hgb_pwy217_pc3","hgb_pwy303_pc1","lymph_pwy153_pc4","lymph_pwy255_pc1","mch_pwy285_pc2","mch_pwy88_pc4","mchc_pwy89_pc5","mcv_pwy270_pc5","mcv_pwy286_pc5","mcv_pwy288_pc5","mcv_pwy296_pc5","mcv_pwy88_pc4","neut_pwy18_pc4","pct_pwy161_pc3","pct_pwy192_pc1","pct_pwy197_pc5","pct_pwy202_pc5","pct_pwy221_pc5","pct_pwy300_pc3","rdw_pwy299_pc2","ret_pwy111_pc5","ukb.allasthma_pwy18_pc1","ukb.allasthma_pwy18_pc5","ukb.allasthma_pwy192_pc5","wbc_pwy319_pc2"],["04668","05100","03015","05210","05212","04142","04664","04014","04062","04151","04261","04622","00010","05169","05214","05220","04724","04915","05213","04261","05010","05162","01521","01522","05131","05163","05165","05203","01521","00230","04371","04670","04720","04725","04919","05210","05206","04010","00230","00230","04670","05235"],["TNF signaling pathway","Bacterial invasion of epithelial cells","mRNA surveillance pathway","Colorectal cancer","Pancreatic cancer","Lysosome","Fc epsilon RI signaling pathway","Ras signaling pathway","Chemokine signaling pathway","PI3K-Akt signaling pathway","Adrenergic signaling in cardiomyocytes","RIG-I-like receptor signaling pathway","Glycolysis / Gluconeogenesis","Epstein-Barr virus infection","Glioma","Chronic myeloid leukemia","Glutamatergic synapse","Estrogen signaling pathway","Endometrial cancer","Adrenergic signaling in cardiomyocytes","Alzheimer disease","Measles","EGFR tyrosine kinase inhibitor resistance","Endocrine resistance","Shigellosis","Human cytomegalovirus infection","Human papillomavirus infection","Viral carcinogenesis","EGFR tyrosine kinase inhibitor resistance","Purine metabolism","Apelin signaling pathway","Leukocyte transendothelial migration","Long-term potentiation","Cholinergic synapse","Thyroid hormone signaling pathway","Colorectal cancer","MicroRNAs in cancer","MAPK signaling pathway","Purine metabolism","Purine metabolism","Leukocyte transendothelial migration","PD-L1 expression and PD-1 checkpoint pathway in cancer"],["75","48","71","64","56","106","49","120","112","177","71","42","47","159","53","63","42","67","43","71","114","99","49","55","161","147","193","136","49","68","70","67","41","57","83","64","117","167","68","68","67","74"],[2,2,2,2,2,4,2,2,2,2,2,2,2,2,2,2,2,2,3,3,2,2,3,2,2,2,2,2,3,2,2,2,2,2,2,2,2,2,2,2,2,2],[2,2,2,2,2,4,2,2,2,2,2,2,2,2,2,2,2,2,3,3,2,2,3,2,2,2,2,2,3,2,2,2,2,2,2,2,2,2,2,2,2,2],[0.003,0.019,0.006,0.001,0.024,0.004,0.01,0.076,0.089,0.077,0.121,0.037,0.017,0.076,0.09,0.095,0.005,0.012,0.006,0.005,0.013,0.009,0.008,0.018,0.008,0.011,0.015,0.003,0.002,0.008,0.027,0.026,0.001,0.041,0.016,0.006,0.003,0.013,0.028,0.004,0.008,0.002],[0.462,0.332,-0.268,0.084,-0.081,0.054,0.061,-0.057,0.059,0.048,-0.06,0.063,-0.063,0.058,-0.077,-0.065,-0.081,-0.078,0.125,0.087,0.056,-0.097,-0.125,0.055,0.034,0.036,-0.028,0.033,-0.108,0.055,0.059,-0.085,0.06,-0.058,-0.061,0.099,-0.08,0.04,0.138,0.08,0.1,-0.089],[1,0.999,1,1,0.999,0.952,1,0.986,0.981,0.986,0.965,0.997,0.999,0.986,0.98,0.978,1,1,0.989,0.99,1,1,0.984,0.999,1,1,0.999,1,0.996,1,0.998,0.998,1,0.996,0.999,1,1,1,0.998,1,1,1],[0.007,0.018,0.011,0,0.021,0.042,0.034,0.013,0.013,0.013,0.027,0.006,0.004,0.013,0.013,0.013,0.003,0.042,0.03,0.011,0.03,0.015,0.023,0.039,0.017,0.025,0.036,0.01,0.008,0.019,0.037,0.015,0,0.033,0.016,0.007,0.008,0.025,0.03,0.014,0.019,0.002]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>factor<\/th>\n      <th>pwy<\/th>\n      <th>pwy_name<\/th>\n      <th>pwy_gene_num<\/th>\n      <th>num_snp_fdr.0.2<\/th>\n      <th>num_snp_noMHC<\/th>\n      <th>pval_in_fitting<\/th>\n      <th>slope_in_fitting<\/th>\n      <th>rsq_in_fitting<\/th>\n      <th>pval_in_resampling<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[5,6,7,8,9,10]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="checking-the-reverse-causality-2" class="section level4">
<h4>Checking the reverse causality</h4>
<p>**The suffix &quot;_reverse&quot; are the results for traits -&gt; LV (using all candidate SNPs after LD Clumping), the suffix &quot;_rremove&quot; are the results for traits -&gt; LV (using all candidate SNPs after LD Clumping but remove the SNPs have fdr &lt; 0.2)**</p>
<div id="htmlwidget-7d2b77686a7a668a5a8b" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-7d2b77686a7a668a5a8b">{"x":{"filter":"none","data":[["EUR.IBD_pwy191_pc4","EUR.IBD_pwy266_pc2","T1D_pwy96_pc4","allergy_pwy300_pc5","allergy_pwy302_pc5","baso_pwy138_pc1","baso_pwy189_pc4","eo_pwy113_pc1","eo_pwy120_pc1","eo_pwy143_pc1","eo_pwy153_pc1","eo_pwy178_pc4","eo_pwy1_pc1","eo_pwy292_pc1","eo_pwy304_pc1","eo_pwy310_pc1","hgb_pwy201_pc2","hgb_pwy217_pc3","hgb_pwy303_pc1","lymph_pwy153_pc4","lymph_pwy255_pc1","mch_pwy285_pc2","mch_pwy88_pc4","mchc_pwy89_pc5","mcv_pwy270_pc5","mcv_pwy286_pc5","mcv_pwy288_pc5","mcv_pwy296_pc5","mcv_pwy88_pc4","neut_pwy18_pc4","pct_pwy161_pc3","pct_pwy192_pc1","pct_pwy197_pc5","pct_pwy202_pc5","pct_pwy221_pc5","pct_pwy300_pc3","rdw_pwy299_pc2","ret_pwy111_pc5","ukb.allasthma_pwy18_pc1","ukb.allasthma_pwy18_pc5","ukb.allasthma_pwy192_pc5","wbc_pwy319_pc2"],[0.00473446786587006,-0.00886808967094499,0.0125939931981165,0.0269829155386674,-0.0284336069618689,0.0654196782718515,0.00148591133276632,0.0107241465200616,0.0319789822561489,0.00291082590705827,0.0417845171726867,-0.0213900709884385,-0.00601196188090825,-0.0381138419705938,0.00244303409966129,0.000647994044516764,0.0237115629309841,-0.0167342502217593,0.0147191785594583,0.00804669372727318,-0.0028423698544735,-0.0208231118629634,-0.0224253607015452,-0.0902876403309048,0.0329652459438918,-0.0657953863902279,0.0784550734200877,0.0772728073524434,-0.0469341184714267,0.0239653808592273,-0.00450560145556977,0.0334260213005315,-0.0250529093136098,0.0235579193899831,0.112004773793515,0.0237061038983502,0.0175465066516125,0.00147783557234149,-0.0301426056806502,-0.0247122522494543,-0.0102161707158643,0.041088267336967],[0.919579915905607,0.873247080235338,0.356566992920476,0.840659021597896,0.599053603363694,0.659185069099854,0.992387284665009,0.89718453081005,0.970621430797749,0.973538552342767,0.93595963896279,0.952114818617866,0.973215881576651,0.98640309236554,0.990022950136338,0.99770824928631,0.95670739242662,0.96983695013275,0.960038392358675,0.988221710572621,0.996301244977825,0.946543097393807,0.903955639756878,0.574422795685608,0.984700539240999,0.939087562925865,0.982427154646294,0.935710048741326,0.817999912177174,0.965323616541825,0.990626343446792,0.964570902267466,0.99068101456299,0.94388611727908,0.807936127471152,0.958042564811669,0.918756447697975,0.993544214332917,0.717699460919093,0.898440322685432,0.93320428057513,0.737916208646284],[0.00371090992489087,-0.0061975274020023,0.00657689055753607,0.0241934998602833,-0.0251940451968551,0.0965412396947904,0.047364152871781,0.00117055416096434,0.0230326406082568,-0.012473769126732,0.0531578342534775,-0.0218780726847339,-0.0123132270543653,-0.0112629563134838,0.0143910127767999,0.00655476986042736,0.0370174678687594,-0.0184869529319496,0.00662128707787346,0.0109025265022247,0.00453585289648828,-0.0085853075381106,-0.0224081435937951,-0.090871080315658,0.0280699395268229,-0.027579847373012,0.132289693354366,-0.0419226339345542,-0.0465045788200555,-0.00530815811050189,-0.0310289857560448,0.0027038947928073,-0.00477330273132992,0.0314838826132977,0.110670035056824,0.0153565915542302,-0.0154246811527887,0.0114874153673838,-0.0155576817584731,-0.0457058788664302,0.0063823665746181,0.0288445699375117],[0.936924471114687,0.91153688113876,0.616102316500312,0.85808673231023,0.639959667321337,0.510987418712146,0.761891447457102,0.98890644548196,0.978418647418997,0.883668135996625,0.916793945095069,0.95011329250469,0.945146282788153,0.99584633418318,0.941951137262996,0.977757961584497,0.932367427868529,0.966656900405891,0.981223923977092,0.983399163118009,0.994062716050499,0.978291206548569,0.905921395085425,0.583844103530283,0.987226413165055,0.974341252658828,0.970391833314754,0.965584524960512,0.824629842611233,0.992302018968197,0.93560602389366,0.997050298394226,0.998254493743619,0.92473151991119,0.804938965058321,0.972962281875699,0.928361279169773,0.950514173710507,0.850989810673152,0.810009999811063,0.957776416086945,0.813698094016959]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>mbe_est_reverse<\/th>\n      <th>mbe_pval_reverse<\/th>\n      <th>mbe_est_rremove<\/th>\n      <th>mbe_pval_rremove<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[1,2,3,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
<p><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5568528/">Together these data support the hypothesis that TNF plays a beneficial role in enhancing Wnt/β-catenin signaling during ulcer healing in IBD.</a></p>
</div>
</div>
<div id="neutrophils-cd15" class="section level3">
<h3>neutrophils CD15</h3>
<div id="histogram-to-show-the-number-of-snps-that-each-pair-contains-3" class="section level4">
<h4>Histogram to show the number of SNPs that each pair contains</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/snp_count_cd15_cedar_qc_kegg_40genes.jpeg" width="80%">
</figure>
</div>
<div id="effectsize-fitting-summary-3" class="section level4">
<h4>effectsize fitting summary</h4>
<div id="htmlwidget-1e347d3f3a7c82c854df" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-1e347d3f3a7c82c854df">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21"],["allergy_pwy136_pc5","baso_pwy320_pc5","EUR.IBD_pwy144_pc5","EUR.IBD_pwy256_pc5","EUR.IBD_pwy310_pc3","EUR.UC_pwy256_pc5","gran_pwy138_pc2","hgb_pwy121_pc2","mch_pwy298_pc2","mcv_pwy136_pc2","mpv_pwy131_pc2","mpv_pwy268_pc5","mpv_pwy91_pc1","myeloid_wbc_pwy144_pc2","neut_pwy138_pc2","pct_pwy135_pc3","pdw_pwy287_pc3","pdw_pwy288_pc4","plt_pwy151_pc1","rdw_pwy280_pc4","ret_pwy305_pc5"],["04140","05310","04152","05012","05220","05012","04142","04064","05205","04140","04120","05120","01524","04152","04142","04137","05164","05165","04218","05146","05215"],["Autophagy - animal","Asthma","AMPK signaling pathway","Parkinson disease","Chronic myeloid leukemia","Parkinson disease","Lysosome","NF-kappa B signaling pathway","Proteoglycans in cancer","Autophagy - animal","Ubiquitin mediated proteolysis","Epithelial cell signaling in Helicobacter pylori infection","Platinum drug resistance","AMPK signaling pathway","Lysosome","Mitophagy - animal","Influenza A","Human papillomavirus infection","Cellular senescence","Amoebiasis","Prostate cancer"],["100","10","71","79","56","79","94","69","102","100","96","47","50","71","94","43","107","175","92","41","59"],[2,2,2,4,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,3,2],[2,1,2,4,2,2,2,1,2,2,2,2,2,2,2,2,2,2,2,3,2],[0.009,0.015,0.091,0.002,0.002,0.013,0.083,0.011,0.015,0.034,0.016,0.029,0.014,0.011,0.074,0.006,0.008,0.015,0.058,0.009,0.03],[-0.067,0.145,0.326,0.555,0.612,0.684,0.067,-0.131,0.094,-0.102,0.09,-0.048,-0.099,0.092,0.071,0.072,-0.029,0.037,-0.169,0.064,0.073],[1,0.999,0.98,0.969,1,1,0.983,1,0.999,0.997,0.999,0.998,1,1,0.987,1,1,0.999,0.992,0.982,0.998],[0.024,0.025,0.024,0.027,0.014,0.025,0.012,0.011,0.038,0.039,0.028,0.047,0.015,0.034,0.014,0.015,0.022,0.034,0.041,0.025,0.049]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>factor<\/th>\n      <th>pwy<\/th>\n      <th>pwy_name<\/th>\n      <th>pwy_gene_num<\/th>\n      <th>num_snp_fdr.0.2<\/th>\n      <th>num_snp_noMHC<\/th>\n      <th>pval_in_fitting<\/th>\n      <th>slope_in_fitting<\/th>\n      <th>rsq_in_fitting<\/th>\n      <th>pval_in_resampling<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[5,6,7,8,9,10]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="checking-the-reverse-causality-3" class="section level4">
<h4>Checking the reverse causality</h4>
<p>**The suffix &quot;_reverse&quot; are the results for traits -&gt; LV (using all candidate SNPs after LD Clumping), the suffix &quot;_rremove&quot; are the results for traits -&gt; LV (using all candidate SNPs after LD Clumping but remove the SNPs have fdr &lt; 0.2)**</p>
<div id="htmlwidget-32695f625600a1e2a463" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-32695f625600a1e2a463">{"x":{"filter":"none","data":[["allergy_pwy136_pc5","baso_pwy320_pc5","EUR.IBD_pwy144_pc5","EUR.IBD_pwy256_pc5","EUR.IBD_pwy310_pc3","EUR.UC_pwy256_pc5","gran_pwy138_pc2","hgb_pwy121_pc2","mch_pwy298_pc2","mcv_pwy136_pc2","mpv_pwy131_pc2","mpv_pwy268_pc5","mpv_pwy91_pc1","myeloid_wbc_pwy144_pc2","neut_pwy138_pc2","pct_pwy135_pc3","pdw_pwy287_pc3","pdw_pwy288_pc4","plt_pwy151_pc1","rdw_pwy280_pc4","ret_pwy305_pc5"],[-0.0680202526454217,0.0741042894498865,-0.0225340232698343,-0.0371688500371681,-0.030520143302855,0.00408052101886636,-0.0714606105209121,-0.00231579521484626,-0.011536674226285,0.023070169835373,0.0019642156936257,0.00454193962294269,0.00990260959797062,0.00235286917533095,0.0359318524392211,0.0511203578042707,0.0600386619163729,0.0338873266707509,-0.0419620124221964,0.00249266692548034,-0.0357296649890166],[0.750323531157474,0.81349010133747,0.386449230675957,0.0850318660299042,0.512944023762307,0.599024356764774,0.961304009972517,0.997618568770831,0.975199680543909,0.992269100012129,0.993560904875469,0.977802650744188,0.916201522131665,0.993666319836788,0.981625905143597,0.73911507452853,0.966068308217071,0.967999282325702,0.737815185728443,0.997424792795026,0.877962191922337],[-0.0566539944107625,0.0587688756192986,-0.0213030213038783,-0.0342660840870397,-0.043402734872084,0.00731328507588414,-0.0612307759317812,-0.00196606481792472,-0.00963574690858771,0.0165930777784684,0.00138064835832452,-0.0484820813971041,-0.00859181350766747,-0.0340640929809108,-0.00570751389221691,0.0205516067361788,-0.0498960674815123,0.0433372468338575,-0.0458269239374651,-0.00853119291531357,-0.0403349943837767],[0.78891055435179,0.8457517495924,0.423038446895962,0.106245911626787,0.355097485646796,0.343362864353305,0.967450124123225,0.998000029592214,0.979140050795723,0.99449380254197,0.995542661141472,0.763947812840464,0.927306208422097,0.909786247973706,0.997005016718666,0.896647143241268,0.972181089107796,0.960217384775869,0.709531666976792,0.991231095904947,0.860406439722521]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>mbe_est_reverse<\/th>\n      <th>mbe_pval_reverse<\/th>\n      <th>mbe_est_rremove<\/th>\n      <th>mbe_pval_rremove<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[1,2,3,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div id="t-cellcd8" class="section level3">
<h3>T cell(CD8)</h3>
<div id="histogram-to-show-the-number-of-snps-that-each-pair-contains-4" class="section level4">
<h4>Histogram to show the number of SNPs that each pair contains</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/snp_count_thy_cedar_qc_kegg_40genes.jpeg" width="80%">
</figure>
</div>
<div id="effectsize-fitting-summary-4" class="section level4">
<h4>effectsize fitting summary</h4>
<div id="htmlwidget-312b9245b08df84edcee" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-312b9245b08df84edcee">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38"],["EUR.CD_pwy237_pc5","EUR.UC_pwy151_pc3","EUR.UC_pwy228_pc2","EUR.UC_pwy314_pc2","T1D_pwy101_pc2","allergy_pwy259_pc3","eo_pwy1_pc1","hgb_pwy131_pc1","hgb_pwy282_pc3","hgb_pwy283_pc5","hgb_pwy311_pc4","hgb_pwy89_pc3","hgb_pwy95_pc1","mchc_pwy111_pc2","mchc_pwy200_pc1","mchc_pwy268_pc3","mono_pwy140_pc4","mono_pwy162_pc3","mono_pwy171_pc3","mono_pwy218_pc2","mpv_pwy101_pc2","myeloid_wbc_pwy295_pc4","pct_pwy203_pc2","pdw_pwy127_pc4","pdw_pwy162_pc1","rbc_pwy275_pc4","rdw_pwy159_pc5","rdw_pwy200_pc5","wbc_pwy118_pc5","wbc_pwy143_pc4","wbc_pwy155_pc5","wbc_pwy159_pc5","wbc_pwy161_pc4","wbc_pwy211_pc2","wbc_pwy221_pc5","wbc_pwy286_pc4","wbc_pwy295_pc4","wbc_pwy304_pc4"],["04935","04218","04926","05224","03040","05017","00010","04120","05152","05160","05221","01522","03013","04010","04723","05120","04145","04380","04550","04916","03040","05202","04726","04080","04380","05140","04360","04723","04060","04151","04310","04360","04371","04810","04919","05163","05202","05214"],["Growth hormone synthesis, secretion and action","Cellular senescence","Relaxin signaling pathway","Breast cancer","Spliceosome","Spinocerebellar ataxia","Glycolysis / Gluconeogenesis","Ubiquitin mediated proteolysis","Tuberculosis","Hepatitis C","Acute myeloid leukemia","Endocrine resistance","RNA transport","MAPK signaling pathway","Retrograde endocannabinoid signaling","Epithelial cell signaling in Helicobacter pylori infection","Phagosome","Osteoclast differentiation","Signaling pathways regulating pluripotency of stem cells","Melanogenesis","Spliceosome","Transcriptional misregulation in cancer","Serotonergic synapse","Neuroactive ligand-receptor interaction","Osteoclast differentiation","Leishmaniasis","Axon guidance","Retrograde endocannabinoid signaling","Cytokine-cytokine receptor interaction","PI3K-Akt signaling pathway","Wnt signaling pathway","Axon guidance","Apelin signaling pathway","Regulation of actin cytoskeleton","Thyroid hormone signaling pathway","Human cytomegalovirus infection","Transcriptional misregulation in cancer","Glioma"],["71","117","69","73","112","58","44","116","120","109","52","57","137","168","71","50","97","91","63","48","112","97","45","71","91","54","87","71","118","178","82","87","74","125","84","152","97","53"],[2,2,2,3,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,3,2,2,6,3,2,2,2,2,3,2,2,2,2,2,2,2,2],[2,2,1,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,3,2,2,6,3,2,2,2,2,3,2,2,2,2,2,2,2,2],[0.004,0.012,0.003,0,0.017,0.009,0.002,0.004,0.005,0.01,0.005,0.013,0.012,0.023,0.012,0.018,0.001,0.014,0.013,0.005,0.007,0.02,0.004,0.003,0.013,0.03,0.007,0.015,0.013,0.003,0.009,0.011,0.008,0.019,0.011,0.006,0.01,0.016],[0.272,0.488,-0.355,-0.495,0.395,-0.131,-0.106,-0.067,0.051,-0.051,0.046,-0.059,0.067,-0.072,-0.072,0.056,-0.051,-0.063,-0.073,-0.071,-0.107,0.059,0.068,-0.088,0.057,-0.048,-0.049,-0.083,-0.041,-0.034,0.037,0.043,0.038,0.032,0.051,-0.029,0.058,-0.044],[1,1,1,1,0.999,1,1,1,1,1,1,1,1,0.999,1,0.999,1,1,1,1,0.987,0.999,1,0.848,0.974,0.998,1,0.999,1,0.995,1,1,1,0.999,1,1,1,0.999],[0.011,0.044,0.01,0,0.028,0.017,0.007,0.006,0.016,0.04,0.016,0.042,0.032,0.047,0.041,0.041,0,0.04,0.02,0.01,0.024,0.032,0.013,0.01,0.035,0.015,0.017,0.032,0.035,0.014,0.026,0.039,0.033,0.048,0.046,0.008,0.022,0.04]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>factor<\/th>\n      <th>pwy<\/th>\n      <th>pwy_name<\/th>\n      <th>pwy_gene_num<\/th>\n      <th>num_snp_fdr.0.2<\/th>\n      <th>num_snp_noMHC<\/th>\n      <th>pval_in_fitting<\/th>\n      <th>slope_in_fitting<\/th>\n      <th>rsq_in_fitting<\/th>\n      <th>pval_in_resampling<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[5,6,7,8,9,10]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="checking-the-reverse-causality-4" class="section level4">
<h4>Checking the reverse causality</h4>
<p>**The suffix &quot;_reverse&quot; are the results for traits -&gt; LV (using all candidate SNPs after LD Clumping), the suffix &quot;_rremove&quot; are the results for traits -&gt; LV (using all candidate SNPs after LD Clumping but remove the SNPs have fdr &lt; 0.2)**</p>
<div id="htmlwidget-3cd2a26b2bcc583003e2" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-3cd2a26b2bcc583003e2">{"x":{"filter":"none","data":[["EUR.CD_pwy237_pc5","EUR.UC_pwy151_pc3","EUR.UC_pwy228_pc2","EUR.UC_pwy314_pc2","T1D_pwy101_pc2","allergy_pwy259_pc3","eo_pwy1_pc1","hgb_pwy131_pc1","hgb_pwy282_pc3","hgb_pwy283_pc5","hgb_pwy311_pc4","hgb_pwy89_pc3","hgb_pwy95_pc1","mchc_pwy111_pc2","mchc_pwy200_pc1","mchc_pwy268_pc3","mono_pwy140_pc4","mono_pwy162_pc3","mono_pwy171_pc3","mono_pwy218_pc2","mpv_pwy101_pc2","myeloid_wbc_pwy295_pc4","pct_pwy203_pc2","pdw_pwy127_pc4","pdw_pwy162_pc1","rbc_pwy275_pc4","rdw_pwy159_pc5","rdw_pwy200_pc5","wbc_pwy118_pc5","wbc_pwy143_pc4","wbc_pwy155_pc5","wbc_pwy159_pc5","wbc_pwy161_pc4","wbc_pwy211_pc2","wbc_pwy221_pc5","wbc_pwy286_pc4","wbc_pwy295_pc4","wbc_pwy304_pc4"],[0.00633609535632029,0.0236142989569164,0.005009314731657,0.0145456500260961,0.0231486092760966,0.0114831348336057,-0.0391211055552496,-0.0159564562803816,0.0340368166260241,0.0334734254007643,-0.0329804775306535,-0.0581787415155723,-0.0309551509661272,0.0189409709937394,-0.0302491559020588,-0.180565511499168,0.0458540102490943,0.0128923208797076,-0.0375852664420933,-0.00884135333826919,0.0397114321228429,-0.00480567244648711,0.0375080560802452,-0.0108332410961189,0.029567340437211,0.0173771828828286,-0.0916847420116476,-0.0342483915672958,-0.0150521774377665,-0.138562601763944,0.0167444074125633,0.0294254433597523,-0.066483395709021,0.0131069309646747,-0.014113810481259,-0.0501809827627326,0.00623704000249248,-0.0245033358585651],[0.816863747510735,0.634633079859871,0.839655172052736,0.279710448158856,0.392455217016153,0.894571898611389,0.718544778358204,0.937921646140581,0.965534722629516,0.962625283456555,0.930627797832612,0.94971626031651,0.831460836238045,0.836030129464031,0.818987640098407,0.191311247243742,0.891146484363325,0.975002218986727,0.806242097712286,0.99089718486983,0.913227739142689,0.985863600168745,0.971394664387056,0.961781607577736,0.956531682334265,0.940085466700268,0.853561306024929,0.901610826697563,0.98714548246265,0.984149205619086,0.98848540054947,0.953871035722967,0.963422598038668,0.960494241888605,0.992003919137145,0.926811842480733,0.980926085315602,0.993287409805839],[0.0103212085926092,0.0127843416429432,0.00280419192644032,0.0199723995071528,0.0300602756534554,0.0148226482052767,-0.0263623098835986,-0.0024559712839114,0.0153409135527944,0.0548209438967291,-0.0177370508949188,-0.0134380856172562,-0.0468720567303862,0.0256113851600759,0.0133299947683803,-0.218481989648692,0.0487064668916553,0.0321921428239129,-0.037206212691661,-0.00407867982828414,0.0361432134627986,0.0076900380206304,0.004484808527355,0.0406830032173175,0.0452926358134711,-0.0309941507321838,-0.0611152602838949,-0.0352388451853365,0.00244859512423901,-0.118643143716923,-0.0234963404056856,0.0161134159625611,0.0114771738086645,0.0760710115165466,0.0347021974987811,0.0734698710826862,-0.0352959502444037,0.0122140074191108],[0.701119625371732,0.797476710212891,0.906509101990869,0.130766709894894,0.263100736063257,0.864029286199525,0.806313112185493,0.990556955221616,0.984958849240907,0.937950934220988,0.963194209884357,0.988270109829768,0.748311334355943,0.775064232393088,0.918154274742533,0.114437841953035,0.88505183820697,0.939957871544947,0.809127447919045,0.995729503125422,0.922190766127622,0.977173017481482,0.996526211822746,0.853108720675818,0.932069921730859,0.893913285407327,0.901544549948075,0.898411880492201,0.997957616118459,0.985919654935691,0.984429599887811,0.974921505631019,0.993560718926054,0.76980712969589,0.980465740912291,0.892458758840859,0.889842335750925,0.996637224794647]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>mbe_est_reverse<\/th>\n      <th>mbe_pval_reverse<\/th>\n      <th>mbe_est_rremove<\/th>\n      <th>mbe_pval_rremove<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[1,2,3,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div id="platelet" class="section level3">
<h3>Platelet</h3>
<div id="histogram-to-show-the-number-of-snps-that-each-pair-contains-5" class="section level4">
<h4>Histogram to show the number of SNPs that each pair contains</h4>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/snp_count_plt_cedar_qc_kegg_40genes.jpeg" width="80%">
</figure>
</div>
<div id="effectsize-fitting-summary-5" class="section level4">
<h4>effectsize fitting summary</h4>
<div id="htmlwidget-9ad70954d77b9fd07fab" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-9ad70954d77b9fd07fab">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33"],["EUR.UC_pwy165_pc2","EUR.UC_pwy211_pc2","allergy_pwy259_pc5","eo_pwy116_pc1","hct_pwy234_pc3","hgb_pwy121_pc3","hgb_pwy159_pc3","hgb_pwy180_pc5","hgb_pwy234_pc3","hgb_pwy255_pc5","hgb_pwy315_pc3","mch_pwy113_pc2","mch_pwy122_pc5","mch_pwy323_pc5","mchc_pwy117_pc3","mchc_pwy153_pc2","mchc_pwy272_pc2","mono_pwy161_pc4","mono_pwy228_pc4","mono_pwy331_pc4","pct_pwy56_pc4","plt_pwy138_pc3","plt_pwy185_pc1","plt_pwy186_pc1","rbc_pwy188_pc5","rdw_pwy287_pc5","ret_pwy161_pc1","ret_pwy169_pc5","ret_pwy294_pc5","ret_pwy316_pc4","ukb.allasthma_pwy138_pc1","wbc_pwy224_pc5","wbc_pwy237_pc2"],["04510","04810","05017","04022","04932","04064","04360","04625","04932","05010","05225","04014","04066","05322","04024","04261","05133","04371","04926","05416","00564","04142","04658","04659","04662","05164","04371","04530","05200","05226","04142","04922","04935"],["Focal adhesion","Regulation of actin cytoskeleton","Spinocerebellar ataxia","cGMP-PKG signaling pathway","Non-alcoholic fatty liver disease","NF-kappa B signaling pathway","Axon guidance","C-type lectin receptor signaling pathway","Non-alcoholic fatty liver disease","Alzheimer disease","Hepatocellular carcinoma","Ras signaling pathway","HIF-1 signaling pathway","Systemic lupus erythematosus","cAMP signaling pathway","Adrenergic signaling in cardiomyocytes","Pertussis","Apelin signaling pathway","Relaxin signaling pathway","Viral myocarditis","Glycerophospholipid metabolism","Lysosome","Th1 and Th2 cell differentiation","Th17 cell differentiation","B cell receptor signaling pathway","Influenza A","Apelin signaling pathway","Tight junction","Pathways in cancer","Gastric cancer","Lysosome","Glucagon signaling pathway","Growth hormone synthesis, secretion and action"],["105","120","52","81","108","69","84","73","108","111","99","111","72","62","91","64","49","72","65","46","45","99","64","79","70","119","72","93","284","71","99","62","67"],[2,2,2,2,2,3,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,3,2],[2,2,2,2,2,3,2,2,2,2,2,2,1,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,3,2],[0.015,0.004,0.002,0.01,0.001,0.006,0.037,0.03,0.003,0.049,0.079,0.028,0.004,0.04,0.002,0.014,0.005,0.005,0.017,0.028,0.005,0.026,0.023,0.018,0.009,0.002,0.021,0.012,0.006,0.018,0.007,0.004,0.01],[-0.504,0.499,0.117,0.146,0.109,-0.069,0.056,0.06,0.102,-0.044,-0.052,-0.081,0.12,0.074,-0.121,-0.081,-0.095,0.052,-0.053,-0.075,0.091,0.087,-0.127,0.104,-0.078,0.058,0.072,0.067,0.043,-0.083,-0.105,-0.054,-0.105],[0.999,1,1,1,1,0.988,0.997,0.998,1,0.994,0.985,0.998,1,0.996,1,1,1,1,0.999,0.998,1,0.998,0.999,0.999,1,1,0.999,1,1,0.999,1,0.991,1],[0.049,0.005,0.012,0.025,0.002,0.044,0.027,0.024,0.014,0.031,0.042,0.039,0.014,0.045,0.004,0.026,0.007,0.015,0.037,0.048,0.005,0.037,0.021,0.011,0.021,0.004,0.027,0.025,0.007,0.033,0.023,0.02,0.032]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>factor<\/th>\n      <th>pwy<\/th>\n      <th>pwy_name<\/th>\n      <th>pwy_gene_num<\/th>\n      <th>num_snp_fdr.0.2<\/th>\n      <th>num_snp_noMHC<\/th>\n      <th>pval_in_fitting<\/th>\n      <th>slope_in_fitting<\/th>\n      <th>rsq_in_fitting<\/th>\n      <th>pval_in_resampling<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[5,6,7,8,9,10]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="checking-the-reverse-causality-5" class="section level4">
<h4>Checking the reverse causality</h4>
<p>**The suffix &quot;_reverse&quot; are the results for traits -&gt; LV (using all candidate SNPs after LD Clumping), the suffix &quot;_rremove&quot; are the results for traits -&gt; LV (using all candidate SNPs after LD Clumping but remove the SNPs have fdr &lt; 0.2)**</p>
<div id="htmlwidget-12eab46753bdd639aac3" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-12eab46753bdd639aac3">{"x":{"filter":"none","data":[["EUR.UC_pwy165_pc2","EUR.UC_pwy211_pc2","allergy_pwy259_pc5","eo_pwy116_pc1","hct_pwy234_pc3","hgb_pwy121_pc3","hgb_pwy159_pc3","hgb_pwy180_pc5","hgb_pwy234_pc3","hgb_pwy255_pc5","hgb_pwy315_pc3","mch_pwy113_pc2","mch_pwy122_pc5","mch_pwy323_pc5","mchc_pwy117_pc3","mchc_pwy153_pc2","mchc_pwy272_pc2","mono_pwy161_pc4","mono_pwy228_pc4","mono_pwy331_pc4","pct_pwy56_pc4","plt_pwy138_pc3","plt_pwy185_pc1","plt_pwy186_pc1","rbc_pwy188_pc5","rdw_pwy287_pc5","ret_pwy161_pc1","ret_pwy169_pc5","ret_pwy294_pc5","ret_pwy316_pc4","ukb.allasthma_pwy138_pc1","wbc_pwy224_pc5","wbc_pwy237_pc2"],[0.0159318429610726,-0.0283318450517207,0.0102184279379482,0.0487726875652186,0.0366284501608867,0.0270950482947363,0.00449578490937697,-0.0373506512272321,-0.0129688592446779,-0.0530390119913022,-0.0774234923720627,0.022091836882737,0.0323154813699045,-0.0258617992489647,0.0176360053675086,-0.120638507955309,0.0179029338716923,-0.00687720007585568,0.00748552464155239,0.107814608205928,-0.0166529342126616,-0.030109002527972,-0.04384996549223,-0.00405133730779283,-0.0235639720038563,-0.0354942210412901,0.014226913893193,-0.0141745874976813,0.00535720037938603,0.0260001530550849,-0.0940872095335212,0.0333968112972016,0.0229009850839503],[0.265357479062023,0.159833557705757,0.938497109227227,0.829112787412366,0.607372243765366,0.901091776117401,0.999191919246935,0.754249543713983,0.861743246909501,0.700288449581416,0.710485131713374,0.977734419122975,0.974622696953685,0.985309184626829,0.809364674501557,0.974161743765073,0.870596232522165,0.962152752106653,0.973871946165318,0.658265518752969,0.966054981319258,0.948827598664822,0.895180430418587,0.998671442938058,0.924413142436059,0.978882238198393,0.946687424304278,0.953433664325173,0.981010582856336,0.978118408077272,0.636465787865512,0.905081058301344,0.912736968012126],[0.0114936679908797,-0.0225864971417007,0.0186508920016415,0.047903506588721,0.0192414702966701,0.0103977052690354,0.0548372298343871,-0.0308819830790448,-0.00210358221814033,-0.0101150860914672,-0.0744722582721593,0.00824149988005907,-0.0236754698316357,0.000640457805236139,0.00212924738907283,-0.0996526263632909,0.00881600426519924,-0.03127506028342,0.0283965483823678,0.0717554418131421,-0.00852858879298068,-0.00172422862940635,-0.0449003568360915,-0.00813130073327528,-0.00856363479920574,-0.0454649093726687,-0.00518951115278909,-0.0147108789809671,0.0637847852421451,0.00608292880471595,-0.119330232676488,0.0321466180658767,-0.00991555047394321],[0.403600134961324,0.27154956699553,0.889476795413025,0.830033935551214,0.787122360243633,0.961851379496996,0.990089042410892,0.798391602346201,0.977381093350112,0.941155981106073,0.71699486783318,0.99163515531342,0.981128043601215,0.999627743351178,0.976443853332108,0.978772720082944,0.934420844285449,0.831788034118058,0.900750674817502,0.770646319933821,0.982480622370337,0.997021469663891,0.894212540615998,0.997330259519125,0.972611685623279,0.972047326832732,0.980726290992226,0.951057365635477,0.78310393510015,0.994897140682729,0.546667562744323,0.907836770542883,0.961792696702314]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>mbe_est_reverse<\/th>\n      <th>mbe_pval_reverse<\/th>\n      <th>mbe_est_rremove<\/th>\n      <th>mbe_pval_rremove<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[1,2,3,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] workflowr_1.6.2

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.5        rstudioapi_0.11   whisker_0.3-2     knitr_1.30       
 [5] magrittr_1.5      R6_2.4.1          rlang_0.4.8       highr_0.8        
 [9] stringr_1.4.0     tools_3.6.1       DT_0.15           xfun_0.18        
[13] git2r_0.26.1      crosstalk_1.1.0.1 htmltools_0.5.0   ellipsis_0.3.1   
[17] rprojroot_1.3-2   yaml_2.2.1        digest_0.6.25     tibble_3.0.3     
[21] lifecycle_0.2.0   crayon_1.3.4      later_1.1.0.1     htmlwidgets_1.5.2
[25] vctrs_0.3.4       promises_1.1.1    fs_1.5.0          glue_1.4.2       
[29] evaluate_0.14     rmarkdown_1.13    stringi_1.5.3     compiler_3.6.1   
[33] pillar_1.4.6      backports_1.1.10  jsonlite_1.7.1    httpuv_1.5.1     
[37] pkgconfig_2.0.3  </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
