<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="XSun" />

<meta name="date" content="2021-02-19" />

<title>Genetic relationship between the expression factors and traits</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">factor_analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/xsun1229/factor_analysis">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Genetic relationship between the expression factors and traits</h1>
<h4 class="author">XSun</h4>
<h4 class="date">2021-02-19</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-08-28
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 2 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>factor_analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomxsun1229factoranalysistree49b5f4b037c82ef4c1274d6e55febcf52d7ade88targetblank49b5f4ba"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/xsun1229/factor_analysis/tree/49b5f4b037c82ef4c1274d6e55febcf52d7ade88" target="_blank">49b5f4b</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomxsun1229factoranalysistree49b5f4b037c82ef4c1274d6e55febcf52d7ade88targetblank49b5f4ba" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/xsun1229/factor_analysis/tree/49b5f4b037c82ef4c1274d6e55febcf52d7ade88" target="_blank">49b5f4b</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    analysis/.Rhistory
    Ignored:    output/.Rhistory

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/factors_traits.Rmd</code>) and HTML (<code>docs/factors_traits.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/49b5f4b037c82ef4c1274d6e55febcf52d7ade88/analysis/factors_traits.Rmd" target="_blank">49b5f4b</a>
</td>
<td>
XSun
</td>
<td>
2021-08-28
</td>
<td>
wflow_publish(“analysis/factors_traits.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/e2d11074d6a61838a5dd03ca01f427e5716dc699/docs/factors_traits.html" target="_blank">e2d1107</a>
</td>
<td>
XSun
</td>
<td>
2021-08-08
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/aa7d35ae5695016c5e8ed6b2e258f373767aa218/analysis/factors_traits.Rmd" target="_blank">aa7d35a</a>
</td>
<td>
XSun
</td>
<td>
2021-08-08
</td>
<td>
wflow_publish(“analysis/factors_traits.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/c4fd028ab7e6419a3732d06bc3832eb7e30768ac/docs/factors_traits.html" target="_blank">c4fd028</a>
</td>
<td>
XSun
</td>
<td>
2021-08-08
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/3e4002c3584f83c71918bc30ae8f675e7df01e6f/analysis/factors_traits.Rmd" target="_blank">3e4002c</a>
</td>
<td>
XSun
</td>
<td>
2021-08-08
</td>
<td>
wflow_publish(“analysis/factors_traits.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/e8c967882f74cc4b58ab13dc9572305559317f62/docs/factors_traits.html" target="_blank">e8c9678</a>
</td>
<td>
XSun
</td>
<td>
2021-08-08
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/1b1c35256a1da484bdf08edaf89e89ff2a555c95/analysis/factors_traits.Rmd" target="_blank">1b1c352</a>
</td>
<td>
XSun
</td>
<td>
2021-08-08
</td>
<td>
wflow_publish(“analysis/factors_traits.Rmd”)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/99de2c5ae8e523b1c6943d386b528cb7b18b8e3c/analysis/factors_traits.Rmd" target="_blank">99de2c5</a>
</td>
<td>
XSun
</td>
<td>
2021-08-08
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/22c057e42e210c7e88cff80145575379b5bfaa73/docs/factors_traits.html" target="_blank">22c057e</a>
</td>
<td>
XSun
</td>
<td>
2021-08-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/204f550172763fd8eae3222fb8d306832a00ac45/analysis/factors_traits.Rmd" target="_blank">204f550</a>
</td>
<td>
XSun
</td>
<td>
2021-08-02
</td>
<td>
wflow_publish(c(“analysis/factors_traits.Rmd”, “analysis/cedar_kegg_26_qc_d1k_pruned_Npeer.Rmd”))
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/253a9ec9dd18d8a987069f1b1e4bb394c2b913c9/analysis/factors_traits.Rmd" target="_blank">253a9ec</a>
</td>
<td>
XSun
</td>
<td>
2021-08-02
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/e5373fd31dd9c93047fede89ed3f1ea1df2792c4/docs/factors_traits.html" target="_blank">e5373fd</a>
</td>
<td>
XSun
</td>
<td>
2021-08-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/d7e4da918908536c4103b2bc43b0f68eb017c5e0/analysis/factors_traits.Rmd" target="_blank">d7e4da9</a>
</td>
<td>
XSun
</td>
<td>
2021-08-02
</td>
<td>
wflow_publish(c(“analysis/CEDAR_data_process.Rmd”, “analysis/cedar_kegg_26_qc_d1k_pruned_Npeer_40genes.Rmd”,
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/b1f7f1209dcd09704d2619ab212e6e6733719c39/docs/factors_traits.html" target="_blank">b1f7f12</a>
</td>
<td>
XSun
</td>
<td>
2021-08-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/8b23583a52e57cb94a57852b1668ef55464501f7/analysis/factors_traits.Rmd" target="_blank">8b23583</a>
</td>
<td>
XSun
</td>
<td>
2021-08-02
</td>
<td>
wflow_publish(c(“analysis/cedar_kegg_26_qc_d1k_pruned_Npeer.Rmd”,
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/1823d675de9e72a3e6f84426a004ab7cf120a68d/analysis/factors_traits.Rmd" target="_blank">1823d67</a>
</td>
<td>
XSun
</td>
<td>
2021-08-02
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/3f6fb22e012a382d091b8ee25497314edf7a6ae6/docs/factors_traits.html" target="_blank">3f6fb22</a>
</td>
<td>
XSun
</td>
<td>
2021-07-31
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/e16ea96987369b741166ebb2c99d6ae60a5a5117/analysis/factors_traits.Rmd" target="_blank">e16ea96</a>
</td>
<td>
XSun
</td>
<td>
2021-07-31
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/58fc0835ffe4b77ff4f00d535aae0307f0e66a0c/analysis/factors_traits.Rmd" target="_blank">58fc083</a>
</td>
<td>
XSun
</td>
<td>
2021-07-29
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/66bed798ba3b1cd0d828c9b672f13a37518ad0e1/docs/factors_traits.html" target="_blank">66bed79</a>
</td>
<td>
XSun
</td>
<td>
2021-07-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/46b163e27c78e3a8f8c6cfdab26d6f1c95de3e3e/analysis/factors_traits.Rmd" target="_blank">46b163e</a>
</td>
<td>
XSun
</td>
<td>
2021-07-17
</td>
<td>
wflow_publish(c(“analysis/factors_traits.Rmd”, “analysis/cedar_kegg_26_qc_d1k_pruned.Rmd”,
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/4948837a14cc4b1e1dd1917b270ae420d4b883b8/docs/factors_traits.html" target="_blank">4948837</a>
</td>
<td>
XSun
</td>
<td>
2021-07-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/d57572a2275a64a967b734afc881d3ad61393f98/analysis/factors_traits.Rmd" target="_blank">d57572a</a>
</td>
<td>
XSun
</td>
<td>
2021-07-17
</td>
<td>
wflow_publish(“analysis/factors_traits.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/0c26bee42ffafb7b89791230015fecf83df4b5d3/docs/factors_traits.html" target="_blank">0c26bee</a>
</td>
<td>
XSun
</td>
<td>
2021-07-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/c6bd0e4afc6bc2db2838e3f4b6de4da34248929b/analysis/factors_traits.Rmd" target="_blank">c6bd0e4</a>
</td>
<td>
XSun
</td>
<td>
2021-07-17
</td>
<td>
wflow_publish(c(“analysis/CEDAR_data_process.Rmd”, “analysis/EM_websites.Rmd”,
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/8d68502f993697ba2a37909bdb4f5a3f0eb8df94/docs/factors_traits.html" target="_blank">8d68502</a>
</td>
<td>
XSun
</td>
<td>
2021-06-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/2f5d3e81e0786dc1438236f1973728c9f70e766d/analysis/factors_traits.Rmd" target="_blank">2f5d3e8</a>
</td>
<td>
XSun
</td>
<td>
2021-06-21
</td>
<td>
wflow_publish(“analysis/factors_traits.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/600864c34081c6265a12b54008f09f898162132a/docs/factors_traits.html" target="_blank">600864c</a>
</td>
<td>
XSun
</td>
<td>
2021-05-31
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/2aefc66e05a1c4f6b218d85c260eedebc547caa9/analysis/factors_traits.Rmd" target="_blank">2aefc66</a>
</td>
<td>
XSun
</td>
<td>
2021-05-31
</td>
<td>
wflow_publish(c(“analysis/factor_traits_setting_compare_CEDAR_imputed.Rmd”,
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/85914ab3cf51cd98a34243802e78dcf536c3046c/docs/factors_traits.html" target="_blank">85914ab</a>
</td>
<td>
XSun
</td>
<td>
2021-04-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/3df38daf871e32f38ba87d18e9ce9b21b7f4fda5/analysis/factors_traits.Rmd" target="_blank">3df38da</a>
</td>
<td>
XSun
</td>
<td>
2021-04-22
</td>
<td>
wflow_publish(c(“analysis/factors_traits.Rmd”, “analysis/factor_traits_setting_compare.Rmd”))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/55cdc151850f97ad9891bbeadca0b1f3c7156f08/docs/factors_traits.html" target="_blank">55cdc15</a>
</td>
<td>
XSun
</td>
<td>
2021-04-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/3c4502751734ffe4e448ac40651091d322b4565a/analysis/factors_traits.Rmd" target="_blank">3c45027</a>
</td>
<td>
XSun
</td>
<td>
2021-04-18
</td>
<td>
wflow_publish(c(“analysis/factors_traits.Rmd”, “analysis/reactome_pcs_factor_traits_regress_5_d1k.Rmd”))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/be6391ca7bb32fd534a037502d58c647127ee6f9/docs/factors_traits.html" target="_blank">be6391c</a>
</td>
<td>
XSun
</td>
<td>
2021-04-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/3c98342ab578e959a74643c921919c7b3ced8aad/analysis/factors_traits.Rmd" target="_blank">3c98342</a>
</td>
<td>
XSun
</td>
<td>
2021-04-14
</td>
<td>
wflow_publish(c(“analysis/factors_traits.Rmd”, “analysis/reactome_pcs_factor_traits_regress_5_d500d_liver.Rmd”,
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/c6fa58c63ed7334240151d1eb2be33ac0fe56897/docs/factors_traits.html" target="_blank">c6fa58c</a>
</td>
<td>
XSun
</td>
<td>
2021-04-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/542a947b878f5a69def98a94e0111f71ce3ef1b4/docs/factors_traits.html" target="_blank">542a947</a>
</td>
<td>
XSun
</td>
<td>
2021-04-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/55fa28043551cbc0bc63773cc71c0ebb40c64810/analysis/factors_traits.Rmd" target="_blank">55fa280</a>
</td>
<td>
XSun
</td>
<td>
2021-04-14
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/cf2670cf89ae58ac634c9e8941c813107b519c9c/docs/factors_traits.html" target="_blank">cf2670c</a>
</td>
<td>
XSun
</td>
<td>
2021-04-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/b52df9401929119c676d595dc060c5146b9af6d6/analysis/factors_traits.Rmd" target="_blank">b52df94</a>
</td>
<td>
XSun
</td>
<td>
2021-04-10
</td>
<td>
wflow_publish(c(“analysis/kegg_pcs_factor_traits_regress_5_d500d_liver.Rmd”,
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/bde2114b78931d4a08d22795e7b8f751a12abca8/docs/factors_traits.html" target="_blank">bde2114</a>
</td>
<td>
XSun
</td>
<td>
2021-04-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/b5b832f52f8a0ef321231ea63c69e12dc268ec18/analysis/factors_traits.Rmd" target="_blank">b5b832f</a>
</td>
<td>
XSun
</td>
<td>
2021-04-03
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/d06e6735199aa9e4b394b28866b54b5d9603dc1a/docs/factors_traits.html" target="_blank">d06e673</a>
</td>
<td>
XSun
</td>
<td>
2021-03-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/efa811459eaeded923131639352a3c4efbbbe131/analysis/factors_traits.Rmd" target="_blank">efa8114</a>
</td>
<td>
XSun
</td>
<td>
2021-03-17
</td>
<td>
wflow_publish(c(“analysis/kegg_pcs_factor_traits_regress_5.Rmd”,
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/7c3af5ff5b153fd71053a290685d9f3e49d9c276/docs/factors_traits.html" target="_blank">7c3af5f</a>
</td>
<td>
XSun
</td>
<td>
2021-03-12
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/0cc34324fd8a4786d1875056172c50f9594237f2/analysis/factors_traits.Rmd" target="_blank">0cc3432</a>
</td>
<td>
XSun
</td>
<td>
2021-03-12
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/b056b3358f78fecdf2a71690bd88d1053e8eb66b/docs/factors_traits.html" target="_blank">b056b33</a>
</td>
<td>
XSun
</td>
<td>
2021-03-05
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/17cfa842350dfa127cde0777f5fa257abae8ce38/analysis/factors_traits.Rmd" target="_blank">17cfa84</a>
</td>
<td>
XSun
</td>
<td>
2021-03-05
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/b0fb96efbf0227b2fb6dbd7898a6f02ba4557527/docs/factors_traits.html" target="_blank">b0fb96e</a>
</td>
<td>
XSun
</td>
<td>
2021-02-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/d8628015b38f897442bb0b2b7d7762dc59daa032/analysis/factors_traits.Rmd" target="_blank">d862801</a>
</td>
<td>
XSun
</td>
<td>
2021-02-23
</td>
<td>
wflow_publish(“analysis/factors_traits.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/5c0e1ef04e43c8e1f539946872ed7e9ea1722e7c/docs/factors_traits.html" target="_blank">5c0e1ef</a>
</td>
<td>
XSun
</td>
<td>
2021-02-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/b53847b25f60d5b8b841d12f235be2109915ca96/analysis/factors_traits.Rmd" target="_blank">b53847b</a>
</td>
<td>
XSun
</td>
<td>
2021-02-23
</td>
<td>
wflow_publish(“analysis/factors_traits.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/770d0b846ce7db86b964f930cc2b858e059fd6c0/docs/factors_traits.html" target="_blank">770d0b8</a>
</td>
<td>
XSun
</td>
<td>
2021-02-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/312103d679850dce50e5bcf5a13fffe1c682e2ea/analysis/factors_traits.Rmd" target="_blank">312103d</a>
</td>
<td>
XSun
</td>
<td>
2021-02-22
</td>
<td>
wflow_publish(c(“analysis/cispathway.Rmd”, “analysis/Susie_results.Rmd”,
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/00467495ee08fb6f83bdfc942f39bcb660469b75/docs/factors_traits.html" target="_blank">0046749</a>
</td>
<td>
XSun
</td>
<td>
2021-02-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/2a60188433cd7920db50301652644465c3f41c3c/analysis/factors_traits.Rmd" target="_blank">2a60188</a>
</td>
<td>
XSun
</td>
<td>
2021-02-19
</td>
<td>
wflow_publish(“analysis/factors_traits.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/98aecea78b3c9e6466b468eeccb867e4a3a6ea59/docs/factors_traits.html" target="_blank">98aecea</a>
</td>
<td>
XSun
</td>
<td>
2021-02-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/bebebccac130569144e93c1ba4cd92947f555378/analysis/factors_traits.Rmd" target="_blank">bebebcc</a>
</td>
<td>
XSun
</td>
<td>
2021-02-19
</td>
<td>
wflow_publish(“analysis/factors_traits.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/fa63c4513463c7b592b6506b8af36f93e7a2b5a0/docs/factors_traits.html" target="_blank">fa63c45</a>
</td>
<td>
XSun
</td>
<td>
2021-02-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/1907e0e3e63e25edc3db228518a33845a7c237d6/analysis/factors_traits.Rmd" target="_blank">1907e0e</a>
</td>
<td>
XSun
</td>
<td>
2021-02-19
</td>
<td>
wflow_publish(c(“analysis/index.Rmd”, “analysis/variants_factors.Rmd”,
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="overview" class="section level1">
<h1><strong>1. Overview</strong></h1>
<p>In this part, we are trying to study effects of factors on phenotypes with Mendelian Randomization. Factors are inferred from expression matrix</p>
<p>The hypothesis is that the factors have a causal effect on outcome(traits). But it’s difficult to establish.<br />
So we state a weaker hypothesis first: GWAS variants of a trait often converge to a relatively small number of processes, which can be represented as expression factors.</p>
<p>Then we strengthen hypothesis: if the effects of SNPs on LV is correlated with their effects on trait when there is a causal effect of LV on trait.</p>
<p>Another hypothesis in this part is that if expression factors can influence the certain phenotype (trait or disease), the variants of the phenotype should be enriched in the factors.</p>
<p>We tried GTEx and CEDAR data set for the analysis in this part.</p>
</div>
<div id="data" class="section level1">
<h1><strong>2. Data</strong></h1>
<p>The GWAS summary statistics data we used are <a href="https://xsun1229.github.io/factor_analysis/GWAS_data.html">here</a></p>
</div>
<div id="pipeline" class="section level1">
<h1><strong>3. Pipeline</strong></h1>
<div id="ascertaining-diseasetrait-associated-variants" class="section level2">
<h2><strong>3.1 Ascertaining disease/trait-associated variants</strong></h2>
<ol style="list-style-type: decimal">
<li><p>The candidate variants associated with diseases/traits were selected using p-values &lt; 5e-8 in GWAS summary statistics.</p></li>
<li><p>Then we did 3-step LD Clumping for the SNPs in step 1) to find independent variants to be used in MR.</p></li>
</ol>
<ul>
<li>Use PLINK to do the preliminary filter: in 1Mb block, the SNPs have LD R2 &gt; 0.1 will be filtered by their p-values</li>
<li>For the remaining SNPs within 1Mb, keep the one have lowest p-value, remove the others.</li>
<li>Compute the LD R2 for the remaining SNPs, for the SNP pairs have R2 &gt; 0.1, filter them again according their p-values.</li>
</ul>
<p>After step 1) and 2), for each trait, a subset of variants associated to it was obtained. Each pair from these variants has LD R2 &lt; 0.1 and physical distance &gt; 1Mb.</p>
<p><strong><a href="https://xsun1229.github.io/factor_analysis/CEDAR_number_variants_each_trait">The number of trait-associated variants is here</a></strong></p>
</div>
<div id="sign-consistency-checking" class="section level2">
<h2><strong>3.2 Sign consistency checking</strong></h2>
<div id="association-tests" class="section level3">
<h3>3.2.1 Association tests</h3>
<ol style="list-style-type: decimal">
<li>we did association tests for expression factors ~ SNPs + 10 genotype PCs + batch covariates +age +sex</li>
</ol>
<p>The 10 genotype PCs were computed using 2 approaches:</p>
<ul>
<li><p>Computing PCs on whole genome using PLINK1.9</p></li>
<li><p>Pruning the whole genome first and compute the PCs (annotated as pruned_PCs, details are <a href="https://xsun1229.github.io/factor_analysis/CEDAR_data_process.html#2_genotype_pcs">here</a>)</p></li>
</ul>
<ol start="2" style="list-style-type: decimal">
<li><p>We recorded the trait-factor pairs that have more than one SNPs have FDR&lt;0.2. Fitted beta.outcome ~ beta.exposure + 0 using the SNPs associated to these pairs. Recorded the fitting p-values.</p></li>
<li><p>We harmonized the data using TwoSampleMR R package</p></li>
<li><p>For the trait-factor pairs in step 2), fitting beta.outcome ~ beta.exposure + 0 using the variants associated to these pairs. Recorded the fitting p-values</p></li>
</ol>
</div>
<div id="resampling" class="section level3">
<h3>3.2.2 Resampling</h3>
<p>To make our results more reliable, we did resampling.</p>
<ol style="list-style-type: decimal">
<li><p>We resampled the beta.outcome and beta.exposure from the origin dataset(the data set containing all SNPs) without replacement, made the fitting again and recorded the r-squareds to compute the p-values in resampling.</p></li>
<li><p>The ‘p-value from resampling’ is computed by: (number of more extreme values)/(times of resampling).The resampling was repeated 1000 times.</p></li>
<li><p>We computed the enrichment of the pairs show sign consistency than expect follows: enrichment=num_(pairs show sign consistancy) /(num_(total pairs) × threshold)</p></li>
</ol>
<ul>
<li><p>We first tried to do resampling 1000 times for each pair have &gt;2 SNPs at FDR &lt; 0.2</p></li>
<li><p>We also tried an adaptive method to do the resampling. In the first stage, 10e3 simulations will be performed. If the resulting empirical p value is less than 0.1, 10e4 simulations will be performed. If the empirical p value from 10e4 simulations is less than 0.001, the program will perform 10e6 simulations. <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2896770/">reference</a></p></li>
</ul>
</div>
<div id="checking-reverse-causality" class="section level3">
<h3>3.2.3 Checking reverse causality</h3>
<p>To check the reverse causality, we exchanged the beta.outcome and beta.exposure of all SNPs associated with traits(pval&lt;5E-8) to compute the MBE estimator and the p-values of the estimator.</p>
<p>Weighted mode-based estimate (MBE) was adopted to check the reverse causality. mr_mbe() function from MendelianRandomization R package(version 0.3.0) was used to compute the weighted MBE with default settings.</p>
</div>
</div>
<div id="colocalization-analysis" class="section level2">
<h2><strong>3.3 Colocalization analysis</strong></h2>
<p>We did colocalization analysis whether factor-association signals and GWAS signals at a locus are consistent with having a shared casual variant. The colocalization analysis was performed using the approximate Bayes factor test implemented by coloc.abf() function in ‘coloc’ R package. ‘coloc’ computes five posterior probabilities (PP0, PP1, PP2, PP3 and PP4), each corresponding to a hypothesis: H0, no association with either trait; H1, association with trait 1 but not with trait 2; H2, association with trait 2 but not with trait 1; H3, association with trait 1 and trait 2, two independent SNPs; H4, association with trait 1 and trait 2, one shared SNP. We ran ‘coloc’ with the default parameters and used PP4 to assess evidence of colocalization.</p>
<p>We will do colocalization for the variants from the pair shows sign consistency, which are named as factor-QTLs by us. The colocalization was performed follows:</p>
<ol style="list-style-type: decimal">
<li><p>For a factor-QTL, the variants less than 500kb were selected to do the analysis</p></li>
<li><p>Association tests follows 3.3.3.1 were performed for these variants. The p-values were recorded.</p></li>
<li><p>Colocalization analysis was did for the association signals in step 2) and GWAS signals.</p></li>
<li><p>The colocalization results were visualized the using the ‘LocusCompareR’ package (version 1.0.0, downloaded from <a href="https://github.com/boxiangliu/locuscomparer" class="uri">https://github.com/boxiangliu/locuscomparer</a>).</p></li>
</ol>
</div>
<div id="ora-enrichment-analysis" class="section level2">
<h2><strong>3.4 ORA enrichment analysis</strong></h2>
<p>ORA: For each factor, we sorted the genes that have non-zero loadings by their loadings, taking the top 25% or 100% as the gene set. The function database here is geneontology Biological Process. The reference set affy hugene 2 0 st v1. Minimum number of genes for a category is 5, maximum number of genes for a category is 2000(default settings). All categories that have fdr&lt;0.05 are listed.</p>
</div>
</div>
<div id="gtex" class="section level1">
<h1><strong>4. GTEx</strong></h1>
<div id="data-1" class="section level2">
<h2><strong>4.1 Data</strong></h2>
<div id="expression-factors" class="section level3">
<h3>4.1.1 Expression factors</h3>
<p>We tried four types of expression factors: 1)PLIER factors 2)PCs from KEGG/Reactome pathways 3)PEER factors 4)FLASH factors 5)PEER factors from KEGG/Reactome pathways. They are the same factors with those used in the prediction model training part.</p>
<p>When using PCs, we regressed out the first 5 PEER factors computed from the GTEx expression data.</p>
<p>The KEGG contains 332 pathways. We have 1866 Reactome pathways. Combining with our GTEx data, some of them have less than 5 genes that have expression data. Since we will infer 5 PCs for each pathway, we just use the pathways have more than 5 genes. The number of pathways we use for whole blood is 1221, the number for liver is 1262.</p>
<table>
<thead>
<tr class="header">
<th>Type</th>
<th>Factor number</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>PLIER - Canonical pathways</td>
<td>128</td>
</tr>
<tr class="even">
<td>PLIER - MSigDB pathways</td>
<td>98</td>
</tr>
<tr class="odd">
<td>PCs - KEGG</td>
<td>1646</td>
</tr>
<tr class="even">
<td>PCs - Reactome</td>
<td>6105</td>
</tr>
<tr class="odd">
<td>PEER</td>
<td>60</td>
</tr>
<tr class="even">
<td>FLASH</td>
<td>100</td>
</tr>
</tbody>
</table>
</div>
<div id="genotype-data" class="section level3">
<h3>4.1.2 Genotype data</h3>
<p>GTEx genotype data. The variants have maf &gt;0.01 were selected, variants on chrX were excluded). We have 10,403,249 variants in total.</p>
<p>The genotype PCs used in this part are computed from whole genome, using PLINK1.9.</p>
</div>
<div id="association-test-covariates" class="section level3">
<h3>4.1.3 Association test covariates</h3>
<p>For GTEx data, the batch covariates were sequencing platform and sequencing protocol. 10 genotype PCs were computed on whole genome using PLINK 1.9.</p>
</div>
</div>
<div id="results-under-different-settings" class="section level2">
<h2><strong>4.2 Results under different settings</strong></h2>
<div id="plier-msigdb-factors-latest-results-0721" class="section level3">
<h3>PLIER-MSigDB factors <strong>(latest results 07/21)</strong></h3>
<ul>
<li><a href="https://xsun1229.github.io/factor_analysis/gtex_plier_d1k.html">Whole Blood</a></li>
</ul>
</div>
<div id="plier-canonical-factors" class="section level3">
<h3>PLIER-Canonical factors</h3>
<ul>
<li><a href="https://xsun1229.github.io/factor_analysis/catalog_gwas_pliercanon_sep_ldclumping_r01d1000.html">Whole Blood</a></li>
</ul>
<p>For the factors in this part, we also tried EM algorithm to estimate the causal effects. <a href="https://xsun1229.github.io/factor_analysis/EM_websites.html">Details</a></p>
<p>For the factors in this part, we used SuSiE to test if our colocalization results are reliable. <a href="https://xsun1229.github.io/factor_analysis/Susie_results.html">For more details</a></p>
<ul>
<li><p><a href="https://xsun1229.github.io/factor_analysis/catalog_gwas_pliercanon_sep_ldclumping_r01d1000_pval5e7.html">Whole Blood, first filtered by pval&lt;5e-7</a></p></li>
<li><p><a href="https://xsun1229.github.io/factor_analysis/catalog_gwas_pliercanon_sep_ldclumping_r01d1000_adipose_sub.html">Adipose Subcutaneous</a></p></li>
<li><p><a href="https://xsun1229.github.io/factor_analysis/catalog_gwas_pliercanon_sep_ldclumping_r01d1000_muscle.html">Skeletal muscle</a></p></li>
<li><p><a href="https://xsun1229.github.io/factor_analysis/catalog_gwas_pliercanon_sep_ldclumping_r01d500d_liver.html">Liver</a></p></li>
<li><p><a href="https://xsun1229.github.io/factor_analysis/catalog_gwas_pliercanon_sep_ldclumping_r01d1000_geu.html">GEUVADIS</a></p></li>
</ul>
</div>
<div id="peer" class="section level3">
<h3>PEER</h3>
<ul>
<li><a href="https://xsun1229.github.io/factor_analysis/catalog_gwas_peer_sep_ldclumping_r01d1000.html">Whole Blood</a></li>
</ul>
</div>
<div id="flash" class="section level3">
<h3>FLASH</h3>
<ul>
<li><a href="https://xsun1229.github.io/factor_analysis/catalog_gwas_flash_sep_ldclumping_r01d1000.html">Whole Blood</a></li>
</ul>
</div>
<div id="pcs---kegg" class="section level3">
<h3>PCs - KEGG</h3>
<p><a href="https://xsun1229.github.io/factor_analysis/gtex_kegg_26_40genes.html">Whole blood</a> <strong>Latest results</strong></p>
<p><a href="https://xsun1229.github.io/factor_analysis/kegg_pcs_factor_traits_regress_5_d1k.html">Whole blood</a></p>
<p><a href="https://xsun1229.github.io/factor_analysis/kegg_pcs_factor_traits_regress_5_d500d_liver.html">Liver</a></p>
</div>
<div id="peer---kegg" class="section level3">
<h3>PEER - KEGG</h3>
<p><a href="https://xsun1229.github.io/factor_analysis/kegg_peer_factor_traits.html">Whole Blood</a></p>
</div>
<div id="pcs---reactome" class="section level3">
<h3>PCs - REACTOME</h3>
<p><a href="https://xsun1229.github.io/factor_analysis/reactome_pcs_factor_traits_regress_5_d1k.html">Whole Blood</a></p>
<p><a href="https://xsun1229.github.io/factor_analysis/reactome_pcs_factor_traits_regress_5_d1k_liver.html">Liver</a></p>
<p>We compared the enrichment results under different settings(PCs from KEGG and REACTOME, PLIER - Canonical).</p>
<p><a href="https://xsun1229.github.io/factor_analysis/factor_traits_setting_compare.html">Comparison</a></p>
</div>
</div>
</div>
<div id="cedar" class="section level1">
<h1><strong>5. CEDAR</strong></h1>
<div id="data-2" class="section level2">
<h2><strong>5.1 Data</strong></h2>
<div id="cedar-data-processing" class="section level3">
<h3>5.1.1 CEDAR data processing</h3>
<p><a href="https://xsun1229.github.io/factor_analysis/CEDAR_data_process.html">Data overview</a></p>
</div>
<div id="expression-factors-1" class="section level3">
<h3>5.1.2 Expression factors</h3>
<p>We tried 2 types of expression factors: 1)PLIER factors 2)PCs from KEGG pathways.</p>
<p><strong>Expression factor numbers for 6 cell types</strong></p>
<table>
<thead>
<tr class="header">
<th>Cell type</th>
<th>Number of LVs</th>
<th>Number of PCs</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>CD4+ T cells</td>
<td>146</td>
<td>1627</td>
</tr>
<tr class="even">
<td>CD8+ T cells</td>
<td>134</td>
<td>1621</td>
</tr>
<tr class="odd">
<td>CD19+ B cells</td>
<td>136</td>
<td>1620</td>
</tr>
<tr class="even">
<td>CD14+ monocytes</td>
<td>88</td>
<td>1628</td>
</tr>
<tr class="odd">
<td>CD15+ neutrophils</td>
<td>146</td>
<td>1614</td>
</tr>
<tr class="even">
<td>platelets</td>
<td>132</td>
<td>1593</td>
</tr>
</tbody>
</table>
</div>
<div id="association-test-covariates-1" class="section level3">
<h3>5.1.3 Association test covariates</h3>
<p>When using PCs as exposure, we regressed out the 5 or N PEER factors computed from the GTEx expression data</p>
<div id="peer-factors" class="section level4">
<h4>5 PEER factors</h4>
<p>The 5 PEER factors are the first 5 factors computed <a href="https://xsun1229.github.io/factor_analysis/CEDAR_data_process.html#2_peer_factors">here</a></p>
</div>
<div id="n-peer-factors" class="section level4">
<h4>N PEER factors</h4>
<p>One ‘danger’ of adjusting for many PEER factors is that the factors may be affected by the genotypes. In other words, we could have this model: genetic variant -&gt; PEER factor -&gt; pathway. We should not adjust for the factor, as it will explain away the variant &gt; pathway association.</p>
<p>So, when testing variant ~ pathway association, we did a association test for variant and a PEER factor, if its p-value is small (&lt; 0.05), we did not adjust that factor. For example, we are testing association of a KEGG pathway with a SNP, and we have 20 PEER factor. We first test association of SNP with each of the 20 PEER factors: if p &lt; 0.05, then we do NOT regress out that factor.</p>
<p>We tried first 20 and 10 candidate PEER factors as covariates to be regressed out.</p>
</div>
</div>
</div>
<div id="results-under-different-settings-1" class="section level2">
<h2><strong>5.2 Results under different settings</strong></h2>
<p>The 10 genotype pcs using as covariates in factor ~ SNP association tests are computed from pruned genotype data set with bigsnpr.</p>
<div id="plier---msigdb-factors" class="section level3">
<h3>PLIER - MSigDB factors</h3>
<p><a href="https://xsun1229.github.io/factor_analysis/cedar_plier_validation.html">Workflow validation</a></p>
<p><a href="https://xsun1229.github.io/factor_analysis/cedar_plier_26_qc_d1k_pruned.html">Details</a></p>
</div>
<div id="pc---kegg" class="section level3">
<h3>PC - KEGG</h3>
<p><strong>regress out 5 PEER factors</strong></p>
<p>In this part, when doing association test for PCs ~ variants, we added first 5 PEER factors as covariates.</p>
<p><a href="https://xsun1229.github.io/factor_analysis/cedar_kegg_26_qc_d1k_pruned.html">Details - all pathways included</a></p>
<p><a href="https://xsun1229.github.io/factor_analysis/cedar_kegg_26_qc_d1k_40genes_pruned.html">Details - only pathways have &gt; 40 genes included</a></p>
<p><strong>regress out N PEER factors, N &lt;= 20</strong></p>
<p>In this part, when doing association test for PCs ~ variants, we added first N PEER factors as covariates(N &lt;= 20). The details are described in 5.1.3.</p>
<p><a href="https://xsun1229.github.io/factor_analysis/cedar_kegg_26_qc_d1k_pruned_Npeer.html">Details - all pathways</a></p>
<p><a href="https://xsun1229.github.io/factor_analysis/cedar_kegg_26_qc_d1k_pruned_Npeer_20genes.html">Details - pathways&gt; 20 genes</a></p>
<p><a href="https://xsun1229.github.io/factor_analysis/cedar_kegg_26_qc_d1k_pruned_Npeer_40genes.html">Details - pathways&gt; 40 genes</a></p>
<p><strong>regress out N PEER factors, N &lt;= 10</strong></p>
<p>In this part, when doing association test for PCs ~ variants, we added first N PEER factors as covariates(N &lt;= 10). The details are described in 5.1.3.</p>
<p><a href="https://xsun1229.github.io/factor_analysis/cedar_kegg_26_qc_d1k_pruned_10peer.html">Details - all pathways</a></p>
<p><a href="https://xsun1229.github.io/factor_analysis/cedar_kegg_26_qc_d1k_pruned_10peer_20genes.html">Details - pathways&gt; 20 genes</a></p>
<p><a href="https://xsun1229.github.io/factor_analysis/cedar_kegg_26_qc_d1k_pruned_10peer_40genes.html">Details - pathways&gt; 40 genes</a></p>
</div>
<div id="earlier-results" class="section level3">
<h3>Earlier results</h3>
<p>The 10 genotype pcs using as covariates in factor ~ SNP association tests are computed from whole genome with PLINK 1.9.</p>
<div id="plier---msigdb-factors-1" class="section level4">
<h4>PLIER - MSigDB factors</h4>
<p><a href="https://xsun1229.github.io/factor_analysis/cedar_plier_26_qc_d1k.html">Details</a></p>
</div>
<div id="pc---kegg-1" class="section level4">
<h4>PC - KEGG</h4>
<p><a href="https://xsun1229.github.io/factor_analysis/cedar_kegg_26_qc_d1k.html">Details - regress out 5 PEER factors</a></p>
<p><a href="https://xsun1229.github.io/factor_analysis/cedar_kegg_26_qc_d1k_40genes.html">Details - regress out 5 PEER factors &gt;40 genes</a></p>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
