<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Factor Analysis</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">factor_analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/xsun1229/factor_analysis">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Factor Analysis</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-09-18
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 2 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>factor_analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomxsun1229factoranalysistree912d18fd832768c26af250b1f50737e011420162targetblank912d18fa"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/xsun1229/factor_analysis/tree/912d18fd832768c26af250b1f50737e011420162" target="_blank">912d18f</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomxsun1229factoranalysistree912d18fd832768c26af250b1f50737e011420162targetblank912d18fa" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/xsun1229/factor_analysis/tree/912d18fd832768c26af250b1f50737e011420162" target="_blank">912d18f</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    analysis/.Rhistory

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/index.Rmd</code>) and HTML (<code>docs/index.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/912d18fd832768c26af250b1f50737e011420162/analysis/index.Rmd" target="_blank">912d18f</a>
</td>
<td>
XSun
</td>
<td>
2020-09-18
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/67decceb35fb14db200766b38e04398b51d82509/docs/index.html" target="_blank">67decce</a>
</td>
<td>
XSun
</td>
<td>
2020-08-27
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/9b2b31462b8c241e6f15bf212d6ffcd291d45e52/analysis/index.Rmd" target="_blank">9b2b314</a>
</td>
<td>
XSun
</td>
<td>
2020-08-27
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/2eac5ab00693f7a31f9790040312d63ca8ac3f45/docs/index.html" target="_blank">2eac5ab</a>
</td>
<td>
XSun
</td>
<td>
2020-08-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/ff043f35390e361ef0d6767064ef273c28af5a71/analysis/index.Rmd" target="_blank">ff043f3</a>
</td>
<td>
XSun
</td>
<td>
2020-08-25
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/060e639c3a7b1638e0618d60063e621f1f0a7d75/docs/index.html" target="_blank">060e639</a>
</td>
<td>
XSun
</td>
<td>
2020-08-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/2891c67fe3111154df9b6abcf6bccc1f32151e2c/analysis/index.Rmd" target="_blank">2891c67</a>
</td>
<td>
XSun
</td>
<td>
2020-08-16
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/83950f0a3b1297820d1bed98ad5169c0f05b95d0/docs/index.html" target="_blank">83950f0</a>
</td>
<td>
XSun
</td>
<td>
2020-08-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/87573852db134bfe813a9a4cc1e6d6bedb007634/analysis/index.Rmd" target="_blank">8757385</a>
</td>
<td>
XSun
</td>
<td>
2020-08-16
</td>
<td>
wflow_publish(c(“analysis/cg-onesample.Rmd”, “analysis/index.Rmd”,
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/44e1617b5e8ff3b9fd2d5becefdd3aa4ea563203/docs/index.html" target="_blank">44e1617</a>
</td>
<td>
XSun
</td>
<td>
2020-08-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/d887dc2afea790cf9134c2cfe2848a148eca4daa/analysis/index.Rmd" target="_blank">d887dc2</a>
</td>
<td>
XSun
</td>
<td>
2020-08-01
</td>
<td>
wflow_publish(c(“analysis/index.Rmd”, “analysis/weak_bias.Rmd”,
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/f614cdabd0d5b025728d5561a11463d17cc417e4/docs/index.html" target="_blank">f614cda</a>
</td>
<td>
XSun
</td>
<td>
2020-07-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/23f45efa0b59790f64af4d662dd874660fe3dcb5/analysis/index.Rmd" target="_blank">23f45ef</a>
</td>
<td>
XSun
</td>
<td>
2020-07-20
</td>
<td>
wflow_publish(c(“analysis/index.Rmd”, “analysis/T1D.Rmd”))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/cbc665990ad7e125af519bd794a1681538d274ae/docs/index.html" target="_blank">cbc6659</a>
</td>
<td>
XSun
</td>
<td>
2020-07-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/f88704ff1b2c99b173c4d828725f8c26af4bc536/analysis/index.Rmd" target="_blank">f88704f</a>
</td>
<td>
XSun
</td>
<td>
2020-07-13
</td>
<td>
wflow_publish(c(“analysis/index.Rmd”))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/3781fda60e220bc14a5e286089143e4716b100c5/docs/index.html" target="_blank">3781fda</a>
</td>
<td>
XSun
</td>
<td>
2020-07-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/a0378aff77704f45a9b757383957c017ebdaf6ae/analysis/index.Rmd" target="_blank">a0378af</a>
</td>
<td>
XSun
</td>
<td>
2020-07-13
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/5c8044ef89474d686e040f6574fed8f2dcfb465b/docs/index.html" target="_blank">5c8044e</a>
</td>
<td>
XSun
</td>
<td>
2020-07-11
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/f5320404814803d591515051ff09e068b367627a/analysis/index.Rmd" target="_blank">f532040</a>
</td>
<td>
XSun
</td>
<td>
2020-07-11
</td>
<td>
wflow_publish(c(“analysis/index.Rmd”))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/5535a7a42ac0a38815304cadd04615106f7a9741/docs/index.html" target="_blank">5535a7a</a>
</td>
<td>
XSun
</td>
<td>
2020-07-11
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/5f65ccf6577306b939fdba93e84aa86f6f94bc04/analysis/index.Rmd" target="_blank">5f65ccf</a>
</td>
<td>
XSun
</td>
<td>
2020-07-11
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/ec5e5dd3cd6871f4d920ba49a36447664876cfd3/docs/index.html" target="_blank">ec5e5dd</a>
</td>
<td>
XSun
</td>
<td>
2020-07-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/9abe220c380520f6b8373528391b7c1f8dc52d1d/analysis/index.Rmd" target="_blank">9abe220</a>
</td>
<td>
XSun
</td>
<td>
2020-07-09
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/7e16686b1584a3c2790c3fa6a6647cd72e5641ea/docs/index.html" target="_blank">7e16686</a>
</td>
<td>
XSun
</td>
<td>
2020-07-08
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/0153beecfda6be6b91d2150e3520be96aacb5602/analysis/index.Rmd" target="_blank">0153bee</a>
</td>
<td>
XSun
</td>
<td>
2020-07-08
</td>
<td>
wflow_publish(c(“analysis/factor_plier_canon.Rmd”, “analysis/modu_factor_pliercanon.Rmd”,
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/d5698613af55f5bfd00b2709f07970bd32718489/analysis/index.Rmd" target="_blank">d569861</a>
</td>
<td>
XSun
</td>
<td>
2020-07-08
</td>
<td>
updatatetetete
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/7f3739fd05fe732067e3da950a21403baa8a0a61/docs/index.html" target="_blank">7f3739f</a>
</td>
<td>
XSun
</td>
<td>
2020-07-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/25e3bcb8f034e0e1c1b0964c8510a69bb7f15f6d/analysis/index.Rmd" target="_blank">25e3bcb</a>
</td>
<td>
XSun
</td>
<td>
2020-07-01
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/1042f4eb06f450af353d84c79c1462be39e48670/docs/index.html" target="_blank">1042f4e</a>
</td>
<td>
XSun
</td>
<td>
2020-06-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/82513dc84bf6a483539d394ab8235e8537497278/analysis/index.Rmd" target="_blank">82513dc</a>
</td>
<td>
XSun
</td>
<td>
2020-06-29
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/89244cbef8fb6d60e616239d6bed67bda2822fa6/docs/index.html" target="_blank">89244cb</a>
</td>
<td>
XSun
</td>
<td>
2020-06-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/73c95ec09c4aa7f1cc14c39b286696a951955697/analysis/index.Rmd" target="_blank">73c95ec</a>
</td>
<td>
XSun
</td>
<td>
2020-06-26
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/ebadf8425741a9a7185d28a572c68008c155130e/docs/index.html" target="_blank">ebadf84</a>
</td>
<td>
XSun
</td>
<td>
2020-06-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/c08d79e7315f01271f8bd1a046935864593a17a2/analysis/index.Rmd" target="_blank">c08d79e</a>
</td>
<td>
XSun
</td>
<td>
2020-06-25
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/87bb9e0e6332c23c08fcea6fc4ad7581c737296b/docs/index.html" target="_blank">87bb9e0</a>
</td>
<td>
XSun
</td>
<td>
2020-06-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/78a63351fa7081ba52ed1e3ebd2437eb274439ab/analysis/index.Rmd" target="_blank">78a6335</a>
</td>
<td>
XSun
</td>
<td>
2020-06-24
</td>
<td>
wflow_publish(all = T)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/329ae59816d549b249c87cb61efb706da43637db/analysis/index.Rmd" target="_blank">329ae59</a>
</td>
<td>
XSun
</td>
<td>
2020-06-23
</td>
<td>
Start workflowr project.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>This page is about factor analysis. Below are details of analyses.</p>
<div id="factor-construction" class="section level1">
<h1><strong>Factor Construction</strong></h1>
<p>I used three softwares, <a href="https://github.com/wgmao/PLIER">PLIER</a>, <a href="https://github.com/stephenslab/flashr">FLASH</a>, and <a href="https://github.com/PMBio/peer/">PEER</a> to construct factors from GTEx gene expression matrix of whole blood(670 samples, 19,696 genes). Genes on chrX were excluded when extracting factors using these softwares.</p>
<div id="plier" class="section level2">
<h2>PLIER</h2>
<p>PLIER(pathway-level information extractor) is a broadly applicable method to accurately infer relevant biological insights, such as variation in cell-type proportion or pathway activity, from global gene expression studies. It approximates the expression pattern of every gene as a linear combination of eigengene-like latent variables (LVs).</p>
<ul>
<li>plier-canonicalPathways</li>
</ul>
<p>In this part, I just used canonicalPathways which include KEGG and Reactome pathways to extract factors and obtained 128 factors. <a href="https://xsun1229.github.io/factor_analysis/factor_plier_canon.html">For more details</a></p>
<ul>
<li>plier-allpathways</li>
</ul>
<p>I used ‘PLIER’ package(with default settings) to construct factors related to all pathways provided by the software and got 96 factors. <a href="https://xsun1229.github.io/factor_analysis/factor_plier_allpathway.html">For more details</a></p>
</div>
<div id="flash" class="section level2">
<h2>FLASH</h2>
<p>Using ‘FLASH’, I obtained 100 factors from our GTEx expression matrix.</p>
</div>
<div id="peer" class="section level2">
<h2>PEER</h2>
<p>Using ‘PEER’, I obtained 60 factors from the GTEx expression matrix.</p>
</div>
<div id="principal-component" class="section level2">
<h2>Principal Component</h2>
<p>I used PCA to infer several PCs for some special cases.</p>
</div>
</div>
<div id="factor-evaluation" class="section level1">
<h1><strong>Factor evaluation</strong></h1>
<p>I first evaluated the heritability of factors using <a href="https://cnsgenomics.com/software/gcta/#GREML">GCTA</a> and made histogrames to show the distribution of the heritabilities.</p>
<p>When computing heritability, I used GTEx genotype data(the variants with maf &gt;0.05 and maf &gt;0.01 were selected, variants on chrX are excluded). 6,568,202 variants have maf &gt;0.05 and 10,403,249 variants have maf &gt;0.01.</p>
<div id="plier-1" class="section level2">
<h2>PLIER</h2>
<ul>
<li>plier-canonicalPathways</li>
</ul>
<p><a href="https://xsun1229.github.io/factor_analysis/eval_plier_canon.html">For more details</a></p>
<ul>
<li>plier-allpathways</li>
</ul>
<p><a href="https://xsun1229.github.io/factor_analysis/eval_plier_allpathway.html">For more details</a></p>
</div>
<div id="flash-1" class="section level2">
<h2>FLASH</h2>
<p><a href="https://xsun1229.github.io/factor_analysis/eval_flash.html">For more details</a></p>
</div>
<div id="peer-1" class="section level2">
<h2>PEER</h2>
<p><a href="https://xsun1229.github.io/factor_analysis/eval_peer.html">For more details</a></p>
</div>
<div id="pcpeer-from-kegg-pathways" class="section level2">
<h2>PC&amp;PEER from KEGG pathways</h2>
<p><a href="https://xsun1229.github.io/factor_analysis/eval_kegg.html">For more details</a></p>
</div>
</div>
<div id="prediction-models-for-factors" class="section level1">
<h1><strong>Prediction models for factors</strong></h1>
<p>In this part, I tried to train prediction models for factors ~ genotype using Lasso and Elastic-net. I used 670 samples from GTEx. I used two groups of variants to train models. The first group contains all variants provided by GTEx with maf &gt; 0.01(10,403,249 variants) and maf &gt; 0.05(6,568,202 variants). The second group contains all significant cis-eQTLs provided by GTEx (FDR &gt; 0.05).</p>
<p>The workflow:</p>
<ol style="list-style-type: decimal">
<li><p>I splited the data set into training(80%, 536 samples) and test(20%, 134 samples) set.</p></li>
<li><p>In the training data, performed association test between the factors and all SNPs using Plink. I corrected the results for genotype PCs (We used Plink to compute 10 PCs and used them as corvariates in association test).</p></li>
<li><p>From 3, we selected all SNPs with p &lt; 1e-02 as candidate variants.</p></li>
<li><p>Trained Lasso/Elastic-net models using SNPs from step4 in the training data and evaluated the performance in the testing data.</p></li>
<li><p>For factors with relatively low pval_test_l, I did GWAS analysis using all GTEx samples(670 samples). In the association tests, first 10 genotype pcs were included as covariants.</p></li>
</ol>
<div id="training-models-with-all-variants" class="section level2">
<h2><strong>Training models with all variants</strong></h2>
<p>There are some missing values in our dataset. I used mean imputation (substitue the missing values with mean value of the corresponding variants) to impute the missing value.</p>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/ms_snp.jpeg" width="50%">
</figure>
<p>The missing data pattern of our dataset. x-axis represents the proportion of missing value for each variant. y-axis represents the number of the variants.</p>
<div id="plier-2" class="section level3">
<h3>PLIER</h3>
<p><a href="https://xsun1229.github.io/factor_analysis/train_plier_canon.html">Results for plier-canonicalPathways factors</a></p>
<p><a href="https://xsun1229.github.io/factor_analysis/train_plier_allpathway.html">Results for plier-allpathway factors</a></p>
</div>
<div id="flash-2" class="section level3">
<h3>FLASH</h3>
<p><a href="https://xsun1229.github.io/factor_analysis/train_flash.html">Results for FLASH factors</a></p>
</div>
<div id="peer-2" class="section level3">
<h3>PEER</h3>
<p><a href="https://xsun1229.github.io/factor_analysis/train_peer.html">Results for PEER factors</a></p>
</div>
</div>
<div id="training-models-with-cis-eqtls" class="section level2">
<h2><strong>Training models with cis-eQTLs</strong></h2>
<div id="plier-3" class="section level3">
<h3>PLIER</h3>
<p><a href="https://xsun1229.github.io/factor_analysis/train_plier_canon_ciseqtl.html">Results for plier-canonicalPathways factors</a></p>
</div>
</div>
<div id="making-the-prediction-models-better" class="section level2">
<h2><strong>Making the prediction models better</strong></h2>
<div id="finding-functional-snps" class="section level3">
<h3><strong><em>Finding functional SNPs</em></strong></h3>
<p>In step 6, the GWAS power of some factors is low. So I tried to find functional SNPs to increase the GWAS power and reduce the testing burden.</p>
</div>
<div id="ld-score-regression" class="section level3">
<h3>LD Score Regression</h3>
<p>I adopted LD score regression to estimate SNP-based heritability and partation the heritability into seperate categories. The software I used here is <a href="https://github.com/bulik/ldsc">ldsc</a>.</p>
<p>The ldsc material and methods of my project can be found <a href="https://xsun1229.github.io/factor_analysis/ldsc.html">here</a></p>
<div id="plier-4" class="section level4">
<h4>PLIER</h4>
<p><a href="https://xsun1229.github.io/factor_analysis/ldsc_plier_canon.html">Results for plier-canonicalPathways factors</a></p>
<p><a href="https://xsun1229.github.io/factor_analysis/ldsc_plier_all.html">Results for plier-allpathway factors</a></p>
</div>
<div id="flash-3" class="section level4">
<h4>FLASH</h4>
<p><a href="https://xsun1229.github.io/factor_analysis/ldsc_flash.html">Results for FLASH factors</a></p>
</div>
<div id="peer-3" class="section level4">
<h4>PEER</h4>
<p><a href="https://xsun1229.github.io/factor_analysis/ldsc_peer.html">Results for PEER factors</a></p>
</div>
</div>
<div id="using-catalog-gwas-data" class="section level3">
<h3>Using catalog GWAS data</h3>
<p>To address issues with respect to computational time and multiple testing, I confined our analysis to those SNPs present in the <a href="https://www.ebi.ac.uk/gwas/">Catalog of Published GWAS</a>. I selected the SNPs with fdr &gt; 0.01. Then, I did GWAS analysis for the plier_canonical factors with the SNPs.</p>
<p><a href="https://xsun1229.github.io/factor_analysis/catalog_gwas_pliercanon.html">For more details</a></p>
</div>
<div id="finding-functional-cis-genes" class="section level3">
<h3><strong><em>Finding functional cis-genes</em></strong></h3>
<p>After preliminary <a href="https://xsun1229.github.io/factor_analysis/gwas_power.html">rough calculations</a>, we found that there might be overfitting issue. So we tried one/two sample settings to test and eliminate the overfitting issue.</p>
<p>The rought process are below, it differs slightly in different settings.</p>
<ol style="list-style-type: decimal">
<li><p>Train weights from expression matrix and genotype matrix. Variants are cis-snps (±500kb from the TSS of genes)</p></li>
<li><p>Impute the cis genetic component part of gene expression (cis-genes):</p>
<p>cis-genes = weights * genotype of cis-genes</p></li>
<li><p>Association test: factors ~ cis-genes + 10 genotype pcs. Record the p-values for each gene.</p></li>
<li><p>Count the number of significant associations at various p-value threshold for each factor.</p></li>
</ol>
<p><a href="https://xsun1229.github.io/factor_analysis/cg-onesample.html">One Sample Setting</a></p>
<p><a href="https://xsun1229.github.io/factor_analysis/cg-twosample.html">Two Sample Setting</a></p>
<p><a href="https://xsun1229.github.io/factor_analysis/cg-onesample-ciseqtls.html">One Sample Setting- only cis-eQTLs</a></p>
</div>
<div id="dealing-with-weak-iv-bias" class="section level3">
<h3><strong><em>Dealing with weak IV bias</em></strong></h3>
<p>Apart from using external weights(<a href="https://xsun1229.github.io/factor_analysis/cg-twosample.html">Two Sample Setting</a>), I tried other methods to eliminate the weak IV bias issue (divide the cis-eQTLS into different groups according to the p-values).</p>
<p><a href="https://xsun1229.github.io/factor_analysis/weak_bias.html">For more details</a></p>
</div>
</div>
</div>
<div id="module-based-factor-analysis" class="section level1">
<h1><strong>Module-based factor analysis</strong></h1>
<div id="module-construction---wgcna" class="section level2">
<h2>Module construction - WGCNA</h2>
<p>I used WGCNA to find modules in GTEx expression data. These modules will be used in subsequent analysis. <a href="https://xsun1229.github.io/factor_analysis/wgcna_module.html">For more details</a></p>
</div>
<div id="factor-analysis" class="section level2">
<h2>Factor analysis</h2>
<p>When employing CanonicalPathway to run PLIER, only the first 3 modules can be used in the software and get the factors. Others cannot be used correctly in PLIER because the datasets are small. The table below shows the number of genes and factors in module.</p>
<table>
<thead>
<tr class="header">
<th>Module</th>
<th>1</th>
<th>2</th>
<th>3</th>
<th>4</th>
<th>5</th>
<th>6</th>
<th>7</th>
<th>8</th>
<th>9</th>
<th>10</th>
<th>11</th>
<th>12</th>
<th>13</th>
<th>14</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>genes per module</td>
<td>1888</td>
<td>774</td>
<td>431</td>
<td>328</td>
<td>245</td>
<td>226</td>
<td>204</td>
<td>156</td>
<td>114</td>
<td>103</td>
<td>58</td>
<td>36</td>
<td>34</td>
<td>33</td>
</tr>
<tr class="even">
<td>factors per module</td>
<td>120</td>
<td>114</td>
<td>118</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
</tr>
</tbody>
</table>
<p>The websites below contain ‘factor construction’, ‘factor evaluation’, ‘prediction models’ and ‘ldsc’ parts.</p>
<ul>
<li><p><a href="https://xsun1229.github.io/factor_analysis/modu1_factor_pliercanon.html">Factors from Module 1</a></p></li>
<li><p><a href="https://xsun1229.github.io/factor_analysis/modu2_factor_pliercanon.html">Factors from Module 2</a></p></li>
<li><p><a href="https://xsun1229.github.io/factor_analysis/modu3_factor_pliercanon.html">Factors from Module 3</a></p></li>
</ul>
</div>
</div>
<div id="case-studies" class="section level1">
<h1><strong>Case Studies</strong></h1>
<div id="idin" class="section level2">
<h2>IDIN</h2>
<p>In <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3241411/">Bayesian detection of expression quantitative trait [Genetics, 2011]</a>, authors used HESS to analyze the larger IDIN(IRF7-driven inflammatory network) and found two SNPs, <a href="https://www.ncbi.nlm.nih.gov/snp/rs9557207">rs9557207</a> and <a href="https://www.ncbi.nlm.nih.gov/snp/rs11616269">rs11616269</a>,which are detected as hotspots for the IDIN expression.</p>
<p>I extracted sevral factors from IDIN gene expression matrix (GTEx whole blood data)/part of IDIN gene expression matrix and performed factor analysis to find if any factor associates with the two SNPs.</p>
<ul>
<li><p><a href="https://xsun1229.github.io/factor_analysis/idin_plier_allpathway.html">PLIER factors</a></p></li>
<li><p><a href="https://xsun1229.github.io/factor_analysis/idin_flash.html">FLASH factors</a> contains association test for:</p>
<p>Factors ~ rs11616269/rs9557207.</p>
<p>Expression data of SNP associated genes ~ SNPs</p>
<p>Factors ~ EBI2 expression</p></li>
</ul>
</div>
<div id="t1d-pc5" class="section level2">
<h2>T1D-PC5</h2>
<p>From <a href="https://xsun1229.github.io/factor_analysis/eval_kegg.html">PC&amp;PEER from KEGG pathways</a>, I found that the 5th PC of Type I diabetes mellitus pathway has the heritability over 0.99. So I did further analysis for this factor. <a href="https://xsun1229.github.io/factor_analysis/T1D.html">For more details</a></p>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
