<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="XSun" />

<meta name="date" content="2021-02-19" />

<title>Understanding genetics of expression factors</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">factor_analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/xsun1229/factor_analysis">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Understanding genetics of expression factors</h1>
<h4 class="author">XSun</h4>
<h4 class="date">2021-02-19</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-07-17
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 2 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>factor_analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomxsun1229factoranalysistreec6bd0e4afc6bc2db2838e3f4b6de4da34248929btargetblankc6bd0e4a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/xsun1229/factor_analysis/tree/c6bd0e4afc6bc2db2838e3f4b6de4da34248929b" target="_blank">c6bd0e4</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomxsun1229factoranalysistreec6bd0e4afc6bc2db2838e3f4b6de4da34248929btargetblankc6bd0e4a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/xsun1229/factor_analysis/tree/c6bd0e4afc6bc2db2838e3f4b6de4da34248929b" target="_blank">c6bd0e4</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    analysis/.Rhistory
    Ignored:    output/.Rhistory

Untracked files:
    Untracked:  analysis/delete_jul.Rmd

Unstaged changes:
    Deleted:    analysis/thesis.Rmd.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/variants_factors.Rmd</code>) and HTML (<code>docs/variants_factors.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/c6bd0e4afc6bc2db2838e3f4b6de4da34248929b/analysis/variants_factors.Rmd" target="_blank">c6bd0e4</a>
</td>
<td>
XSun
</td>
<td>
2021-07-17
</td>
<td>
wflow_publish(c(“analysis/CEDAR_data_process.Rmd”, “analysis/EM_websites.Rmd”,
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/7350b75d7096a1a0daadb61606cdee8f20935edf/docs/variants_factors.html" target="_blank">7350b75</a>
</td>
<td>
XSun
</td>
<td>
2021-02-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/88051cf0b1fe50134742018c598e73e1b31a1ab8/analysis/variants_factors.Rmd" target="_blank">88051cf</a>
</td>
<td>
XSun
</td>
<td>
2021-02-19
</td>
<td>
wflow_publish(“analysis/variants_factors.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis/fa63c4513463c7b592b6506b8af36f93e7a2b5a0/docs/variants_factors.html" target="_blank">fa63c45</a>
</td>
<td>
XSun
</td>
<td>
2021-02-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis/blob/1907e0e3e63e25edc3db228518a33845a7c237d6/analysis/variants_factors.Rmd" target="_blank">1907e0e</a>
</td>
<td>
XSun
</td>
<td>
2021-02-19
</td>
<td>
wflow_publish(c(“analysis/index.Rmd”, “analysis/variants_factors.Rmd”,
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level1">
<h1><strong>1. Introduction</strong></h1>
<p>In this part, we’re trying to answer the questions below:</p>
<ul>
<li><p>Are expression factors genetically determined? Can we build prediction model of expression factors using genetic variants?</p></li>
<li><p>What are the SNPs and genes controlling expression factors?</p></li>
</ul>
</div>
<div id="data" class="section level1">
<h1><strong>2. Data</strong></h1>
<ul>
<li><p>Expression data: GTEx whole blood expression data which contains 670 samples and 19,696 genes. Genes on chrX were excluded.</p></li>
<li><p>Genotype data: GTEx genotype data(the variants with maf &gt;0.05 and maf &gt;0.01 were selected, variants on chrX were excluded). 6,568,202 variants have maf &gt;0.05 and 10,403,249 variants have maf &gt;0.01.</p></li>
</ul>
</div>
<div id="analyses" class="section level1">
<h1><strong>3. Analyses</strong></h1>
<div id="factor-construction-and-evaluation" class="section level2">
<h2><strong>3.1 Factor Construction and evaluation</strong></h2>
<ul>
<li><p>We used 4 methods to infer expression factors from the expression matrix. The <a href="https://github.com/wgmao/PLIER">PLIER</a>, <a href="https://github.com/stephenslab/flashr">FLASH</a>, and <a href="https://github.com/PMBio/peer/">PEER</a> worked on GTEx gene expression matrix of whole blood(670 samples, 19,696 genes). PCA worked on KEGG 332 pathways.</p></li>
<li><p>We evaluated the heritability of factors using <a href="https://cnsgenomics.com/software/gcta/#GREML">GCTA</a> and made histogrames to show the distribution of the heritabilities.</p></li>
</ul>
<p><strong>PLIER</strong></p>
<p><a href="https://xsun1229.github.io/factor_analysis/factor_plier_canon.html">plier - canonicalPathways - factor consruction</a></p>
<p><a href="https://xsun1229.github.io/factor_analysis/eval_plier_canon.html">plier - canonicalPathways - factor evaluation</a></p>
<p><a href="https://xsun1229.github.io/factor_analysis/factor_plier_allpathway.html">plier - allpathways - factor evaluation</a></p>
<p><a href="https://xsun1229.github.io/factor_analysis/eval_plier_allpathway.html">plier - allpathways - factor evaluation</a></p>
<p><strong>FLASH</strong></p>
<p>Using ‘FLASH’, we obtained 100 factors.</p>
<p><a href="https://xsun1229.github.io/factor_analysis/eval_flash.html">FLASH - factor evaluation</a></p>
<p><strong>PEER</strong></p>
<p>Using ‘PEER’, We obtained 60 factors.</p>
<p><a href="https://xsun1229.github.io/factor_analysis/eval_peer.html">PEER - factor evaluation</a></p>
<p><strong>PC&amp;PEER from KEGG pathways</strong></p>
<p>I used PCA to infer 1-5 PCs for each pathway. We have 1646 PCs in total.</p>
<p><a href="https://xsun1229.github.io/factor_analysis/eval_kegg.html">For more details</a></p>
</div>
<div id="prediction-models-for-factors" class="section level2">
<h2><strong>3.2 Prediction models for factors</strong></h2>
<p>I used two groups of variants to train models. The first group contains all variants have maf &gt; 0.01(10,403,249 variants) and maf &gt; 0.05(6,568,202 variants). The second group contains all significant cis-eQTLs provided by GTEx (FDR &gt; 0.05).</p>
<p>The workflow:</p>
<ol style="list-style-type: decimal">
<li>I splited the data set into training(80%, 536 samples) and test(20%, 134 samples) set.</li>
<li><p>In the training data, performed association test between the factors and all SNPs using Plink. I corrected the results for genotype PCs (We used Plink to compute 10 PCs and used them as corvariates in association test).</p></li>
<li><p>From 3, we selected all SNPs with p &lt; 1e-02 as candidate variants.</p></li>
<li><p>Trained Lasso/Elastic-net models using SNPs from step4 in the training data and evaluated the performance in the testing data.</p></li>
<li><p>For factors with relatively low pval_test_l, I did GWAS analysis using all GTEx samples(670 samples). In the association tests, first 10 genotype pcs were included as covariants.</p></li>
</ol>
<div id="training-models-with-all-variants" class="section level3">
<h3>3.2.1 Training models with all variants</h3>
<p>There are some missing values in our dataset. I used mean imputation (substituing the missing values with mean value of the corresponding variants) to impute the missing value.</p>
<figure class="half">
<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/ms_snp.jpeg" width="50%">
</figure>
<p>The missing data pattern of our dataset. x-axis represents the proportion of missing value for each variant. y-axis represents the number of the variants.</p>
<p><strong>PLIER</strong></p>
<p><a href="https://xsun1229.github.io/factor_analysis/train_plier_canon.html">Results for plier-canonicalPathways factors</a></p>
<p><a href="https://xsun1229.github.io/factor_analysis/train_plier_allpathway.html">Results for plier-allpathway factors</a></p>
<p><a href="https://xsun1229.github.io/factor_analysis/train_98lvs.html">Results for plier- MSigDB pathway factors</a></p>
<p><strong>FLASH</strong></p>
<p><a href="https://xsun1229.github.io/factor_analysis/train_flash.html">Results for FLASH factors</a></p>
<p><strong>PEER</strong></p>
<p><a href="https://xsun1229.github.io/factor_analysis/train_peer.html">Results for PEER factors</a></p>
</div>
<div id="training-models-with-cis-eqtls" class="section level3">
<h3>3.2.2 Training models with cis-eQTLs</h3>
<p><strong>PLIER</strong></p>
<p><a href="https://xsun1229.github.io/factor_analysis/train_plier_canon_ciseqtl.html">Results for plier-canonicalPathways factors</a></p>
</div>
<div id="trying-to-make-the-prediction-models-better" class="section level3">
<h3>3.2.3 Trying to make the prediction models better</h3>
<div id="finding-functional-snps---ld-score-regression" class="section level4">
<h4>Finding functional SNPs - LD Score Regression</h4>
<p>The GWAS power of some factors is low. So we tried to find functional SNPs to increase the GWAS power and reduce the testing burden.</p>
<p>We adopted LD score regression to estimate SNP-based heritability and partation the heritability into seperate categories. The software we used here is <a href="https://github.com/bulik/ldsc">ldsc</a>.</p>
<p>The ldsc material and methods of my project can be found <a href="https://xsun1229.github.io/factor_analysis/ldsc.html">here</a></p>
<p><strong>PLIER</strong></p>
<p><a href="https://xsun1229.github.io/factor_analysis/ldsc_plier_canon.html">Results for plier-canonicalPathways factors</a></p>
<p><a href="https://xsun1229.github.io/factor_analysis/ldsc_plier_all.html">Results for plier-allpathway factors</a></p>
<p><strong>FLASH</strong></p>
<p><a href="https://xsun1229.github.io/factor_analysis/ldsc_flash.html">Results for FLASH factors</a></p>
<p><strong>PEER</strong></p>
<p><a href="https://xsun1229.github.io/factor_analysis/ldsc_peer.html">Results for PEER factors</a></p>
</div>
<div id="finding-functional-cis-genes" class="section level4">
<h4>Finding functional cis-genes</h4>
<p>After preliminary <a href="https://xsun1229.github.io/factor_analysis/gwas_power.html">rough calculations</a>, we found that there might be overfitting issue. So we tried one/two sample settings to test and eliminate the overfitting issue.</p>
<p>The rought process are below, it differs slightly in different settings.</p>
<ol style="list-style-type: decimal">
<li><p>Train weights from expression matrix and genotype matrix. Variants are cis-snps (±500kb from the TSS of genes)</p></li>
<li><p>Impute the cis genetic component part of gene expression (cis-genes):</p>
<p>cis-genes = weights * genotype of cis-genes</p></li>
<li><p>Association test: factors ~ cis-genes + 10 genotype pcs. Record the p-values for each gene.</p></li>
<li><p>Count the number of significant associations at various p-value threshold for each factor.</p></li>
</ol>
<p><a href="https://xsun1229.github.io/factor_analysis/cg-onesample.html">One Sample Setting</a></p>
<p><a href="https://xsun1229.github.io/factor_analysis/cg-twosample.html">Two Sample Setting</a></p>
<p><a href="https://xsun1229.github.io/factor_analysis/cg-onesample-ciseqtls.html">One Sample Setting- only cis-eQTLs</a></p>
</div>
<div id="dealing-with-weak-iv-bias" class="section level4">
<h4>Dealing with weak IV bias</h4>
<p>Apart from using external weights(<a href="https://xsun1229.github.io/factor_analysis/cg-twosample.html">Two Sample Setting</a>), I tried other methods to eliminate the weak IV bias issue (divide the cis-eQTLS into different groups according to the p-values).</p>
<p><a href="https://xsun1229.github.io/factor_analysis/weak_bias.html">For more details</a></p>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
