---
title: "ACAT analysis for random SNPs"
author: "XSun"
date: "2022-07-26"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---
```{r, echo = FALSE}
suppressMessages(library(ggplot2))
suppressMessages(library(gridExtra))
suppressMessages(library(ACAT))
suppressMessages(library(data.table))
```


# Introduction

It was shown that ACAT p-values deviate significantly from null. One concern is that the GWAS SNPs may be somewhat different from other SNPs, so the results may be due to chance. It would be useful to repeat the analysis with random SNPs that match GWAS SNPs for allele frequency and LD.


# Original ACAT results

```{r echo=FALSE, message=FALSE, warning=FALSE, paged.print=FALSE,fig.height=10, fig.width=15}
setwd("/project2/xinhe/xsun/CEDAR/4.B_cell_kegg/7.geneset40_peer")

type <- c("B cell (CD19+)","Monocyte (CD14+)","Leukocyte (CD15+)","Platelet","T cell (CD4+)","T cell (CD8+)")
path <- c("4.B_cell_kegg","6.CD14_positive_monocyte_kegg","8.CD15_positive_leukocyte_kegg","16.platelet_kegg","10.T_cell_kegg","14.thymocyte_kegg")

p <- list()
for (i in 1:length(type)){
  
  file <- paste0("/project2/xinhe/xsun/CEDAR/",path[i],"/7.geneset40_peer/acat.rdata")
  load(file)
  
  plotdata <- data.frame(observed = -log10(sort(as.numeric(as.character(list$pval_acat)))),
                         expected = -log10(ppoints(nrow(list))))
  #  
  
  p[[i]] <- ggplot(plotdata) + theme_bw(base_line_size =0.3) +
    geom_point(aes(expected, observed), shape = 1, size = 1.5, color= "steelblue") +
    geom_abline(intercept = 0, slope = 1, alpha = 0.6, color = "red") +
    ggtitle(type[i]) + theme(plot.title = element_text(hjust = 0.5)) +
    labs(x = expression(paste("Expected -log"[10],"(p-value)")),
         y = expression(paste("Observed -log"[10],"(p-value)"))) +
    
    theme(axis.title.x = element_text(size = 10),
          axis.text.x = element_text(size = 8, color = "black"),
          axis.title.y = element_text(size = 10),
          axis.text.y = element_text(size = 8, color = "black"),
          text= element_text(family="Times New Roman")) 
  
  
}

all <- grid.arrange(p[[1]],p[[2]],p[[3]],p[[4]],p[[5]],p[[6]], nrow = 2)
```

# Random ACAT results

We did the analysis follows:

1. Find random SNPs that match GWAS SNPs for allele frequency and LD using [vSampler](http://www.mulinlab.org/vsampler/index.html)

2. For each SNP, do association test PEER ~ SNP, select the PEER factors do not associate with SNP as covariates.
For each expression factor, fo association test factor ~ SNP + SNP-specific covariates.

3. Aggregate p-values using ACAT.

4. Make QQ plot for each cell type.  

```{r echo=FALSE, message=FALSE, warning=FALSE, paged.print=FALSE,fig.height=10, fig.width=15}

load("/project2/xinhe/xsun/CEDAR/ACAT_randomsnp/ACAT_results.rdata")

type <- c("B cell (CD19+)","Monocyte (CD14+)","Leukocyte (CD15+)","Platelet","T cell (CD4+)","T cell (CD8+)")

plot <- list()
for (i in 1:length(type)){
  
  p <- acat_allcelltypes[[i]]
  p[is.na(p)] <- 1
  plotdata <- data.frame(observed = -log10(sort(p)),
                         expected = -log10(ppoints(length(p))))
  #  
  
  plot[[i]] <- ggplot(plotdata) + theme_bw(base_line_size =0.3) +
    geom_point(aes(expected, observed), shape = 1, size = 1.5,color = "steelblue") +
    geom_abline(intercept = 0, slope = 1, alpha = 0.6, color = "red") +
    ggtitle(type[i]) + theme(plot.title = element_text(hjust = 0.5)) +
    labs(x = expression(paste("Expected -log"[10],"(p-value)")),
         y = expression(paste("Observed -log"[10],"(p-value)"))) +
    
    theme(axis.title.x = element_text(size = 10),
          axis.text.x = element_text(size = 8, color = "black"),
          axis.title.y = element_text(size = 10),
          axis.text.y = element_text(size = 8, color = "black"),
          text= element_text(family="Times New Roman")) 
  
  
}
all <- grid.arrange(plot[[1]],plot[[2]],plot[[3]],plot[[4]],plot[[5]],plot[[6]], nrow = 2)

```

The QQ plots shift to y-axis and show strong signal, which is not expected for random SNPs. This could be due to: (1) Even random SNPs tend to be associated with the expression factors. (2) There is really no such trend, but ACAT gave significant p-values, in other words, the ACAT p-values are not calibrated.

# Results checking

Take random SNPs matching a GWAS trait, say asthma, or IBD, or some blood cell traits. We have p-values of these SNPs with all factors. We show the QQ plots of these p-values. 

Then we chose a pair of trait-factors that have very strong ACAT signal and made QQ plot the p-values from factor ~ SNP association tests. 

```{r echo=FALSE, message=FALSE, warning=FALSE}
load("/project2/xinhe/xsun/CEDAR/ACAT_randomsnp/ACAT_results.rdata")
type <- c("B cell (CD19+)","Monocyte (CD14+)","Leukocyte (CD15+)","Platelet","T cell (CD4+)","T cell (CD8+)")
```


## For all traits

Several immune diseases and cell type trait can have strong signals (IBD,rbc,asthma, see below) while others not. 

[All plots are here](https://photos.app.goo.gl/ouXA6kiRXWu2NCio8)

## EUR.IBD

```{r echo=FALSE, message=FALSE, warning=FALSE, paged.print=FALSE,fig.height=6, fig.width=9}

trait <- "EUR.IBD"
plot <- list()
for (i in 1:length(acat_allcelltypes)){
  
  acat_cell <- acat_allcelltypes[[i]]
  index <- grep(trait,names(acat_cell))  
  p <- acat_cell[index]
  p[is.na(p)] <- 1
  plotdata <- data.frame(observed = -log10(sort(p)),
                         expected = -log10(ppoints(length(p))))
  #  
  
  plot[[i]] <- ggplot(plotdata) + theme_bw(base_line_size =0.3) +
    geom_point(aes(expected, observed), shape = 1, size = 1.5,color = "steelblue") +
    geom_abline(intercept = 0, slope = 1, alpha = 0.6, color = "red") +
    ggtitle(type[i]) + theme(plot.title = element_text(hjust = 0.5)) +
    labs(x = expression(paste("Expected -log"[10],"(p-value)")),
         y = expression(paste("Observed -log"[10],"(p-value)"))) +
    
    theme(axis.title.x = element_text(size = 10),
          axis.text.x = element_text(size = 8, color = "black"),
          axis.title.y = element_text(size = 10),
          axis.text.y = element_text(size = 8, color = "black"),
          text= element_text(family="Times New Roman")) 
  
}

all <- grid.arrange(plot[[1]],plot[[2]],plot[[3]],plot[[4]],plot[[5]],plot[[6]], nrow = 2)

```

### pair 1 from IBD trait
From B cell dataset, pair: pwy203_pc4 - EUR.IBD, ACAT p-value:2.57244e-57

```{r echo=FALSE, message=FALSE, warning=FALSE, paged.print=FALSE,fig.height=4, fig.width=4}
load("~/scratch-midway2/CEDAR/ACAT_randomsnp/assoc/B_cell/EUR.IBD/pwy203_pc4/pwy203_pc4.assoc.rdata")

plotdata <- data.frame(observed = -log10(sort(as.numeric(as.character(results_linear$P)))),
                       expected = -log10(ppoints(nrow(results_linear))))


ggplot(plotdata) + theme_bw(base_line_size =0.3) +
  geom_point(aes(expected, observed), shape = 1, size = 1.5,color = "steelblue") +
  geom_abline(intercept = 0, slope = 1, alpha = 0.6, color = "red") +
  theme(plot.title = element_text(hjust = 0.5)) +
  labs(x = expression(paste("Expected -log"[10],"(p-value)")),
       y = expression(paste("Observed -log"[10],"(p-value)"))) +
  
  theme(axis.title.x = element_text(size = 10),
        axis.text.x = element_text(size = 8, color = "black"),
        axis.title.y = element_text(size = 10),
        axis.text.y = element_text(size = 8, color = "black"),
        text= element_text(family="Times New Roman")) 

```

top snps
```{r echo=T}
index <- which.min(as.numeric(as.character(results_linear$P)))
results_linear[index,]
```
```{r echo=F}
load("~/scratch-midway2/CEDAR/ACAT_randomsnp/genoforvsampler/B_cell/EUR.IBD.rdata")
snp <- results_linear[index,]$SNP
index_origin <- which(final_map$rs_vs == snp)
```

The origin snp:
```{r echo=T}
final_map[index_origin,]
index_origin <- which(final_map$rs_vs == snp)
```


QQ plot for the real data.

```{r echo=FALSE, message=FALSE, warning=FALSE, paged.print=FALSE,fig.height=4, fig.width=4}
load("~/xsun_xin/CEDAR/4.B_cell_kegg/7.geneset40_peer/gwas_sum/EUR.IBD.rdata")
results_linear <- gwas_sum[["pwy203_pc4"]]

plotdata <- data.frame(observed = -log10(sort(as.numeric(as.character(results_linear$P)))),
                       expected = -log10(ppoints(nrow(results_linear))))


ggplot(plotdata) + theme_bw(base_line_size =0.3) +
  geom_point(aes(expected, observed), shape = 1, size = 1.5,color = "steelblue") +
  geom_abline(intercept = 0, slope = 1, alpha = 0.6, color = "red") +
  theme(plot.title = element_text(hjust = 0.5)) +
  labs(x = expression(paste("Expected -log"[10],"(p-value)")),
       y = expression(paste("Observed -log"[10],"(p-value)"))) +
  
  theme(axis.title.x = element_text(size = 10),
        axis.text.x = element_text(size = 8, color = "black"),
        axis.title.y = element_text(size = 10),
        axis.text.y = element_text(size = 8, color = "black"),
        text= element_text(family="Times New Roman")) 

```
```{r echo=T}
p_acat <- ACAT(as.numeric(as.character(results_linear$P)))
p_acat
```


### Pair 2 from IBD trait

From CD14_positive_monocyte, pair: pwy312_pc3 - EUR.IBD, ACAT p-value:1.604200e-59
```{r echo=FALSE, message=FALSE, warning=FALSE, paged.print=FALSE,fig.height=4, fig.width=4}
load("~/scratch-midway2/CEDAR/ACAT_randomsnp/assoc/CD14_positive_monocyte/EUR.IBD/pwy312_pc3/pwy312_pc3.assoc.rdata")

plotdata <- data.frame(observed = -log10(sort(as.numeric(as.character(results_linear$P)))),
                       expected = -log10(ppoints(nrow(results_linear))))


ggplot(plotdata) + theme_bw(base_line_size =0.3) +
  geom_point(aes(expected, observed), shape = 1, size = 1.5,color = "steelblue") +
  geom_abline(intercept = 0, slope = 1, alpha = 0.6, color = "red") +
  theme(plot.title = element_text(hjust = 0.5)) +
  labs(x = expression(paste("Expected -log"[10],"(p-value)")),
       y = expression(paste("Observed -log"[10],"(p-value)"))) +
  
  theme(axis.title.x = element_text(size = 10),
        axis.text.x = element_text(size = 8, color = "black"),
        axis.title.y = element_text(size = 10),
        axis.text.y = element_text(size = 8, color = "black"),
        text= element_text(family="Times New Roman")) 

```

top snps
```{r echo=T}
index <- which.min(as.numeric(as.character(results_linear$P)))
results_linear[index,]
```

```{r echo=F}
load("~/scratch-midway2/CEDAR/ACAT_randomsnp/genoforvsampler/CD14_positive_monocyte/EUR.IBD.rdata")
snp <- results_linear[index,]$SNP
index_origin <- which(final_map$rs_vs == snp)
```

The origin snp:
```{r echo=T}
final_map[index_origin,]
```

QQ plot for the real data.

```{r echo=FALSE, message=FALSE, warning=FALSE, paged.print=FALSE,fig.height=4, fig.width=4}
load("~/xsun_xin/CEDAR/6.CD14_positive_monocyte_kegg/7.geneset40_peer/gwas_sum/EUR.IBD.rdata")
results_linear <- gwas_sum[["pwy312_pc3"]]

plotdata <- data.frame(observed = -log10(sort(as.numeric(as.character(results_linear$P)))),
                       expected = -log10(ppoints(nrow(results_linear))))


ggplot(plotdata) + theme_bw(base_line_size =0.3) +
  geom_point(aes(expected, observed), shape = 1, size = 1.5,color = "steelblue") +
  geom_abline(intercept = 0, slope = 1, alpha = 0.6, color = "red") +
  theme(plot.title = element_text(hjust = 0.5)) +
  labs(x = expression(paste("Expected -log"[10],"(p-value)")),
       y = expression(paste("Observed -log"[10],"(p-value)"))) +
  
  theme(axis.title.x = element_text(size = 10),
        axis.text.x = element_text(size = 8, color = "black"),
        axis.title.y = element_text(size = 10),
        axis.text.y = element_text(size = 8, color = "black"),
        text= element_text(family="Times New Roman")) 

```

```{r echo=T}
p_acat <- ACAT(as.numeric(as.character(results_linear$P)))
p_acat
```
## rbc

```{r echo=FALSE, message=FALSE, warning=FALSE, paged.print=FALSE,fig.height=6, fig.width=9}

trait <- "rbc"
plot <- list()
for (i in 1:length(acat_allcelltypes)){
  
  acat_cell <- acat_allcelltypes[[i]]
  index <- grep(trait,names(acat_cell))  
  p <- acat_cell[index]
  p[is.na(p)] <- 1
  plotdata <- data.frame(observed = -log10(sort(p)),
                         expected = -log10(ppoints(length(p))))
  #  
  
  plot[[i]] <- ggplot(plotdata) + theme_bw(base_line_size =0.3) +
    geom_point(aes(expected, observed), shape = 1, size = 1.5,color = "steelblue") +
    geom_abline(intercept = 0, slope = 1, alpha = 0.6, color = "red") +
    ggtitle(type[i]) + theme(plot.title = element_text(hjust = 0.5)) +
    labs(x = expression(paste("Expected -log"[10],"(p-value)")),
         y = expression(paste("Observed -log"[10],"(p-value)"))) +
    
    theme(axis.title.x = element_text(size = 10),
          axis.text.x = element_text(size = 8, color = "black"),
          axis.title.y = element_text(size = 10),
          axis.text.y = element_text(size = 8, color = "black"),
          text= element_text(family="Times New Roman")) 
  
}

all <- grid.arrange(plot[[1]],plot[[2]],plot[[3]],plot[[4]],plot[[5]],plot[[6]], nrow = 2)

```


### Pair 1 from rbc

From T cell (CD8+) dataset, pair: pwy311_pc5 - rbc, ACAT p-value:2.637203e-11
```{r echo=FALSE, message=FALSE, warning=FALSE, paged.print=FALSE,fig.height=4, fig.width=4}
load("~/scratch-midway2/CEDAR/ACAT_randomsnp/assoc/thymocyte/rbc/pwy311_pc5/pwy311_pc5.assoc.rdata")

plotdata <- data.frame(observed = -log10(sort(as.numeric(as.character(results_linear$P)))),
                       expected = -log10(ppoints(nrow(results_linear))))


ggplot(plotdata) + theme_bw(base_line_size =0.3) +
  geom_point(aes(expected, observed), shape = 1, size = 1.5,color = "steelblue") +
  geom_abline(intercept = 0, slope = 1, alpha = 0.6, color = "red") +
  theme(plot.title = element_text(hjust = 0.5)) +
  labs(x = expression(paste("Expected -log"[10],"(p-value)")),
       y = expression(paste("Observed -log"[10],"(p-value)"))) +
  
  theme(axis.title.x = element_text(size = 10),
        axis.text.x = element_text(size = 8, color = "black"),
        axis.title.y = element_text(size = 10),
        axis.text.y = element_text(size = 8, color = "black"),
        text= element_text(family="Times New Roman")) 

```

top snps
```{r echo=T}
index1 <- which.min(as.numeric(as.character(results_linear$P)))
results_linear[index1,]
index2 <- which(as.numeric(as.character(results_linear$P)) == sort(as.numeric(as.character(results_linear$P)))[2])
results_linear[index2,]
```

```{r echo=F}
load("~/scratch-midway2/CEDAR/ACAT_randomsnp/genoforvsampler/thymocyte/rbc.rdata")
snp <- results_linear[index1,]$SNP
index_origin1 <- which(final_map$rs_vs == snp)
snp <- results_linear[index2,]$SNP
index_origin2 <- which(final_map$rs_vs == snp)
```

The origin snp:
```{r echo=T}
final_map[index_origin1,]
final_map[index_origin2,]
```

These two origin variants have LD R2 < 0.1 and physical distance > 1Mb

QQ plot for the real data.
```{r echo=FALSE, message=FALSE, warning=FALSE, paged.print=FALSE,fig.height=4, fig.width=4}
load("~/xsun_xin/CEDAR/14.thymocyte_kegg/7.geneset40_peer/gwas_sum/rbc.rdata")
results_linear <- gwas_sum[["pwy311_pc5"]]

plotdata <- data.frame(observed = -log10(sort(as.numeric(as.character(results_linear$P)))),
                       expected = -log10(ppoints(nrow(results_linear))))


ggplot(plotdata) + theme_bw(base_line_size =0.3) +
  geom_point(aes(expected, observed), shape = 1, size = 1.5,color = "steelblue") +
  geom_abline(intercept = 0, slope = 1, alpha = 0.6, color = "red") +
  theme(plot.title = element_text(hjust = 0.5)) +
  labs(x = expression(paste("Expected -log"[10],"(p-value)")),
       y = expression(paste("Observed -log"[10],"(p-value)"))) +
  
  theme(axis.title.x = element_text(size = 10),
        axis.text.x = element_text(size = 8, color = "black"),
        axis.title.y = element_text(size = 10),
        axis.text.y = element_text(size = 8, color = "black"),
        text= element_text(family="Times New Roman")) 

```

```{r echo=T}
p_acat <- ACAT(as.numeric(as.character(results_linear$P)))
p_acat
```


## Allergy

```{r echo=FALSE, message=FALSE, warning=FALSE, paged.print=FALSE,fig.height=6, fig.width=9}

trait <- "allergy"
plot <- list()
for (i in 1:length(acat_allcelltypes)){
  
  acat_cell <- acat_allcelltypes[[i]]
  index <- grep(trait,names(acat_cell))  
  p <- acat_cell[index]
  p[is.na(p)] <- 1
  plotdata <- data.frame(observed = -log10(sort(p)),
                         expected = -log10(ppoints(length(p))))
  #  
  
  plot[[i]] <- ggplot(plotdata) + theme_bw(base_line_size =0.3) +
    geom_point(aes(expected, observed), shape = 1, size = 1.5,color = "steelblue") +
    geom_abline(intercept = 0, slope = 1, alpha = 0.6, color = "red") +
    ggtitle(type[i]) + theme(plot.title = element_text(hjust = 0.5)) +
    labs(x = expression(paste("Expected -log"[10],"(p-value)")),
         y = expression(paste("Observed -log"[10],"(p-value)"))) +
    
    theme(axis.title.x = element_text(size = 10),
          axis.text.x = element_text(size = 8, color = "black"),
          axis.title.y = element_text(size = 10),
          axis.text.y = element_text(size = 8, color = "black"),
          text= element_text(family="Times New Roman")) 
  
}

all <- grid.arrange(plot[[1]],plot[[2]],plot[[3]],plot[[4]],plot[[5]],plot[[6]], nrow = 2)

```

### Pair 1 from allergy

From B cell dataset, pair: pwy236_pc4 - rbc, ACAT p-value:2.1553e-17
```{r echo=FALSE, message=FALSE, warning=FALSE, paged.print=FALSE,fig.height=3, fig.width=3}
load("~/scratch-midway2/CEDAR/ACAT_randomsnp/assoc/B_cell/allergy/pwy236_pc4/pwy236_pc4.assoc.rdata")

plotdata <- data.frame(observed = -log10(sort(as.numeric(as.character(results_linear$P)))),
                       expected = -log10(ppoints(nrow(results_linear))))


ggplot(plotdata) + theme_bw(base_line_size =0.3) +
  geom_point(aes(expected, observed), shape = 1, size = 1.5,color = "steelblue") +
  geom_abline(intercept = 0, slope = 1, alpha = 0.6, color = "red") +
  theme(plot.title = element_text(hjust = 0.5)) +
  labs(x = expression(paste("Expected -log"[10],"(p-value)")),
       y = expression(paste("Observed -log"[10],"(p-value)"))) +
  
  theme(axis.title.x = element_text(size = 10),
        axis.text.x = element_text(size = 8, color = "black"),
        axis.title.y = element_text(size = 10),
        axis.text.y = element_text(size = 8, color = "black"),
        text= element_text(family="Times New Roman")) 

```

top snps

```{r echo=T}
index1 <- which.min(as.numeric(as.character(results_linear$P)))
results_linear[index1,]
index2 <- which(as.numeric(as.character(results_linear$P)) == sort(as.numeric(as.character(results_linear$P)))[2])
results_linear[index2,]
```


```{r echo=F}
load("~/scratch-midway2/CEDAR/ACAT_randomsnp/genoforvsampler/B_cell/allergy.rdata")
snp <- results_linear[index1,]$SNP
index_origin1 <- which(final_map$rs_vs == snp)
snp <- results_linear[index2,]$SNP
index_origin2 <- which(final_map$rs_vs == snp)
```

The origin snp:
```{r echo=T}
final_map[index_origin1,]
final_map[index_origin2,]
```

These two origin variants have LD R2 < 0.1 and physical distance > 1Mb

QQ plot for the real data.
```{r echo=FALSE, message=FALSE, warning=FALSE, paged.print=FALSE,fig.height=4, fig.width=4}
load("~/xsun_xin/CEDAR/4.B_cell_kegg/7.geneset40_peer/gwas_sum/allergy.rdata")
results_linear <- gwas_sum[["pwy236_pc4"]]

plotdata <- data.frame(observed = -log10(sort(as.numeric(as.character(results_linear$P)))),
                       expected = -log10(ppoints(nrow(results_linear))))


ggplot(plotdata) + theme_bw(base_line_size =0.3) +
  geom_point(aes(expected, observed), shape = 1, size = 1.5,color = "steelblue") +
  geom_abline(intercept = 0, slope = 1, alpha = 0.6, color = "red") +
  theme(plot.title = element_text(hjust = 0.5)) +
  labs(x = expression(paste("Expected -log"[10],"(p-value)")),
       y = expression(paste("Observed -log"[10],"(p-value)"))) +
  
  theme(axis.title.x = element_text(size = 10),
        axis.text.x = element_text(size = 8, color = "black"),
        axis.title.y = element_text(size = 10),
        axis.text.y = element_text(size = 8, color = "black"),
        text= element_text(family="Times New Roman")) 

```

```{r echo=T}
p_acat <- ACAT(as.numeric(as.character(results_linear$P)))
p_acat
```


# Comparing ACAT for real data and random data -- MHC removed

We found that the variants in MHC area may lead to the deviation in random ACAT. So we removed the SNPs in MHC area and re-plot the qq plots. 

```{r echo=FALSE, message=FALSE, warning=FALSE, paged.print=FALSE,fig.height=10, fig.width=15}

load("/project2/xinhe/xsun/CEDAR/ACAT_randomsnp/ACAT_results_remove_MHC_random.rdata")
acat_random <- acat_allcelltypes
load("/project2/xinhe/xsun/CEDAR/ACAT_randomsnp/ACAT_results_remove_MHC_realdata.rdata")
acat_real <- acat_allcelltypes

type <- c("B cell (CD19+)","Monocyte (CD14+)","Leukocyte (CD15+)","Platelet","T cell (CD4+)","T cell (CD8+)")


p <- list()
for (i in 1:length(acat_random)){
  
  p_random <- acat_random[[i]]
  p_real <- acat_real[[i]]
  
  if (length(p_random) > length(p_real)) {
    p_real[length(p_real)+1] <- 1
  }
  p_random[is.na(p_random)] <- 1
  
  plotdata <- data.frame(observed_random = -log10(sort(as.numeric(as.character(p_random)))),
                         expected = -log10(ppoints(length(p_random))),
                         observed_real = -log10(sort(as.numeric(as.character(p_real)))) )
  dat.m <- melt(plotdata, id.vars = "expected")
  
  
  p[[i]]<- ggplot(dat.m, aes(expected, value, colour = variable)) + theme_bw(base_line_size =0.3) +
    geom_point(shape = 1, size = 1.5) +
    geom_abline(intercept = 0, slope = 1, alpha = 0.6, color = "red") +
    labs(x = expression(paste("Expected -log"[10],"(p-value)")),
         y = expression(paste("Observed -log"[10],"(p-value)"))) +
    
    theme(axis.title.x = element_text(size = 10),
          axis.text.x = element_text(size = 8, color = "black"),
          axis.title.y = element_text(size = 10),
          axis.text.y = element_text(size = 8, color = "black"),
          text= element_text(family="Times New Roman")) +
    
    scale_colour_manual(values = c("red", "steelblue")) +
    
    ggtitle(type[i]) + theme(plot.title = element_text(hjust = 0.5))
  
}

all <- grid.arrange(p[[1]],p[[2]],p[[3]],p[[4]],p[[5]],p[[6]], nrow = 2)



```

We pooled all cell types to check if the inflations from random data (Leukocyte (CD15+), Platelet, T cell (CD8+)) dominant the whole data set.

```{r echo=FALSE, message=FALSE, warning=FALSE, paged.print=FALSE,fig.height=5, fig.width=6}

load("/project2/xinhe/xsun/CEDAR/ACAT_randomsnp/ACAT_results_remove_MHC_random.rdata")
acat_random <- acat_allcelltypes
load("/project2/xinhe/xsun/CEDAR/ACAT_randomsnp/ACAT_results_remove_MHC_realdata.rdata")
acat_real <- acat_allcelltypes

acat_pool_real <- c()
for(i in 1:length(acat_real)) {
  
  acat_pool_real <- c(acat_pool_real,acat_real[[i]])
  
}

acat_pool_random <- c()
for(i in 1:length(acat_random)) {
  
  acat_pool_random <- c(acat_pool_random,acat_random[[i]])
  
}

if (length(acat_pool_random) > length(acat_pool_real)) {
  acat_pool_real[length(acat_pool_real)+1] <- 1
}
acat_pool_random[is.na(acat_pool_random)] <- 1

plotdata <- data.frame(observed_random = -log10(sort(as.numeric(as.character(acat_pool_random)))),
                       expected = -log10(ppoints(length(acat_pool_random))),
                       observed_real = -log10(sort(as.numeric(as.character(acat_pool_real)))) )
dat.m <- melt(plotdata, id.vars = "expected")

p <- ggplot(dat.m, aes(expected, value, colour = variable)) + theme_bw(base_line_size =0.3) +
  geom_point(shape = 1, size = 1.5) +
  geom_abline(intercept = 0, slope = 1, alpha = 0.6, color = "red") +
  labs(x = expression(paste("Expected -log"[10],"(p-value)")),
       y = expression(paste("Observed -log"[10],"(p-value)"))) +
  
  theme(axis.title.x = element_text(size = 10),
        axis.text.x = element_text(size = 8, color = "black"),
        axis.title.y = element_text(size = 10),
        axis.text.y = element_text(size = 8, color = "black"),
        text= element_text(family="Times New Roman")) +
  
  scale_colour_manual(values = c("red", "steelblue")) +
  
  ggtitle("Pooled data") + theme(plot.title = element_text(hjust = 0.5))

p

```



We noticed that there is some inflation for random data (Leukocyte (CD15+), Platelet, T cell (CD8+)). We compute genomic inflation factors for all cell types. 


$$ chisq = qchisq(1-pvalue,1)$$
$$ lambda = \frac{median(chisq)}{qchisq(0.5,1)} $$
The lambda are 1.103569 1.110509 1.133734 1.179766 1.045502 4.582142 accordingly. 

Then we corrected the p-values with the inflation factor -- lambda (devide the chi-squares by inflation factors accoring to https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3133943/ and compute the p-values):

$$ chisq_{adjusted} = \frac{chisq}{lambda}$$

$$ pvalue_{adjusted} = 1 - pchisq(chisq_{adjusted}) $$

The adjusted p-values are shown in the qqplots below (comparing unadjusted real p-values). 


```{r echo=FALSE, message=FALSE, warning=FALSE, paged.print=FALSE,fig.height=10, fig.width=15}
load("/project2/xinhe/xsun/CEDAR/ACAT_randomsnp/pval_gc_corrected_random_MHCremoved.rdata")
acat_random <- pval_corrected
load("/project2/xinhe/xsun/CEDAR/ACAT_randomsnp/ACAT_results_remove_MHC_realdata.rdata")
acat_real <- acat_allcelltypes

type <- c("B cell (CD19+)","Monocyte (CD14+)","Leukocyte (CD15+)","Platelet","T cell (CD4+)","T cell (CD8+)")


p <- list()
for (i in 1:length(acat_random)){
  
  p_random <- acat_random[[i]]
  p_real <- acat_real[[i]]
  
  if (length(p_random) > length(p_real)) {
    p_real[length(p_real)+1] <- 1
  }
  p_random[is.na(p_random)] <- 1
  
  plotdata <- data.frame(observed_random = -log10(sort(as.numeric(as.character(p_random)))),
                         expected = -log10(ppoints(length(p_random))),
                         observed_real = -log10(sort(as.numeric(as.character(p_real)))) )
  dat.m <- melt(plotdata, id.vars = "expected")
  
  
  p[[i]]<- ggplot(dat.m, aes(expected, value, colour = variable)) + theme_bw(base_line_size =0.3) +
    geom_point(shape = 1, size = 1.5) +
    geom_abline(intercept = 0, slope = 1, alpha = 0.6, color = "red") +
    labs(x = expression(paste("Expected -log"[10],"(p-value)")),
         y = expression(paste("Observed -log"[10],"(p-value)"))) +
    
    theme(axis.title.x = element_text(size = 10),
          axis.text.x = element_text(size = 8, color = "black"),
          axis.title.y = element_text(size = 10),
          axis.text.y = element_text(size = 8, color = "black"),
          text= element_text(family="Times New Roman")) +
    
    scale_colour_manual(values = c("red", "steelblue")) +
    
    ggtitle(type[i]) + theme(plot.title = element_text(hjust = 0.5))
  
}

all <- grid.arrange(p[[1]],p[[2]],p[[3]],p[[4]],p[[5]],p[[6]], nrow = 2)
```


# Top pairs from T cell CD8+ 

## Random ACAT

We check several top pairs (top20) that have lowest p-values. 

```{r echo=FALSE, message=FALSE, warning=FALSE}

load("/project2/xinhe/xsun/CEDAR/ACAT_randomsnp/ACAT_results_remove_MHC_random.rdata")
acat_random <- acat_allcelltypes
acat_tcellcd8 <- acat_random$thymocyte

num_top <- 20
acat_top <- acat_tcellcd8[order(acat_tcellcd8)][1:num_top]

acat_top

index1 <- seq(1,3*length(acat_top),by=3)
index2 <- seq(2,3*length(acat_top),by=3)
index3 <- seq(3,3*length(acat_top),by=3)
traits <- unlist(strsplit(names(acat_top), split = "_"))[index1]
pwy <- unlist(strsplit(names(acat_top), split = "_"))[index2]
pc <- unlist(strsplit(names(acat_top), split = "_"))[index3]

table(traits)
```

12 of 20 are related to trait pdw (platelet distribution width).

## Real ACAT

For comparison, we also checked the top 20 pairs from real data


```{r echo=FALSE, message=FALSE, warning=FALSE}

load("/project2/xinhe/xsun/CEDAR/ACAT_randomsnp/ACAT_results_remove_MHC_realdata.rdata")
acat_real <- acat_allcelltypes
acat_tcellcd8_real <- acat_real$thymocyte

num_top <- 20
acat_top_real <- acat_tcellcd8_real[order(acat_tcellcd8_real)][1:num_top]

acat_top_real


index1 <- seq(1,3*length(acat_top_real),by=3)
index2 <- seq(2,3*length(acat_top_real),by=3)
index3 <- seq(3,3*length(acat_top_real),by=3)
traits_real <- unlist(strsplit(names(acat_top_real), split = "_"))[index1]
pwy_real <- unlist(strsplit(names(acat_top_real), split = "_"))[index2]
pc_real <- unlist(strsplit(names(acat_top_real), split = "_"))[index3]
table(traits_real)
```

The top pairs are not related to pdw.

We removed the pairs related to pdw for both real data and random data, then made the qq-plot:


```{r echo=FALSE, message=FALSE, warning=FALSE, paged.print=FALSE,fig.height=5, fig.width=12}

load("/project2/xinhe/xsun/CEDAR/ACAT_randomsnp/ACAT_results_remove_MHC_random.rdata")
acat_random <- acat_allcelltypes
acat_tcellcd8 <- acat_random$thymocyte

pair_index <- grep("pdw",names(acat_tcellcd8))
acat_tcellcd8_remove <- acat_tcellcd8[-pair_index]
acat_tcellcd8_keep <- acat_tcellcd8[pair_index]
top1_remove <- acat_tcellcd8_remove[order(acat_tcellcd8_remove)][1]


load("/project2/xinhe/xsun/CEDAR/ACAT_randomsnp/ACAT_results_remove_MHC_realdata.rdata")
acat_real <- acat_allcelltypes
acat_tcellcd8_real <- acat_real$thymocyte

pair_index <- grep("pdw",names(acat_tcellcd8_real))
acat_tcellcd8_remove_real <- acat_tcellcd8_real[-pair_index]
acat_tcellcd8_keep_real <- acat_tcellcd8_real[pair_index]
top1_remove_real <- acat_tcellcd8_remove_real[order(acat_tcellcd8_remove_real)][1]

plotdata <- data.frame(observed_random = -log10(sort(acat_tcellcd8_remove)),
                       expected = -log10(ppoints(length(acat_tcellcd8_remove))),
                       observed_real = -log10(sort(acat_tcellcd8_remove_real)) )
dat.m <- melt(plotdata, id.vars = "expected")


p1<- ggplot(dat.m, aes(expected, value, colour = variable)) + theme_bw(base_line_size =0.3) +
  geom_point(shape = 1, size = 1.5) +
  geom_abline(intercept = 0, slope = 1, alpha = 0.6, color = "red") +
  labs(x = expression(paste("Expected -log"[10],"(p-value)")),
       y = expression(paste("Observed -log"[10],"(p-value)"))) +
  
  theme(axis.title.x = element_text(size = 10),
        axis.text.x = element_text(size = 8, color = "black"),
        axis.title.y = element_text(size = 10),
        axis.text.y = element_text(size = 8, color = "black"),
        text= element_text(family="Times New Roman")) +
  
  scale_colour_manual(values = c("red", "steelblue")) +
  ggtitle("pdw removed") + theme(plot.title = element_text(hjust = 0.5))

plotdata <- data.frame(observed_random = -log10(sort(acat_tcellcd8_keep)),
                       expected = -log10(ppoints(length(acat_tcellcd8_keep))),
                       observed_real = -log10(sort(acat_tcellcd8_keep_real)) )
dat.m <- melt(plotdata, id.vars = "expected")

p2<- ggplot(dat.m, aes(expected, value, colour = variable)) + theme_bw(base_line_size =0.3) +
  geom_point(shape = 1, size = 1.5) +
  geom_abline(intercept = 0, slope = 1, alpha = 0.6, color = "red") +
  labs(x = expression(paste("Expected -log"[10],"(p-value)")),
       y = expression(paste("Observed -log"[10],"(p-value)"))) +
  
  theme(axis.title.x = element_text(size = 10),
        axis.text.x = element_text(size = 8, color = "black"),
        axis.title.y = element_text(size = 10),
        axis.text.y = element_text(size = 8, color = "black"),
        text= element_text(family="Times New Roman")) +
  
  scale_colour_manual(values = c("red", "steelblue")) +
  ggtitle("pdw only") + theme(plot.title = element_text(hjust = 0.5))

all <- grid.arrange(p1,p2, nrow = 1) 
all


sprintf("Left: The top1 pair of random data  (top 1 dot in red) is %s ", names(top1_remove))
sprintf("p-value = %s ", top1_remove)

sprintf("Left: The top1 pair of real data (top 1 dot in blue) is %s ", names(top1_remove_real))
sprintf("p-value = %s ", top1_remove_real)


```

# Top pairs from Leukocyte (CD15+)

## Random ACAT

We also check the top 20 pairs from Leukocyte (CD15+). 

```{r echo=FALSE, message=FALSE, warning=FALSE}

load("/project2/xinhe/xsun/CEDAR/ACAT_randomsnp/ACAT_results_remove_MHC_random.rdata")
acat_random <- acat_allcelltypes
acat_leukocyte <- acat_random$CD15_positive_leukocyte

num_top <- 20
acat_top <- acat_leukocyte[order(acat_leukocyte)][1:num_top]

acat_top

index1 <- seq(1,3*length(acat_top),by=3)
index2 <- seq(2,3*length(acat_top),by=3)
index3 <- seq(3,3*length(acat_top),by=3)
traits <- unlist(strsplit(names(acat_top), split = "_"))[index1]
pwy <- unlist(strsplit(names(acat_top), split = "_"))[index2]
pc <- unlist(strsplit(names(acat_top), split = "_"))[index3]

table(traits)
table(pwy)
```

There is not a dominant pathway or trait.

## Real ACAT 
For comparison, we also checked the top 20 pairs from real data


```{r echo=FALSE, message=FALSE, warning=FALSE}

load("/project2/xinhe/xsun/CEDAR/ACAT_randomsnp/ACAT_results_remove_MHC_realdata.rdata")
acat_real <- acat_allcelltypes
acat_leukocyte_real <- acat_real$thymocyte

num_top <- 20
acat_top_real <- acat_leukocyte_real[order(acat_leukocyte_real)][1:num_top]

acat_top_real


index1 <- seq(1,3*length(acat_top_real),by=3)
index2 <- seq(2,3*length(acat_top_real),by=3)
index3 <- seq(3,3*length(acat_top_real),by=3)
traits_real <- unlist(strsplit(names(acat_top_real), split = "_"))[index1]
pwy_real <- unlist(strsplit(names(acat_top_real), split = "_"))[index2]
pc_real <- unlist(strsplit(names(acat_top_real), split = "_"))[index3]
table(traits_real)
```

There are three dominant traits in random data (gran, hct and neut), hct is also the dominant trait in real data. So we removed gran and neut in both random data and real data.

```{r echo=FALSE, message=FALSE, warning=FALSE, paged.print=FALSE,fig.height=5, fig.width=6}

load("/project2/xinhe/xsun/CEDAR/ACAT_randomsnp/ACAT_results_remove_MHC_random.rdata")
acat_random <- acat_allcelltypes
acat_leukocyte <- acat_random$CD15_positive_leukocyte

pair_index <- grep("gran",names(acat_leukocyte))
acat_leukocyte_remove <- acat_leukocyte[-pair_index]
pair_index <- grep("neut",names(acat_leukocyte_remove))
acat_leukocyte_remove <- acat_leukocyte_remove[-pair_index]
top1_remove <- acat_leukocyte_remove[order(acat_leukocyte_remove)][1:3]


load("/project2/xinhe/xsun/CEDAR/ACAT_randomsnp/ACAT_results_remove_MHC_realdata.rdata")
acat_real <- acat_allcelltypes
acat_leukocyte_real <- acat_real$CD15_positive_leukocyte

pair_index <- grep("gran",names(acat_leukocyte_real))
acat_leukocyte_remove_real <- acat_leukocyte_real[-pair_index]
pair_index <- grep("neut",names(acat_leukocyte_remove_real))
acat_leukocyte_remove_real <- acat_leukocyte_remove_real[-pair_index]
top1_remove_real <- acat_leukocyte_remove_real[order(acat_leukocyte_remove_real)][1:3]

plotdata <- data.frame(observed_random = -log10(sort(acat_leukocyte_remove)),
                       expected = -log10(ppoints(length(acat_leukocyte_remove))),
                       observed_real = -log10(sort(acat_leukocyte_remove_real)) )
dat.m <- melt(plotdata, id.vars = "expected")


p<- ggplot(dat.m, aes(expected, value, colour = variable)) + theme_bw(base_line_size =0.3) +
  geom_point(shape = 1, size = 1.5) +
  geom_abline(intercept = 0, slope = 1, alpha = 0.6, color = "red") +
  labs(x = expression(paste("Expected -log"[10],"(p-value)")),
       y = expression(paste("Observed -log"[10],"(p-value)"))) +
  
  theme(axis.title.x = element_text(size = 10),
        axis.text.x = element_text(size = 8, color = "black"),
        axis.title.y = element_text(size = 10),
        axis.text.y = element_text(size = 8, color = "black"),
        text= element_text(family="Times New Roman")) +
  
  scale_colour_manual(values = c("red", "steelblue"))

p

sprintf("The top3 pairs of random data  (top 3 dot in red) is %s ", paste(names(top1_remove),collapse ="," ))
sprintf("p-value = %s ", paste(top1_remove,collapse =","))

sprintf("The top3 pairs of real data (top 1 dot in blue) is %s ", paste(names(top1_remove_real),collapse =","))
sprintf("p-value = %s ", paste(top1_remove_real,collapse =","))

```

If we remove the hct as well, we have: 


```{r echo=FALSE, message=FALSE, warning=FALSE, paged.print=FALSE,fig.height=5, fig.width=6}

load("/project2/xinhe/xsun/CEDAR/ACAT_randomsnp/ACAT_results_remove_MHC_random.rdata")
acat_random <- acat_allcelltypes
acat_leukocyte <- acat_random$CD15_positive_leukocyte

pair_index <- grep("gran",names(acat_leukocyte))
acat_leukocyte_remove <- acat_leukocyte[-pair_index]
acat_leukocyte_remove_gran_keep <- acat_leukocyte[pair_index]
pair_index <- grep("neut",names(acat_leukocyte_remove))
acat_leukocyte_remove <- acat_leukocyte_remove[-pair_index]
acat_leukocyte_remove_neut_keep <- acat_leukocyte_remove[pair_index]
pair_index <- grep("hct",names(acat_leukocyte_remove))
acat_leukocyte_remove <- acat_leukocyte_remove[-pair_index]
acat_leukocyte_remove_hct_keep <-acat_leukocyte_remove[pair_index]
top1_remove <- acat_leukocyte_remove[order(acat_leukocyte_remove)][1]


load("/project2/xinhe/xsun/CEDAR/ACAT_randomsnp/ACAT_results_remove_MHC_realdata.rdata")
acat_real <- acat_allcelltypes
acat_leukocyte_real <- acat_real$CD15_positive_leukocyte

pair_index <- grep("gran",names(acat_leukocyte_real))
acat_leukocyte_remove_real <- acat_leukocyte_real[-pair_index]
acat_leukocyte_remove_real_gran_keep <- acat_leukocyte_real[pair_index]
pair_index <- grep("neut",names(acat_leukocyte_remove_real))
acat_leukocyte_remove_real <- acat_leukocyte_remove_real[-pair_index]
acat_leukocyte_remove_real_neut_keep <- acat_leukocyte_remove_real[pair_index]
pair_index <- grep("hct",names(acat_leukocyte_remove_real))
acat_leukocyte_remove_real <- acat_leukocyte_remove_real[-pair_index]
acat_leukocyte_remove_real_hct_keep <- acat_leukocyte_remove_real[pair_index]
top1_remove_real <- acat_leukocyte_remove_real[order(acat_leukocyte_remove_real)][1]

plotdata <- data.frame(observed_random = -log10(sort(acat_leukocyte_remove)),
                       expected = -log10(ppoints(length(acat_leukocyte_remove))),
                       observed_real = -log10(sort(acat_leukocyte_remove_real)) )
dat.m <- melt(plotdata, id.vars = "expected")


p<- ggplot(dat.m, aes(expected, value, colour = variable)) + theme_bw(base_line_size =0.3) +
  geom_point(shape = 1, size = 1.5) +
  geom_abline(intercept = 0, slope = 1, alpha = 0.6, color = "red") +
  labs(x = expression(paste("Expected -log"[10],"(p-value)")),
       y = expression(paste("Observed -log"[10],"(p-value)"))) +
  
  theme(axis.title.x = element_text(size = 10),
        axis.text.x = element_text(size = 8, color = "black"),
        axis.title.y = element_text(size = 10),
        axis.text.y = element_text(size = 8, color = "black"),
        text= element_text(family="Times New Roman")) +
  
  scale_colour_manual(values = c("red", "steelblue"))

p

sprintf("The top1 pair of random data  (top 1 dot in red) is %s ", names(top1_remove))
sprintf("p-value = %s ", top1_remove)

sprintf("The top1 pair of real data (top 1 dot in blue) is %s ", names(top1_remove_real))
sprintf("p-value = %s ", top1_remove_real)
```


```{r echo=FALSE, message=FALSE, warning=FALSE, paged.print=FALSE,fig.height=5, fig.width=18}
plotdata <- data.frame(observed_random = -log10(sort(acat_leukocyte_remove_gran_keep)),
                       expected = -log10(ppoints(length(acat_leukocyte_remove_gran_keep))),
                       observed_real = -log10(sort(acat_leukocyte_remove_real_gran_keep)) )
dat.m <- melt(plotdata, id.vars = "expected")


p1<- ggplot(dat.m, aes(expected, value, colour = variable)) + theme_bw(base_line_size =0.3) +
  geom_point(shape = 1, size = 1.5) +
  geom_abline(intercept = 0, slope = 1, alpha = 0.6, color = "red") +
  labs(x = expression(paste("Expected -log"[10],"(p-value)")),
       y = expression(paste("Observed -log"[10],"(p-value)"))) +
  
  theme(axis.title.x = element_text(size = 10),
        axis.text.x = element_text(size = 8, color = "black"),
        axis.title.y = element_text(size = 10),
        axis.text.y = element_text(size = 8, color = "black"),
        text= element_text(family="Times New Roman")) +
  
  scale_colour_manual(values = c("red", "steelblue")) +
  ggtitle("gran only") + theme(plot.title = element_text(hjust = 0.5))


plotdata <- data.frame(observed_random = -log10(sort(acat_leukocyte_remove_neut_keep)),
                       expected = -log10(ppoints(length(acat_leukocyte_remove_neut_keep))),
                       observed_real = -log10(sort(acat_leukocyte_remove_real_neut_keep)) )
dat.m <- melt(plotdata, id.vars = "expected")


p2<- ggplot(dat.m, aes(expected, value, colour = variable)) + theme_bw(base_line_size =0.3) +
  geom_point(shape = 1, size = 1.5) +
  geom_abline(intercept = 0, slope = 1, alpha = 0.6, color = "red") +
  labs(x = expression(paste("Expected -log"[10],"(p-value)")),
       y = expression(paste("Observed -log"[10],"(p-value)"))) +
  
  theme(axis.title.x = element_text(size = 10),
        axis.text.x = element_text(size = 8, color = "black"),
        axis.title.y = element_text(size = 10),
        axis.text.y = element_text(size = 8, color = "black"),
        text= element_text(family="Times New Roman")) +
  
  scale_colour_manual(values = c("red", "steelblue")) +
  ggtitle("neut only") + theme(plot.title = element_text(hjust = 0.5))


plotdata <- data.frame(observed_random = -log10(sort(acat_leukocyte_remove_hct_keep)),
                       expected = -log10(ppoints(length(acat_leukocyte_remove_hct_keep))),
                       observed_real = -log10(sort(acat_leukocyte_remove_real_hct_keep)) )
dat.m <- melt(plotdata, id.vars = "expected")


p3<- ggplot(dat.m, aes(expected, value, colour = variable)) + theme_bw(base_line_size =0.3) +
  geom_point(shape = 1, size = 1.5) +
  geom_abline(intercept = 0, slope = 1, alpha = 0.6, color = "red") +
  labs(x = expression(paste("Expected -log"[10],"(p-value)")),
       y = expression(paste("Observed -log"[10],"(p-value)"))) +
  
  theme(axis.title.x = element_text(size = 10),
        axis.text.x = element_text(size = 8, color = "black"),
        axis.title.y = element_text(size = 10),
        axis.text.y = element_text(size = 8, color = "black"),
        text= element_text(family="Times New Roman")) +
  
  scale_colour_manual(values = c("red", "steelblue")) +
  ggtitle("hct only") + theme(plot.title = element_text(hjust = 0.5))

all <- grid.arrange(p1,p2,p3, nrow = 1)
all
```


# Checking the top 20 pairs for all cell types

## Random ACAT:

```{r echo=FALSE, message=FALSE, warning=FALSE}
type <- c("B cell (CD19+)","Monocyte (CD14+)","Leukocyte (CD15+)","Platelet","T cell (CD4+)","T cell (CD8+)")

num_top <- 20


load("/project2/xinhe/xsun/CEDAR/ACAT_randomsnp/ACAT_results_remove_MHC_random.rdata")
acat_random <- acat_allcelltypes

acat_top_all <-c()
for (i in 1:length(acat_random)) {
  
  acat_sep <- acat_random[[i]]
  acat_top <- acat_sep[order(acat_sep)][1:num_top]
  
  acat_top_all <- cbind(acat_top_all, names(acat_top))
  
  traits <- c()
  pwy <- c()
  pc <- c()
  for (j in 1:length(acat_top)) {
    
    split_tmp <- unlist(strsplit(names(acat_top)[j], split = "_"))
    
    if (length(split_tmp) == 3) {
      
      index1 <- 1
      index2 <- 2
      index3 <- 3
      traits[j] <- split_tmp[index1]
      pwy[j] <- split_tmp[index2]
      pc[j] <- split_tmp[index3]
      
    }  
    if (length(split_tmp) == 4) {
      
      index1 <- 1
      index2 <- 2
      index3 <- 3
      index4 <- 4
      traits[j] <- paste0(split_tmp[index1],"_",split_tmp[index2])
      pwy[j] <- split_tmp[index3]
      pc[j] <- split_tmp[index4]
      
    }
    

  }
  
    print(sprintf("The traits included in top pairs, %s ", type[i]))
    print(table(traits))
    print(sprintf("The pathways included in top pairs, %s ", type[i]))
    print(table(pwy))
  
  
}

colnames(acat_top_all) <- type
rownames(acat_top_all) <- seq(1,nrow(acat_top_all))
DT::datatable(acat_top_all, options = list(pageLength =20))
```

## Real ACAT

```{r echo=FALSE, message=FALSE, warning=FALSE}
type <- c("B cell (CD19+)","Monocyte (CD14+)","Leukocyte (CD15+)","Platelet","T cell (CD4+)","T cell (CD8+)")

num_top <- 20


load("/project2/xinhe/xsun/CEDAR/ACAT_randomsnp/ACAT_results_remove_MHC_realdata.rdata")
acat_real <- acat_allcelltypes

acat_top_all <-c()
for (i in 1:length(acat_real)) {
  
  acat_sep <- acat_real[[i]]
  acat_top <- acat_sep[order(acat_sep)][1:num_top]
  
  acat_top_all <- cbind(acat_top_all, names(acat_top))
  
  traits <- c()
  pwy <- c()
  pc <- c()
  for (j in 1:length(acat_top)) {
    
    split_tmp <- unlist(strsplit(names(acat_top)[j], split = "_"))
    
    if (length(split_tmp) == 3) {
      
      index1 <- 1
      index2 <- 2
      index3 <- 3
      traits[j] <- split_tmp[index1]
      pwy[j] <- split_tmp[index2]
      pc[j] <- split_tmp[index3]
      
    }  
    if (length(split_tmp) == 4) {
      
      index1 <- 1
      index2 <- 2
      index3 <- 3
      index4 <- 4
      traits[j] <- paste0(split_tmp[index1],"_",split_tmp[index2])
      pwy[j] <- split_tmp[index3]
      pc[j] <- split_tmp[index4]
      
    }
    
  }
  
  
    print(sprintf("The traits included in top pairs, %s ", type[i]))
    print(table(traits))
    print(sprintf("The pathways included in top pairs, %s ", type[i]))
    print(table(pwy))
  
}

colnames(acat_top_all) <- type
rownames(acat_top_all) <- seq(1,nrow(acat_top_all))
DT::datatable(acat_top_all, options = list(pageLength =20))
```


# Calibrate ACAT p-values by trait.

We can get a conclusion that results are driven mostly by some traits, instead of pathways. So we will correct ACAT p-values using random SNPs from the same trait.

That is, for each trait (trait X), use the random ACAT p-values related to trait X (all factor-trait X pairs) as null distribution, then correct the real ACAT p-values. All calibration will be done within each trait.
In this way, we will not need to make a QQ-plot showing the random ACAT and real ACAT p-values to demonstrate the signal from real ACAT. Showing the QQ-plot of calibrated real ACAT p-values will be enough.

We calibrated p-values according to: 

p_corr[t] <- sum(p_real[t] > p_random)/length(p_random)

For those p_corr == 0, we set p_corr[p_corr == 0] <- 0.5/length(p_random)

```{r echo=FALSE, message=FALSE, warning=FALSE, paged.print=FALSE,fig.height=10, fig.width=18}
load("/project2/xinhe/xsun/CEDAR/ACAT_randomsnp/ACAT_results_remove_MHC_random.rdata")
acat_random <- acat_allcelltypes
load("/project2/xinhe/xsun/CEDAR/ACAT_randomsnp/ACAT_results_remove_MHC_realdata.rdata")
acat_real <- acat_allcelltypes

traits <- c("EUR.UC","EUR.CD","EUR.IBD","mch","mchc","mcv","rdw","ret","baso","plt","pct","pdw","mpv","hct","hgb","mono","eo","gran","lymph","myeloid_wbc","neut","rbc","wbc","ukb.allasthma","allergy","T1D")

acat_corr_all <- list()
for (i in 1:length(acat_real)) {
  
  acat_corr <- c()
  for (j in 1:length(traits)) {
    
    acat_real_celltype <- acat_real[[i]]
    acat_random_celltype <- acat_random[[i]]
    acat_random_celltype[is.na(acat_random_celltype)] <- 1
    
    index_trait_real <- grep(traits[j],names(acat_real_celltype))
    index_trait_random <- grep(traits[j],names(acat_random_celltype))
    
    p_real <- acat_real_celltype[index_trait_real]
    p_random <- acat_random_celltype[index_trait_random]
    
    p_corr_tmp <- c()
    for (t in 1:length(p_real)) {
      
      p_corr_tmp[t] <- sum(p_real[t] > p_random)/length(p_real)
      
    }
    p_corr_tmp[p_corr_tmp == 0] <- 0.5/length(p_real)
    acat_corr <- c(acat_corr,p_corr_tmp)
    
  }
  
  acat_corr_all[[i]] <- acat_corr
  
}

names(acat_corr_all) <- names(acat_real)

##### qqplot

type <- c("B cell (CD19+)","Monocyte (CD14+)","Leukocyte (CD15+)","Platelet","T cell (CD4+)","T cell (CD8+)")
plot <- list()
for (i in 1:length(type)){
  
  p <- acat_corr_all[[i]]

  plotdata <- data.frame(observed = -log10(sort(p)),
                         expected = -log10(ppoints(length(p))))
  #  
  
  plot[[i]] <- ggplot(plotdata) + theme_bw(base_line_size =0.3) +
    geom_point(aes(expected, observed), shape = 1, size = 1.5,color = "steelblue") +
    geom_abline(intercept = 0, slope = 1, alpha = 0.6, color = "red") +
    ggtitle(type[i]) + theme(plot.title = element_text(hjust = 0.5)) +
    labs(x = expression(paste("Expected -log"[10],"(p-value)")),
         y = expression(paste("Observed -log"[10],"(p-value)"))) +
    
    theme(axis.title.x = element_text(size = 10),
          axis.text.x = element_text(size = 8, color = "black"),
          axis.title.y = element_text(size = 10),
          axis.text.y = element_text(size = 8, color = "black"),
          text= element_text(family="Times New Roman")) 
  
  
}
all <- grid.arrange(plot[[1]],plot[[2]],plot[[3]],plot[[4]],plot[[5]],plot[[6]], nrow = 2)
```

There are many pairs have same calibrated p-values, especially for the pairs whose real ACAT p-values are small. The reason is that the number of (p_random < p_real[t]) is small for these pairs. So they are all set to same values. 

# FDR correction for calibrated ACAT p-values

We computed the FDR for the calibrated ACAT p-values.

We computed both by traits and by cell types.

By cell types means: for each cell type, use all calibrated p-values as inputs to compute FDR. 

By traits means, for each cell type, we calibrated the p-values by traits in last step, we used all p-values within each trait as inputs to compute FDR. 

## Results for 'by cell types'

```{r echo=FALSE, message=FALSE, warning=FALSE}
load("/project2/xinhe/xsun/CEDAR/ACAT_randomsnp/calibrated_fdr/summary_correctfdr_correct_by_celltype.rdata")
DT::datatable(summary, options = list(pageLength =20))
```


## Results for 'by traits'

```{r echo=FALSE, message=FALSE, warning=FALSE}
load("/project2/xinhe/xsun/CEDAR/ACAT_randomsnp/calibrated_fdr/summary_correctfdr_correct_by_trait.rdata")
DT::datatable(summary, options = list(pageLength =20))
```


## Try to find out why the results differ a lot

The number of pairs passing the cutoffs differs a lot under two settings. We will try to find out the reasons for it.

We first made a scatter plot showing the fdr computed from 'by cell types' V.S. fdr computed from 'by traits'.

```{r echo=FALSE, message=FALSE, warning=FALSE, paged.print=FALSE,fig.height=10, fig.width=18}
type <- c("B cell (CD19+)","Monocyte (CD14+)","Leukocyte (CD15+)","Platelet","T cell (CD4+)","T cell (CD8+)")

load("/project2/xinhe/xsun/CEDAR/ACAT_randomsnp/calibrated_fdr/correctfdr_calibrated_acat_real_p_by_celltypes.rdata")
fdr_bycell <- fdr_calibrated_real_p

load("/project2/xinhe/xsun/CEDAR/ACAT_randomsnp/calibrated_fdr/correctfdr_calibrated_acat_real_p_by_trait.rdata")
fdr_bytraits <- fdr_corr_all

plot <- list()
for (i in 1:length(fdr_bycell)) {
  
  fdr_c <- data.frame(cbind(names(fdr_bycell[[i]]), fdr_bycell[[i]]))
  colnames(fdr_c) <- c("pair", "p_val_bycell")
  
  fdr_t <- data.frame(cbind(names(fdr_bytraits[[i]]), fdr_bytraits[[i]]))
  colnames(fdr_t) <- c("pair", "p_val_bytrait")
  
  fdr_combine <- cbind(fdr_c, fdr_t$p_val_bytrait)
  colnames(fdr_combine)[3] <- "p_val_bytrait"
  
  fdr_combine$p_val_bycell <- as.numeric(fdr_combine$p_val_bycell)
  fdr_combine$p_val_bytrait <- as.numeric(fdr_combine$p_val_bytrait)
  
  plot[[i]] <- ggplot(fdr_combine, aes(x = p_val_bycell,y =p_val_bytrait)) +
  geom_point(shape = 1, size = 1.5,color = "steelblue") +theme_bw(base_line_size =0.3) +
    ggtitle(type[i]) + theme(plot.title = element_text(hjust = 0.5)) +
    labs(x = "FDR computed from 'by cell types '",
         y = "FDR computed from 'by traits'") +
    theme(axis.title.x = element_text(size = 10),
          axis.text.x = element_text(size = 8, color = "black"),
          axis.title.y = element_text(size = 10),
          axis.text.y = element_text(size = 8, color = "black"),
          text= element_text(family="Times New Roman")) 
  
}
all <- grid.arrange(plot[[1]],plot[[2]],plot[[3]],plot[[4]],plot[[5]],plot[[6]], nrow = 2)

```

The reason for the large difference is that 1) the FDR computed from 'by traits' are similar for some traits (i.e. trait EUR.UC in B cell and T cell CD8+) 2)the FDR computed from 'by traits' and 'by cell type' are different (i.e. EUR.UC in Platelet, T cell CD4+). See the plots below

```{r echo=FALSE, message=FALSE, warning=FALSE, paged.print=FALSE,fig.height=10, fig.width=18}

for (i in 1:length(fdr_bycell)) {
  
  fdr_c <- data.frame(cbind(names(fdr_bycell[[i]]), fdr_bycell[[i]]))
  colnames(fdr_c) <- c("pair", "p_val_bycell")
  
  fdr_t <- data.frame(cbind(names(fdr_bytraits[[i]]), fdr_bytraits[[i]]))
  colnames(fdr_t) <- c("pair", "p_val_bytrait")
  
  #fdr_combine <-merge(fdr_c, fdr_t, by = "pair")
  
  fdr_combine <- cbind(fdr_c, fdr_t$p_val_bytrait)
  colnames(fdr_combine)[3] <- "p_val_bytrait"
  
  fdr_combine$p_val_bycell <- as.numeric(fdr_combine$p_val_bycell)
  fdr_combine$p_val_bytrait <- as.numeric(fdr_combine$p_val_bytrait)
  
  index <- grep("EUR.UC",fdr_combine$pair)
  eur.uc <- fdr_combine[index,]
  plot[[i]] <- ggplot(eur.uc, aes(x = p_val_bycell,y =p_val_bytrait)) +
  geom_point(shape = 1, size = 1.5,color = "steelblue") +theme_bw(base_line_size =0.3) +
    ggtitle(paste0("EUR.UC from ", type[i])) + theme(plot.title = element_text(hjust = 0.5)) +
    labs(x = "FDR computed from 'by cell types '",
         y = "FDR computed from 'by traits'") +
    theme(axis.title.x = element_text(size = 10),
          axis.text.x = element_text(size = 8, color = "black"),
          axis.title.y = element_text(size = 10),
          axis.text.y = element_text(size = 8, color = "black"),
          text= element_text(family="Times New Roman")) 


}
all <- grid.arrange(plot[[1]],plot[[2]],plot[[3]],plot[[4]],plot[[5]],plot[[6]], nrow = 2)

```

For EUR.UC from B cell, we made a plot showing calibrated ACAT p-values V.S. FDR (by traits) and FDR (by cell types)

```{r echo=FALSE, message=FALSE, warning=FALSE, paged.print=FALSE,fig.height=5, fig.width=10}

i <- 1

fdr_c <- data.frame(cbind(names(fdr_bycell[[i]]), fdr_bycell[[i]]))
colnames(fdr_c) <- c("pair", "p_val_bycell")
  
fdr_t <- data.frame(cbind(names(fdr_bytraits[[i]]), fdr_bytraits[[i]]))
colnames(fdr_t) <- c("pair", "p_val_bytrait")
  
  #fdr_combine <-merge(fdr_c, fdr_t, by = "pair")
  
fdr_combine <- cbind(fdr_c, fdr_t$p_val_bytrait)
colnames(fdr_combine)[3] <- "p_val_bytrait"
  
fdr_combine$p_val_bycell <- as.numeric(fdr_combine$p_val_bycell)
fdr_combine$p_val_bytrait <- as.numeric(fdr_combine$p_val_bytrait)
  
index <- grep("EUR.UC",fdr_combine$pair)
eur.uc <- fdr_combine[index,]


load("/project2/xinhe/xsun/CEDAR/ACAT_randomsnp/ACAT_results_remove_MHC_random.rdata")
acat_random <- acat_allcelltypes
load("/project2/xinhe/xsun/CEDAR/ACAT_randomsnp/ACAT_results_remove_MHC_realdata.rdata")
acat_real <- acat_allcelltypes

acat_real_celltype <- acat_real[[1]]
acat_random_celltype <- acat_random[[1]]
acat_random_celltype[is.na(acat_random_celltype)] <- 1
    
index_trait_real <- grep("EUR.UC",names(acat_real_celltype))
index_trait_random <- grep("EUR.UC",names(acat_random_celltype))
    
p_real <- acat_real_celltype[index_trait_real]
p_random <- acat_random_celltype[index_trait_random]
    
p_corr_tmp <- c()
for (t in 1:length(p_real)) {
  p_corr_tmp[t] <- sum(p_real[t] > p_random)/length(p_real)
}
p_corr_tmp[p_corr_tmp == 0] <- 0.5/length(p_real)
p_calibrated <- p_corr_tmp

df <- data.frame(cbind(p_calibrated,eur.uc$p_val_bycell))
colnames(df) <- c("p_calibrated", "fdr")
p1 <- ggplot(df, aes(x = p_calibrated,y =fdr )) +
  geom_point(shape = 1, size = 1.5,color = "steelblue") +theme_bw(base_line_size =0.3) +
    ggtitle("p_calibrated vs fdr_by cell") + theme(plot.title = element_text(hjust = 0.5)) +
    labs(x = "p_calibrated",
         y = "FDR computed from 'by cell'") +
    theme(axis.title.x = element_text(size = 10),
          axis.text.x = element_text(size = 8, color = "black"),
          axis.title.y = element_text(size = 10),
          axis.text.y = element_text(size = 8, color = "black"),
          text= element_text(family="Times New Roman")) 


df <- data.frame(cbind(p_calibrated,eur.uc$p_val_bytrait))
colnames(df) <- c("p_calibrated", "fdr")
p2 <- ggplot(df, aes(x = p_calibrated,y =fdr )) +
  geom_point(shape = 1, size = 1.5,color = "steelblue") +theme_bw(base_line_size =0.3) +
    ggtitle("p_calibrated vs fdr_by trait") + theme(plot.title = element_text(hjust = 0.5)) +
    labs(x = "p_calibrated",
         y = "FDR computed from 'by trait'") +
    theme(axis.title.x = element_text(size = 10),
          axis.text.x = element_text(size = 8, color = "black"),
          axis.title.y = element_text(size = 10),
          axis.text.y = element_text(size = 8, color = "black"),
          text= element_text(family="Times New Roman")) 

all <- grid.arrange(p1,p2, nrow =1)
hist(p_calibrated)
```

For EUR.UC from T cell CD8+, we also made a plot showing calibrated ACAT p-values V.S. FDR (by traits) and FDR (by cell types)

```{r echo=FALSE, message=FALSE, warning=FALSE, paged.print=FALSE,fig.height=5, fig.width=10}

i <- 6

fdr_c <- data.frame(cbind(names(fdr_bycell[[i]]), fdr_bycell[[i]]))
colnames(fdr_c) <- c("pair", "p_val_bycell")
  
fdr_t <- data.frame(cbind(names(fdr_bytraits[[i]]), fdr_bytraits[[i]]))
colnames(fdr_t) <- c("pair", "p_val_bytrait")
  
  #fdr_combine <-merge(fdr_c, fdr_t, by = "pair")
  
fdr_combine <- cbind(fdr_c, fdr_t$p_val_bytrait)
colnames(fdr_combine)[3] <- "p_val_bytrait"
  
fdr_combine$p_val_bycell <- as.numeric(fdr_combine$p_val_bycell)
fdr_combine$p_val_bytrait <- as.numeric(fdr_combine$p_val_bytrait)
  
index <- grep("EUR.UC",fdr_combine$pair)
eur.uc <- fdr_combine[index,]


load("/project2/xinhe/xsun/CEDAR/ACAT_randomsnp/ACAT_results_remove_MHC_random.rdata")
acat_random <- acat_allcelltypes
load("/project2/xinhe/xsun/CEDAR/ACAT_randomsnp/ACAT_results_remove_MHC_realdata.rdata")
acat_real <- acat_allcelltypes

acat_real_celltype <- acat_real[[6]]
acat_random_celltype <- acat_random[[6]]
acat_random_celltype[is.na(acat_random_celltype)] <- 1
    
index_trait_real <- grep("EUR.UC",names(acat_real_celltype))
index_trait_random <- grep("EUR.UC",names(acat_random_celltype))
    
p_real <- acat_real_celltype[index_trait_real]
p_random <- acat_random_celltype[index_trait_random]
    
p_corr_tmp <- c()
for (t in 1:length(p_real)) {
  p_corr_tmp[t] <- sum(p_real[t] > p_random)/length(p_real)
}
p_corr_tmp[p_corr_tmp == 0] <- 0.5/length(p_real)
p_calibrated <- p_corr_tmp
names(p_calibrated) <- names(p_real)


df <- data.frame(cbind(p_calibrated,eur.uc$p_val_bycell))
colnames(df) <- c("p_calibrated","fdr")
p1 <- ggplot(df, aes(x = p_calibrated,y =fdr )) +
  geom_point(shape = 1, size = 1.5,color = "steelblue") +theme_bw(base_line_size =0.3) +
    ggtitle("p_calibrated vs fdr_by cell") + theme(plot.title = element_text(hjust = 0.5)) +
    labs(x = "p_calibrated",
         y = "FDR computed from 'by cell'") +
    theme(axis.title.x = element_text(size = 10),
          axis.text.x = element_text(size = 8, color = "black"),
          axis.title.y = element_text(size = 10),
          axis.text.y = element_text(size = 8, color = "black"),
          text= element_text(family="Times New Roman")) 


df <- data.frame(cbind(p_calibrated,eur.uc$p_val_bytrait))
colnames(df) <- c("p_calibrated", "fdr")
p2 <- ggplot(df, aes(x = p_calibrated,y =fdr )) +
  geom_point(shape = 1, size = 1.5,color = "steelblue") +theme_bw(base_line_size =0.3) +
    ggtitle("p_calibrated vs fdr_by trait") + theme(plot.title = element_text(hjust = 0.5)) +
    labs(x = "p_calibrated",
         y = "FDR computed from 'by trait'") +
    theme(axis.title.x = element_text(size = 10),
          axis.text.x = element_text(size = 8, color = "black"),
          axis.title.y = element_text(size = 10),
          axis.text.y = element_text(size = 8, color = "black"),
          text= element_text(family="Times New Roman")) 

all <- grid.arrange(p1,p2, nrow =1)

hist(p_calibrated)

```


<!-- We also compared the adjusted random p-values with adjusted real p-values. The inflation factors for real p-values are 5.775381 6.671074 4.687092 4.605726 5.453085 4.84691.  -->

<!-- ```{r echo=FALSE, message=FALSE, warning=FALSE, paged.print=FALSE,fig.height=10, fig.width=15} -->
<!-- load("/project2/xinhe/xsun/CEDAR/ACAT_randomsnp/pval_gc_corrected_random_MHCremoved.rdata") -->
<!-- acat_random <- pval_corrected -->
<!-- load("/project2/xinhe/xsun/CEDAR/ACAT_randomsnp/pval_gc_corrected_real_MHCremoved.rdata") -->
<!-- acat_real <- pval_corrected -->

<!-- type <- c("B cell (CD19+)","Monocyte (CD14+)","Leukocyte (CD15+)","Platelet","T cell (CD4+)","T cell (CD8+)") -->


<!-- p <- list() -->
<!-- for (i in 1:length(acat_random)){ -->

<!--   p_random <- acat_random[[i]] -->
<!--   p_real <- acat_real[[i]] -->

<!--   if (length(p_random) > length(p_real)) { -->
<!--     p_real[length(p_real)+1] <- 1 -->
<!--   } -->
<!--   p_random[is.na(p_random)] <- 1 -->

<!--   plotdata <- data.frame(observed_random = -log10(sort(as.numeric(as.character(p_random)))), -->
<!--                          expected = -log10(ppoints(length(p_random))), -->
<!--                          observed_real = -log10(sort(as.numeric(as.character(p_real)))) ) -->
<!--   dat.m <- melt(plotdata, id.vars = "expected") -->


<!--   p[[i]]<- ggplot(dat.m, aes(expected, value, colour = variable)) + theme_bw(base_line_size =0.3) + -->
<!--     geom_point(shape = 1, size = 1.5) + -->
<!--     geom_abline(intercept = 0, slope = 1, alpha = 0.6, color = "red") + -->
<!--     labs(x = expression(paste("Expected -log"[10],"(p-value)")), -->
<!--          y = expression(paste("Observed -log"[10],"(p-value)"))) + -->

<!--     theme(axis.title.x = element_text(size = 10), -->
<!--           axis.text.x = element_text(size = 8, color = "black"), -->
<!--           axis.title.y = element_text(size = 10), -->
<!--           axis.text.y = element_text(size = 8, color = "black"), -->
<!--           text= element_text(family="Times New Roman")) + -->

<!--     scale_colour_manual(values = c("red", "steelblue")) + -->

<!--     ggtitle(type[i]) + theme(plot.title = element_text(hjust = 0.5)) -->

<!-- } -->

<!-- all <- grid.arrange(p[[1]],p[[2]],p[[3]],p[[4]],p[[5]],p[[6]], nrow = 2) -->
<!-- ``` -->


<!-- ```{r echo=FALSE, message=FALSE, warning=FALSE, paged.print=FALSE,fig.height=10, fig.width=15} -->

<!-- load("/project2/xinhe/xsun/CEDAR/ACAT_randomsnp/ACAT_results_remove_MHC_realdata.rdata") -->

<!-- type <- c("B cell (CD19+)","Monocyte (CD14+)","Leukocyte (CD15+)","Platelet","T cell (CD4+)","T cell (CD8+)") -->

<!-- plot <- list() -->
<!-- for (i in 1:length(type)){ -->

<!--   p <- acat_allcelltypes[[i]] -->
<!--   p[is.na(p)] <- 1 -->
<!--   plotdata <- data.frame(observed = -log10(sort(p)), -->
<!--                          expected = -log10(ppoints(length(p)))) -->
<!--   #   -->

<!--   plot[[i]] <- ggplot(plotdata) + theme_bw(base_line_size =0.3) + -->
<!--     geom_point(aes(expected, observed), shape = 1, size = 1.5,color = "steelblue") + -->
<!--     geom_abline(intercept = 0, slope = 1, alpha = 0.6, color = "red") + -->
<!--     ggtitle(type[i]) + theme(plot.title = element_text(hjust = 0.5)) + -->
<!--     labs(x = expression(paste("Expected -log"[10],"(p-value)")), -->
<!--          y = expression(paste("Observed -log"[10],"(p-value)"))) + -->

<!--     theme(axis.title.x = element_text(size = 10), -->
<!--           axis.text.x = element_text(size = 8, color = "black"), -->
<!--           axis.title.y = element_text(size = 10), -->
<!--           axis.text.y = element_text(size = 8, color = "black"), -->
<!--           text= element_text(family="Times New Roman"))  -->


<!-- } -->
<!-- all <- grid.arrange(plot[[1]],plot[[2]],plot[[3]],plot[[4]],plot[[5]],plot[[6]], nrow = 2) -->

<!-- ``` -->


<!-- ## ACAT for random data -- MHC removed -->

<!-- ```{r echo=FALSE, message=FALSE, warning=FALSE, paged.print=FALSE,fig.height=10, fig.width=15} -->

<!-- load("/project2/xinhe/xsun/CEDAR/ACAT_randomsnp/ACAT_results_remove_MHC_random.rdata") -->

<!-- type <- c("B cell (CD19+)","Monocyte (CD14+)","Leukocyte (CD15+)","Platelet","T cell (CD4+)","T cell (CD8+)") -->

<!-- plot <- list() -->
<!-- for (i in 1:length(type)){ -->

<!--   p <- acat_allcelltypes[[i]] -->
<!--   p[is.na(p)] <- 1 -->
<!--   plotdata <- data.frame(observed = -log10(sort(p)), -->
<!--                          expected = -log10(ppoints(length(p)))) -->
<!--   #   -->

<!--   plot[[i]] <- ggplot(plotdata) + theme_bw(base_line_size =0.3) + -->
<!--     geom_point(aes(expected, observed), shape = 1, size = 1.5,color = "steelblue") + -->
<!--     geom_abline(intercept = 0, slope = 1, alpha = 0.6, color = "red") + -->
<!--     ggtitle(type[i]) + theme(plot.title = element_text(hjust = 0.5)) + -->
<!--     labs(x = expression(paste("Expected -log"[10],"(p-value)")), -->
<!--          y = expression(paste("Observed -log"[10],"(p-value)"))) + -->

<!--     theme(axis.title.x = element_text(size = 10), -->
<!--           axis.text.x = element_text(size = 8, color = "black"), -->
<!--           axis.title.y = element_text(size = 10), -->
<!--           axis.text.y = element_text(size = 8, color = "black"), -->
<!--           text= element_text(family="Times New Roman"))  -->


<!-- } -->
<!-- all <- grid.arrange(plot[[1]],plot[[2]],plot[[3]],plot[[4]],plot[[5]],plot[[6]], nrow = 2) -->

<!-- ``` -->

<!-- # Correct original ACAT results with random ones -- All SNPs included -->

<!-- There is still slight deviation from null in the random SNPs after removing SNPs in MHC area. So we correct for real ACAT p-values using the distribution of ACAT p-values from random SNPs as null. In this way, we got empirical ACAT p-values computed from   sum(p_origin[j] < acat_p_random)/length(p_origin). -->


<!-- ```{r echo=FALSE, message=FALSE, warning=FALSE, paged.print=FALSE,fig.height=10, fig.width=15} -->

<!-- setwd("/project2/xinhe/xsun/CEDAR/") -->

<!-- type <- c("B cell (CD19+)","Monocyte (CD14+)","Leukocyte (CD15+)","Platelet","T cell (CD4+)","T cell (CD8+)") -->
<!-- path <- c("4.B_cell_kegg","6.CD14_positive_monocyte_kegg","8.CD15_positive_leukocyte_kegg","16.platelet_kegg","10.T_cell_kegg","14.thymocyte_kegg") -->

<!-- load("/project2/xinhe/xsun/CEDAR/ACAT_randomsnp/ACAT_results.rdata") -->

<!-- p_correct_allcelltypes <- list() -->
<!-- for (i in 1:length(type)){ -->

<!--   file <- paste0("/project2/xinhe/xsun/CEDAR/",path[i],"/7.geneset40_peer/acat.rdata") -->
<!--   load(file) -->
<!--   p_origin <- as.numeric(list$pval_acat) -->

<!--   acat_cell_random <- acat_allcelltypes[[i]] -->
<!--   acat_cell_random[acat_cell_random == "NaN"] <- 0.00001 -->
<!--   p_correct <- c() -->
<!--   for (j in 1:length(p_origin)) { -->

<!--     p_correct[j] <- sum(p_origin[j] < acat_cell_random)/length(p_origin) -->
<!--   } -->

<!--   p_correct_allcelltypes[[i]] <- p_correct -->

<!-- } -->

<!-- p <- list() -->
<!-- for (i in 1:length(p_correct_allcelltypes)){ -->

<!--   plotdata <- data.frame(observed = -log10(sort(as.numeric(as.character(p_correct_allcelltypes[[i]])))), -->
<!--                          expected = -log10(ppoints(length(p_correct_allcelltypes[[i]])))) -->
<!--   #   -->

<!--   p[[i]] <- ggplot(plotdata) + theme_bw(base_line_size =0.3) + -->
<!--     geom_point(aes(expected, observed), shape = 1, size = 1.5, color= "steelblue") + -->
<!--     geom_abline(intercept = 0, slope = 1, alpha = 0.6, color = "red") + -->
<!--     ggtitle(type[i]) + theme(plot.title = element_text(hjust = 0.5)) + -->
<!--     labs(x = expression(paste("Expected -log"[10],"(p-value)")), -->
<!--          y = expression(paste("Observed -log"[10],"(p-value)"))) + -->

<!--     theme(axis.title.x = element_text(size = 10), -->
<!--           axis.text.x = element_text(size = 8, color = "black"), -->
<!--           axis.title.y = element_text(size = 10), -->
<!--           axis.text.y = element_text(size = 8, color = "black"), -->
<!--           text= element_text(family="Times New Roman"))  -->

<!-- } -->
<!-- all <- grid.arrange(p[[1]],p[[2]],p[[3]],p[[4]],p[[5]],p[[6]], nrow = 2) -->


<!-- ``` -->


<!-- # Correct original ACAT results with random ones -- MHC area removed -->

<!-- ```{r echo=FALSE, message=FALSE, warning=FALSE, paged.print=FALSE,fig.height=10, fig.width=15} -->

<!-- type <- c("B cell (CD19+)","Monocyte (CD14+)","Leukocyte (CD15+)","Platelet","T cell (CD4+)","T cell (CD8+)") -->

<!-- load("/project2/xinhe/xsun/CEDAR/ACAT_randomsnp/ACAT_results_remove_MHC_random.rdata") -->
<!-- acat_random <- acat_allcelltypes -->
<!-- load("/project2/xinhe/xsun/CEDAR/ACAT_randomsnp/ACAT_results_remove_MHC_realdata.rdata") -->
<!-- acat_real <- acat_allcelltypes -->


<!-- p_correct_allcelltypes <- list() -->
<!-- for (i in 1:length(acat_real)){ -->

<!--   p_origin <- as.numeric(as.character(acat_real[[i]])) -->

<!--   acat_cell_random <- as.numeric(as.character(acat_random[[i]])) -->
<!--   acat_cell_random[acat_cell_random == "NaN"] <- 0.00001 -->
<!--   p_correct <- c() -->
<!--   for (j in 1:length(p_origin)) { -->

<!--     p_correct[j] <- sum(p_origin[j] < acat_cell_random)/length(p_origin) -->
<!--   } -->

<!--   p_correct_allcelltypes[[i]] <- p_correct -->

<!-- } -->


<!-- p <- list() -->
<!-- for (i in 1:length(p_correct_allcelltypes)){ -->

<!--   plotdata <- data.frame(observed = -log10(sort(as.numeric(as.character(p_correct_allcelltypes[[i]])))), -->
<!--                          expected = -log10(ppoints(length(p_correct_allcelltypes[[i]])))) -->
<!--   #   -->

<!--   p[[i]] <- ggplot(plotdata) + theme_bw(base_line_size =0.3) + -->
<!--     geom_point(aes(expected, observed), shape = 1, size = 1.5, color= "steelblue") + -->
<!--     geom_abline(intercept = 0, slope = 1, alpha = 0.6, color = "red") + -->
<!--     ggtitle(type[i]) + theme(plot.title = element_text(hjust = 0.5)) + -->
<!--     labs(x = expression(paste("Expected -log"[10],"(p-value)")), -->
<!--          y = expression(paste("Observed -log"[10],"(p-value)"))) + -->

<!--     theme(axis.title.x = element_text(size = 10), -->
<!--           axis.text.x = element_text(size = 8, color = "black"), -->
<!--           axis.title.y = element_text(size = 10), -->
<!--           axis.text.y = element_text(size = 8, color = "black"), -->
<!--           text= element_text(family="Times New Roman"))  -->

<!-- } -->
<!-- all <- grid.arrange(p[[1]],p[[2]],p[[3]],p[[4]],p[[5]],p[[6]], nrow = 2) -->


<!-- ``` -->