---
title: "catalog_gwas_pliercanon_sep_ldclumping"
author: "XSun"
date: "2020-09-24"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

## Introduction

In this part, I considered the traits separately. I selected the SNPs with fdr < 0.01 and pval < 5e-8 for each traits. Then, I did LD Clumping for these SNPs to eliminate the LD and select a smaller subset of SNPs. Finally, I did GWAS analysis for the plier_canonical factors with the SNPs.

## Material and Methods

1. I used the GWAS results of 12 immune traits: platelet count, white blood cell count, myeloid white cell count, lymphocyte counts, red blood cell count, granulocyte count, eosinophil count, neutrophil count. These data are from *Astle WJ, Elding H, Jiang T, et al. The Allelic Landscape of Human Blood Cell Trait Variation and Links to Common Complex Disease. Cell. 2016;167(5):1415-1429.e19. doi:10.1016/j.cell.2016.10.042* . I also used IBD,Ulcerative colitist,Crohn's disease and asthma collected by our lab.

2. Computed the fdr of each SNP according to their p-values from GWAS data. Chose those SNPs with fdr < 0.01 and pval < 5e-8 separately. 

3. Did LD Clumping for the SNPs in obtained from step 2. I used default settings of PLINK LD clumping: 
      
     --clump-p1 0.0001            Significance threshold for index SNPs
     
     --clump-p2 0.01              Secondary significance threshold for clumped SNPs
 
     --clump-r2 0.2               LD threshold for clumping

     --clump-kb 250               Physical distance threshold for clumping

For each trait, I got a subset of SNPs that are not in LD with each other.

4. Did GWAS analysis for plier_canonical factors and SNPs from 3. The association tests were corrected by 10 genotype PCs of whole genome.

5. For Crohn's disease, IBD, and Ulcerative colitist, I sorted the LVs by 'fdr020_num' and selected the top 3/4 LVs. For each LV, I made a plot of association with LV(indicating by beta in GWAS) vs association with trait(indicating by ln(odds ratio) in meta-GWAS) to show if the variants have the same effect direction.

## Results - fdr < 0.01

### Overview

After filtering by 'fdr<0.01' and LD Clumping, for each trait, I got :

platelet count trait contains  1001  SNPs with fdr<0.01. 

white blood cell count trait contains 552 SNPs with fdr<0.01. 

myeloid white cell count trait contains 440 SNPs with fdr<0.01. 

lymphocyte count trait contains 646 SNPs with fdr<0.01. 

red blood cell count trait contains 669 SNPs with fdr<0.01. 

granulocyte count trait contains 810 SNPs with fdr<0.01. 

eosinophil count trait contains  736 SNPs with fdr<0.01. 

neutrophil count trait contains 807 SNPs with fdr<0.01. 

IBD trait contains 309 SNPs with fdr<0.01. 

Ulcerative colitist trait contains 177 SNPs with fdr<0.01. 

Crohn's disease trait contains 231 SNPs with fdr<0.01. 

asthma trait contains 453 SNPs with fdr<0.01. 

### Summary table

I used 'qvalue' R package to compute the fdr from p-values for each SNP and made a table to show the number of SNPs that pass the threshold. The thresholds are 'fdr < 0.1','fdr < 0.2','pval < 5e-8'.

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/sum_sep_tog_fdr_pliercanon_wgpc_ldcl.rdata")
DT::datatable(summary, options = list(pageLength =20))
```

### Info table

For each trait, I made a table to show the info of snps with fdr>0.2 in the factor ~ SNP + genotype pcs association test. The suffix '_assoc' here means that results are from factor ~ SNP + genotype pcs association test. The suffix '_gwas' here means results are from original GWAS results files. For EUR.CD, EUR.IBD, EUR.UC, the effectsize_gwas here means 'ln(OR)', for others, it means 'beta'. 

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/info_fdr001_pliercanon.rdata")
DT::datatable(final_full, options = list(pageLength =20))
```


### Effect size plots

For Crohn's disease, IBD, and Ulcerative colitist, I sorted the LVs by 'fdr020_num' and selected the top 2 LVs. For each LV, I made a plot of association with LV(indicating by beta in GWAS) vs association with trait(indicating by ln(odds ratio) in meta-GWAS) to show if the variants have the same effect direction. Each dot indicates a variant.

I add thress lines to make the plots clear. They are x=0, y=0 and y=x.

#### Crohn's disease

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv18_CD_fdr001_pliercanon.jpeg" width="50%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv88_CD_fdr001_pliercanon.jpeg" width="50%">
</figure>

#### IBD

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv24_IBD_fdr001_pliercanon.jpeg" width="50%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv88_IBD_fdr001_pliercanon.jpeg" width="50%">
</figure>

#### Ulcerative colitist

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv98_UC_fdr001_pliercanon.jpeg" width="50%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv100_UC_fdr001_pliercanon.jpeg" width="50%">
</figure>

### QQ plots of the top factor-trait associations. 

For Crohn's disease, IBD and Ulcerative colitist, I made QQ plots of the top factor-trait associations. 


#### Crohn's disease -- LV88 has 4 SNPs with fdr<0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qq_unif_lv88_CD_fdr_pliercanon.jpeg" width="50%">
</figure> 

#### IBD -- LV88 has 7 SNPs with fdr<0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qq_unif_lv88_IBD_fdr_pliercanon.jpeg" width="50%">
</figure>

#### Ulcerative colitist -- LV100 has 2 SNPs with fdr<0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qq_unif_lv100_UC_fdr_pliercanon.jpeg" width="50%">
</figure>


## Results - pval < 5e-8

### Overview

After filtering by 'fdr<0.01' and LD Clumping, for each trait, I got :

platelet count trait contains 1,073  SNPs with pval<5e-8. 

white blood cell count trait contains 591 SNPs with pval<5e-8. 

myeloid white cell count trait contains 490 SNPs with pval<5e-8. 

lymphocyte count trait contains 699 SNPs with pval<5e-8. 

red blood cell count trait contains 731 SNPs with pval<5e-8. 

granulocyte count trait contains 464 SNPs with pval<5e-8. 

eosinophil count trait contains 786 SNPs with pval<5e-8. 

neutrophil count trait contains 476 SNPs with pval<5e-8. 

IBD trait contains 178 SNPs with pval<5e-8. 

Ulcerative colitist trait contains 103 SNPs with pval<5e-8. 

Crohn's disease trait contains 142 SNPs with pval<5e-8. 

asthma trait contains 283 SNPs with pval<5e-8. 

### Summary table

I used 'qvalue' R package to compute the fdr from p-values for each SNP and made a table to show the number of SNPs that pass the threshold. The thresholds are 'fdr < 0.1','fdr < 0.2','pval < 5e-8'.

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/sum_sep_tog_pval_pliercanon_wgpc_ldcl.rdata")
DT::datatable(summary, options = list(pageLength =20))
```


### Info table

For each trait, I made a table to show the info of snps with fdr>0.2 in the factor ~ SNP + genotype pcs association test. The suffix '_assoc' here means that results are from factor ~ SNP + genotype pcs association test. The suffix '_gwas' here means results are from original GWAS results files. For EUR.CD, EUR.IBD, EUR.UC, the effectsize_gwas here means 'ln(OR)', for others, it means 'beta'. 

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/info_pval5e8_pliercanon.rdata")
DT::datatable(final_full, options = list(pageLength =20))
```

### Effect size plots

For Crohn's disease, IBD, and Ulcerative colitist, I sorted the LVs by 'fdr020_num' and selected the top 2 LVs. For each LV, I made a plot of association with LV(indicating by beta in GWAS) vs association with trait(indicating by ln(odds ratio) in meta-GWAS) to show if the variants have the same effect direction.Each dot indicates a variant.

I add thress lines to make the plots clear. They are x=0, y=0 and y=x.

#### Crohn's disease

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv18_CD_pval5e8_pliercanon.jpeg" width="50%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv88_CD_pval5e8_pliercanon.jpeg" width="50%">
</figure>

#### IBD

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv88_IBD_pval5e8_pliercanon.jpeg" width="50%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv98_IBD_pval5e8_pliercanon.jpeg" width="50%">
</figure>

#### Ulcerative colitist

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv98_UC_pval5e8_pliercanon.jpeg" width="50%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv100_UC_pval5e8_pliercanon.jpeg" width="50%">

</figure>


### QQ plots of the top factor-trait associations. 

For Crohn's disease, IBD and Ulcerative colitist, I made QQ plots of the top factor-trait associations. 


#### Crohn's disease -- LV88 has 5 SNPs with fdr<0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qq_unif_lv88_CD_pval_pliercanon.jpeg" width="50%">
</figure>

#### IBD -- LV88 has 4 SNPs with fdr<0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qq_unif_lv88_IBD_pval_pliercanon.jpeg" width="50%">
</figure>

#### Ulcerative colitist -- LV98 has 2 SNPs with fdr<0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qq_unif_lv98_UC_pval_pliercanon.jpeg" width="50%">
</figure>
