---
title: "catalog_gwas_flash_sep_ldclumping_r01d1000"
author: "XSun"
date: "2020-10-23"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

## Introduction

In this part, I considered the traits separately. I selected the SNPs with pval < 5e-8 for each traits. Then, I did LD Clumping for these SNPs to eliminate the LD and select a smaller subset of SNPs. After that, I did association tests for the FLASH factors with the SNPs. I also did colozalization analysis for some significant factor~snp pairs.


## Material and Methods

1. Catalog GWAS data:  

- Platelet count, white blood cell count, myeloid white cell count, lymphocyte counts, red blood cell count, granulocyte count, eosinophil count, neutrophil count from *Astle WJ, Elding H, Jiang T, et al. The Allelic Landscape of Human Blood Cell Trait Variation and Links to Common Complex Disease. Cell. 2016;167(5):1415-1429.e19. doi:10.1016/j.cell.2016.10.042*. 

- T2D. I first used data from our lab collaction *Morris et al. Large-scale association analysis provides insights into the genetic architecture and pathophysiology of type 2 diabetes. Nat Genet. 2012 Sep;44(9):981-90. doi: 10.1038/ng.2383. Epub 2012 Aug 12. PMID: 22885922; PMCID: PMC3442244.* but it doesn't contain MAF info of variants.So I added this from [GWAS Catalog](https://www.ebi.ac.uk/gwas/home): *Wood AR et al. Variants in the FTO and CDKAL1 loci have recessive effects on risk of obesity and type 2 diabetes, respectively. Diabetologia. 2016 Jun;59(6):1214-21. doi: 10.1007/s00125-016-3908-5. Epub 2016 Mar 10. PMID: 26961502; PMCID: PMC4869698.*.

- Asthma. I first used data from our lab collaction *Zhu et al, Shared Genetics of Asthma and Mental Health Disorders: A Large-Scale Genome-Wide Cross-Trait Analysis. European Respiratory Journal, 2019 (PMID: 31619474)* but it doesn't contain MAF info of variants.So I added this from [GWAS Catalog](https://www.ebi.ac.uk/gwas/home): *Manuel A.R. et al. Genetic Architectures of Childhood- and Adult-Onset Asthma Are Partly Distinct,The American Journal of Human Genetics,Volume 104, Issue 4,2019,Pages 665-684,ISSN 0002-9297,https://doi.org/10.1016/j.ajhg.2019.02.022.*.

- IBD,Ulcerative colitist,Crohn's disease data are from lab collection *Liu, van Sommeren et al, Nature Genetics, 2015*

- Waist-hip ratio data are from *Shungin et al. New genetic loci link adipose and insulin biology to body fat distribution. Nature. 2015 Feb 12;518(7538):187-196. doi: 10.1038/nature14132. PMID: 25673412; PMCID: PMC4338562*.

- BMI data are also from lab collection: *Locke AE, Kahali B, Berndt SI, Justice AE, Pers TH, Day FR, Powell C, Vedantam S, Buchkovich ML, Yang J, Croteau-Chonka DC, Esko T et al. (2015). Genetic studies of body mass index yield new insights for obesity biology. Nature 518, 197-206*

- HDL & LDL data are downloaded from [GWAS Catalog](https://www.ebi.ac.uk/gwas/home).  *Global Lipids Genetics Consortium., Willer, C., Schmidt, E. et al. Discovery and refinement of loci associated with lipid levels. Nat Genet 45, 12741283 (2013). https://doi.org/10.1038/ng.2797*


2. Filtered the SNPs using pval < 5e-8 as cut off from the GWAS Catalog data. 

3. Did LD Clumping for the SNPs in step 2. The PLINK LD Clumping patameters are: 
      
     --clump-p1 0.0001            Significance threshold for index SNPs
     
     --clump-p2 0.01              Secondary significance threshold for clumped SNPs
 
     --clump-r2 0.1               LD threshold for clumping

     --clump-kb 1000               Physical distance threshold for clumping

For each trait, I got a subset of SNPs that are not in LD with each other.

4. Did association tests for plier_canonical factors and SNPs in 3. The association tests were corrected by 1)10 genotype PCs of whole genome; 2)10 PCs + GTEx:Sequencing platform,Sequencing protocol,Sex; 3)10 PCs + GTEx:Sequencing platform,Sequencing protocol,Sex + AGE

5. For each trait, I made a plot of association with LV(indicating by beta in GWAS) vs association with trait(indicating by ln(odds ratio) or beta in GWAS) to show if the variants have the correlated effect direction. The effect sizes of Catalog GWAS and factor association tests are harmonized by [TwoSampleMR](https://mrcieu.github.io/TwoSampleMR/articles/introduction.html) R package to make the effect alleles in these two analysis identical. The LVs have more than one significant SNPs with FDR<0.2 are included in the plotting.Besides, for each plots, I fitted the points with intercept = 0. The pvalues and r-squared are shown on the plots.

6. For the traits and LVs in 5, I made an info table to show more details of the SNPs. 

7. For several LVs we are interested in, I did gene set enrichment analysis to test if the LVs are correlated with some KEGG/REACTOME pathways. I used two kind gene sets to do GSEA: 1. genes that used to compute LVs; 2. Sorting the genes in 1 by their loadings, take the top 25% as the gene set. For both gene sets, the gene scores used as input of GSEA are the gene loadings.

8. Resampling. For some promising trait-factor pairs, I did resampling. I resampled the SNPs without replacement, I fitted the points with intercept = 0 again and recorded the pvalues and r-squared. The resampling was repeated 1000 times. The following plots are the resampling results.


## Snps after filter

After filtering by 'pval < 5e-8' and LD Clumping, for each trait, I got :

platelet count trait contains 688 SNPs with pval<5e-8. 

white blood cell count trait contains 368 SNPs with pval<5e-8. 

myeloid white cell count trait contains 319 SNPs with pval<5e-8. 

lymphocyte count trait contains 436 SNPs with pval<5e-8. 

red blood cell count trait contains 466 SNPs with pval<5e-8. 

granulocyte count trait contains 316 SNPs with pval<5e-8. 

eosinophil count trait contains 491 SNPs with pval<5e-8. 

neutrophil count trait contains 317 SNPs with pval<5e-8. 

IBD trait contains 116 SNPs with pval<5e-8. 

Ulcerative colitist trait contains 73 SNPs with pval<5e-8. 

Crohn's disease trait contains 96 SNPs with pval<5e-8. 

BMI trait contains 104 SNPs with pval<5e-8. 

T2D contains 14 SNPs with pval<5e-8. T2D_2 contains 4 SNPs with pval<5e-8.

Asthma trait contains 186 SNPs with pval<5e-8. Asthma_2 trait contains 112 SNPs with pval<5e-8.

HDL trait contains 227 SNPs with pval<5e-8.

LDL trait contains 204 SNPs with pval<5e-8.

WHR trait contains 36 SNPs with pval<5e-8.


## Results - pval < 5e-8 & 100 FLASH factors & covariants: 10 PCs + GTEx:Sequencing platform,Sequencing protocol,Sex + AGE

### Summary table

I used 'qvalue' R package to compute the fdr from p-values for each SNP and made a table to show the number of SNPs that pass the threshold. The thresholds are 'fdr < 0.1','fdr < 0.2','pval < 5e-8'. The 'num_significant_pairs' indicates the number of significant pairs under each threshold. If a trait~factor pair has as least 1 significant SNP, we named it as 'significant pair'. 

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/sum_sep_tog_pval_flash_wgpc_ldcl_d1k_ageco.rdata")
DT::datatable(summary, options = list(pageLength =20))
```


### Info table

For each trait, I made a table to show the info of snps with fdr>0.2 in the factor ~ SNP + genotype pcs association test. For each trait,The LVs have more than one significant SNPs with FDR<0.2 are included. 

The suffix '_assoc' here means that results are from factor ~ SNP + genotype pcs association test. The suffix '_gwas' here means results are from original GWAS results files. For EUR.CD, EUR.IBD, EUR.UC, the effectsize_gwas here means 'ln(OR)', for others, it means 'beta'. 

'snp_ld' here means the snps that in LD with the snp in each line.'ld_r2' means the LD r-squared which is corresponding to the 'snp_ld' column. 'cis-eqtl' column indicates whether the snp is a cis-eqtl according to GTEx data. 'cis_gene_hgnc' and 'cis_gene_hgnc' is the genes that the snp influence when it act as cis-eqtl. 'func' and 'func_gene' are obtained from ANNOVAR, which indicating the snp function within the genes.


```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/info_pval5e8_flash_ld_d1k_ageco.rdata")
DT::datatable(final_full, options = list(pageLength =20))
```


### Enrichment analysis

There are some promising trait-factor pairs when considering SNPs at fdr<0.2 (i.e. PLT - LV79, T2D-LV23, Asthma - LV30,). When relaxing the cutoff to fdr<0.5, there appeared some other promising pairs(i.e. Neutrophil - LV31).

I did ORA for all the factors mentioned above with WebGestalt. The analysis settings: for each factor, I sorted the genes their absolute loading values, taking the top 5% as the gene set. The function database here are Reactome pathway; Disgenet + GLAD4U + OMIM disease dataset; geneontology biological process. The reference set affy hugene 2 0 st v1. Minimum number of genes for a category is 5, maximum number of genes for a category is 2000(default settings).


**PLT - LV79**

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/lv79top005_flash.rdata")
DT::datatable(ORA_results, options = list(pageLength =5))
```

**T2D - LV23**

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/lv30top005_flash.rdata")
DT::datatable(ORA_results, options = list(pageLength =5))
```

**Asthma - LV30**

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/lv30top005_flash.rdata")
DT::datatable(ORA_results, options = list(pageLength =5))
```

**WHR - LV1**

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/lv1top005_flash.rdata")
DT::datatable(ORA_results, options = list(pageLength =5))
```


### Effect size plots &Resampling

For each trait, I made a plot of association with LV(indicating by beta in GWAS) vs association with trait(indicating by ln(odds ratio) or beta in GWAS) to show if the variants have the correlated effect direction. The effect sizes of Catalog GWAS and factor association tests are harmonized by [TwoSampleMR](https://mrcieu.github.io/TwoSampleMR/articles/introduction.html) R package to make the effect alleles in these two analysis identical. The LVs have more than one significant SNPs with FDR<0.2 are included in the plotting.Besides, for each plots, I fitted the points with intercept = 0. The pvalues and r-squared are shown on the plots.

I also relaxed the fdr threshold of the SNPs that used to make effect size plots(from 0.2 to 0.3/0.5)

For all pairs , I did resampling. I resampled the SNPs without replacement and fitted the points with intercept = 0 again and recorded the pvalues and r-squared. The resampling was repeated 1000 times.

I made a histogram to show the pvalues/rsquared distribution from resampling. The red line in the plots are the pvalue/rsquared in the origin analysis. The p_mean/r_mean values in the plots are the mean values of the pvalues and rsquared in point fitting. The 'prob of getting more extreme values' is computed by: (number of more extreme values)/(times of resampling).

#### Asthma_2

LV2

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv2_asthma_2_pval5e8_flash_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_flash_2_asthma_2.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv2_asthma_2_pval5e8_flash_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_flash_2_asthma_2_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv2_asthma_2_pval5e8_flash_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_flash_2_asthma_2_fdr05.jpeg" width="30%">
</figure>

LV79

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv79_asthma_2_pval5e8_flash_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_flash_79_asthma_2.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv79_asthma_2_pval5e8_flash_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_flash_79_asthma_2_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv79_asthma_2_pval5e8_flash_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_flash_79_asthma_2_fdr05.jpeg" width="30%">
</figure>



#### BMI 

LV90

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv90_bmi_pval5e8_flash_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_flash_90_bmi.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv90_bmi_pval5e8_flash_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_flash_90_bmi_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv90_bmi_pval5e8_flash_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_flash_90_bmi_fdr05.jpeg" width="30%">
</figure>



#### Eosinophil count 


LV73

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv73_eo_pval5e8_flash_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_flash_73_eo.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv73_eo_pval5e8_flash_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_flash_73_eo_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv73_eo_pval5e8_flash_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_flash_73_eo_fdr05.jpeg" width="30%">
</figure>


#### Crohn's disease

LV12

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv12_EUR.CD_pval5e8_flash_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_flash_12_EUR.CD.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv12_EUR.CD_pval5e8_flash_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_flash_12_EUR.CD_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv12_EUR.CD_pval5e8_flash_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_flash_12_EUR.CD_fdr05.jpeg" width="30%">
</figure>


LV6

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv6_EUR.CD_pval5e8_flash_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_flash_6_EUR.CD.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv6_EUR.CD_pval5e8_flash_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_flash_6_EUR.CD_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv6_EUR.CD_pval5e8_flash_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_flash_6_EUR.CD_fdr05.jpeg" width="30%">
</figure>


#### IBD

None of the LVs have >1 SNPs at FDR<0.2.

#### Ulcerative colitist


LV66

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv66_EUR.UC_pval5e8_flash_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_flash_66_EUR.UC.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv66_EUR.UC_pval5e8_flash_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_flash_66_EUR.UC_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv66_EUR.UC_pval5e8_flash_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_flash_66_EUR.UC_fdr05.jpeg" width="30%">
</figure>

LV91

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv91_EUR.UC_pval5e8_flash_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_flash_91_EUR.UC.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv91_EUR.UC_pval5e8_flash_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_flash_91_EUR.UC_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv91_EUR.UC_pval5e8_flash_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_flash_91_EUR.UC_fdr05.jpeg" width="30%">
</figure>


#### granulocyte count

None of the LVs have >1 SNPs at FDR<0.2.


#### HDL

LV60

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv60_HDL_pval5e8_flash_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_flash_60_HDL.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv60_HDL_pval5e8_flash_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_flash_60_HDL_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv60_HDL_pval5e8_flash_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_flash_60_HDL_fdr05.jpeg" width="30%">
</figure>



#### LDL

LV92

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv92_LDL_pval5e8_flash_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_flash_92_LDL.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv92_LDL_pval5e8_flash_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_flash_92_LDL_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv92_LDL_pval5e8_flash_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_flash_92_LDL_fdr05.jpeg" width="30%">
</figure>


#### lymphocyte count


LV100

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv100_lymph_pval5e8_flash_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_flash_100_lymph.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv100_lymph_pval5e8_flash_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_flash_100_lymph_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv100_lymph_pval5e8_flash_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_flash_100_lymph_fdr05.jpeg" width="30%">
</figure>


#### myeloid white cell count

None of the LVs have >1 SNPs at FDR<0.2.

#### neutrophil count


LV31

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv31_neut_pval5e8_flash_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_flash_31_neut.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv31_neut_pval5e8_flash_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_flash_31_neut_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv31_neut_pval5e8_flash_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_flash_31_neut_fdr05.jpeg" width="30%">
</figure>

#### plt

LV79

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv79_plt_pval5e8_flash_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_flash_79_plt.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv79_plt_pval5e8_flash_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_flash_79_plt_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv79_plt_pval5e8_flash_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_flash_79_plt_fdr05.jpeg" width="30%">
</figure>


LV65

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv65_plt_pval5e8_flash_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_flash_65_plt.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv65_plt_pval5e8_flash_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_flash_65_plt_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv65_plt_pval5e8_flash_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_flash_65_plt_fdr05.jpeg" width="30%">
</figure>



#### rbc

LV37

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv37_rbc_pval5e8_flash_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_flash_37_rbc.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv37_rbc_pval5e8_flash_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_flash_37_rbc_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv37_rbc_pval5e8_flash_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_flash_37_rbc_fdr05.jpeg" width="30%">
</figure>


LV48

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv48_rbc_pval5e8_flash_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_flash_48_rbc.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv48_rbc_pval5e8_flash_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_flash_48_rbc_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv48_rbc_pval5e8_flash_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_flash_48_rbc_fdr05.jpeg" width="30%">
</figure>


#### T2D_2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv26_T2D_2_pval5e8_flash_d1k_ageco.jpeg" width="30%">
</figure>

Since there were only 4 SNPs left after filtering by p-values in GWAS summary data and all of them have fdr<0.2 in association test. So there is no other SNPs to do resampling.


#### T2D

LV23

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv23_T2D_pval5e8_flash_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_flash_23_T2D.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv23_T2D_pval5e8_flash_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_flash_23_T2D_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv23_T2D_pval5e8_flash_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_flash_23_T2D_fdr05.jpeg" width="30%">
</figure>


#### asthma


LV87

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv87_ukb.allasthma_pval5e8_flash_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_flash_87_ukb.allasthma.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv87_ukb.allasthma_pval5e8_flash_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_flash_87_ukb.allasthma_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv87_ukb.allasthma_pval5e8_flash_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_flash_87_ukb.allasthma_fdr05.jpeg" width="30%">
</figure>


LV30

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv30_ukb.allasthma_pval5e8_flash_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_flash_30_ukb.allasthma.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv30_ukb.allasthma_pval5e8_flash_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_flash_30_ukb.allasthma_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv30_ukb.allasthma_pval5e8_flash_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_flash_30_ukb.allasthma_fdr05.jpeg" width="30%">
</figure>



LV42

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv42_ukb.allasthma_pval5e8_flash_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_flash_42_ukb.allasthma.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv42_ukb.allasthma_pval5e8_flash_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_flash_42_ukb.allasthma_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv42_ukb.allasthma_pval5e8_flash_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_flash_42_ukb.allasthma_fdr05.jpeg" width="30%">
</figure>


#### wbc

None of the LVs have >1 SNPs at FDR<0.2.


#### WHR


LV1

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv1_WHR_pval5e8_flash_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_flash_1_WHR.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv1_WHR_pval5e8_flash_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_flash_1_WHR_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv1_WHR_pval5e8_flash_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_flash_1_WHR_fdr05.jpeg" width="30%">
</figure>


LV19

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv19_WHR_pval5e8_flash_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_flash_19_WHR.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv19_WHR_pval5e8_flash_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_flash_19_WHR_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv19_WHR_pval5e8_flash_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_flash_19_WHR_fdr05.jpeg" width="30%">
</figure>


LV49

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv49_WHR_pval5e8_flash_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_flash_49_WHR.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv49_WHR_pval5e8_flash_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_flash_49_WHR_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv49_WHR_pval5e8_flash_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_flash_49_WHR_fdr05.jpeg" width="30%">
</figure>


### Effect size plots - checking the reverse causality

To check if the effect size correlation is due to reverse causality: i.e. trait -> LV (trait causally affect LV), instead of LV -> trait (which is what we like to see). I used all SNPs associated with traits(pval<5E-8). The x-axis is the effects of these SNPs on trait, and y-axis is the effects on LV. 

Some pair show p < 0.05, the result may be driven by the possible causal effect of LV -> trait. To test this, I removed the SNPs that are associated with LVs at FDR < 0.2 and made the plots again.

#### Neutrophil count

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv31_neut_pval5e8_flash_d1k_ageco_reverse.jpeg" width="30%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv31_neut_pval5e8_flash_d1k_ageco_reverse_remove.jpeg" width="30%">
</figure>

#### platelet count

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv79_plt_pval5e8_flash_d1k_ageco_reverse.jpeg" width="30%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv79_plt_pval5e8_flash_d1k_ageco_reverse_remove.jpeg" width="30%">
</figure>

#### T2D

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv23_T2D_pval5e8_flash_d1k_ageco_reverse.jpeg" width="30%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv23_T2D_pval5e8_flash_d1k_ageco_reverse_remove.jpeg" width="30%">
</figure>

#### asthma

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv30_ukb.allasthma_pval5e8_flash_d1k_ageco_reverse.jpeg" width="30%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv30_ukb.allasthma_pval5e8_flash_d1k_ageco_reverse_remove.jpeg" width="30%">
</figure>


#### WHR

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv1_WHR_pval5e8_flash_d1k_ageco_reverse.jpeg" width="30%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv1_WHR_pval5e8_flash_d1k_ageco_reverse_remove.jpeg" width="30%">
</figure>



### QQplots - promising pairs

#### Neutrophil count

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qq_unif_lv31_neut_flash.jpeg" width="30%">
</figure>

#### platelet count

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qq_unif_lv79_plt_flash.jpeg" width="30%">
</figure>

#### T2D

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qq_unif_lv23_T2D_flash.jpeg" width="30%">
</figure>

#### asthma

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qq_unif_lv30_ukb.allasthma_flash.jpeg" width="30%">
</figure>

#### WHR

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qq_unif_lv1_WHR_flash.jpeg" width="30%">
</figure>


### Colocalization - 500kb

The colocalization analysis was performed using the approximate Bayes factor test implemented in the Coloc package. Coloc computes five posterior probabilities (PP0, PP1, PP2, PP3 and PP4), each corresponding to a hypothesis: H0, no association with either trait; H1, association with trait 1 but not with trait 2; H2, association with trait 2 but not with trait 1; H3, association with trait 1 and trait 2, two independent SNPs; H4, association with trait 1 and trait 2, one shared SNP. We ran Coloc with the default parameters and used PP4 to assess evidence of colocalization. We visualized the colocalization of factor - QTLs and GWAS associations using the [LocusCompareR package](https://github.com/boxiangliu/locuscomparer).

Chromosome selection: I first sorted the SNPs by their pvalues from factor association tests, then I selected the first two SNPs and chose their chromosome to do the colocalization test adn visualization. 


#### platelet count

lv79

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/plt_79_30343875.coloc_flash.results_500.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "pvalues in coloclization")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/plt_79_30343875.coloc_flash_500.jpeg" width="80%">
</figure>

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/plt_79_44179062.coloc_flash.results_500.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "pvalues in coloclization")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/plt_79_44179062.coloc_flash_500.jpeg" width="80%">
</figure>

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/plt_79_44377654.coloc_flash.results_500.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "pvalues in coloclization")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/plt_79_44377654.coloc_flash_500.jpeg" width="80%">
</figure>

#### WHR

lv79

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/WHR_1_172377408.coloc_flash.results_500.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "pvalues in coloclization")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/WHR_1_172377408.coloc_flash_500.jpeg" width="80%">
</figure>

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/WHR_1_58565242.coloc_flash.results_500.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "pvalues in coloclization")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/WHR_1_58565242.coloc_flash_500.jpeg" width="80%">
</figure>





