---
title: "catalog_gwas_pliercanon_sep_ldclumping_r01d1000_geu"
author: "XSun"
date: "2020-11-02"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

## Introduction

In this part, I used GEUVADIS in LCL(462 samples) and inferred 152 plier-canonical LVs from the expression matrix. Then I used these factors to do the same analysis with the [GTEx LVs](https://xsun1229.github.io/factor_analysis/catalog_gwas_pliercanon_sep_ldclumping_r01d1000.html). Since the analysis pipeline is the same with former analysis(an exception: the covariants in association tests are: 10 whole genome PCs, gender, population, lab,), I just post the results here. 


## Data processing

1. Genotype data. 

The catalog-GWAS data use rsid, so I first converted the GEUVADIS SNP ID(SNP_chr_pos) into rs ID. I downloaded the common dbSNP153 variants(maf >0.01) from the [ucsc table browser](https://genome.ucsc.edu/cgi-bin/hgTables) and mapped the converted SNPs according to these variants.

2. Expression data.

The GEUVADIS expression data are unnormalized. I followed the [GEUVADIS paper](https://www.nature.com/articles/nature12531#Sec8) to do the normalization. Ran PEER for the genes and inferred 10 factors. Then, regress out these factors from the expression data. Transform the expression data to standard normal distribution.

## Snps after filter

After filtering by 'pval < 5e-8' and LD Clumping(r2 = 0.1, physical dist = 1Mb), for each trait, I got :

platelet count trait contains 917 SNPs with pval<5e-8. 

white blood cell count trait contains 563 SNPs with pval<5e-8. 

myeloid white cell count trait contains 477 SNPs with pval<5e-8. 

lymphocyte count trait contains 653 SNPs with pval<5e-8. 

red blood cell count trait contains 671 SNPs with pval<5e-8. 

granulocyte count trait contains 457 SNPs with pval<5e-8. 

eosinophil count trait contains 730 SNPs with pval<5e-8. 

neutrophil count trait contains 460 SNPs with pval<5e-8. 

IBD trait contains 250 SNPs with pval<5e-8. 

Ulcerative colitist trait contains 115 SNPs with pval<5e-8. 

Crohn's disease trait contains 189 SNPs with pval<5e-8. 

asthma trait contains 215 SNPs with pval<5e-8. 

BMI trait contains 106 SNPs with pval<5e-8. 

T2D contains 15 SNPs with pval<5e-8. 

## Results

### Summary table

I used 'qvalue' R package to compute the fdr from p-values for each SNP and made a table to show the number of SNPs that pass the threshold. The thresholds are 'fdr < 0.1','fdr < 0.2','pval < 5e-8'.

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/sum_sep_tog_pval_pliercanon_wgpc_ldcl_d1k_geu.rdata")
DT::datatable(summary, options = list(pageLength =20))
```


### Info table

For each trait, I made a table to show the info of snps with fdr>0.2 in the factor ~ SNP + genotype pcs + gender association test. For each trait,The LVs have more than one significant SNPs with FDR<0.2 are included. 

The suffix '_assoc' here means that results are from factor ~ SNP + genotype pcs association test. The suffix '_gwas' here means results are from original GWAS results files. For EUR.CD, EUR.IBD, EUR.UC, T2D, asthma the effectsize_gwas here means 'ln(OR)', for others, it means 'beta'. 

'snp_ld' here means the snps that in LD with the snp in each line.'ld_r2' means the LD r-squared which is corresponding to the 'snp_ld' column. 'cis-eqtl' column indicates whether the snp is a cis-eqtl according to GTEx data. 'cis_gene_hgnc' and 'cis_gene_hgnc' is the genes that the snp influence when it act as cis-eqtl. 'func' and 'func_gene' are obtained from ANNOVAR, which indicating the snp function within the genes.

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/info_pval5e8_pliercanon_ld_d1k_geu.rdata")
rownames(final_full) <- seq(1,nrow(final_full), by=1)
DT::datatable(final_full, options = list(pageLength =20))
```


### Summary table  - lower fdr threshold

I used 'qvalue' R package to compute the fdr from p-values for each SNP and made a table to show the number of SNPs that pass the threshold. The thresholds are 'fdr < 0.1','fdr < 0.2','pval < 5e-8'.

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/sum_sep_tog_pval_pliercanon_wgpc_ldcl_d1k_geu_low.rdata")
DT::datatable(summary, options = list(pageLength =20))
```


### Info table  - lower fdr threshold

For each trait, I made a table to show the info of snps with fdr>0.2 in the factor ~ SNP + genotype pcs + gender association test. For each trait,The LVs have more than one significant SNPs with FDR<0.2 are included. 

The suffix '_assoc' here means that results are from factor ~ SNP + genotype pcs association test. The suffix '_gwas' here means results are from original GWAS results files. For EUR.CD, EUR.IBD, EUR.UC, T2D, asthma the effectsize_gwas here means 'ln(OR)', for others, it means 'beta'. 

'snp_ld' here means the snps that in LD with the snp in each line.'ld_r2' means the LD r-squared which is corresponding to the 'snp_ld' column. 'cis-eqtl' column indicates whether the snp is a cis-eqtl according to GTEx data. 'cis_gene_hgnc' and 'cis_gene_hgnc' is the genes that the snp influence when it act as cis-eqtl. 'func' and 'func_gene' are obtained from ANNOVAR, which indicating the snp function within the genes.

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/info_pval5e8_pliercanon_ld_d1k_geu_low.rdata")
rownames(final_full) <- seq(1,nrow(final_full), by=1)
DT::datatable(final_full, options = list(pageLength =20))
```

