---
title: "catalog_gwas_peer_sep_ldclumping_r01d1000"
author: "XSun"
date: "2020-10-19"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

## Introduction

In this part, I considered the traits separately. I selected the SNPs with pval < 5e-8 for each traits. Then, I did LD Clumping for these SNPs to eliminate the LD and select a smaller subset of SNPs. Finally, I did GWAS analysis for the PEER factors with the SNPs.


## Material and Methods

1. I used the GWAS results of 14 immune traits: platelet count, white blood cell count, myeloid white cell count, lymphocyte counts, red blood cell count, granulocyte count, eosinophil count, neutrophil count. These data are from *Astle WJ, Elding H, Jiang T, et al. The Allelic Landscape of Human Blood Cell Trait Variation and Links to Common Complex Disease. Cell. 2016;167(5):1415-1429.e19. doi:10.1016/j.cell.2016.10.042* . I also used IBD,Ulcerative colitist,Crohn's disease, asthma, BMI,and T2D collected by our lab.

2. Computed the fdr of each SNP according to their p-values from GWAS data. Chose those SNPs with pval < 5e-8. 

3. Did LD Clumping for the SNPs in obtained from step 2. I used default settings of PLINK LD clumping: 
      
     --clump-p1 0.0001            Significance threshold for index SNPs
     
     --clump-p2 0.01              Secondary significance threshold for clumped SNPs
 
     --clump-r2 0.1               LD threshold for clumping

     --clump-kb 1000               Physical distance threshold for clumping

For each trait, I got a subset of SNPs that are not in LD with each other.

4. Did GWAS analysis for plier_canonical factors and SNPs from 3. The association tests were corrected by 10 genotype PCs of whole genome.

5. For each trait, I made a plot of association with LV(indicating by beta in GWAS) vs association with trait(indicating by ln(odds ratio) or beta in GWAS) to show if the variants have the correlated effect direction. The effect sizes of Catalog GWAS and factor association tests are harmonized by [TwoSampleMR](https://mrcieu.github.io/TwoSampleMR/articles/introduction.html) R package to make the effect alleles in these two analysis identical. The LVs have more than one significant SNPs with FDR<0.2 are included in the plotting.Besides, for each plots, I fitted the points with intercept = 0. The pvalues and r-squared are shown on the plots.

6. For the traits and LVs in 5, I made an info table to show more details of the SNPs. 

7. For several LVs we are interested in, I did gene set enrichment analysis to test if the LVs are correlated with some KEGG/REACTOME pathways. I used two kind gene sets to do GSEA: 1. genes that used to compute LVs; 2. Sorting the genes in 1 by their loadings, take the top 25% as the gene set. For both gene sets, the gene scores used as input of GSEA are the gene loadings.


## SNPs after LD Clumping

After filtering by 'pval < 5e-8' and LD Clumping(r01,d1Mb), for each trait, I got :

platelet count trait contains 688 SNPs with pval<5e-8. 

white blood cell count trait contains 368 SNPs with pval<5e-8. 

myeloid white cell count trait contains 319 SNPs with pval<5e-8. 

lymphocyte count trait contains 436 SNPs with pval<5e-8. 

red blood cell count trait contains 466 SNPs with pval<5e-8. 

granulocyte count trait contains 316 SNPs with pval<5e-8. 

eosinophil count trait contains 491 SNPs with pval<5e-8. 

neutrophil count trait contains 317 SNPs with pval<5e-8. 

IBD trait contains 116 SNPs with pval<5e-8. 

Ulcerative colitist trait contains 73 SNPs with pval<5e-8. 

Crohn's disease trait contains 96 SNPs with pval<5e-8. 

asthma trait contains 186 SNPs with pval<5e-8. 

BMI trait contains 104 SNPs with pval<5e-8. 

T2D contains 14 SNPs with pval<5e-8. 





## Results - pval < 5e-8 & 120 PEER factors & covariants: 10 PCs + GTEx:Sequencing platform,Sequencing protocol,Sex + AGE

### Summary table

I used 'qvalue' R package to compute the fdr from p-values for each SNP and made a table to show the number of SNPs that pass the threshold. The thresholds are 'fdr < 0.1','fdr < 0.2','pval < 5e-8'.

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/sum_sep_tog_pval_peer_wgpc_ldcl_d1k_ageco.rdata")
DT::datatable(summary, options = list(pageLength =20))
```


### Info table

For each trait, I made a table to show the info of snps with fdr>0.2 in the factor ~ SNP + genotype pcs association test. For each trait,The LVs have more than one significant SNPs with FDR<0.2 are included. 

The suffix '_assoc' here means that results are from factor ~ SNP + genotype pcs association test. The suffix '_gwas' here means results are from original GWAS results files. For EUR.CD, EUR.IBD, EUR.UC, the effectsize_gwas here means 'ln(OR)', for others, it means 'beta'. 

'snp_ld' here means the snps that in LD with the snp in each line.'ld_r2' means the LD r-squared which is corresponding to the 'snp_ld' column. 'cis-eqtl' column indicates whether the snp is a cis-eqtl according to GTEx data. 'cis_gene_hgnc' and 'cis_gene_hgnc' is the genes that the snp influence when it act as cis-eqtl. 'func' and 'func_gene' are obtained from ANNOVAR, which indicating the snp function within the genes.

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/info_pval5e8_peer_ld_d1k_ageco.rdata")
DT::datatable(final_full, options = list(pageLength =20))
```


### Enrichment analysis

There are some promising trait-factor pairs when considering SNPs at fdr<0.2 (i.e. BMI-LV14, Ulcerative colitist - LV77,  Lymphocyte-LV66,76, Neutrophil - LV43, RBC-LV73, Asthma - LV47, WBC-LV11,68). When relaxing the cutoff to fdr<0.5, there appeared some other promising pairs(i.e. Eocinophil count - LV9, Granulocyte count - LV43,96, Myeloid white cell count -LV43, WBC-LV43).

I did ORA for all the factors mentioned above with WebGestalt. The analysis settings: for each factor, I sorted the genes their absolute loading values, taking the top 5% as the gene set. The function database here are Reactome pathway; Disgenet + GLAD4U + OMIM disease dataset; geneontology biological process. The reference set affy hugene 2 0 st v1. Minimum number of genes for a category is 5, maximum number of genes for a category is 2000(default settings). All categories that have fdr<0.01 are listed. If all the categories don't have fdr<0.01, I just listed the top 20 categories that related to each factors. 


[BMI - LV14](http://www.webgestalt.org/results/1603472381/#)

[Eocinophil - LV9](http://www.webgestalt.org/results/1603472369/#)

[Ulcerative colitist - LV77](http://www.webgestalt.org/results/1603502010/#)

[Granulocyte/Neutrophil/WBC/Myeloid white cell - LV43](http://www.webgestalt.org/results/1603472386/#)

[Granulocyte - LV96](http://www.webgestalt.org/results/1603472435/#)

[Lymphocyte-LV66](http://www.webgestalt.org/results/1603502255/#)

[Lymphocyte-LV76](http://www.webgestalt.org/results/1603472431/#)

[RBC - LV73](http://www.webgestalt.org/results/1603502267/#)

[Asthma - LV47](http://www.webgestalt.org/results/1603472390/#)

[WBC - LV11](http://www.webgestalt.org/results/1603472375/#)

[WBC - LV68](http://www.webgestalt.org/results/1603472406/#)



### Effect size plots

For each trait, I made a plot of association with LV(indicating by beta in GWAS) vs association with trait(indicating by ln(odds ratio) or beta in GWAS) to show if the variants have the correlated effect direction. The effect sizes of Catalog GWAS and factor association tests are harmonized by [TwoSampleMR](https://mrcieu.github.io/TwoSampleMR/articles/introduction.html) R package to make the effect alleles in these two analysis identical. The LVs have more than one significant SNPs with FDR<0.2 are included in the plotting.Besides, for each plots, I fitted the points with intercept = 0. The pvalues and r-squared are shown on the plots.

#### BMI 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv14_bmi_pval5e8_peer_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv79_bmi_pval5e8_peer_d1k_ageco.jpeg" width="30%">
</figure>


#### Eosinophil count 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv9_eo_pval5e8_peer_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv57_eo_pval5e8_peer_d1k_ageco.jpeg" width="30%">
</figure>

#### Crohn's disease

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv75_EUR.CD_pval5e8_peer_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv91_EUR.CD_pval5e8_peer_d1k_ageco.jpeg" width="30%">
</figure>

#### IBD

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv91_EUR.IBD_pval5e8_peer_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv120_EUR.IBD_pval5e8_peer_d1k_ageco.jpeg" width="30%">
</figure>

#### Ulcerative colitist

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv77_EUR.UC_pval5e8_peer_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv110_EUR.UC_pval5e8_peer_d1k_ageco.jpeg" width="30%">
</figure>


#### granulocyte count

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_gran_pval5e8_peer_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv79_gran_pval5e8_peer_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv96_gran_pval5e8_peer_d1k_ageco.jpeg" width="30%">
</figure>


#### lymphocyte count

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv72_lymph_pval5e8_peer_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv20_lymph_pval5e8_peer_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv44_lymph_pval5e8_peer_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv66_lymph_pval5e8_peer_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv76_lymph_pval5e8_peer_d1k_ageco.jpeg" width="30%">
</figure>

#### myeloid white cell count

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_myeloid_wbc_pval5e8_peer_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv79_myeloid_wbc_pval5e8_peer_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv72_myeloid_wbc_pval5e8_peer_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv88_myeloid_wbc_pval5e8_peer_d1k_ageco.jpeg" width="30%">
</figure>

#### neutrophil count

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv79_neut_pval5e8_peer_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_neut_pval5e8_peer_d1k_ageco.jpeg" width="30%">
</figure>

#### plt

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv50_plt_pval5e8_peer_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv81_plt_pval5e8_peer_d1k_ageco.jpeg" width="30%">
</figure>

#### rbc

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv19_rbc_pval5e8_peer_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv41_rbc_pval5e8_peer_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv73_rbc_pval5e8_peer_d1k_ageco.jpeg" width="30%">
</figure>

#### T2D

<img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv19_T2D_pval5e8_peer_d1k_ageco.jpeg" width="30%">

#### asthma

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv24_ukb.allasthma_pval5e8_peer_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv94_ukb.allasthma_pval5e8_peer_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv102_ukb.allasthma_pval5e8_peer_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv47_ukb.allasthma_pval5e8_peer_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv58_ukb.allasthma_pval5e8_peer_d1k_ageco.jpeg" width="30%">
</figure>

#### wbc

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_wbc_pval5e8_peer_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv11_wbc_pval5e8_peer_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv68_wbc_pval5e8_peer_d1k_ageco.jpeg" width="30%">
</figure>

### Effect size plots- more SNPs

For some promising trait-factor pairs , I relaxed the fdr threshold of the SNPs that used to make effect size plots(from 0.2 to 0.3/0.5)


#### BMI 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv14_bmi_pval5e8_peer_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv14_bmi_pval5e8_peer_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv14_bmi_pval5e8_peer_d1k_ageco_fdr05.jpeg" width="30%">
</figure>

#### Eosinophil count 

The Eo ~lv9 is not very promising pairs when considering SNPs at fdr<0.2, but the fitting result at fdr <0.5 is better than the former result . So I post the plots here too.


<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv9_eo_pval5e8_peer_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv9_eo_pval5e8_peer_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv9_eo_pval5e8_peer_d1k_ageco_fdr05.jpeg" width="30%">
</figure>


#### Ulcerative colitist

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv77_EUR.UC_pval5e8_peer_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv77_EUR.UC_pval5e8_peer_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv77_EUR.UC_pval5e8_peer_d1k_ageco_fdr05.jpeg" width="30%">
</figure>

#### Granulocyte count 

The gran ~lv43,96 is not very promising pairs when considering SNPs at fdr<0.2, but the fitting result at fdr <0.5 is better than the former result . So I post the plots here too.


<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_gran_pval5e8_peer_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_gran_pval5e8_peer_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_gran_pval5e8_peer_d1k_ageco_fdr05.jpeg" width="30%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv96_gran_pval5e8_peer_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv96_gran_pval5e8_peer_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv96_gran_pval5e8_peer_d1k_ageco_fdr05.jpeg" width="30%">
</figure>


#### lymphocyte count

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv66_lymph_pval5e8_peer_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv66_lymph_pval5e8_peer_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv66_lymph_pval5e8_peer_d1k_ageco_fdr05.jpeg" width="30%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv76_lymph_pval5e8_peer_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv76_lymph_pval5e8_peer_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv76_lymph_pval5e8_peer_d1k_ageco_fdr05.jpeg" width="30%">
</figure>

#### Myeloid white cell count 

The myeloid_wbc ~lv43 is not very promising pairs when considering SNPs at fdr<0.2, but the fitting result at fdr <0.5 is better than the former result . So I post the plots here too.


<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_myeloid_wbc_pval5e8_peer_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_myeloid_wbc_pval5e8_peer_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_myeloid_wbc_pval5e8_peer_d1k_ageco_fdr05.jpeg" width="30%">
</figure>

#### Neutrophil count

The neut ~lv43 is not very promising pairs when considering SNPs at fdr<0.2, but the fitting result at fdr <0.5 is better than the former result . So I post the plots here too.


<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_neut_pval5e8_peer_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_neut_pval5e8_peer_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_neut_pval5e8_peer_d1k_ageco_fdr05.jpeg" width="30%">
</figure>


#### red blood cell count

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv73_rbc_pval5e8_peer_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv73_rbc_pval5e8_peer_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv73_rbc_pval5e8_peer_d1k_ageco_fdr05.jpeg" width="30%">
</figure>


#### asthma

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv47_ukb.allasthma_pval5e8_peer_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv47_ukb.allasthma_pval5e8_peer_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv47_ukb.allasthma_pval5e8_peer_d1k_ageco_fdr05.jpeg" width="30%">
</figure>


#### white blood cell count

The wbc ~lv43 is not very promising pairs when considering SNPs at fdr<0.2, but the fitting result at fdr <0.5 is better than the former result . So I post the plots here too.


<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv11_wbc_pval5e8_peer_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv11_wbc_pval5e8_peer_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv11_wbc_pval5e8_peer_d1k_ageco_fdr05.jpeg" width="30%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv68_wbc_pval5e8_peer_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv68_wbc_pval5e8_peer_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv68_wbc_pval5e8_peer_d1k_ageco_fdr05.jpeg" width="30%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_wbc_pval5e8_peer_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_wbc_pval5e8_peer_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_wbc_pval5e8_peer_d1k_ageco_fdr05.jpeg" width="30%">
</figure>


### Effect size plots - checking the reverse causality

To check if the effect size correlation is due to reverse causality: i.e. trait -> LV (trait causally affect LV), instead of LV -> trait (which is what we like to see). I used all SNPs associated with traits(pval<5E-8). The x-axis is the effects of these SNPs on trait, and y-axis is the effects on LV. 

Some pair show p < 0.05, the result may be driven by the possible causal effect of LV -> trait. To test this, I removed the SNPs that are associated with LVs at FDR < 0.2 and made the plots again.

#### BMI 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv14_bmi_pval5e8_peer_d1k_ageco_reverse.jpeg" width="30%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv14_bmi_pval5e8_peer_d1k_ageco_reverse_remove.jpeg" width="30%">
</figure>

#### Eosinophil count 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv9_eo_pval5e8_peer_d1k_ageco_reverse.jpeg" width="30%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv9_eo_pval5e8_peer_d1k_ageco_reverse_remove.jpeg" width="30%">
</figure>

#### Ulcerative colistist

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv77_EUR.UC_pval5e8_peer_d1k_ageco_reverse.jpeg" width="30%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv77_EUR.UC_pval5e8_peer_d1k_ageco_reverse_remove.jpeg" width="30%">
</figure>


#### Granulocyte count

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_gran_pval5e8_peer_d1k_ageco_reverse.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv96_gran_pval5e8_peer_d1k_ageco_reverse.jpeg" width="30%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_gran_pval5e8_peer_d1k_ageco_reverse_remove.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv96_gran_pval5e8_peer_d1k_ageco_reverse_remove.jpeg" width="30%">
</figure>

#### lymphocyte count

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv66_lymph_pval5e8_peer_d1k_ageco_reverse.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv76_lymph_pval5e8_peer_d1k_ageco_reverse.jpeg" width="30%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv66_lymph_pval5e8_peer_d1k_ageco_reverse_remove.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv76_lymph_pval5e8_peer_d1k_ageco_reverse_remove.jpeg" width="30%">
</figure>


#### Myeloid white cell count

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_myeloid_wbc_pval5e8_peer_d1k_ageco_reverse.jpeg" width="30%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_myeloid_wbc_pval5e8_peer_d1k_ageco_reverse_remove.jpeg" width="30%">
</figure>

#### Neutrophil count

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_neut_pval5e8_peer_d1k_ageco_reverse.jpeg" width="30%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_neut_pval5e8_peer_d1k_ageco_reverse_remove.jpeg" width="30%">
</figure>

#### red blood cell count

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv73_rbc_pval5e8_peer_d1k_ageco_reverse.jpeg" width="30%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv73_rbc_pval5e8_peer_d1k_ageco_reverse_remove.jpeg" width="30%">
</figure>

#### asthma

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv47_ukb.allasthma_pval5e8_peer_d1k_ageco_reverse.jpeg" width="30%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv47_ukb.allasthma_pval5e8_peer_d1k_ageco_reverse_remove.jpeg" width="30%">
</figure>

#### white blood cell count

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv11_wbc_pval5e8_peer_d1k_ageco_reverse.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv68_wbc_pval5e8_peer_d1k_ageco_reverse.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_wbc_pval5e8_peer_d1k_ageco_reverse.jpeg" width="30%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv11_wbc_pval5e8_peer_d1k_ageco_reverse_remove.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv68_wbc_pval5e8_peer_d1k_ageco_reverse_remove.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_wbc_pval5e8_peer_d1k_ageco_reverse_remove.jpeg" width="30%">
</figure>


### QQplots - promising pairs

#### BMI 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qq_unif_lv14_bmi_peer_d1k_ageco.jpeg" width="30%">
</figure>

#### Eosinophil count 

The Eo ~lv9 is not very promising pairs when considering SNPs at fdr<0.2, but the fitting result at fdr <0.5 is better than the former result . So I post the plots here too.


<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qq_unif_lv9_eo_peer_d1k_ageco.jpeg" width="30%">
</figure>


#### Ulcerative colitist

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qq_unif_lv77_EUR.UC_peer_d1k_ageco.jpeg" width="30%">
</figure>

#### Granulocyte count 

The gran ~lv43,96 is not very promising pairs when considering SNPs at fdr<0.2, but the fitting result at fdr <0.5 is better than the former result . So I post the plots here too.


<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qq_unif_lv43_gran_peer_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qq_unif_lv96_gran_peer_d1k_ageco.jpeg" width="30%">
</figure>



#### lymphocyte count

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qq_unif_lv66_lymph_peer_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qq_unif_lv76_lymph_peer_d1k_ageco.jpeg" width="30%">
</figure>


#### Myeloid white cell count 

The myeloid_wbc ~lv43 is not very promising pairs when considering SNPs at fdr<0.2, but the fitting result at fdr <0.5 is better than the former result . So I post the plots here too.


<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qq_unif_lv43_myeloid_wbc_peer_d1k_ageco.jpeg" width="30%">
</figure>

#### Neutrophil count

The neut ~lv43 is not very promising pairs when considering SNPs at fdr<0.2, but the fitting result at fdr <0.5 is better than the former result . So I post the plots here too.


<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qq_unif_lv43_neut_peer_d1k_ageco.jpeg" width="30%">
</figure>


#### red blood cell count

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qq_unif_lv73_rbc_peer_d1k_ageco.jpeg" width="30%">
</figure>


#### asthma

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qq_unif_lv47_ukb.allasthma_peer_d1k_ageco.jpeg" width="30%">
</figure>


#### white blood cell count

The wbc ~lv43 is not very promising pairs when considering SNPs at fdr<0.2, but the fitting result at fdr <0.5 is better than the former result . So I post the plots here too.


<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qq_unif_lv11_wbc_peer_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qq_unif_lv68_wbc_peer_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qq_unif_lv43_wbc_peer_d1k_ageco.jpeg" width="30%">
</figure>


### Colocalization

The colocalization analysis was performed using the approximate Bayes factor test implemented in the Coloc package. Coloc computes five posterior probabilities (PP0, PP1, PP2, PP3 and PP4), each corresponding to a hypothesis: H0, no association with either trait; H1, association with trait 1 but not with trait 2; H2, association with trait 2 but not with trait 1; H3, association with trait 1 and trait 2, two independent SNPs; H4, association with trait 1 and trait 2, one shared SNP. We ran Coloc with the default parameters and used PP4 to assess evidence of colocalization. We visualized the colocalization of factor - QTLs and GWAS associations using the [LocusCompareR package](https://github.com/boxiangliu/locuscomparer).

Chromosome selection: I first sorted the SNPs by their pvalues from factor association tests, then I selected the first two SNPs and chose their chromosome to do the colocalization test adn visualization. 

#### BMI 

LV14 top1 snp and its chromosome

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/bmi_lv14_coloc.results_peer_d1k.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "pvalues in coloclization")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/coloc_bmi_lv14_peer_d1k.jpeg" width="80%">
</figure>


LV14 the second snp and its chromosome

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/bmi_lv14_coloc_top2.results_peer_d1k.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "pvalues in coloclization")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/coloc_bmi_lv14_peer_d1k_top2.jpeg" width="80%">
</figure>

#### Eosinophil count 

The Eo ~lv9 is not very promising pairs when considering SNPs at fdr<0.2, but the fitting result at fdr <0.5 is better than the former result . So I post the plots here too.


LV9 top1 snp and its chromosome

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/eo_lv9_coloc.results_peer_d1k.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "pvalues in coloclization")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/coloc_eo_lv9_peer_d1k.jpeg" width="80%">
</figure>


LV9 the second snp and its chromosome

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/eo_lv9_coloc_top2.results_peer_d1k.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "pvalues in coloclization")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/coloc_eo_lv9_peer_d1k_top2.jpeg" width="80%">
</figure>


#### Ulcerative colitist

lv77 top1 snp and its chromosome

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/EUR.UC_lv77_coloc.results_peer_d1k.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "pvalues in coloclization")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/coloc_EUR.UC_lv77_peer_d1k.jpeg" width="80%">
</figure>


lv77 the second snp and its chromosome

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/EUR.UC_lv77_coloc_top2.results_peer_d1k.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "pvalues in coloclization")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/coloc_EUR.UC_lv77_peer_d1k_top2.jpeg" width="80%">
</figure>

#### Granulocyte count 

The gran ~lv43,96 is not very promising pairs when considering SNPs at fdr<0.2, but the fitting result at fdr <0.5 is better than the former result . So I post the plots here too.


lv43 top1 snp and its chromosome

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/gran_lv43_coloc.results_peer_d1k.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "pvalues in coloclization")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/coloc_gran_lv43_peer_d1k.jpeg" width="80%">
</figure>


lv43 the second snp and its chromosome

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/gran_lv43_coloc_top2.results_peer_d1k.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "pvalues in coloclization")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/coloc_gran_lv43_peer_d1k_top2.jpeg" width="80%">
</figure>


lv96 top1 snp and its chromosome

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/gran_lv96_coloc.results_peer_d1k.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "pvalues in coloclization")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/coloc_gran_lv96_peer_d1k.jpeg" width="80%">
</figure>


lv96 the second snp and its chromosome

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/gran_lv96_coloc_top2.results_peer_d1k.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "pvalues in coloclization")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/coloc_gran_lv96_peer_d1k_top2.jpeg" width="80%">
</figure>


#### lymphocyte count

lv66 top1 snp and its chromosome

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/lymph_lv66_coloc.results_peer_d1k.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "pvalues in coloclization")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/coloc_lymph_lv66_peer_d1k.jpeg" width="80%">
</figure>


lv66 the second snp and its chromosome

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/lymph_lv66_coloc_top2.results_peer_d1k.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "pvalues in coloclization")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/coloc_lymph_lv66_peer_d1k_top2.jpeg" width="80%">
</figure>


lv76 top1 snp and its chromosome

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/lymph_lv76_coloc.results_peer_d1k.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "pvalues in coloclization")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/coloc_lymph_lv76_peer_d1k.jpeg" width="80%">
</figure>


lv76 the second snp and its chromosome

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/lymph_lv76_coloc_top2.results_peer_d1k.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "pvalues in coloclization")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/coloc_lymph_lv76_peer_d1k_top2.jpeg" width="80%">
</figure>

#### Myeloid white cell count 

The myeloid_wbc ~lv43 is not very promising pairs when considering SNPs at fdr<0.2, but the fitting result at fdr <0.5 is better than the former result . So I post the plots here too.

lv43 top1 snp and its chromosome

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/myeloid_wbc_lv43_coloc.results_peer_d1k.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "pvalues in coloclization")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/coloc_myeloid_wbc_lv43_peer_d1k.jpeg" width="80%">
</figure>


lv43 the second snp and its chromosome

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/myeloid_wbc_lv43_coloc_top2.results_peer_d1k.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "pvalues in coloclization")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/coloc_myeloid_wbc_lv43_peer_d1k_top2.jpeg" width="80%">
</figure>

#### Neutrophil count

The neut ~lv43 is not very promising pairs when considering SNPs at fdr<0.2, but the fitting result at fdr <0.5 is better than the former result . So I post the plots here too.

lv43 top1 snp and its chromosome

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/neut_lv43_coloc.results_peer_d1k.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "pvalues in coloclization")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/coloc_neut_lv43_peer_d1k.jpeg" width="80%">
</figure>


lv43 the second snp and its chromosome

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/neut_lv43_coloc_top2.results_peer_d1k.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "pvalues in coloclization")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/coloc_neut_lv43_peer_d1k_top2.jpeg" width="80%">
</figure>

#### red blood cell count

lv73 top1 snp and its chromosome

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/rbc_lv73_coloc.results_peer_d1k.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "pvalues in coloclization")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/coloc_rbc_lv73_peer_d1k.jpeg" width="80%">
</figure>


lv73 the second snp and its chromosome

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/rbc_lv73_coloc_top2.results_peer_d1k.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "pvalues in coloclization")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/coloc_rbc_lv73_peer_d1k_top2.jpeg" width="80%">
</figure>


#### asthma

The asthma GWAS data doesn't contain the MAF of SNPs which is required in 'coloc' package when performing colocalization analysis. So I just post the plots here. 

lv47 top1 snp and its chromosome

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/ukb.allasthma_lv47_coloc.results_peer_d1k.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "pvalues in coloclization")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/coloc_ukb.allasthma_lv47_peer_d1k.jpeg" width="80%">
</figure>


lv47 the second snp and its chromosome

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/ukb.allasthma_lv47_coloc_top2.results_peer_d1k.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "pvalues in coloclization")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/coloc_ukb.allasthma_lv47_peer_d1k_top2.jpeg" width="80%">
</figure>

#### white blood cell count

The wbc ~lv43 is not very promising pairs when considering SNPs at fdr<0.2, but the fitting result at fdr <0.5 is better than the former result . So I post the plots here too.

lv11 top1 snp and its chromosome

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/wbc_lv11_coloc.results_peer_d1k.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "pvalues in coloclization")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/coloc_wbc_lv11_peer_d1k.jpeg" width="80%">
</figure>


lv11 the second snp and its chromosome

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/wbc_lv11_coloc_top2.results_peer_d1k.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "pvalues in coloclization")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/coloc_wbc_lv11_peer_d1k_top2.jpeg" width="80%">
</figure>


lv68 top1 snp and its chromosome

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/wbc_lv68_coloc.results_peer_d1k.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "pvalues in coloclization")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/coloc_wbc_lv68_peer_d1k.jpeg" width="80%">
</figure>


lv68 the second snp and its chromosome

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/wbc_lv68_coloc_top2.results_peer_d1k.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "pvalues in coloclization")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/coloc_wbc_lv68_peer_d1k_top2.jpeg" width="80%">
</figure>


lv43 top1 snp and its chromosome

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/wbc_lv43_coloc.results_peer_d1k.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "pvalues in coloclization")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/coloc_wbc_lv43_peer_d1k.jpeg" width="80%">
</figure>


lv43 the second snp and its chromosome

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/wbc_lv43_coloc_top2.results_peer_d1k.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "pvalues in coloclization")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/coloc_wbc_lv43_peer_d1k_top2.jpeg" width="80%">
</figure>


## Results(former results) - pval < 5e-8 & 60 PEER factors & covariants: 10 whole genome PCs

In this part, the PEER factors are consistent with the the [former PEER factors](https://xsun1229.github.io/factor_analysis/eval_peer.html)


### Summary table

I used 'qvalue' R package to compute the fdr from p-values for each SNP and made a table to show the number of SNPs that pass the threshold. The thresholds are 'fdr < 0.1','fdr < 0.2','pval < 5e-8'.

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/sum_sep_tog_pval_peer_wgpc_ldcl_d1k.rdata")
DT::datatable(summary, options = list(pageLength =20))
```

### Info table

For each trait, I made a table to show the info of snps with fdr>0.2 in the factor ~ SNP + genotype pcs association test. For each trait,The LVs have more than one significant SNPs with FDR<0.2 are included. 

The suffix '_assoc' here means that results are from factor ~ SNP + genotype pcs association test. The suffix '_gwas' here means results are from original GWAS results files. For EUR.CD, EUR.IBD, EUR.UC, the effectsize_gwas here means 'ln(OR)', for others, it means 'beta'. 

'snp_ld' here means the snps that in LD with the snp in each line.'ld_r2' means the LD r-squared which is corresponding to the 'snp_ld' column. 'cis-eqtl' column indicates whether the snp is a cis-eqtl according to GTEx data. 'cis_gene_hgnc' and 'cis_gene_hgnc' is the genes that the snp influence when it act as cis-eqtl. 'func' and 'func_gene' are obtained from ANNOVAR, which indicating the snp function within the genes.

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/info_pval5e8_peer_ld_d1k.rdata")
DT::datatable(final_full, options = list(pageLength =20))
```


### Effect size plots

For each trait, I made a plot of association with LV(indicating by beta in GWAS) vs association with trait(indicating by ln(odds ratio) or beta in GWAS) to show if the variants have the correlated effect direction. The effect sizes of Catalog GWAS and factor association tests are harmonized by [TwoSampleMR](https://mrcieu.github.io/TwoSampleMR/articles/introduction.html) R package to make the effect alleles in these two analysis identical. The LVs have more than one significant SNPs with FDR<0.2 are included in the plotting.Besides, for each plots, I fitted the points with intercept = 0. The pvalues and r-squared are shown on the plots.

#### BMI 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv14_bmi_pval5e8_d1k_peer.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv40_bmi_pval5e8_d1k_peer.jpeg" width="30%">
</figure>


#### Eosinophil count 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv57_eo_pval5e8_d1k_peer.jpeg" width="30%">
</figure>

#### Crohn's disease

None of the LVs have >1 SNPs at FDR<0.2.

#### IBD

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv6_EUR.IBD_pval5e8_d1k_peer.jpeg" width="30%">
</figure>

#### Ulcerative colitist

None of the LVs have >1 SNPs at FDR<0.2.


#### granulocyte count

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_gran_pval5e8_d1k_peer.jpeg" width="30%">
</figure>


#### lymphocyte count

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv44_lymph_pval5e8_d1k_peer.jpeg" width="30%">
</figure>

#### myeloid white cell count

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_myeloid_wbc_pval5e8_d1k_peer.jpeg" width="30%">
</figure>

#### neutrophil count

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_neut_pval5e8_d1k_peer.jpeg" width="30%">
</figure>

#### plt

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv7_plt_pval5e8_d1k_peer.jpeg" width="30%">
</figure>

#### rbc

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv11_rbc_pval5e8_d1k_peer.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv28_rbc_pval5e8_d1k_peer.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv41_rbc_pval5e8_d1k_peer.jpeg" width="30%">
</figure>


#### T2D

None of the LVs have >1 SNPs at FDR<0.2.

#### asthma

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv24_ukb.allasthma_pval5e8_d1k_peer.jpeg" width="30%">
</figure>

#### wbc

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_wbc_pval5e8_d1k_peer.jpeg" width="30%">
</figure>


### effect size plots -- more SNPs

For some promising trait-factor pairs (i.e. BMI-LV14, LV40,LV76, RBC-LV28), I relaxed the fdr threshold of the SNPs that used to make effect size plots(from 0.2 to 0.3/0.5)

I also made a plot to show the the distribution of the SNPs' fdr.

#### BMI 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv14_bmi_pval5e8_d1k_peer.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv14_bmi_pval5e8_d1k_peer_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv14_bmi_pval5e8_d1k_peer_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/fdrhist_lv14_bmi_pval5e8_d1k_peer.jpeg" width="30%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv40_bmi_pval5e8_d1k_peer.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv40_bmi_pval5e8_d1k_peer_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv40_bmi_pval5e8_d1k_peer_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/fdrhist_lv40_bmi_pval5e8_d1k_peer.jpeg" width="30%">
</figure>

#### IBD

The IBD~lv6 is not very promising pairs when considering SNPs at fdr<0.2, but the fitting result at fdr <0.5 is better than the former result. So I post the plots here too.

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv6_EUR.IBD_pval5e8_d1k_peer.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv6_EUR.IBD_pval5e8_d1k_peer_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv6_EUR.IBD_pval5e8_d1k_peer_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/fdrhist_lv6_EUR.IBD_pval5e8_d1k_peer.jpeg" width="30%">
</figure>

#### myeloid white cell count

The myeloid white cell count~lv6 is not very promising pairs when considering SNPs at fdr<0.2, but the fitting result at fdr <0.5 is better than the former result. So I post the plots here too.

<figure class="half">
     <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_myeloid_wbc_pval5e8_d1k_peer.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_myeloid_wbc_pval5e8_d1k_peer_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_myeloid_wbc_pval5e8_d1k_peer_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/fdrhist_lv43_myeloid_wbc_pval5e8_d1k_peer.jpeg" width="30%">
</figure>

#### neutrophil count

The neutrophil count~lv6 is not very promising pairs when considering SNPs at fdr<0.2, but the fitting result at fdr <0.5 is better than the former result. So I post the plots here too.

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_neut_pval5e8_d1k_peer.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_neut_pval5e8_d1k_peer_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_neut_pval5e8_d1k_peer_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/fdrhist_lv43_neut_pval5e8_d1k_peer.jpeg" width="30%">
</figure>


#### red blood cell count 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv28_rbc_pval5e8_d1k_peer.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv28_rbc_pval5e8_d1k_peer_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv28_rbc_pval5e8_d1k_peer_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/fdrhist_lv28_rbc_pval5e8_d1k_peer.jpeg" width="30%">
</figure>


#### wbc

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_wbc_pval5e8_d1k_peer.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_wbc_pval5e8_d1k_peer_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_wbc_pval5e8_d1k_peer_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/fdrhist_lv43_wbc_pval5e8_d1k_peer.jpeg" width="30%">
</figure>

### Effect size plots - checking the reverse causality

To check if the effect size correlation is due to reverse causality: i.e. trait -> LV (trait causally affect LV), instead of LV -> trait (which is what we like to see). I used all SNPs associated with traits(pval<5E-8). The x-axis is the effects of these SNPs on trait, and y-axis is the effects on LV. 

Some pair show p < 0.05, the result may be driven by the possible causal effect of LV -> trait. To test this, I removed the SNPs that are associated with LVs at FDR < 0.2 and made the plots again.

#### BMI 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv14_bmi_pval5e8_d1k_reverse_peer.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv40_bmi_pval5e8_d1k_reverse_peer.jpeg" width="30%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv14_bmi_pval5e8_d1k_reverse_remove_peer.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv40_bmi_pval5e8_d1k_reverse_remove_peer.jpeg" width="30%">
</figure>

#### IBD

The IBD~lv6 is not very promising pairs when considering SNPs at fdr<0.2, but the fitting result at fdr <0.5 is better than the former result. So I post the plots here too.

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv6_EUR.IBD_pval5e8_d1k_reverse_peer.jpeg" width="30%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv6_EUR.IBD_pval5e8_d1k_reverse_remove_peer.jpeg" width="30%">
</figure>

#### myeloid white cell count

The myeloid white cell count~lv6 is not very promising pairs when considering SNPs at fdr<0.2, but the fitting result at fdr <0.5 is better than the former result. So I post the plots here too.

<figure class="half">
     <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_myeloid_wbc_pval5e8_d1k_reverse_peer.jpeg" width="30%">
</figure>

<figure class="half">
     <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_myeloid_wbc_pval5e8_d1k_reverse_remove_peer.jpeg" width="30%">
</figure>

#### neutrophil count

The neutrophil count~lv6 is not very promising pairs when considering SNPs at fdr<0.2, but the fitting result at fdr <0.5 is better than the former result. So I post the plots here too.

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_neut_pval5e8_d1k_reverse_peer.jpeg" width="30%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_neut_pval5e8_d1k_reverse_remove_peer.jpeg" width="30%">
</figure>


#### red blood cell count 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv28_rbc_pval5e8_d1k_reverse_peer.jpeg" width="30%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv28_rbc_pval5e8_d1k_reverse_remove_peer.jpeg" width="30%">
</figure>


#### wbc

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_wbc_pval5e8_d1k_reverse_peer.jpeg" width="30%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_wbc_pval5e8_d1k_reverse_remove_peer.jpeg" width="30%">
</figure>


