---
title: "catalog_gwas_pliercanon_sep_ldclumping_r01d1000_adipose_sub"
author: "XSun"
date: "2020-11-19"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---


## Introduction

The analysis in this part is the same with [PLIER-Canonical](https://xsun1229.github.io/factor_analysis/catalog_gwas_pliercanon_sep_ldclumping_r01d1000.html). But in this part, I used Adipose Subcutaneous expression data. It contains 581 samples. 


## Material and Methods

This part is the same with [PLIER-Canonical](https://xsun1229.github.io/factor_analysis/catalog_gwas_pliercanon_sep_ldclumping_r01d1000.html).


## Snps after filter

After filtering by 'pval < 5e-8' and LD Clumping, for each trait, I got :

BMI trait contains 102 SNPs with pval<5e-8. 

T2D contains 14 SNPs with pval<5e-8. T2D_2 contains 4 SNPs with pval<5e-8.

HDL trait contains 225 SNPs with pval<5e-8.

LDL trait contains 203 SNPs with pval<5e-8.

WHR trait contains 37 SNPs with pval<5e-8.


## Results - pval < 5e-8 & 120 PEER factors & covariants: 10 PCs + GTEx:Sequencing platform,Sequencing protocol,Sex + AGE

### Summary table

I used 'qvalue' R package to compute the fdr from p-values for each SNP and made a table to show the number of SNPs that pass the threshold. The thresholds are 'fdr < 0.1','fdr < 0.2','pval < 5e-8'. The 'num_significant_pairs' indicates the number of significant pairs under each threshold. If a trait~factor pair has as least 1 significant SNP, we named it as 'significant pair'. 

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/sum_sep_tog_pval_pliercanon_wgpc_ldcl_d1k_ageco_adp.rdata")
DT::datatable(summary, options = list(pageLength =20))
```


### Info table

For each trait, I made a table to show the info of snps with fdr>0.2 in the factor ~ SNP + genotype pcs association test. For each trait,The LVs have more than one significant SNPs with FDR<0.2 are included. 

The suffix '_assoc' here means that results are from factor ~ SNP + genotype pcs association test. The suffix '_gwas' here means results are from original GWAS results files. 

'snp_ld' here means the snps that in LD with the snp in each line.'ld_r2' means the LD r-squared which is corresponding to the 'snp_ld' column. 'cis-eqtl' column indicates whether the snp is a cis-eqtl according to GTEx data. 'cis_gene_hgnc' and 'cis_gene_hgnc' is the genes that the snp influence when it act as cis-eqtl. 'func' and 'func_gene' are obtained from ANNOVAR, which indicating the snp function within the genes.


```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/info_pval5e8_pliercanon_ld_d1k_ageco_adp.rdata")
DT::datatable(final_full, options = list(pageLength =20))
```


### Enrichment analysis

There is only one promising pair: LDL-LV35

**LDL - LV35**

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/lv35top025_pliercanon_adp.rdata")
DT::datatable(ORA_results, options = list(pageLength =5))
```


### Effect size plots & Resampling

For each trait, I made a plot of association with LV(indicating by beta in GWAS) vs association with trait(indicating by ln(odds ratio) or beta in GWAS) to show if the variants have the correlated effect direction. The effect sizes of Catalog GWAS and factor association tests are harmonized by [TwoSampleMR](https://mrcieu.github.io/TwoSampleMR/articles/introduction.html) R package to make the effect alleles in these two analysis identical. The LVs have more than one significant SNPs with FDR<0.2 are included in the plotting.Besides, for each plots, I fitted the points with intercept = 0. The pvalues and r-squared are shown on the plots.


#### BMI 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv78_bmi_pval5e8_pliercanon_d1k_ageco_adp.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv78_bmi_pval5e8_pliercanon_d1k_ageco_fdr03_adp.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv78_bmi_pval5e8_pliercanon_d1k_ageco_fdr05_adp.jpeg" width="30%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_78_bmi_adp.jpeg" width="30%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv100_bmi_pval5e8_pliercanon_d1k_ageco_adp.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv100_bmi_pval5e8_pliercanon_d1k_ageco_fdr03_adp.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv100_bmi_pval5e8_pliercanon_d1k_ageco_fdr05_adp.jpeg" width="30%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_100_bmi_adp.jpeg" width="30%">
</figure>

#### HDL

None of the LVs have >1 SNPs at FDR<0.2.


#### LDL

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv35_LDL_pval5e8_pliercanon_d1k_ageco_adp.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv35_LDL_pval5e8_pliercanon_d1k_ageco_fdr03_adp.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv35_LDL_pval5e8_pliercanon_d1k_ageco_fdr05_adp.jpeg" width="30%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_35_LDL_adp.jpeg" width="30%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv94_LDL_pval5e8_pliercanon_d1k_ageco_adp.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv94_LDL_pval5e8_pliercanon_d1k_ageco_fdr03_adp.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv94_LDL_pval5e8_pliercanon_d1k_ageco_fdr05_adp.jpeg" width="30%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_94_LDL_adp.jpeg" width="30%">
</figure>


#### T2D_2

None of the LVs have >1 SNPs at FDR<0.2.

#### T2D

None of the LVs have >1 SNPs at FDR<0.2.


#### WHR

None of the LVs have >1 SNPs at FDR<0.2.




#### LDL

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv35_LDL_pval5e8_pliercanon_d1k_ageco_reverse_adp.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv35_LDL_pval5e8_pliercanon_d1k_ageco_reverse_remove_adp.jpeg" width="30%">
</figure>




### QQplots - promising pairs


#### LDL

LV35

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qq_unif_lv35_LDL_pliercanon_adp.jpeg" width="30%">
</figure>



### Colocalization -- 500kb

The colocalization analysis was performed using the approximate Bayes factor test implemented in the Coloc package. Coloc computes five posterior probabilities (PP0, PP1, PP2, PP3 and PP4), each corresponding to a hypothesis: H0, no association with either trait; H1, association with trait 1 but not with trait 2; H2, association with trait 2 but not with trait 1; H3, association with trait 1 and trait 2, two independent SNPs; H4, association with trait 1 and trait 2, one shared SNP. We ran Coloc with the default parameters and used PP4 to assess evidence of colocalization. We visualized the colocalization of factor - QTLs and GWAS associations using the [LocusCompareR package](https://github.com/boxiangliu/locuscomparer).

Chromosome selection: I first sorted the SNPs by their pvalues from factor association tests, then I selected the first two SNPs and chose their chromosome to do the colocalization test adn visualization. 


#### LDL

lv35

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/LDL_35_20816003.coloc.results_adp_500.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "pvalues in coloclization")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/LDL_35_20816003.coloc_adp_500.jpeg" width="80%">
</figure>

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/LDL_35_45149969.coloc.results_adp_500.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "pvalues in coloclization")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/LDL_35_45149969.coloc_adp_500.jpeg" width="80%">
</figure>





### Colocalization - 200 kb

The colocalization analysis was performed using the approximate Bayes factor test implemented in the Coloc package. Coloc computes five posterior probabilities (PP0, PP1, PP2, PP3 and PP4), each corresponding to a hypothesis: H0, no association with either trait; H1, association with trait 1 but not with trait 2; H2, association with trait 2 but not with trait 1; H3, association with trait 1 and trait 2, two independent SNPs; H4, association with trait 1 and trait 2, one shared SNP. We ran Coloc with the default parameters and used PP4 to assess evidence of colocalization. We visualized the colocalization of factor - QTLs and GWAS associations using the [LocusCompareR package](https://github.com/boxiangliu/locuscomparer).

Chromosome selection: I first sorted the SNPs by their pvalues from factor association tests, then I selected the first two SNPs and chose their chromosome to do the colocalization test adn visualization. 


#### LDL

lv35

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/LDL_35_20816003.coloc.results_adp.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "pvalues in coloclization")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/LDL_35_20816003.coloc_adp.jpeg" width="80%">
</figure>

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/LDL_35_45149969.coloc.results_adp.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "pvalues in coloclization")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/LDL_35_45149969.coloc_adp.jpeg" width="80%">
</figure>





