---
title: "catalog_gwas_pliercanon_sep_ldclumping_r01d1000"
author: "XSun"
date: "2020-10-14"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

## Introduction

In this part, I considered the traits separately. I selected the SNPs with pval < 5e-8 for each traits. Then, I did LD Clumping for these SNPs to eliminate the LD and select a smaller subset of SNPs. After that, I did association tests for the plier_canonical factors with the SNPs. I also did colozalization analysis for some significant factor~snp pairs.


## Material and Methods

1. Catalog GWAS data:  

- platelet count (plt), white blood cell count, myeloid white cell count (myeloid wbc), lymphocyte count, red blood cell count(wbc), granulocyte count(gran), eosinophil count(eo), neutrophil count (neut) from *Astle WJ, Elding H, Jiang T, et al. The Allelic Landscape of Human Blood Cell Trait Variation and Links to Common Complex Disease. Cell. 2016;167(5):1415-1429.e19. doi:10.1016/j.cell.2016.10.042*. 

- T2D. I first used data from our lab collaction *Morris et al. Large-scale association analysis provides insights into the genetic architecture and pathophysiology of type 2 diabetes. Nat Genet. 2012 Sep;44(9):981-90. doi: 10.1038/ng.2383. Epub 2012 Aug 12. PMID: 22885922; PMCID: PMC3442244.* but it doesn't contain MAF info of variants.So I added this from [GWAS Catalog](https://www.ebi.ac.uk/gwas/home): *Wood AR et al. Variants in the FTO and CDKAL1 loci have recessive effects on risk of obesity and type 2 diabetes, respectively. Diabetologia. 2016 Jun;59(6):1214-21. doi: 10.1007/s00125-016-3908-5. Epub 2016 Mar 10. PMID: 26961502; PMCID: PMC4869698.*.

- T1D data are also from lab collection: *Onengut-Gumuscu S et al. Fine mapping of type 1 diabetes susceptibility loci and evidence for colocalization of causal variants with lymphoid gene enhancers. Nat Genet. 2015 Apr;47(4):381-6. doi: 10.1038/ng.3245. Epub 2015 Mar 9. PMID: 25751624; PMCID: PMC4380767.*

- Asthma. I first used data from our lab collaction *Zhu et al, Shared Genetics of Asthma and Mental Health Disorders: A Large-Scale Genome-Wide Cross-Trait Analysis. European Respiratory Journal, 2019 (PMID: 31619474)* but it doesn't contain MAF info of variants.So I added this from [GWAS Catalog](https://www.ebi.ac.uk/gwas/home): *Manuel A.R. et al. Genetic Architectures of Childhood- and Adult-Onset Asthma Are Partly Distinct,The American Journal of Human Genetics,Volume 104, Issue 4,2019,Pages 665-684,ISSN 0002-9297,https://doi.org/10.1016/j.ajhg.2019.02.022.*.

- IBD,Ulcerative colitist(UC),Crohn's disease(CD) data are from lab collection *Liu, van Sommeren et al, Nature Genetics, 2015*

- Waist-hip ratio data are from *Shungin et al. New genetic loci link adipose and insulin biology to body fat distribution. Nature. 2015 Feb 12;518(7538):187-196. doi: 10.1038/nature14132. PMID: 25673412; PMCID: PMC4338562*.

- BMI data are also from lab collection: *Locke AE, Kahali B, Berndt SI, Justice AE, Pers TH, Day FR, Powell C, Vedantam S, Buchkovich ML, Yang J, Croteau-Chonka DC, Esko T et al. (2015). Genetic studies of body mass index yield new insights for obesity biology. Nature 518, 197-206*

- HDL & LDL & TG &TC data are downloaded from [GWAS Catalog](https://www.ebi.ac.uk/gwas/home).  *Global Lipids Genetics Consortium., Willer, C., Schmidt, E. et al. Discovery and refinement of loci associated with lipid levels. Nat Genet 45, 12741283 (2013). https://doi.org/10.1038/ng.2797*

- Height data are from lab collection: *Yengo L, Sidorenko J, Kemper KE, et al. Meta-analysis of genome-wide association studies for height and body mass index in <700000 individuals of European ancestry. Hum Mol Genet. 2018;27(20):3641-3649. doi:10.1093/hmg/ddy271*

- Schizophrenia data are from lab collection: *Schizophrenia Working Group of the Psychiatric Genomics Consortium., Ripke, S., Neale, B. et al. Biological insights from 108 schizophrenia-associated genetic loci. Nature 511, 421427 (2014). https://doi.org/10.1038/nature13595*

- Vitiligo data are from lab collection: *Jin Y et al. Genome-wide association studies of autoimmune vitiligo identify 23 new risk loci and highlight key pathways and regulatory variants. Nat Genet. 2016 Nov;48(11):1418-1424. doi: 10.1038/ng.3680. Epub 2016 Oct 10. PMID: 27723757; PMCID: PMC5120758.*

- Allergy data are from lab collection: *Ferreira MA et al. Shared genetic origin of asthma, hay fever and eczema elucidates allergic disease biology. Nat Genet. 2017 Dec;49(12):1752-1757. doi: 10.1038/ng.3985. Epub 2017 Oct 30. PMID: 29083406; PMCID: PMC5989923.*

2. Filtered the SNPs using pval < 5e-8 as cut off from the GWAS Catalog data. 

3. Did LD Clumping for the SNPs in step 2. The PLINK LD Clumping patameters are: 
      
     --clump-p1 0.0001            Significance threshold for index SNPs
     
     --clump-p2 0.01              Secondary significance threshold for clumped SNPs
 
     --clump-r2 0.1               LD threshold for clumping

     --clump-kb 1000               Physical distance threshold for clumping

For each trait, I got a subset of SNPs that are not in LD with each other.

4. Did association tests for plier_canonical factors and SNPs in 3. The association tests were corrected by 1)10 genotype PCs of whole genome; 2)10 PCs + GTEx:Sequencing platform,Sequencing protocol,Sex; 3)10 PCs + GTEx:Sequencing platform,Sequencing protocol,Sex + AGE

5. For each trait, I made a plot of association with LV(indicating by beta in GWAS) vs association with trait(indicating by ln(odds ratio) or beta in GWAS) to show if the variants have the correlated effect direction. The effect sizes of Catalog GWAS and factor association tests are harmonized by [TwoSampleMR](https://mrcieu.github.io/TwoSampleMR/articles/introduction.html) R package to make the effect alleles in these two analysis identical. The LVs have more than one significant SNPs with FDR<0.2 are included in the plotting.Besides, for each plots, I fitted the points with intercept = 0. The pvalues and r-squared are shown on the plots.

6. For the traits and LVs in 5, I made an info table to show more details of the SNPs. 

7. For several LVs we are interested in, I did gene set enrichment analysis to test if the LVs are correlated with some KEGG/REACTOME pathways. I used two kind gene sets to do GSEA: 1. genes that used to compute LVs; 2. Sorting the genes in 1 by their loadings, take the top 25% as the gene set. For both gene sets, the gene scores used as input of GSEA are the gene loadings.

8. Resampling. For some promising trait-factor pairs, I did resampling. I resampled the SNPs without replacement, I fitted the points with intercept = 0 again and recorded the pvalues and r-squared. The resampling was repeated 1000 times. The following plots are the resampling results.

## Snps after filter

After filtering by 'pval < 5e-8' and LD Clumping, for each trait, I got :

platelet count (plt) trait contains 688 SNPs with pval<5e-8. 

white blood cell count(wbc)trait contains 368 SNPs with pval<5e-8. 

myeloid white cell count (myeloid wbc) trait contains 319 SNPs with pval<5e-8. 

lymphocyte count(lymph) trait contains 436 SNPs with pval<5e-8. 

red blood cell count(wbc) trait contains 466 SNPs with pval<5e-8. 

granulocyte count(gran) trait contains 316 SNPs with pval<5e-8. 

eosinophil count(eo) trait contains 491 SNPs with pval<5e-8. 

neutrophil count (neut) trait contains 317 SNPs with pval<5e-8. 

IBD trait contains 116 SNPs with pval<5e-8. 

Ulcerative colitist(UC) trait contains 73 SNPs with pval<5e-8. 

Crohn's disease(CD) trait contains 96 SNPs with pval<5e-8. 

BMI trait contains 104 SNPs with pval<5e-8. 

T2D contains 14 SNPs with pval<5e-8. T2D_2 contains 4 SNPs with pval<5e-8.

Asthma trait contains 186 SNPs with pval<5e-8. Asthma_2 trait contains 112 SNPs with pval<5e-8.

HDL trait contains 227 SNPs with pval<5e-8.

LDL trait contains 204 SNPs with pval<5e-8.

WHR trait contains 36 SNPs with pval<5e-8.

TC trait contains 250 SNPs with pval<5e-8.

TG trait contains 155 SNPs with pval<5e-8.

Height trait contains 5846 SNPs with pval<5e-8.

SCZ trait contains 135 SNPs with pval<5e-8.

Vitiligo trait contains 101 SNPs with pval<5e-8.

T1D trait contains 69 SNPs with pval<5e-8.

Allergy trait contains 158 SNPs with pval<5e-8.

## Results - pval < 5e-8 & association test covariants: 10 PCs + GTEx:Sequencing platform,Sequencing protocol,Sex + AGE


### Summary table (26 traits)

I used 'qvalue' R package to compute the fdr from p-values for each SNP and made a table to show the number of SNPs that pass the threshold. The thresholds are 'fdr < 0.1','fdr < 0.2','pval < 5e-8'. The 'num_significant_pairs' indicates the number of significant pairs under each threshold. If a trait~factor pair has as least 1 significant SNP, we named it as 'significant pair'. 

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/sum_pliercanon_wgpc_ldcl_d1k_ageco.rdata")
knitr::kable(sum, caption = "number of significant trait-LV associations across all traits")
```

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/sum_sep_tog_pval_pliercanon_wgpc_ldcl_d1k_ageco.rdata")
DT::datatable(summary, options = list(pageLength =20))
```

### Info table (26 traits)

For each trait, I made a table to show the info of snps with fdr>0.2 in the factor ~ SNP + genotype pcs association test. For each trait,The LVs have more than one significant SNPs with FDR<0.2 are included. 

The suffix '_assoc' here means that results are from factor ~ SNP + genotype pcs association test. The suffix '_gwas' here means results are from original GWAS results files. For EUR.CD, EUR.IBD, EUR.UC,T2D, asthma, the effectsize_gwas here means 'ln(OR)', for others, it means 'beta'. 

'snp_ld' here means the snps that in LD with the snp in each line.'ld_r2' means the LD r-squared which is corresponding to the 'snp_ld' column. 'cis-eqtl' column indicates whether the snp is a cis-eqtl according to GTEx data. 'cis_gene_hgnc' and 'cis_gene_hgnc' is the genes that the snp influence when it act as cis-eqtl. 'func' and 'func_gene' are obtained from ANNOVAR, which indicating the snp function within the genes.

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/info_pval5e8_pliercanon_ld_d1k_ageco.rdata")
DT::datatable(final_full, options = list(pageLength =20))
```



### Effect size plots & Resampling (26 traits)

For each trait, I made a plot of association with LV(indicating by beta in GWAS) vs association with trait(indicating by ln(odds ratio) or beta in GWAS) to show if the variants have the correlated effect direction. The effect sizes of Catalog GWAS and factor association tests are harmonized by [TwoSampleMR](https://mrcieu.github.io/TwoSampleMR/articles/introduction.html) R package to make the effect alleles in these two analysis identical. The LVs have more than one significant SNPs with FDR<0.2 are included in the plotting.Besides, for each plots, I fitted the points with intercept = 0. The pvalues and r-squared are shown on the plots.

I also relaxed the fdr threshold of the SNPs that used to make effect size plots(from 0.2 to 0.3/0.5)

For all pairs , I did resampling. I resampled the SNPs without replacement and fitted the points with intercept = 0 again and recorded the pvalues and r-squared. The resampling was repeated 1000 times.

I made a histogram to show the rsquared distribution from resampling. The red line in the plots are the rsquared in the origin analysis. The r_mean values in the plots are the mean values of rsquared in point fitting. The 'p-value from resampling' is computed by: (number of more extreme values)/(times of resampling).

#### Allergy

LV33

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv33_allergy_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_33_allergy.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv33_allergy_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_33_allergy_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv33_allergy_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_33_allergy_fdr05.jpeg" width="30%">
</figure>

LV74

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv74_allergy_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_74_allergy.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv74_allergy_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_74_allergy_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv74_allergy_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_74_allergy_fdr05.jpeg" width="30%">
</figure>


LV110

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv110_allergy_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_110_allergy.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv110_allergy_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_110_allergy_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv110_allergy_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_110_allergy_fdr05.jpeg" width="30%">
</figure>

#### Asthma_2 


LV88

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv88_asthma_2_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_88_asthma_2.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv88_asthma_2_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_88_asthma_2_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv88_asthma_2_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_88_asthma_2_fdr05.jpeg" width="30%">
</figure>



LV91

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv91_asthma_2_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_91_asthma_2.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv91_asthma_2_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_91_asthma_2_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv91_asthma_2_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_91_asthma_2_fdr05.jpeg" width="30%">
</figure>



#### BMI 

LV27

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv27_bmi_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_27_bmi.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv27_bmi_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_27_bmi_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv27_bmi_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_27_bmi_fdr05.jpeg" width="30%">
</figure>


LV30

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv30_bmi_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_30_bmi.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv30_bmi_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_30_bmi_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv30_bmi_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_30_bmi_fdr05.jpeg" width="30%">
</figure>

LV76

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv76_bmi_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_76_bmi.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv76_bmi_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_76_bmi_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv76_bmi_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_76_bmi_fdr05.jpeg" width="30%">
</figure>

LV90

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv90_bmi_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_90_bmi.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv90_bmi_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_90_bmi_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv90_bmi_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_90_bmi_fdr05.jpeg" width="30%">
</figure>


#### eosinophil count(eo) 

LV106

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv106_eo_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_106_eo.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv106_eo_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_106_eo_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv106_eo_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_106_eo_fdr05.jpeg" width="30%">
</figure>

LV125

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv125_eo_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_125_eo.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv125_eo_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_125_eo_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv125_eo_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_125_eo_fdr05.jpeg" width="30%">
</figure>


#### Crohn's disease(CD)

LV106

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv126_EUR.CD_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_126_EUR.CD.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv126_EUR.CD_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_126_EUR.CD_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv126_EUR.CD_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_126_EUR.CD_fdr05.jpeg" width="30%">
</figure>


#### IBD

LV98

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv98_EUR.IBD_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_98_EUR.IBD.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv98_EUR.IBD_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_98_EUR.IBD_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv98_EUR.IBD_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_98_EUR.IBD_fdr05.jpeg" width="30%">
</figure>


#### Ulcerative colitist(UC)

LV23

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv23_EUR.UC_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_23_EUR.UC.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv23_EUR.UC_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_23_EUR.UC_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv23_EUR.UC_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_23_EUR.UC_fdr05.jpeg" width="30%">
</figure>

LV24

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv24_EUR.UC_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_24_EUR.UC.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv24_EUR.UC_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_24_EUR.UC_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv24_EUR.UC_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_24_EUR.UC_fdr05.jpeg" width="30%">
</figure>


LV108

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv108_EUR.UC_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_108_EUR.UC.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv108_EUR.UC_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_108_EUR.UC_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv108_EUR.UC_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_108_EUR.UC_fdr05.jpeg" width="30%">
</figure>


#### HDL

LV3

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv3_HDL_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_3_HDL.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv3_HDL_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_3_HDL_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv3_HDL_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_3_HDL_fdr05.jpeg" width="30%">
</figure>


LV28

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv28_HDL_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_28_HDL.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv28_HDL_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_28_HDL_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv28_HDL_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_28_HDL_fdr05.jpeg" width="30%">
</figure>

LV69

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv69_HDL_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_69_HDL.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv69_HDL_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_69_HDL_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv69_HDL_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_69_HDL_fdr05.jpeg" width="30%">
</figure>


LV72

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv72_HDL_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_72_HDL.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv72_HDL_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_72_HDL_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv72_HDL_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_72_HDL_fdr05.jpeg" width="30%">
</figure>


LV101

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv101_HDL_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_101_HDL.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv101_HDL_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_101_HDL_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv101_HDL_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_101_HDL_fdr05.jpeg" width="30%">
</figure>



#### Height

LV24

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv24_height_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_24_height.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv24_height_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_24_height_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv24_height_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_24_height_fdr05.jpeg" width="30%">
</figure>

LV87

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv87_height_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_87_height.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv87_height_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_87_height_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv87_height_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_87_height_fdr05.jpeg" width="30%">
</figure>

LV96

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv96_height_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_96_height.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv96_height_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_96_height_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv96_height_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_96_height_fdr05.jpeg" width="30%">
</figure>

#### LDL

LV23

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv23_LDL_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_23_LDL.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv23_LDL_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_23_LDL_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv23_LDL_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_23_LDL_fdr05.jpeg" width="30%">
</figure>


LV79

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv79_LDL_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_79_LDL.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv79_LDL_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_79_LDL_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv79_LDL_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_79_LDL_fdr05.jpeg" width="30%">
</figure>


LV113

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv113_LDL_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_113_LDL.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv113_LDL_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_113_LDL_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv113_LDL_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_113_LDL_fdr05.jpeg" width="30%">
</figure>


LV117

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv117_LDL_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_117_LDL.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv117_LDL_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_117_LDL_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv117_LDL_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_117_LDL_fdr05.jpeg" width="30%">
</figure>


LV125

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv125_LDL_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_125_LDL.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv125_LDL_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_125_LDL_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv125_LDL_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_125_LDL_fdr05.jpeg" width="30%">
</figure>



#### TG

LV30

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv30_TG_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_30_TG.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv30_TG_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_30_TG_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv30_TG_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_30_TG_fdr05.jpeg" width="30%">
</figure>

#### granulocyte count(gran)

None of the LVs have >1 SNPs at FDR<0.2.


#### lymphocyte count(lymph)

LV23

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv23_lymph_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_23_lymph.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv23_lymph_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_23_lymph_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv23_lymph_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_23_lymph_fdr05.jpeg" width="30%">
</figure>


LV26

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv26_lymph_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_26_lymph.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv26_lymph_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_26_lymph_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv26_lymph_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_26_lymph_fdr05.jpeg" width="30%">
</figure>

LV37

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv37_lymph_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_37_lymph.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv37_lymph_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_37_lymph_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv37_lymph_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_37_lymph_fdr05.jpeg" width="30%">
</figure>


LV65

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv65_lymph_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_65_lymph.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv65_lymph_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_65_lymph_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv65_lymph_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_65_lymph_fdr05.jpeg" width="30%">
</figure>

LV123

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv123_lymph_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_123_lymph.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv123_lymph_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_123_lymph_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv123_lymph_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_123_lymph_fdr05.jpeg" width="30%">
</figure>


#### myeloid white cell count (myeloid wbc)

LV65

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv65_myeloid_wbc_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_65_myeloid_wbc.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv65_myeloid_wbc_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_65_myeloid_wbc_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv65_myeloid_wbc_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_65_myeloid_wbc_fdr05.jpeg" width="30%">
</figure>


LV94

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv94_myeloid_wbc_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_94_myeloid_wbc.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv94_myeloid_wbc_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_94_myeloid_wbc_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv94_myeloid_wbc_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_94_myeloid_wbc_fdr05.jpeg" width="30%">
</figure>


LV119

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv119_myeloid_wbc_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_119_myeloid_wbc.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv119_myeloid_wbc_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_119_myeloid_wbc_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv119_myeloid_wbc_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_119_myeloid_wbc_fdr05.jpeg" width="30%">
</figure>




#### neutrophil count (neut)


LV6

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv6_neut_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_6_neut.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv6_neut_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_6_neut_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv6_neut_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_6_neut_fdr05.jpeg" width="30%">
</figure>

LV95

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv95_neut_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_95_neut.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv95_neut_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_95_neut_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv95_neut_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_95_neut_fdr05.jpeg" width="30%">
</figure>


#### plt

LV49

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv49_plt_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_49_plt.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv49_plt_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_49_plt_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv49_plt_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_49_plt_fdr05.jpeg" width="30%">
</figure>

LV97

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv97_plt_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_97_plt.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv97_plt_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_97_plt_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv97_plt_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_97_plt_fdr05.jpeg" width="30%">
</figure>


#### rbc

LV3

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv3_rbc_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_3_rbc.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv3_rbc_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_3_rbc_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv3_rbc_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_3_rbc_fdr05.jpeg" width="30%">
</figure>


LV20

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv20_rbc_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_20_rbc.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv20_rbc_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_20_rbc_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv20_rbc_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_20_rbc_fdr05.jpeg" width="30%">
</figure>

LV82

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv82_rbc_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_82_rbc.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv82_rbc_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_82_rbc_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv82_rbc_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_82_rbc_fdr05.jpeg" width="30%">
</figure>


LV118

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv118_rbc_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_118_rbc.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv118_rbc_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_118_rbc_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv118_rbc_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_118_rbc_fdr05.jpeg" width="30%">
</figure>


#### SCZ

LV7

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv7_SCZ_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_7_SCZ.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv7_SCZ_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_7_SCZ_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv7_SCZ_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_7_SCZ_fdr05.jpeg" width="30%">
</figure>

LV98

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv98_SCZ_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_98_SCZ.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv98_SCZ_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_98_SCZ_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv98_SCZ_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_98_SCZ_fdr05.jpeg" width="30%">
</figure>

LV100

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv100_SCZ_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_100_SCZ.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv100_SCZ_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_100_SCZ_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv100_SCZ_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_100_SCZ_fdr05.jpeg" width="30%">
</figure>

LV118

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv118_SCZ_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_118_SCZ.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv118_SCZ_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_118_SCZ_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv118_SCZ_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_118_SCZ_fdr05.jpeg" width="30%">
</figure>


#### T1D

LV30

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv30_T1D_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_30_T1D.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv30_T1D_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_30_T1D_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv30_T1D_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_30_T1D_fdr05.jpeg" width="30%">
</figure>


#### T2D_2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv38_T2D_2_pval5e8_d1k_ageco.jpeg" width="30%">
</figure>

Since there were only 4 SNPs left after filtering by p-values in GWAS summary data and all of them have fdr<0.2 in association test. So there is no other SNPs to do resampling.

#### T2D


LV1

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv1_T2D_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_1_T2D.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv1_T2D_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_1_T2D_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv1_T2D_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_1_T2D_fdr05.jpeg" width="30%">
</figure>



#### TC

LV20

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv20_TC_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_20_TC.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv20_TC_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_20_TC_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv20_TC_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_20_TC_fdr05.jpeg" width="30%">
</figure>

LV23

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv23_TC_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_23_TC.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv23_TC_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_23_TC_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv23_TC_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_23_TC_fdr05.jpeg" width="30%">
</figure>

LV125

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv125_TC_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_125_TC.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv125_TC_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_125_TC_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv125_TC_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_125_TC_fdr05.jpeg" width="30%">
</figure>

#### TG

LV30

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv30_TG_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_30_TG.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv30_TG_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_30_TG_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv30_TG_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_30_TG_fdr05.jpeg" width="30%">
</figure>

#### asthma


LV21

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv21_ukb.allasthma_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_21_ukb.allasthma.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv21_ukb.allasthma_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_21_ukb.allasthma_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv21_ukb.allasthma_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_21_ukb.allasthma_fdr05.jpeg" width="30%">
</figure>


LV36

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv36_ukb.allasthma_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_36_ukb.allasthma.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv36_ukb.allasthma_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_36_ukb.allasthma_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv36_ukb.allasthma_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_36_ukb.allasthma_fdr05.jpeg" width="30%">
</figure>


LV39

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv39_ukb.allasthma_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_39_ukb.allasthma.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv39_ukb.allasthma_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_39_ukb.allasthma_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv39_ukb.allasthma_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_39_ukb.allasthma_fdr05.jpeg" width="30%">
</figure>


LV68

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv68_ukb.allasthma_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_68_ukb.allasthma.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv68_ukb.allasthma_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_68_ukb.allasthma_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv68_ukb.allasthma_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_68_ukb.allasthma_fdr05.jpeg" width="30%">
</figure>

LV82

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv82_ukb.allasthma_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_82_ukb.allasthma.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv82_ukb.allasthma_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_82_ukb.allasthma_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv82_ukb.allasthma_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_82_ukb.allasthma_fdr05.jpeg" width="30%">
</figure>


#### Vitiligo

LV35

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv35_vitiligo_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_35_vitiligo.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv35_vitiligo_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_35_vitiligo_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv35_vitiligo_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_35_vitiligo_fdr05.jpeg" width="30%">
</figure>

LV69

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv69_vitiligo_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_69_vitiligo.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv69_vitiligo_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_69_vitiligo_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv69_vitiligo_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_69_vitiligo_fdr05.jpeg" width="30%">
</figure>

LV82

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv82_vitiligo_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_82_vitiligo.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv82_vitiligo_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_82_vitiligo_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv82_vitiligo_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_82_vitiligo_fdr05.jpeg" width="30%">
</figure>

LV97

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv97_vitiligo_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_97_vitiligo.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv97_vitiligo_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_97_vitiligo_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv97_vitiligo_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_97_vitiligo_fdr05.jpeg" width="30%">
</figure>

LV109

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv109_vitiligo_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_109_vitiligo.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv109_vitiligo_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_109_vitiligo_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv109_vitiligo_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_109_vitiligo_fdr05.jpeg" width="30%">
</figure>

#### wbc

LV6

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv6_wbc_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_6_wbc.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv6_wbc_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_6_wbc_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv6_wbc_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_6_wbc_fdr05.jpeg" width="30%">
</figure>


LV119

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv119_wbc_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_119_wbc.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv119_wbc_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_119_wbc_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv119_wbc_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_119_wbc_fdr05.jpeg" width="30%">
</figure>




#### WHR

LV1

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv1_WHR_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_1_WHR.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv1_WHR_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_1_WHR_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv1_WHR_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_1_WHR_fdr05.jpeg" width="30%">
</figure>



LV47

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv47_WHR_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_47_WHR.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv47_WHR_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_47_WHR_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv47_WHR_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_47_WHR_fdr05.jpeg" width="30%">
</figure>



LV82

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv82_WHR_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_82_WHR.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv82_WHR_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_82_WHR_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv82_WHR_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_82_WHR_fdr05.jpeg" width="30%">
</figure>


LV106

fdr0.2

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv106_WHR_pval5e8_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_106_WHR.jpeg" width="30%">
</figure>


fdr0.3 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv106_WHR_pval5e8_d1k_ageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_106_WHR_fdr03.jpeg" width="30%">
</figure>


fdr0.5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv106_WHR_pval5e8_d1k_ageco_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/resampling_hist_pliercanon_106_WHR_fdr05.jpeg" width="30%">
</figure>



### Effectsize summary (26 traits)

To summarize the results above, I made a scatter plot of all trait-LV pairs. The x-axis shows the R2 in fitting, and the y-axis show the -log10-p values from resampling. Below are plots for three FDR cutoffs, 0.2, 0.3 and 0.5. To make the plot more informative, Each point represent a trait-LV pair. Different colors show the number of SNPs passing the FDR threshold for each pair.


<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsizesum_0.2_pliercanon.jpeg" width="60%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsizesum_0.3_pliercanon.jpeg" width="60%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsizesum_0.5_pliercanon.jpeg" width="60%">
</figure>

[Summary Table](https://drive.google.com/file/d/1CTK2pwSTkhDHREYJ71PWjL5YZpHy84IP/view?usp=sharing)

### Colocalization -- 1000kb (26 traits)

The colocalization analysis was performed using the approximate Bayes factor test implemented in the Coloc package. Coloc computes five posterior probabilities (PP0, PP1, PP2, PP3 and PP4), each corresponding to a hypothesis: H0, no association with either trait; H1, association with trait 1 but not with trait 2; H2, association with trait 2 but not with trait 1; H3, association with trait 1 and trait 2, two independent SNPs; H4, association with trait 1 and trait 2, one shared SNP. We ran Coloc with the default parameters and used PP4 to assess evidence of colocalization. We visualized the colocalization of factor - QTLs and GWAS associations using the [LocusCompareR package](https://github.com/boxiangliu/locuscomparer).

SNP selection: 1. Chose the SNPs in the info table.  2. For each SNP, the region used in colocalization analysis is between [pos-500kb, pos+500kb]. 3. All SNPs in this region are included in alalysis.


[Summary Table](https://drive.google.com/file/d/1KU0eiDuiyXS_qQZexTBcniLDWN0ub4zr/view?usp=sharing)

#### Allergy

LV33

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/allergy_33_chr1_152347096_T_C_b38_rs61816766.coloc.results.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "PPs in coloclization analysis")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/allergy_33_chr1_152347096_T_C_b38_rs61816766.coloc.jpeg" width="80%">
</figure>


```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/allergy_33_chr4_102648126_A_G_b38_rs228619.coloc.results.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "PPs in coloclization analysis")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/allergy_33_chr4_102648126_A_G_b38_rs228619.coloc.jpeg" width="80%">
</figure>

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/allergy_33_chr4_122464858_C_T_b38_rs58939053.coloc.results.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "PPs in coloclization analysis")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/allergy_33_chr4_122464858_C_T_b38_rs58939053.coloc.jpeg" width="80%">
</figure>


LV110

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/allergy_110_chr2_102250748_C_G_b38_rs12464251.coloc.results.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "PPs in coloclization analysis")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/allergy_110_chr2_102250748_C_G_b38_rs12464251.coloc.jpeg" width="80%">
</figure>

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/allergy_110_chr5_137534265_T_C_b38_rs11746779.coloc.results.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "PPs in coloclization analysis")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/allergy_110_chr5_137534265_T_C_b38_rs11746779.coloc.jpeg" width="80%">
</figure>


#### BMI 

LV27 

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/bmi_27_53759010.coloc_pliercanon_d1k_500.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "PPs in coloclization analysis")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/bmi_27_53759010.coloc_pliercanon_d1k_500.jpeg" width="80%">
</figure>


```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/bmi_27_85758440.coloc_pliercanon_d1k_500.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "PPs in coloclization analysis")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/bmi_27_85758440.coloc_pliercanon_d1k_500.jpeg" width="80%">
</figure>

LV76 

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/bmi_76_40380914.coloc_pliercanon_d1k_500.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "PPs in coloclization analysis")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/bmi_76_40380914.coloc_pliercanon_d1k_500.jpeg" width="80%">
</figure>

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/bmi_76_43842728.coloc_pliercanon_d1k_500.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "PPs in coloclization analysis")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/bmi_76_43842728.coloc_pliercanon_d1k_500.jpeg" width="80%">
</figure>


#### HDL

LV28

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/HDL_28_chr15_58337260_A_G_b38_rs2566091.coloc.results.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "PPs in coloclization analysis")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/HDL_28_chr15_58337260_A_G_b38_rs2566091.coloc.jpeg" width="80%">
</figure>


```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/HDL_28_chr16_56950678_T_C_b38_rs9938160.coloc.results.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "PPs in coloclization analysis")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/HDL_28_chr16_56950678_T_C_b38_rs9938160.coloc.jpeg" width="80%">
</figure>

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/HDL_28_chr16_56987521_A_G_b38_rs12934552.coloc.results.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "PPs in coloclization analysis")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/HDL_28_chr16_56987521_A_G_b38_rs12934552.coloc.jpeg" width="80%">
</figure>

#### LDL

LV125

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/LDL_125_160687504.coloc_pliercanon_d1k_500.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "PPs in coloclization analysis")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/LDL_125_160687504.coloc_pliercanon_d1k_500.jpeg" width="80%">
</figure>


```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/LDL_125_44943964.coloc_pliercanon_d1k_500.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "PPs in coloclization analysis")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/LDL_125_44943964.coloc_pliercanon_d1k_500.jpeg" width="80%">
</figure>



```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/LDL_125_56955678.coloc_pliercanon_d1k_500.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "PPs in coloclization analysis")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/LDL_125_56955678.coloc_pliercanon_d1k_500.jpeg" width="80%">
</figure>

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/LDL_125_56965346.coloc_pliercanon_d1k_500.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "PPs in coloclization analysis")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/LDL_125_56965346.coloc_pliercanon_d1k_500.jpeg" width="80%">
</figure>


#### lymphocyte count(lymph)

LV26

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/lymph_26_41686157.coloc_pliercanon_d1k_500.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "PPs in coloclization")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/lymph_26_41686157.coloc_pliercanon_d1k_500.jpeg" width="80%">
</figure>


```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/lymph_26_59258463.coloc_pliercanon_d1k_500.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "PPs in coloclization")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/lymph_26_59258463.coloc_pliercanon_d1k_500.jpeg" width="80%">
</figure>


#### platelet count (plt) 

LV49 

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/plt_49_34414160.coloc_pliercanon_d1k_500.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "PPs in coloclization")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/plt_49_34414160.coloc_pliercanon_d1k_500.jpeg" width="80%">
</figure>

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/plt_49_56786080.coloc_pliercanon_d1k_500.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "PPs in coloclization")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/plt_49_56786080.coloc_pliercanon_d1k_500.jpeg" width="80%">
</figure>

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/plt_49_88978375.coloc_pliercanon_d1k_500.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "PPs in coloclization")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/plt_49_88978375.coloc_pliercanon_d1k_500.jpeg" width="80%">
</figure>


#### red blood cell count(rbc) 

LV82 

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/rbc_82_32506416.coloc_pliercanon_d1k_500.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "PPs in coloclization")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/rbc_82_32506416.coloc_pliercanon_d1k_500.jpeg" width="80%">
</figure>

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/rbc_82_42718545.coloc_pliercanon_d1k_500.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "PPs in coloclization")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/rbc_82_42718545.coloc_pliercanon_d1k_500.jpeg" width="80%">
</figure>


```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/rbc_82_54482753.coloc_pliercanon_d1k_500.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "PPs in coloclization")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/rbc_82_54482753.coloc_pliercanon_d1k_500.jpeg" width="80%">
</figure>

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/rbc_82_8614076.coloc_pliercanon_d1k_500.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "PPs in coloclization")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/rbc_82_8614076.coloc_pliercanon_d1k_500.jpeg" width="80%">
</figure>



#### white blood cell count(wbc)



LV6

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/wbc_6_129655816.coloc_pliercanon_d1k_500.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "PPs in coloclization")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/wbc_6_129655816.coloc_pliercanon_d1k_500.jpeg" width="80%">
</figure>

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/wbc_6_81575948.coloc_pliercanon_d1k_500.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "PPs in coloclization")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/wbc_6_81575948.coloc_pliercanon_d1k_500.jpeg" width="80%">
</figure>




LV119

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/wbc_119_16589041.coloc_pliercanon_d1k_500.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "PPs in coloclization")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/wbc_119_16589041.coloc_pliercanon_d1k_500.jpeg" width="80%">
</figure>

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/wbc_119_31355703.coloc_pliercanon_d1k_500.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "PPs in coloclization")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/wbc_119_31355703.coloc_pliercanon_d1k_500.jpeg" width="80%">
</figure>


#### TC

LV125

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/TC_125_chr16_56955678_C_T_b38_rs247616.coloc.results.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "PPs in coloclization")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/TC_125_chr16_56955678_C_T_b38_rs247616.coloc.jpeg" width="80%">
</figure>

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/TC_125_chr16_56965346_A_C_b38_rs7203984.coloc.results.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "PPs in coloclization")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/TC_125_chr16_56965346_A_C_b38_rs7203984.coloc.jpeg" width="80%">
</figure>

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/TC_125_chr6_160687504_C_T_b38_rs6935921.coloc.results.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "PPs in coloclization")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/TC_125_chr6_160687504_C_T_b38_rs6935921.coloc.jpeg" width="80%">
</figure>



#### Vitiligo 

LV69

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/vitiligo_69_chr11_35253282_A_G_b38_rs1043101.coloc.results.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "PPs in coloclization")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/vitiligo_69_chr11_35253282_A_G_b38_rs1043101.coloc.jpeg" width="80%">
</figure>

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/vitiligo_69_chr13_54518347_G_A_b38_rs111260018.coloc.results.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "PPs in coloclization")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/vitiligo_69_chr13_54518347_G_A_b38_rs111260018.coloc.jpeg" width="80%">
</figure>

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/vitiligo_69_chr2_55623198_C_T_b38_rs9309267.coloc.results.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "PPs in coloclization")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/vitiligo_69_chr2_55623198_C_T_b38_rs9309267.coloc.jpeg" width="80%">
</figure>


LV109

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/vitiligo_109_chr20_34077942_A_G_b38_rs6059655.coloc.results.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "PPs in coloclization")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/vitiligo_109_chr20_34077942_A_G_b38_rs6059655.coloc.jpeg" width="80%">
</figure>


```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/vitiligo_109_chr20_35102207_C_T_b38_rs4911466.coloc.results.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "PPs in coloclization")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/vitiligo_109_chr20_35102207_C_T_b38_rs4911466.coloc.jpeg" width="80%">
</figure>

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/vitiligo_109_chr6_31933996_G_A_b38_rs9332735.coloc.results.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "PPs in coloclization")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/vitiligo_109_chr6_31933996_G_A_b38_rs9332735.coloc.jpeg" width="80%">
</figure>

#### ukb.asthma

#### WHR

LV47

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/WHR_47_53955989.coloc_pliercanon_d1k_500.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "PPs in coloclization")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/WHR_47_53955989.coloc_pliercanon_d1k_500.jpeg" width="80%">
</figure>

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/WHR_47_6742916.coloc_pliercanon_d1k_500.rdata")
note <- c(NA,"no association with either trait","association with trait 1 but not with trait 2","association with trait 2 but not with trait 1","association with trait 1 and trait 2,two independent SNPs"," association with trait 1 and trait 2, one shared SNP")
pval <- cbind(results$summary,note)
knitr::kable(pval, caption = "PPs in coloclization")
```

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/WHR_47_6742916.coloc_pliercanon_d1k_500.jpeg" width="80%">
</figure>






### Enrichment analysis (19 traits)

For some promising trait-factor pairs (i.e. BMI-LV3, LV27, RBC-LV82, AsthmaLV68,  WBC-LV119, Lymphocyte-LV23, LV78), I did enrichment analysis with WebGestalt. The analysis are under different settings:

1. ORA: For each factor, I sorted the genes that have non-zero loadings by their loadings, taking the top 25% as the gene set. The function database here are Reactome pathway; Disgenet + GLAD4U + OMIM disease dataset; geneontology biological process. The reference set affy hugene 2 0 st v1. Minimum number of genes for a category is 5, maximum number of genes for a category is 2000(default settings). All categories that have fdr<0.01 are listed.


**BMI-LV27**

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/lv27top025_pliercanon.rdata")
DT::datatable(ORA_results, options = list(pageLength =5))
```

**BMI-LV76**

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/lv76top025_pliercanon.rdata")
DT::datatable(ORA_results, options = list(pageLength =5))
```

**BMI-LV90**

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/lv90top025_pliercanon.rdata")
DT::datatable(ORA_results, options = list(pageLength =5))
```

**PLT-LV49**

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/lv49top025_pliercanon.rdata")
DT::datatable(ORA_results, options = list(pageLength =5))
```

**RBC-LV82**

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/lv82top025_pliercanon.rdata")
DT::datatable(ORA_results, options = list(pageLength =5))
```

**Asthma-LV36**

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/lv36top025_pliercanon.rdata")
DT::datatable(ORA_results, options = list(pageLength =5))
```

**Asthma-LV39**

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/lv39top025_pliercanon.rdata")
DT::datatable(ORA_results, options = list(pageLength =5))
```

**WBC-LV6**

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/lv6top025_pliercanon.rdata")
DT::datatable(ORA_results, options = list(pageLength =5))
```

**WBC-LV119**

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/lv119top025_pliercanon.rdata")
DT::datatable(ORA_results, options = list(pageLength =5))
```

**Lymphocyte-LV23**

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/lv23top025_pliercanon.rdata")
DT::datatable(ORA_results, options = list(pageLength =5))
```

**Lymphocyte-LV26**

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/lv26top025_pliercanon.rdata")
DT::datatable(ORA_results, options = list(pageLength =5))
```

**LDL-LV125**

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/lv125top025_pliercanon.rdata")
DT::datatable(ORA_results, options = list(pageLength =5))
```

**WHR**

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/lv47top025_pliercanon.rdata")
DT::datatable(ORA_results, options = list(pageLength =5))
```

### Effect size plots - checking the reverse causality (19 traits)

To check if the effect size correlation is due to reverse causality: i.e. trait -> LV (trait causally affect LV), instead of LV -> trait (which is what we like to see). I used all SNPs associated with traits(pval<5E-8). The x-axis is the effects of these SNPs on trait, and y-axis is the effects on LV. 

Some pair show p < 0.05, the result may be driven by the possible causal effect of LV -> trait. To test this, I removed the SNPs that are associated with LVs at FDR < 0.2 and made the plots again.

#### BMI 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv27_bmi_pval5e8_d1k_ageco_reverse.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv76_bmi_pval5e8_d1k_ageco_reverse.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv90_bmi_pval5e8_d1k_ageco_reverse.jpeg" width="30%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv27_bmi_pval5e8_d1k_ageco_reverse_remove.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv76_bmi_pval5e8_d1k_ageco_reverse_remove.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv90_bmi_pval5e8_d1k_ageco_reverse_remove.jpeg" width="30%">
</figure>


#### LDL

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv125_LDL_pval5e8_d1k_ageco_reverse.jpeg" width="30%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv125_LDL_pval5e8_d1k_ageco_reverse_remove.jpeg" width="30%">
</figure>


#### lymphocyte count(lymph)

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv23_lymph_pval5e8_d1k_ageco_reverse.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv26_lymph_pval5e8_d1k_ageco_reverse.jpeg" width="30%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv23_lymph_pval5e8_d1k_ageco_reverse_remove.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv26_lymph_pval5e8_d1k_ageco_reverse_remove.jpeg" width="30%">
</figure>

#### platelet count (plt) 


<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv49_plt_pval5e8_d1k_ageco_reverse.jpeg" width="30%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv49_plt_pval5e8_d1k_ageco_reverse_remove.jpeg" width="30%">
</figure>

#### red blood cell count(wbc) 


<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv82_rbc_pval5e8_d1k_ageco_reverse.jpeg" width="30%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv82_rbc_pval5e8_d1k_ageco_reverse_remove.jpeg" width="30%">
</figure>


#### asthma

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv36_ukb.allasthma_pval5e8_d1k_ageco_reverse.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv39_ukb.allasthma_pval5e8_d1k_ageco_reverse.jpeg" width="30%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv36_ukb.allasthma_pval5e8_d1k_ageco_reverse_remove.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv39_ukb.allasthma_pval5e8_d1k_ageco_reverse_remove.jpeg" width="30%">
</figure>

#### white blood cell count

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv6_wbc_pval5e8_d1k_ageco_reverse.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv119_wbc_pval5e8_d1k_ageco_reverse.jpeg" width="30%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv6_wbc_pval5e8_d1k_ageco_reverse_remove.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv119_wbc_pval5e8_d1k_ageco_reverse_remove.jpeg" width="30%">
</figure>

#### WHR

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv47_WHR_pval5e8_d1k_ageco_reverse.jpeg" width="30%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv47_WHR_pval5e8_d1k_ageco_reverse_remove.jpeg" width="30%">
</figure>


### QQplots (19 traits)

#### BMI 

LV27, LV76 and LV90

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qq_unif_lv27_bmi_pliercanon_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qq_unif_lv76_bmi_pliercanon_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qq_unif_lv90_bmi_pliercanon_d1k_ageco.jpeg" width="30%">
</figure>


#### LDL

LV125

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qq_unif_lv125_LDL_pliercanon_d1k_ageco.jpeg" width="30%">\
</figure>



#### platelet count (plt) 

LV49

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qq_unif_lv49_plt_pliercanon_d1k_ageco.jpeg" width="30%">
</figure>



#### red blood cell count(wbc) 

LV82

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qq_unif_lv82_rbc_pliercanon_d1k_ageco.jpeg" width="30%">
</figure>



#### asthma

LV36 and LV79

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qq_unif_lv36_ukb.allasthma_pliercanon_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qq_unif_lv79_ukb.allasthma_pliercanon_d1k_ageco.jpeg" width="30%">
</figure>


#### white blood cell count

LV6 and LV119

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qq_unif_lv6_wbc_pliercanon_d1k_ageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qq_unif_lv119_wbc_pliercanon_d1k_ageco.jpeg" width="30%">
</figure>


#### WHR

LV47

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qq_unif_lv47_WHR_pliercanon_d1k_ageco.jpeg" width="30%">
</figure>


## Results - pval < 5e-8 & association test covariants: 10 PCs

### Summary table

I used 'qvalue' R package to compute the fdr from p-values for each SNP and made a table to show the number of SNPs that pass the threshold. The thresholds are 'fdr < 0.1','fdr < 0.2','pval < 5e-8'.

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/sum_sep_tog_pval_pliercanon_wgpc_ldcl_d1k.rdata")
DT::datatable(summary, options = list(pageLength =20))
```


### Info table

For each trait, I made a table to show the info of snps with fdr>0.2 in the factor ~ SNP + genotype pcs association test. For each trait,The LVs have more than one significant SNPs with FDR<0.2 are included. 

The suffix '_assoc' here means that results are from factor ~ SNP + genotype pcs association test. The suffix '_gwas' here means results are from original GWAS results files. For EUR.CD, EUR.IBD, EUR.UC, the effectsize_gwas here means 'ln(OR)', for others, it means 'beta'. 

'snp_ld' here means the snps that in LD with the snp in each line.'ld_r2' means the LD r-squared which is corresponding to the 'snp_ld' column. 'cis-eqtl' column indicates whether the snp is a cis-eqtl according to GTEx data. 'cis_gene_hgnc' and 'cis_gene_hgnc' is the genes that the snp influence when it act as cis-eqtl. 'func' and 'func_gene' are obtained from ANNOVAR, which indicating the snp function within the genes.

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/info_pval5e8_pliercanon_ld_d1k.rdata")
DT::datatable(final_full, options = list(pageLength =20))
```

### Enrichment analysis

For some promising trait-factor pairs (i.e. BMI-LV3, LV27, RBC-LV82, AsthmaLV68,  WBC-LV119, Lymphocyte-LV23, LV78), I did enrichment analysis with WebGestalt. The analysis are under different settings:

1. ORA: For each factor, I sorted the genes in 1 by their loadings, taking the top 25% as the gene set. The function database here are Reactome pathway; Disgenet + GLAD4U + OMIM disease dataset; geneontology biological process. The reference set affy hugene 2 0 st v1. Minimum number of genes for a category is 5, maximum number of genes for a category is 2000(default settings). All categories that have fdr<0.01 are listed.

[BMI-LV3](http://www.webgestalt.org/results/1603046168/#)

[BMI-LV27](http://www.webgestalt.org/results/1603046183/#)

[BMI-LV76](http://www.webgestalt.org/results/1603046194/#)

[BMI-LV90](http://www.webgestalt.org/results/1603129380/#)

[RBC-LV82](http://www.webgestalt.org/results/1603046204/#)

[Asthma-LV68](http://www.webgestalt.org/results/1603046188/#)

[WBC-LV119](http://www.webgestalt.org/results/1603046209/#)

[Lymphocyte-LV23](http://www.webgestalt.org/results/1603046175/#)

[Lymphocyte-LV78](http://www.webgestalt.org/results/1603046199/#)

2. GSEA: For each factor, I used all genes (zero loading genes included), taking the gene loadings as their gene score in GSEA. The function database is Reactome pathway.

[BMI-LV3-Reactome](http://www.webgestalt.org/results/1602817719/#)
[BMI-LV3-GO](http://www.webgestalt.org/results/1603047206/#)
[BMI-LV3-phenotype](http://www.webgestalt.org/results/1603047219/#)
[BMI-LV3-disease-Disgenet](http://www.webgestalt.org/results/1603047233/#)


[BMI-LV27-Reactome](http://www.webgestalt.org/results/1602817777/#)
[BMI-LV27-GO](http://www.webgestalt.org/results/1603057432/#)
[BMI-LV27-phenotype](http://www.webgestalt.org/results/1603056104/#)
[BMI-LV27-disease-Disgenet](http://www.webgestalt.org/results/1603048331/#)

[BMI-LV76-Reactome](http://www.webgestalt.org/results/1602879472/#)
[BMI-LV76-GO](http://www.webgestalt.org/results/1603057390/#)
[BMI-LV76-phenotype](http://www.webgestalt.org/results/1603051035/#)
[BMI-LV76-disease-Disgenet](http://www.webgestalt.org/results/1603050188/#)

[RBC-LV82-Reactome](http://www.webgestalt.org/results/1602817795/#)
[RBC-LV82-GO](http://www.webgestalt.org/results/1603055953/#)
[RBC-LV82-phenotype](http://www.webgestalt.org/results/1603049647/#)
[RBC-LV82-disease-Disgenet](http://www.webgestalt.org/results/1603048694/#)

[Asthma-LV68-Reactome](http://www.webgestalt.org/results/1602817784/#)
[Asthma-LV68-GO](http://www.webgestalt.org/results/1603057396/#)
[Asthma-LV68-phenotype](http://www.webgestalt.org/results/1603056098/#)
[Asthma-LV68-disease-Disgenet](http://www.webgestalt.org/results/1603050957/#)

[WBC-LV119-Reactome](http://www.webgestalt.org/results/1602817802/#)
[WBC-LV119-GO](http://www.webgestalt.org/results/1603055946/#)
[WBC-LV119-phenotype](http://www.webgestalt.org/results/1603051028/#)
[WBC-LV119-disease-Disgenet](http://www.webgestalt.org/results/1603048685/#)

[Lymphocyte-LV23-Reactome](http://www.webgestalt.org/results/1602817770/#)
[Lymphocyte-LV23-GO](http://www.webgestalt.org/results/1603058325/#)
[Lymphocyte-LV23-phenotype](http://www.webgestalt.org/results/1603057276/#)
[Lymphocyte-LV23-disease-Disgenet](http://www.webgestalt.org/results/1603048322/#)

[Lymphocyte-LV78](http://www.webgestalt.org/results/1602817790/#)
[Lymphocyte-LV78-GO](http://www.webgestalt.org/results/1603055931/#)
[Lymphocyte-LV78-phenotype](http://www.webgestalt.org/results/1603049640/#)
[Lymphocyte-LV78-disease-Disgenet](http://www.webgestalt.org/results/1603050962/#)

3. GSEA: For each factor, I used all genes (zero loading genes included), taking the gene loadings as their gene score in GSEA. For those loadings = 0, I assigned a random number from normal distribution N(0,1e-5) to it to avoid ties. The function database is Reactome pathway.

[BMI-LV3](http://www.webgestalt.org/results/1602818962/#)

[BMI-LV27](http://www.webgestalt.org/results/1602818974/#)

[BMI-LV76](http://www.webgestalt.org/results/1602879491/#)

[RBC-LV82](http://www.webgestalt.org/results/1602818985/#)

[Asthma-LV68](http://www.webgestalt.org/results/1602818980/#)

[WBC-LV119](http://www.webgestalt.org/results/1602818997/#)

[Lymphocyte-LV23](http://www.webgestalt.org/results/1602818968/#)

[Lymphocyte-LV78](http://www.webgestalt.org/results/1602818985/#)


### Effect size plots

For each trait, I made a plot of association with LV(indicating by beta in GWAS) vs association with trait(indicating by ln(odds ratio) or beta in GWAS) to show if the variants have the correlated effect direction. The effect sizes of Catalog GWAS and factor association tests are harmonized by [TwoSampleMR](https://mrcieu.github.io/TwoSampleMR/articles/introduction.html) R package to make the effect alleles in these two analysis identical. The LVs have more than one significant SNPs with FDR<0.2 are included in the plotting.Besides, for each plots, I fitted the points with intercept = 0. The pvalues and r-squared are shown on the plots.

#### BMI 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv3_bmi_pval5e8_d1k.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv27_bmi_pval5e8_d1k.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv30_bmi_pval5e8_d1k.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv76_bmi_pval5e8_d1k.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv90_bmi_pval5e8_d1k.jpeg" width="30%">
</figure>


#### eosinophil count(eo) 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv125_eo_pval5e8_d1k.jpeg" width="30%">
</figure>

#### Crohn's disease(CD)

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv88_EUR.CD_pval5e8_d1k.jpeg" width="30%">
</figure>

#### IBD

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv98_EUR.IBD_pval5e8_d1k.jpeg" width="30%">
</figure>

#### Ulcerative colitist(UC)

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv23_EUR.UC_pval5e8_d1k.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv24_EUR.UC_pval5e8_d1k.jpeg" width="30%">
</figure>


#### granulocyte count(gran)

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv120_gran_pval5e8_d1k.jpeg" width="30%">
</figure>


#### lymphocyte count(lymph)

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv23_lymph_pval5e8_d1k.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv37_lymph_pval5e8_d1k.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv65_lymph_pval5e8_d1k.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv78_lymph_pval5e8_d1k.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv123_lymph_pval5e8_d1k.jpeg" width="30%">
</figure>

#### myeloid white cell count (myeloid wbc)

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv94_myeloid_wbc_pval5e8_d1k.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv20_myeloid_wbc_pval5e8_d1k.jpeg" width="30%">
</figure>

#### neutrophil count (neut)

None of the LVs have >1 SNPs at FDR<0.2.

#### plt

None of the LVs have >1 SNPs at FDR<0.2.

#### rbc

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv3_rbc_pval5e8_d1k.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv20_rbc_pval5e8_d1k.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv70_rbc_pval5e8_d1k.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv59_rbc_pval5e8_d1k.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv82_rbc_pval5e8_d1k.jpeg" width="30%">
</figure>

#### asthma

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv68_ukb.allasthma_pval5e8_d1k.jpeg" width="30%">
</figure>

#### wbc

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv119_wbc_pval5e8_d1k.jpeg" width="30%">
</figure>


### effect size plots -- more SNPs

For some promising trait-factor pairs (i.e. BMI-LV3, LV27,LV76, RBC-LV82, Asthma-LV68,  WBC-LV119, Lymphocyte-LV23, LV78), I relaxed the fdr threshold of the SNPs that used to make effect size plots(from 0.2 to 0.3/0.5)

I also made a plot to show the the distribution of the SNPs' fdr.

#### BMI 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv3_bmi_pval5e8_d1k.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv3_bmi_pval5e8_d1k_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv3_bmi_pval5e8_d1k_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/fdrhist_lv3_bmi_pval5e8_d1k.jpeg" width="30%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv27_bmi_pval5e8_d1k.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv27_bmi_pval5e8_d1k_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv27_bmi_pval5e8_d1k_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/fdrhist_lv27_bmi_pval5e8_d1k.jpeg" width="30%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv76_bmi_pval5e8_d1k.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv76_bmi_pval5e8_d1k_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv76_bmi_pval5e8_d1k_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/fdrhist_lv76_bmi_pval5e8_d1k.jpeg" width="30%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv90_bmi_pval5e8_d1k.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv90_bmi_pval5e8_d1k_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv90_bmi_pval5e8_d1k_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/fdrhist_lv90_bmi_pval5e8_d1k.jpeg" width="30%">
</figure>

#### Crohn's disease(CD)

The CD~lv88 is not very promising pairs when considering SNPs at fdr<0.2, but the fitting result at fdr <0.5 is better than the former result . So I post the plots here too.

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv88_EUR.CD_pval5e8_d1k.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv88_EUR.CD_pval5e8_d1k_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv88_EUR.CD_pval5e8_d1k_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/fdrhist_lv88_EUR.CD_pval5e8_d1k.jpeg" width="30%">
</figure>


#### lymphocyte count(lymph)

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv23_lymph_pval5e8_d1k.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv23_lymph_pval5e8_d1k_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv23_lymph_pval5e8_d1k_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/fdrhist_lv23_lymph_pval5e8_d1k.jpeg" width="30%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv78_lymph_pval5e8_d1k.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv78_lymph_pval5e8_d1k_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv78_lymph_pval5e8_d1k_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/fdrhist_lv78_lymph_pval5e8_d1k.jpeg" width="30%">
</figure>

#### red blood cell count(wbc) 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv82_rbc_pval5e8_d1k.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv82_rbc_pval5e8_d1k_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv82_rbc_pval5e8_d1k_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/fdrhist_lv82_rbc_pval5e8_d1k.jpeg" width="30%">
</figure>

The rbc~lv42 and rbc~lv59 pairs are not very promising pairs when considering SNPs at fdr<0.2, but the fitting results at fdr <0.5 are better than the former results. So I post the plots here too.

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv42_rbc_pval5e8_d1k.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv42_rbc_pval5e8_d1k_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv42_rbc_pval5e8_d1k_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/fdrhist_lv42_rbc_pval5e8_d1k.jpeg" width="30%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv59_rbc_pval5e8_d1k.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv59_rbc_pval5e8_d1k_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv59_rbc_pval5e8_d1k_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/fdrhist_lv59_rbc_pval5e8_d1k.jpeg" width="30%">
</figure>


#### asthma

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv68_ukb.allasthma_pval5e8_d1k.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv68_ukb.allasthma_pval5e8_d1k_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv68_ukb.allasthma_pval5e8_d1k_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/fdrhist_lv68_ukb.allasthma_pval5e8_d1k.jpeg" width="30%">
</figure>

#### white blood cell count

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv119_wbc_pval5e8_d1k.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv119_wbc_pval5e8_d1k_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv119_wbc_pval5e8_d1k_fdr05.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/fdrhist_lv119_wbc_pval5e8_d1k.jpeg" width="30%">
</figure>

### Effect size plots - checking the reverse causality

To check if the effect size correlation is due to reverse causality: i.e. trait -> LV (trait causally affect LV), instead of LV -> trait (which is what we like to see). I used all SNPs associated with traits(pval<5E-8). The x-axis is the effects of these SNPs on trait, and y-axis is the effects on LV. 

Some pair show p < 0.05, the result may be driven by the possible causal effect of LV -> trait. To test this, I removed the SNPs that are associated with LVs at FDR < 0.2 and made the plots again.

#### BMI 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv3_bmi_pval5e8_d1k_reverse.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv27_bmi_pval5e8_d1k_reverse.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv76_bmi_pval5e8_d1k_reverse.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv90_bmi_pval5e8_d1k_reverse.jpeg" width="30%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv3_bmi_pval5e8_d1k_reverse_remove.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv27_bmi_pval5e8_d1k_reverse_remove.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv76_bmi_pval5e8_d1k_reverse_remove.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv90_bmi_pval5e8_d1k_reverse_remove.jpeg" width="30%">
</figure>



#### Crohn's disease(CD)

The CD~lv88 is not very promising pairs when considering SNPs at fdr<0.2, but the fitting result at fdr <0.5 is better than the former result . So I post the plots here too.

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv88_EUR.CD_pval5e8_d1k_reverse.jpeg" width="30%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv88_EUR.CD_pval5e8_d1k_reverse_remove.jpeg" width="30%">
</figure>


#### lymphocyte count(lymph)

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv23_lymph_pval5e8_d1k_reverse.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv78_lymph_pval5e8_d1k_reverse.jpeg" width="30%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv23_lymph_pval5e8_d1k_reverse_remove.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv78_lymph_pval5e8_d1k_reverse_remove.jpeg" width="30%">
</figure>

#### red blood cell count(wbc) 

The rbc~lv42 and rbc~lv59 pairs are not very promising pairs when considering SNPs at fdr<0.2, but the fitting results at fdr <0.5 are better than the former results. So I post the plots here too.

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv82_rbc_pval5e8_d1k_reverse.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv42_rbc_pval5e8_d1k_reverse.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv59_rbc_pval5e8_d1k_reverse.jpeg" width="30%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv82_rbc_pval5e8_d1k_reverse_remove.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv42_rbc_pval5e8_d1k_reverse_remove.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv59_rbc_pval5e8_d1k_reverse_remove.jpeg" width="30%">
</figure>


#### asthma

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv68_ukb.allasthma_pval5e8_d1k_reverse.jpeg" width="30%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv68_ukb.allasthma_pval5e8_d1k_reverse_remove.jpeg" width="30%">
</figure>

#### white blood cell count

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv119_wbc_pval5e8_d1k_reverse.jpeg" width="30%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv119_wbc_pval5e8_d1k_reverse_remove.jpeg" width="30%">
</figure>


## Results - pval < 5e-8 & association test covariants: 10 PCs + GTEx:Sequencing platform,Sequencing protocol,Sex 

### Summary table

I used 'qvalue' R package to compute the fdr from p-values for each SNP and made a table to show the number of SNPs that pass the threshold. The thresholds are 'fdr < 0.1','fdr < 0.2','pval < 5e-8'.

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/sum_sep_tog_pval_pliercanon_wgpc_ldcl_d1k_noageco.rdata")
DT::datatable(summary, options = list(pageLength =20))
```


### Info table

For each trait, I made a table to show the info of snps with fdr>0.2 in the factor ~ SNP + genotype pcs association test. For each trait,The LVs have more than one significant SNPs with FDR<0.2 are included. 

The suffix '_assoc' here means that results are from factor ~ SNP + genotype pcs association test. The suffix '_gwas' here means results are from original GWAS results files. For EUR.CD, EUR.IBD, EUR.UC, the effectsize_gwas here means 'ln(OR)', for others, it means 'beta'. 

'snp_ld' here means the snps that in LD with the snp in each line.'ld_r2' means the LD r-squared which is corresponding to the 'snp_ld' column. 'cis-eqtl' column indicates whether the snp is a cis-eqtl according to GTEx data. 'cis_gene_hgnc' and 'cis_gene_hgnc' is the genes that the snp influence when it act as cis-eqtl. 'func' and 'func_gene' are obtained from ANNOVAR, which indicating the snp function within the genes.

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/info_pval5e8_pliercanon_ld_d1k_noageco.rdata")
DT::datatable(final_full, options = list(pageLength =20))
```

### Effect size plots

For each trait, I made a plot of association with LV(indicating by beta in GWAS) vs association with trait(indicating by ln(odds ratio) or beta in GWAS) to show if the variants have the correlated effect direction. The effect sizes of Catalog GWAS and factor association tests are harmonized by [TwoSampleMR](https://mrcieu.github.io/TwoSampleMR/articles/introduction.html) R package to make the effect alleles in these two analysis identical. The LVs have more than one significant SNPs with FDR<0.2 are included in the plotting.Besides, for each plots, I fitted the points with intercept = 0. The pvalues and r-squared are shown on the plots.

#### BMI 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv3_bmi_pval5e8_d1k_noageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv27_bmi_pval5e8_d1k_noageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv30_bmi_pval5e8_d1k_noageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv76_bmi_pval5e8_d1k_noageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv90_bmi_pval5e8_d1k_noageco.jpeg" width="30%">
</figure>


#### eosinophil count(eo) 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv125_eo_pval5e8_d1k_noageco.jpeg" width="30%">
</figure>

#### Crohn's disease(CD)

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv49_EUR.CD_pval5e8_d1k_noageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv126_EUR.CD_pval5e8_d1k_noageco.jpeg" width="30%">
</figure>

#### IBD

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv98_EUR.IBD_pval5e8_d1k_noageco.jpeg" width="30%">
</figure>

#### Ulcerative colitist(UC)

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv23_EUR.UC_pval5e8_d1k_noageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv24_EUR.UC_pval5e8_d1k_noageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv37_EUR.UC_pval5e8_d1k_noageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv108_EUR.UC_pval5e8_d1k_noageco.jpeg" width="30%">
</figure>


#### granulocyte count(gran)

None of the LVs have >1 SNPs at FDR<0.2.


#### lymphocyte count(lymph)

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv23_lymph_pval5e8_d1k_noageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv26_lymph_pval5e8_d1k_noageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv37_lymph_pval5e8_d1k_noageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv65_lymph_pval5e8_d1k_noageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv78_lymph_pval5e8_d1k_noageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv123_lymph_pval5e8_d1k_noageco.jpeg" width="30%">
</figure>

#### myeloid white cell count (myeloid wbc)

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv94_myeloid_wbc_pval5e8_d1k_noageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv119_myeloid_wbc_pval5e8_d1k_noageco.jpeg" width="30%">
</figure>

#### neutrophil count (neut)

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv95_neut_pval5e8_d1k_noageco.jpeg" width="30%">
</figure>

#### plt

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv39_plt_pval5e8_d1k_noageco.jpeg" width="30%">
</figure>

#### rbc

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv3_rbc_pval5e8_d1k_noageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv20_rbc_pval5e8_d1k_noageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv70_rbc_pval5e8_d1k_noageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv43_rbc_pval5e8_d1k_noageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv82_rbc_pval5e8_d1k_noageco.jpeg" width="30%">
</figure>

#### T2D

None of the LVs have >1 SNPs at FDR<0.2.

#### asthma

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv21_ukb.allasthma_pval5e8_d1k_noageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv36_ukb.allasthma_pval5e8_d1k_noageco.jpeg" width="30%">
</figure>

#### wbc

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv119_wbc_pval5e8_d1k_noageco.jpeg" width="30%">
</figure>

### Effect size plots- more SNPs

For some promising trait-factor pairs , I relaxed the fdr threshold of the SNPs that used to make effect size plots(from 0.2 to 0.3/0.5)


#### BMI 

The BMI~lv90 is not very promising pairs when considering SNPs at fdr<0.2, but the fitting result at fdr <0.5 is better than the former result . So I post the plots here too.

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv3_bmi_pval5e8_d1k_noageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv3_bmi_pval5e8_d1k_noageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv3_bmi_pval5e8_d1k_noageco_fdr05.jpeg" width="30%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv27_bmi_pval5e8_d1k_noageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv27_bmi_pval5e8_d1k_noageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv27_bmi_pval5e8_d1k_noageco_fdr05.jpeg" width="30%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv76_bmi_pval5e8_d1k_noageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv76_bmi_pval5e8_d1k_noageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv76_bmi_pval5e8_d1k_noageco_fdr05.jpeg" width="30%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv90_bmi_pval5e8_d1k_noageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv90_bmi_pval5e8_d1k_noageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv90_bmi_pval5e8_d1k_noageco_fdr05.jpeg" width="30%">
</figure>


#### lymphocyte count(lymph)

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv26_lymph_pval5e8_d1k_noageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv26_lymph_pval5e8_d1k_noageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv26_lymph_pval5e8_d1k_noageco_fdr05.jpeg" width="30%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv78_lymph_pval5e8_d1k_noageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv78_lymph_pval5e8_d1k_noageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv78_lymph_pval5e8_d1k_noageco_fdr05.jpeg" width="30%">
</figure>

#### red blood cell count(wbc)

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv82_rbc_pval5e8_d1k_noageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv82_rbc_pval5e8_d1k_noageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv82_rbc_pval5e8_d1k_noageco_fdr05.jpeg" width="30%">
</figure>


#### asthma

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv36_ukb.allasthma_pval5e8_d1k_noageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv36_ukb.allasthma_pval5e8_d1k_noageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv36_ukb.allasthma_pval5e8_d1k_noageco_fdr05.jpeg" width="30%">
</figure>

#### white blood cell count

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv119_wbc_pval5e8_d1k_noageco.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv119_wbc_pval5e8_d1k_noageco_fdr03.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv119_wbc_pval5e8_d1k_noageco_fdr05.jpeg" width="30%">
</figure>


### Effect size plots - checking the reverse causality

To check if the effect size correlation is due to reverse causality: i.e. trait -> LV (trait causally affect LV), instead of LV -> trait (which is what we like to see). I used all SNPs associated with traits(pval<5E-8). The x-axis is the effects of these SNPs on trait, and y-axis is the effects on LV. 

Some pair show p < 0.05, the result may be driven by the possible causal effect of LV -> trait. To test this, I removed the SNPs that are associated with LVs at FDR < 0.2 and made the plots again.

#### BMI 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv3_bmi_pval5e8_d1k_noageco_reverse.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv27_bmi_pval5e8_d1k_noageco_reverse.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv76_bmi_pval5e8_d1k_noageco_reverse.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv90_bmi_pval5e8_d1k_noageco_reverse.jpeg" width="30%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv3_bmi_pval5e8_d1k_noageco_reverse_remove.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv27_bmi_pval5e8_d1k_noageco_reverse_remove.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv76_bmi_pval5e8_d1k_noageco_reverse_remove.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv90_bmi_pval5e8_d1k_noageco_reverse_remove.jpeg" width="30%">
</figure>

#### lymphocyte count(lymph)

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv26_lymph_pval5e8_d1k_noageco_reverse.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv78_lymph_pval5e8_d1k_noageco_reverse.jpeg" width="30%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv26_lymph_pval5e8_d1k_noageco_reverse_remove.jpeg" width="30%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv78_lymph_pval5e8_d1k_noageco_reverse_remove.jpeg" width="30%">
</figure>

#### red blood cell count(wbc) 


<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv82_rbc_pval5e8_d1k_noageco_reverse.jpeg" width="30%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv82_rbc_pval5e8_d1k_noageco_reverse_remove.jpeg" width="30%">
</figure>


#### asthma

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv36_ukb.allasthma_pval5e8_d1k_noageco_reverse.jpeg" width="30%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv36_ukb.allasthma_pval5e8_d1k_noageco_reverse_remove.jpeg" width="30%">
</figure>

#### white blood cell count

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv119_wbc_pval5e8_d1k_noageco_reverse.jpeg" width="30%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_lv119_wbc_pval5e8_d1k_noageco_reverse_remove.jpeg" width="30%">
</figure>



