---
title: "KEGG PCs summary for 6 cell types from CEDAR dataset (pathways have > 40 genes only)"
author: "XSun"
date: "2021-06-28"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

## **Introduction**

In this part, we just used the pathways have > 40 genes.

## **Summary** 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/CEDAR_cell_sum_40genes.png" width="80%">
</figure>

## **Detailed results for 6 cell types**

We just listed the pairs show sign consistancy (at p-value < 0.05 in resampling) here

We did some literature searching for some disease related pairs (p-value < 0.01 in resampling) in monocyte and B cell.

### monocyte CD14

#### Histogram to show the number of SNPs that each pair contains

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/snp_count_cd14_cedar_qc_kegg_40genes.jpeg" width="80%">
</figure>

#### effectsize fitting summary 

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/pwy_name_cd14_cedar_qc_kegg_40genes.rdata")
DT::datatable(final_full, options = list(pageLength =5))
```

EUR.IBD_pwy121_pc1:  [Dysregulation in the NF-κβ pathway and factors that regulate it lead to a state of uncontrolled inflammation and altered immunity, as typically observed in IBD](https://www.frontiersin.org/articles/10.3389/fped.2018.00317/full)

ukb.allasthma_pwy155_pc1 [Taken together, these findings demonstrate that the Wnt/β-catenin signaling pathway is highly expressed and regulates the development of airway remodeling in chronic asthma](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4686695/)

EUR.UC_pwy97_pc2 [Numerous miRNAs were found to be differentially expressed in IBD and were hypothesized to act as critical mediators in the IBD pathogenesis by regulating proteins in vital regulatory pathways such as in the expression of inflammatory cytokines](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6077315/)

#### Checking the reverse causality

**The suffix "_reverse" are the results for traits -> LV (using all candidate SNPs after LD Clumping), the suffix "_rremove" are the results for traits -> LV (using all candidate SNPs after LD Clumping but remove the SNPs have fdr < 0.2)**

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/mbe_cd14_cedar_qc_kegg_40genes.rdata")
DT::datatable(sum, options = list(pageLength =5))
```

### B cell

#### Histogram to show the number of SNPs that each pair contains

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/snp_count_bcell_cedar_qc_kegg_40genes.jpeg" width="80%">
</figure>

#### effectsize fitting summary 

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/pwy_name_bcell_cedar_qc_kegg_40genes.rdata")
DT::datatable(final_full, options = list(pageLength =5))
```

EUR.CD_pwy180_pc1: [Therefore, CLRs, as innate immune receptors, may play vital roles in IBD progression.](https://academic.oup.com/gastro/article/7/5/312/5530700)

EUR.UC_pwy163_pc5: [The role of the Hippo pathway in the pathogenesis of inflammatory bowel disease](https://www.nature.com/articles/s41419-021-03395-3)


#### Checking the reverse causality

**The suffix "_reverse" are the results for traits -> LV (using all candidate SNPs after LD Clumping), the suffix "_rremove" are the results for traits -> LV (using all candidate SNPs after LD Clumping but remove the SNPs have fdr < 0.2)**

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/mbe_bcell_cedar_qc_kegg_40genes.rdata")
DT::datatable(sum, options = list(pageLength =5))
```


### T cell(CD4)

#### Histogram to show the number of SNPs that each pair contains

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/snp_count_tcell_cedar_qc_kegg_40genes.jpeg" width="80%">
</figure>

#### effectsize fitting summary 

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/pwy_name_tcell_cedar_qc_kegg_40genes.rdata")
DT::datatable(final_full, options = list(pageLength =5))
```


#### Checking the reverse causality

**The suffix "_reverse" are the results for traits -> LV (using all candidate SNPs after LD Clumping), the suffix "_rremove" are the results for traits -> LV (using all candidate SNPs after LD Clumping but remove the SNPs have fdr < 0.2)**

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/mbe_tcell_cedar_qc_kegg_40genes.rdata")
DT::datatable(sum, options = list(pageLength =5))
```

[Together these data support the hypothesis that TNF plays a beneficial role in enhancing Wnt/β-catenin signaling during ulcer healing in IBD.](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5568528/)

### neutrophils CD15

#### Histogram to show the number of SNPs that each pair contains

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/snp_count_cd15_cedar_qc_kegg_40genes.jpeg" width="80%">
</figure>

#### effectsize fitting summary 

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/pwy_name_cd15_cedar_qc_kegg_40genes.rdata")
DT::datatable(final_full, options = list(pageLength =5))
```


#### Checking the reverse causality

**The suffix "_reverse" are the results for traits -> LV (using all candidate SNPs after LD Clumping), the suffix "_rremove" are the results for traits -> LV (using all candidate SNPs after LD Clumping but remove the SNPs have fdr < 0.2)**

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/mbe_cd15_cedar_qc_kegg_40genes.rdata")
DT::datatable(sum, options = list(pageLength =5))
```



### T cell(CD8)

#### Histogram to show the number of SNPs that each pair contains

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/snp_count_thy_cedar_qc_kegg_40genes.jpeg" width="80%">
</figure>

#### effectsize fitting summary 

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/pwy_name_thy_cedar_qc_kegg_40genes.rdata")
DT::datatable(final_full, options = list(pageLength =5))
```


#### Checking the reverse causality

**The suffix "_reverse" are the results for traits -> LV (using all candidate SNPs after LD Clumping), the suffix "_rremove" are the results for traits -> LV (using all candidate SNPs after LD Clumping but remove the SNPs have fdr < 0.2)**

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/mbe_thy_cedar_qc_kegg_40genes.rdata")
DT::datatable(sum, options = list(pageLength =5))
```


### Platelet 

#### Histogram to show the number of SNPs that each pair contains

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/snp_count_plt_cedar_qc_kegg_40genes.jpeg" width="80%">
</figure>

#### effectsize fitting summary 

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/pwy_name_plt_cedar_qc_kegg_40genes.rdata")
DT::datatable(final_full, options = list(pageLength =5))
```


#### Checking the reverse causality

**The suffix "_reverse" are the results for traits -> LV (using all candidate SNPs after LD Clumping), the suffix "_rremove" are the results for traits -> LV (using all candidate SNPs after LD Clumping but remove the SNPs have fdr < 0.2)**

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/mbe_plt_cedar_qc_kegg_40genes.rdata")
DT::datatable(sum, options = list(pageLength =5))
```

