---
title: "Overview on CEDAR data set"
author: "XSun"
date: "2021-07-15"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

## **Introduction**

We used CEDAR data from [IBD risk loci are enriched in multigenic regulatory modules encompassing putative causative genes](https://www.nature.com/articles/s41467-018-04365-8). The data can be downloaded from ArrayExpress under accession codes [E-MTAB-6666](https://www.ebi.ac.uk/arrayexpress/experiments/E-MTAB-6666/) and [E-MTAB-6667](https://www.ebi.ac.uk/arrayexpress/experiments/E-MTAB-6667/)


## **Genotype data** 

### 1. Data processing

The raw genotype data carried out by Illumina HumanOmniExpress-12 v1_A genotyping arrays were processed follows: 

1)	Format Conversion

  The raw IDAT files were converted into VCF format using gtc2vcf tool (https://github.com/freeseek/gtc2vcf).
        
2)	Quality Control

-	The alleles were aligned with the 1000 Genomes Phase 3 reference panel and excluding variants that could not be aligned using Genotype harmonizer (version 1.4.20).

-	Variants with Hardy-Weinberg equilibrium p-value>10âˆ’6, missingness >0.05 and MAF <0.01 were excluded.

-	Samples with more than 95% of the variants missing were excluded.

3)	Imputation

  After quality control, 602,849 autosomal genetic variants from 323 individuals were included for imputation. Michigan Imputation Server[148] with the 1000 Genomes Phase 3 reference panel was used to perform phasing and imputation.
        
  Bcftools (version 1.9.0) were used to filter the genetic variants after imputation. 8,045,943 variants on autosomes with imputation quality score R2 < 0.8 and MAF < 0.01 were included finally. 


### 2. Genotype PCs

Using whole genome to compute principle components usually captures linkage disequilibrium structure and cannot detect the sample outliers. So, in this chapter, [bigsnpr](https://academic.oup.com/bioinformatics/article/36/16/4449/5838185) was employed to compute the 10 PCs using as covariates in association tests to avoid these issues. The computation pipeline follows:

1)	The relatedness of the samples using was first computed using 2e-4.5 as the cutoff to find the second-degree relations with snp_plinkKINGQC() function from bigsnpr.

2)	After removing the related individuals, the sample outliers was computed using function prob_dist() and visualized by geom_histogram() from ggplot2. Figure below shows, s=1.0 is a reasonable threshold to remove the sample outliers. 

3)	Before doing PCA, the bigsnpr will do LD Pruning using R2 = 0.2 as threshold to remove the SNPs in LD.

4)	Finally, we projected the removed samples using the PCA results in step 3) and obtained the final 10 PCs.



The table below shows the sample size and variant number we used in PCA

|              | sample size - origin | sample size - related removed | sample size - outlier removed | variants number - origin  | variants number - final  |
|--------------|----------------------|-------------------------------|-------------------------------|---------------------------|--------------------------|
| B cell       | 276                  | 268                           | 259                           | 8045943                   | 259604                   |
| CD14         | 282                  | 274                           | 266                           | 8045943                   | 261937                   |
| CD15         | 270                  | 264                           | 258                           | 8045943                   | 261049                   |
| T CELL(CD4+) | 299                  | 291                           | 279                           | 8045943                   | 269355                   |
| T CELL(CD8+) | 286                  | 280                           | 272                           | 8045943                   | 266120                   |
| Platelet     | 237                  | 235                           | 233                           | 8045943                   | 251819                   |


The plot below shows the outliers we removed in step 3.

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/outliers.jpg" width="100%">
</figure>

The plot below shows the PVE of each PCs.

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/PVE.jpg" width="100%">
</figure>


## **Gene expression data**

CEDAR gene expression data generated by A-MEXP-2072 - Illumina HumanHT-12_V4_0_R2_15002873_B contain 6 cell types: CD4+ T cells, CD8+ T cells, CD19+ B cells and CD14+ monocytes, CD15+ neutrophils and platelets. 

### 1. Data processing

1)	Normalization: Fluorescence intensities were Log2 transformed and Robust Spline Normalized (RSN) with Lumi.

2)	The probes presented in more than 75% of our samples for given cell type were kept.

Table below shows the summary after gene expression data procession.


|     Cell type            |     Sample size    |     Number of probes    |     Number of genes(gene   symbol ID)    |
|--------------------------|--------------------|-------------------------|------------------------------------------|
|     CD4+ T cells         |     299            |     17915               |     16897                                |
|     CD8+ T cells         |     286            |     17615               |     16646                                |
|     CD19+ B cells        |     276            |     16866               |     15774                                |
|     CD14+ monocytes      |     282            |     17506               |     16374                                |
|     CD15+ neutrophils    |     270            |     14968               |     13905                                |
|     platelets            |     237            |     9027                |     8388                                 |

### 2. PEER factors

We used PEER(https://github.com/PMBio/peer/wiki) under default settings to infer PEER factors for each expression data set. For the data set have > 250 samples, we inferred 45 factors. For others, we inferred 30 factors. 