---
title: "Genetic relationship between the expression factors and traits"
author: "XSun"
date: "2021-02-19"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---


# **Studying effects of factors on phenotypes with Mendelian Randomization : factors are inferred from expression matrix**

We'd like to show that the factors have a causal effect on outcome(traits), but it's difficult to establish.  
So we state a weaker hypothesis first: GWAS variants of a trait often converge to a relatively small number of processes, which can be represented as expression factors.

Then we strengthen hypothesis: if the effects of SNPs on LV is correlated with their effects on trait when there is a causal effect of LV on trait.


## *1. Effect size & Colocalization tests based on FDR cutoff for different factors*

To address issues with respect to computational time and multiple testing, I confined our analysis to those SNPs present in the [Catalog of Published GWAS](https://www.ebi.ac.uk/gwas/). I selected the SNPs with pval < 5e-8/5e-7. Then, I did association tests for the factors with the SNPs.

Each setting below contains effectsize analysis&resampling, colocalization analysis, enrichment analysis.

### PLIER-Canonical factors

- [Whole Blood](https://xsun1229.github.io/factor_analysis/catalog_gwas_pliercanon_sep_ldclumping_r01d1000.html)

- [Whole Blood, filtered by pval<5e-7](https://xsun1229.github.io/factor_analysis/catalog_gwas_pliercanon_sep_ldclumping_r01d1000_pval5e7.html)

- [Adipose Subcutaneous](https://xsun1229.github.io/factor_analysis/catalog_gwas_pliercanon_sep_ldclumping_r01d1000_adipose_sub.html)

- [Skeletal muscle](https://xsun1229.github.io/factor_analysis/catalog_gwas_pliercanon_sep_ldclumping_r01d1000_muscle.html)

- [Liver](https://xsun1229.github.io/factor_analysis/catalog_gwas_pliercanon_sep_ldclumping_r01d500d_liver.html)

- [GEUVADIS](https://xsun1229.github.io/factor_analysis/catalog_gwas_pliercanon_sep_ldclumping_r01d1000_geu.html)


### PEER 

- [Whole Blood](https://xsun1229.github.io/factor_analysis/catalog_gwas_peer_sep_ldclumping_r01d1000.html)

### FLASH

- [Whole Blood](https://xsun1229.github.io/factor_analysis/catalog_gwas_flash_sep_ldclumping_r01d1000.html)


## *2.SuSiE analysis*

We used SuSiE to test if our colocalization results are reliable. 

[For more details](https://xsun1229.github.io/factor_analysis/Susie_results.html)

## *3.EM algorithm - PLIER-Canonical factors*

- [Real data](https://xsun1229.github.io/factor_analysis/EM_algorithm_pliercanon.html)

- [Simulated data](https://xsun1229.github.io/factor_analysis/EM_simulation.html)

- [EM_pi_j_estimation](https://xsun1229.github.io/factor_analysis/EM_pij_estimation.html)

- [Resampling summary table](https://xsun1229.github.io/factor_analysis/EM_resampling_sum.html) 

- [Resampling summary table -- pval5e7](https://xsun1229.github.io/factor_analysis/EM_resampling_sum_pval5e7.html) 

- [Using truncated normal distribution for IV and Non-IV variables](https://xsun1229.github.io/factor_analysis/EM_algorithm_pliercanon_trunc.html)


# **Studying effects of factors on phenotypes with Mendelian Randomization : factors are inferred from known pathways**

- KEGG pathways

We have 332 human pathways from KEGG. For each pathway, we inferred 5 PCs and 5 PEER factors from the gene expression matrix. We obtained 1646 PCs and 1634 PEER factors in total. 

[PCs](https://xsun1229.github.io/factor_analysis/kegg_pcs_factor_traits.html)

[PCs_regress out 5 GTEx PEER factors](https://xsun1229.github.io/factor_analysis/kegg_pcs_factor_traits_regress_5.html)

[PCs_regress out 5 GTEx PEER factors & candidate SNPs: 'AND' LD Clumping](https://xsun1229.github.io/factor_analysis/kegg_pcs_factor_traits_regress_5_d500d.html)

[PCs_regress out 5 GTEx PEER factors & candidate SNPs: 'AND' LD Clumping(r2 = 0.1, d= 1Mb)](https://xsun1229.github.io/factor_analysis/kegg_pcs_factor_traits_regress_5_d1k.html)


[PCs_regress out 5 GTEx PEER factors & candidate SNPs: 'AND' LD Clumping Liver eqtls](https://xsun1229.github.io/factor_analysis/kegg_pcs_factor_traits_regress_5_d500d_liver.html)


[PEER](https://xsun1229.github.io/factor_analysis/kegg_peer_factor_traits.html)


[Compare factors & factors after regressing out GTEx PEER factors](https://xsun1229.github.io/factor_analysis/compare_regress.html)


- Reactome pathways

We have 1866 Reactome pathways. Combining with our GTEx data, some of them have less than 5 genes that have expression data. Since we will infer 5 PCs for each pathway, we just use the pathways have more than 5 genes. The number of pathways we use for whole blood is 1221, the number for liver is 1262. 

[Whole Blood: PCs_regress out 5 GTEx PEER factors & candidate SNPs: 'AND' LD Clumping](https://xsun1229.github.io/factor_analysis/reactome_pcs_factor_traits_regress_5_d500d.html)

[Whole Blood: PCs_regress out 5 GTEx PEER factors & candidate SNPs: 'AND' LD Clumping (r2 = 0.1, d= 1Mb)](https://xsun1229.github.io/factor_analysis/reactome_pcs_factor_traits_regress_5_d1k.html)

[Liver: PCs_regress out 5 GTEx PEER factors & candidate SNPs: 'AND' LD Clumping](https://xsun1229.github.io/factor_analysis/reactome_pcs_factor_traits_regress_5_d500d_liver.html)

[Liver: PCs_regress out 5 GTEx PEER factors & candidate SNPs: 'AND' LD Clumping(r2 = 0.1, d= 1Mb)](https://xsun1229.github.io/factor_analysis/reactome_pcs_factor_traits_regress_5_d1k_liver.html)


- Comparing the enrichments under different settings

[GTEx](https://xsun1229.github.io/factor_analysis/factor_traits_setting_compare.html)

[CEDAR - original genotype](https://xsun1229.github.io/factor_analysis/factor_traits_setting_compare_CEDAR.html)

[CEDAR_thymocyte](https://xsun1229.github.io/factor_analysis/CEDAR_thymocyte.html)

[CEDAR - imputed genotype](https://xsun1229.github.io/factor_analysis/factor_traits_setting_compare_CEDAR_imputed.html)


# **GWAS convergence on pathways using cis-eQTLs**

[For more details](https://xsun1229.github.io/factor_analysis/cispathway.html)


