---
title: "Comparing the enrichments under different settings_CEADR"
author: "XSun"
date: "2021-05-31"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

## Introduction

In this part, the expression data are the same with [CEDAR_origin](https://xsun1229.github.io/factor_analysis/factor_traits_setting_compare_CEDAR.html). The genotype data were imputed using [Michigan Imputation Server](https://imputationserver.sph.umich.edu/index.html#!) 

In the imputation part, we used 1000G Phase 3 V5 as our reference panel. Mixed population was selected. After imputation, we used rsq > 0.8 to filter the SNPs. We obtained 9,571,582 SNPs. Then we filtered the SNPs using MAF > 0.01 and got 7,741,014 variates. 

Now, we have reasonable SNP numbers when filtering by GWAS p-values. 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/SNPs_count.png" width="100%">
</figure>


## **Comparison under different settings**

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/cedar_imputed_1.png" width="100%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/cedar_imputed_2.png" width="100%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/cedar_imputed_3.png" width="100%">
</figure>

## **Pathway details for some cell types**

### B cell - KEGG

The table below shows the pathway details for the pathway have > 40 genes && pairs have p-value < 0.01 in resampling. 

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/summary_snp3_d1k_cedar_bcell_kegg.rdata")
DT::datatable(final_full, options = list(pageLength =5))
```


1. [Therefore, CLRs, as innate immune receptors, may play vital roles in IBD progression.](https://academic.oup.com/gastro/article/7/5/312/5530700)

2. [In the past few years, studies have revealed the importance of the Hippo pathway in gastrointestinal tract physiology and gastrointestinal diseases, such as colorectal cancer and IBD.](https://www.nature.com/articles/s41419-021-03395-3)


<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_pwy106_pc3_EUR.CD_pval5e8_cedar_bcell_kegg.jpeg" width="80%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_pwy163_pc5_EUR.UC_pval5e8_cedar_bcell_kegg.jpeg" width="80%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_pwy180_pc1_EUR.CD_pval5e8_cedar_bcell_kegg.jpeg" width="80%">
</figure>


### Monocyte(CD14+) - KEGG


The table below shows the pathway details for the pathway have > 40 genes && pairs have p-value < 0.01 in resampling. 

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/summary_snp3_d1k_cedar_cd14_kegg.rdata")
DT::datatable(final_full, options = list(pageLength =5))
```

1. [Dysregulation in the NF-κβ pathway and factors that regulate it lead to a state of uncontrolled inflammation and altered immunity, as typically observed in IBD](https://www.nature.com/articles/srep04112)

2. [ Classically, IBD was thought to be primarily mediated by Th1 cells in CD or Th2 cells in UC, but it is now known that Th17 cells and their related cytokines are crucial mediators in both conditions.](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4005031/)

3. [The present study supports the concept of the Y. enterocolitica infection as a trigger of chronic IBD. ](https://pubmed.ncbi.nlm.nih.gov/15967332/)

7. [ Evidence has implicated both β-catenin-dependent and β-catenin-independent WNT axes with asthma. Its deregulation is important for inflammation, airway remodelling, and airway hyperresponsives](https://www.sciencedirect.com/science/article/pii/S135964461730079X)


<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_pwy121_pc1_EUR.IBD_pval5e8_cedar_CD14_kegg.jpeg" width="80%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_pwy186_pc4_EUR.UC_pval5e8_cedar_CD14_kegg.jpeg" width="80%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_pwy274_pc5_EUR.UC_pval5e8_cedar_CD14_kegg.jpeg" width="80%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_pwy91_pc5_EUR.UC_pval5e8_cedar_CD14_kegg.jpeg" width="80%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_pwy279_pc4_allergy_pval5e8_cedar_CD14_kegg.jpeg" width="80%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_pwy294_pc1_lymph_pval5e8_cedar_CD14_kegg.jpeg" width="80%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_pwy155_pc1_ukb.allasthma_pval5e8_cedar_CD14_kegg.jpeg" width="80%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_pwy112_pc2_wbc_pval5e8_cedar_CD14_kegg.jpeg" width="80%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_pwy222_pc1_wbc_pval5e8_cedar_CD14_kegg.jpeg" width="80%">
</figure>

### Leukocyte(CD15+) -KEGG

The table below shows the pathway details for the pathway have > 40 genes && pairs have p-value < 0.01 in resampling. 

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/summary_snp3_d1k_cedar_cd15_kegg.rdata")
DT::datatable(final_full, options = list(pageLength =5))
```

1. [Oxytocin may provide new potential insights into the pathogenesis and therapy of IBD](https://portlandpress.com/clinsci/article/133/18/1977/220436)

4. [The pathogenesis of inflammatory bowel diseases (IBD) seems to be associated with alterations of immunoregulation. Several lines of evidence suggest that estrogens play a role in the modulation of immune responses and may be related to the etiology of IBD.](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6651503/)


We did enrichment analysis for the genes in 'EUR.IBD_pwy256_pc5'. We tried both ORA and GSEA.

ORA: We just used the gene name in pwy256, the enrichment categories is geneontology_Biological_Process. The reference list: all mapped entrezgene IDs from the selected platform affy_hugene_2_0_st_v1. The detailed results are [here](http://www.webgestalt.org/results/1623581009/#)

The fdrs of the pathways is shown in the table below.

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/pwy256_kegg_cd15_all_cedar.rdata")
DT::datatable(final_full, options = list(pageLength =5))
```

GSEA: We used gene names and gene loading for pc5. The enrichment categories is geneontology_Biological_Process. The fdrs of the reported pathways are all > 0.5. None of them are immune/inflammation pathways The detailed results are [here](http://www.webgestalt.org/results/1623552821/#).

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_pwy223_pc5_EUR.CD_pval5e8_cedar_CD15_kegg.jpeg" width="80%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_pwy272_pc3_EUR.CD_pval5e8_cedar_CD15_kegg.jpeg" width="80%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_pwy256_pc5_EUR.IBD_pval5e8_cedar_CD15_kegg.jpeg" width="80%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_pwy217_pc4_EUR.UC_pval5e8_cedar_CD15_kegg.jpeg" width="80%">
</figure>


