---
title: "Genetic relationship between the expression factors and traits"
author: "XSun"
date: "2021-02-19"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---


# **1. Introduction**

In this part, we are trying to study effects of factors on phenotypes with Mendelian Randomization. Factors are inferred from expression matrix

The hypothesis is that the factors have a causal effect on outcome(traits). But it's difficult to establish.  
So we state a weaker hypothesis first: GWAS variants of a trait often converge to a relatively small number of processes, which can be represented as expression factors.

Then we strengthen hypothesis: if the effects of SNPs on LV is correlated with their effects on trait when there is a causal effect of LV on trait.

We tried GTEx and CEDAR data set for the analysis in this part.

# **2. Pipeline**

1) We first selected some disease/trait-associated SNPs, using p-values < 5E-8 in GWAS summary statistics as cutoff. The GWAS summary statistics data we used are [here](https://xsun1229.github.io/factor_analysis/GWAS_data.html)

2) Then we did 3-step LD Clumping for the SNPs in step 2 to find independent variants to be used in MR. 

- Use PLINK to do the preliminary filter: in 1Mb block, the SNPs have LD R2 > 0.1 will be filtered by their p-values
- For the remaining SNPs within 1Mb, keep the one have lowest p-value, remove the others.
- Compute the LD R2 for the remaining SNPs, for the SNP pairs have R2 > 0.1, filter them again according their p-values.

3) Next, we did association tests for expression factors ~ SNPs + 10 genotype PCs + batch +age +sex 

4) We recorded the trait-factor pairs that have more than one SNPs have FDR<0.2. Fitted beta.outcome ~ beta.exposure + 0 using the SNPs associated to these pairs. Recorded the fitting p-values. 

5) To make our results more reliable, we did resampling. We resampled the beta.outcome and beta.exposure from the origin dataset(the data set containing all SNPs) without replacement, made the fitting again and recorded the r-squareds to compute the p-values in resampling. The ‘p-value from resampling’ is computed by: (number of more extreme values)/(times of resampling).The resampling was repeated 1000 times.

6) To check the reverse causality, we exchanged the beta.outcome and beta.exposure of all SNPs associated with traits(pval<5E-8) to compute the MBE estimator and the p-values of the estimator.

# **3. GTEx**

## **3.1 Data**

### 3.1.1 Expression factors

We tried four types of expression factors: 1)PLIER factors 2)PCs from KEGG/Reactome pathways 3)PEER factors 4)FLASH factors 5)PEER factors from KEGG/Reactome pathways. They are the same factors with those used in the prediction model training part. 

When using PCs, we regressed out the first 5 PEER factors computed from the GTEx expression data.

The KEGG contains 332 pathways. We have 1866 Reactome pathways. Combining with our GTEx data, some of them have less than 5 genes that have expression data. Since we will infer 5 PCs for each pathway, we just use the pathways have more than 5 genes. The number of pathways we use for whole blood is 1221, the number for liver is 1262. 

| Type                       | Factor number |
|----------------------------|---------------|
| PLIER - Canonical pathways | 128           |
| PLIER - MSigDB pathways    | 98            |
| PCs - KEGG                 | 1646          |
| PCs - Reactome             | 6105          |
| PEER                       | 60            |
| FLASH                      | 100           |

### 3.1.2 Genotype data

GTEx genotype data. The variants have maf >0.01 were selected, variants on chrX were excluded). We have 10,403,249 variants in total.

## **3.2 Results under different settings**

### PLIER-MSigDB factors **(latest results 07/21)**

- [Whole Blood](https://xsun1229.github.io/factor_analysis/gtex_plier_d1k.html)

### PLIER-Canonical factors

- [Whole Blood](https://xsun1229.github.io/factor_analysis/catalog_gwas_pliercanon_sep_ldclumping_r01d1000.html)

For the factors in this part, we also tried EM algorithm to estimate the causal effects. [Details](https://xsun1229.github.io/factor_analysis/EM_websites.html) 

For the factors in this part, we used SuSiE to test if our colocalization results are reliable. 
[For more details](https://xsun1229.github.io/factor_analysis/Susie_results.html)

- [Whole Blood, first filtered by pval<5e-7](https://xsun1229.github.io/factor_analysis/catalog_gwas_pliercanon_sep_ldclumping_r01d1000_pval5e7.html)

- [Adipose Subcutaneous](https://xsun1229.github.io/factor_analysis/catalog_gwas_pliercanon_sep_ldclumping_r01d1000_adipose_sub.html)

- [Skeletal muscle](https://xsun1229.github.io/factor_analysis/catalog_gwas_pliercanon_sep_ldclumping_r01d1000_muscle.html)

- [Liver](https://xsun1229.github.io/factor_analysis/catalog_gwas_pliercanon_sep_ldclumping_r01d500d_liver.html)

- [GEUVADIS](https://xsun1229.github.io/factor_analysis/catalog_gwas_pliercanon_sep_ldclumping_r01d1000_geu.html)


### PEER 

- [Whole Blood](https://xsun1229.github.io/factor_analysis/catalog_gwas_peer_sep_ldclumping_r01d1000.html)

### FLASH

- [Whole Blood](https://xsun1229.github.io/factor_analysis/catalog_gwas_flash_sep_ldclumping_r01d1000.html)






### PCs - KEGG

[Whole blood](https://xsun1229.github.io/factor_analysis/kegg_pcs_factor_traits_regress_5_d1k.html)

[Liver](https://xsun1229.github.io/factor_analysis/kegg_pcs_factor_traits_regress_5_d500d_liver.html)

### PEER - KEGG

[Whole Blood](https://xsun1229.github.io/factor_analysis/kegg_peer_factor_traits.html)

### PCs - REACTOME

[Whole Blood](https://xsun1229.github.io/factor_analysis/reactome_pcs_factor_traits_regress_5_d1k.html)

[Liver](https://xsun1229.github.io/factor_analysis/reactome_pcs_factor_traits_regress_5_d1k_liver.html)













We compared the enrichment results under different settings(PCs from KEGG and REACTOME, PLIER - Canonical).

[Comparison](https://xsun1229.github.io/factor_analysis/factor_traits_setting_compare.html)


# **4. CEDAR**

## **4.1 Data**

### 4.1.1 CEDAR data processing 

[Data overview](https://xsun1229.github.io/factor_analysis/CEDAR_data_process.html)

### 4.1.2 Expression factors

We tried 2 types of expression factors: 1)PLIER factors 2)PCs from KEGG pathways. 

When using PCs, we regressed out the first 5 PEER factors computed from the GTEx expression data.

## **4.2 Results under different settings**

### PLIER - MSigDB factors **latest results 07/21**

[Details](https://xsun1229.github.io/factor_analysis/cedar_plier_26_qc_d1k_pruned.html)

### PC - KEGG **latest results 07/21**

[Details](https://xsun1229.github.io/factor_analysis/cedar_kegg_26_qc_d1k_pruned.html)







