---
title: "Genetic relationship between the expression factors and traits"
author: "XSun"
date: "2021-02-19"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---


# **1. Overview**

In this part, we are trying to study effects of factors on phenotypes with Mendelian Randomization. Factors are inferred from expression matrix

The hypothesis is that the factors have a causal effect on outcome(traits). But it's difficult to establish.  
So we state a weaker hypothesis first: GWAS variants of a trait often converge to a relatively small number of processes, which can be represented as expression factors.

Then we strengthen hypothesis: if the effects of SNPs on LV is correlated with their effects on trait when there is a causal effect of LV on trait.

Another hypothesis in this part is that if expression factors can influence the certain phenotype (trait or disease), the variants of the phenotype should be enriched in the factors.

We tried GTEx and CEDAR data set for the analysis in this part.

# **2. Data**

The GWAS summary statistics data we used are [here](https://xsun1229.github.io/factor_analysis/GWAS_data.html)


# **3. Pipeline**

## **3.1 Ascertaining disease/trait-associated variants**

1) The candidate variants associated with diseases/traits were selected using p-values < 5e-8 in GWAS summary statistics. 

2) Then we did 3-step LD Clumping for the SNPs in step 1) to find independent variants to be used in MR. 

- Use PLINK to do the preliminary filter: in 1Mb block, the SNPs have LD R2 > 0.1 will be filtered by their p-values
- For the remaining SNPs within 1Mb, keep the one have lowest p-value, remove the others.
- Compute the LD R2 for the remaining SNPs, for the SNP pairs have R2 > 0.1, filter them again according their p-values.

After step 1) and 2), for each trait, a subset of variants associated to it was obtained. Each pair from these variants has LD R2 < 0.1 and physical distance > 1Mb.

**[The number of trait-associated variants is here](https://xsun1229.github.io/factor_analysis/CEDAR_number_variants_each_trait)**

## **3.2 Sign consistency checking**

### 3.2.1 Association tests

1) we did association tests for expression factors ~ SNPs + 10 genotype PCs + batch covariates +age +sex

The 10 genotype PCs were computed using 2 approaches:

- Computing PCs on whole genome using PLINK1.9

- Pruning the whole genome first and compute the PCs (annotated as pruned_PCs, details are [here](https://xsun1229.github.io/factor_analysis/CEDAR_data_process.html#2_genotype_pcs))

2) We recorded the trait-factor pairs that have more than one SNPs have FDR<0.2. Fitted beta.outcome ~ beta.exposure + 0 using the SNPs associated to these pairs. Recorded the fitting p-values. 

3) We harmonized the data using TwoSampleMR R package

4) For the trait-factor pairs in step 2), fitting beta.outcome ~ beta.exposure + 0 using the variants associated to these pairs. Recorded the fitting p-values 

### 3.2.2 Resampling

To make our results more reliable, we did resampling. 

1) We resampled the beta.outcome and beta.exposure from the origin dataset(the data set containing all SNPs) without replacement, made the fitting again and recorded the r-squareds to compute the p-values in resampling. 

2) The ‘p-value from resampling’ is computed by: (number of more extreme values)/(times of resampling).The resampling was repeated 1000 times.

3) We computed the enrichment of the pairs show sign consistency than expect follows: enrichment=num_(pairs show sign consistancy)  /(num_(total pairs)  × threshold)

- We first tried to do resampling 1000 times for each pair have >2 SNPs at FDR < 0.2

- We also tried an adaptive method to do the resampling. In the first stage, 10e3 simulations will be performed. If the resulting empirical p value is less than 0.1, 10e4 simulations will be performed. If the empirical p value from 10e4 simulations is less than 0.001, the program will perform 10e6 simulations. [reference](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2896770/)


### 3.2.3 Checking reverse causality

To check the reverse causality, we exchanged the beta.outcome and beta.exposure of all SNPs associated with traits(pval<5E-8) to compute the MBE estimator and the p-values of the estimator.

Weighted mode-based estimate (MBE) was adopted to check the reverse causality. mr_mbe() function from MendelianRandomization R package(version 0.3.0) was used to compute the weighted MBE with default settings. 

## **3.3 Colocalization analysis**

We did colocalization analysis whether factor-association signals and GWAS signals at a locus are consistent with having a shared casual variant. The colocalization analysis was performed using the approximate Bayes factor test implemented by coloc.abf() function in ‘coloc’ R package. ‘coloc’ computes five posterior probabilities (PP0, PP1, PP2, PP3 and PP4), each corresponding to a hypothesis: H0, no association with either trait; H1, association with trait 1 but not with trait 2; H2, association with trait 2 but not with trait 1; H3, association with trait 1 and trait 2, two independent SNPs; H4, association with trait 1 and trait 2, one shared SNP. We ran ‘coloc’ with the default parameters and used PP4 to assess evidence of colocalization. 

We will do colocalization for the variants from the pair shows sign consistency, which are named as factor-QTLs by us. The colocalization was performed follows:

1)	For a factor-QTL, the variants less than 500kb were selected to do the analysis

2)	Association tests follows 3.3.3.1 were performed for these variants. The p-values were recorded.

3)	Colocalization analysis was did for the association signals in step 2) and GWAS signals.

4)	The colocalization results were visualized the using the ‘LocusCompareR’ package (version 1.0.0, downloaded from https://github.com/boxiangliu/locuscomparer).  


## **3.4 ORA enrichment analysis** 

ORA: For each factor, we sorted the genes that have non-zero loadings by their loadings, taking the top 25% or 100% as the gene set. The function database here is geneontology Biological Process. The reference set affy hugene 2 0 st v1. Minimum number of genes for a category is 5, maximum number of genes for a category is 2000(default settings). All categories that have fdr<0.05 are listed.

# **4. GTEx**

## **4.1 Data**

### 4.1.1 Expression factors

We tried four types of expression factors: 1)PLIER factors 2)PCs from KEGG/Reactome pathways 3)PEER factors 4)FLASH factors 5)PEER factors from KEGG/Reactome pathways. They are the same factors with those used in the prediction model training part. 

When using PCs, we regressed out the first 5 PEER factors computed from the GTEx expression data.

The KEGG contains 332 pathways. We have 1866 Reactome pathways. Combining with our GTEx data, some of them have less than 5 genes that have expression data. Since we will infer 5 PCs for each pathway, we just use the pathways have more than 5 genes. The number of pathways we use for whole blood is 1221, the number for liver is 1262. 

| Type                       | Factor number |
|----------------------------|---------------|
| PLIER - Canonical pathways | 128           |
| PLIER - MSigDB pathways    | 98            |
| PCs - KEGG                 | 1646          |
| PCs - Reactome             | 6105          |
| PEER                       | 60            |
| FLASH                      | 100           |

### 4.1.2 Genotype data

GTEx genotype data. The variants have maf >0.01 were selected, variants on chrX were excluded). We have 10,403,249 variants in total.

The genotype PCs used in this part are computed from whole genome, using PLINK1.9.

### 4.1.3 Association test covariates

For GTEx data, the batch covariates were sequencing platform and sequencing protocol. 10 genotype PCs were computed on whole genome using PLINK 1.9. 

## **4.2 Results under different settings**

### PLIER-MSigDB factors **(latest results 07/21)**

- [Whole Blood](https://xsun1229.github.io/factor_analysis/gtex_plier_d1k.html)

### PLIER-Canonical factors

- [Whole Blood](https://xsun1229.github.io/factor_analysis/catalog_gwas_pliercanon_sep_ldclumping_r01d1000.html)

For the factors in this part, we also tried EM algorithm to estimate the causal effects. [Details](https://xsun1229.github.io/factor_analysis/EM_websites.html) 

For the factors in this part, we used SuSiE to test if our colocalization results are reliable. 
[For more details](https://xsun1229.github.io/factor_analysis/Susie_results.html)

- [Whole Blood, first filtered by pval<5e-7](https://xsun1229.github.io/factor_analysis/catalog_gwas_pliercanon_sep_ldclumping_r01d1000_pval5e7.html)

- [Adipose Subcutaneous](https://xsun1229.github.io/factor_analysis/catalog_gwas_pliercanon_sep_ldclumping_r01d1000_adipose_sub.html)

- [Skeletal muscle](https://xsun1229.github.io/factor_analysis/catalog_gwas_pliercanon_sep_ldclumping_r01d1000_muscle.html)

- [Liver](https://xsun1229.github.io/factor_analysis/catalog_gwas_pliercanon_sep_ldclumping_r01d500d_liver.html)

- [GEUVADIS](https://xsun1229.github.io/factor_analysis/catalog_gwas_pliercanon_sep_ldclumping_r01d1000_geu.html)


### PEER 

- [Whole Blood](https://xsun1229.github.io/factor_analysis/catalog_gwas_peer_sep_ldclumping_r01d1000.html)

### FLASH

- [Whole Blood](https://xsun1229.github.io/factor_analysis/catalog_gwas_flash_sep_ldclumping_r01d1000.html)


### PCs - KEGG

[Whole blood](https://xsun1229.github.io/factor_analysis/kegg_pcs_factor_traits_regress_5_d1k.html)

[Liver](https://xsun1229.github.io/factor_analysis/kegg_pcs_factor_traits_regress_5_d500d_liver.html)

### PEER - KEGG

[Whole Blood](https://xsun1229.github.io/factor_analysis/kegg_peer_factor_traits.html)

### PCs - REACTOME

[Whole Blood](https://xsun1229.github.io/factor_analysis/reactome_pcs_factor_traits_regress_5_d1k.html)

[Liver](https://xsun1229.github.io/factor_analysis/reactome_pcs_factor_traits_regress_5_d1k_liver.html)


We compared the enrichment results under different settings(PCs from KEGG and REACTOME, PLIER - Canonical).

[Comparison](https://xsun1229.github.io/factor_analysis/factor_traits_setting_compare.html)


# **5. CEDAR**

## **5.1 Data**

### 5.1.1 CEDAR data processing 

[Data overview](https://xsun1229.github.io/factor_analysis/CEDAR_data_process.html)

### 5.1.2 Expression factors

We tried 2 types of expression factors: 1)PLIER factors 2)PCs from KEGG pathways. 

**Expression factor numbers for 6 cell types**

|     Cell type            |     Number of LVs    |     Number of PCs    |
|--------------------------|----------------------|----------------------|
|     CD4+ T cells         |     146              |     1627             |
|     CD8+ T cells         |     134              |     1621             |
|     CD19+ B cells        |     136              |     1620             |
|     CD14+ monocytes      |     88               |     1628             |
|     CD15+ neutrophils    |     146              |     1614             |
|     platelets            |     132              |     1593             |


### 5.1.3 Association test covariates

When using PCs as exposure, we regressed out the 5 or N PEER factors computed from the GTEx expression data 

#### 5 PEER factors

The 5 PEER factors are the first 5 factors computed [here](https://xsun1229.github.io/factor_analysis/CEDAR_data_process.html#2_peer_factors) 

#### N PEER factors

One 'danger' of adjusting for many PEER factors is that the factors may be affected by the genotypes. In other words, we could have this model: genetic variant -> PEER factor -> pathway. We should not adjust for the factor, as it will explain away the variant > pathway association.

So, when testing variant ~ pathway association, we did a association test for variant and a PEER factor, if its p-value is small (< 0.05), we did not adjust that factor. 
For example, we are testing association of a KEGG pathway with a SNP, and we have 20 PEER factor.  We first test association of SNP with each of the 20 PEER factors: if p < 0.05, then we do NOT regress out that factor. 

We tried first 20 and 10 candidate PEER factors as covariates to be regressed out.

## **5.2 Results under different settings**

The 10 genotype pcs using as covariates in factor ~ SNP association tests are computed from pruned genotype data set with bigsnpr. 

### PLIER - MSigDB factors 

[Workflow validation](https://xsun1229.github.io/factor_analysis/cedar_plier_validation.html)

[Details](https://xsun1229.github.io/factor_analysis/cedar_plier_26_qc_d1k_pruned.html)

### PC - KEGG 

**regress out 5 PEER factors**

In this part, when doing association test for PCs ~ variants, we added first 5 PEER factors as covariates.

[Details - all pathways included](https://xsun1229.github.io/factor_analysis/cedar_kegg_26_qc_d1k_pruned.html)

[Details - only pathways have > 40 genes included](https://xsun1229.github.io/factor_analysis/cedar_kegg_26_qc_d1k_40genes_pruned.html)

**regress out N PEER factors, N <= 20**

In this part, when doing association test for PCs ~ variants, we added first N PEER factors as covariates(N <= 20). The details are described in 5.1.3.

[Details - all pathways](https://xsun1229.github.io/factor_analysis/cedar_kegg_26_qc_d1k_pruned_Npeer.html)

[Details - pathways> 20 genes ](https://xsun1229.github.io/factor_analysis/cedar_kegg_26_qc_d1k_pruned_Npeer_20genes.html)

[Details - pathways> 40 genes ](https://xsun1229.github.io/factor_analysis/cedar_kegg_26_qc_d1k_pruned_Npeer_40genes.html)


**regress out N PEER factors, N <= 10**

In this part, when doing association test for PCs ~ variants, we added first N PEER factors as covariates(N <= 10). The details are described in 5.1.3.

[Details - all pathways](https://xsun1229.github.io/factor_analysis/cedar_kegg_26_qc_d1k_pruned_10peer.html)

[Details - pathways> 20 genes ](https://xsun1229.github.io/factor_analysis/cedar_kegg_26_qc_d1k_pruned_10peer_20genes.html)

[Details - pathways> 40 genes ](https://xsun1229.github.io/factor_analysis/cedar_kegg_26_qc_d1k_pruned_10peer_40genes.html)

### Earlier results 

The 10 genotype pcs using as covariates in factor ~ SNP association tests are computed from whole genome with PLINK 1.9. 

#### PLIER - MSigDB factors

[Details](https://xsun1229.github.io/factor_analysis/cedar_plier_26_qc_d1k.html)


#### PC - KEGG 

[Details - regress out 5 PEER factors](https://xsun1229.github.io/factor_analysis/cedar_kegg_26_qc_d1k.html)

[Details - regress out 5 PEER factors >40 genes](https://xsun1229.github.io/factor_analysis/cedar_kegg_26_qc_d1k_40genes.html)