---
title: "gwas_power"
author: "XSun"
date: "2020-07-13"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

## Introduction

In step 6, the power of GWAS is low. So I tried to find functional SNPs to increase the GWAS power and reduce the testing burden.


## Material and Methods. 

1. I used GTEx gene expression data and variant data. The dataset contains 19,696 genes. The variants here have maf > 0.01. The total number of variants is 10,403,249. The genotype of some variants are missing. I used mean-imputation to impute the missing data.


```{r eval=FALSE, echo=T}
indv <- nrow(snp_set)
    for (j in 1: ncol(snp_set) ) {
      missnum <- sum(is.na(snp_set[,j]))
      if (missnum != 0) {
        mean <- sum(snp_set[is.na((snp_set[,j]))==0,j])/ (indv - missnum)
        snp_set[is.na(snp_set[,j]),j] <- mean 
      }
    }
```


2. For each gene, I performed lasso regression for gene expression ~ cis-snps(within 500kb from TSS of gene). I got the weights of each cis-snp in this step. 17,175 of 19,696 genes have cis-snps. So I used these genes in the following procedure.  7895 of 17175 lasso regression selected at least one snp.

```{r eval=FALSE, echo=T}
fit_lasso <- cv.glmnet(x, y, alpha =1, foldid = foldid)
```



3. Using the weights in step2, I can got the cis genetic component part of gene expression. 

```{r eval=FALSE, echo=T}
y_p <- predict(fit_lasso,x,s=fit_lasso$lambda.min)
```

4. For each gene, I performed linear regression for factors ~ cis genetic component part of gene expression + 10 genotype pcs(covariants). The factors I used here are LV90,50,125,47,81,120 of PLIER-CanonicalPathway (the last three are negative control, whose heritability equal to 0.0001 and have very high pval_test_l) and pc5 of TID pathway. 

```{r eval=FALSE, echo=T}
fit_linear <- lm(factor ~ y_p + covariants[,3] + covariants[,4] + covariants[,5]+ covariants[,6] + covariants[,7] + covariants[,8] + covariants[,9] + covariants[,10] + covariants[,11] + covariants[,12])
pval[k] <- summary(fit_linear)$coefficients[2,4]   # the pval of kth gene
```

5. For each factor, I found the overlap between the significant genes in step4 (pval < 0.01) and genes belonging to the factor(the factor loading >0). The factors I used here are the same with step 4.

6. To eliminate weak instrument bias, I did test association only on "cis-heritable" genes instead of all 7895 genes. I estimate cis-heritability of genes(use the cis-snp within 500kb from TSS) with gcta, and choose the genes have LRT pvalue < 0.01 to do factors ~ cis genetic component part of gene expression + 10 genotype pcs(covariants)  test. 

7. The FUSION provided pre-trained YFS whole blood weights. It contains 4700 genes. I used these weights and GTEx genotype data to impute the cis-gene expression of these 4700 genes. Then, used these imputed expression to do the factors ~ cis genetic component part of gene expression + 10 genotype pcs(covariants) association test.


## Results- Original


### Summary

The suffix "_NC" means negative control. 

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/sum_cisgene_all.rdata")
DT::datatable(summary_all, options = list(pageLength =20))
```

The number of genes that have pvalue < 0.05 in association tests:

| Factor             | T1D-pc5    | LV90   | LV50  | LV125  | LV47-NC  | LV81-NC   | LV120-NC |
|--------------------|------|-----|-----|-----|-----|-----|-----|
| Number of genes with pval<0.05   | 474 | 1189 | 645 | 1008 | 1558 | 1788 | 1120 |


### Histogram of pvalues

- T1D pc5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/pv_h_t1dpc5.jpeg" width="40%">
</figure>

- low_pval_test_l lvs

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/pv_h_lv90.jpeg" width="40%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/pv_h_lv50.jpeg" width="40%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/pv_h_lv125.jpeg" width="40%">
</figure>

- negative controls

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/pv_h_lv47.jpeg" width="40%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/pv_h_lv81.jpeg" width="40%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/pv_h_lv120.jpeg" width="40%">
</figure>

### QQ plot against uniform distribution

- T1D pc5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qqplot_pval_t1dpc5.jpeg" width="40%">
</figure>

- low_pval_test_l lvs

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qqplot_pval_lv90.jpeg" width="40%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qqplot_pval_lv50.jpeg" width="40%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qqplot_pval_lv125.jpeg" width="40%">
</figure>

- negative controls

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qqplot_pval_lv47.jpeg" width="40%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qqplot_pval_lv81.jpeg" width="40%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qqplot_pval_lv120.jpeg" width="40%">
</figure>


### Find the overlap

#### T1D PC5

There is not any overlap in this factor. Although T1D pathway contains 174 genes(Ensemble ID), the GTEx expression data just cover 36 of them. None of these 36 genes has pvalue less than 0.01.


#### LV90

460 genes have pvalue < 0.01 in the association test with LV90. Among 5472 genes(PLIER used these genes to infer factors, these genes belong to 545 pathways), the factor loadings of 2544 genes are greater than 0. 55 genes are both significant and have non-zero factor loadings.

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/overlap_lv90.jpeg" width="50%">
</figure>

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/sum_cgl001_lv90.rdata")
DT::datatable(summary_loading_001, options = list(pageLength =10))
```

#### LV50

155 genes have pvalue < 0.01 in the association test with LV60. Among 5472 genes, the factor loadings of 2730 genes are greater than 0. 20 genes are both significant and have non-zero factor loadings.

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/overlap_lv50.jpeg" width="50%">
</figure>

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/sum_cgl001_lv50.rdata")
DT::datatable(summary_loading_001, options = list(pageLength =10))
```

#### LV125

351 genes have pvalue < 0.01 in the association test with LV125. Among 5472 genes, the factor loadings of 3014 genes are greater than 0. 62 genes are both significant and have non-zero factor loadings.

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/overlap_lv125.jpeg" width="50%">
</figure>

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/sum_cgl001_lv125.rdata")
DT::datatable(summary_loading_001, options = list(pageLength =10))
```


#### LV47-NC

702 genes have pvalue < 0.01 in the association test with LV47. Among 5472 genes, the factor loadings of 2576 genes are greater than 0. 100 genes are both significant and have non-zero factor loadings.

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/overlap_lv47.jpeg" width="50%">
</figure>


```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/sum_cgl001_lv47.rdata")
DT::datatable(summary_loading_001, options = list(pageLength =10))
```

#### LV81-NC

871 genes have pvalue < 0.01 in the association test with LV81. Among 5472 genes, the factor loadings of 3124 genes are greater than 0. 191 genes are both significant and have non-zero factor loadings.

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/overlap_lv81.jpeg" width="50%">
</figure>

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/sum_cgl001_lv81.rdata")
DT::datatable(summary_loading_001, options = list(pageLength =10))
```

#### LV120-NC

436 genes have pvalue < 0.01 in the association test with LV47. Among 5472 genes, the factor loadings of 2644 genes are greater than 0. 64 genes are both significant and have non-zero factor loadings.

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/overlap_lv120.jpeg" width="50%">
</figure>


```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/sum_cgl001_lv120.rdata")
DT::datatable(summary_loading_001, options = list(pageLength =10))
```


## Results- eliminate weak instrument bias-using GTEx weights(filted by h2 pval)

I got 1900 cis-heritable genes from 7895 genes(heritability LRT pvalue < 0.01). And used these genes to do factors ~ cis genetic component part of gene expression + 10 genotype pcs(covariants).


### Histogram of pvalues

- T1D pc5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/pv_h_t1dpc5_bias.jpeg" width="40%">
</figure>

- low_pval_test_l lvs

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/pv_h_lv90_bias.jpeg" width="40%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/pv_h_lv50_bias.jpeg" width="40%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/pv_h_lv125_bias.jpeg" width="40%">
</figure>

- negative controls

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/pv_h_lv47_bias.jpeg" width="40%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/pv_h_lv81_bias.jpeg" width="40%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/pv_h_lv120_bias.jpeg" width="40%">
</figure>


### QQ plot against uniform distribution

- T1D pc5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qqplot_pval_t1dpc5_bias.jpeg" width="40%">
</figure>

- low_pval_test_l lvs

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qqplot_pval_lv90_bias.jpeg" width="40%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qqplot_pval_lv50_bias.jpeg" width="40%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qqplot_pval_lv125_bias.jpeg" width="40%">
</figure>

- negative controls

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qqplot_pval_lv47_bias.jpeg" width="40%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qqplot_pval_lv81_bias.jpeg" width="40%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qqplot_pval_lv120_bias.jpeg" width="40%">
</figure>


### Find the overlap

#### T1D PC5

There is not any overlap in this factor. Although T1D pathway contains 174 genes(Ensemble ID), the GTEx expression data just cover 36 of them. None of these 36 genes has pvalue less than 0.01.


#### LV90

90 genes have pvalue < 0.01 in the association test with LV90. Among 5472 genes(PLIER used these genes to infer factors, these genes belong to 545 pathways), the factor loadings of 2544 genes are greater than 0. 11 genes are both significant and have non-zero factor loadings.

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/overlap_lv90_bias.jpeg" width="50%">
</figure>

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/sum_cgl001_lv90_bias.rdata")
DT::datatable(summary_loading_001, options = list(pageLength =10))
```

#### LV50

34 genes have pvalue < 0.01 in the association test with LV60. Among 5472 genes, the factor loadings of 2730 genes are greater than 0. 6 genes are both significant and have non-zero factor loadings.

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/overlap_lv50_bias.jpeg" width="50%">
</figure>

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/sum_cgl001_lv50_bias.rdata")
DT::datatable(summary_loading_001, options = list(pageLength =10))
```

#### LV125

123 genes have pvalue < 0.01 in the association test with LV125. Among 5472 genes, the factor loadings of 3014 genes are greater than 0. 15 genes are both significant and have non-zero factor loadings.

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/overlap_lv125_bias.jpeg" width="50%">
</figure>

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/sum_cgl001_lv125_bias.rdata")
DT::datatable(summary_loading_001, options = list(pageLength =10))
```


#### LV47-NC

114 genes have pvalue < 0.01 in the association test with LV47. Among 5472 genes, the factor loadings of 2576 genes are greater than 0. 15 genes are both significant and have non-zero factor loadings.

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/overlap_lv47_bias.jpeg" width="50%">
</figure>


```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/sum_cgl001_lv47_bias.rdata")
DT::datatable(summary_loading_001, options = list(pageLength =10))
```

#### LV81-NC

166 genes have pvalue < 0.01 in the association test with LV81. Among 5472 genes, the factor loadings of 3124 genes are greater than 0. 25 genes are both significant and have non-zero factor loadings.

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/overlap_lv81_bias.jpeg" width="50%">
</figure>

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/sum_cgl001_lv81_bias.rdata")
DT::datatable(summary_loading_001, options = list(pageLength =10))
```

#### LV120-NC

78 genes have pvalue < 0.01 in the association test with LV47. Among 5472 genes, the factor loadings of 2644 genes are greater than 0. 10 genes are both significant and have non-zero factor loadings.

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/overlap_lv120_bias.jpeg" width="50%">
</figure>


```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/sum_cgl001_lv120_bias.rdata")
DT::datatable(summary_loading_001, options = list(pageLength =10))
```


## Results- eliminate weak instrument bias-using YFS weights

The FUSION provided pre-trained YFS whole blood weights. It contains 4700 genes. 4212 of them can be used to impute the cis-genetic part of gene expression. I used these weights and GTEx genotype data to impute the cis-gene expression of these 4212 genes. Then, used these imputed expression to do the factors ~ cis genetic component part of gene expression + 10 genotype pcs(covariants) association test.


### Histogram of pvalues

- T1D pc5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/pv_h_t1dpc5_YFS.jpeg" width="40%">
</figure>

- low_pval_test_l lvs

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/pv_h_lv90_YFS.jpeg" width="40%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/pv_h_lv50_YFS.jpeg" width="40%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/pv_h_lv125_YFS.jpeg" width="40%">
</figure>

- negative controls

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/pv_h_lv47_YFS.jpeg" width="40%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/pv_h_lv81_YFS.jpeg" width="40%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/pv_h_lv120_YFS.jpeg" width="40%">
</figure>


### QQ plot against uniform distribution

- T1D pc5

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qqplot_pval_t1dpc5_YFS.jpeg" width="40%">
</figure>

- low_pval_test_l lvs

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qqplot_pval_lv90_YFS.jpeg" width="40%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qqplot_pval_lv50_YFS.jpeg" width="40%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qqplot_pval_lv125_YFS.jpeg" width="40%">
</figure>

- negative controls

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qqplot_pval_lv47_YFS.jpeg" width="40%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qqplot_pval_lv81_YFS.jpeg" width="40%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qqplot_pval_lv120_YFS.jpeg" width="40%">
</figure>

### Find the overlap

#### T1D PC5

T1D pathway contains 174 genes(Ensemble ID), the GTEx expression data just cover 36 of them. 8 of these 36 genes has pvalue less than 0.01.   

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/overlap_pc5_YFS.jpeg" width="50%">
</figure>

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/sum_cgl_t1dpc5.rdata")
DT::datatable(summary_loading_001, options = list(pageLength =10))
```

#### LV90

62 genes have pvalue < 0.01 in the association test with LV90. Among 5472 genes(PLIER used these genes to infer factors, these genes belong to 545 pathways), the factor loadings of 2544 genes are greater than 0. 15 genes are both significant and have non-zero factor loadings.

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/overlap_lv90_YFS.jpeg" width="50%">
</figure>

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/sum_cgl001_lv90_YFS.rdata")
DT::datatable(summary_loading_001, options = list(pageLength =10))
```

#### LV50

45 genes have pvalue < 0.01 in the association test with LV60. Among 5472 genes, the factor loadings of 2730 genes are greater than 0. 16 genes are both significant and have non-zero factor loadings.

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/overlap_lv50_YFS.jpeg" width="50%">
</figure>

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/sum_cgl001_lv50_YFS.rdata")
DT::datatable(summary_loading_001, options = list(pageLength =10))
```

#### LV125

53 genes have pvalue < 0.01 in the association test with LV125. Among 5472 genes, the factor loadings of 3014 genes are greater than 0. 17 genes are both significant and have non-zero factor loadings.

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/overlap_lv125_YFS.jpeg" width="50%">
</figure>

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/sum_cgl001_lv125_YFS.rdata")
DT::datatable(summary_loading_001, options = list(pageLength =10))
```


#### LV47-NC

30 genes have pvalue < 0.01 in the association test with LV47. Among 5472 genes, the factor loadings of 2576 genes are greater than 0. 7 genes are both significant and have non-zero factor loadings.

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/overlap_lv47_YFS.jpeg" width="50%">
</figure>


```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/sum_cgl001_lv47_YFS.rdata")
DT::datatable(summary_loading_001, options = list(pageLength =10))
```

#### LV81-NC

59 genes have pvalue < 0.01 in the association test with LV81. Among 5472 genes, the factor loadings of 3124 genes are greater than 0. 25 genes are both significant and have non-zero factor loadings.

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/overlap_lv81_YFS.jpeg" width="50%">
</figure>

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/sum_cgl001_lv81_YFS.rdata")
DT::datatable(summary_loading_001, options = list(pageLength =10))
```

#### LV120-NC

50 genes have pvalue < 0.01 in the association test with LV47. Among 5472 genes, the factor loadings of 2644 genes are greater than 0. 11 genes are both significant and have non-zero factor loadings.

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/overlap_lv120_YFS.jpeg" width="50%">
</figure>


```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/sum_cgl001_lv120_YFS.rdata")
DT::datatable(summary_loading_001, options = list(pageLength =10))
```


