---
title: "gwas_power"
author: "XSun"
date: "2020-07-13"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

## Introduction

In step 6, the power of GWAS is low. So I tried to find functional SNPs to increase the GWAS power and reduce the testing burden.


## Material and Methods. 

### Original analysis
1. I used GTEx gene expression data and variant data. The dataset contains 19,696 genes. The variants here have maf > 0.01. The total number of variants is 10,403,249. The genotype of some variants are missing. I used mean-imputation to impute the missing data.


```{r eval=FALSE, echo=T}
indv <- nrow(snp_set)
    for (j in 1: ncol(snp_set) ) {
      missnum <- sum(is.na(snp_set[,j]))
      if (missnum != 0) {
        mean <- sum(snp_set[is.na((snp_set[,j]))==0,j])/ (indv - missnum)
        snp_set[is.na(snp_set[,j]),j] <- mean 
      }
    }
```


2. For each gene, I performed lasso regression for gene expression ~ cis-snps(within 500kb from TSS of gene). I got the weights of each cis-snp in this step. 17,175 of 19,696 genes have cis-snps. So I used these genes in the following procedure.  7895 of 17175 lasso regression selected at least one snp.

```{r eval=FALSE, echo=T}
fit_lasso <- cv.glmnet(x, y, alpha =1, foldid = foldid)
```


3. Using the weights in step2, I can got the cis genetic component part of gene expression. 

```{r eval=FALSE, echo=T}
y_p <- predict(fit_lasso,x,s=fit_lasso$lambda.min)
```

4. For each gene, I performed linear regression for factors ~ cis genetic component part of gene expression + 10 genotype pcs(covariants). The factors I used here are LV90,50,125,47,81,120 of PLIER-CanonicalPathway (the last three are negative control, whose heritability equal to 0.0001 and have very high pval_test_l) and pc5 of TID pathway. 

```{r eval=FALSE, echo=T}
fit_linear <- lm(factor ~ y_p + covariants[,3] + covariants[,4] + covariants[,5]+ covariants[,6] + covariants[,7] + covariants[,8] + covariants[,9] + covariants[,10] + covariants[,11] + covariants[,12])
pval[k] <- summary(fit_linear)$coefficients[2,4]   # the pval of kth gene
```

### ELiminate weak instrument bias-using GTEx weights(filted by h2 pval)

5. To eliminate weak instrument bias, I did test association only on "cis-heritable" genes instead of all 7895 genes. I estimate cis-heritability of genes(use the cis-snp within 500kb from TSS) with gcta, and choose the genes have LRT pvalue < 0.01 to do factors ~ cis genetic component part of gene expression + 10 genotype pcs(covariants)  test. 


### ELiminate weak instrument bias-using YFS weights

6. The FUSION provided pre-trained YFS whole blood weights. It contains 4700 genes. I used these weights and GTEx genotype data to impute the cis-gene expression of these 4700 genes. Then, used these imputed expression to do the factors ~ cis genetic component part of gene expression + 10 genotype pcs(covariants) association test.


## Results- Original

The table below shows the number of significant associations at various p-value threshold for each LV

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/sum_pval_lvpc.rdata")
DT::datatable(summary_loading_001, options = list(pageLength =10))
```



## Results- eliminate weak instrument bias-using GTEx weights(filted by h2 pval)

The table below shows the number of significant associations at various p-value threshold for each LV

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/sum_pval_lvpc_bias.rdata")
DT::datatable(summary_loading_001, options = list(pageLength =10))
```

## Results- eliminate weak instrument bias-using YFS weights

The table below shows the number of significant associations at various p-value threshold for each LV

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/sum_pval_lvpc_YFS.rdata")
DT::datatable(summary_all, options = list(pageLength =10))
```


