---
title: "Studying effects of PCs on phenotypes (KEGG) – After regressing out 5 GTEx PEER factors, 'AND' LD Clumping "
author: "XSun"
date: "2021-04-02"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---


## **Introduction**

In this part, we inferred 1646 PCs from 332 KEGG pathways and studied the effects of these PCs on phenotype. We regressed out top 5 PEER factors from the GTEx data set. 

## **Candidate SNPs**

The candidate SNPs in this part are trait-related SNPs. We first filtered the SNPs using pval < 5e-8 as cut off. Then we did LD Clumping for those SNPs.The PLINK LD Clumping patameters are:

– clump-p1 0.0001 Significance threshold for index SNPs

– clump-p2 0.01 Secondary significance threshold for clumped SNPs

– clump-r2 0.1 LD threshold for clumping

– clump-kb 500 Physical distance threshold for clumping

After LD Clumping, we removed SNPs that are too close (within 500kb).

## **Effectsize analysis -- effect size fitting**

For the pairs have more than 2 SNPs with FDR <0.2 (920 pairs), we also tested the effectsize correlation(effect size in GWAS and effect size in factos-association). We fitted the effectsize using: effectsize in GWAS ~ 0 +effectsize in factor-association. We recorded the p-values and r-squareds in the fitting.

To make our results more reliable, we did resampling. We resampled the SNPs without replacement and fitted the points with intercept = 0 again and recorded the r-squareds to compute the p-values in resampling. The 'p-value from resampling' is computed by: (number of more extreme values)/(times of resampling).The resampling was repeated 1000 times.

There are 215 pairs show sign consistency (have p-values < 0.05 in resampling, 4.67 fold enrichment). There are 75 pairs at p-value < 0.01, 8.3 fold enrichment). 


```{r echo=FALSE}
sum <- read.table("/project2/xinhe/xsun/website/factor_analysis/output/simple_sum_pcs_reg_5_d500d.txt", header = T)
DT::datatable(sum, options = list(pageLength =5))
```

After deleting the pairs whose 'num_snp_noMHC' == 0, we got 824 pairs. We used extended MHC area here ([chr6:25726063-33400644](https://www.nature.com/articles/nrg1489)) There are 163 pairs show sign consistency (have p-values < 0.05 in resampling, 3.95 fold enrichment). There are 45 pairs at p-value < 0.01, 5.46 fold enrichment).  

```{r echo=FALSE}
sum <- read.table("/project2/xinhe/xsun/website/factor_analysis/output/simple_sum_noMHC_pcs_reg_5_extend_d500d.txt", header = T)
DT::datatable(sum, options = list(pageLength =5))
```


To summarize the results, We made a scatter plot of all trait-LV pairs. The x-axis shows the R2 in fitting, and the y-axis show the -log10-p values from resampling. To make the plot more informative, Each point represent a trait-LV pair. Different colors show the number of SNPs passing the FDR threshold for each pair.


<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsizesum_0.2_kegg_pc_reg_5_d500d.jpeg" width="80%">
</figure>


## **Detailed results for 'eo_pwy211_pc3' **

The related pathway is : Regulation of actin cytoskeleton [ several reports have examined the role of the actin cytoskeleton in activation of human eosinophils](https://www.jacionline.org/article/S0091-6749(03)01249-1/fulltext)

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/info_table_annot_pval_kegg_pc_reg_5_d500d_eo.rdata")

DT::datatable(final_full, options = list(pageLength =5))
```

<<<<<<< HEAD
<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_pwy211_pc3_eo_pval5e8_d500d.jpeg" width="80%">
</figure>
=======
![](https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_pwy211_pc3_eo_pval5e8_d500d.jpeg)
>>>>>>> 8d321c96ac8958e047f2ca1ac2f5d72d7f7823ff


## **Detailed results for 'TG_pwy211_pc2' **

The related pathway is : Primary bile acid biosynthesis 

It seems that this pathway is not biological plausible.

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/info_table_annot_pval_kegg_pc_reg_5_d500d_TG.rdata")

DT::datatable(final_full, options = list(pageLength =5))
```

<<<<<<< HEAD
<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_pwy13_pc2_TG_pval5e8_d500d.jpeg" width="80%">
</figure>
=======
![](https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_pwy13_pc2_TG_pval5e8_d500d.jpeg)
>>>>>>> 8d321c96ac8958e047f2ca1ac2f5d72d7f7823ff


## **Detailed results for granulocyte count **

The table below shows the SNP details. We deleted the pairs that only have SNPs in MHC area.

We found 4 SNPs drive different pathways:

```{r echo=T, eval=FALSE}
chr3_128603030_AG_A_b38 : row 4,29
chr8_8237348_G_A_b38 : row 2,19,22,26
chr8_9061626_G_A_b38 : row 3,23
chr11_108378047_C_G_b38 : row 1,18
```

```{r echo=FALSE}
sum <- read.table("/project2/xinhe/xsun/website/factor_analysis/output/gran_kegg_pc_reg_5_d500d.txt", header = T, fill = T, sep = "\t")
DT::datatable(sum, options = list(pageLength =5))
```


There are 9 pathways in this table: 

[1] "Circadian rhythm"  [Circadian rhythms are present in many cellular and humoral components of the immune system. Granulocytes and monocytes exhibit circadian oscillations in their numbers in blood, both in humans and mice ](https://www.frontiersin.org/articles/10.3389/fimmu.2020.00576/full)                                     

[2] "Ether lipid metabolism"                                

[3] "Fatty acid biosynthesis"                               

[4] "Glycosylphosphatidylinositol (GPI)-anchor biosynthesis"

[5] "Lysine degradation"                                    

[6] "Pyruvate metabolism"                                   

[7] "Regulation of actin cytoskeleton"                      

[8] "Ribosome"                                              

[9] "Steroid hormone biosynthesis"   [may be related through some other mediator: These findings implicate GM-CSF as a local mediator of steroid-driven remodeling events in the cycling and preimplantation endometrium, possibly acting through the recruitment and behavioral regulation of granulocytes and macrophages. ](https://pubmed.ncbi.nlm.nih.gov/8838016/)   


```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/info_table_annot_pval_kegg_pc_reg_5_d500d_gran.rdata")

DT::datatable(final_full, options = list(pageLength =5))
```


## **Detailed results for ukb.allasthma **

The table below shows the SNP details. We deleted the pairs that only have SNPs in MHC area.

We found 3 SNPs drive different pathways:

```{r echo=T, eval=FALSE}
chr1_152313385_G_A_b38 : row 2,7,9,13,15,17
chr5_119355086_C_G_b38 : row 6,8,12,16
chr7_20547220_A_G_b38 : row 5,14,18
```

```{r echo=FALSE}
sum <- read.table("/project2/xinhe/xsun/website/factor_analysis/output/ukb.allasthma_kegg_pc_reg_5_d500d.txt", header = T, fill = T, sep = "\t")
DT::datatable(sum, options = list(pageLength =5))
```

There are 9 pathways in this table: 


[1] "Adrenergic signaling in cardiomyocytes"        

[2] "Arrhythmogenic right ventricular cardiomyopathy"   asthma related to Right ventricular diastolic dysfunction but not exactly ARVC : [Asthmatic patients develop right heart affection from early stages through different mechanisms. The current study was performed aiming to assess the right ventricular diastolic dysfunction in different asthmatic stages.](https://www.sciencedirect.com/science/article/pii/S0422763813002094)
 
[3] "Calcium signaling pathway"   [Ca2+ signaling pathways part in paper : 'Targeting cell signaling in allergic asthma'](https://www.nature.com/articles/s41392-019-0079-0)                  
[4] "Cardiac muscle contraction"  [no paper shows the direct relationship, but I'm considering the Ca2+ may acts as a cofounder in it.]

[5] "Dilated cardiomyopathy"   [A statistically significant, independent association was observed between idiopathic DCM and a history of asthma ](https://pubmed.ncbi.nlm.nih.gov/2801484/)        

[6] "Fatty acid biosynthesis"      [fatty acid metabolism](https://onlinelibrary.wiley.com/doi/abs/10.1111/all.13180)

[7] "Hypertrophic cardiomyopathy"    [not related: Mistakes and delays in diagnosisabound. Hypertrophic cardiomyopathy commonly“masquerades”as asthma](https://www.amjmed.com/article/S0002-9343(15)00771-8/pdf)      

[8] "Necroptosis"   [Finally, the human and murine necroptosis inhibitor, GW806742X, blocked necroptosis and IL-33 release in vitro and reduced eosinophilia in Aspergillus fumigatus extract-induced asthma in vivo, an allergic inflammation model that is highly dependent on IL-33](https://pubmed.ncbi.nlm.nih.gov/30576068/)  

[9] "Tight junction" 


```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/info_table_annot_pval_kegg_pc_reg_5_d500d_ukb.allasthma.rdata")

DT::datatable(final_full, options = list(pageLength =5))
```

## **Detailed results for EUR.IBD **

The table below shows the SNP details. We deleted the pairs that only have SNPs in MHC area.

We found 8 SNPs drive different pathways:

```{r echo=T, eval=FALSE}
chr2_43226582_G_A_b38 : row 32,34
chr3_48717444_G_T_b38 : row 4,10,12,14,21,23,27,31,36,42,46,50,52
chr5_135109292_C_A_b38: row 33,35,47
chr5_131126137_G_T_b38: row 19,44
chr6_14721666_T_C_b38 : row 2,3,9,11,13,20,22,25,26,30,41,45,49,51,57
chr12_40430996_T_C_b38 : row 7,17,28
chr16_81883307_A_G_b38 : row 6,15,37,39
chr22_21561877_T_C_b38: row 8,38,40,54,56
```

```{r echo=FALSE}
sum <- read.table("/project2/xinhe/xsun/website/factor_analysis/output/EUR.IBD_kegg_pc_reg_5_d500d.txt", header = T, fill = T, sep = "\t")
DT::datatable(sum, options = list(pageLength =5))
```

There are 26 pathways in this table: 

 [1] "Alcoholism"    paper: [Alcohol consumption is a potential trigger for flare in Inflammatory Bowel Disease (IBD) flare ](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3708696/)                          
 
 [2] "Amyotrophic lateral sclerosis"           
 
 [3] "Axon guidance"                           
 
 [4] "Bacterial invasion of epithelial cells"  paper: [The epithelial layer is essential for the proper functioning of the gastrointestinal tract, and its increased permeability to the luminal antigens may lead to the inflammatory processes and mucosal damage observed in IBD](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2604805/) 
 
 [5] "Basal transcription factors"   paper: [Genome-wide association studies (GWAS) have detected single-nucleotide polymorphisms(SNPs) in genes encoding cytokines or transcription factors that control cytokine-mediated activationof immune cells associated with the pathogenesis of IBD](https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=&cad=rja&uact=8&ved=2ahUKEwiTptulluTvAhWBsJ4KHc41C9IQFjABegQIAhAD&url=https%3A%2F%2Fwww.mdpi.com%2F1422-0067%2F21%2F9%2F3207%2Fpdf&usg=AOvVaw0WoX2S3bSMdIymOtAy6p1E)          
 
 [6] "beta-Alanine metabolism" paper : [the most impacted by IBD pathways are: glycine, serine and threonine metabolism, taurine and hypotaurine metabolism, alanine, aspartate and glutamate metabolism, arginine and proline metabolism, glycerophospholipid metabolism, pyruvate metabolism, aminoacyl-tRNA biosynthesis, phenylalanine metabolism, sulphur metabolism, primary bile acid biosynthesis, and beta-alanine metabolism](https://www.sciencedirect.com/science/article/pii/S1896112619300264)                
 
 [7] "Cardiac muscle contraction"  [IBD may relates to Cardiovascular diseases](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6348818/)            
 
 [8] "Complement and coagulation cascades"   paper : [We review the evidence base for the role and potential therapeutic manipulation of the complement cascade in inflammatory bowel disease](https://pubmed.ncbi.nlm.nih.gov/24831561/)  
 
 [9] "Dilated cardiomyopathy"   paper :same with [7]               

[10] "Fatty acid degradation"  paper: [1](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4100336/) [2](https://www.sciencedirect.com/science/article/pii/S0925443905001870)            

[11] "Human immunodeficiency virus 1 infection"  paper: [Human immunodeficiency virus infection, the acquired immunodeficiency syndrome, and inflammatory bowel disease ](https://pubmed.ncbi.nlm.nih.gov/8835895/)

[12] "Hypertrophic cardiomyopathy"     paper :same with [7]         

[13] "Malaria"                                 

[14] "Mitophagy - animal"  paper : [Inflammatory Bowel Disease at the Intersection of Autophagy and Immunity: Insights from Human Genetics](https://link.springer.com/chapter/10.1007/978-3-030-28703-0_14)           

[15] "Non-small cell lung cancer"              

[16] "Other glycan degradation"                

[17] "Pancreatic cancer"   paper : [People with IBD had a 3.36-fold higher risk of pancreatic cancer](https://www.webmd.com/ibd-crohns-disease/news/20100505/ibd-may-raise-risk-of-pancreatic-cancer)                    

[18] "Parkinson disease"  paper : [Inflammatory bowel disease and Parkinson’s disease: common pathophysiological links](https://gut.bmj.com/content/70/2/408)     

[19] "Pertussis"   paper : [Many Inflammatory Bowel Disease Patients Are Not Immune to Measles or Pertussis](https://pubmed.ncbi.nlm.nih.gov/27557706/)                 

[20] "Phosphonate and phosphinate metabolism"  

[21] "Porphyrin and chlorophyll metabolism"    

[22] "Renal cell carcinoma"                    

[23] "Renin-angiotensin system"   paper : [Upregulation of circulating  components of the alternative  renin-angiotensin system in inflammatory bowel disease: A pilot study](https://journals.sagepub.com/doi/pdf/10.1177/1470320314521086)             

[24] "Ribosome"                                

[25] "RNA transport"                           

[26] "Viral carcinogenesis"  



```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/info_table_annot_pval_kegg_pc_reg_5_d500d_EUR.IBD.rdata")

DT::datatable(final_full, options = list(pageLength =5))
```