---
title: "Studying effects of PCs on phenotypes (KEGG) -- After regressing out 5 GTEx PEER factors"
author: "XSun"
date: "2021-03-17"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

## **Introduction**

The factors in this part are the [PCs](https://xsun1229.github.io/factor_analysis/kegg_pcs_factor_traits.html) after regressing out 5 GTEx PEER factors. There're 1646 factors in total.


## **Association tests**

We did association tests for PCs and SNPs have p-value < 5E-8 in each trait-GWAS. The association tests were corrected by 10 whole genome PCs + GTEx:Sequencing platform,Sequencing protocol,Sex + AGE

We used 'qvalue' R package to compute the fdr from p-values for each SNP.


## **Effectsize analysis -- effect size fitting**

For the pairs have more than 2 SNPs with FDR <0.2 (1092 pairs), we also tested the effectsize correlation(effect size in GWAS and effect size in factos-association). We fitted the effectsize using: effectsize in GWAS ~ 0 +effectsize in factor-association. We recorded the p-values and r-squareds in the fitting.

To make our results more reliable, we did resampling. We resampled the SNPs without replacement and fitted the points with intercept = 0 again and recorded the r-squareds to compute the p-values in resampling. The 'p-value from resampling' is computed by: (number of more extreme values)/(times of resampling).The resampling was repeated 1000 times.

There are 228 pairs show sign consistency (have p-values < 0.05 in resampling). There are 82 pairs at p-value < 0.01). 

```{r echo=FALSE}
sum <- read.table("/project2/xinhe/xsun/website/factor_analysis/output/simple_sum_pcs_reg_5.txt", header = T)
DT::datatable(sum, options = list(pageLength =20))
```

After deleting the pairs whose 'num_snp_noMHC' == 0, we got 768 pairs. There are 137 pairs show sign consistency (have p-values < 0.05 in resampling). There are 37 pairs at p-value < 0.01).  

```{r echo=FALSE}
sum <- read.table("/project2/xinhe/xsun/website/factor_analysis/output/simple_sum_noMHC_pcs_reg_5.txt", header = T)
sum <- sum[complete.cases(sum$pval_in_resampling),]
DT::datatable(sum, options = list(pageLength =20))
```


To summarize the results, We made a scatter plot of all trait-LV pairs. The x-axis shows the R2 in fitting, and the y-axis show the -log10-p values from resampling. To make the plot more informative, Each point represent a trait-LV pair. Different colors show the number of SNPs passing the FDR threshold for each pair.


<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsizesum_0.2_kegg_pc_reg_5.jpeg" width="60%">
</figure>

## **Detailed results for White Blood Cell Count **

The WBC trait has significant effect size correlations in most pathways (p-value < 0.05 in resampling). For this trait, we studied the details of the driving SNPs(SNPs have FDR < 0.2 in association tests). 

The table below shows the SNP details. The snp in 'gtexid' is in LD with the snp in ‘snp_ld’. ’ld_r2’ is the LD r-squared which is corresponding to the ‘snp_ld’ column. ‘cis-eqtl’ column indicates whether the snp is a cis-eqtl according to GTEx data. ‘cis_gene_hgnc’ and ‘cis_gene_ensembl’ is the genes that the snp influence when it act as cis-eqtl. ‘func’ and ‘func_gene’ are obtained from ANNOVAR, which indicating the snp function within the genes. 


```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/info_table_annot_pval_MHC_kegg_pc_reg_5_wbc.rdata")
DT::datatable(final_full, options = list(pageLength =5))
```

There are too many SNPs in MHC area in the table above. We deleted the pairs that only have SNPs in MHC area, the MHC-excluded table is below. The MHC area is chr6: 28,477,797-33,448,354. (The last SNP in this table is in MHC area. This SNP is included, because this pair does not only have SNPs in MHC area)

We found 5 SNPs drive different pathways:

```{r echo=T, eval=FALSE}
chr10_102727686_G_A_b38 : row 11 and row 18
chr3_141286732_A_C_b38 : row 23 ,row 25 and row 27
chr19_1026478_G_T_b38 : row 24 and row 26
chr6_26463347_G_T_b38: row 1 and row 28
chr6_27873148_T_C_b38: row 6 and row 29
```

There are 11 pathways in this table: 

```{r echo=T, eval=FALSE}
 [1] "Circadian rhythm"                                   
 [2] "Drug metabolism - cytochrome P450"                  
 [3] "Glyoxylate and dicarboxylate metabolism"            
 [4] "Malaria"                                            
 [5] "Metabolism of xenobiotics by cytochrome P450"       
 [6] "Mineral absorption"                                 
 [7] "Pantothenate and CoA biosynthesis"                  
 [8] "Phenylalanine, tyrosine and tryptophan biosynthesis"
 [9] "Phosphonate and phosphinate metabolism"             
[10] "RNA transport"                                      
[11] "Terpenoid backbone biosynthesis"    
```

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/info_table_annot_pval_kegg_pc_reg_5_wbc.rdata")
DT::datatable(final_full, options = list(pageLength =5))
```


### Effectsize plots

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_pwy74_pc4_wbc_pval5e8_kegg_pc_reg_5.jpeg" width="60%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/effectsize_pwy194_pc1_wbc_pval5e8_kegg_pc_reg_5.jpeg" width="60%">
</figure>

## **Detailed results for ukb.allasthma **


The table below shows the SNP details. We deleted the pairs that only have SNPs in MHC area.

We found 3 SNPs drive different pathways:

```{r echo=T, eval=FALSE}
chr10_102727686_G_A_b38 : row 5, row 7
chr1_152313385_G_A_b38 : row 2, row 10
chr7_20547220_A_G_b38 : row 11, row 9
```

There are 6 pathways in this table: 

```{r echo=T, eval=FALSE}
[1] "Arrhythmogenic right ventricular cardiomyopathy"
[2] "Bile secretion"                                 
[3] "Calcium signaling pathway"                      
[4] "Fatty acid biosynthesis"                        
[5] "Malaria"                                        
[6] "Proximal tubule bicarbonate reclamation"    
```

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/info_table_annot_pval_kegg_pc_reg_5_ukb.allasthma.rdata")
DT::datatable(final_full, options = list(pageLength =5))
```


## **Detailed results for Type I Diabetes **

The table below shows the SNP details. We deleted the pairs that only have SNPs in MHC area.

We found 7 SNPs drive different pathways:

```{r echo=T, eval=FALSE}
chr1_113611678_T_C_b38 : row 2, row 6, row 9, row 23, row 31,
chr1_113276758_C_T_b38 : row 22, row 25, row 30
chr1_113623939_A_G_b38: row 24, row 32
chr10_6086584_G_T_b38 : row 11, row 14
chr18_12809341_A_G_b38: row 16, row 27
chr19_48702851_C_G_b38: row 19, row 29
chr22_30135102_C_T_b38: row 18, row 28
```

There are 10 pathways in this table: 

```{r echo=T, eval=FALSE}
 [1] "Chemical carcinogenesis"                   
 [2] "Cholesterol metabolism"                    
 [3] "ECM-receptor interaction"                  
 [4] "Hippo signaling pathway - multiple species"
 [5] "Inositol phosphate metabolism"             
 [6] "Protein export"                            
 [7] "Sulfur relay system"                       
 [8] "T cell receptor signaling pathway"         
 [9] "Transcriptional misregulation in cancer"   
[10] "Tyrosine metabolism"    
```

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis/output/info_table_annot_pval_kegg_pc_reg_5_T1D.rdata")
DT::datatable(final_full, options = list(pageLength =5))
```


