---
title: "modu2_factor_pliercanon"
author: "XSun"
date: "2020-07-10"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

## Introduction

I used 'PLIER' package(with default settings) to construct factors related to Canonical Pathways from module 3.

## Factor Construction

    - The gene expression data from GTEx uses Ensemnl ID. In order to use PLIER R package, we need to convert Ensemnl ID to HGNC Gene Symbol. There are 19,696 genes in GTEx data. After conversion, 16,634 left(not all Ensemnl IDs have a corresponding HGNC Gene Symbol). 177 of 16,634 were used in this part.
    - I used CanonicalPathways in PLIER R package, which contains 545 curated pathways from Reactome and KEGG.
    - I obtained 118 LVs from PLIER, 16 of them have AUC > 0.70.

## Heritability of factors


<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/hist_H_lv_pliercanon_mod3.jpeg" width="50%">
</figure>

Details about the factors are listed below. The last column are the top 3 pathways related to each factor.

```{r echo=FALSE}
aa <- read.table("/project2/xinhe/xsun/website/factor_analysis/output/H_lv_processed_sorted_pliercanon_mod3.txt", header = T)
DT::datatable(aa, options = list(pageLength =10))

```

## Prediction model

In this part, I adopted lasso/elastic_net regression to train prediction models. The dependent variables here were LVs and the independent variables were genotype of certain variants. (LVs ~ variant genotype)
  
There are 431 genes in this module. PLIER used 177 of them to infer factors. I got 114 factors for this module.
  
  
### Results-maf001

### Heritability vs R-squared

This figure shows the prediction performance (R2 of Lasso/Elastic_net regression on training set in red) in comparison to LV heritability estimates (black; standard errors in gray). Performance was assessed using tenfold cross-validation in the GTEx whole-blood cohort (on training set, n = 536) with the elastic net and lasso models.
  
<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/devide_snp001_le_mod3_pliercanon.jpeg" width="150%">
</figure>


### Summary of Heritability 

The table below shows the heritability of LVs and performance of the training models. The table has been sorted by Heritability from small to large.

Some notations to explain: **rsq_min** is the r-squared when lambda=lambda.min, **ncoef_nzero_min** is the number of non-zero coefficients when lambda=lambda.min, **rsq_test** is the r-squared in testing set (obtained from lm(lv.obs ~ lv.pred)) ,**pval_test** is the pvalue of F-statistics in testing set. When the suffix is '_1se', the notations mean lambda=lambda.1se. Suffix '_l' means lasso regression and '_e' means elastic_net


```{r eval=FALSE, echo=T}
  ###Lasso regression. 'k' represents the kth LV.
    fit_lasso <- cv.glmnet(x, y, alpha =1, foldid = foldid)   ##'foldid = foldid' can fix the data partition in cross validation, making the codes reproducible
    fit_r_train_l[k]  <- fit_lasso$glmnet.fit$dev.ratio[which(fit_lasso$glmnet.fit$lambda == fit_lasso$lambda.min)]  ###r2 when lambda = lambda.min
    predict_l <- predict(fit_lasso, nx, s = fit_lasso$lambda.min )
    
    fit_linear <- lm(y_test ~ predict_l)
    pval_test_l[i] <- summary(fit_linear)$coefficients[,4]
    rsq_test_l[i] <- summary(fit_linear)$r.squared

```


In this table, there are several NA value. These NA values come from linear regression(lm(y_test ~ predict_l)). It means the variant in regression is linearly with the intercept. After checking the predict_l, we found the predict_l for all samples are the same.

```{r echo=FALSE}
aa <- read.table("/project2/xinhe/xsun/website/factor_analysis/output/summary_H_le_sorted_mod3_pliercanon.txt",header = T)

DT::datatable(aa, options = list(pageLength =20))

```


### LVs with relatively low pval_test_l

By sorting the pval_test_l in the table above, I foung that only LV47,95,112 have relatively low pvalues on testing set. We made scatter plots of predicted LVs vs. observed LVs for them. Besides, we did GWAS analysis.

The red line in manhattan plot means p-value is 5e-8. 

#### lv47

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/LV47_mod3_pliercanon.jpeg" width="200%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qq_unif_lv47_mod3_pliercanon.jpeg" width="40%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/man_lv47_mod3_pliercanon.jpeg" width="40%">
</figure>

#### lv95

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/LV95_mod3_pliercanon.jpeg" width="200%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qq_unif_lv95_mod3_pliercanon.jpeg" width="40%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/man_lv95_mod3_pliercanon.jpeg" width="40%">
</figure>

#### lv112

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/LV112_mod3_pliercanon.jpeg" width="200%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/qq_unif_lv112_mod3_pliercanon.jpeg" width="40%">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/man_lv112wflow__mod3_pliercanon.jpeg" width="40%">
</figure>

