---
title: "Analyses using PCs where PEER factors were regressed out"
author: "XSun"
date: "2022-09-30"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

# Introduction

We re-do our analyses with new PCs where the first several PEER factors computed from global gene expression matrix were regressed out 

# Results

1. We first determined the number of PEER factors that need to be regressed out

[Details](https://xsun1229.github.io/factor_analysis/num_peer.html)

2. ECT 

  We determined to regress out 2 PEER factors before doing PCA on KEGG pathways. Then, we did ECT for these factor-trait pairs. 

[SNP FDR cutoff: 0.2](https://xsun1229.github.io/factor_analysis/ECT_2peer.html)

[SNP FDR cutoff: 0.1](https://xsun1229.github.io/factor_analysis/ECT_2peer_fdr01.html)

3. ACAT

[Details](https://xsun1229.github.io/factor_analysis/acat_2peer.html)

4. MAGMA

[Details](https://xsun1229.github.io/factor_analysis/magma_2peer.html)

5. Heritability

[Details](https://xsun1229.github.io/factor_analysis/hsq_pc_regressed.html)
