---
title: "Understanding genetics of expression factors"
author: "XSun"
date: "2021-02-19"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

# **1. Introduction**

In this part, we're trying to answer the questions below:

- Are expression factors genetically determined? Can we build prediction model of expression factors using genetic variants?

- What are the SNPs and genes controlling expression factors?

# **2. Data**

- Expression data: GTEx whole blood expression data which contains 670 samples and 19,696 genes. Genes on chrX were excluded.

- Genotype data: GTEx genotype data(the variants with maf >0.05 and maf >0.01 were selected, variants on chrX were excluded). 6,568,202 variants have maf >0.05 and 10,403,249 variants have maf >0.01.

# **3. Analyses**

## **3.1 Factor Construction and evaluation**

- 4 methods were used to infer expression factors from the expression matrix. The [PLIER](https://github.com/wgmao/PLIER), [FLASH]( https://github.com/stephenslab/flashr), and [PEER](https://github.com/PMBio/peer/) worked on GTEx gene expression matrix of whole blood(670 samples, 19,696 genes). PCA worked on KEGG 332 pathways. 

- Heritability of factors was computed using [GCTA](
https://cnsgenomics.com/software/gcta/#GREML) and histograms were made to show the heritability distribution.


**PLIER**
  
  [plier - canonicalPathways - factor consruction ](https://xsun1229.github.io/factor_analysis/factor_plier_canon.html)
  
  [plier - canonicalPathways - factor evaluation ](https://xsun1229.github.io/factor_analysis/eval_plier_canon.html)
  
  [plier - allpathways - factor evaluation ](https://xsun1229.github.io/factor_analysis/factor_plier_allpathway.html)
  
  [plier - allpathways - factor evaluation ](https://xsun1229.github.io/factor_analysis/eval_plier_allpathway.html)
  
   plier - MSigDB results are in Prediction part below (**latest results 07/21**)
  
**FLASH**

  Using 'FLASH', we obtained 100 factors.

  [FLASH - factor evaluation](https://xsun1229.github.io/factor_analysis/eval_flash.html)

**PEER**
  
  Using 'PEER', We obtained 60 factors.
  
  [PEER - factor evaluation](https://xsun1229.github.io/factor_analysis/eval_peer.html)
  
**PC&PEER from KEGG pathways**
  
  1-5 PCs for each KEGG pathway were calculated by PCA. We have 1646 PCs in total.
  
  [For more details](https://xsun1229.github.io/factor_analysis/eval_kegg.html)


## **3.2 Prediction models for factors**

We used two groups of variants to train models. The first group contains all variants have maf > 0.01(10,403,249 variants) and maf > 0.05(6,568,202 variants). The second group contains all significant cis-eQTLs provided by GTEx (FDR > 0.05).

The workflow of the prediction models is described below. The model training and performance evaluation were performed in R (version 4.0.4).
        
The model training follows:
	
1. The 670 GTEx samples were randomly split into training (80%, 536 samples) and testing (20%, 134 samples) set. Association tests follows: expression factors ~ variant+10 genotype PCs were performed on training set to select candidate variants in model training. 10 genotype PCs were computed by PLINK 1.9 on whole genome data. Variants have p-value less than 0.01 were selected.
	
2. The model training was realized by LassoThe outcomes in Lasso models are expression factors while the covariate matrices are the genotype matrix of candidate variants. Ten folds cross validation realized by cv.glmnet() function from glmnet R package were used to train the models. The values of r-squared (rsq on training set), number of variants have non-zero coefficients (num non-zero), at λ =lambda. min (that gives minimum mean cross-validated error lambda in LASSO regression) were recorded.  

The prediction model evaluation follows:

1. The LASSO models at λ =lambda.min were applied to testing sets and thus a series predicted expression factors can be obtained.

2. Linear regression was performed for predicted factors and observed factors following: predicted factor ~ observed factor. The performance of prediction models was evaluated by the p-values of F-statistics from linear regressions (noted as p-values on testing set below) as criterion to evaluate the performance of prediction models. R-squared on testing set was also recorded (rsq on testing set) 

For factors with relatively low pval_test_l, we did GWAS analysis using all GTEx samples(670 samples) following: expression factors ~ variant+10 genotype PCs. 

### 3.2.1 Training models with all variants

There are some missing values in our dataset. We used mean imputation (substituing the missing values with mean value of the corresponding variants) to impute the missing value. 

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis/raw/master/output/ms_snp.jpeg" width="50%">
</figure>

The missing data pattern of our dataset. x-axis represents the proportion of missing value for each variant. y-axis represents the number of the variants.


**PLIER**

  [Results for plier-canonicalPathways factors](https://xsun1229.github.io/factor_analysis/train_plier_canon.html)
    
  [Results for plier-allpathway factors](https://xsun1229.github.io/factor_analysis/train_plier_allpathway.html)
  
  [Results for plier- MSigDB pathway factors](https://xsun1229.github.io/factor_analysis/plier_msigdb.html)

**FLASH**

  [Results for FLASH factors](https://xsun1229.github.io/factor_analysis/train_flash.html)

**PEER**

  [Results for PEER factors](https://xsun1229.github.io/factor_analysis/train_peer.html)


### 3.2.2 Training models with cis-eQTLs

**PLIER**

  [Results for plier-canonicalPathways factors](https://xsun1229.github.io/factor_analysis/train_plier_canon_ciseqtl.html)


### 3.2.3 Trying to make the prediction models better

#### Finding functional SNPs  - LD Score Regression 

The GWAS power of some factors is low. So we tried to find functional SNPs to increase the GWAS power and reduce the testing burden.

We adopted LD score regression to estimate SNP-based heritability and partation the heritability into seperate categories. The software we used here is [ldsc](https://github.com/bulik/ldsc).

The ldsc material and methods of my project can be found [here](https://xsun1229.github.io/factor_analysis/ldsc.html)

**PLIER**

  [Results for plier-canonicalPathways factors](https://xsun1229.github.io/factor_analysis/ldsc_plier_canon.html)
    
  [Results for plier-allpathway factors](https://xsun1229.github.io/factor_analysis/ldsc_plier_all.html)

**FLASH**

  [Results for FLASH factors](https://xsun1229.github.io/factor_analysis/ldsc_flash.html)

**PEER**

  [Results for PEER factors](https://xsun1229.github.io/factor_analysis/ldsc_peer.html)
  

#### Finding functional cis-genes
  
  After preliminary [rough calculations](https://xsun1229.github.io/factor_analysis/gwas_power.html), we found that there might be overfitting issue. So we tried one/two sample settings to test and eliminate the overfitting issue. 
  
  The rought process are below, it differs slightly in different settings.
  
1. Train weights from expression matrix and genotype matrix. Variants are cis-snps (±500kb from the TSS of genes)

2. Impute the cis genetic component part of gene expression (cis-genes):

	 cis-genes = weights * genotype of cis-genes

3. Association test: factors ~ cis-genes + 10 genotype pcs. Record the p-values for each gene.

4. Count the number of significant associations at various p-value threshold for each factor.
  
  [One Sample Setting](https://xsun1229.github.io/factor_analysis/cg-onesample.html)
  
  [Two Sample Setting](https://xsun1229.github.io/factor_analysis/cg-twosample.html)
  
  [One Sample Setting- only cis-eQTLs](https://xsun1229.github.io/factor_analysis/cg-onesample-ciseqtls.html)
  
#### Dealing with weak IV bias

  Apart from using external weights([Two Sample Setting](https://xsun1229.github.io/factor_analysis/cg-twosample.html)), I tried other methods to eliminate the weak IV bias issue (divide the cis-eQTLS into different groups according to the p-values). 
  
  [For more details](https://xsun1229.github.io/factor_analysis/weak_bias.html)